
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Appendix &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.e8f53015daec13862f6db5e763c41738.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Bibliography" href="reference.html" />
    <link rel="prev" title="Glossary" href="glossary.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating
     <code class="docutils literal notranslate">
      <span class="pre">
       snowflake
      </span>
     </code>
     inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. How Ontolopy works
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#simulating-rna-seq-data-to-test-batch-correction-across-experiments">
   Simulating RNA-Seq data to test batch-correction across experiments
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#parameters-for-simulation">
   Parameters for simulation
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#count-parameters">
     Count parameters
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#experimental-design-of-simulated-data">
     Experimental design of simulated data
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#estimating-co-expression-between-genes">
     Estimating co-expression between genes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#distribution-of-fold-changes-for-tissue-specific-genes">
     Distribution of fold-changes for tissue specific genes
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#simulating-tissue-specific-rna-seq-counts">
   Simulating tissue-specific RNA-Seq counts
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#next-steps">
   Next Steps
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="appendix">
<h1>Appendix<a class="headerlink" href="#appendix" title="Permalink to this headline">¶</a></h1>
<!--
## Gene study
[//]: # (TODO: Gene study)
[//]: # (TODO: If I don't put in anything major e.g. Athletes, put it here)
-->
<div class="section" id="simulating-rna-seq-data-to-test-batch-correction-across-experiments">
<span id="simulation-appendix"></span><h2>Simulating RNA-Seq data to test batch-correction across experiments<a class="headerlink" href="#simulating-rna-seq-data-to-test-batch-correction-across-experiments" title="Permalink to this headline">¶</a></h2>
<p>Simulated data can be used to test that methodologies are applicable to new data types.
Since simulated data has a well-defined ground truth, we can test the performance and accuracy of a methodology using it.
As long as the real data is similar to the simulated data, we can assume that methodologies will perform similarly on the real data.</p>
<p>In order to test whether it is feasible to use batch correction to adjust the RNA-Seq experiments chosen (considering the unbalanced design), I want to create a simulated data set of tissue-specific batch-affected gene expression data.
This appendix contains some preliminary work towards this goal, in estimating parameters from the combined data set that will be useful in creating the data set.</p>
</div>
<div class="section" id="parameters-for-simulation">
<h2>Parameters for simulation<a class="headerlink" href="#parameters-for-simulation" title="Permalink to this headline">¶</a></h2>
<p>In order to create simulated data that is similar to the real thing, decisions must be made about how to parameterise the distribution of counts per sample and how these relate to tissue specific effects and batch effects.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">polyester</span></code> R package<span id="id1">[<a class="reference internal" href="reference.html#id87"><span>261</span></a>]</span> can be used to simulate RNA-seq count data with the same design of tissues, samples, and experiments as in the combined data set, particularly the <code class="docutils literal notranslate"><span class="pre">create_read_numbers</span></code> function, which requires a model matrix that specifies the experimental design and a matrix of coefficients <span class="math notranslate nohighlight">\(\beta\)</span> that specify the sample-specific effects.</p>
<div class="section" id="count-parameters">
<h3>Count parameters<a class="headerlink" href="#count-parameters" title="Permalink to this headline">¶</a></h3>
<p>For gene expression count data, a zero-truncated negative binomial distribution is commonly used to represent the underlying gene expression counts
because the distribution is always positive, does not assume mean and variance are equal, and can be tuned to have many zero counts as we see in real data.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">get_params</span></code> function from <code class="docutils literal notranslate"><span class="pre">polyester</span></code> handles the paramerisation of the zero-inflated negative binomial that it uses to simulate count data, using an example data set as input.
I used a cleaned version of the FANTOM5 data as input which was restricted only to genes that are common between all experiments (HDBR, HPA, FANTOM and GTEx), removing all zero rows, and set NaN counts to 0.
Parameters calculated by polyester include means per gene and size, and probability of a zero count per gene.</p>
</div>
<div class="section" id="experimental-design-of-simulated-data">
<h3>Experimental design of simulated data<a class="headerlink" href="#experimental-design-of-simulated-data" title="Permalink to this headline">¶</a></h3>
<p>I already have “model-matrix”, specifying the experimental design of the combined data set, in terms of batch and the 10 more general <a class="reference internal" href="../c08-combining/3-data-wrangling.html#tissue-groups"><span class="std std-ref">tissue groups</span></a> that I mapped samples to using Ontolopy.
These more general tissue groups contain the same specificity (and some of the exact same) terms (e.g. brain) that are in the Human Protein Atlas, which was used to parameterise the simulated count data, which is why this model matrix is a better choice than the more specific 129 Uberon terms that the samples also map to.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span> 
<span class="kn">from</span> <span class="nn">myst_nb</span> <span class="kn">import</span> <span class="n">glue</span>

<span class="n">design_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/design.csv&#39;</span>
<span class="n">design</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">design_file</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;,&#39;</span> <span class="p">,</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                     <span class="n">usecols</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">,</span><span class="s1">&#39;Experiment&#39;</span><span class="p">,</span><span class="s1">&#39;Tissue (UBERON)&#39;</span><span class="p">])</span>

<span class="n">tissue_groups_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../c08-combining/data/tissues_groups.csv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">design</span><span class="p">[</span><span class="s1">&#39;Group name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">design</span><span class="p">[</span><span class="s1">&#39;Tissue (UBERON)&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">tissue_groups_df</span><span class="p">[</span><span class="s1">&#39;Group name&#39;</span><span class="p">])</span>

<span class="n">tissue_groups</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">design</span><span class="p">[</span><span class="s1">&#39;Group name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">dropna</span><span class="p">()</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
<span class="n">mod</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">index</span> <span class="o">=</span> <span class="n">design</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="k">for</span> <span class="n">tissue_group</span> <span class="ow">in</span> <span class="n">tissue_groups</span><span class="p">:</span>
    <span class="c1"># Note: zeros correspond to a normal fold change of 1 (as mod contains logfold changes)</span>
    <span class="n">mod</span><span class="p">[</span><span class="n">tissue_group</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">design</span><span class="p">[</span><span class="s1">&#39;Group name&#39;</span><span class="p">]</span><span class="o">==</span><span class="n">tissue_group</span><span class="p">)</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="s1">&#39;int&#39;</span><span class="p">)</span>

<span class="n">overall_design</span> <span class="o">=</span> <span class="n">mod</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">overall_design</span><span class="p">[</span><span class="s1">&#39;Experiment&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">design</span><span class="p">[</span><span class="s1">&#39;Experiment&#39;</span><span class="p">]</span>
<span class="n">overall_design</span> <span class="o">=</span> <span class="n">overall_design</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="s1">&#39;Experiment&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">overall_design</span><span class="p">[</span><span class="s1">&#39;Samples per experiment&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">overall_design</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">overall_design</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;Total&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">overall_design</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">percentage</span> <span class="o">=</span> <span class="n">overall_design</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">overall_design</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;Total&#39;</span><span class="p">,</span> <span class="s1">&#39;Samples per experiment&#39;</span><span class="p">])</span>
<span class="n">tissue_groups_to_discard</span> <span class="o">=</span> <span class="n">percentage</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">percentage</span><span class="o">&lt;</span><span class="mf">0.01</span><span class="p">]</span>
<span class="n">overall_design</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">tissue_groups_to_discard</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">mod</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">tissue_groups_to_discard</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">tissue_groups</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="n">tissue_groups_to_discard</span><span class="p">)</span>

<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;design-balance&#39;</span><span class="p">,</span> <span class="n">overall_design</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;mod-head&#39;</span><span class="p">,</span> <span class="n">mod</span><span class="o">.</span><span class="n">head</span><span class="p">())</span>

<span class="n">mod</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;../c08-combining/data/simulated/group_mod.csv&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="estimating-co-expression-between-genes">
<span id="estimating-coexpression"></span><h3>Estimating co-expression between genes<a class="headerlink" href="#estimating-co-expression-between-genes" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">polyester</span></code> package does not include gene co-expression (a.k.a. co-occurrance): the correlation between genes of expression values, which is due to genes working together in the same networks, although some other packages do have this functionality.</p>
<p>In order to introduce this correlation to some extent, I used FANTOM data to estimate the correlation between gene expression and used this correlation matrix to create the tissue-specific effects over genes.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="n">fantom_exp</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../c08-combining/data/experiments/fantom_gene_expression.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">random_seed</span> <span class="o">=</span> <span class="mi">14042003</span>
<span class="n">num_genes_to_simulate</span> <span class="o">=</span> <span class="mi">1000</span>

<span class="c1"># Calculate gene covariances to include in model data for RNA-seq simulations</span>
<span class="n">sampled</span> <span class="o">=</span> <span class="n">fantom_exp</span><span class="o">.</span><span class="n">sample</span><span class="p">(</span><span class="n">num_genes_to_simulate</span><span class="p">,</span> <span class="n">random_state</span><span class="o">=</span><span class="n">random_seed</span><span class="p">)</span>
<span class="n">corrcoef_genes</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">corrcoef</span><span class="p">(</span><span class="n">sampled</span><span class="p">)</span>
<span class="k">assert</span><span class="p">(</span><span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">shape</span> <span class="o">==</span> <span class="p">(</span><span class="n">num_genes_to_simulate</span><span class="p">,</span> <span class="n">num_genes_to_simulate</span><span class="p">))</span>
<span class="n">corrcoef_genes</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">corrcoef_genes</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="n">sampled</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="n">sampled</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="c1"># display(corrcoef_genes.head())</span>

<span class="c1"># Not exactly 1 due to the linear alg. Set to 1.</span>
<span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">columns</span><span class="p">:</span>
    <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">gene</span><span class="p">,</span> <span class="n">gene</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>

<span class="c1"># all-NaNs due to all-zero genes in FANTOM. Set to zeros.</span>
<span class="k">for</span> <span class="n">gene</span> <span class="ow">in</span> <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">gene</span><span class="p">][</span><span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">gene</span><span class="p">,</span> <span class="p">:]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">(</span><span class="n">num_genes_to_simulate</span><span class="p">)</span> 
    <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">gene</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">zeros</span><span class="p">(</span><span class="n">num_genes_to_simulate</span><span class="p">)</span> 
<span class="k">assert</span><span class="p">(</span><span class="nb">len</span><span class="p">((</span><span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">gene</span><span class="p">][</span><span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">gene</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]))</span><span class="o">==</span><span class="mi">0</span><span class="p">)</span>  <span class="c1"># No more NaNs</span>

<span class="c1"># Save:</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s1">&#39;../c08-combining/data/simulated/corrcoef_matrix_genes.csv&#39;</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
    <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;# corrcoef matrix for </span><span class="si">{</span><span class="n">num_genes_to_simulate</span><span class="si">}</span><span class="s1"> randomly sampled genes from FANTOM gene expression file `fantom_gene_expression.tsv`, seed = </span><span class="si">{</span><span class="n">random_seed</span><span class="si">}</span><span class="s1">.</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
    <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">f</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
    
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;average-correlation-fantom&#39;</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">values</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">plotly.graph_objects</span> <span class="k">as</span> <span class="nn">go</span>
<span class="kn">import</span> <span class="nn">plotly.figure_factory</span> <span class="k">as</span> <span class="nn">ff</span>
<span class="kn">from</span> <span class="nn">scipy.spatial.distance</span> <span class="kn">import</span> <span class="n">pdist</span><span class="p">,</span> <span class="n">squareform</span>

<span class="n">data_array</span> <span class="o">=</span> <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">values</span>
<span class="n">labels</span> <span class="o">=</span> <span class="n">corrcoef_genes</span><span class="o">.</span><span class="n">columns</span>

<span class="c1"># Code for dendogram and heatmap adapted from plotly examples: https://plotly.com/python/dendrogram/</span>
<span class="c1"># Initialize figure by creating upper dendrogram</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">ff</span><span class="o">.</span><span class="n">create_dendrogram</span><span class="p">(</span><span class="n">data_array</span><span class="p">,</span> <span class="n">orientation</span><span class="o">=</span><span class="s1">&#39;bottom&#39;</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="n">labels</span><span class="p">,</span> <span class="p">)</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">fig</span><span class="p">[</span><span class="s1">&#39;data&#39;</span><span class="p">])):</span>
    <span class="n">fig</span><span class="p">[</span><span class="s1">&#39;data&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">][</span><span class="s1">&#39;yaxis&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;y2&#39;</span>

<span class="c1"># Create Side Dendrogram</span>
<span class="n">dendro_side</span> <span class="o">=</span> <span class="n">ff</span><span class="o">.</span><span class="n">create_dendrogram</span><span class="p">(</span><span class="n">data_array</span><span class="p">,</span> <span class="n">orientation</span><span class="o">=</span><span class="s1">&#39;right&#39;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">dendro_side</span><span class="p">[</span><span class="s1">&#39;data&#39;</span><span class="p">])):</span>
    <span class="n">dendro_side</span><span class="p">[</span><span class="s1">&#39;data&#39;</span><span class="p">][</span><span class="n">i</span><span class="p">][</span><span class="s1">&#39;xaxis&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;x2&#39;</span>

<span class="c1"># Add Side Dendrogram Data to Figure</span>
<span class="k">for</span> <span class="n">data</span> <span class="ow">in</span> <span class="n">dendro_side</span><span class="p">[</span><span class="s1">&#39;data&#39;</span><span class="p">]:</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">data</span><span class="p">,)</span>

<span class="c1"># Create Heatmap</span>
<span class="n">dendro_leaves</span> <span class="o">=</span> <span class="n">dendro_side</span><span class="p">[</span><span class="s1">&#39;layout&#39;</span><span class="p">][</span><span class="s1">&#39;yaxis&#39;</span><span class="p">][</span><span class="s1">&#39;ticktext&#39;</span><span class="p">]</span>
<span class="n">dendro_leaves</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">int</span><span class="p">,</span> <span class="n">dendro_leaves</span><span class="p">))</span>
<span class="n">heat_data</span> <span class="o">=</span> <span class="n">data_array</span><span class="p">[</span><span class="n">dendro_leaves</span><span class="p">,:]</span>
<span class="n">heat_data</span> <span class="o">=</span> <span class="n">heat_data</span><span class="p">[:,</span><span class="n">dendro_leaves</span><span class="p">]</span>

<span class="n">heatmap</span> <span class="o">=</span> <span class="p">[</span>
    <span class="n">go</span><span class="o">.</span><span class="n">Heatmap</span><span class="p">(</span>
        <span class="n">x</span> <span class="o">=</span> <span class="n">dendro_leaves</span><span class="p">,</span>
        <span class="n">y</span> <span class="o">=</span> <span class="n">dendro_leaves</span><span class="p">,</span>
        <span class="n">z</span> <span class="o">=</span> <span class="n">heat_data</span><span class="p">,</span>
        <span class="n">colorscale</span> <span class="o">=</span> <span class="s1">&#39;Blues&#39;</span><span class="p">,</span>
        <span class="n">name</span> <span class="o">=</span> <span class="s1">&#39;Correlation&#39;</span>
    <span class="p">)</span>
<span class="p">]</span>

<span class="n">heatmap</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="s1">&#39;x&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">fig</span><span class="p">[</span><span class="s1">&#39;layout&#39;</span><span class="p">][</span><span class="s1">&#39;xaxis&#39;</span><span class="p">][</span><span class="s1">&#39;tickvals&#39;</span><span class="p">]</span>
<span class="n">heatmap</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="s1">&#39;y&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">dendro_side</span><span class="p">[</span><span class="s1">&#39;layout&#39;</span><span class="p">][</span><span class="s1">&#39;yaxis&#39;</span><span class="p">][</span><span class="s1">&#39;tickvals&#39;</span><span class="p">]</span>

<span class="c1"># Add Heatmap Data to Figure</span>
<span class="k">for</span> <span class="n">data</span> <span class="ow">in</span> <span class="n">heatmap</span><span class="p">:</span>
    <span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>

<span class="c1"># Edit Layout</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">({</span><span class="s1">&#39;width&#39;</span><span class="p">:</span><span class="mi">600</span><span class="p">,</span> <span class="s1">&#39;height&#39;</span><span class="p">:</span><span class="mi">600</span><span class="p">,</span>
                         <span class="s1">&#39;showlegend&#39;</span><span class="p">:</span><span class="kc">False</span><span class="p">,</span> <span class="s1">&#39;hovermode&#39;</span><span class="p">:</span> <span class="s1">&#39;closest&#39;</span><span class="p">,</span>
                         <span class="p">})</span>
<span class="c1"># Edit xaxis</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">xaxis</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;domain&#39;</span><span class="p">:</span> <span class="p">[</span><span class="mf">.15</span><span class="p">,</span> <span class="mi">1</span><span class="p">],</span>
                                  <span class="s1">&#39;mirror&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;showgrid&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;showline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;zeroline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                <span class="s1">&#39;showticklabels&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>

                                  <span class="s1">&#39;ticks&#39;</span><span class="p">:</span><span class="s2">&quot;&quot;</span><span class="p">})</span>
<span class="c1"># Edit xaxis2</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">xaxis2</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;domain&#39;</span><span class="p">:</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mf">.15</span><span class="p">],</span>
                                   <span class="s1">&#39;mirror&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;showgrid&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;showline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;zeroline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;showticklabels&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;ticks&#39;</span><span class="p">:</span><span class="s2">&quot;&quot;</span><span class="p">})</span>

<span class="c1"># Edit yaxis</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">yaxis</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;domain&#39;</span><span class="p">:</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mf">.85</span><span class="p">],</span>
                                  <span class="s1">&#39;mirror&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;showgrid&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;showline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;zeroline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;showticklabels&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                  <span class="s1">&#39;ticks&#39;</span><span class="p">:</span> <span class="s2">&quot;&quot;</span>
                        <span class="p">})</span>
<span class="c1"># Edit yaxis2</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">yaxis2</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;domain&#39;</span><span class="p">:[</span><span class="mf">.825</span><span class="p">,</span> <span class="mf">.975</span><span class="p">],</span>
                                   <span class="s1">&#39;mirror&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;showgrid&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;showline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;zeroline&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;showticklabels&#39;</span><span class="p">:</span> <span class="kc">False</span><span class="p">,</span>
                                   <span class="s1">&#39;ticks&#39;</span><span class="p">:</span><span class="s2">&quot;&quot;</span><span class="p">})</span>

<span class="c1"># Plot!</span>
<span class="n">fig</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="heatmap-correlations">
<a class="reference internal image-reference" href="../_images/blank.png"><img alt="../_images/blank.png" src="../_images/blank.png" style="width: 1px;" /></a>
<p class="caption"><span class="caption-number">Fig. 1 </span><span class="caption-text">Heatmap showing the correlation coefficients between the randomly sampled 1000 genes in the FANTOM5 data set.</span><a class="headerlink" href="#heatmap-correlations" title="Permalink to this image">¶</a></p>
</div>
<p>The correlations between genes, which are used to <a class="reference internal" href="../c08-combining/4-simulation.html#creating-coefficient-matrix"><span class="std std-ref">create the coefficient matrix</span></a> are shown in <a class="reference internal" href="#heatmap-correlations"><span class="std std-numref">Fig. 1</span></a>.</p>
</div>
<div class="section" id="distribution-of-fold-changes-for-tissue-specific-genes">
<h3>Distribution of fold-changes for tissue specific genes<a class="headerlink" href="#distribution-of-fold-changes-for-tissue-specific-genes" title="Permalink to this headline">¶</a></h3>
<p>The expected log2-fold change due to tissue-specific effects per gene and per sample (matrix <span class="math notranslate nohighlight">\(\beta\)</span>) must be pre-decided in order to simulate the data set.
The size of the effect and number genes affected were estimated using data from the Human Protein Atlas (HPA) - available <a class="reference external" href="https://www.proteinatlas.org/download/proteinatlas.tsv.zip">here</a> -  which contains for each tissue-specific gene, the transcripts per million (TPM) for tissues that were found to be tissue-enriched (at least a 5 fold change, compared to all other tissues), group-enriched (at least a 5 fold change between the group of 2-7 tissues compared to all other tissues) or tissue enhanced (at least a 5 fold change between the tissue and the average of all other tissues), and the transcripts per million of the most highly expressed tissues that were not.
Taken together (tissue-enriched, group-enriched and tissue-enhanced), we here refer to these genes/tissues as tissue-specific.
An excerpt of the file can be seen in <a class="reference internal" href="#protein-atlas-view-tbl"><span class="std std-numref">Fig. 2</span></a>.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">powerlaw</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../c08-combining/data/proteinatlas.tsv&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span>
                 <span class="n">usecols</span><span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Gene&#39;</span><span class="p">,</span><span class="s1">&#39;RNA tissue category&#39;</span><span class="p">,</span> <span class="s1">&#39;RNA TS&#39;</span><span class="p">,</span> <span class="s1">&#39;RNA TS TPM&#39;</span><span class="p">,</span> <span class="s1">&#39;TPM max in non-specific&#39;</span><span class="p">])</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;protein-atlas-view&#39;</span><span class="p">,</span> <span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">())</span>
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="protein-atlas-view-tbl">
<p class="caption"><span class="caption-number">Fig. 2 </span><span class="caption-text">The Human protein atlas provides a csv file of TPM values for tissues with &gt;5 fold change. This table was used to parameterise the matrix of coefficients <span class="math notranslate nohighlight">\(\beta\)</span>.</span><a class="headerlink" href="#protein-atlas-view-tbl" title="Permalink to this image">¶</a></p>
</div>
<p>Since the HPA data does not include fold-changes of less than 5, I had no information about these changes, and decided to model the distribution of unaffected genes separately to the affected genes.</p>
<p id="lognormal-estimates"><strong>Estimating parameters of lognormal distribution of log2-fold change per gene:</strong>
For any of these tissue-specific genes/tissues, the log2-fold change per tissue per gene was calculated.
I first checked that each of the tissues had some tissue-specific genes according to the HPA data; this was the case.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># First check that all tissues have some tissue-specific effects according to HPA (5fold(TPM) changes)</span>
<span class="n">non_spec_tissues</span> <span class="o">=</span> <span class="nb">set</span><span class="p">([])</span>
<span class="n">spec_tissues</span> <span class="o">=</span> <span class="nb">set</span><span class="p">([])</span>
<span class="k">for</span> <span class="n">gene</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">for</span> <span class="n">non_spec_tissue</span> <span class="ow">in</span> <span class="p">[</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">str</span><span class="p">(</span><span class="n">row</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;RNA TS TPM&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">,</span><span class="s1">&#39;;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)]:</span> 
        <span class="n">non_spec_tissues</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">non_spec_tissue</span><span class="p">)</span>
    <span class="k">for</span> <span class="n">spec_tissue</span> <span class="ow">in</span> <span class="p">[</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">row</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;TPM max in non-specific&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">)]:</span>
        <span class="n">spec_tissues</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">spec_tissue</span><span class="p">)</span>
        
<span class="c1"># remove strings we don&#39;t care about:</span>
<span class="n">spec_tissues</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="s1">&#39;all non-specific tissues&#39;</span><span class="p">)</span>  <span class="c1"># denotes genes that are only expressed in specific tissues</span>
<span class="n">non_spec_tissues</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="s1">&#39;nan&#39;</span><span class="p">)</span>  <span class="c1"># denotes genes that are not tissue-specific</span>

<span class="k">assert</span><span class="p">(</span><span class="n">non_spec_tissues</span><span class="o">==</span><span class="n">spec_tissues</span><span class="p">)</span> <span class="c1"># =&gt; yes all tissues are tissue-specific of these 37</span>
</pre></div>
</div>
</div>
</div>
<p>I then extracted the multipliers from the data, and converted them to log2-fold format (expected by <code class="docutils literal notranslate"><span class="pre">polyester</span></code>).
Since the distribution was long-tailed, I compared the distribution to an exponential, log-normal and power-law distribution using the python <code class="docutils literal notranslate"><span class="pre">powerlaw</span></code> package<span id="id2">[<a class="reference internal" href="reference.html#id270"><span>262</span></a>]</span>.
Comparative tests showed that lognormal was the best fit (with extremely low p-values, see code output below);  <code class="xref std std-numref docutils literal notranslate"><span class="pre">powerlaw-multipliers</span></code> visualises this.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="c1">#Extract the data of the TPM multipliers</span>
<span class="n">multipliers</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">gene</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">row</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;RNA TS TPM&#39;</span><span class="p">]):</span>
        <span class="k">continue</span>
    <span class="n">biggest_ts</span> <span class="o">=</span> <span class="nb">max</span><span class="p">([</span><span class="nb">float</span><span class="p">(</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">strip</span><span class="p">())</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">row</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;RNA TS TPM&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)])</span>
    <span class="n">biggest_non_ts</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">row</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;TPM max in non-specific&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">strip</span><span class="p">())</span>
    <span class="k">if</span> <span class="n">biggest_non_ts</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="k">continue</span> 
    <span class="k">if</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">biggest_ts</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">biggest_non_ts</span><span class="p">))</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
        <span class="k">continue</span>
    <span class="n">multipliers</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">biggest_ts</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">biggest_non_ts</span><span class="p">)))</span>
    
<span class="c1">#Fit distribitions to the TPM multipliers</span>
<span class="n">fit</span> <span class="o">=</span> <span class="n">powerlaw</span><span class="o">.</span><span class="n">Fit</span><span class="p">(</span><span class="n">multipliers</span><span class="p">,</span> <span class="n">xmin</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">discrete</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">compare1</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;lognormal&#39;</span><span class="p">,</span> <span class="s1">&#39;power_law&#39;</span><span class="p">]</span>
<span class="n">R</span><span class="p">,</span> <span class="n">p</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">distribution_compare</span><span class="p">(</span><span class="o">*</span><span class="n">compare1</span><span class="p">)</span>
<span class="k">if</span>  <span class="n">p</span> <span class="o">&lt;</span> <span class="mf">0.05</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution is a better fit than </span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution with p-value=</span><span class="si">{</span><span class="n">p</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

<span class="n">compare2</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;lognormal&#39;</span><span class="p">,</span> <span class="s1">&#39;exponential&#39;</span><span class="p">]</span>
<span class="n">R</span><span class="p">,</span> <span class="n">p</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">distribution_compare</span><span class="p">(</span><span class="o">*</span><span class="n">compare2</span><span class="p">)</span>
<span class="k">if</span> <span class="n">p</span> <span class="o">&lt;</span> <span class="mf">0.05</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution is a better fit than </span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution with p-value=</span><span class="si">{</span><span class="n">p</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
    
<span class="c1"># Parameters for log2-fold coefficients of tissue-specific expression over genes:</span>
<span class="n">mu_ts</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">lognormal</span><span class="o">.</span><span class="n">mu</span>
<span class="n">sigma_ts</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">lognormal</span><span class="o">.</span><span class="n">sigma</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;mu-lognormal-ts&#39;</span><span class="p">,</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">mu_ts</span><span class="si">:</span><span class="s1">.2f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;sigma-lognormal-ts&#39;</span><span class="p">,</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">sigma_ts</span><span class="si">:</span><span class="s1">.2f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bin_edges</span><span class="p">,</span> <span class="n">probs</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">20</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">)))</span>
<span class="n">bin_centres</span> <span class="o">=</span> <span class="mf">0.5</span> <span class="o">*</span> <span class="p">(</span><span class="n">bin_edges</span><span class="p">[:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">+</span> <span class="n">bin_edges</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span>
<span class="n">xlim</span> <span class="o">=</span> <span class="nb">max</span><span class="p">(</span><span class="n">bin_edges</span><span class="p">)</span>
<span class="n">x</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="n">xlim</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">))</span>

<span class="n">lognormal</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">lognormal</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
<span class="n">exponential</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">exponential</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
<span class="n">power_law</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">power_law</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>


<span class="n">fig</span> <span class="o">=</span> <span class="n">go</span><span class="o">.</span><span class="n">Figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">bin_centres</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">probs</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;HPA data&#39;</span><span class="p">,</span>  <span class="n">fill</span><span class="o">=</span><span class="s1">&#39;tozeroy&#39;</span><span class="p">))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">lognormal</span><span class="p">,</span>
                    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;lines&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;lognormal&#39;</span><span class="p">,</span> <span class="n">line</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;darkorange&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                              <span class="n">dash</span><span class="o">=</span><span class="s1">&#39;dot&#39;</span><span class="p">)))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">exponential</span><span class="p">,</span>
                    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;lines&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;exponential&#39;</span><span class="p">,</span> <span class="n">line</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;darkmagenta&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                              <span class="n">dash</span><span class="o">=</span><span class="s1">&#39;dot&#39;</span><span class="p">)))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">power_law</span><span class="p">,</span>
                    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;lines&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;power law&#39;</span><span class="p">,</span> <span class="n">line</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;green&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                              <span class="n">dash</span><span class="o">=</span><span class="s1">&#39;dot&#39;</span><span class="p">)))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">xaxis_title</span><span class="o">=</span><span class="s1">&#39;Log&lt;sub&gt;2&lt;/sub&gt;-fold multipliers tissue-specific gene.&#39;</span><span class="p">,</span>
                 <span class="n">yaxis_title</span><span class="o">=</span><span class="s1">&#39;Probability density&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">400</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_xaxes</span><span class="p">(</span><span class="nb">range</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="nb">max</span><span class="p">(</span><span class="n">bin_centres</span><span class="p">)])</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_yaxes</span><span class="p">(</span><span class="nb">range</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mf">0.008</span><span class="p">])</span>

<span class="n">fig</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="powerlaw-multipliers-fit">
<a class="reference internal image-reference" href="../_images/blank.png"><img alt="../_images/blank.png" src="../_images/blank.png" style="width: 1px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3 </span><span class="caption-text">The distribution of tissue-specific fold-change over all tissue-specific gene-sample pairs from HPA, fitted to lognormal, powerlaw, and exponential distributions, showing the lognormal as the best fit.</span><a class="headerlink" href="#powerlaw-multipliers-fit" title="Permalink to this image">¶</a></p>
</div>
<p>The log-normal distribution was the best fit to the data, see <a class="reference internal" href="#powerlaw-multipliers-fit"><span class="std std-numref">Fig. 3</span></a>).
The parameters fitting the log2-fold changes to the log-normal distribution were estimated as <span class="math notranslate nohighlight">\(\mu=\)</span> <span class="math notranslate nohighlight">\(\sigma=\)</span>.
Visually inspection of <code class="xref std std-numref docutils literal notranslate"><span class="pre">multipliers-fit</span></code> reveals that the data simulated from these parameters appears to fit the data reasonably well, although it may be better parameterised by two overlapping distributions.</p>
<p><strong>Number of tissue-specific genes per tissue:</strong>
The number of tissue-specific genes per tissue was also calculated from the HPA data.
Again, the data was most similar to a lognormal, still with very small p-values (see code output below), but the fit (see <a class="reference internal" href="#genes-per-tissue"><span class="std std-numref">Fig. 4</span></a>) was less convincing, probably due to the small number of tissues: 37.
The distribution was parameterised with <span class="math notranslate nohighlight">\(\mu=\)</span> <span class="math notranslate nohighlight">\(\sigma=\)</span>.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">#Extract the data of genes per sample</span>
<span class="n">df</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;RNA TS TPM&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
<span class="n">count_dict</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">gene</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;RNA TS TPM&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="n">tissue_names</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">row</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;RNA TS TPM&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;;&#39;</span><span class="p">)]</span>
    <span class="k">for</span> <span class="n">tissue</span> <span class="ow">in</span> <span class="n">tissue_names</span><span class="p">:</span>
        <span class="k">try</span><span class="p">:</span>
            <span class="n">count_dict</span><span class="p">[</span><span class="n">tissue</span><span class="p">]</span><span class="o">+=</span><span class="mi">1</span>
        <span class="k">except</span><span class="p">:</span>
            <span class="n">count_dict</span><span class="p">[</span><span class="n">tissue</span><span class="p">]</span><span class="o">=</span><span class="mi">1</span>

<span class="n">genes_per_tissue</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">count_dict</span><span class="o">.</span><span class="n">values</span><span class="p">()]</span>

<span class="c1"># Fit powerlaw</span>
<span class="n">min_fold_change</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="n">fit</span> <span class="o">=</span> <span class="n">powerlaw</span><span class="o">.</span><span class="n">Fit</span><span class="p">(</span><span class="n">genes_per_tissue</span><span class="p">,</span> <span class="n">xmin</span><span class="o">=</span><span class="n">min_fold_change</span><span class="p">,</span> <span class="n">discrete</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">R</span><span class="p">,</span> <span class="n">p</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">distribution_compare</span><span class="p">(</span><span class="o">*</span><span class="n">compare1</span><span class="p">)</span>
<span class="k">if</span>  <span class="n">p</span> <span class="o">&lt;</span> <span class="mf">0.05</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution is a better fit than </span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution with p-value=</span><span class="si">{</span><span class="n">p</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

<span class="n">fit</span><span class="o">.</span><span class="n">distribution_compare</span><span class="p">(</span><span class="o">*</span><span class="n">compare2</span><span class="p">)</span>
<span class="k">if</span> <span class="n">p</span> <span class="o">&lt;</span> <span class="mf">0.05</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution is a better fit than </span><span class="si">{</span><span class="n">compare1</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s2"> distribution with p-value=</span><span class="si">{</span><span class="n">p</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

<span class="c1"># Parameters for number of genes affected by tissues:</span>
<span class="n">mu_ngenes</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">lognormal</span><span class="o">.</span><span class="n">mu</span>
<span class="n">sigma_ngenes</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">lognormal</span><span class="o">.</span><span class="n">sigma</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;mu-lognormal-num-ts-genes&#39;</span><span class="p">,</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">mu_ngenes</span><span class="si">:</span><span class="s1">.2f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;sigma-lognormal-num-ts-genes&#39;</span><span class="p">,</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">sigma_ngenes</span><span class="si">:</span><span class="s1">.2f</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bin_edges</span><span class="p">,</span> <span class="n">probs</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">20</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">)))</span>
<span class="n">bin_centres</span> <span class="o">=</span> <span class="mf">0.5</span> <span class="o">*</span> <span class="p">(</span><span class="n">bin_edges</span><span class="p">[:</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span> <span class="o">+</span> <span class="n">bin_edges</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span>
<span class="n">xlim</span> <span class="o">=</span> <span class="nb">max</span><span class="p">(</span><span class="n">bin_edges</span><span class="p">)</span>
<span class="n">x</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="n">xlim</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">))</span>

<span class="n">lognormal</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">lognormal</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
<span class="n">exponential</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">exponential</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
<span class="n">power_law</span> <span class="o">=</span> <span class="n">fit</span><span class="o">.</span><span class="n">power_law</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>


<span class="n">fig</span> <span class="o">=</span> <span class="n">go</span><span class="o">.</span><span class="n">Figure</span><span class="p">()</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">bin_centres</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">probs</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;HPA data&#39;</span><span class="p">,</span>  <span class="n">fill</span><span class="o">=</span><span class="s1">&#39;tozeroy&#39;</span><span class="p">))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">lognormal</span><span class="p">,</span>
                    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;lines&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;lognormal&#39;</span><span class="p">,</span> <span class="n">line</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;darkorange&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                              <span class="n">dash</span><span class="o">=</span><span class="s1">&#39;dot&#39;</span><span class="p">)))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">exponential</span><span class="p">,</span>
                    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;lines&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;exponential&#39;</span><span class="p">,</span> <span class="n">line</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;darkmagenta&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                              <span class="n">dash</span><span class="o">=</span><span class="s1">&#39;dot&#39;</span><span class="p">)))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span><span class="n">go</span><span class="o">.</span><span class="n">Scatter</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">power_law</span><span class="p">,</span>
                    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;lines&#39;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s1">&#39;power law&#39;</span><span class="p">,</span> <span class="n">line</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s1">&#39;green&#39;</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                              <span class="n">dash</span><span class="o">=</span><span class="s1">&#39;dot&#39;</span><span class="p">)))</span>

<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">xaxis_title</span><span class="o">=</span><span class="s1">&#39;Numbers of tissue-specific genes per tissue&#39;</span><span class="p">,</span>
                 <span class="n">yaxis_title</span><span class="o">=</span><span class="s1">&#39;Probability density&#39;</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">400</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_xaxes</span><span class="p">(</span><span class="nb">range</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="nb">max</span><span class="p">(</span><span class="n">bin_centres</span><span class="p">)])</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_yaxes</span><span class="p">(</span><span class="nb">range</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mf">0.006</span><span class="p">])</span>

<span class="n">fig</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="genes-per-tissue">
<a class="reference internal image-reference" href="../_images/blank.png"><img alt="../_images/blank.png" src="../_images/blank.png" style="width: 1px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4 </span><span class="caption-text">The distribution of the number of tissue-specific genes per tissue from HPA, fitted to lognormal, powerlaw, and exponential distributions, showing the lognormal as the best fit.</span><a class="headerlink" href="#genes-per-tissue" title="Permalink to this image">¶</a></p>
</div>
</div>
</div>
<div class="section" id="simulating-tissue-specific-rna-seq-counts">
<h2>Simulating tissue-specific RNA-Seq counts<a class="headerlink" href="#simulating-tissue-specific-rna-seq-counts" title="Permalink to this headline">¶</a></h2>
<p>Counts can then be simulated using <code class="docutils literal notranslate"><span class="pre">polyester</span></code> (using <code class="xref download docutils literal notranslate"><span class="pre">this</span> <span class="pre">script</span></code>) or an alternative tool.</p>
<p>The simulated data set is given by: <span class="math notranslate nohighlight">\(C_{ijk}\propto Negative Binomal (mean=\mu_{jk},size=r_{jk})\)</span> for replicate <span class="math notranslate nohighlight">\(i\)</span>, gene <span class="math notranslate nohighlight">\(j\)</span>, and sample <span class="math notranslate nohighlight">\(k\)</span>, where:</p>
<ul class="simple">
<li><p>the means are given by <span class="math notranslate nohighlight">\(\mu_{jk}=\mu'_j+\beta_{jk} \cdot mod\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(\mu'_j\)</span> are the estimated base means per gene</p></li>
<li><p><span class="math notranslate nohighlight">\(\beta_{jk}\)</span> are the generated matrix of log-fold changes in matrix format, including both batch and tissue effects (<code class="docutils literal notranslate"><span class="pre">coeffs_batch.csv</span></code>)</p></li>
<li><p><span class="math notranslate nohighlight">\(mod\)</span> is the model design matrix.</p></li>
<li><p>the dispersion parameter (size), <span class="math notranslate nohighlight">\(r_{jk}\)</span> is calculated based on <span class="math notranslate nohighlight">\(\mu_{jk}\)</span> and the fit between mean and size (estimated from the FANTOM5 data).</p></li>
</ul>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>Next steps will be to perform batch-correction on these simulated data sets, e.g. ComBat, ComBat-Seq, and Mutual Nearest Neighbours, and performing differential expression analyses, using the input <span class="math notranslate nohighlight">\(\beta\)</span> matrix to test for ground truths.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./cz-end-matter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="glossary.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title">Glossary</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="reference.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title">Bibliography</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>