
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Bibliography &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="Appendix" href="0-appendix.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Bibliography</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="section" id="bibliography">
<span id="id1"></span><h1>Bibliography<a class="headerlink" href="#bibliography" title="Permalink to this headline">¶</a></h1>
<p>References are shown at the bottom of the page on which they are cited. A full bibliography (of the entire Jupyter Book) appears below:</p>
<hr class="docutils" />
<p id="id2"><dl class="citation">
<dt class="label" id="id277"><span class="brackets">1</span></dt>
<dd><p>Executable Books Community. Jupyter book. February 2020. URL: <a class="reference external" href="https://zenodo.org/record/4539666">https://zenodo.org/record/4539666</a>.</p>
</dd>
<dt class="label" id="id4"><span class="brackets">2</span></dt>
<dd><p>Jan Zaucha, Jonathan Stahlhacke, Matt E Oates, Natalie Thurlby, Owen J L Rackham, Hai Fang, Ben Smithers, and Julian Gough. A proteome quality index. <em>Environ. Microbiol.</em>, 17(1):4–9, 2015. URL: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.12622">https://onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.12622</a>.</p>
</dd>
<dt class="label" id="id3"><span class="brackets">3</span></dt>
<dd><p>Matt E Oates, Jonathan Stahlhacke, Dimitrios V Vavoulis, Ben Smithers, Owen J L Rackham, Adam J Sardar, Jan Zaucha, Natalie Thurlby, Hai Fang, and Julian Gough. The SUPERFAMILY 1.75 database in 2014: a doubling of data. <em>Nucleic Acids Res.</em>, 43(Database issue):D227–33, January 2015.</p>
</dd>
<dt class="label" id="id5"><span class="brackets">4</span></dt>
<dd><p>Julian Gough, Jan Zaucha, and Natalie Thurlby. Determining phenotype from genotype. July 2017. URL: <a class="reference external" href="https://patents.google.com/patent/US20200176085A1/en">https://patents.google.com/patent/US20200176085A1/en</a>.</p>
</dd>
<dt class="label" id="id6"><span class="brackets">5</span></dt>
<dd><p>Naihui Zhou, Yuxiang Jiang, Timothy R Bergquist, Alexandra J Lee, Balint Z Kacsoh, Alex W Crocker, Kimberley A Lewis, George Georghiou, Huy N Nguyen, Md Nafiz Hamid, Larry Davis, Tunca Dogan, Volkan Atalay, Ahmet S Rifaioglu, Alperen Dalkiran, Rengul Cetin-Atalay, Chengxin Zhang, Rebecca L Hurto, Peter L Freddolino, Yang Zhang, Prajwal Bhat, Fran Supek, José M Fernández, Branislava Gemovic, Vladimir R Perovic, Radoslav S Davidović, Neven Sumonja, Nevena Veljkovic, Ehsaneddin Asgari, Mohammad R K Mofrad, Giuseppe Profiti, Castrense Savojardo, Pier Luigi Martelli, Rita Casadio, Florian Boecker, Indika Kahanda, Natalie Thurlby, Alice C McHardy, Alexandre Renaux, Rabie Saidi, Julian Gough, Alex A Freitas, Magdalena Antczak, Fabio Fabris, Mark N Wass, Jie Hou, Jianlin Cheng, Zheng Wang, Alfonso E Romero, Alberto Paccanaro, Haixuan Yang, Tatyana Goldberg, Chenguang Zhao, Liisa Holm, Petri Törönen, Alan J Medlar, Elaine Zosa, Itamar Borukhov, Ilya Novikov, Angela Wilkins, Olivier Lichtarge, Po-Han Chi, Wei-Cheng Tseng, Michal Linial, Peter W Rose, Christophe Dessimoz, Vedrana Vidulin, Saso Dzeroski, Ian Sillitoe, Sayoni Das, Jonathan Gill Lees, David T Jones, Cen Wan, Domenico Cozzetto, Rui Fa, Mateo Torres, Alex Wiarwick Vesztrocy, Jose Manuel Rodriguez, Michael L Tress, Marco Frasca, Marco Notaro, Giuliano Grossi, Alessandro Petrini, Matteo Re, Giorgio Valentini, Marco Mesiti, Daniel B Roche, Jonas Reeb, David W Ritchie, Sabeur Aridhi, Seyed Ziaeddin Alborzi, Marie-Dominique Devignes, Koo Da Chen Emily, Richard Bonneau, Vladimir Gligorijević, Meet Barot, Hai Fang, Stefano Toppo, Enrico Lavezzo, Marco Falda, Michele Berselli, Silvio C E Tosatto, Marco Carraro, Damiano Piovesan, Hafeez Ur Rehman, Qizhong Mao, Shanshan Zhang, Slobodan Vucetic, Gage S Black, Dane Jo, Dallas J Larsen, Ashton R Omdahl, Luke W Sagers, Erica Suh, Jonathan B Dayton, Liam J McGuffin, Danielle A Brackenridge, Patricia C Babbitt, Jeffrey M Yunes, Paolo Fontana, Feng Zhang, Shanfeng Zhu, Ronghui You, Zihan Zhang, Suyang Dai, Shuwei Yao, Weidong Tian, Renzhi Cao, Caleb Chandler, Miguel Amezola, Devon Johnson, Jia-Ming Chang, Wen-Hung Liao, Yi-Wei Liu, Stefano Pascarelli, Yotam Frank, Robert Hoehndorf, Maxat Kulmanov, Imane Boudellioua, Gianfranco Politano, Stefano Di Carlo, Alfredo Benso, Kai Hakala, Filip Ginter, Farrokh Mehryary, Suwisa Kaewphan, Jari Björne, Hans Moen, Martti E E Tolvanen, Tapio Salakoski, Daisuke Kihara, Aashish Jain, Tomislav Šmuc, Adrian Altenhoff, Asa Ben-Hur, Burkhard Rost, Steven E Brenner, Christine A Orengo, Constance J Jeffery, Giovanni Bosco, Deborah A Hogan, Maria J Martin, Claire O'Donovan, Sean D Mooney, Casey S Greene, Predrag Radivojac, and Iddo Friedberg. The CAFA challenge reports improved protein function prediction and new functional annotations for hundreds of genes through experimental screens. <em>bioRxiv</em>, pages 653105, May 2019.</p>
</dd>
<dt class="label" id="id194"><span class="brackets">6</span></dt>
<dd><p>Welcome to the krita 4.4 manual! — krita manual 4.4.0 documentation. <span><a class="reference external" href="#"></a></span>https://docs.krita.org/en/index.html. Accessed: 2021-2-22. URL: <a class="reference external" href="https://docs.krita.org/en/index.html">https://docs.krita.org/en/index.html</a>.</p>
</dd>
<dt class="label" id="id228"><span class="brackets">7</span></dt>
<dd><p>Eric Bonabeau. Agent-based modeling: methods and techniques for simulating human systems. <em>Proc. Natl. Acad. Sci. U. S. A.</em>, 99 Suppl 3:7280–7287, May 2002. URL: <a class="reference external" href="http://dx.doi.org/10.1073/pnas.082080899">http://dx.doi.org/10.1073/pnas.082080899</a>.</p>
</dd>
<dt class="label" id="id229"><span class="brackets">8</span></dt>
<dd><p>A M Turing. The chemical basis of morphogenesis. 1953. <em>Bull. Math. Biol.</em>, 52(1-2):153–97; discussion 119–52, 1990. URL: <a class="reference external" href="http://dx.doi.org/10.1007/BF02459572">http://dx.doi.org/10.1007/BF02459572</a>.</p>
</dd>
<dt class="label" id="id230"><span class="brackets">9</span></dt>
<dd><p>Kristie Whitaker and Olivia Guest. # bropenscience is broken science: kirstie whitaker and olivia guest ask how open `open science'really is. <em>Psychologist</em>, 33:34–37, 2020. URL: <a class="reference external" href="https://pure.mpg.de/rest/items/item_3286863/component/file_3286864/content">https://pure.mpg.de/rest/items/item_3286863/component/file_3286864/content</a>.</p>
</dd>
<dt class="label" id="id201"><span class="brackets">10</span></dt>
<dd><p>Steven Rose. Darwin, race and gender. <em>EMBO Rep.</em>, 10(4):297–298, April 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1038/embor.2009.40">http://dx.doi.org/10.1038/embor.2009.40</a>.</p>
</dd>
<dt class="label" id="id7"><span class="brackets">11</span></dt>
<dd><p>Conway Zirkle. The inheritance of acquired characters and the provisional hypothesis of pangenesis. <em>Am. Nat.</em>, 69(724):417–445, September 1935.</p>
</dd>
<dt class="label" id="id8"><span class="brackets">12</span></dt>
<dd><p>Anthony Grafton and Nancy G Siraisi. Natural particulars: nature and the disciplines in renaissance europe. In <em>acls humanities e-book</em>. MPublishing, University of Michigan Library, 1999.</p>
</dd>
<dt class="label" id="id202"><span class="brackets">13</span></dt>
<dd><p>Charles Darwin. <em>On the Origin of Species by Means of Natural Selection Or the Preservation of Favoured Races in the Struggle for Life</em>. International Book Company, 1913. URL: <a class="reference external" href="http://lnmcp.mf.uni-lj.si/ISN/Darwin_C.doc">http://lnmcp.mf.uni-lj.si/ISN/Darwin_C.doc</a>.</p>
</dd>
<dt class="label" id="id200"><span class="brackets">14</span></dt>
<dd><p>Daniel J Fairbanks. Mendel and darwin: untangling a persistent enigma. <em>Heredity</em>, 124(2):263–273, February 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41437-019-0289-9">http://dx.doi.org/10.1038/s41437-019-0289-9</a>.</p>
</dd>
<dt class="label" id="id147"><span class="brackets">15</span></dt>
<dd><p>Gregor Mendel. Experiments in plant hybridization (1865). <em>Verhandlungen des naturforschenden Vereins Brünn) Available online</em>, 1996. URL: <a class="reference external" href="http://old.esp.org/foundations/genetics/classical/gm-65-a.pdf">http://old.esp.org/foundations/genetics/classical/gm-65-a.pdf</a>.</p>
</dd>
<dt class="label" id="id146"><span class="brackets">16</span></dt>
<dd><p>W Johannsen. The genotype conception of heredity. <em>Am. Nat.</em>, 45(531):129–159, March 1911. URL: <a class="reference external" href="https://doi.org/10.1086/279202">https://doi.org/10.1086/279202</a>.</p>
</dd>
<dt class="label" id="id151"><span class="brackets">17</span></dt>
<dd><p>Richard J Evans. RA fisher and the science of hatred. <em>New Stateman</em>, June 2020. URL: <a class="reference external" href="https://www.newstatesman.com/international/science-tech/2020/07/ra-fisher-and-science-hatred">https://www.newstatesman.com/international/science-tech/2020/07/ra-fisher-and-science-hatred</a>.</p>
</dd>
<dt class="label" id="id148"><span class="brackets">18</span></dt>
<dd><p>J D Watson and F H Crick. Molecular structure of nucleic acids; a structure for deoxyribose nucleic acid. <em>Nature</em>, 171(4356):737–738, April 1953. URL: <a class="reference external" href="http://dx.doi.org/10.1038/171737a0">http://dx.doi.org/10.1038/171737a0</a>.</p>
</dd>
<dt class="label" id="id16"><span class="brackets">19</span></dt>
<dd><p>J C Venter, M D Adams, E W Myers, P W Li, R J Mural, G G Sutton, H O Smith, M Yandell, C A Evans, R A Holt, J D Gocayne, P Amanatides, R M Ballew, D H Huson, J R Wortman, Q Zhang, C D Kodira, X H Zheng, L Chen, M Skupski, G Subramanian, P D Thomas, J Zhang, G L Gabor Miklos, C Nelson, S Broder, A G Clark, J Nadeau, V A McKusick, N Zinder, A J Levine, R J Roberts, M Simon, C Slayman, M Hunkapiller, R Bolanos, A Delcher, I Dew, D Fasulo, M Flanigan, L Florea, A Halpern, S Hannenhalli, S Kravitz, S Levy, C Mobarry, K Reinert, K Remington, J Abu-Threideh, E Beasley, K Biddick, V Bonazzi, R Brandon, M Cargill, I Chandramouliswaran, R Charlab, K Chaturvedi, Z Deng, V Di Francesco, P Dunn, K Eilbeck, C Evangelista, A E Gabrielian, W Gan, W Ge, F Gong, Z Gu, P Guan, T J Heiman, M E Higgins, R R Ji, Z Ke, K A Ketchum, Z Lai, Y Lei, Z Li, J Li, Y Liang, X Lin, F Lu, G V Merkulov, N Milshina, H M Moore, A K Naik, V A Narayan, B Neelam, D Nusskern, D B Rusch, S Salzberg, W Shao, B Shue, J Sun, Z Wang, A Wang, X Wang, J Wang, M Wei, R Wides, C Xiao, C Yan, A Yao, J Ye, M Zhan, W Zhang, H Zhang, Q Zhao, L Zheng, F Zhong, W Zhong, S Zhu, S Zhao, D Gilbert, S Baumhueter, G Spier, C Carter, A Cravchik, T Woodage, F Ali, H An, A Awe, D Baldwin, H Baden, M Barnstead, I Barrow, K Beeson, D Busam, A Carver, A Center, M L Cheng, L Curry, S Danaher, L Davenport, R Desilets, S Dietz, K Dodson, L Doup, S Ferriera, N Garg, A Gluecksmann, B Hart, J Haynes, C Haynes, C Heiner, S Hladun, D Hostin, J Houck, T Howland, C Ibegwam, J Johnson, F Kalush, L Kline, S Koduru, A Love, F Mann, D May, S McCawley, T McIntosh, I McMullen, M Moy, L Moy, B Murphy, K Nelson, C Pfannkoch, E Pratts, V Puri, H Qureshi, M Reardon, R Rodriguez, Y H Rogers, D Romblad, B Ruhfel, R Scott, C Sitter, M Smallwood, E Stewart, R Strong, E Suh, R Thomas, N N Tint, S Tse, C Vech, G Wang, J Wetter, S Williams, M Williams, S Windsor, E Winn-Deen, K Wolfe, J Zaveri, K Zaveri, J F Abril, R Guigó, M J Campbell, K V Sjolander, B Karlak, A Kejariwal, H Mi, B Lazareva, T Hatton, A Narechania, K Diemer, A Muruganujan, N Guo, S Sato, V Bafna, S Istrail, R Lippert, R Schwartz, B Walenz, S Yooseph, D Allen, A Basu, J Baxendale, L Blick, M Caminha, J Carnes-Stine, P Caulk, Y H Chiang, M Coyne, C Dahlke, A Mays, M Dombroski, M Donnelly, D Ely, S Esparham, C Fosler, H Gire, S Glanowski, K Glasser, A Glodek, M Gorokhov, K Graham, B Gropman, M Harris, J Heil, S Henderson, J Hoover, D Jennings, C Jordan, J Jordan, J Kasha, L Kagan, C Kraft, A Levitsky, M Lewis, X Liu, J Lopez, D Ma, W Majoros, J McDaniel, S Murphy, M Newman, T Nguyen, N Nguyen, M Nodell, S Pan, J Peck, M Peterson, W Rowe, R Sanders, J Scott, M Simpson, T Smith, A Sprague, T Stockwell, R Turner, E Venter, M Wang, M Wen, D Wu, M Wu, A Xia, A Zandieh, and X Zhu. The sequence of the human genome. <em>Science</em>, 291(5507):1304–1351, February 2001.</p>
</dd>
<dt class="label" id="id149"><span class="brackets">20</span></dt>
<dd><p>James Meek. Decoding DNA. <em>The Guardian</em>, June 2000. URL: <a class="reference external" href="http://www.theguardian.com/science/2000/jun/27/genetics.uknews">http://www.theguardian.com/science/2000/jun/27/genetics.uknews</a>.</p>
</dd>
<dt class="label" id="id150"><span class="brackets">21</span></dt>
<dd><p>Amy Harmon. James watson had a chance to salvage his reputation on race. he made things worse. <em>The New York Times</em>, January 2019. URL: <a class="reference external" href="https://www.nytimes.com/2019/01/01/science/watson-dna-genetics-race.html">https://www.nytimes.com/2019/01/01/science/watson-dna-genetics-race.html</a>.</p>
</dd>
<dt class="label" id="id145"><span class="brackets">22</span></dt>
<dd><p>Timothé Cynober. Why are there only 11 cell and gene therapies in europe? <em>Labiotech</em>, September 2020. URL: <a class="reference external" href="https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/">https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/</a>.</p>
</dd>
<dt class="label" id="id144"><span class="brackets">23</span></dt>
<dd><p>Guardian staff reporter. Cancer patients in england to be offered chance to avoid toxic side-effects. <em>The Guardian</em>, December 2020. URL: <a class="reference external" href="http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects">http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects</a>.</p>
</dd>
<dt class="label" id="id278"><span class="brackets">24</span></dt>
<dd><p>Michael Le Page. We don't know what a fifth of our genes do – and won't find out soon. <em>New Scientist</em>, February 2019. URL: <a class="reference external" href="https://www.newscientist.com/article/2194516-we-dont-know-what-a-fifth-of-our-genes-do-and-wont-find-out-soon/">https://www.newscientist.com/article/2194516-we-dont-know-what-a-fifth-of-our-genes-do-and-wont-find-out-soon/</a>.</p>
</dd>
<dt class="label" id="id155"><span class="brackets">25</span></dt>
<dd><p>Elizabeth Pennisi. Genomics. DNA study forces rethink of what it means to be a gene. <em>Science</em>, 316(5831):1556–1557, June 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.316.5831.1556">http://dx.doi.org/10.1126/science.316.5831.1556</a>.</p>
</dd>
<dt class="label" id="id9"><span class="brackets">26</span></dt>
<dd><p>Gallery 19: DNA model, 1953 :: DNA learning center. <span><a class="reference external" href="#"></a></span>https://www.dnalc.org/view/16430-Gallery-19-DNA-model-1953.html. Accessed: 2019-6-2.</p>
</dd>
<dt class="label" id="id12"><span class="brackets">27</span></dt>
<dd><p>G J Mulder. Ueber die zusammensetzung einiger thierischen substanzen. <em>J. Prakt. Chem.</em>, 1839.</p>
</dd>
<dt class="label" id="id179"><span class="brackets">28</span></dt>
<dd><p>V N Uversky. Posttranslational modification. In Stanley Maloy and Kelly Hughes, editors, <em>Brenner's Encyclopedia of Genetics (Second Edition)</em>, pages 425–430. Academic Press, San Diego, January 2013. URL: <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/B9780123749840012031">https://www.sciencedirect.com/science/article/pii/B9780123749840012031</a>.</p>
</dd>
<dt class="label" id="id279"><span class="brackets">29</span></dt>
<dd><p>Robin C Friedman, Kyle Kai-How Farh, Christopher B Burge, and David P Bartel. Most mammalian mRNAs are conserved targets of microRNAs. <em>Genome Res.</em>, 19(1):92–105, January 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.082701.108">http://dx.doi.org/10.1101/gr.082701.108</a>.</p>
</dd>
<dt class="label" id="id232"><span class="brackets">30</span></dt>
<dd><p>Valer Gotea and Wojciech Makałowski. Do transposable elements really contribute to proteomes? <em>Trends Genet.</em>, 22(5):260–267, May 2006. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.tig.2006.03.006">http://dx.doi.org/10.1016/j.tig.2006.03.006</a>.</p>
</dd>
<dt class="label" id="id203"><span class="brackets">31</span></dt>
<dd><p>Seth W Cheetham, Geoffrey J Faulkner, and Marcel E Dinger. Overcoming challenges and dogmas to understand the functions of pseudogenes. <em>Nat. Rev. Genet.</em>, 21(3):191–201, March 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41576-019-0196-1">http://dx.doi.org/10.1038/s41576-019-0196-1</a>.</p>
</dd>
<dt class="label" id="id15"><span class="brackets">32</span></dt>
<dd><p>Nicolas Jarraud. Mecabricks. <span><a class="reference external" href="#"></a></span>https://mecabricks.com/, 2019.</p>
</dd>
<dt class="label" id="id44"><span class="brackets">33</span></dt>
<dd><p>A G Murzin, S E Brenner, T Hubbard, and C Chothia. SCOP: a structural classification of proteins database for the investigation of sequences and structures. <em>J. Mol. Biol.</em>, 247(4):536–540, April 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.1995.0159">http://dx.doi.org/10.1006/jmbi.1995.0159</a>.</p>
</dd>
<dt class="label" id="id161"><span class="brackets">34</span></dt>
<dd><p>Eiko I Fried. What are psychological constructs? on the nature and statistical modelling of emotions, intelligence, personality traits and mental disorders. <em>Health Psychol. Rev.</em>, 11(2):130–134, June 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1080/17437199.2017.1306718">http://dx.doi.org/10.1080/17437199.2017.1306718</a>.</p>
</dd>
<dt class="label" id="id162"><span class="brackets">35</span></dt>
<dd><p>Ian Hacking, Emeritus University Professor Ian Hacking, and Jan Hacking. <em>The Social Construction of What?</em> Harvard University Press, 1999. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=XkCR1p2YMRwC">https://play.google.com/store/books/details?id=XkCR1p2YMRwC</a>.</p>
</dd>
<dt class="label" id="id163"><span class="brackets">36</span></dt>
<dd><p>Thomas Szasz. The myth of mental illness. In James M Humber and Robert F Almeder, editors, <em>Biomedical Ethics and the Law</em>, pages 113–122. Springer US, Boston, MA, 1976. URL: <a class="reference external" href="https://doi.org/10.1007/978-1-4684-2223-8_10">https://doi.org/10.1007/978-1-4684-2223-8_10</a>.</p>
</dd>
<dt class="label" id="id159"><span class="brackets">37</span></dt>
<dd><p>Jocelyn Kaiser. Genetics may explain up to 25% of same-sex behavior, giant analysis reveals. <em>Science Magazine</em>, August 2019. URL: <a class="reference external" href="https://www.sciencemag.org/news/2019/08/genetics-may-explain-25-same-sex-behavior-giant-analysis-reveals">https://www.sciencemag.org/news/2019/08/genetics-may-explain-25-same-sex-behavior-giant-analysis-reveals</a>.</p>
</dd>
<dt class="label" id="id160"><span class="brackets">38</span></dt>
<dd><p>Ken Richardson. What IQ tests test. <em>Theory Psychol.</em>, 12(3):283–314, June 2002. URL: <a class="reference external" href="https://doi.org/10.1177/0959354302012003012">https://doi.org/10.1177/0959354302012003012</a>.</p>
</dd>
<dt class="label" id="id158"><span class="brackets">39</span></dt>
<dd><p>Angela Saini. The disturbing return of scientific racism. <span><a class="reference external" href="#"></a></span>https://www.wired.co.uk/article/superior-the-return-of-race-science-angela-saini. Accessed: 2021-2-11. URL: <a class="reference external" href="https://www.wired.co.uk/article/superior-the-return-of-race-science-angela-saini">https://www.wired.co.uk/article/superior-the-return-of-race-science-angela-saini</a>.</p>
</dd>
<dt class="label" id="id153"><span class="brackets">40</span></dt>
<dd><p>U Schüklenk, E Stein, J Kerin, and W Byne. The ethics of genetic research on sexual orientation. <em>Hastings Cent. Rep.</em>, 27(4):6–13, July 1997. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/9271716">https://www.ncbi.nlm.nih.gov/pubmed/9271716</a>.</p>
</dd>
<dt class="label" id="id166"><span class="brackets">41</span></dt>
<dd><p>Xiaolin Wu and Xi Zhang. Responses to critiques on machine learning of criminality perceptions (addendum of arxiv:1611.04135). <em>arXiv</em>, November 2016. URL: <a class="reference external" href="http://arxiv.org/abs/1611.04135">http://arxiv.org/abs/1611.04135</a>, <a class="reference external" href="https://arxiv.org/abs/1611.04135">arXiv:1611.04135</a>.</p>
</dd>
<dt class="label" id="id167"><span class="brackets">42</span></dt>
<dd><p>Luke Stark. Facial recognition, emotion and race in animated social media. <em>First Monday</em>, September 2018. URL: <a class="reference external" href="http://journals.uic.edu/ojs/index.php/fm/article/view/9406">http://journals.uic.edu/ojs/index.php/fm/article/view/9406</a>.</p>
</dd>
<dt class="label" id="id164"><span class="brackets">43</span></dt>
<dd><p>Andrew Pulrang. Disabled people explained: why we say we don't want to be cured — disability thinking. <span><a class="reference external" href="#"></a></span>https://disabilitythinking.com/disabilitythinking/2019/4/22/disabled-people-explained-why-we-say-we-dont-want-to-be-cured, April 2019. Accessed: 2021-2-11. URL: <a class="reference external" href="https://disabilitythinking.com/disabilitythinking/2019/4/22/disabled-people-explained-why-we-say-we-dont-want-to-be-cured">https://disabilitythinking.com/disabilitythinking/2019/4/22/disabled-people-explained-why-we-say-we-dont-want-to-be-cured</a>.</p>
</dd>
<dt class="label" id="id165"><span class="brackets">44</span></dt>
<dd><p>Paul Steven Miller and Rebecca Leah Levine. Avoiding genetic genocide: understanding good intentions and eugenics in the complex dialogue between the medical and disability communities. <em>Genet. Med.</em>, 15(2):95–102, February 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/gim.2012.102">http://dx.doi.org/10.1038/gim.2012.102</a>.</p>
</dd>
<dt class="label" id="id19"><span class="brackets">45</span></dt>
<dd><p>Franziska Pfeiffer, Carsten Gröber, Michael Blank, Kristian Händler, Marc Beyer, Joachim L Schultze, and Günter Mayer. Systematic evaluation of error rates and causes in short samples in next-generation sequencing. <em>Sci. Rep.</em>, 8(1):10950, July 2018.</p>
</dd>
<dt class="label" id="id20"><span class="brackets">46</span></dt>
<dd><p>A P Jason de Koning, Wanjun Gu, Todd A Castoe, Mark A Batzer, and David D Pollock. Repetitive elements may comprise over two-thirds of the human genome. <em>PLoS Genet.</em>, 7(12):e1002384, December 2011.</p>
</dd>
<dt class="label" id="id18"><span class="brackets">47</span></dt>
<dd><p>Brian J Haas and Michael C Zody. Advancing RNA-Seq analysis. <em>Nat. Biotechnol.</em>, 28(5):421–423, May 2010.</p>
</dd>
<dt class="label" id="id17"><span class="brackets">48</span></dt>
<dd><p>Roger Bumgarner. Overview of DNA microarrays: types, applications, and their future. <em>Curr. Protoc. Mol. Biol.</em>, Chapter 22:Unit 22.1., January 2013.</p>
</dd>
<dt class="label" id="id182"><span class="brackets">49</span></dt>
<dd><p>Karen H Miga. Completing the human genome: the progress and challenge of satellite DNA assembly. <em>Chromosome Res.</em>, 23(3):421–426, September 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s10577-015-9488-2">http://dx.doi.org/10.1007/s10577-015-9488-2</a>.</p>
</dd>
<dt class="label" id="id82"><span class="brackets">50</span></dt>
<dd><p>W James Kent, Charles W Sugnet, Terrence S Furey, Krishna M Roskin, Tom H Pringle, Alan M Zahler, and David Haussler. The human genome browser at UCSC. <em>Genome Res.</em>, 12(6):996–1006, June 2002. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.229102">http://dx.doi.org/10.1101/gr.229102</a>.</p>
</dd>
<dt class="label" id="id81"><span class="brackets">51</span></dt>
<dd><p>C Harger, G Chen, A Farmer, W Huang, J Inman, D Kiphart, F Schilkey, M P Skupski, and J Weller. The genome sequence DataBase. <em>Nucleic Acids Res.</em>, 28(1):31–32, January 2000. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/28.1.31">http://dx.doi.org/10.1093/nar/28.1.31</a>.</p>
</dd>
<dt class="label" id="id83"><span class="brackets">52</span></dt>
<dd><p>T Hubbard, D Barker, E Birney, G Cameron, Y Chen, L Clark, T Cox, J Cuff, V Curwen, T Down, R Durbin, E Eyras, J Gilbert, M Hammond, L Huminiecki, A Kasprzyk, H Lehvaslaiho, P Lijnzaad, C Melsopp, E Mongin, R Pettett, M Pocock, S Potter, A Rust, E Schmidt, S Searle, G Slater, J Smith, W Spooner, A Stabenau, J Stalker, E Stupka, A Ureta-Vidal, I Vastrik, and M Clamp. The ensembl genome database project. <em>Nucleic Acids Res.</em>, 30(1):38–41, January 2002. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/30.1.38">http://dx.doi.org/10.1093/nar/30.1.38</a>.</p>
</dd>
<dt class="label" id="id142"><span class="brackets">53</span></dt>
<dd><p>Genome browser FAQ. <span><a class="reference external" href="#"></a></span>https://genome.ucsc.edu/FAQ/FAQreleases.html. Accessed: 2020-12-13. URL: <a class="reference external" href="https://genome.ucsc.edu/FAQ/FAQreleases.html">https://genome.ucsc.edu/FAQ/FAQreleases.html</a>.</p>
</dd>
<dt class="label" id="id143"><span class="brackets">54</span></dt>
<dd><p>GATK Team. Human genome reference builds - GRCh38 or hg38 - b37 - hg19. <span><a class="reference external" href="#"></a></span>https://gatk.broadinstitute.org/hc/en-us/articles/360035890951, June 2020. Accessed: 2020-12-13. URL: <a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890951">https://gatk.broadinstitute.org/hc/en-us/articles/360035890951</a>.</p>
</dd>
<dt class="label" id="id11"><span class="brackets">55</span></dt>
<dd><p>Steven L Salzberg. Open questions: how many genes do we have? <em>BMC Biol.</em>, 16(1):94, August 2018.</p>
</dd>
<dt class="label" id="id152"><span class="brackets">56</span></dt>
<dd><p>Alice Meadows, Laurel L Haak, and Josh Brown. Persistent identifiers: the building blocks of the research information infrastructure. <em>Insights Imaging</em>, 32(1):9, March 2019. URL: <a class="reference external" href="http://insights.uksg.org/articles/10.1629/uksg.457/">http://insights.uksg.org/articles/10.1629/uksg.457/</a>.</p>
</dd>
<dt class="label" id="id154"><span class="brackets">57</span></dt>
<dd><p>James Vincent. Scientists rename human genes to stop microsoft excel from misreading them as dates. <span><a class="reference external" href="#"></a></span>https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates, August 2020. Accessed: 2021-2-7. URL: <a class="reference external" href="https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates">https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates</a>.</p>
</dd>
<dt class="label" id="id34"><span class="brackets">58</span></dt>
<dd><p>S T Sherry. dbSNP: the NCBI database of genetic variation. <em>Nucleic Acids Res.</em>, 29(1):308–311, 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/29.1.308">http://dx.doi.org/10.1093/nar/29.1.308</a>.</p>
</dd>
<dt class="label" id="id177"><span class="brackets">59</span></dt>
<dd><p>Ryan J Andrews, Levi Baber, and Walter N Moss. RNAStructuromeDB: a genome-wide database for RNA structural inference. <em>Sci. Rep.</em>, 7(1):17269, December 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41598-017-17510-y">http://dx.doi.org/10.1038/s41598-017-17510-y</a>.</p>
</dd>
<dt class="label" id="id178"><span class="brackets">60</span></dt>
<dd><p>Katherine E Richardson, Charles C Kirkpatrick, and Brent M Znosko. RNA CoSSMos 2.0: an improved searchable database of secondary structure motifs in RNA three-dimensional structures. <em>Database</em>, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/database/baz153">http://dx.doi.org/10.1093/database/baz153</a>.</p>
</dd>
<dt class="label" id="id35"><span class="brackets">61</span></dt>
<dd><p>Robert Petryszak, Maria Keays, Y Amy Tang, Nuno A Fonseca, Elisabet Barrera, Tony Burdett, Anja Füllgrabe, Alfonso Muñoz-Pomer Fuentes, Simon Jupp, Satu Koskinen, Oliver Mannion, Laura Huerta, Karine Megy, Catherine Snow, Eleanor Williams, Mitra Barzine, Emma Hastings, Hendrik Weisser, James Wright, Pankaj Jaiswal, Wolfgang Huber, Jyoti Choudhary, Helen E Parkinson, and Alvis Brazma. Expression atlas update—an integrated database of gene and protein expression in humans, animals and plants. 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv1045">http://dx.doi.org/10.1093/nar/gkv1045</a>.</p>
</dd>
<dt class="label" id="id176"><span class="brackets">62</span></dt>
<dd><p>Irene Papatheodorou, Pablo Moreno, Jonathan Manning, Alfonso Muñoz-Pomer Fuentes, Nancy George, Silvie Fexova, Nuno A Fonseca, Anja Füllgrabe, Matthew Green, Ni Huang, Laura Huerta, Haider Iqbal, Monica Jianu, Suhaib Mohammed, Lingyun Zhao, Andrew F Jarnuczak, Simon Jupp, John Marioni, Kerstin Meyer, Robert Petryszak, Cesar Augusto Prada Medina, Carlos Talavera-López, Sarah Teichmann, Juan Antonio Vizcaino, and Alvis Brazma. Expression atlas update: from tissues to single cells. <em>Nucleic Acids Res.</em>, 48(D1):D77–D83, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkz947">http://dx.doi.org/10.1093/nar/gkz947</a>.</p>
</dd>
<dt class="label" id="id206"><span class="brackets">63</span></dt>
<dd><p>Ashraful Haque, Jessica Engel, Sarah A. Teichmann &amp; Tapio Lönnberg. A practical guide to single-cell RNA-sequencing for biomedical research and clinical applications. <em>Genome Medicine</em>, August 2017. URL: <a class="reference external" href="https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-017-0467-4">https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-017-0467-4</a>.</p>
</dd>
<dt class="label" id="id36"><span class="brackets">64</span></dt>
<dd><p>I Illumina. Understanding illumina quality scores. <em>Technical Note: Informatics</em>, 2014. URL: <a class="reference external" href="https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf">https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf</a>.</p>
</dd>
<dt class="label" id="id37"><span class="brackets">65</span></dt>
<dd><p>Günter P Wagner, Koryu Kin, and Vincent J Lynch. Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples. <em>Theory Biosci.</em>, 131(4):281–285, December 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s12064-012-0162-3">http://dx.doi.org/10.1007/s12064-012-0162-3</a>.</p>
</dd>
<dt class="label" id="id38"><span class="brackets">66</span></dt>
<dd><p>H Pimentel. What the FPKM? a review of RNA-Seq expression units. 2014. URL: <a class="reference external" href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/</a>.</p>
</dd>
<dt class="label" id="id39"><span class="brackets">67</span></dt>
<dd><p>M D Robinson, D J McCarthy, and G K Smyth. Edger: a bioconductor package for differential expression analysis of digital gene expression data. <em>Bioinformatics</em>, 26(1):139–140, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btp616">http://dx.doi.org/10.1093/bioinformatics/btp616</a>.</p>
</dd>
<dt class="label" id="id40"><span class="brackets">68</span></dt>
<dd><p>Michael I Love, Wolfgang Huber, and Simon Anders. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. <em>Genome Biol.</em>, 15(12):550, 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-014-0550-8">http://dx.doi.org/10.1186/s13059-014-0550-8</a>.</p>
</dd>
<dt class="label" id="id13"><span class="brackets">69</span></dt>
<dd><p>Robert T Hersh. Atlas of protein sequence and structure, 1966. <em>Syst. Biol.</em>, 16(3):262–263, September 1967.</p>
</dd>
<dt class="label" id="id55"><span class="brackets">70</span></dt>
<dd><p>Sangya Pundir, Maria J Martin, Claire O'Donovan, and The UniProt Consortium. UniProt tools. <em>Curr. Protoc. Bioinformatics</em>, pages 1.29.1–1.29.15, 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1002/0471250953.bi0129s53">http://dx.doi.org/10.1002/0471250953.bi0129s53</a>.</p>
</dd>
<dt class="label" id="id180"><span class="brackets">71</span></dt>
<dd><p>M Wang, M Weiss, M Simonovic, G Haertinger, S P Schrimpf, M O Hengartner, and C von Mering. PaxDb, a database of protein abundance averages across all three domains of life. <em>Mol. Cell. Proteomics</em>, 11(8):492–500, August 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1074/mcp.O111.014704">http://dx.doi.org/10.1074/mcp.O111.014704</a>.</p>
</dd>
<dt class="label" id="id181"><span class="brackets">72</span></dt>
<dd><p>Patroklos Samaras, Tobias Schmidt, Martin Frejno, Siegfried Gessulat, Maria Reinecke, Anna Jarzab, Jana Zecha, Julia Mergner, Piero Giansanti, Hans-Christian Ehrlich, Stephan Aiche, Johannes Rank, Harald Kienegger, Helmut Krcmar, Bernhard Kuster, and Mathias Wilhelm. ProteomicsDB: a multi-omics and multi-organism resource for life science research. <em>Nucleic Acids Res.</em>, 48(D1):D1153–D1163, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkz974">http://dx.doi.org/10.1093/nar/gkz974</a>.</p>
</dd>
<dt class="label" id="id91"><span class="brackets">73</span></dt>
<dd><p>Björn Schwanhäusser, Dorothea Busse, Na Li, Gunnar Dittmar, Johannes Schuchhardt, Jana Wolf, Wei Chen, and Matthias Selbach. Global quantification of mammalian gene expression control. <em>Nature</em>, 473(7347):337–342, May 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nature10098">http://dx.doi.org/10.1038/nature10098</a>.</p>
</dd>
<dt class="label" id="id92"><span class="brackets">74</span></dt>
<dd><p>S P Gygi, Y Rochon, B R Franza, and R Aebersold. Correlation between protein and mRNA abundance in yeast. <em>Mol. Cell. Biol.</em>, 19(3):1720–1730, March 1999. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/10022859">https://www.ncbi.nlm.nih.gov/pubmed/10022859</a>.</p>
</dd>
<dt class="label" id="id93"><span class="brackets">75</span></dt>
<dd><p>Idit Kosti, Nishant Jain, Dvir Aran, Atul J Butte, and Marina Sirota. Cross-tissue analysis of gene and protein expression in normal and cancer tissues. <em>Sci. Rep.</em>, 6:24799, May 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/srep24799">http://dx.doi.org/10.1038/srep24799</a>.</p>
</dd>
<dt class="label" id="id14"><span class="brackets">76</span></dt>
<dd><p>Protein data bank.</p>
</dd>
<dt class="label" id="id168"><span class="brackets">77</span></dt>
<dd><p>Adrian Furnham. Response bias, social desirability and dissimulation. <em>Pers. Individ. Dif.</em>, 7(3):385–400, January 1986. URL: <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/0191886986900140">https://www.sciencedirect.com/science/article/pii/0191886986900140</a>.</p>
</dd>
<dt class="label" id="id169"><span class="brackets">78</span></dt>
<dd><p>B Knäuper and H U Wittchen. Diagnosing major depression in the elderly: evidence for response bias in standardized diagnostic interviews? <em>J. Psychiatr. Res.</em>, 28(2):147–164, March 1994. URL: <a class="reference external" href="http://dx.doi.org/10.1016/0022-3956(94)90026-4">http://dx.doi.org/10.1016/0022-3956(94)90026-4</a>.</p>
</dd>
<dt class="label" id="id130"><span class="brackets">79</span></dt>
<dd><p>Golding, Golding, Pembrey, Jones, and The Alspac Study Team. ALSPAC-The avon longitudinal study of parents and children. <em>Paediatr. Perinat. Epidemiol.</em>, 15(1):74–87, 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1046/j.1365-3016.2001.00325.x">http://dx.doi.org/10.1046/j.1365-3016.2001.00325.x</a>.</p>
</dd>
<dt class="label" id="id170"><span class="brackets">80</span></dt>
<dd><p>Clare Bycroft, Colin Freeman, Desislava Petkova, Gavin Band, Lloyd T Elliott, Kevin Sharp, Allan Motyer, Damjan Vukcevic, Olivier Delaneau, Jared O'Connell, Adrian Cortes, Samantha Welsh, Alan Young, Mark Effingham, Gil McVean, Stephen Leslie, Naomi Allen, Peter Donnelly, and Jonathan Marchini. The UK biobank resource with deep phenotyping and genomic data. <em>Nature</em>, 562(7726):203–209, October 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41586-018-0579-z">http://dx.doi.org/10.1038/s41586-018-0579-z</a>.</p>
</dd>
<dt class="label" id="id171"><span class="brackets">81</span></dt>
<dd><p>Kendall Powell. The broken promise that undermines human genome research. <em>Nature</em>, 590(7845):198–201, February 2021. URL: <a class="reference external" href="https://www.nature.com/articles/d41586-021-00331-5">https://www.nature.com/articles/d41586-021-00331-5</a>.</p>
</dd>
<dt class="label" id="id22"><span class="brackets">82</span></dt>
<dd><p>Annalisa Buniello, Jacqueline A L MacArthur, Maria Cerezo, Laura W Harris, James Hayhurst, Cinzia Malangone, Aoife McMahon, Joannella Morales, Edward Mountjoy, Elliot Sollis, Daniel Suveges, Olga Vrousgou, Patricia L Whetzel, Ridwan Amode, Jose A Guillen, Harpreet S Riat, Stephen J Trevanion, Peggy Hall, Heather Junkins, Paul Flicek, Tony Burdett, Lucia A Hindorff, Fiona Cunningham, and Helen Parkinson. The NHGRI-EBI GWAS catalog of published genome-wide association studies, targeted arrays and summary statistics 2019. <em>Nucleic Acids Res.</em>, 47(D1):D1005–D1012, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gky1120">http://dx.doi.org/10.1093/nar/gky1120</a>.</p>
</dd>
<dt class="label" id="id23"><span class="brackets">83</span></dt>
<dd><p>J Macarthur L. Emery. GWAS catalog: exploring SNP-trait associations. <span><a class="reference external" href="#"></a></span>http://europepmc.org/article/CTX/C7914, December 2017. Accessed: 2020-9-3. URL: <a class="reference external" href="http://europepmc.org/article/CTX/C7914">http://europepmc.org/article/CTX/C7914</a>.</p>
</dd>
<dt class="label" id="id248"><span class="brackets">84</span></dt>
<dd><p>Joshua C Denny, Marylyn D Ritchie, Melissa A Basford, Jill M Pulley, Lisa Bastarache, Kristin Brown-Gentry, Deede Wang, Dan R Masys, Dan M Roden, and Dana C Crawford. PheWAS: demonstrating the feasibility of a phenome-wide scan to discover gene–disease associations. <em>Bioinformatics</em>, 26(9):1205–1210, March 2010. URL: <a class="reference external" href="https://academic.oup.com/bioinformatics/article-abstract/26/9/1205/201211">https://academic.oup.com/bioinformatics/article-abstract/26/9/1205/201211</a>.</p>
</dd>
<dt class="label" id="id21"><span class="brackets">85</span></dt>
<dd><p>Ivana Barbaric, Gaynor Miller, and T Neil Dear. Appearances can be deceiving: phenotypes of knockout mice. <em>Brief. Funct. Genomic. Proteomic.</em>, 6(2):91–103, June 2007.</p>
</dd>
<dt class="label" id="id172"><span class="brackets">86</span></dt>
<dd><p>Aihua Zhang, Hui Sun, Guangli Yan, Ping Wang, and Xijun Wang. Mass spectrometry-based metabolomics: applications to biomarker and metabolic pathway research. <em>Biomed. Chromatogr.</em>, 30(1):7–12, January 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1002/bmc.3453">http://dx.doi.org/10.1002/bmc.3453</a>.</p>
</dd>
<dt class="label" id="id173"><span class="brackets">87</span></dt>
<dd><p>Ganesh A Viswanathan, Jeremy Seto, Sonali Patil, German Nudelman, and Stuart C Sealfon. Getting started in biological pathway construction and analysis. <em>PLoS Comput. Biol.</em>, 4(2):e16, February 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pcbi.0040016">http://dx.doi.org/10.1371/journal.pcbi.0040016</a>.</p>
</dd>
<dt class="label" id="id174"><span class="brackets">88</span></dt>
<dd><p>Antonio Fabregat, Steven Jupe, Lisa Matthews, Konstantinos Sidiropoulos, Marc Gillespie, Phani Garapati, Robin Haw, Bijay Jassal, Florian Korninger, Bruce May, Marija Milacic, Corina Duenas Roca, Karen Rothfels, Cristoffer Sevilla, Veronica Shamovsky, Solomon Shorser, Thawfeek Varusai, Guilherme Viteri, Joel Weiser, Guanming Wu, Lincoln Stein, Henning Hermjakob, and Peter D'Eustachio. The reactome pathway knowledgebase. <em>Nucleic Acids Res.</em>, 46(D1):D649–D655, January 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkx1132">http://dx.doi.org/10.1093/nar/gkx1132</a>.</p>
</dd>
<dt class="label" id="id175"><span class="brackets">89</span></dt>
<dd><p>Minoru Kanehisa, Michihiro Araki, Susumu Goto, Masahiro Hattori, Mika Hirakawa, Masumi Itoh, Toshiaki Katayama, Shuichi Kawashima, Shujiro Okuda, Toshiaki Tokimatsu, and Yoshihiro Yamanishi. KEGG for linking genomes to life and the environment. <em>Nucleic Acids Res.</em>, 36(Database issue):D480–4, January 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkm882">http://dx.doi.org/10.1093/nar/gkm882</a>.</p>
</dd>
<dt class="label" id="id25"><span class="brackets">90</span></dt>
<dd><p>Wilfrid Blunt. <em>The Compleat Naturalist: A Life of Linnaeus</em>. Frances Lincoln, 2001. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=B3YOvgAACAAJ">https://play.google.com/store/books/details?id=B3YOvgAACAAJ</a>.</p>
</dd>
<dt class="label" id="id26"><span class="brackets">91</span></dt>
<dd><p>Ehret. <em>Plantae et papiliones rariores</em>. Volume 1748. [London :s.n.],, 1748. URL: <a class="reference external" href="https://www.biodiversitylibrary.org/item/205762">https://www.biodiversitylibrary.org/item/205762</a>.</p>
</dd>
<dt class="label" id="id141"><span class="brackets">92</span></dt>
<dd><p>Dr Isabelle Charmantier. Linnaeus and race. <span><a class="reference external" href="#"></a></span>https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race. Accessed: 2020-11-30. URL: <a class="reference external" href="https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race">https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race</a>.</p>
</dd>
<dt class="label" id="id24"><span class="brackets">93</span></dt>
<dd><p>Lars J Jensen and Peer Bork. Ontologies in quantitative biology: a basis for comparison, integration, and discovery. <em>PLoS Biol.</em>, 8(5):e1000374, May 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1000374">http://dx.doi.org/10.1371/journal.pbio.1000374</a>.</p>
</dd>
<dt class="label" id="id68"><span class="brackets">94</span></dt>
<dd><p>James A Overton, Heiko Dietze, Shahim Essaid, David Osumi-Sutherland, and Christopher J Mungall. ROBOT: a command-line tool for ontology development. In <em>ICBO</em>. ceur-ws.org, 2015. URL: <a class="reference external" href="http://ceur-ws.org/Vol-1515/demo6.pdf">http://ceur-ws.org/Vol-1515/demo6.pdf</a>.</p>
</dd>
<dt class="label" id="id72"><span class="brackets">95</span></dt>
<dd><p>Eran Eden, Roy Navon, Israel Steinfeld, Doron Lipson, and Zohar Yakhini. GOrilla: a tool for discovery and visualization of enriched GO terms in ranked gene lists. <em>BMC Bioinformatics</em>, 10:48, February 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-10-48">http://dx.doi.org/10.1186/1471-2105-10-48</a>.</p>
</dd>
<dt class="label" id="id70"><span class="brackets">96</span></dt>
<dd><p>M Ashburner, C A Ball, J A Blake, D Botstein, H Butler, J M Cherry, A P Davis, K Dolinski, S S Dwight, J T Eppig, M A Harris, D P Hill, L Issel-Tarver, A Kasarskis, S Lewis, J C Matese, J E Richardson, M Ringwald, G M Rubin, and G Sherlock. Gene ontology: tool for the unification of biology. the gene ontology consortium. <em>Nat. Genet.</em>, 25(1):25–29, May 2000. URL: <a class="reference external" href="http://dx.doi.org/10.1038/75556">http://dx.doi.org/10.1038/75556</a>.</p>
</dd>
<dt class="label" id="id71"><span class="brackets">97</span></dt>
<dd><p>Paul D Thomas. The gene ontology and the meaning of biological function. <em>Methods Mol. Biol.</em>, 1446:15–24, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1007/978-1-4939-3743-1_2">http://dx.doi.org/10.1007/978-1-4939-3743-1_2</a>.</p>
</dd>
<dt class="label" id="id183"><span class="brackets">98</span></dt>
<dd><p>Evelyn Camon, Michele Magrane, Daniel Barrell, Vivian Lee, Emily Dimmer, John Maslen, David Binns, Nicola Harte, Rodrigo Lopez, and Rolf Apweiler. The gene ontology annotation (GOA) database: sharing knowledge in uniprot with gene ontology. <em>Nucleic Acids Res.</em>, 32(Database issue):D262–6, January 2004. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkh021">http://dx.doi.org/10.1093/nar/gkh021</a>.</p>
</dd>
<dt class="label" id="id50"><span class="brackets">99</span></dt>
<dd><p>Christopher J Mungall, Carlo Torniai, Georgios V Gkoutos, Suzanna E Lewis, and Melissa A Haendel. Uberon, an integrative multi-species anatomy ontology. <em>Genome Biol.</em>, 13(1):R5, January 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gb-2012-13-1-r5">http://dx.doi.org/10.1186/gb-2012-13-1-r5</a>.</p>
</dd>
<dt class="label" id="id49"><span class="brackets">100</span></dt>
<dd><p>Venkat S Malladi, Drew T Erickson, Nikhil R Podduturi, Laurence D Rowe, Esther T Chan, Jean M Davidson, Benjamin C Hitz, Marcus Ho, Brian T Lee, Stuart Miyasato, Gregory R Roe, Matt Simison, Cricket A Sloan, J Seth Strattan, Forrest Tanaka, W James Kent, J Michael Cherry, and Eurie L Hong. Ontology application and use at the ENCODE DCC. <em>Database</em>, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/database/bav010">http://dx.doi.org/10.1093/database/bav010</a>.</p>
</dd>
<dt class="label" id="id186"><span class="brackets">101</span></dt>
<dd><p>Lynn M Schriml, Elvira Mitraka, James Munro, Becky Tauber, Mike Schor, Lance Nickle, Victor Felix, Linda Jeng, Cynthia Bearer, Richard Lichenstein, Katharine Bisordi, Nicole Campion, Brooke Hyman, David Kurland, Connor Patrick Oates, Siobhan Kibbey, Poorna Sreekumar, Chris Le, Michelle Giglio, and Carol Greene. Human disease ontology 2018 update: classification, content and workflow expansion. <em>Nucleic Acids Res.</em>, 47(D1):D955–D962, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gky1032">http://dx.doi.org/10.1093/nar/gky1032</a>.</p>
</dd>
<dt class="label" id="id187"><span class="brackets">102</span></dt>
<dd><p>Sebastian Köhler, Michael Gargano, Nicolas Matentzoglu, Leigh C Carmody, David Lewis-Smith, Nicole A Vasilevsky, Daniel Danis, Ganna Balagura, Gareth Baynam, Amy M Brower, Tiffany J Callahan, Christopher G Chute, Johanna L Est, Peter D Galer, Shiva Ganesan, Matthias Griese, Matthias Haimel, Julia Pazmandi, Marc Hanauer, Nomi L Harris, Michael J Hartnett, Maximilian Hastreiter, Fabian Hauck, Yongqun He, Tim Jeske, Hugh Kearney, Gerhard Kindle, Christoph Klein, Katrin Knoflach, Roland Krause, David Lagorce, Julie A McMurry, Jillian A Miller, Monica C Munoz-Torres, Rebecca L Peters, Christina K Rapp, Ana M Rath, Shahmir A Rind, Avi Z Rosenberg, Michael M Segal, Markus G Seidel, Damian Smedley, Tomer Talmy, Yarlalu Thomas, Samuel A Wiafe, Julie Xian, Zafer Yüksel, Ingo Helbig, Christopher J Mungall, Melissa A Haendel, and Peter N Robinson. The human phenotype ontology in 2021. <em>Nucleic Acids Res.</em>, 49(D1):D1207–D1217, January 2021. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkaa1043">http://dx.doi.org/10.1093/nar/gkaa1043</a>.</p>
</dd>
<dt class="label" id="id188"><span class="brackets">103</span></dt>
<dd><p>James Malone, Ele Holloway, Tomasz Adamusiak, Misha Kapushesky, Jie Zheng, Nikolay Kolesnikov, Anna Zhukova, Alvis Brazma, and Helen Parkinson. Modeling sample variables with an experimental factor ontology. <em>Bioinformatics</em>, 26(8):1112–1118, April 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btq099">http://dx.doi.org/10.1093/bioinformatics/btq099</a>.</p>
</dd>
<dt class="label" id="id43"><span class="brackets">104</span></dt>
<dd><p>S F Altschul, W Gish, W Miller, E W Myers, and D J Lipman. Basic local alignment search tool. <em>J. Mol. Biol.</em>, 215(3):403–410, October 1990. URL: <a class="reference external" href="http://dx.doi.org/10.1016/S0022-2836(05)80360-2">http://dx.doi.org/10.1016/S0022-2836(05)80360-2</a>.</p>
</dd>
<dt class="label" id="id45"><span class="brackets">105</span></dt>
<dd><p>Antonina Andreeva, Dave Howorth, Cyrus Chothia, Eugene Kulesha, and Alexey G Murzin. SCOP2 prototype: a new approach to protein structure mining. <em>Nucleic Acids Res.</em>, 42(Database issue):D310–4, January 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkt1242">http://dx.doi.org/10.1093/nar/gkt1242</a>.</p>
</dd>
<dt class="label" id="id46"><span class="brackets">106</span></dt>
<dd><p>C A Orengo, A D Michie, S Jones, D T Jones, M B Swindells, and J M Thornton. CATH – a hierarchic classification of protein domain structures. <em>Structure</em>, 5(8):1093–1109, 1997. URL: <a class="reference external" href="http://dx.doi.org/10.1016/s0969-2126(97)00260-8">http://dx.doi.org/10.1016/s0969-2126(97)00260-8</a>.</p>
</dd>
<dt class="label" id="id47"><span class="brackets">107</span></dt>
<dd><p>Gergely Csaba, Fabian Birzele, and Ralf Zimmer. Systematic comparison of SCOP and CATH: a new gold standard for protein structure analysis. <em>BMC Struct. Biol.</em>, 9:23, April 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1472-6807-9-23">http://dx.doi.org/10.1186/1472-6807-9-23</a>.</p>
</dd>
<dt class="label" id="id48"><span class="brackets">108</span></dt>
<dd><p>J Gough, K Karplus, R Hughey, and C Chothia. Assignment of homology to genome sequences using a library of hidden markov models that represent all proteins of known structure. <em>J. Mol. Biol.</em>, 313(4):903–919, November 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.2001.5080">http://dx.doi.org/10.1006/jmbi.2001.5080</a>.</p>
</dd>
<dt class="label" id="id51"><span class="brackets">109</span></dt>
<dd><p>Hai Fang and Julian Gough. DcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more. <em>Nucleic Acids Res.</em>, 41(Database issue):D536–44, January 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gks1080">http://dx.doi.org/10.1093/nar/gks1080</a>.</p>
</dd>
<dt class="label" id="id52"><span class="brackets">110</span></dt>
<dd><p>Hai Fang and Julian Gough. A domain-centric solution to functional genomics via dcGO predictor. <em>BMC Bioinformatics</em>, 14 Suppl 3:S9, February 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-14-S3-S9">http://dx.doi.org/10.1186/1471-2105-14-S3-S9</a>.</p>
</dd>
<dt class="label" id="id113"><span class="brackets">111</span></dt>
<dd><p>Yuxiang Jiang, Tal Ronnen Oron, Wyatt T Clark, Asma R Bankapur, Daniel D'Andrea, Rosalba Lepore, Christopher S Funk, Indika Kahanda, Karin M Verspoor, Asa Ben-Hur, Da Chen Emily Koo, Duncan Penfold-Brown, Dennis Shasha, Noah Youngs, Richard Bonneau, Alexandra Lin, Sayed M E Sahraeian, Pier Luigi Martelli, Giuseppe Profiti, Rita Casadio, Renzhi Cao, Zhaolong Zhong, Jianlin Cheng, Adrian Altenhoff, Nives Skunca, Christophe Dessimoz, Tunca Dogan, Kai Hakala, Suwisa Kaewphan, Farrokh Mehryary, Tapio Salakoski, Filip Ginter, Hai Fang, Ben Smithers, Matt Oates, Julian Gough, Petri Törönen, Patrik Koskinen, Liisa Holm, Ching-Tai Chen, Wen-Lian Hsu, Kevin Bryson, Domenico Cozzetto, Federico Minneci, David T Jones, Samuel Chapman, Dukka Bkc, Ishita K Khan, Daisuke Kihara, Dan Ofer, Nadav Rappoport, Amos Stern, Elena Cibrian-Uhalte, Paul Denny, Rebecca E Foulger, Reija Hieta, Duncan Legge, Ruth C Lovering, Michele Magrane, Anna N Melidoni, Prudence Mutowo-Meullenet, Klemens Pichler, Aleksandra Shypitsyna, Biao Li, Pooya Zakeri, Sarah ElShal, Léon-Charles Tranchevent, Sayoni Das, Natalie L Dawson, David Lee, Jonathan G Lees, Ian Sillitoe, Prajwal Bhat, Tamás Nepusz, Alfonso E Romero, Rajkumar Sasidharan, Haixuan Yang, Alberto Paccanaro, Jesse Gillis, Adriana E Sedeño-Cortés, Paul Pavlidis, Shou Feng, Juan M Cejuela, Tatyana Goldberg, Tobias Hamp, Lothar Richter, Asaf Salamov, Toni Gabaldon, Marina Marcet-Houben, Fran Supek, Qingtian Gong, Wei Ning, Yuanpeng Zhou, Weidong Tian, Marco Falda, Paolo Fontana, Enrico Lavezzo, Stefano Toppo, Carlo Ferrari, Manuel Giollo, Damiano Piovesan, Silvio C E Tosatto, Angela Del Pozo, José M Fernández, Paolo Maietta, Alfonso Valencia, Michael L Tress, Alfredo Benso, Stefano Di Carlo, Gianfranco Politano, Alessandro Savino, Hafeez Ur Rehman, Matteo Re, Marco Mesiti, Giorgio Valentini, Joachim W Bargsten, Aalt D J van Dijk, Branislava Gemovic, Sanja Glisic, Vladmir Perovic, Veljko Veljkovic, Nevena Veljkovic, Danillo C Almeida-E-Silva, Ricardo Z N Vencio, Malvika Sharan, Jörg Vogel, Lakesh Kansakar, Shanshan Zhang, Slobodan Vucetic, Zheng Wang, Michael J E Sternberg, Mark N Wass, Rachael P Huntley, Maria J Martin, Claire O'Donovan, Peter N Robinson, Yves Moreau, Anna Tramontano, Patricia C Babbitt, Steven E Brenner, Michal Linial, Christine A Orengo, Burkhard Rost, Casey S Greene, Sean D Mooney, Iddo Friedberg, and Predrag Radivojac. An expanded evaluation of protein function prediction methods shows an improvement in accuracy. <em>Genome Biol.</em>, 17(1):184, September 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-016-1037-6">http://dx.doi.org/10.1186/s13059-016-1037-6</a>.</p>
</dd>
<dt class="label" id="id114"><span class="brackets">112</span></dt>
<dd><p>Predrag Radivojac, Wyatt T Clark, Tal Ronnen Oron, Alexandra M Schnoes, Tobias Wittkop, Artem Sokolov, Kiley Graim, Christopher Funk, Karin Verspoor, Asa Ben-Hur, Gaurav Pandey, Jeffrey M Yunes, Ameet S Talwalkar, Susanna Repo, Michael L Souza, Damiano Piovesan, Rita Casadio, Zheng Wang, Jianlin Cheng, Hai Fang, Julian Gough, Patrik Koskinen, Petri Törönen, Jussi Nokso-Koivisto, Liisa Holm, Domenico Cozzetto, Daniel W A Buchan, Kevin Bryson, David T Jones, Bhakti Limaye, Harshal Inamdar, Avik Datta, Sunitha K Manjari, Rajendra Joshi, Meghana Chitale, Daisuke Kihara, Andreas M Lisewski, Serkan Erdin, Eric Venner, Olivier Lichtarge, Robert Rentzsch, Haixuan Yang, Alfonso E Romero, Prajwal Bhat, Alberto Paccanaro, Tobias Hamp, Rebecca Kaßner, Stefan Seemayer, Esmeralda Vicedo, Christian Schaefer, Dominik Achten, Florian Auer, Ariane Boehm, Tatjana Braun, Maximilian Hecht, Mark Heron, Peter Hönigschmid, Thomas A Hopf, Stefanie Kaufmann, Michael Kiening, Denis Krompass, Cedric Landerer, Yannick Mahlich, Manfred Roos, Jari Björne, Tapio Salakoski, Andrew Wong, Hagit Shatkay, Fanny Gatzmann, Ingolf Sommer, Mark N Wass, Michael J E Sternberg, Nives Škunca, Fran Supek, Matko Bošnjak, Panče Panov, Sašo Džeroski, Tomislav Šmuc, Yiannis A I Kourmpetis, Aalt D J van Dijk, Cajo J F ter Braak, Yuanpeng Zhou, Qingtian Gong, Xinran Dong, Weidong Tian, Marco Falda, Paolo Fontana, Enrico Lavezzo, Barbara Di Camillo, Stefano Toppo, Liang Lan, Nemanja Djuric, Yuhong Guo, Slobodan Vucetic, Amos Bairoch, Michal Linial, Patricia C Babbitt, Steven E Brenner, Christine Orengo, Burkhard Rost, Sean D Mooney, and Iddo Friedberg. A large-scale evaluation of computational protein function prediction. <em>Nat. Methods</em>, 10(3):221–227, March 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.2340">http://dx.doi.org/10.1038/nmeth.2340</a>.</p>
</dd>
<dt class="label" id="id53"><span class="brackets">113</span></dt>
<dd><p>Hashem A Shihab, Julian Gough, David N Cooper, Peter D Stenson, Gary L A Barker, Keith J Edwards, Ian N M Day, and Tom R Gaunt. Predicting the functional, molecular, and phenotypic consequences of amino acid substitutions using hidden markov models. <em>Hum. Mutat.</em>, 34(1):57–65, January 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1002/humu.22225">http://dx.doi.org/10.1002/humu.22225</a>.</p>
</dd>
<dt class="label" id="id54"><span class="brackets">114</span></dt>
<dd><p>Peter D Stenson, Matthew Mort, Edward V Ball, Katy Howells, Andrew D Phillips, Nick St Thomas, and David N Cooper. The human gene mutation database: 2008 update. <em>Genome Med.</em>, 1(1):13, January 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gm13">http://dx.doi.org/10.1186/gm13</a>.</p>
</dd>
<dt class="label" id="id56"><span class="brackets">115</span></dt>
<dd><p>William McLaren, Laurent Gil, Sarah E Hunt, Harpreet Singh Riat, Graham R S Ritchie, Anja Thormann, Paul Flicek, and Fiona Cunningham. The ensembl variant effect predictor. <em>Genome Biol.</em>, 17(1):122, June 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-016-0974-4">http://dx.doi.org/10.1186/s13059-016-0974-4</a>.</p>
</dd>
<dt class="label" id="id131"><span class="brackets">116</span></dt>
<dd><p>Gregory McInnes, Roxana Daneshjou, Panagiostis Katsonis, Olivier Lichtarge, Rajgopal Srinivasan, Sadhna Rana, Predrag Radivojac, Sean D Mooney, Kymberleigh A Pagel, Moses Stamboulian, Yuxiang Jiang, Emidio Capriotti, Yanran Wang, Yana Bromberg, Samuele Bovo, Castrense Savojardo, Pier Luigi Martelli, Rita Casadio, Lipika R Pal, John Moult, Steven E Brenner, and Russ Altman. Predicting venous thromboembolism risk from exomes in the critical assessment of genome interpretation (CAGI) challenges. <em>Hum. Mutat.</em>, 40(9):1314–1320, September 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1002/humu.23825">http://dx.doi.org/10.1002/humu.23825</a>.</p>
</dd>
<dt class="label" id="id258"><span class="brackets">117</span></dt>
<dd><p>Laura Kasak, Jesse M Hunter, Rupa Udani, Constantina Bakolitsa, Zhiqiang Hu, Aashish N Adhikari, Giulia Babbi, Rita Casadio, Julian Gough, Rafael F Guerrero, Yuxiang Jiang, Thomas Joseph, Panagiotis Katsonis, Sujatha Kotte, Kunal Kundu, Olivier Lichtarge, Pier Luigi Martelli, Sean D Mooney, John Moult, Lipika R Pal, Jennifer Poitras, Predrag Radivojac, Aditya Rao, Naveen Sivadasan, Uma Sunderam, V G Saipradeep, Yizhou Yin, Jan Zaucha, Steven E Brenner, and M Stephen Meyn. CAGI SickKids challenges: assessment of phenotype and variant predictions derived from clinical and genomic data of children with undiagnosed diseases. <em>Hum. Mutat.</em>, 40(9):1373–1391, September 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1002/humu.23874">http://dx.doi.org/10.1002/humu.23874</a>.</p>
</dd>
<dt class="label" id="id219"><span class="brackets">118</span></dt>
<dd><p>The Turing Way Community, Becky Arnold, Louise Bowler, Sarah Gibson, Patricia Herterich, Rosie Higman, Anna Krystalli, Alexander Morley, Martin O'Reilly, and Kirstie Whitaker. The turing way: a handbook for reproducible data science. March 2019. URL: <a class="reference external" href="https://zenodo.org/record/3233986">https://zenodo.org/record/3233986</a>.</p>
</dd>
<dt class="label" id="id57"><span class="brackets">119</span></dt>
<dd><p>C Glenn Begley and Lee M Ellis. Drug development: raise standards for preclinical cancer research. <em>Nature</em>, 483(7391):531–533, March 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1038/483531a">http://dx.doi.org/10.1038/483531a</a>.</p>
</dd>
<dt class="label" id="id58"><span class="brackets">120</span></dt>
<dd><p>Florian Prinz, Thomas Schlange, and Khusru Asadullah. Believe it or not: how much can we rely on published data on potential drug targets? <em>Nat. Rev. Drug Discov.</em>, 10(9):712–712, 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrd3439-c1">http://dx.doi.org/10.1038/nrd3439-c1</a>.</p>
</dd>
<dt class="label" id="id59"><span class="brackets">121</span></dt>
<dd><p>Open Science Collaboration. PSYCHOLOGY. estimating the reproducibility of psychological science. <em>Science</em>, 349(6251):aac4716, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.aac4716">http://dx.doi.org/10.1126/science.aac4716</a>.</p>
</dd>
<dt class="label" id="id60"><span class="brackets">122</span></dt>
<dd><p>Monya Baker. 1,500 scientists lift the lid on reproducibility. <em>Nature</em>, 533(7604):452–454, May 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/533452a">http://dx.doi.org/10.1038/533452a</a>.</p>
</dd>
<dt class="label" id="id61"><span class="brackets">123</span></dt>
<dd><p>Dorothy Bishop. Rein in the four horsemen of irreproducibility. <em>Nature</em>, 568(7753):435, April 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/d41586-019-01307-2">http://dx.doi.org/10.1038/d41586-019-01307-2</a>.</p>
</dd>
<dt class="label" id="id234"><span class="brackets">124</span></dt>
<dd><p>Daniele Fanelli. How many scientists fabricate and falsify research? a systematic review and meta-analysis of survey data. <em>PLoS One</em>, 4(5):e5738, May 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pone.0005738">http://dx.doi.org/10.1371/journal.pone.0005738</a>.</p>
</dd>
<dt class="label" id="id62"><span class="brackets">125</span></dt>
<dd><p>John P A Ioannidis. Why most published research findings are false. <em>PLoS Med.</em>, 2(8):e124, August 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pmed.0020124">http://dx.doi.org/10.1371/journal.pmed.0020124</a>.</p>
</dd>
<dt class="label" id="id240"><span class="brackets">126</span></dt>
<dd><p>Daniel Lakens, Federico G Adolfi, Casper J Albers, Farid Anvari, Matthew A J Apps, Shlomo E Argamon, Thom Baguley, Raymond B Becker, Stephen D Benning, Daniel E Bradford, Erin M Buchanan, Aaron R Caldwell, Ben Van Calster, Rickard Carlsson, Sau-Chin Chen, Bryan Chung, Lincoln J Colling, Gary S Collins, Zander Crook, Emily S Cross, Sameera Daniels, Henrik Danielsson, Lisa DeBruine, Daniel J Dunleavy, Brian D Earp, Michele I Feist, Jason D Ferrell, James G Field, Nicholas W Fox, Amanda Friesen, Caio Gomes, Monica Gonzalez-Marquez, James A Grange, Andrew P Grieve, Robert Guggenberger, James Grist, Anne-Laura van Harmelen, Fred Hasselman, Kevin D Hochard, Mark R Hoffarth, Nicholas P Holmes, Michael Ingre, Peder M Isager, Hanna K Isotalus, Christer Johansson, Konrad Juszczyk, David A Kenny, Ahmed A Khalil, Barbara Konat, Junpeng Lao, Erik Gahner Larsen, Gerine M A Lodder, Jiří Lukavský, Christopher R Madan, David Manheim, Stephen R Martin, Andrea E Martin, Deborah G Mayo, Randy J McCarthy, Kevin McConway, Colin McFarland, Amanda Q X Nio, Gustav Nilsonne, Cilene Lino de Oliveira, Jean-Jacques Orban de Xivry, Sam Parsons, Gerit Pfuhl, Kimberly A Quinn, John J Sakon, S Adil Saribay, Iris K Schneider, Manojkumar Selvaraju, Zsuzsika Sjoerds, Samuel G Smith, Tim Smits, Jeffrey R Spies, Vishnu Sreekumar, Crystal N Steltenpohl, Neil Stenhouse, Wojciech Świątkowski, Miguel A Vadillo, Marcel A L M Van Assen, Matt N Williams, Samantha E Williams, Donald R Williams, Tal Yarkoni, Ignazio Ziano, and Rolf A Zwaan. Justify your alpha. <em>Nature Human Behaviour</em>, 2(3):168–171, February 2018. URL: <a class="reference external" href="https://www.nature.com/articles/s41562-018-0311-x">https://www.nature.com/articles/s41562-018-0311-x</a>.</p>
</dd>
<dt class="label" id="id241"><span class="brackets">127</span></dt>
<dd><p>Daniel J Benjamin, James O Berger, Magnus Johannesson, Brian A Nosek, E-J Wagenmakers, Richard Berk, Kenneth A Bollen, Björn Brembs, Lawrence Brown, Colin Camerer, David Cesarini, Christopher D Chambers, Merlise Clyde, Thomas D Cook, Paul De Boeck, Zoltan Dienes, Anna Dreber, Kenny Easwaran, Charles Efferson, Ernst Fehr, Fiona Fidler, Andy P Field, Malcolm Forster, Edward I George, Richard Gonzalez, Steven Goodman, Edwin Green, Donald P Green, Anthony G Greenwald, Jarrod D Hadfield, Larry V Hedges, Leonhard Held, Teck Hua Ho, Herbert Hoijtink, Daniel J Hruschka, Kosuke Imai, Guido Imbens, John P A Ioannidis, Minjeong Jeon, James Holland Jones, Michael Kirchler, David Laibson, John List, Roderick Little, Arthur Lupia, Edouard Machery, Scott E Maxwell, Michael McCarthy, Don A Moore, Stephen L Morgan, Marcus Munafó, Shinichi Nakagawa, Brendan Nyhan, Timothy H Parker, Luis Pericchi, Marco Perugini, Jeff Rouder, Judith Rousseau, Victoria Savalei, Felix D Schönbrodt, Thomas Sellke, Betsy Sinclair, Dustin Tingley, Trisha Van Zandt, Simine Vazire, Duncan J Watts, Christopher Winship, Robert L Wolpert, Yu Xie, Cristobal Young, Jonathan Zinman, and Valen E Johnson. Redefine statistical significance. <em>Nat Hum Behav</em>, 2(1):6–10, January 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41562-017-0189-z">http://dx.doi.org/10.1038/s41562-017-0189-z</a>.</p>
</dd>
<dt class="label" id="id233"><span class="brackets">128</span></dt>
<dd><p>Norbert L Kerr. HARKing: hypothesizing after the results are known. 1998. URL: <a class="reference external" href="http://dx.doi.org/10.1207/s15327957pspr0203_4">http://dx.doi.org/10.1207/s15327957pspr0203_4</a>.</p>
</dd>
<dt class="label" id="id63"><span class="brackets">129</span></dt>
<dd><p>Megan L Head, Luke Holman, Rob Lanfear, Andrew T Kahn, and Michael D Jennions. The extent and consequences of p-hacking in science. <em>PLoS Biol.</em>, 13(3):e1002106, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1002106">http://dx.doi.org/10.1371/journal.pbio.1002106</a>.</p>
</dd>
<dt class="label" id="id64"><span class="brackets">130</span></dt>
<dd><p>R A Fisher and F Yates. <em>Statistical Methods, Experimental Design, and Scientific Inference: A Re-Issue of Statistical Methods for Research Workers, the Design of Experiments, and Statistical Methods and Scientific Inference</em>. OUP Oxford, April 1990. URL: <a class="reference external" href="https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290">https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290</a>.</p>
</dd>
<dt class="label" id="id65"><span class="brackets">131</span></dt>
<dd><p>Olive Jean Dunn. Estimation of the means of dependent variables. <em>The Annals of Mathematical Statistics</em>, 29(4):1095–1111, 1958. URL: <a class="reference external" href="http://dx.doi.org/10.1214/aoms/1177706443">http://dx.doi.org/10.1214/aoms/1177706443</a>.</p>
</dd>
<dt class="label" id="id66"><span class="brackets">132</span></dt>
<dd><p>Yoav Benjamini and Yosef Hochberg. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal Statistical Society: Series B (Methodological)</em>, 57(1):289–300, 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x">http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x</a>.</p>
</dd>
<dt class="label" id="id238"><span class="brackets">133</span></dt>
<dd><p>R Silberzahn, E L Uhlmann, D P Martin, P Anselmi, F Aust, E Awtrey, Š Bahník, F Bai, C Bannard, E Bonnier, R Carlsson, F Cheung, G Christensen, R Clay, M A Craig, A Dalla Rosa, L Dam, M H Evans, I Flores Cervantes, N Fong, M Gamez-Djokic, A Glenz, S Gordon-McKeon, T J Heaton, K Hederos, M Heene, A J Hofelich Mohr, F Högden, K Hui, M Johannesson, J Kalodimos, E Kaszubowski, D M Kennedy, R Lei, T A Lindsay, S Liverani, C R Madan, D Molden, E Molleman, R D Morey, L B Mulder, B R Nijstad, N G Pope, B Pope, J M Prenoveau, F Rink, E Robusto, H Roderique, A Sandberg, E Schlüter, F D Schönbrodt, M F Sherman, S A Sommer, K Sotak, S Spain, C Spörlein, T Stafford, L Stefanutti, S Tauber, J Ullrich, M Vianello, E-J Wagenmakers, M Witkowiak, S Yoon, and B A Nosek. Many analysts, one data set: making transparent how variations in analytic choices affect results. <em>Advances in Methods and Practices in Psychological Science</em>, 1(3):337–356, September 2018. URL: <a class="reference external" href="https://doi.org/10.1177/2515245917747646">https://doi.org/10.1177/2515245917747646</a>.</p>
</dd>
<dt class="label" id="id239"><span class="brackets">134</span></dt>
<dd><p>Rotem Botvinik-Nezer, Felix Holzmeister, Colin F Camerer, Anna Dreber, Juergen Huber, Magnus Johannesson, Michael Kirchler, Roni Iwanir, Jeanette A Mumford, R Alison Adcock, Paolo Avesani, Blazej M Baczkowski, Aahana Bajracharya, Leah Bakst, Sheryl Ball, Marco Barilari, Nadège Bault, Derek Beaton, Julia Beitner, Roland G Benoit, Ruud M W J Berkers, Jamil P Bhanji, Bharat B Biswal, Sebastian Bobadilla-Suarez, Tiago Bortolini, Katherine L Bottenhorn, Alexander Bowring, Senne Braem, Hayley R Brooks, Emily G Brudner, Cristian B Calderon, Julia A Camilleri, Jaime J Castrellon, Luca Cecchetti, Edna C Cieslik, Zachary J Cole, Olivier Collignon, Robert W Cox, William A Cunningham, Stefan Czoschke, Kamalaker Dadi, Charles P Davis, Alberto De Luca, Mauricio R Delgado, Lysia Demetriou, Jeffrey B Dennison, Xin Di, Erin W Dickie, Ekaterina Dobryakova, Claire L Donnat, Juergen Dukart, Niall W Duncan, Joke Durnez, Amr Eed, Simon B Eickhoff, Andrew Erhart, Laura Fontanesi, G Matthew Fricke, Shiguang Fu, Adriana Galván, Remi Gau, Sarah Genon, Tristan Glatard, Enrico Glerean, Jelle J Goeman, Sergej A E Golowin, Carlos González-García, Krzysztof J Gorgolewski, Cheryl L Grady, Mikella A Green, João F Guassi Moreira, Olivia Guest, Shabnam Hakimi, J Paul Hamilton, Roeland Hancock, Giacomo Handjaras, Bronson B Harry, Colin Hawco, Peer Herholz, Gabrielle Herman, Stephan Heunis, Felix Hoffstaedter, Jeremy Hogeveen, Susan Holmes, Chuan-Peng Hu, Scott A Huettel, Matthew E Hughes, Vittorio Iacovella, Alexandru D Iordan, Peder M Isager, Ayse I Isik, Andrew Jahn, Matthew R Johnson, Tom Johnstone, Michael J E Joseph, Anthony C Juliano, Joseph W Kable, Michalis Kassinopoulos, Cemal Koba, Xiang-Zhen Kong, Timothy R Koscik, Nuri Erkut Kucukboyaci, Brice A Kuhl, Sebastian Kupek, Angela R Laird, Claus Lamm, Robert Langner, Nina Lauharatanahirun, Hongmi Lee, Sangil Lee, Alexander Leemans, Andrea Leo, Elise Lesage, Flora Li, Monica Y C Li, Phui Cheng Lim, Evan N Lintz, Schuyler W Liphardt, Annabel B Losecaat Vermeer, Bradley C Love, Michael L Mack, Norberto Malpica, Theo Marins, Camille Maumet, Kelsey McDonald, Joseph T McGuire, Helena Melero, Adriana S Méndez Leal, Benjamin Meyer, Kristin N Meyer, Glad Mihai, Georgios D Mitsis, Jorge Moll, Dylan M Nielson, Gustav Nilsonne, Michael P Notter, Emanuele Olivetti, Adrian I Onicas, Paolo Papale, Kaustubh R Patil, Jonathan E Peelle, Alexandre Pérez, Doris Pischedda, Jean-Baptiste Poline, Yanina Prystauka, Shruti Ray, Patricia A Reuter-Lorenz, Richard C Reynolds, Emiliano Ricciardi, Jenny R Rieck, Anais M Rodriguez-Thompson, Anthony Romyn, Taylor Salo, Gregory R Samanez-Larkin, Emilio Sanz-Morales, Margaret L Schlichting, Douglas H Schultz, Qiang Shen, Margaret A Sheridan, Jennifer A Silvers, Kenny Skagerlund, Alec Smith, David V Smith, Peter Sokol-Hessner, Simon R Steinkamp, Sarah M Tashjian, Bertrand Thirion, John N Thorp, Gustav Tinghög, Loreen Tisdall, Steven H Tompson, Claudio Toro-Serey, Juan Jesus Torre Tresols, Leonardo Tozzi, Vuong Truong, Luca Turella, Anna E van 't Veer, Tom Verguts, Jean M Vettel, Sagana Vijayarajah, Khoi Vo, Matthew B Wall, Wouter D Weeda, Susanne Weis, David J White, David Wisniewski, Alba Xifra-Porxas, Emily A Yearling, Sangsuk Yoon, Rui Yuan, Kenneth S L Yuen, Lei Zhang, Xu Zhang, Joshua E Zosky, Thomas E Nichols, Russell A Poldrack, and Tom Schonberg. Variability in the analysis of a single neuroimaging dataset by many teams. <em>Nature</em>, 582(7810):84–88, June 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41586-020-2314-9">http://dx.doi.org/10.1038/s41586-020-2314-9</a>.</p>
</dd>
<dt class="label" id="id235"><span class="brackets">135</span></dt>
<dd><p>Lauren Cadwallader, Jason A Papin, Feilim Mac Gabhann, and Rebecca Kirk. Collaborating with our community to increase code sharing. <em>PLoS Comput. Biol.</em>, 17(3):e1008867, March 2021. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pcbi.1008867">http://dx.doi.org/10.1371/journal.pcbi.1008867</a>.</p>
</dd>
<dt class="label" id="id237"><span class="brackets">136</span></dt>
<dd><p>Mark D Wilkinson, Michel Dumontier, I Jsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, Jan-Willem Boiten, Luiz Bonino da Silva Santos, Philip E Bourne, Jildau Bouwman, Anthony J Brookes, Tim Clark, Mercè Crosas, Ingrid Dillo, Olivier Dumon, Scott Edmunds, Chris T Evelo, Richard Finkers, Alejandra Gonzalez-Beltran, Alasdair J G Gray, Paul Groth, Carole Goble, Jeffrey S Grethe, Jaap Heringa, Peter A C 't Hoen, Rob Hooft, Tobias Kuhn, Ruben Kok, Joost Kok, Scott J Lusher, Maryann E Martone, Albert Mons, Abel L Packer, Bengt Persson, Philippe Rocca-Serra, Marco Roos, Rene van Schaik, Susanna-Assunta Sansone, Erik Schultes, Thierry Sengstag, Ted Slater, George Strawn, Morris A Swertz, Mark Thompson, Johan van der Lei, Erik van Mulligen, Jan Velterop, Andra Waagmeester, Peter Wittenburg, Katherine Wolstencroft, Jun Zhao, and Barend Mons. The FAIR guiding principles for scientific data management and stewardship. <em>Sci Data</em>, 3:160018, March 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/sdata.2016.18">http://dx.doi.org/10.1038/sdata.2016.18</a>.</p>
</dd>
<dt class="label" id="id236"><span class="brackets">137</span></dt>
<dd><p>Anna-Lena Lamprecht, Leyla Garcia, Mateusz Kuzak, Carlos Martinez, Ricardo Arcila, Eva Martin Del Pico, Victoria Dominguez Del Angel, Stephanie Van De Sandt, Jon Ison, Paula Andrea Martinez, and Others. Towards FAIR principles for research software. <em>Data Science</em>, 3(1):37–59, 2020. URL: <a class="reference external" href="https://content.iospress.com/articles/data-science/ds190026">https://content.iospress.com/articles/data-science/ds190026</a>.</p>
</dd>
<dt class="label" id="id184"><span class="brackets">138</span></dt>
<dd><p>SLOW-SCIENCE.org — bear with us, while we think. <span><a class="reference external" href="#"></a></span>http://slow-science.org/. Accessed: 2021-2-14. URL: <a class="reference external" href="http://slow-science.org/">http://slow-science.org/</a>.</p>
</dd>
<dt class="label" id="id27"><span class="brackets">139</span></dt>
<dd><p>Hai Fang, Matt E Oates, Ralph B Pethica, Jenny M Greenwood, Adam J Sardar, Owen J L Rackham, Philip C J Donoghue, Alexandros Stamatakis, David A de Lima Morais, and Julian Gough. A daily-updated tree of (sequenced) life as a reference for genome research. <em>Sci. Rep.</em>, 3:2015, 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/srep02015">http://dx.doi.org/10.1038/srep02015</a>.</p>
</dd>
<dt class="label" id="id29"><span class="brackets">140</span></dt>
<dd><p>Elias Dohmen, Lukas P M Kremer, Erich Bornberg-Bauer, and Carsten Kemena. DOGMA: domain-based transcriptome and proteome quality assessment. <em>Bioinformatics</em>, 32(17):2577–2581, September 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btw231">http://dx.doi.org/10.1093/bioinformatics/btw231</a>.</p>
</dd>
<dt class="label" id="id28"><span class="brackets">141</span></dt>
<dd><p>Joel Cracraft and Michael J Donoghue. <em>Assembling the Tree of Life</em>. Oxford University Press, July 2004. URL: <a class="reference external" href="https://www.amazon.co.uk/Assembling-Tree-Life-Joel-Cracraft/dp/0195172345">https://www.amazon.co.uk/Assembling-Tree-Life-Joel-Cracraft/dp/0195172345</a>.</p>
</dd>
<dt class="label" id="id245"><span class="brackets">142</span></dt>
<dd><p>Iupac-Iub Comm on Biochem Nomencl and Iupac-Iub Comm on. A one-letter notation for amino acid sequences. tentative rules. 1968. URL: <a class="reference external" href="http://dx.doi.org/10.1021/bi00848a001">http://dx.doi.org/10.1021/bi00848a001</a>.</p>
</dd>
<dt class="label" id="id246"><span class="brackets">143</span></dt>
<dd><p>NCBI Resource Coordinators and NCBI Resource Coordinators. Database resources of the national center for biotechnology information. 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkw1071">http://dx.doi.org/10.1093/nar/gkw1071</a>.</p>
</dd>
<dt class="label" id="id243"><span class="brackets">144</span></dt>
<dd><p>G Parra, K Bradnam, and I Korf. CEGMA: a pipeline to accurately annotate core genes in eukaryotic genomes. 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btm071">http://dx.doi.org/10.1093/bioinformatics/btm071</a>.</p>
</dd>
<dt class="label" id="id244"><span class="brackets">145</span></dt>
<dd><p>Eugene V Koonin, Natalie D Fedorova, John D Jackson, Aviva R Jacobs, Dmitri M Krylov, Kira S Makarova, Raja Mazumder, Sergei L Mekhedov, Anastasia N Nikolskaya, B Sridhar Rao, Igor B Rogozin, Sergei Smirnov, Alexander V Sorokin, Alexander V Sverdlov, Sona Vasudevan, Yuri I Wolf, Jodie J Yin, and Darren A Natale. A comprehensive evolutionary classification of proteins encoded in complete eukaryotic genomes. <em>Genome Biol.</em>, 5(2):R7, January 2004. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gb-2004-5-2-r7">http://dx.doi.org/10.1186/gb-2004-5-2-r7</a>.</p>
</dd>
<dt class="label" id="id30"><span class="brackets">146</span></dt>
<dd><p>Felipe A Simão, Robert M Waterhouse, Panagiotis Ioannidis, Evgenia V Kriventseva, and Evgeny M Zdobnov. BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs. <em>Bioinformatics</em>, 31(19):3210–3212, October 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btv351">http://dx.doi.org/10.1093/bioinformatics/btv351</a>.</p>
</dd>
<dt class="label" id="id242"><span class="brackets">147</span></dt>
<dd><p>Roman L Tatusov, Natalie D Fedorova, John D Jackson, Aviva R Jacobs, Boris Kiryutin, Eugene V Koonin, Dmitri M Krylov, Raja Mazumder, Sergei L Mekhedov, Anastasia N Nikolskaya, B Sridhar Rao, Sergei Smirnov, Alexander V Sverdlov, Sona Vasudevan, Yuri I Wolf, Jodie J Yin, and Darren A Natale. The COG database: an updated version includes eukaryotes. <em>BMC Bioinformatics</em>, 4:41, September 2003. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-4-41">http://dx.doi.org/10.1186/1471-2105-4-41</a>.</p>
</dd>
<dt class="label" id="id247"><span class="brackets">148</span></dt>
<dd><p>Evgenia V Kriventseva, Dmitry Kuznetsov, Fredrik Tegenfeldt, Mosè Manni, Renata Dias, Felipe A Simão, and Evgeny M Zdobnov. OrthoDB v10: sampling the diversity of animal, plant, fungal, protist, bacterial and viral genomes for evolutionary and functional annotations of orthologs. <em>Nucleic Acids Res.</em>, 47(D1):D807–D811, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gky1053">http://dx.doi.org/10.1093/nar/gky1053</a>.</p>
</dd>
<dt class="label" id="id31"><span class="brackets">149</span></dt>
<dd><p>Paul A Kitts, Deanna M Church, Françoise Thibaud-Nissen, Jinna Choi, Vichet Hem, Victor Sapojnikov, Robert G Smith, Tatiana Tatusova, Charlie Xiang, Andrey Zherikov, Michael DiCuccio, Terence D Murphy, Kim D Pruitt, and Avi Kimchi. Assembly: a resource for assembled genomes at NCBI. <em>Nucleic Acids Res.</em>, 44(D1):D73–80, January 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv1226">http://dx.doi.org/10.1093/nar/gkv1226</a>.</p>
</dd>
<dt class="label" id="id32"><span class="brackets">150</span></dt>
<dd><p>David Sims, Ian Sudbery, Nicholas E Ilott, Andreas Heger, and Chris P Ponting. Sequencing depth and coverage: key considerations in genomic analyses. <em>Nat. Rev. Genet.</em>, 15(2):121–132, February 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3642">http://dx.doi.org/10.1038/nrg3642</a>.</p>
</dd>
<dt class="label" id="id33"><span class="brackets">151</span></dt>
<dd><p>Mick Watson and Amanda Warr. Errors in long-read assemblies can critically affect protein prediction. <em>Nat. Biotechnol.</em>, 37(2):124–126, February 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41587-018-0004-z">http://dx.doi.org/10.1038/s41587-018-0004-z</a>.</p>
</dd>
<dt class="label" id="id73"><span class="brackets">152</span></dt>
<dd><p>Yehudit Hasin, Marcus Seldin, and Aldons Lusis. Multi-omics approaches to disease. <em>Genome Biol.</em>, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-017-1215-1">http://dx.doi.org/10.1186/s13059-017-1215-1</a>.</p>
</dd>
<dt class="label" id="id74"><span class="brackets">153</span></dt>
<dd><p>Marylyn D Ritchie, Emily R Holzinger, Ruowang Li, Sarah A Pendergrass, and Dokyoon Kim. Methods of integrating data to uncover genotype-phenotype interactions. <em>Nat. Rev. Genet.</em>, 16(2):85–97, February 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3868">http://dx.doi.org/10.1038/nrg3868</a>.</p>
</dd>
<dt class="label" id="id75"><span class="brackets">154</span></dt>
<dd><p>Vessela N Kristensen, Ole Christian Lingjærde, Hege G Russnes, Hans Kristian M Vollan, Arnoldo Frigessi, and Anne-Lise Børresen-Dale. Principles and methods of integrative genomic analyses in cancer. <em>Nat. Rev. Cancer</em>, 14(5):299–313, May 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrc3721">http://dx.doi.org/10.1038/nrc3721</a>.</p>
</dd>
<dt class="label" id="id252"><span class="brackets">155</span></dt>
<dd><p>Asif Javed, Saloni Agrawal, and Pauline C Ng. Phen-Gen: combining phenotype and genotype to analyze rare disorders. <em>Nat. Methods</em>, 11(9):935–937, September 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.3046">http://dx.doi.org/10.1038/nmeth.3046</a>.</p>
</dd>
<dt class="label" id="id257"><span class="brackets">156</span></dt>
<dd><p>Damian Smedley, Anika Oellrich, Sebastian Köhler, Barbara Ruef, Sanger Mouse Genetics Project, Monte Westerfield, Peter Robinson, Suzanna Lewis, and Christopher Mungall. PhenoDigm: analyzing curated annotations to associate animal models with human diseases. <em>Database</em>, 2013:bat025, May 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1093/database/bat025">http://dx.doi.org/10.1093/database/bat025</a>.</p>
</dd>
<dt class="label" id="id254"><span class="brackets">157</span></dt>
<dd><p>Peter N Robinson, Sebastian Köhler, Anika Oellrich, Sanger Mouse Genetics Project, Kai Wang, Christopher J Mungall, Suzanna E Lewis, Nicole Washington, Sebastian Bauer, Dominik Seelow, Peter Krawitz, Christian Gilissen, Melissa Haendel, and Damian Smedley. Improved exome prioritization of disease genes through cross-species phenotype comparison. <em>Genome Res.</em>, 24(2):340–348, February 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.160325.113">http://dx.doi.org/10.1101/gr.160325.113</a>.</p>
</dd>
<dt class="label" id="id256"><span class="brackets">158</span></dt>
<dd><p>Martin Kircher, Daniela M Witten, Preti Jain, Brian J O'Roak, Gregory M Cooper, and Jay Shendure. A general framework for estimating the relative pathogenicity of human genetic variants. <em>Nat. Genet.</em>, 46(3):310–315, March 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ng.2892">http://dx.doi.org/10.1038/ng.2892</a>.</p>
</dd>
<dt class="label" id="id255"><span class="brackets">159</span></dt>
<dd><p>Damian Smedley, Julius O B Jacobsen, Marten Jäger, Sebastian Köhler, Manuel Holtgrewe, Max Schubach, Enrico Siragusa, Tomasz Zemojtel, Orion J Buske, Nicole L Washington, William P Bone, Melissa A Haendel, and Peter N Robinson. Next-generation diagnostics and disease-gene discovery with the exomiser. <em>Nat. Protoc.</em>, 10(12):2004–2015, November 2015. URL: <a class="reference external" href="https://www.nature.com/articles/nprot.2015.124">https://www.nature.com/articles/nprot.2015.124</a>.</p>
</dd>
<dt class="label" id="id253"><span class="brackets">160</span></dt>
<dd><p>Hui Yang, Peter N Robinson, and Kai Wang. Phenolyzer: phenotype-based prioritization of candidate genes for human diseases. <em>Nat. Methods</em>, 12(9):841–843, September 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.3484">http://dx.doi.org/10.1038/nmeth.3484</a>.</p>
</dd>
<dt class="label" id="id251"><span class="brackets">161</span></dt>
<dd><p>Damian Smedley, Max Schubach, Julius O B Jacobsen, Sebastian Köhler, Tomasz Zemojtel, Malte Spielmann, Marten Jäger, Harry Hochheiser, Nicole L Washington, Julie A McMurry, Melissa A Haendel, Christopher J Mungall, Suzanna E Lewis, Tudor Groza, Giorgio Valentini, and Peter N Robinson. A Whole-Genome analysis framework for effective identification of pathogenic regulatory variants in mendelian disease. <em>Am. J. Hum. Genet.</em>, 99(3):595–606, September 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.ajhg.2016.07.005">http://dx.doi.org/10.1016/j.ajhg.2016.07.005</a>.</p>
</dd>
<dt class="label" id="id250"><span class="brackets">162</span></dt>
<dd><p>Tomasz Zemojtel, Sebastian Köhler, Luisa Mackenroth, Marten Jäger, Jochen Hecht, Peter Krawitz, Luitgard Graul-Neumann, Sandra Doelken, Nadja Ehmke, Malte Spielmann, Nancy Christine Oien, Michal R Schweiger, Ulrike Krüger, Götz Frommer, Björn Fischer, Uwe Kornak, Ricarda Flöttmann, Amin Ardeshirdavani, Yves Moreau, Suzanna E Lewis, Melissa Haendel, Damian Smedley, Denise Horn, Stefan Mundlos, and Peter N Robinson. Effective diagnosis of genetic disease by computational phenotype analysis of the disease-associated genome. <em>Sci. Transl. Med.</em>, 6(252):252ra123, September 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1126/scitranslmed.3009262">http://dx.doi.org/10.1126/scitranslmed.3009262</a>.</p>
</dd>
<dt class="label" id="id249"><span class="brackets">163</span></dt>
<dd><p>Sebastian Köhler, Marcel H Schulz, Peter Krawitz, Sebastian Bauer, Sandra Dölken, Claus E Ott, Christine Mundlos, Denise Horn, Stefan Mundlos, and Peter N Robinson. Clinical diagnostics in human genetics with semantic similarity searches in ontologies. <em>Am. J. Hum. Genet.</em>, 85(4):457–464, October 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.ajhg.2009.09.003">http://dx.doi.org/10.1016/j.ajhg.2009.09.003</a>.</p>
</dd>
<dt class="label" id="id268"><span class="brackets">164</span></dt>
<dd><p>Peristera Paschou, Elad Ziv, Esteban G Burchard, Shweta Choudhry, William Rodriguez-Cintron, Michael W Mahoney, and Petros Drineas. PCA-correlated SNPs for structure identification in worldwide human populations. <em>PLoS Genet.</em>, 3(9):1672–1686, September 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pgen.0030160">http://dx.doi.org/10.1371/journal.pgen.0030160</a>.</p>
</dd>
<dt class="label" id="id267"><span class="brackets">165</span></dt>
<dd><p>M B Eisen, P T Spellman, P O Brown, and D Botstein. Cluster analysis and display of genome-wide expression patterns. <em>Proc. Natl. Acad. Sci. U. S. A.</em>, 95(25):14863–14868, December 1998. URL: <a class="reference external" href="http://dx.doi.org/10.1073/pnas.95.25.14863">http://dx.doi.org/10.1073/pnas.95.25.14863</a>.</p>
</dd>
<dt class="label" id="id270"><span class="brackets">166</span></dt>
<dd><p>Habtom W Ressom, Rency S Varghese, Zhen Zhang, Jianhua Xuan, and Robert Clarke. Classification algorithms for phenotype prediction in genomics and proteomics. <em>Front. Biosci.</em>, 13:691–708, January 2008. URL: <a class="reference external" href="http://dx.doi.org/10.2741/2712">http://dx.doi.org/10.2741/2712</a>.</p>
</dd>
<dt class="label" id="id269"><span class="brackets">167</span></dt>
<dd><p>Michael H Cho, George R Washko, Thomas J Hoffmann, Gerard J Criner, Eric A Hoffman, Fernando J Martinez, Nan Laird, John J Reilly, and Edwin K Silverman. Cluster analysis in severe emphysema subjects using phenotype and genotype data: an exploratory investigation. <em>Respir. Res.</em>, 11:30, March 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1465-9921-11-30">http://dx.doi.org/10.1186/1465-9921-11-30</a>.</p>
</dd>
<dt class="label" id="id124"><span class="brackets">168</span></dt>
<dd><p>Richard Bellman. <em>Dynamic Programming</em>. Princeton University Press, 1957. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=wdtoPwAACAAJ">https://play.google.com/store/books/details?id=wdtoPwAACAAJ</a>.</p>
</dd>
<dt class="label" id="id125"><span class="brackets">169</span></dt>
<dd><p>A Zimek, E Schubert, and H P Kriegel. A survey on unsupervised outlier detection in high‐dimensional numerical data. <em>Stat. Anal. Data Min.</em>, 2012. URL: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/abs/10.1002/sam.11161">https://onlinelibrary.wiley.com/doi/abs/10.1002/sam.11161</a>.</p>
</dd>
<dt class="label" id="id118"><span class="brackets">170</span></dt>
<dd><p>The 1000 Genomes Project Consortium and The 1000 Genomes Project Consortium. A global reference for human genetic variation. <em>Nature</em>, 526(7571):68–74, 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nature15393">http://dx.doi.org/10.1038/nature15393</a>.</p>
</dd>
<dt class="label" id="id119"><span class="brackets">171</span></dt>
<dd><p>Cynthia L Smith, Carroll-Ann W Goldsmith, and Janan T Eppig. The mammalian phenotype ontology as a tool for annotating, analyzing and comparing phenotypic information. <em>Genome Biol.</em>, 6(1):R7, 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gb-2004-6-1-r7">http://dx.doi.org/10.1186/gb-2004-6-1-r7</a>.</p>
</dd>
<dt class="label" id="id120"><span class="brackets">172</span></dt>
<dd><p>P N Robinson and S Mundlos. The human phenotype ontology. <em>Clin. Genet.</em>, 77(6):525–534, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1111/j.1399-0004.2010.01436.x">http://dx.doi.org/10.1111/j.1399-0004.2010.01436.x</a>.</p>
</dd>
<dt class="label" id="id121"><span class="brackets">173</span></dt>
<dd><p>Lynn Marie Schriml, Cesar Arze, Suvarna Nadendla, Yu-Wei Wayne Chang, Mark Mazaitis, Victor Felix, Gang Feng, and Warren Alden Kibbe. Disease ontology: a backbone for disease semantic integration. <em>Nucleic Acids Res.</em>, 40(Database issue):D940–6, January 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkr972">http://dx.doi.org/10.1093/nar/gkr972</a>.</p>
</dd>
<dt class="label" id="id122"><span class="brackets">174</span></dt>
<dd><p>Alex Bateman, Ewan Birney, Lorenzo Cerruti, Richard Durbin, Laurence Etwiller, Sean R Eddy, Sam Griffiths-Jones, Kevin L Howe, Mhairi Marshall, and Erik L L Sonnhammer. The pfam protein families database. <em>Nucleic Acids Res.</em>, 30(1):276–280, January 2002. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/30.1.276">http://dx.doi.org/10.1093/nar/30.1.276</a>.</p>
</dd>
<dt class="label" id="id123"><span class="brackets">175</span></dt>
<dd><p>Jan Poland and Thomas Zeugmann. Clustering pairwise distances with missing data: maximum cuts versus normalized cuts. <em>Discovery Science</em>, pages 197–208, 2006. URL: <a class="reference external" href="http://dx.doi.org/10.1007/11893318_21">http://dx.doi.org/10.1007/11893318_21</a>.</p>
</dd>
<dt class="label" id="id226"><span class="brackets">176</span></dt>
<dd><p>Barry Smith, Werner Ceusters, Bert Klagges, Jacob Köhler, Anand Kumar, Jane Lomax, Chris Mungall, Fabian Neuhaus, Alan L Rector, and Cornelius Rosse. Relations in biomedical ontologies. <em>Genome Biol.</em>, 6(5):R46, April 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gb-2005-6-5-r46">http://dx.doi.org/10.1186/gb-2005-6-5-r46</a>.</p>
</dd>
<dt class="label" id="id231"><span class="brackets">177</span></dt>
<dd><p>H J Lowe and G O Barnett. Understanding and using the medical subject headings (MeSH) vocabulary to perform literature searches. <em>JAMA</em>, 271(14):1103–1108, April 1994. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/8151853">https://www.ncbi.nlm.nih.gov/pubmed/8151853</a>.</p>
</dd>
<dt class="label" id="id137"><span class="brackets">178</span></dt>
<dd><p>1000 Genomes Project Consortium, Goncalo R Abecasis, Adam Auton, Lisa D Brooks, Mark A DePristo, Richard M Durbin, Robert E Handsaker, Hyun Min Kang, Gabor T Marth, and Gil A McVean. An integrated map of genetic variation from 1,092 human genomes. <em>Nature</em>, 491(7422):56–65, November 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nature11632">http://dx.doi.org/10.1038/nature11632</a>.</p>
</dd>
<dt class="label" id="id138"><span class="brackets">179</span></dt>
<dd><p>Susan Fairley, Ernesto Lowy-Gallego, Emily Perry, and Paul Flicek. The international genome sample resource (IGSR) collection of open human genomic variation resources. <em>Nucleic Acids Res.</em>, 48(D1):D941–D947, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkz836">http://dx.doi.org/10.1093/nar/gkz836</a>.</p>
</dd>
<dt class="label" id="id265"><span class="brackets">180</span></dt>
<dd><p>Heng Li. Tabix: fast retrieval of sequence features from generic TAB-delimited files. <em>Bioinformatics</em>, 27(5):718–719, March 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btq671">http://dx.doi.org/10.1093/bioinformatics/btq671</a>.</p>
</dd>
<dt class="label" id="id84"><span class="brackets">181</span></dt>
<dd><p>Cath Tyner. The UCSC genome browser coordinate counting systems. <span><a class="reference external" href="#"></a></span>http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/, December 2016. Accessed: 2021-2-7. URL: <a class="reference external" href="http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/">http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/</a>.</p>
</dd>
<dt class="label" id="id128"><span class="brackets">182</span></dt>
<dd><p>G M Church. The personal genome project. <em>Mol. Syst. Biol.</em>, 1:2005.0030, December 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1038/msb4100040">http://dx.doi.org/10.1038/msb4100040</a>.</p>
</dd>
<dt class="label" id="id129"><span class="brackets">183</span></dt>
<dd><p>Philipp G Sand. A lesson not learned: allele misassignment. <em>Behav. Brain Funct.</em>, 3(1):65, 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1744-9081-3-65">http://dx.doi.org/10.1186/1744-9081-3-65</a>.</p>
</dd>
<dt class="label" id="id127"><span class="brackets">184</span></dt>
<dd><p>2.3. clustering — scikit-learn 0.21.2 documentation. <span><a class="reference external" href="#"></a></span>https://scikit-learn.org/stable/modules/clustering.html. Accessed: 2019-6-9. URL: <a class="reference external" href="https://scikit-learn.org/stable/modules/clustering.html">https://scikit-learn.org/stable/modules/clustering.html</a>.</p>
</dd>
<dt class="label" id="id126"><span class="brackets">185</span></dt>
<dd><p>Lucien Marie Le Cam and Jerzy Neyman. <em>Proceedings of the Fifth Berkeley Symposium on Mathematical Statistics and Probability</em>. University of California Press, 1967. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=EN6isXsFdKgC">https://play.google.com/store/books/details?id=EN6isXsFdKgC</a>.</p>
</dd>
<dt class="label" id="id134"><span class="brackets">186</span></dt>
<dd><p>Chenguang Zhao and Zheng Wang. GOGO: an improved algorithm to measure the semantic similarity between gene ontology terms. <em>Sci. Rep.</em>, 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41598-018-33219-y">http://dx.doi.org/10.1038/s41598-018-33219-y</a>.</p>
</dd>
<dt class="label" id="id135"><span class="brackets">187</span></dt>
<dd><p>James Z Wang, Zhidian Du, Rapeeporn Payattakool, Philip S Yu, and Chin-Fu Chen. A new method to measure the semantic similarity of GO terms. <em>Bioinformatics</em>, 23(10):1274–1281, May 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btm087">http://dx.doi.org/10.1093/bioinformatics/btm087</a>.</p>
</dd>
<dt class="label" id="id136"><span class="brackets">188</span></dt>
<dd><p>Mark F Rogers, Hashem A Shihab, Matthew Mort, David N Cooper, Tom R Gaunt, and Colin Campbell. FATHMM-XF: accurate prediction of pathogenic point mutations via extended features. <em>Bioinformatics</em>, 34(3):511–513, February 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btx536">http://dx.doi.org/10.1093/bioinformatics/btx536</a>.</p>
</dd>
<dt class="label" id="id266"><span class="brackets">189</span></dt>
<dd><p>Alice B Popejoy and Stephanie M Fullerton. Genomics is failing on diversity. <em>Nature</em>, 538(7624):161–164, October 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/538161a">http://dx.doi.org/10.1038/538161a</a>.</p>
</dd>
<dt class="label" id="id264"><span class="brackets">190</span></dt>
<dd><p>Momoko Horikoshi, Hanieh Yaghootkar, Dennis O Mook-Kanamori, Ulla Sovio, H Rob Taal, Branwen J Hennig, Jonathan P Bradfield, Beate St Pourcain, David M Evans, Pimphen Charoen, Marika Kaakinen, Diana L Cousminer, Terho Lehtimäki, Eskil Kreiner-Møller, Nicole M Warrington, Mariona Bustamante, Bjarke Feenstra, Diane J Berry, Elisabeth Thiering, Thiemo Pfab, Sheila J Barton, Beverley M Shields, Marjan Kerkhof, Elisabeth M van Leeuwen, Anthony J Fulford, Zoltán Kutalik, Jing Hua Zhao, Marcel den Hoed, Anubha Mahajan, Virpi Lindi, Liang-Kee Goh, Jouke-Jan Hottenga, Ying Wu, Olli T Raitakari, Marie N Harder, Aline Meirhaeghe, Ioanna Ntalla, Rany M Salem, Karen A Jameson, Kaixin Zhou, Dorota M Monies, Vasiliki Lagou, Mirna Kirin, Jani Heikkinen, Linda S Adair, Fowzan S Alkuraya, Ali Al-Odaib, Philippe Amouyel, Ehm Astrid Andersson, Amanda J Bennett, Alexandra I F Blakemore, Jessica L Buxton, Jean Dallongeville, Shikta Das, Eco J C de Geus, Xavier Estivill, Claudia Flexeder, Philippe Froguel, Frank Geller, Keith M Godfrey, Frédéric Gottrand, Christopher J Groves, Torben Hansen, Joel N Hirschhorn, Albert Hofman, Mads V Hollegaard, David M Hougaard, Elina Hyppönen, Hazel M Inskip, Aaron Isaacs, Torben Jørgensen, Christina Kanaka-Gantenbein, John P Kemp, Wieland Kiess, Tuomas O Kilpeläinen, Norman Klopp, Bridget A Knight, Christopher W Kuzawa, George McMahon, John P Newnham, Harri Niinikoski, Ben A Oostra, Louise Pedersen, Dirkje S Postma, Susan M Ring, Fernando Rivadeneira, Neil R Robertson, Sylvain Sebert, Olli Simell, Torsten Slowinski, Carla M T Tiesler, Anke Tönjes, Allan Vaag, Jorma S Viikari, Jacqueline M Vink, Nadja Hawwa Vissing, Nicholas J Wareham, Gonneke Willemsen, Daniel R Witte, Haitao Zhang, Jianhua Zhao, Meta-Analyses of Glucose- and Insulin-related traits Consortium (MAGIC), James F Wilson, Michael Stumvoll, Andrew M Prentice, Brian F Meyer, Ewan R Pearson, Colin A G Boreham, Cyrus Cooper, Matthew W Gillman, George V Dedoussis, Luis A Moreno, Oluf Pedersen, Maiju Saarinen, Karen L Mohlke, Dorret I Boomsma, Seang-Mei Saw, Timo A Lakka, Antje Körner, Ruth J F Loos, Ken K Ong, Peter Vollenweider, Cornelia M van Duijn, Gerard H Koppelman, Andrew T Hattersley, John W Holloway, Berthold Hocher, Joachim Heinrich, Chris Power, Mads Melbye, Mònica Guxens, Craig E Pennell, Klaus Bønnelykke, Hans Bisgaard, Johan G Eriksson, Elisabeth Widén, Hakon Hakonarson, André G Uitterlinden, Anneli Pouta, Debbie A Lawlor, George Davey Smith, Timothy M Frayling, Mark I McCarthy, Struan F A Grant, Vincent W V Jaddoe, Marjo-Riitta Jarvelin, Nicholas J Timpson, Inga Prokopenko, Rachel M Freathy, and Early Growth Genetics (EGG) Consortium. New loci associated with birth weight identify genetic links between intrauterine growth and adult height and metabolism. <em>Nat. Genet.</em>, 45(1):76–82, January 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ng.2477">http://dx.doi.org/10.1038/ng.2477</a>.</p>
</dd>
<dt class="label" id="id260"><span class="brackets">191</span></dt>
<dd><p>David A Hinds, George McMahon, Amy K Kiefer, Chuong B Do, Nicholas Eriksson, David M Evans, Beate St Pourcain, Susan M Ring, Joanna L Mountain, Uta Francke, George Davey-Smith, Nicholas J Timpson, and Joyce Y Tung. A genome-wide association meta-analysis of self-reported allergy identifies shared and allergy-specific susceptibility loci. <em>Nat. Genet.</em>, 45(8):907–911, August 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ng.2686">http://dx.doi.org/10.1038/ng.2686</a>.</p>
</dd>
<dt class="label" id="id259"><span class="brackets">192</span></dt>
<dd><p>Elise B Robinson, Beate St Pourcain, Verneri Anttila, Jack A Kosmicki, Brendan Bulik-Sullivan, Jakob Grove, Julian Maller, Kaitlin E Samocha, Stephan J Sanders, Stephan Ripke, Joanna Martin, Mads V Hollegaard, Thomas Werge, David M Hougaard, iPSYCH-SSI-Broad Autism Group, Benjamin M Neale, David M Evans, David Skuse, Preben Bo Mortensen, Anders D Børglum, Angelica Ronald, George Davey Smith, and Mark J Daly. Genetic risk for autism spectrum disorders and neuropsychiatric variation in the general population. <em>Nat. Genet.</em>, 48(5):552–555, May 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ng.3529">http://dx.doi.org/10.1038/ng.3529</a>.</p>
</dd>
<dt class="label" id="id262"><span class="brackets">193</span></dt>
<dd><p>Beate St Pourcain, C M A Haworth, O S P Davis, Kai Wang, Nicholas J Timpson, David M Evans, John P Kemp, Angelica Ronald, Tom Price, Emma Meaburn, Susan M Ring, Jean Golding, Hakon Hakonarson, R Plomin, and George Davey Smith. Heritability and genome-wide analyses of problematic peer relationships during childhood and adolescence. <em>Hum. Genet.</em>, 134(6):539–551, June 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s00439-014-1514-5">http://dx.doi.org/10.1007/s00439-014-1514-5</a>.</p>
</dd>
<dt class="label" id="id261"><span class="brackets">194</span></dt>
<dd><p>Emmanouela Repapi, Ian Sayers, Louise V Wain, Paul R Burton, Toby Johnson, Ma'en Obeidat, Jing Hua Zhao, Adaikalavan Ramasamy, Guangju Zhai, Veronique Vitart, Jennifer E Huffman, Wilmar Igl, Eva Albrecht, Panos Deloukas, John Henderson, Raquel Granell, Wendy L McArdle, Alicja R Rudnicka, Wellcome Trust Case Control Consortium, Inês Barroso, Ruth J F Loos, Nicholas J Wareham, Linda Mustelin, Taina Rantanen, Ida Surakka, Medea Imboden, H Erich Wichmann, Ivica Grkovic, Stipan Jankovic, Lina Zgaga, Anna-Liisa Hartikainen, Leena Peltonen, Ulf Gyllensten, Asa Johansson, Ghazal Zaboli, Harry Campbell, Sarah H Wild, James F Wilson, Sven Gläser, Georg Homuth, Henry Völzke, Massimo Mangino, Nicole Soranzo, Tim D Spector, Ozren Polasek, Igor Rudan, Alan F Wright, Markku Heliövaara, Samuli Ripatti, Anneli Pouta, Asa Torinsson Naluai, Anna-Carin Olin, Kjell Torén, Matthew N Cooper, Alan L James, Lyle J Palmer, Aroon D Hingorani, S Goya Wannamethee, Peter H Whincup, George Davey Smith, Shah Ebrahim, Tricia M McKeever, Ian D Pavord, Andrew K MacLeod, Andrew D Morris, David J Porteous, Cyrus Cooper, Elaine Dennison, Seif Shaheen, Stefan Karrasch, Eva Schnabel, Holger Schulz, Harald Grallert, Nabila Bouatia-Naji, Jérôme Delplanque, Philippe Froguel, John D Blakey, NSHD Respiratory Study Team, John R Britton, Richard W Morris, John W Holloway, Debbie A Lawlor, Jennie Hui, Fredrik Nyberg, Marjo-Riitta Jarvelin, Cathy Jackson, Mika Kähönen, Jaakko Kaprio, Nicole M Probst-Hensch, Beate Koch, Caroline Hayward, David M Evans, Paul Elliott, David P Strachan, Ian P Hall, and Martin D Tobin. Genome-wide association study identifies five loci associated with lung function. <em>Nat. Genet.</em>, 42(1):36–44, January 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ng.501">http://dx.doi.org/10.1038/ng.501</a>.</p>
</dd>
<dt class="label" id="id263"><span class="brackets">195</span></dt>
<dd><p>Naomi R Wray, Jian Yang, Ben J Hayes, Alkes L Price, Michael E Goddard, and Peter M Visscher. Pitfalls of predicting complex traits from SNPs. <em>Nat. Rev. Genet.</em>, 14(7):507–515, July 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3457">http://dx.doi.org/10.1038/nrg3457</a>.</p>
</dd>
<dt class="label" id="id115"><span class="brackets">196</span></dt>
<dd><p>Kasper Lage, Niclas Tue Hansen, E Olof Karlberg, Aron C Eklund, Francisco S Roque, Patricia K Donahoe, Zoltan Szallasi, Thomas Skøt Jensen, and Søren Brunak. A large-scale analysis of tissue-specific pathology and gene expression of human disease genes and complexes. <em>Proc. Natl. Acad. Sci. U. S. A.</em>, 105(52):20870–20875, December 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1073/pnas.0810772105">http://dx.doi.org/10.1073/pnas.0810772105</a>.</p>
</dd>
<dt class="label" id="id90"><span class="brackets">197</span></dt>
<dd><p>Eitan E Winter, Leo Goodstadt, and Chris P Ponting. Elevated rates of protein secretion, evolution, and disease among tissue-specific genes. <em>Genome Res.</em>, 14(1):54–61, January 2004. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.1924004">http://dx.doi.org/10.1101/gr.1924004</a>.</p>
</dd>
<dt class="label" id="id116"><span class="brackets">198</span></dt>
<dd><p>Owen J L Rackham, Hashem A Shihab, Michael R Johnson, and Enrico Petretto. EvoTol: a protein-sequence based evolutionary intolerance framework for disease-gene prioritization. <em>Nucleic Acids Res.</em>, 43(5):e33, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gku1322">http://dx.doi.org/10.1093/nar/gku1322</a>.</p>
</dd>
<dt class="label" id="id117"><span class="brackets">199</span></dt>
<dd><p>Agne Antanaviciute, Catherine Daly, Laura A Crinnion, Alexander F Markham, Christopher M Watson, David T Bonthron, and Ian M Carr. GeneTIER: prioritization of candidate disease genes using tissue-specific gene expression profiles. <em>Bioinformatics</em>, 31(16):2728–2735, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btv196">http://dx.doi.org/10.1093/bioinformatics/btv196</a>.</p>
</dd>
<dt class="label" id="id205"><span class="brackets">200</span></dt>
<dd><p>Arjun Raj and Alexander van Oudenaarden. Nature, nurture, or chance: stochastic gene expression and its consequences. <em>Cell</em>, 135(2):216–226, October 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.cell.2008.09.050">http://dx.doi.org/10.1016/j.cell.2008.09.050</a>.</p>
</dd>
<dt class="label" id="id210"><span class="brackets">201</span></dt>
<dd><p>Jong Kyoung Kim, Aleksandra A Kolodziejczyk, Tomislav Ilicic, Sarah A Teichmann, and John C Marioni. Characterizing noise structure in single-cell RNA-seq distinguishes genuine from technical stochastic allelic expression. <em>Nat. Commun.</em>, 6:8687, October 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ncomms9687">http://dx.doi.org/10.1038/ncomms9687</a>.</p>
</dd>
<dt class="label" id="id209"><span class="brackets">202</span></dt>
<dd><p>Nils Eling, Michael D Morgan, and John C Marioni. Challenges in measuring and understanding biological noise. <em>Nat. Rev. Genet.</em>, 20(9):536–548, September 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41576-019-0130-6">http://dx.doi.org/10.1038/s41576-019-0130-6</a>.</p>
</dd>
<dt class="label" id="id207"><span class="brackets">203</span></dt>
<dd><p>Simon Anders, Davis J McCarthy, Yunshun Chen, Michal Okoniewski, Gordon K Smyth, Wolfgang Huber, and Mark D Robinson. Count-based differential expression analysis of RNA sequencing data using R and bioconductor. <em>Nat. Protoc.</em>, 8(9):1765–1786, September 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nprot.2013.099">http://dx.doi.org/10.1038/nprot.2013.099</a>.</p>
</dd>
<dt class="label" id="id208"><span class="brackets">204</span></dt>
<dd><p>Michael I Love, Simon Anders, and Wolfgang Huber. DESeq2 vignette: analyzing RNA-seq data with DESeq2. <span><a class="reference external" href="#"></a></span>http://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html, February 2021. Accessed: 2021-3-21. URL: <a class="reference external" href="http://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">http://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a>.</p>
</dd>
<dt class="label" id="id211"><span class="brackets">205</span></dt>
<dd><p>Ana Conesa, Pedro Madrigal, Sonia Tarazona, David Gomez-Cabrero, Alejandra Cervera, Andrew McPherson, Michał Wojciech Szcześniak, Daniel J Gaffney, Laura L Elo, Xuegong Zhang, and Ali Mortazavi. A survey of best practices for RNA-seq data analysis. <em>Genome Biol.</em>, 17:13, January 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-016-0881-8">http://dx.doi.org/10.1186/s13059-016-0881-8</a>.</p>
</dd>
<dt class="label" id="id185"><span class="brackets">206</span></dt>
<dd><p>Anton Pottegård, Maija Bruun Haastrup, Tore Bjerregaard Stage, Morten Rix Hansen, Kasper Søltoft Larsen, Peter Martin Meegaard, Line Haugaard Vrdlovec Meegaard, Henrik Horneberg, Charlotte Gils, Dorthe Dideriksen, Lise Aagaard, Anna Birna Almarsdottir, Jesper Hallas, and Per Damkier. SearCh for humouristic and extravagant acronyms and thoroughly inappropriate names for important clinical trials (SCIENTIFIC): qualitative and quantitative systematic study. <em>BMJ</em>, 349:g7092, December 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1136/bmj.g7092">http://dx.doi.org/10.1136/bmj.g7092</a>.</p>
</dd>
<dt class="label" id="id204"><span class="brackets">207</span></dt>
<dd><p>Imad Abugessaisa, Shuhei Noguchi, Akira Hasegawa, Jayson Harshbarger, Atsushi Kondo, Marina Lizio, Jessica Severin, Piero Carninci, Hideya Kawaji, and Takeya Kasukawa. FANTOM5 CAGE profiles of human and mouse reprocessed for GRCh38 and GRCm38 genome assemblies. <em>Sci Data</em>, 4:170107, August 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1038/sdata.2017.107">http://dx.doi.org/10.1038/sdata.2017.107</a>.</p>
</dd>
<dt class="label" id="id198"><span class="brackets">208</span></dt>
<dd><p>Alexandra Witze. Wealthy funder pays reparations for use of HeLa cells. <em>Nature</em>, 587(7832):20–21, November 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1038/d41586-020-03042-5">http://dx.doi.org/10.1038/d41586-020-03042-5</a>.</p>
</dd>
<dt class="label" id="id195"><span class="brackets">209</span></dt>
<dd><p>Danielle M Pastor, Lisa S Poritz, Thomas L Olson, Christina L Kline, Leonard R Harris, Walter A Koltun, Vernon M Chinchilli, and Rosalyn B Irby. Primary cell lines: false representation or model system? a comparison of four human colorectal tumors and their coordinately established cell lines. <em>Int. J. Clin. Exp. Med.</em>, 3(1):69–83, February 2010. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/20369042">https://www.ncbi.nlm.nih.gov/pubmed/20369042</a>.</p>
</dd>
<dt class="label" id="id196"><span class="brackets">210</span></dt>
<dd><p>Gurvinder Kaur and Jannette M Dufour. Cell lines: valuable tools or useless artifacts. <em>Spermatogenesis</em>, 2(1):1–5, January 2012. URL: <a class="reference external" href="http://dx.doi.org/10.4161/spmg.19885">http://dx.doi.org/10.4161/spmg.19885</a>.</p>
</dd>
<dt class="label" id="id197"><span class="brackets">211</span></dt>
<dd><p>Graham Bell. Replicates and repeats. <em>BMC Biol.</em>, 14:28, April 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s12915-016-0254-5">http://dx.doi.org/10.1186/s12915-016-0254-5</a>.</p>
</dd>
<dt class="label" id="id191"><span class="brackets">212</span></dt>
<dd><p>Peter J A Cock, Tiago Antao, Jeffrey T Chang, Brad A Chapman, Cymon J Cox, Andrew Dalke, Iddo Friedberg, Thomas Hamelryck, Frank Kauff, Bartek Wilczynski, and Michiel J L de Hoon. Biopython: freely available python tools for computational molecular biology and bioinformatics. <em>Bioinformatics</em>, 25(11):1422–1423, June 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btp163">http://dx.doi.org/10.1093/bioinformatics/btp163</a>.</p>
</dd>
<dt class="label" id="id192"><span class="brackets">213</span></dt>
<dd><p>Sangya Pundir, Maria J Martin, Claire O'Donovan, and UniProt Consortium. UniProt tools. <em>Curr. Protoc. Bioinformatics</em>, 53:1.29.1–1.29.15, March 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1002/0471250953.bi0129s53">http://dx.doi.org/10.1002/0471250953.bi0129s53</a>.</p>
</dd>
<dt class="label" id="id193"><span class="brackets">214</span></dt>
<dd><p>Hai Fang. dcGOR: an R package for analysing ontologies and protein domain annotations. <em>PLoS Comput. Biol.</em>, 10(10):e1003929, October 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pcbi.1003929">http://dx.doi.org/10.1371/journal.pcbi.1003929</a>.</p>
</dd>
<dt class="label" id="id189"><span class="brackets">215</span></dt>
<dd><p>Fran Supek, Matko Bošnjak, Nives Škunca, and Tomislav Šmuc. REVIGO summarizes and visualizes long lists of gene ontology terms. <em>PLoS One</em>, 6(7):e21800, July 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pone.0021800">http://dx.doi.org/10.1371/journal.pone.0021800</a>.</p>
</dd>
<dt class="label" id="id213"><span class="brackets">216</span></dt>
<dd><p>Christopher Buccitelli and Matthias Selbach. mRNAs, proteins and the emerging principles of gene expression control. <em>Nat. Rev. Genet.</em>, 21(10):630–644, October 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41576-020-0258-4">http://dx.doi.org/10.1038/s41576-020-0258-4</a>.</p>
</dd>
<dt class="label" id="id212"><span class="brackets">217</span></dt>
<dd><p>Goro Terai and Kiyoshi Asai. Improving the prediction accuracy of protein abundance in escherichia coli using mRNA accessibility. <em>Nucleic Acids Res.</em>, 48(14):e81, August 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkaa481">http://dx.doi.org/10.1093/nar/gkaa481</a>.</p>
</dd>
<dt class="label" id="id215"><span class="brackets">218</span></dt>
<dd><p>D V Klopfenstein, Liangsheng Zhang, Brent S Pedersen, Fidel Ramírez, Alex Warwick Vesztrocy, Aurélien Naldi, Christopher J Mungall, Jeffrey M Yunes, Olga Botvinnik, Mark Weigel, Will Dampier, Christophe Dessimoz, Patrick Flick, and Haibao Tang. GOATOOLS: a python library for gene ontology analyses. <em>Sci. Rep.</em>, 8(1):10872, July 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41598-018-28948-z">http://dx.doi.org/10.1038/s41598-018-28948-z</a>.</p>
</dd>
<dt class="label" id="id216"><span class="brackets">219</span></dt>
<dd><p>Edison Ong, Zuoshuang Xiang, Bin Zhao, Yue Liu, Yu Lin, Jie Zheng, Chris Mungall, Mélanie Courtot, Alan Ruttenberg, and Yongqun He. Ontobee: a linked ontology data server to support ontology term dereferencing, linkage, query and integration. <em>Nucleic Acids Res.</em>, 45(D1):D347–D352, January 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkw918">http://dx.doi.org/10.1093/nar/gkw918</a>.</p>
</dd>
<dt class="label" id="id214"><span class="brackets">220</span></dt>
<dd><p>Martin Larralde, Alex Henrie, Philipp A., Spencer Mitchell, and Tatsuya Sakaguchi. Althonos/pronto: 2.4.1. February 2021. URL: <a class="reference external" href="https://zenodo.org/record/4552164">https://zenodo.org/record/4552164</a>.</p>
</dd>
<dt class="label" id="id224"><span class="brackets">221</span></dt>
<dd><p>Andrea C F Albuquerque, Jose L Campos dos Santos, and Alberto N de Castro. OntoBio: a biodiversity domain ontology for amazonian biological collected objects. In <em>2015 48th Hawaii International Conference on System Sciences</em>, 3770–3779. January 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1109/HICSS.2015.453">http://dx.doi.org/10.1109/HICSS.2015.453</a>.</p>
</dd>
<dt class="label" id="id190"><span class="brackets">222</span></dt>
<dd><p>Steven Bird. NLTK: the natural language toolkit. In <em>Proceedings of the COLING/ACL 2006 Interactive Presentation Sessions</em>, 69–72. aclweb.org, 2006. URL: <a class="reference external" href="https://www.aclweb.org/anthology/P06-4018.pdf">https://www.aclweb.org/anthology/P06-4018.pdf</a>.</p>
</dd>
<dt class="label" id="id217"><span class="brackets">223</span></dt>
<dd><p>Charles R. Harris, K. Jarrod Millman, St'efan J. van der Walt, Ralf Gommers, Pauli Virtanen, David Cournapeau, Eric Wieser, Julian Taylor, Sebastian Berg, Nathaniel J. Smith, Robert Kern, Matti Picus, Stephan Hoyer, Marten H. van Kerkwijk, Matthew Brett, Allan Haldane, Jaime Fern'andez del R'ıo, Mark Wiebe, Pearu Peterson, Pierre G'erard-Marchant, Kevin Sheppard, Tyler Reddy, Warren Weckesser, Hameer Abbasi, Christoph Gohlke, and Travis E. Oliphant. Array programming with NumPy. <em>Nature</em>, 585(7825):357–362, September 2020. URL: <a class="reference external" href="https://doi.org/10.1038/s41586-020-2649-2">https://doi.org/10.1038/s41586-020-2649-2</a>, <a class="reference external" href="https://doi.org/10.1038/s41586-020-2649-2">doi:10.1038/s41586-020-2649-2</a>.</p>
</dd>
<dt class="label" id="id218"><span class="brackets">224</span></dt>
<dd><p>The pandas development team. Pandas-dev/pandas: pandas. February 2020. URL: <a class="reference external" href="https://doi.org/10.5281/zenodo.3509134">https://doi.org/10.5281/zenodo.3509134</a>, <a class="reference external" href="https://doi.org/10.5281/zenodo.3509134">doi:10.5281/zenodo.3509134</a>.</p>
</dd>
<dt class="label" id="id220"><span class="brackets">225</span></dt>
<dd><p>Christopher Woods. Developer's guide — MetaWards documentation. <span><a class="reference external" href="#"></a></span>https://metawards.org/development.html?highlight=style. Accessed: 2021-5-1. URL: <a class="reference external" href="https://metawards.org/development.html?highlight=style">https://metawards.org/development.html?highlight=style</a>.</p>
</dd>
<dt class="label" id="id221"><span class="brackets">226</span></dt>
<dd><p>Donald Stufft. Welcome to twine's documentation! — twine 3.4.2.dev1+geff3a45 documentation. <span><a class="reference external" href="#"></a></span>https://twine.readthedocs.io/en/latest/, 2019. Accessed: 2021-5-1. URL: <a class="reference external" href="https://twine.readthedocs.io/en/latest/">https://twine.readthedocs.io/en/latest/</a>.</p>
</dd>
<dt class="label" id="id222"><span class="brackets">227</span></dt>
<dd><p>Holger Krekel, Bruno Oliveira, Ronny Pfannschmidt, Floris Bruynooghe, Brianna Laugher, and Florian Bruhin. Pytest x.y. 2004. URL: <a class="reference external" href="https://github.com/pytest-dev/pytest">https://github.com/pytest-dev/pytest</a>.</p>
</dd>
<dt class="label" id="id223"><span class="brackets">228</span></dt>
<dd><p>PyData Community. The PyData sphinx theme — PyData sphinx theme documentation. <span><a class="reference external" href="#"></a></span>https://pydata-sphinx-theme.readthedocs.io/, 2019. Accessed: 2021-5-1. URL: <a class="reference external" href="https://pydata-sphinx-theme.readthedocs.io/">https://pydata-sphinx-theme.readthedocs.io/</a>.</p>
</dd>
<dt class="label" id="id225"><span class="brackets">229</span></dt>
<dd><p>Marina Lizio, Jayson Harshbarger, Hisashi Shimoji, Jessica Severin, Takeya Kasukawa, Serkan Sahin, Imad Abugessaisa, Shiro Fukuda, Fumi Hori, Sachi Ishikawa-Kato, Christopher J Mungall, Erik Arner, J Kenneth Baillie, Nicolas Bertin, Hidemasa Bono, Michiel de Hoon, Alexander D Diehl, Emmanuel Dimont, Tom C Freeman, Kaori Fujieda, Winston Hide, Rajaram Kaliyaperumal, Toshiaki Katayama, Timo Lassmann, Terrence F Meehan, Koro Nishikata, Hiromasa Ono, Michael Rehli, Albin Sandelin, Erik A Schultes, Peter A C 't Hoen, Zuotian Tatum, Mark Thompson, Tetsuro Toyoda, Derek W Wright, Carsten O Daub, Masayoshi Itoh, Piero Carninci, Yoshihide Hayashizaki, Alistair R R Forrest, Hideya Kawaji, and FANTOM consortium. Gateways to the FANTOM5 promoter level mammalian expression atlas. <em>Genome Biol.</em>, 16:22, January 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-014-0560-6">http://dx.doi.org/10.1186/s13059-014-0560-6</a>.</p>
</dd>
<dt class="label" id="id227"><span class="brackets">230</span></dt>
<dd><p>Jean-Baptiste Lamy. Owlready: ontology-oriented programming in python with automatic classification and high level constructs for biomedical ontologies. <em>Artif. Intell. Med.</em>, 80:11–28, July 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.artmed.2017.07.002">http://dx.doi.org/10.1016/j.artmed.2017.07.002</a>.</p>
</dd>
<dt class="label" id="id111"><span class="brackets">231</span></dt>
<dd><p>David Gomez-Cabrero, Imad Abugessaisa, Dieter Maier, Andrew Teschendorff, Matthias Merkenschlager, Andreas Gisel, Esteban Ballestar, Erik Bongcam-Rudloff, Ana Conesa, and Jesper Tegnér. Data integration in the era of omics: current and future challenges. <em>BMC Syst. Biol.</em>, 8 Suppl 2:I1, March 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1752-0509-8-S2-I1">http://dx.doi.org/10.1186/1752-0509-8-S2-I1</a>.</p>
</dd>
<dt class="label" id="id110"><span class="brackets">232</span></dt>
<dd><p>Gordon Bell, Tony Hey, and Alex Szalay. Computer science. beyond the data deluge. <em>Science</em>, 323(5919):1297–1298, March 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.1170411">http://dx.doi.org/10.1126/science.1170411</a>.</p>
</dd>
<dt class="label" id="id103"><span class="brackets">233</span></dt>
<dd><p>S Cortijo, Z Aydin, S Ahnert, and others. Widespread inter‐individual gene expression variability in arabidopsis thaliana. <em>Mol. Syst. Biol.</em>, 2019. URL: <a class="reference external" href="http://msb.embopress.org/content/15/1/e8591.abstract">http://msb.embopress.org/content/15/1/e8591.abstract</a>.</p>
</dd>
<dt class="label" id="id105"><span class="brackets">234</span></dt>
<dd><p>Ana Viñuela, Andrew A Brown, Alfonso Buil, Pei-Chien Tsai, Matthew N Davies, Jordana T Bell, Emmanouil T Dermitzakis, Timothy D Spector, and Kerrin S Small. Age-dependent changes in mean and variance of gene expression across tissues in a twin cohort. <em>Human molecular genetics</em>, 27(4):732–741, 2018.</p>
</dd>
<dt class="label" id="id104"><span class="brackets">235</span></dt>
<dd><p>Jialiang Yang, Tao Huang, Francesca Petralia, Quan Long, Bin Zhang, Carmen Argmann, Yong Zhao, Charles V Mobbs, Eric E Schadt, Jun Zhu, Zhidong Tu, and GTEx Consortium. Synchronized age-related gene expression changes across multiple tissues in human and the link to complex diseases. <em>Sci. Rep.</em>, 5:15145, October 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/srep15145">http://dx.doi.org/10.1038/srep15145</a>.</p>
</dd>
<dt class="label" id="id106"><span class="brackets">236</span></dt>
<dd><p>B Reinius and E Jazin. Prenatal sex differences in the human brain. <em>Mol. Psychiatry</em>, 14(11):987, 988–9, November 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1038/mp.2009.79">http://dx.doi.org/10.1038/mp.2009.79</a>.</p>
</dd>
<dt class="label" id="id107"><span class="brackets">237</span></dt>
<dd><p>Gregory Stone, Ashley Choi, Meritxell Oliva, Joshua Gorham, Mahyar Heydarpour, Christine E Seidman, Jon G Seidman, Sary F Aranki, Simon C Body, Vincent J Carey, Benjamin A Raby, Barbara E Stranger, and Jochen D Muehlschlegel. Sex differences in gene expression in response to ischemia in the human left ventricular myocardium. <em>Hum. Mol. Genet.</em>, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/hmg/ddz014">http://dx.doi.org/10.1093/hmg/ddz014</a>.</p>
</dd>
<dt class="label" id="id108"><span class="brackets">238</span></dt>
<dd><p>Ueli Schibler. The daily timing of gene expression and physiology in mammals. <em>Dialogues Clin. Neurosci.</em>, 9(3):257–272, 2007. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/17969863">https://www.ncbi.nlm.nih.gov/pubmed/17969863</a>.</p>
</dd>
<dt class="label" id="id109"><span class="brackets">239</span></dt>
<dd><p>Valentine Svensson, Sarah A Teichmann, and Oliver Stegle. SpatialDE: identification of spatially variable genes. <em>Nat. Methods</em>, 15(5):343–346, May 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.4636">http://dx.doi.org/10.1038/nmeth.4636</a>.</p>
</dd>
<dt class="label" id="id112"><span class="brackets">240</span></dt>
<dd><p>Qingguo Wang, Joshua Armenia, Chao Zhang, Alexander V Penson, Ed Reznik, Liguo Zhang, Thais Minet, Angelica Ochoa, Benjamin E Gross, Christine A Iacobuzio-Donahue, Doron Betel, Barry S Taylor, Jianjiong Gao, and Nikolaus Schultz. Unifying cancer and normal RNA sequencing data from different sources. <em>Sci Data</em>, 5:180061, April 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/sdata.2018.61">http://dx.doi.org/10.1038/sdata.2018.61</a>.</p>
</dd>
<dt class="label" id="id41"><span class="brackets">241</span></dt>
<dd><p>Jeffrey T Leek, Robert B Scharpf, Héctor Corrada Bravo, David Simcha, Benjamin Langmead, W Evan Johnson, Donald Geman, Keith Baggerly, and Rafael A Irizarry. Tackling the widespread and critical impact of batch effects in high-throughput data. <em>Nat. Rev. Genet.</em>, 11(10):733–739, October 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg2825">http://dx.doi.org/10.1038/nrg2825</a>.</p>
</dd>
<dt class="label" id="id42"><span class="brackets">242</span></dt>
<dd><p>Rafael A Irizarry, Daniel Warren, Forrest Spencer, Irene F Kim, Shyam Biswal, Bryan C Frank, Edward Gabrielson, Joe G N Garcia, Joel Geoghegan, Gregory Germino, Constance Griffin, Sara C Hilmer, Eric Hoffman, Anne E Jedlicka, Ernest Kawasaki, Francisco Martínez-Murillo, Laura Morsberger, Hannah Lee, David Petersen, John Quackenbush, Alan Scott, Michael Wilson, Yanqin Yang, Shui Qing Ye, and Wayne Yu. Multiple-laboratory comparison of microarray platforms. <em>Nat. Methods</em>, 2(5):345–350, May 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth756">http://dx.doi.org/10.1038/nmeth756</a>.</p>
</dd>
<dt class="label" id="id69"><span class="brackets">243</span></dt>
<dd><p>W Evan Johnson, Cheng Li, and Ariel Rabinovic. Adjusting batch effects in microarray expression data using empirical bayes methods. <em>Biostatistics</em>, 8(1):118–127, January 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1093/biostatistics/kxj037">http://dx.doi.org/10.1093/biostatistics/kxj037</a>.</p>
</dd>
<dt class="label" id="id85"><span class="brackets">244</span></dt>
<dd><p>Jeffrey T Leek and John D Storey. Capturing heterogeneity in gene expression studies by surrogate variable analysis. <em>PLoS Genet.</em>, 3(9):1724–1735, September 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pgen.0030161">http://dx.doi.org/10.1371/journal.pgen.0030161</a>.</p>
</dd>
<dt class="label" id="id86"><span class="brackets">245</span></dt>
<dd><p>Chao Chen, Kay Grennan, Judith Badner, Dandan Zhang, Elliot Gershon, Li Jin, and Chunyu Liu. Removing batch effects in analysis of expression microarray data: an evaluation of six batch adjustment methods. <em>PLoS One</em>, 6(2):e17238, February 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pone.0017238">http://dx.doi.org/10.1371/journal.pone.0017238</a>.</p>
</dd>
<dt class="label" id="id87"><span class="brackets">246</span></dt>
<dd><p>Q Liu and M Markatou. Evaluation of methods in removing batch effects on RNA-seq data. <em>Infectious Diseases and Translational Medicine</em>, 2016. URL: <a class="reference external" href="http://www.tran-med.com/CN/article/downloadArticleFile.do?attachType=PDF&amp;id=24">http://www.tran-med.com/CN/article/downloadArticleFile.do?attachType=PDF&amp;id=24</a>.</p>
</dd>
<dt class="label" id="id89"><span class="brackets">247</span></dt>
<dd><p>Nuno A Fonseca, Robert Petryszak, John Marioni, and Alvis Brazma. iRAP - an integrated RNA-seq analysis pipeline. Preprint, June 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1101/005991">http://dx.doi.org/10.1101/005991</a>.</p>
</dd>
<dt class="label" id="id94"><span class="brackets">248</span></dt>
<dd><p>Tobias Maier, Marc Güell, and Luis Serrano. Correlation of mRNA and protein in complex biological samples. <em>FEBS Lett.</em>, 583(24):3966–3973, December 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.febslet.2009.10.036">http://dx.doi.org/10.1016/j.febslet.2009.10.036</a>.</p>
</dd>
<dt class="label" id="id95"><span class="brackets">249</span></dt>
<dd><p>Andreas Beyer, Jens Hollunder, Heinz-Peter Nasheuer, and Thomas Wilhelm. Post-transcriptional expression regulation in the yeast saccharomyces cerevisiae on a genomic scale. <em>Mol. Cell. Proteomics</em>, 3(11):1083–1092, November 2004. URL: <a class="reference external" href="http://dx.doi.org/10.1074/mcp.M400099-MCP200">http://dx.doi.org/10.1074/mcp.M400099-MCP200</a>.</p>
</dd>
<dt class="label" id="id96"><span class="brackets">250</span></dt>
<dd><p>Gang Wu, Lei Nie, and Weiwen Zhang. Integrative analyses of posttranscriptional regulation in the yeast saccharomyces cerevisiae using transcriptomic and proteomic data. <em>Curr. Microbiol.</em>, 57(1):18–22, July 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s00284-008-9145-5">http://dx.doi.org/10.1007/s00284-008-9145-5</a>.</p>
</dd>
<dt class="label" id="id97"><span class="brackets">251</span></dt>
<dd><p>Nancy Yiu-Lin Yu, Björn M Hallström, Linn Fagerberg, Fredrik Ponten, Hideya Kawaji, Piero Carninci, Alistair R R Forrest, Fantom Consortium, Yoshihide Hayashizaki, Mathias Uhlén, and Carsten O Daub. Complementing tissue characterization by integrating transcriptome profiling from the human protein atlas and from the FANTOM5 consortium. <em>Nucleic Acids Res.</em>, 43(14):6787–6798, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv608">http://dx.doi.org/10.1093/nar/gkv608</a>.</p>
</dd>
<dt class="label" id="id156"><span class="brackets">252</span></dt>
<dd><p>The human protein atlas. <span><a class="reference external" href="#"></a></span>https://www.proteinatlas.org/. Accessed: 2021-1-29. URL: <a class="reference external" href="https://www.proteinatlas.org/">https://www.proteinatlas.org/</a>.</p>
</dd>
<dt class="label" id="id98"><span class="brackets">253</span></dt>
<dd><p>Hideya Kawaji, Marina Lizio, Masayoshi Itoh, Mutsumi Kanamori-Katayama, Ai Kaiho, Hiromi Nishiyori-Sueki, Jay W Shin, Miki Kojima-Ishiyama, Mitsuoki Kawano, Mitsuyoshi Murata, Noriko Ninomiya-Fukuda, Sachi Ishikawa-Kato, Sayaka Nagao-Sato, Shohei Noma, Yoshihide Hayashizaki, Alistair R R Forrest, Piero Carninci, and FANTOM Consortium. Comparison of CAGE and RNA-seq transcriptome profiling using clonally amplified and single-molecule next-generation sequencing. <em>Genome Res.</em>, 24(4):708–717, April 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.156232.113">http://dx.doi.org/10.1101/gr.156232.113</a>.</p>
</dd>
<dt class="label" id="id99"><span class="brackets">254</span></dt>
<dd><p>Mathias Uhlen, Per Oksvold, Linn Fagerberg, Emma Lundberg, Kalle Jonasson, Mattias Forsberg, Martin Zwahlen, Caroline Kampf, Kenneth Wester, Sophia Hober, Henrik Wernerus, Lisa Björling, and Fredrik Ponten. Towards a knowledge-based human protein atlas. <em>Nature Biotechnology</em>, 28(12):1248–1250, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nbt1210-1248">http://dx.doi.org/10.1038/nbt1210-1248</a>.</p>
</dd>
<dt class="label" id="id100"><span class="brackets">255</span></dt>
<dd><p>Mathias Uhlén, Linn Fagerberg, Björn M Hallström, Cecilia Lindskog, Per Oksvold, Adil Mardinoglu, Åsa Sivertsson, Caroline Kampf, Evelina Sjöstedt, Anna Asplund, Ingmarie Olsson, Karolina Edlund, Emma Lundberg, Sanjay Navani, Cristina Al-Khalili Szigyarto, Jacob Odeberg, Dijana Djureinovic, Jenny Ottosson Takanen, Sophia Hober, Tove Alm, Per-Henrik Edqvist, Holger Berling, Hanna Tegel, Jan Mulder, Johan Rockberg, Peter Nilsson, Jochen M Schwenk, Marica Hamsten, Kalle von Feilitzen, Mattias Forsberg, Lukas Persson, Fredric Johansson, Martin Zwahlen, Gunnar von Heijne, Jens Nielsen, and Fredrik Pontén. Proteomics. tissue-based map of the human proteome. <em>Science</em>, 347(6220):1260419, January 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.1260419">http://dx.doi.org/10.1126/science.1260419</a>.</p>
</dd>
<dt class="label" id="id101"><span class="brackets">256</span></dt>
<dd><p>GTEx Consortium. The Genotype-Tissue expression (GTEx) project. <em>Nat. Genet.</em>, 45(6):580–585, June 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ng.2653">http://dx.doi.org/10.1038/ng.2653</a>.</p>
</dd>
<dt class="label" id="id102"><span class="brackets">257</span></dt>
<dd><p>Susan J Lindsay, Yaobo Xu, Steven N Lisgo, Lauren F Harkin, Andrew J Copp, Dianne Gerrelli, Gavin J Clowry, Aysha Talbot, Michael J Keogh, Jonathan Coxhead, Mauro Santibanez-Koref, and Patrick F Chinnery. HDBR expression: a unique resource for global and individual gene expression studies during early human brain development. <em>Front. Neuroanat.</em>, 10:86, October 2016. URL: <a class="reference external" href="http://dx.doi.org/10.3389/fnana.2016.00086">http://dx.doi.org/10.3389/fnana.2016.00086</a>.</p>
</dd>
<dt class="label" id="id157"><span class="brackets">258</span></dt>
<dd><p>M Keays. ExpressionAtlas: download datasets from EMBL-EBI expression atlas. R package version 1.10.0. 2018. URL: <a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/ExpressionAtlas.html">https://bioconductor.org/packages/release/bioc/html/ExpressionAtlas.html</a>.</p>
</dd>
<dt class="label" id="id275"><span class="brackets">259</span></dt>
<dd><p>Damian Smedley, Syed Haider, Benoit Ballester, Richard Holland, Darin London, Gudmundur Thorisson, and Arek Kasprzyk. BioMart–biological queries made easy. <em>BMC Genomics</em>, 10:22, January 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2164-10-22">http://dx.doi.org/10.1186/1471-2164-10-22</a>.</p>
</dd>
<dt class="label" id="id276"><span class="brackets">260</span></dt>
<dd><p>Clarissa M Koch, Stephen F Chiu, Mahzad Akbarpour, Ankit Bharat, Karen M Ridge, Elizabeth T Bartom, and Deborah R Winter. A beginner's guide to analysis of RNA sequencing data. <em>Am. J. Respir. Cell Mol. Biol.</em>, 59(2):145–157, August 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1165/rcmb.2017-0430TR">http://dx.doi.org/10.1165/rcmb.2017-0430TR</a>.</p>
</dd>
<dt class="label" id="id76"><span class="brackets">261</span></dt>
<dd><p>Aaron M Newman, Chih Long Liu, Michael R Green, Andrew J Gentles, Weiguo Feng, Yue Xu, Chuong D Hoang, Maximilian Diehn, and Ash A Alizadeh. Robust enumeration of cell subsets from tissue expression profiles. <em>Nat. Methods</em>, 12(5):453–457, May 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.3337">http://dx.doi.org/10.1038/nmeth.3337</a>.</p>
</dd>
<dt class="label" id="id77"><span class="brackets">262</span></dt>
<dd><p>Maayan Baron, Adrian Veres, Samuel L Wolock, Aubrey L Faust, Renaud Gaujoux, Amedeo Vetere, Jennifer Hyoje Ryu, Bridget K Wagner, Shai S Shen-Orr, Allon M Klein, Douglas A Melton, and Itai Yanai. A Single-Cell transcriptomic map of the human and mouse pancreas reveals inter- and intra-cell population structure. <em>Cell Syst</em>, 3(4):346–360.e4, October 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.cels.2016.08.011">http://dx.doi.org/10.1016/j.cels.2016.08.011</a>.</p>
</dd>
<dt class="label" id="id78"><span class="brackets">263</span></dt>
<dd><p>Xuran Wang, Jihwan Park, Katalin Susztak, Nancy R Zhang, and Mingyao Li. Bulk tissue cell type deconvolution with multi-subject single-cell expression reference. <em>Nat. Commun.</em>, 10(1):380, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41467-018-08023-x">http://dx.doi.org/10.1038/s41467-018-08023-x</a>.</p>
</dd>
<dt class="label" id="id274"><span class="brackets">264</span></dt>
<dd><p>Yuqing Zhang, Giovanni Parmigiani, and W Evan Johnson. ComBat-seq: batch effect adjustment for RNA-seq count data. <em>NAR Genom Bioinform</em>, 2(3):lqaa078, September 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nargab/lqaa078">http://dx.doi.org/10.1093/nargab/lqaa078</a>.</p>
</dd>
<dt class="label" id="id79"><span class="brackets">265</span></dt>
<dd><p>Laleh Haghverdi, Aaron T L Lun, Michael D Morgan, and John C Marioni. Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors. <em>Nat. Biotechnol.</em>, 36(5):421–427, June 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nbt.4091">http://dx.doi.org/10.1038/nbt.4091</a>.</p>
</dd>
<dt class="label" id="id88"><span class="brackets">266</span></dt>
<dd><p>Alyssa C Frazee, Andrew E Jaffe, Ben Langmead, and Jeffrey T Leek. Polyester: simulating RNA-seq datasets with differential transcript expression. <em>Bioinformatics</em>, 31(17):2778–2784, September 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btv272">http://dx.doi.org/10.1093/bioinformatics/btv272</a>.</p>
</dd>
<dt class="label" id="id271"><span class="brackets">267</span></dt>
<dd><p>Jeff Alstott, Ed Bullmore, and Dietmar Plenz. Powerlaw: a python package for analysis of heavy-tailed distributions. <em>PLoS One</em>, 9(1):e85777, January 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pone.0085777">http://dx.doi.org/10.1371/journal.pone.0085777</a>.</p>
</dd>
</dl>
</p>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./cz-end-matter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="0-appendix.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Appendix</p>
        </div>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Natalie Zelenka<br/>
    
        &copy; Copyright 2020.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>