(proteins-organised)=

### Looking more closely at proteins: domains and disorder

Just as DNA has been classified at different levels (SNP, gene, genome), so too have proteins.

#### Primary, Secondary, Tertiary, and Quaternary

Proteins are described and classified in terms of the their primary, secondary, tertiary, and quaternary structure. 

The primary structure is simply the amino acid makeup of the protein, which describes the protein's chemical makeup, but not it's three-dimensional structure. 
These amino acid strings tend to form into a small number of familiar (secondary) three-dimensional structures, for example beta strands, beta sheets, and alpha helices.

At the next level, the tertiary structure describes folds: combinations of secondary structures, for example a TIM barrels (a torroidal structure made up of alpha helices and beta strands). 
At this level, similar structures do not imply an evolutionary or functional similarity.

##### Quaternary structures: protein domains

```{figure} ../images/lego.png
---
height: 100px
name: lego
alt: The figure shows a red lego block linked to a yellow lego block, which is then linked to another yellow lego block. The links are thin grey lego blocks.
---
An illustration of the lego analogy for protein domains. Coloured bricks represent protein domains - colour represents a specific protein domain type, while thin grey bricks represent polypeptide linkers which link domains. Image created using mecabricks{cite}`Jarraud2019-jj`
``` ``

[//]: # (TODO: delete extra ```)

The quaternary structures of proteins - protein domains - have proved particularly interesting for research. 
A simple and oft-used metaphor is to think of protein domains as lego building blocks ({numref}`lego`) which can be linked by polypeptide chains to make up a protein. 
These polypeptide chains (known as linkers) are often inflexible, in order to allow only one conformation of the protein. 
Small and simple proteins often consist of just one domain, while bigger proteins can contain many domains. 
An individual domain can be found in many different proteins, and multiple times in the same protein. 

[//]: # (TODO: Definitely definitely need this to be cited.)
Protein domains are interesting because they are highly conserved in evolution, and are thought of as units of function, evolution, and/or structure. 
The functions of proteins, at both low-level (e.g. “calcium signalling protein”) and high-level (e.g. annotated to “liver disease”), are costly and difficult to discern, so there are many proteins about which little is known. 
For this reason, often proteins are classified according to their similarity to proteins about which functions are known, for example those containing the same protein domains.

##### Disorder
While protein domains exist in one confirmation, not all linkers are rigid. 
Flexible regions of proteins which allow for various conformations are referred to as disordered. 
Disordered regions are not only relegated to linkers between domains. 
Proteins can be constituted entirely of disordered regions, or may have large disordered regions.

Intrinsically disordered proteins can exist in a number of conformations, rather than one fixed structure. 
On some occasions, the disordered regions are known to be functional, while on others, proteins may be non-functional until they bind with another macromolecule which forces them into a fixed conformation.

[//]: # (TODO: Potentially add something about protein families and superfamilies here.)

#### The protein folding problem
As mentioned above, translated strings of amino acids fold automatically (or sometimes with the help of *chaperone* proteins) into the 3D structure of proteins. 
It’s the structure of proteins which define their function since their shape effects what they can bind to. 
Predicting what the structure of a protein will be from the string of amino acids is known as the protein folding problem. 
It’s has been one of the grandest challenges in biology for over half a century. If it were solved, it would revolutionise protein function and phenotype prediction. 

[//]: # (TODO: Mention google deep mind?)

---
**Page References**

```{bibliography} /_bibliography/references.bib
:filter: docname in docnames
:style: unsrt
```