
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>6.1. Background &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.c441f2ba0852f4cabcb80105e3a46ae6.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="6.2. Data Acquisition" href="2-data.html" />
    <link rel="prev" title="6. Combining RNA-seq datasets" href="0-index.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c0-front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-filter/0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/2-uberon.html">
     5.2. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/4-data.html">
     5.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/5-methods.html">
     5.4. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/6-results.html">
     5.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/7-discussion.html">
     5.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/8-future-work.html">
     5.7. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-batch-correction.html">
     6.4. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html">
     6.5. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html#combined-data-set">
     6.6. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     6.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="8-future-work.html">
     6.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#motivation-genotype-to-phenotype-predictions">
   6.1.1. Motivation: genotype to phenotype predictions
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#tissue-specific-gene-expression-data">
   6.1.2. Tissue-specific gene expression data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#challenges-in-combining-data-sets">
   6.1.3. Challenges in combining data sets
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#harmonising-meta-data">
     6.1.3.1. Harmonising meta-data
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#batch-effects">
     6.1.3.2. Batch effects
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#batch-effect-correction">
     6.1.3.3. Batch effect correction
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#combat">
       6.1.3.3.1. ComBat
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#pca-to-visualise-batch-effect-removal">
       6.1.3.3.2. PCA to visualise batch effect removal
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="background">
<h1><span class="section-number">6.1. </span>Background<a class="headerlink" href="#background" title="Permalink to this headline">¶</a></h1>
<p>There is general agreement that integrating omics datasets is one of the primary challenges to overcome if we wish to harness the full information contained within them<a href="#id1"><span class="problematic" id="id2">:cite:`Gomez-Cabrero2014-gk`</span></a>.
Combining gene expression data sets is a challenge in terms of data management, statistics, and harmonisation of meta-data for interoperability.</p>
<div class="section" id="motivation-genotype-to-phenotype-predictions">
<h2><span class="section-number">6.1.1. </span>Motivation: genotype to phenotype predictions<a class="headerlink" href="#motivation-genotype-to-phenotype-predictions" title="Permalink to this headline">¶</a></h2>
<p>The latter is crucial for downstream use as we’ll see in <a class="reference internal" href="../c05-filter/0-index.html#c06-filter"><span class="std std-ref">the next Chapter</span></a>.</p>
<p>Furthermore, having access to a (combined) data set with more repeated measurements, and more variation in samples would allow researchers to apply additional statistical techniques, and ask additional questions, by increasing the statistical power.
For example, tissue-specific data sets are often used to identify housekeeping genes, and to build models of gene regulatory networks.</p>
</div>
<div class="section" id="tissue-specific-gene-expression-data">
<h2><span class="section-number">6.1.2. </span>Tissue-specific gene expression data<a class="headerlink" href="#tissue-specific-gene-expression-data" title="Permalink to this headline">¶</a></h2>
<p>To answer these questions, we need gene expression data.
Falling costs and rapid advances in sequencing technologies have resulted in what many have described as a deluge of omics data<a href="#id3"><span class="problematic" id="id4">:cite:`Bell2009-uq`</span></a>.
And this includes huge amount of <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html#rna-measurements"><span class="std std-ref">gene expression data</span></a>, as demonstrated by the 3,564 studies and 112,225 assays currently available through the European Molecular Biology Laboratory’s (EMBL) Gene Expression Atlas (GxA) website <a href="#id5"><span class="problematic" id="id6">:cite:`Petryszak2016-je`</span></a>.</p>
<p>Yet each individual measure of expression is only a snapshot of what a gene can do.
It only tells us about the transcription of proteins at that one time, in that one sample.
Gene expression can also vary by tissue and cell type, individual organism<a href="#id7"><span class="problematic" id="id8">:cite:`Cortijo2019-wi`</span></a>, age<a href="#id9"><span class="problematic" id="id10">:cite:`vinuela2018age,Yang2015-mo`</span></a>, sex<a href="#id11"><span class="problematic" id="id12">:cite:`Reinius2009-ir,Stone2019-fk`</span></a>, time of day<a href="#id13"><span class="problematic" id="id14">:cite:`Cortijo2019-wi,Schibler2007-jx`</span></a>, and spatial location within a tissue or culture<a href="#id15"><span class="problematic" id="id16">:cite:`Svensson2018-oa`</span></a>.
There are also interactions between these different sources of variation, for example certain genes may only exhibit differential expression based on time of day in certain cell types.
If we want a full understanding of what a gene does, we must understand how it’s expressed in a variety of scenarios, for example, in different tissues, from different people, at different times of day, and across many repeats.</p>
<p>Many phenotypes are naturally linked to certain locations.
Humans have many diseases and features which are particular to certain locations on the body and certain tissues, be it blood, brain, skin, or lung.
So to answer the question of whether a protein is produced in a context relevant to a phenotype, we not only need gene expression information, but <em>tissue-specific</em> gene expression information: is this gene ever expressed in the heart?</p>
<p>For a typical (human) next-generation sequencing transcriptomics experiment, data is collected for over 20,000 genes, but generally far fewer samples, and very few replicates of a certain kind of sample (e.g. tissue).
For context, the largest experiment in the Ensembl Gene Expression Atlas (GxA) is the Genotype-Tissue Expression (GTEx) Project with 18736 samples.
It is currently too expensive for one experiment to measure enough samples to give us a comprehensive understanding of how genes can behave.
So, when it comes to gene expression, we have “big data” because we are measuring so many genes, but in any one experiment we don’t collect enough to have a deep understanding of how these genes behave in different contexts.</p>
</div>
<div class="section" id="challenges-in-combining-data-sets">
<h2><span class="section-number">6.1.3. </span>Challenges in combining data sets<a class="headerlink" href="#challenges-in-combining-data-sets" title="Permalink to this headline">¶</a></h2>
<p>Combining expression data from many different experiments has the potential to create a data set containing a more representative view of gene expression.
This has already been done for two experiments<a href="#id17"><span class="problematic" id="id18">:cite:`Wang2018-rz`</span></a>.</p>
<div class="section" id="harmonising-meta-data">
<h3><span class="section-number">6.1.3.1. </span>Harmonising meta-data<a class="headerlink" href="#harmonising-meta-data" title="Permalink to this headline">¶</a></h3>
<p>Data about samples was recorded at different levels of specificity.
This was a particular challenge for tissue where some samples are simply labelled “brain”, while others are labelled “medulla oblongata”, and yet others are identified by cell type.
Harmonising this cell and tissue meta-data was one of the challenges of combining this data set, which was done using the Uberon cross-species anatomy ontology<a href="#id19"><span class="problematic" id="id20">:cite:`Mungall2012-nc`</span></a>, and the Cell Ontology<a href="#id21"><span class="problematic" id="id22">:cite:`Malladi2015-iq`</span></a> (CL), which is integrated with Uberon.
Samples were primarily assigned Uberon term identifiers by searching for matching text between sample information files and CL or Uberon term names or descriptions.
Where existing terms did not turn up a match, samples were assigned an Uberon term by hand.
Then using the Uberon ontology, tissues could be understood in relation to each other, being mapped to tissues and more general tissue groups.</p>
</div>
<div class="section" id="batch-effects">
<h3><span class="section-number">6.1.3.2. </span>Batch effects<a class="headerlink" href="#batch-effects" title="Permalink to this headline">¶</a></h3>
<p>Combining gene expression data sets, however, is not trivial: a major problem is their well known susceptibility to batch effects (differences in measurements due to technical artefacts of sequencing batch)<a href="#id23"><span class="problematic" id="id24">:cite:`Leek2010-yw`</span></a>.
When combining and comparing gene expression data from two (or more) experiments, it’s not obvious how much of our signal comes from real biological differences in transcription, and how much comes from unwanted variation associated with the batch it was sequenced in, resulting from unknown variation during the process of sequencing for example the date, time, or location of sequencing<a href="#id25"><span class="problematic" id="id26">:cite:`Irizarry2005-ie`</span></a>, or the technician doing the work</p>
<p>To complicate matters, some batch effects may be due to factors that might be expected to genuinely influence expression of genes, such as temperature, time of year, humidity, diet, individual, age, etc.
Covariates such as these are often unrecorded and/or not reported, so it is not easy to distinguish these from those due to protocol differences, such as reagents, personnel doing the sequencing, hardware, processing pipeline, etc.
As such the problem of batch effects is closely related to the problem of recording sample metadata.</p>
<p>Batch effects can often confound and obscure the biological differences of interest between samples (e.g. tumour versus healthy tissue).
At best, batch effects add random variation to expression measurements, which obscure signals.
Often they can also add systematic differences that can lead to incorrect biological conclusions<a href="#id27"><span class="problematic" id="id28">:cite:`Leek2010-yw`</span></a>.
They are a problem for analysing the output of an individual experiment where there are multiple sequencing batches, but pose a particular problem in combining data from different experiments, as there is almost certainly more variations between analysis pipelines.</p>
</div>
<div class="section" id="batch-effect-correction">
<h3><span class="section-number">6.1.3.3. </span>Batch effect correction<a class="headerlink" href="#batch-effect-correction" title="Permalink to this headline">¶</a></h3>
<p>Batch effects may affect only specific subsets of genes, and may affect different genes in different ways<a href="#id29"><span class="problematic" id="id30">:cite:`Leek2010-yw`</span></a>. This means that <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html#rna-normalisation"><span class="std std-ref">normalisation</span></a> (e.g. TPM, FKPM) will not account for batch.
However, when it is known, date of sequence processing is often used as a surrogate for batch, enabling researchers to check for, and then remove, batch effects if necessary.</p>
<p>There are a number of batch correction analyses which attempt to remove batch effects from RNA-seq data, for example ComBat<a href="#id31"><span class="problematic" id="id32">:cite:`Johnson2007-zh`</span></a> and Surrogate Variable Analysis (SVA)<a href="#id33"><span class="problematic" id="id34">:cite:`Leek2007-ba`</span></a>
Batch correction can be very useful for understanding baseline gene expression, but can lead to inflated p-values for downstream analysis (notably for differential gene expression, using ComBat<a href="#id35"><span class="problematic" id="id36">:cite:`Johnson2007-zh`</span></a>), where a more sensible approach is to include batch as a confounder for statistical tests.</p>
<div class="section" id="combat">
<span id="combat-description"></span><h4><span class="section-number">6.1.3.3.1. </span>ComBat<a class="headerlink" href="#combat" title="Permalink to this headline">¶</a></h4>
<p>ComBat<a href="#id37"><span class="problematic" id="id38">:cite:`Johnson2007-zh`</span></a> is a popular batch effect removal procedure, which was first developed for use with microarray data, but continues to be a popular choice for RNA-seq data.
Generally, it is a well-trusted method for both of these types of gene expression data<a href="#id39"><span class="problematic" id="id40">:cite:`Chen2011-ke`</span></a>, although there is some evidence that it may “over-correct” batches for some RNA-seq data<a href="#id41"><span class="problematic" id="id42">:cite:`Liu2016-wa`</span></a>.</p>
<p>ComBat is an Empirical Bayes method, meaning that the prior distribution is estimated from the data.
It is designed to “borrow/share information” between genes in order to get a better estimate of batch effects, and assumes that batch effects affect many genes in similar ways.</p>
</div>
<div class="section" id="pca-to-visualise-batch-effect-removal">
<h4><span class="section-number">6.1.3.3.2. </span>PCA to visualise batch effect removal<a class="headerlink" href="#pca-to-visualise-batch-effect-removal" title="Permalink to this headline">¶</a></h4>
<p>Principal Components Analysis (PCA) is often used to visually inspect experimental results for batch effects; when biologically alike samples cluster together rather than those from like-batches, batch effects are often ignored.
In order to do this, we must also have enough meaningful information recorded per sample, access to data in raw count format, and mapping between the data that is given and similar samples in other datasets, as well as computational problems (data storage, optimisation of running operations on many or large files, etc).</p>
<p>Cell type is one of the better understood influences on gene expression.
We know that the same DNA is in every cell, and yet the morphology and function of each cell is determined by its cell type, due to its gene expression.
We can expect largely similar patterns of gene expression in similar cell types, which means that when we know cell type of samples, this information can be used to aid in visually checking the results of batch correction using PCA.</p>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c06-combining"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="0-index.html" title="previous page"><span class="section-number">6. </span>Combining RNA-seq datasets</a>
    <a class='right-next' id="next-link" href="2-data.html" title="next page"><span class="section-number">6.2. </span>Data Acquisition</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>