

<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Bibliography &#8212; Phenotype from Genotype</title>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css" integrity="sha384-KA6wR/X5RY4zFAHpv/CnoG2UW1uogYfdnP67Uv7eULvTveboZJg0qUpmJZb5VqzN" crossorigin="anonymous">
    <link href="../_static/css/index.css" rel="stylesheet">
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-dropdown.css" />
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/sphinx-book-theme.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="8. Conclusion" href="../c07-conclusion/0-conclusion.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="docsearch:language" content="en">



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div>

<form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>

<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../jupyter_book_intro.html">
   Start Here
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/title-page.html">
   Title Page
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-figures.html">
   List of Figures
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-tables.html">
   List of Tables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/declaration.html">
   Declaration
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c02-biology-background/0-bg-biology.html">
   2. Biology Background
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c02-biology-background/0-bg-biology.html#id18">
   3. Biology Background
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c03-compbio-background/0-compbio-bg.html">
   4. Computational Biology Background
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c04-snowflake/0-snowflake.html">
   5. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c05-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c06-filter/0-filter.html">
   7. Filtering computational predictions with tissue-specific expression information
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   8. Conclusion
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Bibliography
  </a>
 </li>
</ul>

</nav>

 <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        <div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    
    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/end-matter/reference.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
    
</div>
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            
        </nav>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="bibliography">
<h1>Bibliography<a class="headerlink" href="#bibliography" title="Permalink to this headline">¶</a></h1>
<p>References are shown at the bottom of the page on which they are cited. A full bibliography (of the entire Jupyter Book) appears below:</p>
<hr class="docutils" />
<p id="bibtex-bibliography-end-matter/reference-0"><dl class="citation">
<dt class="bibtex label" id="zaucha2015-ez"><span class="brackets">1</span></dt>
<dd><p>Jan Zaucha, Jonathan Stahlhacke, Matt E Oates, Natalie Thurlby, Owen J L Rackham, Hai Fang, Ben Smithers, and Julian Gough. A proteome quality index. <em>Environ. Microbiol.</em>, 17(1):4–9, 2015. URL: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.12622">https://onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.12622</a>.</p>
</dd>
<dt class="bibtex label" id="oates2015-li"><span class="brackets">2</span></dt>
<dd><p>Matt E Oates, Jonathan Stahlhacke, Dimitrios V Vavoulis, Ben Smithers, Owen J L Rackham, Adam J Sardar, Jan Zaucha, Natalie Thurlby, Hai Fang, and Julian Gough. The SUPERFAMILY 1.75 database in 2014: a doubling of data. <em>Nucleic Acids Res.</em>, 43(Database issue):D227–33, January 2015.</p>
</dd>
<dt class="bibtex label" id="gough2017-ik"><span class="brackets">3</span></dt>
<dd><p>Julian Gough, Jan Zaucha, and Natalie Thurlby. Determining phenotype from genotype. July 2017.</p>
</dd>
<dt class="bibtex label" id="zhou2019-jk"><span class="brackets">4</span></dt>
<dd><p>Naihui Zhou, Yuxiang Jiang, Timothy R Bergquist, Alexandra J Lee, Balint Z Kacsoh, Alex W Crocker, Kimberley A Lewis, George Georghiou, Huy N Nguyen, Md Nafiz Hamid, Larry Davis, Tunca Dogan, Volkan Atalay, Ahmet S Rifaioglu, Alperen Dalkiran, Rengul Cetin-Atalay, Chengxin Zhang, Rebecca L Hurto, Peter L Freddolino, Yang Zhang, Prajwal Bhat, Fran Supek, José M Fernández, Branislava Gemovic, Vladimir R Perovic, Radoslav S Davidović, Neven Sumonja, Nevena Veljkovic, Ehsaneddin Asgari, Mohammad R K Mofrad, Giuseppe Profiti, Castrense Savojardo, Pier Luigi Martelli, Rita Casadio, Florian Boecker, Indika Kahanda, Natalie Thurlby, Alice C McHardy, Alexandre Renaux, Rabie Saidi, Julian Gough, Alex A Freitas, Magdalena Antczak, Fabio Fabris, Mark N Wass, Jie Hou, Jianlin Cheng, Zheng Wang, Alfonso E Romero, Alberto Paccanaro, Haixuan Yang, Tatyana Goldberg, Chenguang Zhao, Liisa Holm, Petri Törönen, Alan J Medlar, Elaine Zosa, Itamar Borukhov, Ilya Novikov, Angela Wilkins, Olivier Lichtarge, Po-Han Chi, Wei-Cheng Tseng, Michal Linial, Peter W Rose, Christophe Dessimoz, Vedrana Vidulin, Saso Dzeroski, Ian Sillitoe, Sayoni Das, Jonathan Gill Lees, David T Jones, Cen Wan, Domenico Cozzetto, Rui Fa, Mateo Torres, Alex Wiarwick Vesztrocy, Jose Manuel Rodriguez, Michael L Tress, Marco Frasca, Marco Notaro, Giuliano Grossi, Alessandro Petrini, Matteo Re, Giorgio Valentini, Marco Mesiti, Daniel B Roche, Jonas Reeb, David W Ritchie, Sabeur Aridhi, Seyed Ziaeddin Alborzi, Marie-Dominique Devignes, Koo Da Chen Emily, Richard Bonneau, Vladimir Gligorijević, Meet Barot, Hai Fang, Stefano Toppo, Enrico Lavezzo, Marco Falda, Michele Berselli, Silvio C E Tosatto, Marco Carraro, Damiano Piovesan, Hafeez Ur Rehman, Qizhong Mao, Shanshan Zhang, Slobodan Vucetic, Gage S Black, Dane Jo, Dallas J Larsen, Ashton R Omdahl, Luke W Sagers, Erica Suh, Jonathan B Dayton, Liam J McGuffin, Danielle A Brackenridge, Patricia C Babbitt, Jeffrey M Yunes, Paolo Fontana, Feng Zhang, Shanfeng Zhu, Ronghui You, Zihan Zhang, Suyang Dai, Shuwei Yao, Weidong Tian, Renzhi Cao, Caleb Chandler, Miguel Amezola, Devon Johnson, Jia-Ming Chang, Wen-Hung Liao, Yi-Wei Liu, Stefano Pascarelli, Yotam Frank, Robert Hoehndorf, Maxat Kulmanov, Imane Boudellioua, Gianfranco Politano, Stefano Di Carlo, Alfredo Benso, Kai Hakala, Filip Ginter, Farrokh Mehryary, Suwisa Kaewphan, Jari Björne, Hans Moen, Martti E E Tolvanen, Tapio Salakoski, Daisuke Kihara, Aashish Jain, Tomislav Šmuc, Adrian Altenhoff, Asa Ben-Hur, Burkhard Rost, Steven E Brenner, Christine A Orengo, Constance J Jeffery, Giovanni Bosco, Deborah A Hogan, Maria J Martin, Claire O’Donovan, Sean D Mooney, Casey S Greene, Predrag Radivojac, and Iddo Friedberg. The CAFA challenge reports improved protein function prediction and new functional annotations for hundreds of genes through experimental screens. <em>bioRxiv</em>, pages 653105, May 2019.</p>
</dd>
<dt class="bibtex label" id="zirkle1935-sb"><span class="brackets">5</span></dt>
<dd><p>Conway Zirkle. The inheritance of acquired characters and the provisional hypothesis of pangenesis. <em>Am. Nat.</em>, 69(724):417–445, September 1935.</p>
</dd>
<dt class="bibtex label" id="grafton-undated-af"><span class="brackets">6</span></dt>
<dd><p>Anthony Grafton and Nancy G Siraisi. Natural particulars: nature and the disciplines in renaissance europe. In <em>acls humanities e-book</em>. MPublishing, University of Michigan Library, 1999.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ih"><span class="brackets">7</span></dt>
<dd><p>Gallery 19: DNA model, 1953 :: DNA learning center. <span><a class="reference external" href="#"></a></span>https://www.dnalc.org/view/16430-Gallery-19-DNA-model-1953.html. Accessed: 2019-6-2.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ic"><span class="brackets">8</span></dt>
<dd><p>Amino acid codon wheel. <span><a class="reference external" href="#"></a></span>https://www.sigmaaldrich.com/technical-documents/articles/biology/amino-acid-codon-wheel.html. Accessed: 2019-6-2.</p>
</dd>
<dt class="bibtex label" id="salzberg2018-yc"><span class="brackets">9</span></dt>
<dd><p>Steven L Salzberg. Open questions: how many genes do we have? <em>BMC Biol.</em>, 16(1):94, August 2018.</p>
</dd>
<dt class="bibtex label" id="mulder1839-rf"><span class="brackets">10</span></dt>
<dd><p>G J Mulder. Ueber die zusammensetzung einiger thierischen substanzen. <em>J. Prakt. Chem.</em>, 1839.</p>
</dd>
<dt class="bibtex label" id="hersh1967-ox"><span class="brackets">11</span></dt>
<dd><p>Robert T Hersh. Atlas of protein sequence and structure, 1966. <em>Syst. Biol.</em>, 16(3):262–263, September 1967.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ow"><span class="brackets">12</span></dt>
<dd><p>Protein data bank.</p>
</dd>
<dt class="bibtex label" id="jarraud2019-jj"><span class="brackets">13</span></dt>
<dd><p>Nicolas Jarraud. Mecabricks. <span><a class="reference external" href="#"></a></span>https://mecabricks.com/, 2019.</p>
</dd>
<dt class="bibtex label" id="venter2001-wn"><span class="brackets">14</span></dt>
<dd><p>J C Venter, M D Adams, E W Myers, P W Li, R J Mural, G G Sutton, H O Smith, M Yandell, C A Evans, R A Holt, J D Gocayne, P Amanatides, R M Ballew, D H Huson, J R Wortman, Q Zhang, C D Kodira, X H Zheng, L Chen, M Skupski, G Subramanian, P D Thomas, J Zhang, G L Gabor Miklos, C Nelson, S Broder, A G Clark, J Nadeau, V A McKusick, N Zinder, A J Levine, R J Roberts, M Simon, C Slayman, M Hunkapiller, R Bolanos, A Delcher, I Dew, D Fasulo, M Flanigan, L Florea, A Halpern, S Hannenhalli, S Kravitz, S Levy, C Mobarry, K Reinert, K Remington, J Abu-Threideh, E Beasley, K Biddick, V Bonazzi, R Brandon, M Cargill, I Chandramouliswaran, R Charlab, K Chaturvedi, Z Deng, V Di Francesco, P Dunn, K Eilbeck, C Evangelista, A E Gabrielian, W Gan, W Ge, F Gong, Z Gu, P Guan, T J Heiman, M E Higgins, R R Ji, Z Ke, K A Ketchum, Z Lai, Y Lei, Z Li, J Li, Y Liang, X Lin, F Lu, G V Merkulov, N Milshina, H M Moore, A K Naik, V A Narayan, B Neelam, D Nusskern, D B Rusch, S Salzberg, W Shao, B Shue, J Sun, Z Wang, A Wang, X Wang, J Wang, M Wei, R Wides, C Xiao, C Yan, A Yao, J Ye, M Zhan, W Zhang, H Zhang, Q Zhao, L Zheng, F Zhong, W Zhong, S Zhu, S Zhao, D Gilbert, S Baumhueter, G Spier, C Carter, A Cravchik, T Woodage, F Ali, H An, A Awe, D Baldwin, H Baden, M Barnstead, I Barrow, K Beeson, D Busam, A Carver, A Center, M L Cheng, L Curry, S Danaher, L Davenport, R Desilets, S Dietz, K Dodson, L Doup, S Ferriera, N Garg, A Gluecksmann, B Hart, J Haynes, C Haynes, C Heiner, S Hladun, D Hostin, J Houck, T Howland, C Ibegwam, J Johnson, F Kalush, L Kline, S Koduru, A Love, F Mann, D May, S McCawley, T McIntosh, I McMullen, M Moy, L Moy, B Murphy, K Nelson, C Pfannkoch, E Pratts, V Puri, H Qureshi, M Reardon, R Rodriguez, Y H Rogers, D Romblad, B Ruhfel, R Scott, C Sitter, M Smallwood, E Stewart, R Strong, E Suh, R Thomas, N N Tint, S Tse, C Vech, G Wang, J Wetter, S Williams, M Williams, S Windsor, E Winn-Deen, K Wolfe, J Zaveri, K Zaveri, J F Abril, R Guigó, M J Campbell, K V Sjolander, B Karlak, A Kejariwal, H Mi, B Lazareva, T Hatton, A Narechania, K Diemer, A Muruganujan, N Guo, S Sato, V Bafna, S Istrail, R Lippert, R Schwartz, B Walenz, S Yooseph, D Allen, A Basu, J Baxendale, L Blick, M Caminha, J Carnes-Stine, P Caulk, Y H Chiang, M Coyne, C Dahlke, A Mays, M Dombroski, M Donnelly, D Ely, S Esparham, C Fosler, H Gire, S Glanowski, K Glasser, A Glodek, M Gorokhov, K Graham, B Gropman, M Harris, J Heil, S Henderson, J Hoover, D Jennings, C Jordan, J Jordan, J Kasha, L Kagan, C Kraft, A Levitsky, M Lewis, X Liu, J Lopez, D Ma, W Majoros, J McDaniel, S Murphy, M Newman, T Nguyen, N Nguyen, M Nodell, S Pan, J Peck, M Peterson, W Rowe, R Sanders, J Scott, M Simpson, T Smith, A Sprague, T Stockwell, R Turner, E Venter, M Wang, M Wen, D Wu, M Wu, A Xia, A Zandieh, and X Zhu. The sequence of the human genome. <em>Science</em>, 291(5507):1304–1351, February 2001.</p>
</dd>
<dt class="bibtex label" id="bumgarner2013-hg"><span class="brackets">15</span></dt>
<dd><p>Roger Bumgarner. Overview of DNA microarrays: types, applications, and their future. <em>Curr. Protoc. Mol. Biol.</em>, Chapter 22:Unit 22.1., January 2013.</p>
</dd>
<dt class="bibtex label" id="haas2010-lm"><span class="brackets">16</span></dt>
<dd><p>Brian J Haas and Michael C Zody. Advancing RNA-Seq analysis. <em>Nat. Biotechnol.</em>, 28(5):421–423, May 2010.</p>
</dd>
<dt class="bibtex label" id="pfeiffer2018-kt"><span class="brackets">17</span></dt>
<dd><p>Franziska Pfeiffer, Carsten Gröber, Michael Blank, Kristian Händler, Marc Beyer, Joachim L Schultze, and Günter Mayer. Systematic evaluation of error rates and causes in short samples in next-generation sequencing. <em>Sci. Rep.</em>, 8(1):10950, July 2018.</p>
</dd>
<dt class="bibtex label" id="de-koning2011-ac"><span class="brackets">18</span></dt>
<dd><p>A P Jason de Koning, Wanjun Gu, Todd A Castoe, Mark A Batzer, and David D Pollock. Repetitive elements may comprise over two-thirds of the human genome. <em>PLoS Genet.</em>, 7(12):e1002384, December 2011.</p>
</dd>
<dt class="bibtex label" id="buniello2019-cv"><span class="brackets">19</span></dt>
<dd><p>Annalisa Buniello, Jacqueline A L MacArthur, Maria Cerezo, Laura W Harris, James Hayhurst, Cinzia Malangone, Aoife McMahon, Joannella Morales, Edward Mountjoy, Elliot Sollis, Daniel Suveges, Olga Vrousgou, Patricia L Whetzel, Ridwan Amode, Jose A Guillen, Harpreet S Riat, Stephen J Trevanion, Peggy Hall, Heather Junkins, Paul Flicek, Tony Burdett, Lucia A Hindorff, Fiona Cunningham, and Helen Parkinson. The NHGRI-EBI GWAS catalog of published genome-wide association studies, targeted arrays and summary statistics 2019. <em>Nucleic Acids Res.</em>, 47(D1):D1005–D1012, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gky1120">http://dx.doi.org/10.1093/nar/gky1120</a>.</p>
</dd>
<dt class="bibtex label" id="l-emery2017-rd"><span class="brackets">20</span></dt>
<dd><p>J Macarthur L. Emery. GWAS catalog: exploring SNP-trait associations. <span><a class="reference external" href="#"></a></span>http://europepmc.org/article/CTX/C7914, December 2017. Accessed: 2020-9-3. URL: <a class="reference external" href="http://europepmc.org/article/CTX/C7914">http://europepmc.org/article/CTX/C7914</a>.</p>
</dd>
<dt class="bibtex label" id="barbaric2007-zm"><span class="brackets">4</span></dt>
<dd><p>Ivana Barbaric, Gaynor Miller, and T Neil Dear. Appearances can be deceiving: phenotypes of knockout mice. <em>Brief. Funct. Genomic. Proteomic.</em>, 6(2):91–103, June 2007.</p>
</dd>
<dt class="bibtex label" id="jensen2010-fk"><span class="brackets">5</span></dt>
<dd><p>Lars J Jensen and Peer Bork. Ontologies in quantitative biology: a basis for comparison, integration, and discovery. <em>PLoS Biol.</em>, 8(5):e1000374, May 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1000374">http://dx.doi.org/10.1371/journal.pbio.1000374</a>.</p>
</dd>
<dt class="bibtex label" id="blunt2001-wr"><span class="brackets">6</span></dt>
<dd><p>Wilfrid Blunt. <em>The Compleat Naturalist: A Life of Linnaeus</em>. Frances Lincoln, 2001. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=B3YOvgAACAAJ">https://play.google.com/store/books/details?id=B3YOvgAACAAJ</a>.</p>
</dd>
<dt class="bibtex label" id="ehret1748-bx"><span class="brackets">7</span></dt>
<dd><p>Ehret. <em>Plantae et papiliones rariores</em>. Volume 1748. [London :s.n.],, 1748. URL: <a class="reference external" href="https://www.biodiversitylibrary.org/item/205762">https://www.biodiversitylibrary.org/item/205762</a>.</p>
</dd>
<dt class="bibtex label" id="overton2015-vo"><span class="brackets">8</span></dt>
<dd><p>James A Overton, Heiko Dietze, Shahim Essaid, David Osumi-Sutherland, and Christopher J Mungall. ROBOT: a command-line tool for ontology development. In <em>ICBO</em>. ceur-ws.org, 2015. URL: <a class="reference external" href="http://ceur-ws.org/Vol-1515/demo6.pdf">http://ceur-ws.org/Vol-1515/demo6.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="fang2013-et"><span class="brackets">9</span></dt>
<dd><p>Hai Fang, Matt E Oates, Ralph B Pethica, Jenny M Greenwood, Adam J Sardar, Owen J L Rackham, Philip C J Donoghue, Alexandros Stamatakis, David A de Lima Morais, and Julian Gough. A daily-updated tree of (sequenced) life as a reference for genome research. <em>Sci. Rep.</em>, 3:2015, 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/srep02015">http://dx.doi.org/10.1038/srep02015</a>.</p>
</dd>
<dt class="bibtex label" id="cracraft2004-ud"><span class="brackets">10</span></dt>
<dd><p>Joel Cracraft and Michael J Donoghue. <em>Assembling the Tree of Life</em>. Oxford University Press, July 2004. URL: <a class="reference external" href="https://www.amazon.co.uk/Assembling-Tree-Life-Joel-Cracraft/dp/0195172345">https://www.amazon.co.uk/Assembling-Tree-Life-Joel-Cracraft/dp/0195172345</a>.</p>
</dd>
<dt class="bibtex label" id="dohmen2016-iv"><span class="brackets">11</span></dt>
<dd><p>Elias Dohmen, Lukas P M Kremer, Erich Bornberg-Bauer, and Carsten Kemena. DOGMA: domain-based transcriptome and proteome quality assessment. <em>Bioinformatics</em>, 32(17):2577–2581, September 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btw231">http://dx.doi.org/10.1093/bioinformatics/btw231</a>.</p>
</dd>
<dt class="bibtex label" id="simao2015-gc"><span class="brackets">12</span></dt>
<dd><p>Felipe A Simão, Robert M Waterhouse, Panagiotis Ioannidis, Evgenia V Kriventseva, and Evgeny M Zdobnov. BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs. <em>Bioinformatics</em>, 31(19):3210–3212, October 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btv351">http://dx.doi.org/10.1093/bioinformatics/btv351</a>.</p>
</dd>
<dt class="bibtex label" id="kitts2016-eo"><span class="brackets">13</span></dt>
<dd><p>Paul A Kitts, Deanna M Church, Françoise Thibaud-Nissen, Jinna Choi, Vichet Hem, Victor Sapojnikov, Robert G Smith, Tatiana Tatusova, Charlie Xiang, Andrey Zherikov, Michael DiCuccio, Terence D Murphy, Kim D Pruitt, and Avi Kimchi. Assembly: a resource for assembled genomes at NCBI. <em>Nucleic Acids Res.</em>, 44(D1):D73–80, January 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv1226">http://dx.doi.org/10.1093/nar/gkv1226</a>.</p>
</dd>
<dt class="bibtex label" id="sims2014-df"><span class="brackets">14</span></dt>
<dd><p>David Sims, Ian Sudbery, Nicholas E Ilott, Andreas Heger, and Chris P Ponting. Sequencing depth and coverage: key considerations in genomic analyses. <em>Nat. Rev. Genet.</em>, 15(2):121–132, February 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3642">http://dx.doi.org/10.1038/nrg3642</a>.</p>
</dd>
<dt class="bibtex label" id="watson2019-pz"><span class="brackets">15</span></dt>
<dd><p>Mick Watson and Amanda Warr. Errors in long-read assemblies can critically affect protein prediction. <em>Nat. Biotechnol.</em>, 37(2):124–126, February 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41587-018-0004-z">http://dx.doi.org/10.1038/s41587-018-0004-z</a>.</p>
</dd>
<dt class="bibtex label" id="eden2009-ic"><span class="brackets">16</span></dt>
<dd><p>Eran Eden, Roy Navon, Israel Steinfeld, Doron Lipson, and Zohar Yakhini. GOrilla: a tool for discovery and visualization of enriched GO terms in ranked gene lists. <em>BMC Bioinformatics</em>, 10:48, February 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-10-48">http://dx.doi.org/10.1186/1471-2105-10-48</a>.</p>
</dd>
<dt class="bibtex label" id="ashburner2000-cr"><span class="brackets">17</span></dt>
<dd><p>M Ashburner, C A Ball, J A Blake, D Botstein, H Butler, J M Cherry, A P Davis, K Dolinski, S S Dwight, J T Eppig, M A Harris, D P Hill, L Issel-Tarver, A Kasarskis, S Lewis, J C Matese, J E Richardson, M Ringwald, G M Rubin, and G Sherlock. Gene ontology: tool for the unification of biology. the gene ontology consortium. <em>Nat. Genet.</em>, 25(1):25–29, May 2000. URL: <a class="reference external" href="http://dx.doi.org/10.1038/75556">http://dx.doi.org/10.1038/75556</a>.</p>
</dd>
<dt class="bibtex label" id="thomas2017-vm"><span class="brackets">18</span></dt>
<dd><p>Paul D Thomas. The gene ontology and the meaning of biological function. <em>Methods Mol. Biol.</em>, 1446:15–24, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1007/978-1-4939-3743-1_2">http://dx.doi.org/10.1007/978-1-4939-3743-1_2</a>.</p>
</dd>
<dt class="bibtex label" id="sherry2001-nm"><span class="brackets">19</span></dt>
<dd><p>S T Sherry. dbSNP: the NCBI database of genetic variation. <em>Nucleic Acids Res.</em>, 29(1):308–311, 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/29.1.308">http://dx.doi.org/10.1093/nar/29.1.308</a>.</p>
</dd>
<dt class="bibtex label" id="petryszak2016-je"><span class="brackets">20</span></dt>
<dd><p>Robert Petryszak, Maria Keays, Y Amy Tang, Nuno A Fonseca, Elisabet Barrera, Tony Burdett, Anja Füllgrabe, Alfonso Muñoz-Pomer Fuentes, Simon Jupp, Satu Koskinen, Oliver Mannion, Laura Huerta, Karine Megy, Catherine Snow, Eleanor Williams, Mitra Barzine, Emma Hastings, Hendrik Weisser, James Wright, Pankaj Jaiswal, Wolfgang Huber, Jyoti Choudhary, Helen E Parkinson, and Alvis Brazma. Expression atlas update—an integrated database of gene and protein expression in humans, animals and plants. 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv1045">http://dx.doi.org/10.1093/nar/gkv1045</a>.</p>
</dd>
<dt class="bibtex label" id="illumina2014-nd"><span class="brackets">21</span></dt>
<dd><p>I Illumina. Understanding illumina quality scores. <em>Technical Note: Informatics</em>, 2014. URL: <a class="reference external" href="https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf">https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="wagner2012-ac"><span class="brackets">22</span></dt>
<dd><p>Günter P Wagner, Koryu Kin, and Vincent J Lynch. Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples. <em>Theory Biosci.</em>, 131(4):281–285, December 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s12064-012-0162-3">http://dx.doi.org/10.1007/s12064-012-0162-3</a>.</p>
</dd>
<dt class="bibtex label" id="pimentel2014-xm"><span class="brackets">23</span></dt>
<dd><p>H Pimentel. What the FPKM? a review of RNA-Seq expression units. 2014. URL: <a class="reference external" href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/</a>.</p>
</dd>
<dt class="bibtex label" id="robinson2010-kl"><span class="brackets">24</span></dt>
<dd><p>M D Robinson, D J McCarthy, and G K Smyth. Edger: a bioconductor package for differential expression analysis of digital gene expression data. <em>Bioinformatics</em>, 26(1):139–140, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btp616">http://dx.doi.org/10.1093/bioinformatics/btp616</a>.</p>
</dd>
<dt class="bibtex label" id="love2014-vx"><span class="brackets">25</span></dt>
<dd><p>Michael I Love, Wolfgang Huber, and Simon Anders. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. <em>Genome Biol.</em>, 15(12):550, 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-014-0550-8">http://dx.doi.org/10.1186/s13059-014-0550-8</a>.</p>
</dd>
<dt class="bibtex label" id="leek2010-yw"><span class="brackets">26</span></dt>
<dd><p>Jeffrey T Leek, Robert B Scharpf, Héctor Corrada Bravo, David Simcha, Benjamin Langmead, W Evan Johnson, Donald Geman, Keith Baggerly, and Rafael A Irizarry. Tackling the widespread and critical impact of batch effects in high-throughput data. <em>Nat. Rev. Genet.</em>, 11(10):733–739, October 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg2825">http://dx.doi.org/10.1038/nrg2825</a>.</p>
</dd>
<dt class="bibtex label" id="irizarry2005-ie"><span class="brackets">27</span></dt>
<dd><p>Rafael A Irizarry, Daniel Warren, Forrest Spencer, Irene F Kim, Shyam Biswal, Bryan C Frank, Edward Gabrielson, Joe G N Garcia, Joel Geoghegan, Gregory Germino, Constance Griffin, Sara C Hilmer, Eric Hoffman, Anne E Jedlicka, Ernest Kawasaki, Francisco Martínez-Murillo, Laura Morsberger, Hannah Lee, David Petersen, John Quackenbush, Alan Scott, Michael Wilson, Yanqin Yang, Shui Qing Ye, and Wayne Yu. Multiple-laboratory comparison of microarray platforms. <em>Nat. Methods</em>, 2(5):345–350, May 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth756">http://dx.doi.org/10.1038/nmeth756</a>.</p>
</dd>
<dt class="bibtex label" id="altschul1990-zf"><span class="brackets">28</span></dt>
<dd><p>S F Altschul, W Gish, W Miller, E W Myers, and D J Lipman. Basic local alignment search tool. <em>J. Mol. Biol.</em>, 215(3):403–410, October 1990. URL: <a class="reference external" href="http://dx.doi.org/10.1016/S0022-2836(05)80360-2">http://dx.doi.org/10.1016/S0022-2836(05)80360-2</a>.</p>
</dd>
<dt class="bibtex label" id="murzin1995-se"><span class="brackets">29</span></dt>
<dd><p>A G Murzin, S E Brenner, T Hubbard, and C Chothia. SCOP: a structural classification of proteins database for the investigation of sequences and structures. <em>J. Mol. Biol.</em>, 247(4):536–540, April 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.1995.0159">http://dx.doi.org/10.1006/jmbi.1995.0159</a>.</p>
</dd>
<dt class="bibtex label" id="andreeva2014-om"><span class="brackets">30</span></dt>
<dd><p>Antonina Andreeva, Dave Howorth, Cyrus Chothia, Eugene Kulesha, and Alexey G Murzin. SCOP2 prototype: a new approach to protein structure mining. <em>Nucleic Acids Res.</em>, 42(Database issue):D310–4, January 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkt1242">http://dx.doi.org/10.1093/nar/gkt1242</a>.</p>
</dd>
<dt class="bibtex label" id="orengo1997-vf"><span class="brackets">31</span></dt>
<dd><p>C A Orengo, A D Michie, S Jones, D T Jones, M B Swindells, and J M Thornton. CATH – a hierarchic classification of protein domain structures. <em>Structure</em>, 5(8):1093–1109, 1997. URL: <a class="reference external" href="http://dx.doi.org/10.1016/s0969-2126(97)00260-8">http://dx.doi.org/10.1016/s0969-2126(97)00260-8</a>.</p>
</dd>
<dt class="bibtex label" id="csaba2009-of"><span class="brackets">32</span></dt>
<dd><p>Gergely Csaba, Fabian Birzele, and Ralf Zimmer. Systematic comparison of SCOP and CATH: a new gold standard for protein structure analysis. <em>BMC Struct. Biol.</em>, 9:23, April 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1472-6807-9-23">http://dx.doi.org/10.1186/1472-6807-9-23</a>.</p>
</dd>
<dt class="bibtex label" id="gough2001-ct"><span class="brackets">33</span></dt>
<dd><p>J Gough, K Karplus, R Hughey, and C Chothia. Assignment of homology to genome sequences using a library of hidden markov models that represent all proteins of known structure. <em>J. Mol. Biol.</em>, 313(4):903–919, November 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.2001.5080">http://dx.doi.org/10.1006/jmbi.2001.5080</a>.</p>
</dd>
<dt class="bibtex label" id="fang2013-ms"><span class="brackets">34</span></dt>
<dd><p>Hai Fang and Julian Gough. DcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more. <em>Nucleic Acids Res.</em>, 41(Database issue):D536–44, January 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gks1080">http://dx.doi.org/10.1093/nar/gks1080</a>.</p>
</dd>
<dt class="bibtex label" id="fang2013-ix"><span class="brackets">35</span></dt>
<dd><p>Hai Fang and Julian Gough. A domain-centric solution to functional genomics via dcGO predictor. <em>BMC Bioinformatics</em>, 14 Suppl 3:S9, February 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-14-S3-S9">http://dx.doi.org/10.1186/1471-2105-14-S3-S9</a>.</p>
</dd>
<dt class="bibtex label" id="shihab2013-pk"><span class="brackets">36</span></dt>
<dd><p>Hashem A Shihab, Julian Gough, David N Cooper, Peter D Stenson, Gary L A Barker, Keith J Edwards, Ian N M Day, and Tom R Gaunt. Predicting the functional, molecular, and phenotypic consequences of amino acid substitutions using hidden markov models. <em>Hum. Mutat.</em>, 34(1):57–65, January 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1002/humu.22225">http://dx.doi.org/10.1002/humu.22225</a>.</p>
</dd>
<dt class="bibtex label" id="stenson2009-jc"><span class="brackets">37</span></dt>
<dd><p>Peter D Stenson, Matthew Mort, Edward V Ball, Katy Howells, Andrew D Phillips, Nick St Thomas, and David N Cooper. The human gene mutation database: 2008 update. <em>Genome Med.</em>, 1(1):13, January 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gm13">http://dx.doi.org/10.1186/gm13</a>.</p>
</dd>
<dt class="bibtex label" id="pundir2016-ya"><span class="brackets">38</span></dt>
<dd><p>Sangya Pundir, Maria J Martin, Claire O’Donovan, and The UniProt Consortium. UniProt tools. <em>Curr. Protoc. Bioinformatics</em>, pages 1.29.1–1.29.15, 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1002/0471250953.bi0129s53">http://dx.doi.org/10.1002/0471250953.bi0129s53</a>.</p>
</dd>
<dt class="bibtex label" id="mclaren2016-di"><span class="brackets">39</span></dt>
<dd><p>William McLaren, Laurent Gil, Sarah E Hunt, Harpreet Singh Riat, Graham R S Ritchie, Anja Thormann, Paul Flicek, and Fiona Cunningham. The ensembl variant effect predictor. <em>Genome Biol.</em>, 17(1):122, June 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-016-0974-4">http://dx.doi.org/10.1186/s13059-016-0974-4</a>.</p>
</dd>
<dt class="bibtex label" id="begley2012-oc"><span class="brackets">40</span></dt>
<dd><p>C Glenn Begley and Lee M Ellis. Drug development: raise standards for preclinical cancer research. <em>Nature</em>, 483(7391):531–533, March 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1038/483531a">http://dx.doi.org/10.1038/483531a</a>.</p>
</dd>
<dt class="bibtex label" id="prinz2011-rh"><span class="brackets">41</span></dt>
<dd><p>Florian Prinz, Thomas Schlange, and Khusru Asadullah. Believe it or not: how much can we rely on published data on potential drug targets? <em>Nat. Rev. Drug Discov.</em>, 10(9):712–712, 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrd3439-c1">http://dx.doi.org/10.1038/nrd3439-c1</a>.</p>
</dd>
<dt class="bibtex label" id="open-science-collaboration2015-vi"><span class="brackets">42</span></dt>
<dd><p>Open Science Collaboration. PSYCHOLOGY. estimating the reproducibility of psychological science. <em>Science</em>, 349(6251):aac4716, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.aac4716">http://dx.doi.org/10.1126/science.aac4716</a>.</p>
</dd>
<dt class="bibtex label" id="baker2016-tb"><span class="brackets">43</span></dt>
<dd><p>Monya Baker. 1,500 scientists lift the lid on reproducibility. <em>Nature</em>, 533(7604):452–454, May 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/533452a">http://dx.doi.org/10.1038/533452a</a>.</p>
</dd>
<dt class="bibtex label" id="bishop2019-hg"><span class="brackets">44</span></dt>
<dd><p>Dorothy Bishop. Rein in the four horsemen of irreproducibility. <em>Nature</em>, 568(7753):435, April 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/d41586-019-01307-2">http://dx.doi.org/10.1038/d41586-019-01307-2</a>.</p>
</dd>
<dt class="bibtex label" id="ioannidis2005-mo"><span class="brackets">45</span></dt>
<dd><p>John P A Ioannidis. Why most published research findings are false. <em>PLoS Med.</em>, 2(8):e124, August 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pmed.0020124">http://dx.doi.org/10.1371/journal.pmed.0020124</a>.</p>
</dd>
<dt class="bibtex label" id="head2015-ns"><span class="brackets">46</span></dt>
<dd><p>Megan L Head, Luke Holman, Rob Lanfear, Andrew T Kahn, and Michael D Jennions. The extent and consequences of p-hacking in science. <em>PLoS Biol.</em>, 13(3):e1002106, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1002106">http://dx.doi.org/10.1371/journal.pbio.1002106</a>.</p>
</dd>
<dt class="bibtex label" id="fisher1990-ro"><span class="brackets">47</span></dt>
<dd><p>R A Fisher and F Yates. <em>Statistical Methods, Experimental Design, and Scientific Inference: A Re-Issue of Statistical Methods for Research Workers, the Design of Experiments, and Statistical Methods and Scientific Inference</em>. OUP Oxford, April 1990. URL: <a class="reference external" href="https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290">https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290</a>.</p>
</dd>
<dt class="bibtex label" id="dunn1958-sj"><span class="brackets">48</span></dt>
<dd><p>Olive Jean Dunn. Estimation of the means of dependent variables. <em>The Annals of Mathematical Statistics</em>, 29(4):1095–1111, 1958. URL: <a class="reference external" href="http://dx.doi.org/10.1214/aoms/1177706443">http://dx.doi.org/10.1214/aoms/1177706443</a>.</p>
</dd>
<dt class="bibtex label" id="benjamini1995-me"><span class="brackets">49</span></dt>
<dd><p>Yoav Benjamini and Yosef Hochberg. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal Statistical Society: Series B (Methodological)</em>, 57(1):289–300, 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x">http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x</a>.</p>
</dd>
<dt class="bibtex label" id="hardwicke-undated-jj"><span class="brackets">50</span></dt>
<dd><p>Tom Elis Hardwicke and John Ioannidis. Mapping the universe of registered reports. <em>Nature Human Behaviour</em>, 2018. URL: <a class="reference external" href="https://www.nature.com/articles/s41562-018-0444-y">https://www.nature.com/articles/s41562-018-0444-y</a>.</p>
</dd>
<dt class="bibtex label" id="hasin2017-tk"><span class="brackets">51</span></dt>
<dd><p>Yehudit Hasin, Marcus Seldin, and Aldons Lusis. Multi-omics approaches to disease. <em>Genome Biol.</em>, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-017-1215-1">http://dx.doi.org/10.1186/s13059-017-1215-1</a>.</p>
</dd>
<dt class="bibtex label" id="ritchie2015-ds"><span class="brackets">52</span></dt>
<dd><p>Marylyn D Ritchie, Emily R Holzinger, Ruowang Li, Sarah A Pendergrass, and Dokyoon Kim. Methods of integrating data to uncover genotype-phenotype interactions. <em>Nat. Rev. Genet.</em>, 16(2):85–97, February 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3868">http://dx.doi.org/10.1038/nrg3868</a>.</p>
</dd>
<dt class="bibtex label" id="kristensen2014-iw"><span class="brackets">53</span></dt>
<dd><p>Vessela N Kristensen, Ole Christian Lingjærde, Hege G Russnes, Hans Kristian M Vollan, Arnoldo Frigessi, and Anne-Lise Børresen-Dale. Principles and methods of integrative genomic analyses in cancer. <em>Nat. Rev. Cancer</em>, 14(5):299–313, May 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrc3721">http://dx.doi.org/10.1038/nrc3721</a>.</p>
</dd>
</dl>
</p>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./end-matter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="../c07-conclusion/0-conclusion.html" title="previous page"><span class="section-number">8. </span>Conclusion</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    <script src="../_static/js/index.js"></script>
    
  </body>
</html>