
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Bibliography &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="Glossary" href="glossary.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Sources of bias in computational biology datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-error.html">
     3.5. Sources of error in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-combining/0-index.html">
   5. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/1-background.html">
     5.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/2-data.html">
     5.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/3-data-wrangling.html">
     5.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html">
     5.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/5-batch-correction.html">
     5.5. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html">
     5.6. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html#combined-data-set">
     5.7. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/7-discussion.html">
     5.8. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/8-future-work.html">
     5.9. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-data.html">
     6.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-methods.html">
     6.3. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/4-results.html">
     6.4. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-discussion.html">
     6.5. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-future-work.html">
     6.6. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="bibliography">
<h1>Bibliography<a class="headerlink" href="#bibliography" title="Permalink to this headline">¶</a></h1>
<p>References are shown at the bottom of the page on which they are cited. A full bibliography (of the entire Jupyter Book) appears below:</p>
<hr class="docutils" />
<p id="bibtex-bibliography-end-matter/reference-0"><dl class="citation">
<dt class="bibtex label" id="zaucha2015-ez"><span class="brackets">1</span></dt>
<dd><p>Jan Zaucha, Jonathan Stahlhacke, Matt E Oates, Natalie Thurlby, Owen J L Rackham, Hai Fang, Ben Smithers, and Julian Gough. A proteome quality index. <em>Environ. Microbiol.</em>, 17(1):4–9, 2015. URL: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.12622">https://onlinelibrary.wiley.com/doi/abs/10.1111/1462-2920.12622</a>.</p>
</dd>
<dt class="bibtex label" id="oates2015-li"><span class="brackets">2</span></dt>
<dd><p>Matt E Oates, Jonathan Stahlhacke, Dimitrios V Vavoulis, Ben Smithers, Owen J L Rackham, Adam J Sardar, Jan Zaucha, Natalie Thurlby, Hai Fang, and Julian Gough. The SUPERFAMILY 1.75 database in 2014: a doubling of data. <em>Nucleic Acids Res.</em>, 43(Database issue):D227–33, January 2015.</p>
</dd>
<dt class="bibtex label" id="gough2017-ik"><span class="brackets">3</span></dt>
<dd><p>Julian Gough, Jan Zaucha, and Natalie Thurlby. Determining phenotype from genotype. July 2017. URL: <a class="reference external" href="https://patents.google.com/patent/US20200176085A1/en">https://patents.google.com/patent/US20200176085A1/en</a>.</p>
</dd>
<dt class="bibtex label" id="zhou2019-jk"><span class="brackets">4</span></dt>
<dd><p>Naihui Zhou, Yuxiang Jiang, Timothy R Bergquist, Alexandra J Lee, Balint Z Kacsoh, Alex W Crocker, Kimberley A Lewis, George Georghiou, Huy N Nguyen, Md Nafiz Hamid, Larry Davis, Tunca Dogan, Volkan Atalay, Ahmet S Rifaioglu, Alperen Dalkiran, Rengul Cetin-Atalay, Chengxin Zhang, Rebecca L Hurto, Peter L Freddolino, Yang Zhang, Prajwal Bhat, Fran Supek, José M Fernández, Branislava Gemovic, Vladimir R Perovic, Radoslav S Davidović, Neven Sumonja, Nevena Veljkovic, Ehsaneddin Asgari, Mohammad R K Mofrad, Giuseppe Profiti, Castrense Savojardo, Pier Luigi Martelli, Rita Casadio, Florian Boecker, Indika Kahanda, Natalie Thurlby, Alice C McHardy, Alexandre Renaux, Rabie Saidi, Julian Gough, Alex A Freitas, Magdalena Antczak, Fabio Fabris, Mark N Wass, Jie Hou, Jianlin Cheng, Zheng Wang, Alfonso E Romero, Alberto Paccanaro, Haixuan Yang, Tatyana Goldberg, Chenguang Zhao, Liisa Holm, Petri Törönen, Alan J Medlar, Elaine Zosa, Itamar Borukhov, Ilya Novikov, Angela Wilkins, Olivier Lichtarge, Po-Han Chi, Wei-Cheng Tseng, Michal Linial, Peter W Rose, Christophe Dessimoz, Vedrana Vidulin, Saso Dzeroski, Ian Sillitoe, Sayoni Das, Jonathan Gill Lees, David T Jones, Cen Wan, Domenico Cozzetto, Rui Fa, Mateo Torres, Alex Wiarwick Vesztrocy, Jose Manuel Rodriguez, Michael L Tress, Marco Frasca, Marco Notaro, Giuliano Grossi, Alessandro Petrini, Matteo Re, Giorgio Valentini, Marco Mesiti, Daniel B Roche, Jonas Reeb, David W Ritchie, Sabeur Aridhi, Seyed Ziaeddin Alborzi, Marie-Dominique Devignes, Koo Da Chen Emily, Richard Bonneau, Vladimir Gligorijević, Meet Barot, Hai Fang, Stefano Toppo, Enrico Lavezzo, Marco Falda, Michele Berselli, Silvio C E Tosatto, Marco Carraro, Damiano Piovesan, Hafeez Ur Rehman, Qizhong Mao, Shanshan Zhang, Slobodan Vucetic, Gage S Black, Dane Jo, Dallas J Larsen, Ashton R Omdahl, Luke W Sagers, Erica Suh, Jonathan B Dayton, Liam J McGuffin, Danielle A Brackenridge, Patricia C Babbitt, Jeffrey M Yunes, Paolo Fontana, Feng Zhang, Shanfeng Zhu, Ronghui You, Zihan Zhang, Suyang Dai, Shuwei Yao, Weidong Tian, Renzhi Cao, Caleb Chandler, Miguel Amezola, Devon Johnson, Jia-Ming Chang, Wen-Hung Liao, Yi-Wei Liu, Stefano Pascarelli, Yotam Frank, Robert Hoehndorf, Maxat Kulmanov, Imane Boudellioua, Gianfranco Politano, Stefano Di Carlo, Alfredo Benso, Kai Hakala, Filip Ginter, Farrokh Mehryary, Suwisa Kaewphan, Jari Björne, Hans Moen, Martti E E Tolvanen, Tapio Salakoski, Daisuke Kihara, Aashish Jain, Tomislav Šmuc, Adrian Altenhoff, Asa Ben-Hur, Burkhard Rost, Steven E Brenner, Christine A Orengo, Constance J Jeffery, Giovanni Bosco, Deborah A Hogan, Maria J Martin, Claire O’Donovan, Sean D Mooney, Casey S Greene, Predrag Radivojac, and Iddo Friedberg. The CAFA challenge reports improved protein function prediction and new functional annotations for hundreds of genes through experimental screens. <em>bioRxiv</em>, pages 653105, May 2019.</p>
</dd>
<dt class="bibtex label" id="zirkle1935-sb"><span class="brackets">5</span></dt>
<dd><p>Conway Zirkle. The inheritance of acquired characters and the provisional hypothesis of pangenesis. <em>Am. Nat.</em>, 69(724):417–445, September 1935.</p>
</dd>
<dt class="bibtex label" id="grafton-undated-af"><span class="brackets">6</span></dt>
<dd><p>Anthony Grafton and Nancy G Siraisi. Natural particulars: nature and the disciplines in renaissance europe. In <em>acls humanities e-book</em>. MPublishing, University of Michigan Library, 1999.</p>
</dd>
<dt class="bibtex label" id="mendel1996-gi"><span class="brackets">7</span></dt>
<dd><p>Gregor Mendel. Experiments in plant hybridization (1865). <em>Verhandlungen des naturforschenden Vereins Brünn) Available online</em>, 1996. URL: <a class="reference external" href="http://old.esp.org/foundations/genetics/classical/gm-65-a.pdf">http://old.esp.org/foundations/genetics/classical/gm-65-a.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="johannsen1911-rh"><span class="brackets">8</span></dt>
<dd><p>W Johannsen. The genotype conception of heredity. <em>Am. Nat.</em>, 45(531):129–159, March 1911. URL: <a class="reference external" href="https://doi.org/10.1086/279202">https://doi.org/10.1086/279202</a>.</p>
</dd>
<dt class="bibtex label" id="evans2020-dq"><span class="brackets">9</span></dt>
<dd><p>Richard J Evans. RA fisher and the science of hatred. <em>New Stateman</em>, June 2020. URL: <a class="reference external" href="https://www.newstatesman.com/international/science-tech/2020/07/ra-fisher-and-science-hatred">https://www.newstatesman.com/international/science-tech/2020/07/ra-fisher-and-science-hatred</a>.</p>
</dd>
<dt class="bibtex label" id="watson1953-np"><span class="brackets">10</span></dt>
<dd><p>J D Watson and F H Crick. Molecular structure of nucleic acids; a structure for deoxyribose nucleic acid. <em>Nature</em>, 171(4356):737–738, April 1953. URL: <a class="reference external" href="http://dx.doi.org/10.1038/171737a0">http://dx.doi.org/10.1038/171737a0</a>.</p>
</dd>
<dt class="bibtex label" id="venter2001-wn"><span class="brackets">11</span></dt>
<dd><p>J C Venter, M D Adams, E W Myers, P W Li, R J Mural, G G Sutton, H O Smith, M Yandell, C A Evans, R A Holt, J D Gocayne, P Amanatides, R M Ballew, D H Huson, J R Wortman, Q Zhang, C D Kodira, X H Zheng, L Chen, M Skupski, G Subramanian, P D Thomas, J Zhang, G L Gabor Miklos, C Nelson, S Broder, A G Clark, J Nadeau, V A McKusick, N Zinder, A J Levine, R J Roberts, M Simon, C Slayman, M Hunkapiller, R Bolanos, A Delcher, I Dew, D Fasulo, M Flanigan, L Florea, A Halpern, S Hannenhalli, S Kravitz, S Levy, C Mobarry, K Reinert, K Remington, J Abu-Threideh, E Beasley, K Biddick, V Bonazzi, R Brandon, M Cargill, I Chandramouliswaran, R Charlab, K Chaturvedi, Z Deng, V Di Francesco, P Dunn, K Eilbeck, C Evangelista, A E Gabrielian, W Gan, W Ge, F Gong, Z Gu, P Guan, T J Heiman, M E Higgins, R R Ji, Z Ke, K A Ketchum, Z Lai, Y Lei, Z Li, J Li, Y Liang, X Lin, F Lu, G V Merkulov, N Milshina, H M Moore, A K Naik, V A Narayan, B Neelam, D Nusskern, D B Rusch, S Salzberg, W Shao, B Shue, J Sun, Z Wang, A Wang, X Wang, J Wang, M Wei, R Wides, C Xiao, C Yan, A Yao, J Ye, M Zhan, W Zhang, H Zhang, Q Zhao, L Zheng, F Zhong, W Zhong, S Zhu, S Zhao, D Gilbert, S Baumhueter, G Spier, C Carter, A Cravchik, T Woodage, F Ali, H An, A Awe, D Baldwin, H Baden, M Barnstead, I Barrow, K Beeson, D Busam, A Carver, A Center, M L Cheng, L Curry, S Danaher, L Davenport, R Desilets, S Dietz, K Dodson, L Doup, S Ferriera, N Garg, A Gluecksmann, B Hart, J Haynes, C Haynes, C Heiner, S Hladun, D Hostin, J Houck, T Howland, C Ibegwam, J Johnson, F Kalush, L Kline, S Koduru, A Love, F Mann, D May, S McCawley, T McIntosh, I McMullen, M Moy, L Moy, B Murphy, K Nelson, C Pfannkoch, E Pratts, V Puri, H Qureshi, M Reardon, R Rodriguez, Y H Rogers, D Romblad, B Ruhfel, R Scott, C Sitter, M Smallwood, E Stewart, R Strong, E Suh, R Thomas, N N Tint, S Tse, C Vech, G Wang, J Wetter, S Williams, M Williams, S Windsor, E Winn-Deen, K Wolfe, J Zaveri, K Zaveri, J F Abril, R Guigó, M J Campbell, K V Sjolander, B Karlak, A Kejariwal, H Mi, B Lazareva, T Hatton, A Narechania, K Diemer, A Muruganujan, N Guo, S Sato, V Bafna, S Istrail, R Lippert, R Schwartz, B Walenz, S Yooseph, D Allen, A Basu, J Baxendale, L Blick, M Caminha, J Carnes-Stine, P Caulk, Y H Chiang, M Coyne, C Dahlke, A Mays, M Dombroski, M Donnelly, D Ely, S Esparham, C Fosler, H Gire, S Glanowski, K Glasser, A Glodek, M Gorokhov, K Graham, B Gropman, M Harris, J Heil, S Henderson, J Hoover, D Jennings, C Jordan, J Jordan, J Kasha, L Kagan, C Kraft, A Levitsky, M Lewis, X Liu, J Lopez, D Ma, W Majoros, J McDaniel, S Murphy, M Newman, T Nguyen, N Nguyen, M Nodell, S Pan, J Peck, M Peterson, W Rowe, R Sanders, J Scott, M Simpson, T Smith, A Sprague, T Stockwell, R Turner, E Venter, M Wang, M Wen, D Wu, M Wu, A Xia, A Zandieh, and X Zhu. The sequence of the human genome. <em>Science</em>, 291(5507):1304–1351, February 2001.</p>
</dd>
<dt class="bibtex label" id="meek2000-ec"><span class="brackets">12</span></dt>
<dd><p>James Meek. Decoding DNA. <em>The Guardian</em>, June 2000. URL: <a class="reference external" href="http://www.theguardian.com/science/2000/jun/27/genetics.uknews">http://www.theguardian.com/science/2000/jun/27/genetics.uknews</a>.</p>
</dd>
<dt class="bibtex label" id="harmon2019-na"><span class="brackets">13</span></dt>
<dd><p>Amy Harmon. James watson had a chance to salvage his reputation on race. he made things worse. <em>The New York Times</em>, January 2019. URL: <a class="reference external" href="https://www.nytimes.com/2019/01/01/science/watson-dna-genetics-race.html">https://www.nytimes.com/2019/01/01/science/watson-dna-genetics-race.html</a>.</p>
</dd>
<dt class="bibtex label" id="cynober2020-rx"><span class="brackets">14</span></dt>
<dd><p>Timothé Cynober. Why are there only 11 cell and gene therapies in europe? <em>Labiotech</em>, September 2020. URL: <a class="reference external" href="https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/">https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/</a>.</p>
</dd>
<dt class="bibtex label" id="guardian-staff-reporter2020-lb"><span class="brackets">15</span></dt>
<dd><p>Guardian staff reporter. Cancer patients in england to be offered chance to avoid toxic side-effects. <em>The Guardian</em>, December 2020. URL: <a class="reference external" href="http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects">http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects</a>.</p>
</dd>
<dt class="bibtex label" id="pennisi2007-yv"><span class="brackets">16</span></dt>
<dd><p>Elizabeth Pennisi. Genomics. DNA study forces rethink of what it means to be a gene. <em>Science</em>, 316(5831):1556–1557, June 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.316.5831.1556">http://dx.doi.org/10.1126/science.316.5831.1556</a>.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ih"><span class="brackets">17</span></dt>
<dd><p>Gallery 19: DNA model, 1953 :: DNA learning center. <span><a class="reference external" href="#"></a></span>https://www.dnalc.org/view/16430-Gallery-19-DNA-model-1953.html. Accessed: 2019-6-2.</p>
</dd>
<dt class="bibtex label" id="mulder1839-rf"><span class="brackets">18</span></dt>
<dd><p>G J Mulder. Ueber die zusammensetzung einiger thierischen substanzen. <em>J. Prakt. Chem.</em>, 1839.</p>
</dd>
<dt class="bibtex label" id="uversky2013-bt"><span class="brackets">19</span></dt>
<dd><p>V N Uversky. Posttranslational modification. In Stanley Maloy and Kelly Hughes, editors, <em>Brenner’s Encyclopedia of Genetics (Second Edition)</em>, pages 425–430. Academic Press, San Diego, January 2013. URL: <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/B9780123749840012031">https://www.sciencedirect.com/science/article/pii/B9780123749840012031</a>.</p>
</dd>
<dt class="bibtex label" id="jarraud2019-jj"><span class="brackets">20</span></dt>
<dd><p>Nicolas Jarraud. Mecabricks. <span><a class="reference external" href="#"></a></span>https://mecabricks.com/, 2019.</p>
</dd>
<dt class="bibtex label" id="murzin1995-se"><span class="brackets">21</span></dt>
<dd><p>A G Murzin, S E Brenner, T Hubbard, and C Chothia. SCOP: a structural classification of proteins database for the investigation of sequences and structures. <em>J. Mol. Biol.</em>, 247(4):536–540, April 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.1995.0159">http://dx.doi.org/10.1006/jmbi.1995.0159</a>.</p>
</dd>
<dt class="bibtex label" id="fried2017-zu"><span class="brackets">22</span></dt>
<dd><p>Eiko I Fried. What are psychological constructs? on the nature and statistical modelling of emotions, intelligence, personality traits and mental disorders. <em>Health Psychol. Rev.</em>, 11(2):130–134, June 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1080/17437199.2017.1306718">http://dx.doi.org/10.1080/17437199.2017.1306718</a>.</p>
</dd>
<dt class="bibtex label" id="hacking1999-gh"><span class="brackets">23</span></dt>
<dd><p>Ian Hacking, Emeritus University Professor Ian Hacking, and Jan Hacking. <em>The Social Construction of What?</em> Harvard University Press, 1999. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=XkCR1p2YMRwC">https://play.google.com/store/books/details?id=XkCR1p2YMRwC</a>.</p>
</dd>
<dt class="bibtex label" id="szasz1976-ee"><span class="brackets">24</span></dt>
<dd><p>Thomas Szasz. The myth of mental illness. In James M Humber and Robert F Almeder, editors, <em>Biomedical Ethics and the Law</em>, pages 113–122. Springer US, Boston, MA, 1976. URL: <a class="reference external" href="https://doi.org/10.1007/978-1-4684-2223-8_10">https://doi.org/10.1007/978-1-4684-2223-8_10</a>.</p>
</dd>
<dt class="bibtex label" id="kaiser2019-tu"><span class="brackets">25</span></dt>
<dd><p>Jocelyn Kaiser. Genetics may explain up to 25% of same-sex behavior, giant analysis reveals. <em>Science Magazine</em>, August 2019. URL: <a class="reference external" href="https://www.sciencemag.org/news/2019/08/genetics-may-explain-25-same-sex-behavior-giant-analysis-reveals">https://www.sciencemag.org/news/2019/08/genetics-may-explain-25-same-sex-behavior-giant-analysis-reveals</a>.</p>
</dd>
<dt class="bibtex label" id="richardson2002-nx"><span class="brackets">26</span></dt>
<dd><p>Ken Richardson. What IQ tests test. <em>Theory Psychol.</em>, 12(3):283–314, June 2002. URL: <a class="reference external" href="https://doi.org/10.1177/0959354302012003012">https://doi.org/10.1177/0959354302012003012</a>.</p>
</dd>
<dt class="bibtex label" id="saini-undated-ng"><span class="brackets">27</span></dt>
<dd><p>Angela Saini. The disturbing return of scientific racism. <span><a class="reference external" href="#"></a></span>https://www.wired.co.uk/article/superior-the-return-of-race-science-angela-saini. Accessed: 2021-2-11. URL: <a class="reference external" href="https://www.wired.co.uk/article/superior-the-return-of-race-science-angela-saini">https://www.wired.co.uk/article/superior-the-return-of-race-science-angela-saini</a>.</p>
</dd>
<dt class="bibtex label" id="schuklenk1997-ag"><span class="brackets">28</span></dt>
<dd><p>U Schüklenk, E Stein, J Kerin, and W Byne. The ethics of genetic research on sexual orientation. <em>Hastings Cent. Rep.</em>, 27(4):6–13, July 1997. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/9271716">https://www.ncbi.nlm.nih.gov/pubmed/9271716</a>.</p>
</dd>
<dt class="bibtex label" id="wu2016-ic"><span class="brackets">29</span></dt>
<dd><p>Xiaolin Wu and Xi Zhang. Responses to critiques on machine learning of criminality perceptions (addendum of arxiv:1611.04135). <em>arXiv</em>, November 2016. URL: <a class="reference external" href="http://arxiv.org/abs/1611.04135">http://arxiv.org/abs/1611.04135</a>, <a class="reference external" href="https://arxiv.org/abs/1611.04135">arXiv:1611.04135</a>.</p>
</dd>
<dt class="bibtex label" id="stark2018-lk"><span class="brackets">30</span></dt>
<dd><p>Luke Stark. Facial recognition, emotion and race in animated social media. <em>First Monday</em>, September 2018. URL: <a class="reference external" href="http://journals.uic.edu/ojs/index.php/fm/article/view/9406">http://journals.uic.edu/ojs/index.php/fm/article/view/9406</a>.</p>
</dd>
<dt class="bibtex label" id="pulrang2019-gj"><span class="brackets">31</span></dt>
<dd><p>Andrew Pulrang. Disabled people explained: why we say we don’t want to be cured — disability thinking. <span><a class="reference external" href="#"></a></span>https://disabilitythinking.com/disabilitythinking/2019/4/22/disabled-people-explained-why-we-say-we-dont-want-to-be-cured, April 2019. Accessed: 2021-2-11. URL: <a class="reference external" href="https://disabilitythinking.com/disabilitythinking/2019/4/22/disabled-people-explained-why-we-say-we-dont-want-to-be-cured">https://disabilitythinking.com/disabilitythinking/2019/4/22/disabled-people-explained-why-we-say-we-dont-want-to-be-cured</a>.</p>
</dd>
<dt class="bibtex label" id="miller2013-ws"><span class="brackets">32</span></dt>
<dd><p>Paul Steven Miller and Rebecca Leah Levine. Avoiding genetic genocide: understanding good intentions and eugenics in the complex dialogue between the medical and disability communities. <em>Genet. Med.</em>, 15(2):95–102, February 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/gim.2012.102">http://dx.doi.org/10.1038/gim.2012.102</a>.</p>
</dd>
<dt class="bibtex label" id="pfeiffer2018-kt"><span class="brackets">33</span></dt>
<dd><p>Franziska Pfeiffer, Carsten Gröber, Michael Blank, Kristian Händler, Marc Beyer, Joachim L Schultze, and Günter Mayer. Systematic evaluation of error rates and causes in short samples in next-generation sequencing. <em>Sci. Rep.</em>, 8(1):10950, July 2018.</p>
</dd>
<dt class="bibtex label" id="de-koning2011-ac"><span class="brackets">34</span></dt>
<dd><p>A P Jason de Koning, Wanjun Gu, Todd A Castoe, Mark A Batzer, and David D Pollock. Repetitive elements may comprise over two-thirds of the human genome. <em>PLoS Genet.</em>, 7(12):e1002384, December 2011.</p>
</dd>
<dt class="bibtex label" id="haas2010-lm"><span class="brackets">35</span></dt>
<dd><p>Brian J Haas and Michael C Zody. Advancing RNA-Seq analysis. <em>Nat. Biotechnol.</em>, 28(5):421–423, May 2010.</p>
</dd>
<dt class="bibtex label" id="bumgarner2013-hg"><span class="brackets">36</span></dt>
<dd><p>Roger Bumgarner. Overview of DNA microarrays: types, applications, and their future. <em>Curr. Protoc. Mol. Biol.</em>, Chapter 22:Unit 22.1., January 2013.</p>
</dd>
<dt class="bibtex label" id="miga2015-zl"><span class="brackets">37</span></dt>
<dd><p>Karen H Miga. Completing the human genome: the progress and challenge of satellite DNA assembly. <em>Chromosome Res.</em>, 23(3):421–426, September 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s10577-015-9488-2">http://dx.doi.org/10.1007/s10577-015-9488-2</a>.</p>
</dd>
<dt class="bibtex label" id="kent2002-jg"><span class="brackets">38</span></dt>
<dd><p>W James Kent, Charles W Sugnet, Terrence S Furey, Krishna M Roskin, Tom H Pringle, Alan M Zahler, and David Haussler. The human genome browser at UCSC. <em>Genome Res.</em>, 12(6):996–1006, June 2002. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.229102">http://dx.doi.org/10.1101/gr.229102</a>.</p>
</dd>
<dt class="bibtex label" id="harger2000-vr"><span class="brackets">39</span></dt>
<dd><p>C Harger, G Chen, A Farmer, W Huang, J Inman, D Kiphart, F Schilkey, M P Skupski, and J Weller. The genome sequence DataBase. <em>Nucleic Acids Res.</em>, 28(1):31–32, January 2000. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/28.1.31">http://dx.doi.org/10.1093/nar/28.1.31</a>.</p>
</dd>
<dt class="bibtex label" id="hubbard2002-oi"><span class="brackets">40</span></dt>
<dd><p>T Hubbard, D Barker, E Birney, G Cameron, Y Chen, L Clark, T Cox, J Cuff, V Curwen, T Down, R Durbin, E Eyras, J Gilbert, M Hammond, L Huminiecki, A Kasprzyk, H Lehvaslaiho, P Lijnzaad, C Melsopp, E Mongin, R Pettett, M Pocock, S Potter, A Rust, E Schmidt, S Searle, G Slater, J Smith, W Spooner, A Stabenau, J Stalker, E Stupka, A Ureta-Vidal, I Vastrik, and M Clamp. The ensembl genome database project. <em>Nucleic Acids Res.</em>, 30(1):38–41, January 2002. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/30.1.38">http://dx.doi.org/10.1093/nar/30.1.38</a>.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ci"><span class="brackets">41</span></dt>
<dd><p>Genome browser FAQ. <span><a class="reference external" href="#"></a></span>https://genome.ucsc.edu/FAQ/FAQreleases.html. Accessed: 2020-12-13. URL: <a class="reference external" href="https://genome.ucsc.edu/FAQ/FAQreleases.html">https://genome.ucsc.edu/FAQ/FAQreleases.html</a>.</p>
</dd>
<dt class="bibtex label" id="gatk-team2020-au"><span class="brackets">42</span></dt>
<dd><p>GATK Team. Human genome reference builds - GRCh38 or hg38 - b37 - hg19. <span><a class="reference external" href="#"></a></span>https://gatk.broadinstitute.org/hc/en-us/articles/360035890951, June 2020. Accessed: 2020-12-13. URL: <a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890951">https://gatk.broadinstitute.org/hc/en-us/articles/360035890951</a>.</p>
</dd>
<dt class="bibtex label" id="meadows2019-zy"><span class="brackets">43</span></dt>
<dd><p>Alice Meadows, Laurel L Haak, and Josh Brown. Persistent identifiers: the building blocks of the research information infrastructure. <em>Insights Imaging</em>, 32(1):9, March 2019. URL: <a class="reference external" href="http://insights.uksg.org/articles/10.1629/uksg.457/">http://insights.uksg.org/articles/10.1629/uksg.457/</a>.</p>
</dd>
<dt class="bibtex label" id="vincent2020-ih"><span class="brackets">44</span></dt>
<dd><p>James Vincent. Scientists rename human genes to stop microsoft excel from misreading them as dates. <span><a class="reference external" href="#"></a></span>https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates, August 2020. Accessed: 2021-2-7. URL: <a class="reference external" href="https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates">https://www.theverge.com/2020/8/6/21355674/human-genes-rename-microsoft-excel-misreading-dates</a>.</p>
</dd>
<dt class="bibtex label" id="salzberg2018-yc"><span class="brackets">45</span></dt>
<dd><p>Steven L Salzberg. Open questions: how many genes do we have? <em>BMC Biol.</em>, 16(1):94, August 2018.</p>
</dd>
<dt class="bibtex label" id="sherry2001-nm"><span class="brackets">46</span></dt>
<dd><p>S T Sherry. dbSNP: the NCBI database of genetic variation. <em>Nucleic Acids Res.</em>, 29(1):308–311, 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/29.1.308">http://dx.doi.org/10.1093/nar/29.1.308</a>.</p>
</dd>
<dt class="bibtex label" id="andrews2017-tr"><span class="brackets">47</span></dt>
<dd><p>Ryan J Andrews, Levi Baber, and Walter N Moss. RNAStructuromeDB: a genome-wide database for RNA structural inference. <em>Sci. Rep.</em>, 7(1):17269, December 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41598-017-17510-y">http://dx.doi.org/10.1038/s41598-017-17510-y</a>.</p>
</dd>
<dt class="bibtex label" id="richardson2020-wa"><span class="brackets">48</span></dt>
<dd><p>Katherine E Richardson, Charles C Kirkpatrick, and Brent M Znosko. RNA CoSSMos 2.0: an improved searchable database of secondary structure motifs in RNA three-dimensional structures. <em>Database</em>, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/database/baz153">http://dx.doi.org/10.1093/database/baz153</a>.</p>
</dd>
<dt class="bibtex label" id="petryszak2016-je"><span class="brackets">49</span></dt>
<dd><p>Robert Petryszak, Maria Keays, Y Amy Tang, Nuno A Fonseca, Elisabet Barrera, Tony Burdett, Anja Füllgrabe, Alfonso Muñoz-Pomer Fuentes, Simon Jupp, Satu Koskinen, Oliver Mannion, Laura Huerta, Karine Megy, Catherine Snow, Eleanor Williams, Mitra Barzine, Emma Hastings, Hendrik Weisser, James Wright, Pankaj Jaiswal, Wolfgang Huber, Jyoti Choudhary, Helen E Parkinson, and Alvis Brazma. Expression atlas update—an integrated database of gene and protein expression in humans, animals and plants. 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv1045">http://dx.doi.org/10.1093/nar/gkv1045</a>.</p>
</dd>
<dt class="bibtex label" id="papatheodorou2020-ib"><span class="brackets">50</span></dt>
<dd><p>Irene Papatheodorou, Pablo Moreno, Jonathan Manning, Alfonso Muñoz-Pomer Fuentes, Nancy George, Silvie Fexova, Nuno A Fonseca, Anja Füllgrabe, Matthew Green, Ni Huang, Laura Huerta, Haider Iqbal, Monica Jianu, Suhaib Mohammed, Lingyun Zhao, Andrew F Jarnuczak, Simon Jupp, John Marioni, Kerstin Meyer, Robert Petryszak, Cesar Augusto Prada Medina, Carlos Talavera-López, Sarah Teichmann, Juan Antonio Vizcaino, and Alvis Brazma. Expression atlas update: from tissues to single cells. <em>Nucleic Acids Res.</em>, 48(D1):D77–D83, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkz947">http://dx.doi.org/10.1093/nar/gkz947</a>.</p>
</dd>
<dt class="bibtex label" id="illumina2014-nd"><span class="brackets">51</span></dt>
<dd><p>I Illumina. Understanding illumina quality scores. <em>Technical Note: Informatics</em>, 2014. URL: <a class="reference external" href="https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf">https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="wagner2012-ac"><span class="brackets">52</span></dt>
<dd><p>Günter P Wagner, Koryu Kin, and Vincent J Lynch. Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples. <em>Theory Biosci.</em>, 131(4):281–285, December 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s12064-012-0162-3">http://dx.doi.org/10.1007/s12064-012-0162-3</a>.</p>
</dd>
<dt class="bibtex label" id="pimentel2014-xm"><span class="brackets">53</span></dt>
<dd><p>H Pimentel. What the FPKM? a review of RNA-Seq expression units. 2014. URL: <a class="reference external" href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/</a>.</p>
</dd>
<dt class="bibtex label" id="robinson2010-kl"><span class="brackets">54</span></dt>
<dd><p>M D Robinson, D J McCarthy, and G K Smyth. Edger: a bioconductor package for differential expression analysis of digital gene expression data. <em>Bioinformatics</em>, 26(1):139–140, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btp616">http://dx.doi.org/10.1093/bioinformatics/btp616</a>.</p>
</dd>
<dt class="bibtex label" id="love2014-vx"><span class="brackets">55</span></dt>
<dd><p>Michael I Love, Wolfgang Huber, and Simon Anders. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. <em>Genome Biol.</em>, 15(12):550, 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-014-0550-8">http://dx.doi.org/10.1186/s13059-014-0550-8</a>.</p>
</dd>
<dt class="bibtex label" id="hersh1967-ox"><span class="brackets">56</span></dt>
<dd><p>Robert T Hersh. Atlas of protein sequence and structure, 1966. <em>Syst. Biol.</em>, 16(3):262–263, September 1967.</p>
</dd>
<dt class="bibtex label" id="pundir2016-ya"><span class="brackets">57</span></dt>
<dd><p>Sangya Pundir, Maria J Martin, Claire O’Donovan, and The UniProt Consortium. UniProt tools. <em>Curr. Protoc. Bioinformatics</em>, pages 1.29.1–1.29.15, 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1002/0471250953.bi0129s53">http://dx.doi.org/10.1002/0471250953.bi0129s53</a>.</p>
</dd>
<dt class="bibtex label" id="wang2012-pv"><span class="brackets">58</span></dt>
<dd><p>M Wang, M Weiss, M Simonovic, G Haertinger, S P Schrimpf, M O Hengartner, and C von Mering. PaxDb, a database of protein abundance averages across all three domains of life. <em>Mol. Cell. Proteomics</em>, 11(8):492–500, August 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1074/mcp.O111.014704">http://dx.doi.org/10.1074/mcp.O111.014704</a>.</p>
</dd>
<dt class="bibtex label" id="samaras2020-mg"><span class="brackets">59</span></dt>
<dd><p>Patroklos Samaras, Tobias Schmidt, Martin Frejno, Siegfried Gessulat, Maria Reinecke, Anna Jarzab, Jana Zecha, Julia Mergner, Piero Giansanti, Hans-Christian Ehrlich, Stephan Aiche, Johannes Rank, Harald Kienegger, Helmut Krcmar, Bernhard Kuster, and Mathias Wilhelm. ProteomicsDB: a multi-omics and multi-organism resource for life science research. <em>Nucleic Acids Res.</em>, 48(D1):D1153–D1163, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkz974">http://dx.doi.org/10.1093/nar/gkz974</a>.</p>
</dd>
<dt class="bibtex label" id="schwanhausser2011-tm"><span class="brackets">60</span></dt>
<dd><p>Björn Schwanhäusser, Dorothea Busse, Na Li, Gunnar Dittmar, Johannes Schuchhardt, Jana Wolf, Wei Chen, and Matthias Selbach. Global quantification of mammalian gene expression control. <em>Nature</em>, 473(7347):337–342, May 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nature10098">http://dx.doi.org/10.1038/nature10098</a>.</p>
</dd>
<dt class="bibtex label" id="gygi1999-lr"><span class="brackets">61</span></dt>
<dd><p>S P Gygi, Y Rochon, B R Franza, and R Aebersold. Correlation between protein and mRNA abundance in yeast. <em>Mol. Cell. Biol.</em>, 19(3):1720–1730, March 1999. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/10022859">https://www.ncbi.nlm.nih.gov/pubmed/10022859</a>.</p>
</dd>
<dt class="bibtex label" id="kosti2016-gt"><span class="brackets">62</span></dt>
<dd><p>Idit Kosti, Nishant Jain, Dvir Aran, Atul J Butte, and Marina Sirota. Cross-tissue analysis of gene and protein expression in normal and cancer tissues. <em>Sci. Rep.</em>, 6:24799, May 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/srep24799">http://dx.doi.org/10.1038/srep24799</a>.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ow"><span class="brackets">63</span></dt>
<dd><p>Protein data bank.</p>
</dd>
<dt class="bibtex label" id="altschul1990-zf"><span class="brackets">64</span></dt>
<dd><p>S F Altschul, W Gish, W Miller, E W Myers, and D J Lipman. Basic local alignment search tool. <em>J. Mol. Biol.</em>, 215(3):403–410, October 1990. URL: <a class="reference external" href="http://dx.doi.org/10.1016/S0022-2836(05)80360-2">http://dx.doi.org/10.1016/S0022-2836(05)80360-2</a>.</p>
</dd>
<dt class="bibtex label" id="andreeva2014-om"><span class="brackets">65</span></dt>
<dd><p>Antonina Andreeva, Dave Howorth, Cyrus Chothia, Eugene Kulesha, and Alexey G Murzin. SCOP2 prototype: a new approach to protein structure mining. <em>Nucleic Acids Res.</em>, 42(Database issue):D310–4, January 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkt1242">http://dx.doi.org/10.1093/nar/gkt1242</a>.</p>
</dd>
<dt class="bibtex label" id="orengo1997-vf"><span class="brackets">66</span></dt>
<dd><p>C A Orengo, A D Michie, S Jones, D T Jones, M B Swindells, and J M Thornton. CATH – a hierarchic classification of protein domain structures. <em>Structure</em>, 5(8):1093–1109, 1997. URL: <a class="reference external" href="http://dx.doi.org/10.1016/s0969-2126(97)00260-8">http://dx.doi.org/10.1016/s0969-2126(97)00260-8</a>.</p>
</dd>
<dt class="bibtex label" id="csaba2009-of"><span class="brackets">67</span></dt>
<dd><p>Gergely Csaba, Fabian Birzele, and Ralf Zimmer. Systematic comparison of SCOP and CATH: a new gold standard for protein structure analysis. <em>BMC Struct. Biol.</em>, 9:23, April 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1472-6807-9-23">http://dx.doi.org/10.1186/1472-6807-9-23</a>.</p>
</dd>
<dt class="bibtex label" id="gough2001-ct"><span class="brackets">68</span></dt>
<dd><p>J Gough, K Karplus, R Hughey, and C Chothia. Assignment of homology to genome sequences using a library of hidden markov models that represent all proteins of known structure. <em>J. Mol. Biol.</em>, 313(4):903–919, November 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.2001.5080">http://dx.doi.org/10.1006/jmbi.2001.5080</a>.</p>
</dd>
<dt class="bibtex label" id="furnham1986-yt"><span class="brackets">69</span></dt>
<dd><p>Adrian Furnham. Response bias, social desirability and dissimulation. <em>Pers. Individ. Dif.</em>, 7(3):385–400, January 1986. URL: <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/0191886986900140">https://www.sciencedirect.com/science/article/pii/0191886986900140</a>.</p>
</dd>
<dt class="bibtex label" id="knauper1994-wl"><span class="brackets">70</span></dt>
<dd><p>B Knäuper and H U Wittchen. Diagnosing major depression in the elderly: evidence for response bias in standardized diagnostic interviews? <em>J. Psychiatr. Res.</em>, 28(2):147–164, March 1994. URL: <a class="reference external" href="http://dx.doi.org/10.1016/0022-3956(94)90026-4">http://dx.doi.org/10.1016/0022-3956(94)90026-4</a>.</p>
</dd>
<dt class="bibtex label" id="golding2001-oj"><span class="brackets">71</span></dt>
<dd><p>Golding, Golding, Pembrey, Jones, and The Alspac Study Team. ALSPAC-The avon longitudinal study of parents and children. <em>Paediatr. Perinat. Epidemiol.</em>, 15(1):74–87, 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1046/j.1365-3016.2001.00325.x">http://dx.doi.org/10.1046/j.1365-3016.2001.00325.x</a>.</p>
</dd>
<dt class="bibtex label" id="bycroft2018-mw"><span class="brackets">72</span></dt>
<dd><p>Clare Bycroft, Colin Freeman, Desislava Petkova, Gavin Band, Lloyd T Elliott, Kevin Sharp, Allan Motyer, Damjan Vukcevic, Olivier Delaneau, Jared O’Connell, Adrian Cortes, Samantha Welsh, Alan Young, Mark Effingham, Gil McVean, Stephen Leslie, Naomi Allen, Peter Donnelly, and Jonathan Marchini. The UK biobank resource with deep phenotyping and genomic data. <em>Nature</em>, 562(7726):203–209, October 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41586-018-0579-z">http://dx.doi.org/10.1038/s41586-018-0579-z</a>.</p>
</dd>
<dt class="bibtex label" id="powell2021-vc"><span class="brackets">73</span></dt>
<dd><p>Kendall Powell. The broken promise that undermines human genome research. <em>Nature</em>, 590(7845):198–201, February 2021. URL: <a class="reference external" href="https://www.nature.com/articles/d41586-021-00331-5">https://www.nature.com/articles/d41586-021-00331-5</a>.</p>
</dd>
<dt class="bibtex label" id="buniello2019-cv"><span class="brackets">74</span></dt>
<dd><p>Annalisa Buniello, Jacqueline A L MacArthur, Maria Cerezo, Laura W Harris, James Hayhurst, Cinzia Malangone, Aoife McMahon, Joannella Morales, Edward Mountjoy, Elliot Sollis, Daniel Suveges, Olga Vrousgou, Patricia L Whetzel, Ridwan Amode, Jose A Guillen, Harpreet S Riat, Stephen J Trevanion, Peggy Hall, Heather Junkins, Paul Flicek, Tony Burdett, Lucia A Hindorff, Fiona Cunningham, and Helen Parkinson. The NHGRI-EBI GWAS catalog of published genome-wide association studies, targeted arrays and summary statistics 2019. <em>Nucleic Acids Res.</em>, 47(D1):D1005–D1012, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gky1120">http://dx.doi.org/10.1093/nar/gky1120</a>.</p>
</dd>
<dt class="bibtex label" id="l-emery2017-rd"><span class="brackets">75</span></dt>
<dd><p>J Macarthur L. Emery. GWAS catalog: exploring SNP-trait associations. <span><a class="reference external" href="#"></a></span>http://europepmc.org/article/CTX/C7914, December 2017. Accessed: 2020-9-3. URL: <a class="reference external" href="http://europepmc.org/article/CTX/C7914">http://europepmc.org/article/CTX/C7914</a>.</p>
</dd>
<dt class="bibtex label" id="barbaric2007-zm"><span class="brackets">76</span></dt>
<dd><p>Ivana Barbaric, Gaynor Miller, and T Neil Dear. Appearances can be deceiving: phenotypes of knockout mice. <em>Brief. Funct. Genomic. Proteomic.</em>, 6(2):91–103, June 2007.</p>
</dd>
<dt class="bibtex label" id="zhang2016-ty"><span class="brackets">77</span></dt>
<dd><p>Aihua Zhang, Hui Sun, Guangli Yan, Ping Wang, and Xijun Wang. Mass spectrometry-based metabolomics: applications to biomarker and metabolic pathway research. <em>Biomed. Chromatogr.</em>, 30(1):7–12, January 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1002/bmc.3453">http://dx.doi.org/10.1002/bmc.3453</a>.</p>
</dd>
<dt class="bibtex label" id="viswanathan2008-tn"><span class="brackets">78</span></dt>
<dd><p>Ganesh A Viswanathan, Jeremy Seto, Sonali Patil, German Nudelman, and Stuart C Sealfon. Getting started in biological pathway construction and analysis. <em>PLoS Comput. Biol.</em>, 4(2):e16, February 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pcbi.0040016">http://dx.doi.org/10.1371/journal.pcbi.0040016</a>.</p>
</dd>
<dt class="bibtex label" id="fabregat2018-na"><span class="brackets">79</span></dt>
<dd><p>Antonio Fabregat, Steven Jupe, Lisa Matthews, Konstantinos Sidiropoulos, Marc Gillespie, Phani Garapati, Robin Haw, Bijay Jassal, Florian Korninger, Bruce May, Marija Milacic, Corina Duenas Roca, Karen Rothfels, Cristoffer Sevilla, Veronica Shamovsky, Solomon Shorser, Thawfeek Varusai, Guilherme Viteri, Joel Weiser, Guanming Wu, Lincoln Stein, Henning Hermjakob, and Peter D’Eustachio. The reactome pathway knowledgebase. <em>Nucleic Acids Res.</em>, 46(D1):D649–D655, January 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkx1132">http://dx.doi.org/10.1093/nar/gkx1132</a>.</p>
</dd>
<dt class="bibtex label" id="kanehisa2008-wz"><span class="brackets">80</span></dt>
<dd><p>Minoru Kanehisa, Michihiro Araki, Susumu Goto, Masahiro Hattori, Mika Hirakawa, Masumi Itoh, Toshiaki Katayama, Shuichi Kawashima, Shujiro Okuda, Toshiaki Tokimatsu, and Yoshihiro Yamanishi. KEGG for linking genomes to life and the environment. <em>Nucleic Acids Res.</em>, 36(Database issue):D480–4, January 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkm882">http://dx.doi.org/10.1093/nar/gkm882</a>.</p>
</dd>
<dt class="bibtex label" id="blunt2001-wr"><span class="brackets">81</span></dt>
<dd><p>Wilfrid Blunt. <em>The Compleat Naturalist: A Life of Linnaeus</em>. Frances Lincoln, 2001. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=B3YOvgAACAAJ">https://play.google.com/store/books/details?id=B3YOvgAACAAJ</a>.</p>
</dd>
<dt class="bibtex label" id="ehret1748-bx"><span class="brackets">82</span></dt>
<dd><p>Ehret. <em>Plantae et papiliones rariores</em>. Volume 1748. [London :s.n.],, 1748. URL: <a class="reference external" href="https://www.biodiversitylibrary.org/item/205762">https://www.biodiversitylibrary.org/item/205762</a>.</p>
</dd>
<dt class="bibtex label" id="charmantier-undated-wy"><span class="brackets">83</span></dt>
<dd><p>Dr Isabelle Charmantier. Linnaeus and race. <span><a class="reference external" href="#"></a></span>https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race. Accessed: 2020-11-30. URL: <a class="reference external" href="https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race">https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race</a>.</p>
</dd>
<dt class="bibtex label" id="jensen2010-fk"><span class="brackets">84</span></dt>
<dd><p>Lars J Jensen and Peer Bork. Ontologies in quantitative biology: a basis for comparison, integration, and discovery. <em>PLoS Biol.</em>, 8(5):e1000374, May 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1000374">http://dx.doi.org/10.1371/journal.pbio.1000374</a>.</p>
</dd>
<dt class="bibtex label" id="overton2015-vo"><span class="brackets">85</span></dt>
<dd><p>James A Overton, Heiko Dietze, Shahim Essaid, David Osumi-Sutherland, and Christopher J Mungall. ROBOT: a command-line tool for ontology development. In <em>ICBO</em>. ceur-ws.org, 2015. URL: <a class="reference external" href="http://ceur-ws.org/Vol-1515/demo6.pdf">http://ceur-ws.org/Vol-1515/demo6.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="eden2009-ic"><span class="brackets">86</span></dt>
<dd><p>Eran Eden, Roy Navon, Israel Steinfeld, Doron Lipson, and Zohar Yakhini. GOrilla: a tool for discovery and visualization of enriched GO terms in ranked gene lists. <em>BMC Bioinformatics</em>, 10:48, February 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-10-48">http://dx.doi.org/10.1186/1471-2105-10-48</a>.</p>
</dd>
<dt class="bibtex label" id="ashburner2000-cr"><span class="brackets">87</span></dt>
<dd><p>M Ashburner, C A Ball, J A Blake, D Botstein, H Butler, J M Cherry, A P Davis, K Dolinski, S S Dwight, J T Eppig, M A Harris, D P Hill, L Issel-Tarver, A Kasarskis, S Lewis, J C Matese, J E Richardson, M Ringwald, G M Rubin, and G Sherlock. Gene ontology: tool for the unification of biology. the gene ontology consortium. <em>Nat. Genet.</em>, 25(1):25–29, May 2000. URL: <a class="reference external" href="http://dx.doi.org/10.1038/75556">http://dx.doi.org/10.1038/75556</a>.</p>
</dd>
<dt class="bibtex label" id="thomas2017-vm"><span class="brackets">88</span></dt>
<dd><p>Paul D Thomas. The gene ontology and the meaning of biological function. <em>Methods Mol. Biol.</em>, 1446:15–24, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1007/978-1-4939-3743-1_2">http://dx.doi.org/10.1007/978-1-4939-3743-1_2</a>.</p>
</dd>
<dt class="bibtex label" id="begley2012-oc"><span class="brackets">89</span></dt>
<dd><p>C Glenn Begley and Lee M Ellis. Drug development: raise standards for preclinical cancer research. <em>Nature</em>, 483(7391):531–533, March 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1038/483531a">http://dx.doi.org/10.1038/483531a</a>.</p>
</dd>
<dt class="bibtex label" id="prinz2011-rh"><span class="brackets">90</span></dt>
<dd><p>Florian Prinz, Thomas Schlange, and Khusru Asadullah. Believe it or not: how much can we rely on published data on potential drug targets? <em>Nat. Rev. Drug Discov.</em>, 10(9):712–712, 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrd3439-c1">http://dx.doi.org/10.1038/nrd3439-c1</a>.</p>
</dd>
<dt class="bibtex label" id="open-science-collaboration2015-vi"><span class="brackets">91</span></dt>
<dd><p>Open Science Collaboration. PSYCHOLOGY. estimating the reproducibility of psychological science. <em>Science</em>, 349(6251):aac4716, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.aac4716">http://dx.doi.org/10.1126/science.aac4716</a>.</p>
</dd>
<dt class="bibtex label" id="baker2016-tb"><span class="brackets">92</span></dt>
<dd><p>Monya Baker. 1,500 scientists lift the lid on reproducibility. <em>Nature</em>, 533(7604):452–454, May 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/533452a">http://dx.doi.org/10.1038/533452a</a>.</p>
</dd>
<dt class="bibtex label" id="bishop2019-hg"><span class="brackets">93</span></dt>
<dd><p>Dorothy Bishop. Rein in the four horsemen of irreproducibility. <em>Nature</em>, 568(7753):435, April 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/d41586-019-01307-2">http://dx.doi.org/10.1038/d41586-019-01307-2</a>.</p>
</dd>
<dt class="bibtex label" id="ioannidis2005-mo"><span class="brackets">94</span></dt>
<dd><p>John P A Ioannidis. Why most published research findings are false. <em>PLoS Med.</em>, 2(8):e124, August 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pmed.0020124">http://dx.doi.org/10.1371/journal.pmed.0020124</a>.</p>
</dd>
<dt class="bibtex label" id="head2015-ns"><span class="brackets">95</span></dt>
<dd><p>Megan L Head, Luke Holman, Rob Lanfear, Andrew T Kahn, and Michael D Jennions. The extent and consequences of p-hacking in science. <em>PLoS Biol.</em>, 13(3):e1002106, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1002106">http://dx.doi.org/10.1371/journal.pbio.1002106</a>.</p>
</dd>
<dt class="bibtex label" id="fisher1990-ro"><span class="brackets">96</span></dt>
<dd><p>R A Fisher and F Yates. <em>Statistical Methods, Experimental Design, and Scientific Inference: A Re-Issue of Statistical Methods for Research Workers, the Design of Experiments, and Statistical Methods and Scientific Inference</em>. OUP Oxford, April 1990. URL: <a class="reference external" href="https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290">https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290</a>.</p>
</dd>
<dt class="bibtex label" id="dunn1958-sj"><span class="brackets">97</span></dt>
<dd><p>Olive Jean Dunn. Estimation of the means of dependent variables. <em>The Annals of Mathematical Statistics</em>, 29(4):1095–1111, 1958. URL: <a class="reference external" href="http://dx.doi.org/10.1214/aoms/1177706443">http://dx.doi.org/10.1214/aoms/1177706443</a>.</p>
</dd>
<dt class="bibtex label" id="benjamini1995-me"><span class="brackets">98</span></dt>
<dd><p>Yoav Benjamini and Yosef Hochberg. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal Statistical Society: Series B (Methodological)</em>, 57(1):289–300, 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x">http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x</a>.</p>
</dd>
<dt class="bibtex label" id="hardwicke-undated-jj"><span class="brackets">99</span></dt>
<dd><p>Tom Elis Hardwicke and John Ioannidis. Mapping the universe of registered reports. <em>Nature Human Behaviour</em>, 2018. URL: <a class="reference external" href="https://www.nature.com/articles/s41562-018-0444-y">https://www.nature.com/articles/s41562-018-0444-y</a>.</p>
</dd>
<dt class="bibtex label" id="fang2013-et"><span class="brackets">100</span></dt>
<dd><p>Hai Fang, Matt E Oates, Ralph B Pethica, Jenny M Greenwood, Adam J Sardar, Owen J L Rackham, Philip C J Donoghue, Alexandros Stamatakis, David A de Lima Morais, and Julian Gough. A daily-updated tree of (sequenced) life as a reference for genome research. <em>Sci. Rep.</em>, 3:2015, 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/srep02015">http://dx.doi.org/10.1038/srep02015</a>.</p>
</dd>
<dt class="bibtex label" id="cracraft2004-ud"><span class="brackets">101</span></dt>
<dd><p>Joel Cracraft and Michael J Donoghue. <em>Assembling the Tree of Life</em>. Oxford University Press, July 2004. URL: <a class="reference external" href="https://www.amazon.co.uk/Assembling-Tree-Life-Joel-Cracraft/dp/0195172345">https://www.amazon.co.uk/Assembling-Tree-Life-Joel-Cracraft/dp/0195172345</a>.</p>
</dd>
<dt class="bibtex label" id="dohmen2016-iv"><span class="brackets">102</span></dt>
<dd><p>Elias Dohmen, Lukas P M Kremer, Erich Bornberg-Bauer, and Carsten Kemena. DOGMA: domain-based transcriptome and proteome quality assessment. <em>Bioinformatics</em>, 32(17):2577–2581, September 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btw231">http://dx.doi.org/10.1093/bioinformatics/btw231</a>.</p>
</dd>
<dt class="bibtex label" id="simao2015-gc"><span class="brackets">103</span></dt>
<dd><p>Felipe A Simão, Robert M Waterhouse, Panagiotis Ioannidis, Evgenia V Kriventseva, and Evgeny M Zdobnov. BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs. <em>Bioinformatics</em>, 31(19):3210–3212, October 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btv351">http://dx.doi.org/10.1093/bioinformatics/btv351</a>.</p>
</dd>
<dt class="bibtex label" id="kitts2016-eo"><span class="brackets">104</span></dt>
<dd><p>Paul A Kitts, Deanna M Church, Françoise Thibaud-Nissen, Jinna Choi, Vichet Hem, Victor Sapojnikov, Robert G Smith, Tatiana Tatusova, Charlie Xiang, Andrey Zherikov, Michael DiCuccio, Terence D Murphy, Kim D Pruitt, and Avi Kimchi. Assembly: a resource for assembled genomes at NCBI. <em>Nucleic Acids Res.</em>, 44(D1):D73–80, January 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv1226">http://dx.doi.org/10.1093/nar/gkv1226</a>.</p>
</dd>
<dt class="bibtex label" id="sims2014-df"><span class="brackets">105</span></dt>
<dd><p>David Sims, Ian Sudbery, Nicholas E Ilott, Andreas Heger, and Chris P Ponting. Sequencing depth and coverage: key considerations in genomic analyses. <em>Nat. Rev. Genet.</em>, 15(2):121–132, February 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3642">http://dx.doi.org/10.1038/nrg3642</a>.</p>
</dd>
<dt class="bibtex label" id="watson2019-pz"><span class="brackets">106</span></dt>
<dd><p>Mick Watson and Amanda Warr. Errors in long-read assemblies can critically affect protein prediction. <em>Nat. Biotechnol.</em>, 37(2):124–126, February 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41587-018-0004-z">http://dx.doi.org/10.1038/s41587-018-0004-z</a>.</p>
</dd>
<dt class="bibtex label" id="hasin2017-tk"><span class="brackets">107</span></dt>
<dd><p>Yehudit Hasin, Marcus Seldin, and Aldons Lusis. Multi-omics approaches to disease. <em>Genome Biol.</em>, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-017-1215-1">http://dx.doi.org/10.1186/s13059-017-1215-1</a>.</p>
</dd>
<dt class="bibtex label" id="ritchie2015-ds"><span class="brackets">108</span></dt>
<dd><p>Marylyn D Ritchie, Emily R Holzinger, Ruowang Li, Sarah A Pendergrass, and Dokyoon Kim. Methods of integrating data to uncover genotype-phenotype interactions. <em>Nat. Rev. Genet.</em>, 16(2):85–97, February 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3868">http://dx.doi.org/10.1038/nrg3868</a>.</p>
</dd>
<dt class="bibtex label" id="kristensen2014-iw"><span class="brackets">109</span></dt>
<dd><p>Vessela N Kristensen, Ole Christian Lingjærde, Hege G Russnes, Hans Kristian M Vollan, Arnoldo Frigessi, and Anne-Lise Børresen-Dale. Principles and methods of integrative genomic analyses in cancer. <em>Nat. Rev. Cancer</em>, 14(5):299–313, May 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrc3721">http://dx.doi.org/10.1038/nrc3721</a>.</p>
</dd>
<dt class="bibtex label" id="shihab2013-pk"><span class="brackets">110</span></dt>
<dd><p>Hashem A Shihab, Julian Gough, David N Cooper, Peter D Stenson, Gary L A Barker, Keith J Edwards, Ian N M Day, and Tom R Gaunt. Predicting the functional, molecular, and phenotypic consequences of amino acid substitutions using hidden markov models. <em>Hum. Mutat.</em>, 34(1):57–65, January 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1002/humu.22225">http://dx.doi.org/10.1002/humu.22225</a>.</p>
</dd>
<dt class="bibtex label" id="fang2013-ms"><span class="brackets">111</span></dt>
<dd><p>Hai Fang and Julian Gough. DcGO: database of domain-centric ontologies on functions, phenotypes, diseases and more. <em>Nucleic Acids Res.</em>, 41(Database issue):D536–44, January 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gks1080">http://dx.doi.org/10.1093/nar/gks1080</a>.</p>
</dd>
<dt class="bibtex label" id="consortium2015-ci"><span class="brackets">112</span></dt>
<dd><p>The 1000 Genomes Project Consortium and The 1000 Genomes Project Consortium. A global reference for human genetic variation. <em>Nature</em>, 526(7571):68–74, 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nature15393">http://dx.doi.org/10.1038/nature15393</a>.</p>
</dd>
<dt class="bibtex label" id="smith2005-uh"><span class="brackets">113</span></dt>
<dd><p>Cynthia L Smith, Carroll-Ann W Goldsmith, and Janan T Eppig. The mammalian phenotype ontology as a tool for annotating, analyzing and comparing phenotypic information. <em>Genome Biol.</em>, 6(1):R7, 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gb-2004-6-1-r7">http://dx.doi.org/10.1186/gb-2004-6-1-r7</a>.</p>
</dd>
<dt class="bibtex label" id="robinson2010-ga"><span class="brackets">114</span></dt>
<dd><p>P N Robinson and S Mundlos. The human phenotype ontology. <em>Clin. Genet.</em>, 77(6):525–534, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1111/j.1399-0004.2010.01436.x">http://dx.doi.org/10.1111/j.1399-0004.2010.01436.x</a>.</p>
</dd>
<dt class="bibtex label" id="schriml2012-dz"><span class="brackets">115</span></dt>
<dd><p>Lynn Marie Schriml, Cesar Arze, Suvarna Nadendla, Yu-Wei Wayne Chang, Mark Mazaitis, Victor Felix, Gang Feng, and Warren Alden Kibbe. Disease ontology: a backbone for disease semantic integration. <em>Nucleic Acids Res.</em>, 40(Database issue):D940–6, January 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkr972">http://dx.doi.org/10.1093/nar/gkr972</a>.</p>
</dd>
<dt class="bibtex label" id="bateman2002-bz"><span class="brackets">116</span></dt>
<dd><p>Alex Bateman, Ewan Birney, Lorenzo Cerruti, Richard Durbin, Laurence Etwiller, Sean R Eddy, Sam Griffiths-Jones, Kevin L Howe, Mhairi Marshall, and Erik L L Sonnhammer. The pfam protein families database. <em>Nucleic Acids Res.</em>, 30(1):276–280, January 2002. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/30.1.276">http://dx.doi.org/10.1093/nar/30.1.276</a>.</p>
</dd>
<dt class="bibtex label" id="genomes-project-consortium2012-ek"><span class="brackets">117</span></dt>
<dd><p>1000 Genomes Project Consortium, Goncalo R Abecasis, Adam Auton, Lisa D Brooks, Mark A DePristo, Richard M Durbin, Robert E Handsaker, Hyun Min Kang, Gabor T Marth, and Gil A McVean. An integrated map of genetic variation from 1,092 human genomes. <em>Nature</em>, 491(7422):56–65, November 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nature11632">http://dx.doi.org/10.1038/nature11632</a>.</p>
</dd>
<dt class="bibtex label" id="fairley2020-hp"><span class="brackets">118</span></dt>
<dd><p>Susan Fairley, Ernesto Lowy-Gallego, Emily Perry, and Paul Flicek. The international genome sample resource (IGSR) collection of open human genomic variation resources. <em>Nucleic Acids Res.</em>, 48(D1):D941–D947, January 2020. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkz836">http://dx.doi.org/10.1093/nar/gkz836</a>.</p>
</dd>
<dt class="bibtex label" id="poland2006-gd"><span class="brackets">119</span></dt>
<dd><p>Jan Poland and Thomas Zeugmann. Clustering pairwise distances with missing data: maximum cuts versus normalized cuts. <em>Discovery Science</em>, pages 197–208, 2006. URL: <a class="reference external" href="http://dx.doi.org/10.1007/11893318_21">http://dx.doi.org/10.1007/11893318_21</a>.</p>
</dd>
<dt class="bibtex label" id="fang2013-ix"><span class="brackets">120</span></dt>
<dd><p>Hai Fang and Julian Gough. A domain-centric solution to functional genomics via dcGO predictor. <em>BMC Bioinformatics</em>, 14 Suppl 3:S9, February 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-14-S3-S9">http://dx.doi.org/10.1186/1471-2105-14-S3-S9</a>.</p>
</dd>
<dt class="bibtex label" id="stenson2009-jc"><span class="brackets">121</span></dt>
<dd><p>Peter D Stenson, Matthew Mort, Edward V Ball, Katy Howells, Andrew D Phillips, Nick St Thomas, and David N Cooper. The human gene mutation database: 2008 update. <em>Genome Med.</em>, 1(1):13, January 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gm13">http://dx.doi.org/10.1186/gm13</a>.</p>
</dd>
<dt class="bibtex label" id="mclaren2016-di"><span class="brackets">122</span></dt>
<dd><p>William McLaren, Laurent Gil, Sarah E Hunt, Harpreet Singh Riat, Graham R S Ritchie, Anja Thormann, Paul Flicek, and Fiona Cunningham. The ensembl variant effect predictor. <em>Genome Biol.</em>, 17(1):122, June 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-016-0974-4">http://dx.doi.org/10.1186/s13059-016-0974-4</a>.</p>
</dd>
<dt class="bibtex label" id="church2005-zv"><span class="brackets">123</span></dt>
<dd><p>G M Church. The personal genome project. <em>Mol. Syst. Biol.</em>, 1:2005.0030, December 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1038/msb4100040">http://dx.doi.org/10.1038/msb4100040</a>.</p>
</dd>
<dt class="bibtex label" id="sand2007-ed"><span class="brackets">124</span></dt>
<dd><p>Philipp G Sand. A lesson not learned: allele misassignment. <em>Behav. Brain Funct.</em>, 3(1):65, 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1744-9081-3-65">http://dx.doi.org/10.1186/1744-9081-3-65</a>.</p>
</dd>
<dt class="bibtex label" id="le-cam1967-ku"><span class="brackets">125</span></dt>
<dd><p>Lucien Marie Le Cam and Jerzy Neyman. <em>Proceedings of the Fifth Berkeley Symposium on Mathematical Statistics and Probability</em>. University of California Press, 1967. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=EN6isXsFdKgC">https://play.google.com/store/books/details?id=EN6isXsFdKgC</a>.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-nc"><span class="brackets">126</span></dt>
<dd><p>2.3. clustering — scikit-learn 0.21.2 documentation. <span><a class="reference external" href="#"></a></span>https://scikit-learn.org/stable/modules/clustering.html. Accessed: 2019-6-9. URL: <a class="reference external" href="https://scikit-learn.org/stable/modules/clustering.html">https://scikit-learn.org/stable/modules/clustering.html</a>.</p>
</dd>
<dt class="bibtex label" id="bellman1957-dc"><span class="brackets">127</span></dt>
<dd><p>Richard Bellman. <em>Dynamic Programming</em>. Princeton University Press, 1957. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=wdtoPwAACAAJ">https://play.google.com/store/books/details?id=wdtoPwAACAAJ</a>.</p>
</dd>
<dt class="bibtex label" id="zimek2012-fg"><span class="brackets">128</span></dt>
<dd><p>A Zimek, E Schubert, and H P Kriegel. A survey on unsupervised outlier detection in high‐dimensional numerical data. <em>Stat. Anal. Data Min.</em>, 2012. URL: <a class="reference external" href="https://onlinelibrary.wiley.com/doi/abs/10.1002/sam.11161">https://onlinelibrary.wiley.com/doi/abs/10.1002/sam.11161</a>.</p>
</dd>
<dt class="bibtex label" id="greshake2014-mp"><span class="brackets">129</span></dt>
<dd><p>Bastian Greshake, Philipp E Bayer, Helge Rausch, and Julia Reda. openSNP–a crowdsourced web resource for personal genomics. <em>PLoS One</em>, 9(3):e89204, March 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pone.0089204">http://dx.doi.org/10.1371/journal.pone.0089204</a>.</p>
</dd>
<dt class="bibtex label" id="zhao2018-rw"><span class="brackets">130</span></dt>
<dd><p>Chenguang Zhao and Zheng Wang. GOGO: an improved algorithm to measure the semantic similarity between gene ontology terms. <em>Sci. Rep.</em>, 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41598-018-33219-y">http://dx.doi.org/10.1038/s41598-018-33219-y</a>.</p>
</dd>
<dt class="bibtex label" id="wang2007-yc"><span class="brackets">131</span></dt>
<dd><p>James Z Wang, Zhidian Du, Rapeeporn Payattakool, Philip S Yu, and Chin-Fu Chen. A new method to measure the semantic similarity of GO terms. <em>Bioinformatics</em>, 23(10):1274–1281, May 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btm087">http://dx.doi.org/10.1093/bioinformatics/btm087</a>.</p>
</dd>
<dt class="bibtex label" id="rogers2018-rc"><span class="brackets">132</span></dt>
<dd><p>Mark F Rogers, Hashem A Shihab, Matthew Mort, David N Cooper, Tom R Gaunt, and Colin Campbell. FATHMM-XF: accurate prediction of pathogenic point mutations via extended features. <em>Bioinformatics</em>, 34(3):511–513, February 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btx536">http://dx.doi.org/10.1093/bioinformatics/btx536</a>.</p>
</dd>
<dt class="bibtex label" id="mcinnes2019-ov"><span class="brackets">133</span></dt>
<dd><p>Gregory McInnes, Roxana Daneshjou, Panagiostis Katsonis, Olivier Lichtarge, Rajgopal Srinivasan, Sadhna Rana, Predrag Radivojac, Sean D Mooney, Kymberleigh A Pagel, Moses Stamboulian, Yuxiang Jiang, Emidio Capriotti, Yanran Wang, Yana Bromberg, Samuele Bovo, Castrense Savojardo, Pier Luigi Martelli, Rita Casadio, Lipika R Pal, John Moult, Steven E Brenner, and Russ Altman. Predicting venous thromboembolism risk from exomes in the critical assessment of genome interpretation (CAGI) challenges. <em>Hum. Mutat.</em>, 40(9):1314–1320, September 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1002/humu.23825">http://dx.doi.org/10.1002/humu.23825</a>.</p>
</dd>
<dt class="bibtex label" id="zaucha2018-dg"><span class="brackets">134</span></dt>
<dd><p>Jan Maciej Zaucha. <em>Predicting Outlier Phenotypes from SNPS: A Journey Through Molecular Biology</em>. PhD thesis, University of Bristol, 2018. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=_M7QvQEACAAJ">https://play.google.com/store/books/details?id=_M7QvQEACAAJ</a>.</p>
</dd>
<dt class="bibtex label" id="gomez-cabrero2014-gk"><span class="brackets">135</span></dt>
<dd><p>David Gomez-Cabrero, Imad Abugessaisa, Dieter Maier, Andrew Teschendorff, Matthias Merkenschlager, Andreas Gisel, Esteban Ballestar, Erik Bongcam-Rudloff, Ana Conesa, and Jesper Tegnér. Data integration in the era of omics: current and future challenges. <em>BMC Syst. Biol.</em>, 8 Suppl 2:I1, March 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1752-0509-8-S2-I1">http://dx.doi.org/10.1186/1752-0509-8-S2-I1</a>.</p>
</dd>
<dt class="bibtex label" id="bell2009-uq"><span class="brackets">136</span></dt>
<dd><p>Gordon Bell, Tony Hey, and Alex Szalay. Computer science. beyond the data deluge. <em>Science</em>, 323(5919):1297–1298, March 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.1170411">http://dx.doi.org/10.1126/science.1170411</a>.</p>
</dd>
<dt class="bibtex label" id="cortijo2019-wi"><span class="brackets">137</span></dt>
<dd><p>S Cortijo, Z Aydin, S Ahnert, and others. Widespread inter‐individual gene expression variability in arabidopsis thaliana. <em>Mol. Syst. Biol.</em>, 2019. URL: <a class="reference external" href="http://msb.embopress.org/content/15/1/e8591.abstract">http://msb.embopress.org/content/15/1/e8591.abstract</a>.</p>
</dd>
<dt class="bibtex label" id="vinuela2018age"><span class="brackets">138</span></dt>
<dd><p>Ana Viñuela, Andrew A Brown, Alfonso Buil, Pei-Chien Tsai, Matthew N Davies, Jordana T Bell, Emmanouil T Dermitzakis, Timothy D Spector, and Kerrin S Small. Age-dependent changes in mean and variance of gene expression across tissues in a twin cohort. <em>Human molecular genetics</em>, 27(4):732–741, 2018.</p>
</dd>
<dt class="bibtex label" id="yang2015-mo"><span class="brackets">139</span></dt>
<dd><p>Jialiang Yang, Tao Huang, Francesca Petralia, Quan Long, Bin Zhang, Carmen Argmann, Yong Zhao, Charles V Mobbs, Eric E Schadt, Jun Zhu, Zhidong Tu, and GTEx Consortium. Synchronized age-related gene expression changes across multiple tissues in human and the link to complex diseases. <em>Sci. Rep.</em>, 5:15145, October 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/srep15145">http://dx.doi.org/10.1038/srep15145</a>.</p>
</dd>
<dt class="bibtex label" id="reinius2009-ir"><span class="brackets">140</span></dt>
<dd><p>B Reinius and E Jazin. Prenatal sex differences in the human brain. <em>Mol. Psychiatry</em>, 14(11):987, 988–9, November 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1038/mp.2009.79">http://dx.doi.org/10.1038/mp.2009.79</a>.</p>
</dd>
<dt class="bibtex label" id="stone2019-fk"><span class="brackets">141</span></dt>
<dd><p>Gregory Stone, Ashley Choi, Meritxell Oliva, Joshua Gorham, Mahyar Heydarpour, Christine E Seidman, Jon G Seidman, Sary F Aranki, Simon C Body, Vincent J Carey, Benjamin A Raby, Barbara E Stranger, and Jochen D Muehlschlegel. Sex differences in gene expression in response to ischemia in the human left ventricular myocardium. <em>Hum. Mol. Genet.</em>, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/hmg/ddz014">http://dx.doi.org/10.1093/hmg/ddz014</a>.</p>
</dd>
<dt class="bibtex label" id="schibler2007-jx"><span class="brackets">142</span></dt>
<dd><p>Ueli Schibler. The daily timing of gene expression and physiology in mammals. <em>Dialogues Clin. Neurosci.</em>, 9(3):257–272, 2007. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/17969863">https://www.ncbi.nlm.nih.gov/pubmed/17969863</a>.</p>
</dd>
<dt class="bibtex label" id="svensson2018-oa"><span class="brackets">143</span></dt>
<dd><p>Valentine Svensson, Sarah A Teichmann, and Oliver Stegle. SpatialDE: identification of spatially variable genes. <em>Nat. Methods</em>, 15(5):343–346, May 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.4636">http://dx.doi.org/10.1038/nmeth.4636</a>.</p>
</dd>
<dt class="bibtex label" id="wang2018-rz"><span class="brackets">144</span></dt>
<dd><p>Qingguo Wang, Joshua Armenia, Chao Zhang, Alexander V Penson, Ed Reznik, Liguo Zhang, Thais Minet, Angelica Ochoa, Benjamin E Gross, Christine A Iacobuzio-Donahue, Doron Betel, Barry S Taylor, Jianjiong Gao, and Nikolaus Schultz. Unifying cancer and normal RNA sequencing data from different sources. <em>Sci Data</em>, 5:180061, April 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/sdata.2018.61">http://dx.doi.org/10.1038/sdata.2018.61</a>.</p>
</dd>
<dt class="bibtex label" id="mungall2012-nc"><span class="brackets">145</span></dt>
<dd><p>Christopher J Mungall, Carlo Torniai, Georgios V Gkoutos, Suzanna E Lewis, and Melissa A Haendel. Uberon, an integrative multi-species anatomy ontology. <em>Genome Biol.</em>, 13(1):R5, January 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1186/gb-2012-13-1-r5">http://dx.doi.org/10.1186/gb-2012-13-1-r5</a>.</p>
</dd>
<dt class="bibtex label" id="malladi2015-iq"><span class="brackets">146</span></dt>
<dd><p>Venkat S Malladi, Drew T Erickson, Nikhil R Podduturi, Laurence D Rowe, Esther T Chan, Jean M Davidson, Benjamin C Hitz, Marcus Ho, Brian T Lee, Stuart Miyasato, Gregory R Roe, Matt Simison, Cricket A Sloan, J Seth Strattan, Forrest Tanaka, W James Kent, J Michael Cherry, and Eurie L Hong. Ontology application and use at the ENCODE DCC. <em>Database</em>, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/database/bav010">http://dx.doi.org/10.1093/database/bav010</a>.</p>
</dd>
<dt class="bibtex label" id="leek2010-yw"><span class="brackets">147</span></dt>
<dd><p>Jeffrey T Leek, Robert B Scharpf, Héctor Corrada Bravo, David Simcha, Benjamin Langmead, W Evan Johnson, Donald Geman, Keith Baggerly, and Rafael A Irizarry. Tackling the widespread and critical impact of batch effects in high-throughput data. <em>Nat. Rev. Genet.</em>, 11(10):733–739, October 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg2825">http://dx.doi.org/10.1038/nrg2825</a>.</p>
</dd>
<dt class="bibtex label" id="irizarry2005-ie"><span class="brackets">148</span></dt>
<dd><p>Rafael A Irizarry, Daniel Warren, Forrest Spencer, Irene F Kim, Shyam Biswal, Bryan C Frank, Edward Gabrielson, Joe G N Garcia, Joel Geoghegan, Gregory Germino, Constance Griffin, Sara C Hilmer, Eric Hoffman, Anne E Jedlicka, Ernest Kawasaki, Francisco Martínez-Murillo, Laura Morsberger, Hannah Lee, David Petersen, John Quackenbush, Alan Scott, Michael Wilson, Yanqin Yang, Shui Qing Ye, and Wayne Yu. Multiple-laboratory comparison of microarray platforms. <em>Nat. Methods</em>, 2(5):345–350, May 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth756">http://dx.doi.org/10.1038/nmeth756</a>.</p>
</dd>
<dt class="bibtex label" id="johnson2007-zh"><span class="brackets">149</span></dt>
<dd><p>W Evan Johnson, Cheng Li, and Ariel Rabinovic. Adjusting batch effects in microarray expression data using empirical bayes methods. <em>Biostatistics</em>, 8(1):118–127, January 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1093/biostatistics/kxj037">http://dx.doi.org/10.1093/biostatistics/kxj037</a>.</p>
</dd>
<dt class="bibtex label" id="leek2007-ba"><span class="brackets">150</span></dt>
<dd><p>Jeffrey T Leek and John D Storey. Capturing heterogeneity in gene expression studies by surrogate variable analysis. <em>PLoS Genet.</em>, 3(9):1724–1735, September 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pgen.0030161">http://dx.doi.org/10.1371/journal.pgen.0030161</a>.</p>
</dd>
<dt class="bibtex label" id="chen2011-ke"><span class="brackets">151</span></dt>
<dd><p>Chao Chen, Kay Grennan, Judith Badner, Dandan Zhang, Elliot Gershon, Li Jin, and Chunyu Liu. Removing batch effects in analysis of expression microarray data: an evaluation of six batch adjustment methods. <em>PLoS One</em>, 6(2):e17238, February 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pone.0017238">http://dx.doi.org/10.1371/journal.pone.0017238</a>.</p>
</dd>
<dt class="bibtex label" id="liu2016-wa"><span class="brackets">152</span></dt>
<dd><p>Q Liu and M Markatou. Evaluation of methods in removing batch effects on RNA-seq data. <em>Infectious Diseases and Translational Medicine</em>, 2016. URL: <a class="reference external" href="http://www.tran-med.com/CN/article/downloadArticleFile.do?attachType=PDF&amp;id=24">http://www.tran-med.com/CN/article/downloadArticleFile.do?attachType=PDF&amp;id=24</a>.</p>
</dd>
<dt class="bibtex label" id="fonseca2014-hp"><span class="brackets">153</span></dt>
<dd><p>Nuno A Fonseca, Robert Petryszak, John Marioni, and Alvis Brazma. iRAP - an integrated RNA-seq analysis pipeline. Preprint, June 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1101/005991">http://dx.doi.org/10.1101/005991</a>.</p>
</dd>
<dt class="bibtex label" id="maier2009-vw"><span class="brackets">154</span></dt>
<dd><p>Tobias Maier, Marc Güell, and Luis Serrano. Correlation of mRNA and protein in complex biological samples. <em>FEBS Lett.</em>, 583(24):3966–3973, December 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1016/j.febslet.2009.10.036">http://dx.doi.org/10.1016/j.febslet.2009.10.036</a>.</p>
</dd>
<dt class="bibtex label" id="beyer2004-ws"><span class="brackets">155</span></dt>
<dd><p>Andreas Beyer, Jens Hollunder, Heinz-Peter Nasheuer, and Thomas Wilhelm. Post-transcriptional expression regulation in the yeast saccharomyces cerevisiae on a genomic scale. <em>Mol. Cell. Proteomics</em>, 3(11):1083–1092, November 2004. URL: <a class="reference external" href="http://dx.doi.org/10.1074/mcp.M400099-MCP200">http://dx.doi.org/10.1074/mcp.M400099-MCP200</a>.</p>
</dd>
<dt class="bibtex label" id="wu2008-hb"><span class="brackets">156</span></dt>
<dd><p>Gang Wu, Lei Nie, and Weiwen Zhang. Integrative analyses of posttranscriptional regulation in the yeast saccharomyces cerevisiae using transcriptomic and proteomic data. <em>Curr. Microbiol.</em>, 57(1):18–22, July 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s00284-008-9145-5">http://dx.doi.org/10.1007/s00284-008-9145-5</a>.</p>
</dd>
<dt class="bibtex label" id="yu2015-uf"><span class="brackets">157</span></dt>
<dd><p>Nancy Yiu-Lin Yu, Björn M Hallström, Linn Fagerberg, Fredrik Ponten, Hideya Kawaji, Piero Carninci, Alistair R R Forrest, Fantom Consortium, Yoshihide Hayashizaki, Mathias Uhlén, and Carsten O Daub. Complementing tissue characterization by integrating transcriptome profiling from the human protein atlas and from the FANTOM5 consortium. <em>Nucleic Acids Res.</em>, 43(14):6787–6798, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv608">http://dx.doi.org/10.1093/nar/gkv608</a>.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-gi"><span class="brackets">158</span></dt>
<dd><p>The human protein atlas. <span><a class="reference external" href="#"></a></span>https://www.proteinatlas.org/. Accessed: 2021-1-29. URL: <a class="reference external" href="https://www.proteinatlas.org/">https://www.proteinatlas.org/</a>.</p>
</dd>
<dt class="bibtex label" id="kawaji2014-tl"><span class="brackets">159</span></dt>
<dd><p>Hideya Kawaji, Marina Lizio, Masayoshi Itoh, Mutsumi Kanamori-Katayama, Ai Kaiho, Hiromi Nishiyori-Sueki, Jay W Shin, Miki Kojima-Ishiyama, Mitsuoki Kawano, Mitsuyoshi Murata, Noriko Ninomiya-Fukuda, Sachi Ishikawa-Kato, Sayaka Nagao-Sato, Shohei Noma, Yoshihide Hayashizaki, Alistair R R Forrest, Piero Carninci, and FANTOM Consortium. Comparison of CAGE and RNA-seq transcriptome profiling using clonally amplified and single-molecule next-generation sequencing. <em>Genome Res.</em>, 24(4):708–717, April 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.156232.113">http://dx.doi.org/10.1101/gr.156232.113</a>.</p>
</dd>
<dt class="bibtex label" id="winter2004-rr"><span class="brackets">160</span></dt>
<dd><p>Eitan E Winter, Leo Goodstadt, and Chris P Ponting. Elevated rates of protein secretion, evolution, and disease among tissue-specific genes. <em>Genome Res.</em>, 14(1):54–61, January 2004. URL: <a class="reference external" href="http://dx.doi.org/10.1101/gr.1924004">http://dx.doi.org/10.1101/gr.1924004</a>.</p>
</dd>
<dt class="bibtex label" id="uhlen2010-mx"><span class="brackets">161</span></dt>
<dd><p>Mathias Uhlen, Per Oksvold, Linn Fagerberg, Emma Lundberg, Kalle Jonasson, Mattias Forsberg, Martin Zwahlen, Caroline Kampf, Kenneth Wester, Sophia Hober, Henrik Wernerus, Lisa Björling, and Fredrik Ponten. Towards a knowledge-based human protein atlas. <em>Nature Biotechnology</em>, 28(12):1248–1250, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nbt1210-1248">http://dx.doi.org/10.1038/nbt1210-1248</a>.</p>
</dd>
<dt class="bibtex label" id="uhlen2015-at"><span class="brackets">162</span></dt>
<dd><p>Mathias Uhlén, Linn Fagerberg, Björn M Hallström, Cecilia Lindskog, Per Oksvold, Adil Mardinoglu, Åsa Sivertsson, Caroline Kampf, Evelina Sjöstedt, Anna Asplund, Ingmarie Olsson, Karolina Edlund, Emma Lundberg, Sanjay Navani, Cristina Al-Khalili Szigyarto, Jacob Odeberg, Dijana Djureinovic, Jenny Ottosson Takanen, Sophia Hober, Tove Alm, Per-Henrik Edqvist, Holger Berling, Hanna Tegel, Jan Mulder, Johan Rockberg, Peter Nilsson, Jochen M Schwenk, Marica Hamsten, Kalle von Feilitzen, Mattias Forsberg, Lukas Persson, Fredric Johansson, Martin Zwahlen, Gunnar von Heijne, Jens Nielsen, and Fredrik Pontén. Proteomics. tissue-based map of the human proteome. <em>Science</em>, 347(6220):1260419, January 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.1260419">http://dx.doi.org/10.1126/science.1260419</a>.</p>
</dd>
<dt class="bibtex label" id="gtex-consortium2013-gl"><span class="brackets">163</span></dt>
<dd><p>GTEx Consortium. The Genotype-Tissue expression (GTEx) project. <em>Nat. Genet.</em>, 45(6):580–585, June 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/ng.2653">http://dx.doi.org/10.1038/ng.2653</a>.</p>
</dd>
<dt class="bibtex label" id="lindsay2016-en"><span class="brackets">164</span></dt>
<dd><p>Susan J Lindsay, Yaobo Xu, Steven N Lisgo, Lauren F Harkin, Andrew J Copp, Dianne Gerrelli, Gavin J Clowry, Aysha Talbot, Michael J Keogh, Jonathan Coxhead, Mauro Santibanez-Koref, and Patrick F Chinnery. HDBR expression: a unique resource for global and individual gene expression studies during early human brain development. <em>Front. Neuroanat.</em>, 10:86, October 2016. URL: <a class="reference external" href="http://dx.doi.org/10.3389/fnana.2016.00086">http://dx.doi.org/10.3389/fnana.2016.00086</a>.</p>
</dd>
<dt class="bibtex label" id="keays2018-pg"><span class="brackets">165</span></dt>
<dd><p>M Keays. ExpressionAtlas: download datasets from EMBL-EBI expression atlas. R package version 1.10.0. 2018. URL: <a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/ExpressionAtlas.html">https://bioconductor.org/packages/release/bioc/html/ExpressionAtlas.html</a>.</p>
</dd>
<dt class="bibtex label" id="frazee2015-kg"><span class="brackets">166</span></dt>
<dd><p>Alyssa C Frazee, Andrew E Jaffe, Ben Langmead, and Jeffrey T Leek. Polyester: simulating RNA-seq datasets with differential transcript expression. <em>Bioinformatics</em>, 31(17):2778–2784, September 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btv272">http://dx.doi.org/10.1093/bioinformatics/btv272</a>.</p>
</dd>
<dt class="bibtex label" id="newman2015-le"><span class="brackets">167</span></dt>
<dd><p>Aaron M Newman, Chih Long Liu, Michael R Green, Andrew J Gentles, Weiguo Feng, Yue Xu, Chuong D Hoang, Maximilian Diehn, and Ash A Alizadeh. Robust enumeration of cell subsets from tissue expression profiles. <em>Nat. Methods</em>, 12(5):453–457, May 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.3337">http://dx.doi.org/10.1038/nmeth.3337</a>.</p>
</dd>
<dt class="bibtex label" id="wang2019-hc"><span class="brackets">168</span></dt>
<dd><p>Xuran Wang, Jihwan Park, Katalin Susztak, Nancy R Zhang, and Mingyao Li. Bulk tissue cell type deconvolution with multi-subject single-cell expression reference. <em>Nat. Commun.</em>, 10(1):380, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/s41467-018-08023-x">http://dx.doi.org/10.1038/s41467-018-08023-x</a>.</p>
</dd>
<dt class="bibtex label" id="haghverdi2018-ig"><span class="brackets">169</span></dt>
<dd><p>Laleh Haghverdi, Aaron T L Lun, Michael D Morgan, and John C Marioni. Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors. <em>Nat. Biotechnol.</em>, 36(5):421–427, June 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nbt.4091">http://dx.doi.org/10.1038/nbt.4091</a>.</p>
</dd>
<dt class="bibtex label" id="jiang2016-rz"><span class="brackets">170</span></dt>
<dd><p>Yuxiang Jiang, Tal Ronnen Oron, Wyatt T Clark, Asma R Bankapur, Daniel D’Andrea, Rosalba Lepore, Christopher S Funk, Indika Kahanda, Karin M Verspoor, Asa Ben-Hur, Da Chen Emily Koo, Duncan Penfold-Brown, Dennis Shasha, Noah Youngs, Richard Bonneau, Alexandra Lin, Sayed M E Sahraeian, Pier Luigi Martelli, Giuseppe Profiti, Rita Casadio, Renzhi Cao, Zhaolong Zhong, Jianlin Cheng, Adrian Altenhoff, Nives Skunca, Christophe Dessimoz, Tunca Dogan, Kai Hakala, Suwisa Kaewphan, Farrokh Mehryary, Tapio Salakoski, Filip Ginter, Hai Fang, Ben Smithers, Matt Oates, Julian Gough, Petri Törönen, Patrik Koskinen, Liisa Holm, Ching-Tai Chen, Wen-Lian Hsu, Kevin Bryson, Domenico Cozzetto, Federico Minneci, David T Jones, Samuel Chapman, Dukka Bkc, Ishita K Khan, Daisuke Kihara, Dan Ofer, Nadav Rappoport, Amos Stern, Elena Cibrian-Uhalte, Paul Denny, Rebecca E Foulger, Reija Hieta, Duncan Legge, Ruth C Lovering, Michele Magrane, Anna N Melidoni, Prudence Mutowo-Meullenet, Klemens Pichler, Aleksandra Shypitsyna, Biao Li, Pooya Zakeri, Sarah ElShal, Léon-Charles Tranchevent, Sayoni Das, Natalie L Dawson, David Lee, Jonathan G Lees, Ian Sillitoe, Prajwal Bhat, Tamás Nepusz, Alfonso E Romero, Rajkumar Sasidharan, Haixuan Yang, Alberto Paccanaro, Jesse Gillis, Adriana E Sedeño-Cortés, Paul Pavlidis, Shou Feng, Juan M Cejuela, Tatyana Goldberg, Tobias Hamp, Lothar Richter, Asaf Salamov, Toni Gabaldon, Marina Marcet-Houben, Fran Supek, Qingtian Gong, Wei Ning, Yuanpeng Zhou, Weidong Tian, Marco Falda, Paolo Fontana, Enrico Lavezzo, Stefano Toppo, Carlo Ferrari, Manuel Giollo, Damiano Piovesan, Silvio C E Tosatto, Angela Del Pozo, José M Fernández, Paolo Maietta, Alfonso Valencia, Michael L Tress, Alfredo Benso, Stefano Di Carlo, Gianfranco Politano, Alessandro Savino, Hafeez Ur Rehman, Matteo Re, Marco Mesiti, Giorgio Valentini, Joachim W Bargsten, Aalt D J van Dijk, Branislava Gemovic, Sanja Glisic, Vladmir Perovic, Veljko Veljkovic, Nevena Veljkovic, Danillo C Almeida-E-Silva, Ricardo Z N Vencio, Malvika Sharan, Jörg Vogel, Lakesh Kansakar, Shanshan Zhang, Slobodan Vucetic, Zheng Wang, Michael J E Sternberg, Mark N Wass, Rachael P Huntley, Maria J Martin, Claire O’Donovan, Peter N Robinson, Yves Moreau, Anna Tramontano, Patricia C Babbitt, Steven E Brenner, Michal Linial, Christine A Orengo, Burkhard Rost, Casey S Greene, Sean D Mooney, Iddo Friedberg, and Predrag Radivojac. An expanded evaluation of protein function prediction methods shows an improvement in accuracy. <em>Genome Biol.</em>, 17(1):184, September 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-016-1037-6">http://dx.doi.org/10.1186/s13059-016-1037-6</a>.</p>
</dd>
<dt class="bibtex label" id="radivojac2013-wh"><span class="brackets">171</span></dt>
<dd><p>Predrag Radivojac, Wyatt T Clark, Tal Ronnen Oron, Alexandra M Schnoes, Tobias Wittkop, Artem Sokolov, Kiley Graim, Christopher Funk, Karin Verspoor, Asa Ben-Hur, Gaurav Pandey, Jeffrey M Yunes, Ameet S Talwalkar, Susanna Repo, Michael L Souza, Damiano Piovesan, Rita Casadio, Zheng Wang, Jianlin Cheng, Hai Fang, Julian Gough, Patrik Koskinen, Petri Törönen, Jussi Nokso-Koivisto, Liisa Holm, Domenico Cozzetto, Daniel W A Buchan, Kevin Bryson, David T Jones, Bhakti Limaye, Harshal Inamdar, Avik Datta, Sunitha K Manjari, Rajendra Joshi, Meghana Chitale, Daisuke Kihara, Andreas M Lisewski, Serkan Erdin, Eric Venner, Olivier Lichtarge, Robert Rentzsch, Haixuan Yang, Alfonso E Romero, Prajwal Bhat, Alberto Paccanaro, Tobias Hamp, Rebecca Kaßner, Stefan Seemayer, Esmeralda Vicedo, Christian Schaefer, Dominik Achten, Florian Auer, Ariane Boehm, Tatjana Braun, Maximilian Hecht, Mark Heron, Peter Hönigschmid, Thomas A Hopf, Stefanie Kaufmann, Michael Kiening, Denis Krompass, Cedric Landerer, Yannick Mahlich, Manfred Roos, Jari Björne, Tapio Salakoski, Andrew Wong, Hagit Shatkay, Fanny Gatzmann, Ingolf Sommer, Mark N Wass, Michael J E Sternberg, Nives Škunca, Fran Supek, Matko Bošnjak, Panče Panov, Sašo Džeroski, Tomislav Šmuc, Yiannis A I Kourmpetis, Aalt D J van Dijk, Cajo J F ter Braak, Yuanpeng Zhou, Qingtian Gong, Xinran Dong, Weidong Tian, Marco Falda, Paolo Fontana, Enrico Lavezzo, Barbara Di Camillo, Stefano Toppo, Liang Lan, Nemanja Djuric, Yuhong Guo, Slobodan Vucetic, Amos Bairoch, Michal Linial, Patricia C Babbitt, Steven E Brenner, Christine Orengo, Burkhard Rost, Sean D Mooney, and Iddo Friedberg. A large-scale evaluation of computational protein function prediction. <em>Nat. Methods</em>, 10(3):221–227, March 2013. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nmeth.2340">http://dx.doi.org/10.1038/nmeth.2340</a>.</p>
</dd>
<dt class="bibtex label" id="lage2008-gq"><span class="brackets">172</span></dt>
<dd><p>Kasper Lage, Niclas Tue Hansen, E Olof Karlberg, Aron C Eklund, Francisco S Roque, Patricia K Donahoe, Zoltan Szallasi, Thomas Skøt Jensen, and Søren Brunak. A large-scale analysis of tissue-specific pathology and gene expression of human disease genes and complexes. <em>Proc. Natl. Acad. Sci. U. S. A.</em>, 105(52):20870–20875, December 2008. URL: <a class="reference external" href="http://dx.doi.org/10.1073/pnas.0810772105">http://dx.doi.org/10.1073/pnas.0810772105</a>.</p>
</dd>
<dt class="bibtex label" id="rackham2015-jp"><span class="brackets">173</span></dt>
<dd><p>Owen J L Rackham, Hashem A Shihab, Michael R Johnson, and Enrico Petretto. EvoTol: a protein-sequence based evolutionary intolerance framework for disease-gene prioritization. <em>Nucleic Acids Res.</em>, 43(5):e33, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gku1322">http://dx.doi.org/10.1093/nar/gku1322</a>.</p>
</dd>
<dt class="bibtex label" id="antanaviciute2015-ke"><span class="brackets">174</span></dt>
<dd><p>Agne Antanaviciute, Catherine Daly, Laura A Crinnion, Alexander F Markham, Christopher M Watson, David T Bonthron, and Ian M Carr. GeneTIER: prioritization of candidate disease genes using tissue-specific gene expression profiles. <em>Bioinformatics</em>, 31(16):2728–2735, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btv196">http://dx.doi.org/10.1093/bioinformatics/btv196</a>.</p>
</dd>
<dt class="bibtex label" id="dobzhansky2013-no"><span class="brackets">175</span></dt>
<dd><p>Theodosius Dobzhansky. Nothing in biology makes sense except in the light of evolution. <em>ambt</em>, 75(2):87–91, February 2013. URL: <a class="reference external" href="https://bioone.org/journals/the-american-biology-teacher/volume-75/issue-2/4444260/Nothing-in-Biology-Makes-Sense-Except-in-the-Light-of/10.2307/4444260.full">https://bioone.org/journals/the-american-biology-teacher/volume-75/issue-2/4444260/Nothing-in-Biology-Makes-Sense-Except-in-the-Light-of/10.2307/4444260.full</a>.</p>
</dd>
</dl>
</p>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./end-matter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="glossary.html" title="previous page">Glossary</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>