### Genes, Genomes, and Genetic Variation

#### Genes
As mentioned, genes are usually defined as stretches of DNA which encode for RNA. Only a proportion of an organism's DNA is make up of genes. Much of DNA is never expressed (read in order to produce RNA or proteins). This "unused" DNA is known as non-coding DNA. Even genes which can be expressed, are not used all the time, or in every cell in the body. The level of expression depends on the individual environments in those cells.
 
[//]: # (TODO: link for regulating transcription) 
The functionality of non-coding DNA is not fully understood. Some of it is used to regulate expression of other parts of the DNA. 

Genes are also not a straightforward unit. Just because a gene is read to create RNA, does not mean that RNA always makes proteins. Furthermore, there is not a 1:1 mapping of genes:proteins for those that do (one gene could be read to create multiple different proteins). There can be disagreements about the locations of genes on the genome, and popular databases of genes fundamentally disagree on this and on the number of genes[12]. Gene names and symbols change over time, and can be difficult to map between for this reason. 

Genes are also not necessarily a “unit of heritability” - we inherit regions around genes together with them and do not necessarily inherit whole copies of each gene from each parent (although this is rare). 

#### Genomes
The genome is the full amount of DNA belonging to an organism. We can talk about the genome of an individual, or about the genome of an organism (e.g. the human genome). We talk about the human genome in relation to an example human genome called a reference genome, which does not belong to any particular person, but instead is supposed to have the most common nucleotide for humans at each location. The human genome is 3 billion base pairs long, and contains approximately 20,000 genes (which make up around 2% of the genome).

#### Genetic variation
As you are no doubt aware, individuals of the same species share the vast majority of their DNA. Genetic variation is the difference in DNA sequence between individuals in a population, and variants are locations on an organisms genome where we observe variation between individuals. There are different types of genetic variation, for example indels (insertions and deletions)- where small sections of DNA are inserted or deleted from the genome - and copy number variations (CNVs) where larger stretches of the genome are repeated. However, the most well-studied and common type of genetic variation between humans are single nucleotide polymorphisms (SNPs, pronounced “snips”).

The different forms that a variant can take in the population are called alleles, depending on the type of variant. Alleles could be different forms of a whole gene, or as small in length as an individual nucleotide.

##### SNPs and SNVs
A SNP is a location on an organism’s genome where there are differences of a single nucleotide (“A”,”C”,”T”,”G”) between individuals. In some fields, these variations are only considered to be Single Nucleotide *Polymorphisms* if they are relatively commonly occurring in the population (at least 1%), while Single Nucleotide Variants (SNVs) can include both rare and common variants. 

Variation at a location does not imply a disease-causing effect, many SNPs are neutral. Much of the time, the aim of studying such variants is to determine which are which. This is often done through looking at their rarity, either in a specific human population (e.g. people with diabetes), the entire human population, or across the tree of life. 

SNPs are defined by their location on a human reference genome. An individual allele for a given SNP is defined as “wild” type if it matches the reference genome and “mutant” if it does not. The reference genome does not always have the most common allele at each location, although this is it’s aim, so “wild” and “mutant” do not necessarily imply anything about rarity. 

If there are only two nucleotide possibilities for a SNP (e.g. A or C), then it is called bi-allelic; the vast majority of SNPs are of this type. Multi-allelic SNPs (e.g. A, T or C) are much rarer. 

Since (except for X/Y chromosomes) humans have two copies of each chromosome, an individual will have two alleles for each SNP. These may match (which we call homozygous) or not (heterozygous). Sometimes a disease-causing allele can cause problems even for heterozygotes, while in other cases a person needs two copies of the disease-causing allele in order for it to have an effect.

SNPs can occur either in coding or non-coding regions of the genome. In non-coding regions, SNPs can still affect gene expression, for example by altering a promoter site. SNVs in coding regions have two types: synonymous or non-synonymous, based on whether they alter the amino acid sequence.

###### Non-synonymous SNVs
If a SNP alters the amino acid makeup of a protein, it is known as non-synonymous. Non-synonymous SNVs can cause either nonsense or missense mutations. 

Nonsense mutations occur where the SNP substitution results in a stop codon (e.g. TAG) in an unusual position, which signals for a ribosome to stop translating RNA into a protein. This results in an incomplete and usually nonfunctional protein. The effect of a nonsense mutation would be more or less severe depending on the location of the new stop codon. For example, if it was close to the end of the protein, the protein may still be functional. Sufficiently incomplete proteins are usually destroyed by the cell. 

[//]: # (Is this true? About location of stop codon. If so, I want the link!)

On the other hand, missense mutations occur where the SNP substitution results in an amino acid substitution in the protein. Some amino acids can be substituted without causing any difference to the function of the protein, while others can severely impede the protein.

###### Synonymous SNVs
Synonymous SNVs occur where substituting the usual nucleotide with another results in the same amino acid. The resulting protein will have the exact same functionality. However, synonymous SNVs could still have an effect on high-level traits, since different nucleotides are translated at different speeds. 
