
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>7.1. Introduction &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="7.2. Functionality" href="2-functionality.html" />
    <link rel="prev" title="7. Ontolopy" href="0-index.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   7. Ontolopy
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#motivation">
   7.1.1. Motivation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#obo-files">
   7.1.2. OBO files
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#anatomy-of-an-obo-file">
     7.1.2.1. Anatomy of an OBO file
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#purpose">
   7.1.3. Purpose
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#other-available-tools">
   7.1.4. Other available tools
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Introduction</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#motivation">
   7.1.1. Motivation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#obo-files">
   7.1.2. OBO files
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#anatomy-of-an-obo-file">
     7.1.2.1. Anatomy of an OBO file
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#purpose">
   7.1.3. Purpose
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#other-available-tools">
   7.1.4. Other available tools
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="section" id="introduction">
<span id="ontolopy-intro"></span><h1><span class="section-number">7.1. </span>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h1>
<div class="margin sidebar">
<p class="sidebar-title">What are ontologies again?</p>
<p>Ontologies are controlled vocabularies of terms and relationships. You can read more about them in <a class="reference internal" href="../c03-compbio-background/3-ontologies.html#what-are-ontologies"><span class="std std-numref">Section 3.3</span></a>.</p>
</div>
<div class="section" id="motivation">
<span id="opy-motivation"></span><h2><span class="section-number">7.1.1. </span>Motivation<a class="headerlink" href="#motivation" title="Permalink to this headline">¶</a></h2>
<p>I created Ontolopy in order to create a high-coverage mapping between tissues and gene expression samples, which I hoped would aid in phenotype and protein function prediction.
An earlier version of the package was used to create a mapping between gene expression sample names or identifiers to phenotypes that are known to affect that type of tissue for <a class="reference internal" href="../c06-filter/0-index.html#c06-filter"><span class="std std-numref">Section 6</span></a>.
Removing wrong-tissue predictions proved successful in improving protein function predictions, but was constrained by a low coverage, despite using one of the most extensive tissue-specific gene expression experiments.
In order to improve this coverage, I needed to extend it’s functionality to allow mapping more generally between samples and phenotypes according to their tissues.</p>
<p>There are many gene expression data sets, and the reporting for tissue metadata is not at all standardised between them.
This is true even within databases of gene expression data where great care has been taken to harmonise the metadata such as the Gene Expression Atlas.
If tissue type is recorded at all, it is usually manually given a label tissue using a name (e.g. “blood”, “kidney”), or perhaps as part of the sample name (“blood adult donor1”).
In other cases, cell type might be recorded instead (e.g. “leukocyte”, “cardiac fibroblast”).
In other circumstances still, the samples might be annotated to existing ontologies, and some even have their own ontologies of samples (such as FANTOM5).
Names like <em>blood</em> can be useful, but if you’d like to compare across samples, then it’s helpful to have a controlled vocabulary such as ontology terms: that way even a computer can figure out what <em>mature basophils</em>, <em>plasma</em> and <em>fibrinogen complex</em> have in common.</p>
<p>In addition to the benefits of ontologies’ controlled vocabulary (the terms themselves), they also contain a wealth of additional information and links to other ontologies.
For example, Uberon contains information about synonyms for different anatomical entities: the <em>pituitary stalk</em> is also known as the <em>infundibular stem</em> which is <em>part of the brain</em> that <em>connects to the hypothalamus</em>.
Ontologies are therefore also sources of text that could be used to map sample names to terms.
Once samples are mapped to ontologies they can leverage on all of the information inside them, for example, to find all the samples that are capable of <em>hormone secretion</em>.</p>
</div>
<div class="section" id="obo-files">
<span id="id1"></span><h2><span class="section-number">7.1.2. </span>OBO files<a class="headerlink" href="#obo-files" title="Permalink to this headline">¶</a></h2>
<p>There are two file formats which rule the ontology world.</p>
<p>Open Biomedical Ontology (OBO, <code class="docutils literal notranslate"><span class="pre">.obo</span></code> files) is the format that biomedical ontologies such as Gene Ontology or Human Phenotype Ontology were originally built in.
Meanwhile, the other file format is the Ontology Web Language (OWL), which is built upon XML.
Although it has not always been the home of biomedical ontologies, many now release both OBO and OWL versions.
Both file types are human-readable, although the OBO format is a little easier (for humans) to edit and read directly, and is generally considered easier to work with.
The major benefit of the OWL format is that it is formally axiomised and there exists a large suite of tools available for performing logical reasoning (e.g. using HermiT and SPARQL)</p>
<p>I found that at the time of creating, I needed files which were only available in OBO format, and OBO-to-OWL converters were not able to extract all the information that I needed.</p>
<div class="section" id="anatomy-of-an-obo-file">
<span id="obo-anatomy"></span><h3><span class="section-number">7.1.2.1. </span>Anatomy of an OBO file<a class="headerlink" href="#anatomy-of-an-obo-file" title="Permalink to this headline">¶</a></h3>
<p>OBO files are text files.
The top of an OBO file contains metadata about the ontology itself, for example it’s version in terms of format (<code class="docutils literal notranslate"><span class="pre">format-version</span></code>) and contents (<code class="docutils literal notranslate"><span class="pre">data-version</span></code>), name (<code class="docutils literal notranslate"><span class="pre">ontology</span></code>), the definitions of any subsets of ontology terms (<code class="docutils literal notranslate"><span class="pre">subsetdef</span></code>), the definition of synonym types (<code class="docutils literal notranslate"><span class="pre">synonymtypedef</span></code>), among many others.</p>
<p>Below are some example lines from this top section of the extended Uberon ontology file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">format</span><span class="o">-</span><span class="n">version</span><span class="p">:</span> <span class="mf">1.2</span>
<span class="n">ontology</span><span class="p">:</span> <span class="n">uberon</span><span class="o">/</span><span class="n">ext</span>
<span class="n">data</span><span class="o">-</span><span class="n">version</span><span class="p">:</span> <span class="n">uberon</span><span class="o">/</span><span class="n">releases</span><span class="o">/</span><span class="mi">2021</span><span class="o">-</span><span class="mi">02</span><span class="o">-</span><span class="mi">12</span><span class="o">/</span><span class="n">ext</span><span class="o">.</span><span class="n">owl</span>
<span class="n">subsetdef</span><span class="p">:</span> <span class="n">non_informative</span> <span class="s2">&quot;abstract class brought in to group ontology classes but not informative&quot;</span>
<span class="n">synonymtypedef</span><span class="p">:</span> <span class="n">HUMAN_PREFERRED</span> <span class="s2">&quot;preferred term when talking about an instance of this class in Homo sapiens&quot;</span>
</pre></div>
</div>
<p>The rest of the file has the following format, a blank line followed by <code class="docutils literal notranslate"><span class="pre">[Term]</span></code> indicates a new term is being defined, followed by different types of attributes, such as <code class="docutils literal notranslate"><span class="pre">id</span></code>, <code class="docutils literal notranslate"><span class="pre">name</span></code>, definition (<code class="docutils literal notranslate"><span class="pre">def</span></code>), external reference (<code class="docutils literal notranslate"><span class="pre">xref</span></code>), links to parent terms <code class="docutils literal notranslate"><span class="pre">is_a</span></code>, and other relationships such as <code class="docutils literal notranslate"><span class="pre">part_of</span></code> or <code class="docutils literal notranslate"><span class="pre">located_in</span></code>.
See for example the <em>pupillary membrane</em> term, <code class="docutils literal notranslate"><span class="pre">UBERON:0002269</span></code>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[Term]
id: UBERON:0002269
name: pupillary membrane
def: &quot;The pupillary membrane in mammals exists in the fetus as a source of blood supply for the lens. It normally atrophies from the time of birth to the age of four to eight weeks.&quot; [http://en.wikipedia.org/wiki/Persistent_pupillary_membrane]
xref: FMA:77663
xref: http://en.wikipedia.org/wiki/Persistent_pupillary_membrane
xref: MA:0001293
is_a: UBERON:0000158 ! membranous layer
is_a: UBERON:0004121 ! ectoderm-derived structure
relationship: located_in UBERON:0001771 {source=&quot;MA-modified&quot;} ! pupil
relationship: part_of UBERON:0000922 ! embryo
relationship: part_of UBERON:0001769 ! iris
</pre></div>
</div>
</div>
</div>
<div class="section" id="purpose">
<span id="ontolopy-purpose"></span><h2><span class="section-number">7.1.3. </span>Purpose<a class="headerlink" href="#purpose" title="Permalink to this headline">¶</a></h2>
<div class="figure align-default" id="ontolopy-logo">
<a class="reference internal image-reference" href="../_images/ontolopy_logo.png"><img alt="../_images/ontolopy_logo.png" src="../_images/ontolopy_logo.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 7.1 </span><span class="caption-text">Figure showing the purpose of Ontolopy as an ontology.</span><a class="headerlink" href="#ontolopy-logo" title="Permalink to this image">¶</a></p>
</div>
<div class="margin sidebar" id="propagate">
<p class="sidebar-title">Propagating</p>
<p>I use the word <em>propagate</em> to applying information about one ontology term, to describe it’s parents or children.
For example <code class="docutils literal notranslate"><span class="pre">UBERON:0009865</span></code> (<em><a class="reference external" href="http://en.wikipedia.org/wiki/Hatschek%27s_pit">Hatschek’s pit</a></em> - part of a fish-like lancelet) is <code class="docutils literal notranslate"><span class="pre">capable_of</span> <span class="pre">GO:0070254</span></code> (<em>mucus secretion</em>) and also <code class="docutils literal notranslate"><span class="pre">is_a</span> <span class="pre">UBERON:0006846</span></code> (<em>surface groove</em>).</p>
<p>If we propagate this relationship to the parent term, we find out that <em>surface groove</em>s can be <code class="docutils literal notranslate"><span class="pre">capable_of</span></code> <em>mucus secretion</em>.</p>
</div>
<div class="admonition-purpose-statement admonition">
<p class="admonition-title">Purpose statement</p>
<p>Ontolopy makes it easier to work with OBO ontologies in Python using familiar data types such as Python <code class="docutils literal notranslate"><span class="pre">dict</span></code>s (dictionary objects) and Pandas <code class="docutils literal notranslate"><span class="pre">DataFrame</span></code>s.
In particular, this supports finding and <a class="reference internal" href="#propagate"><span class="std std-ref">propagating</span></a> relationships between ontology terms (such as tissue and phenotype terms), and enables matching of sample names to ontology terms.</p>
</div>
<p>To my knowledge, no other existing package fulfils this particular need, however there are of course many other tools for working with ontologies.</p>
</div>
<div class="section" id="other-available-tools">
<span id="other-ont-tools"></span><h2><span class="section-number">7.1.4. </span>Other available tools<a class="headerlink" href="#other-available-tools" title="Permalink to this headline">¶</a></h2>
<p>There are a number of other tools that are available for building and logically querying ontologies, but these tend to be standalone platforms like <a class="reference external" href="https://protege.stanford.edu/">Protégé</a> (a Desktop platform primarily for building ontologies) or OWL-specific like <a class="reference external" href="https://owlready2.readthedocs.io/en/latest/intro.html">OwlReady2</a>.
I tested Protégé (using it’s built-in reasoner HermiT with SPARQL queries) for extracting uberon-phenotype mappings, but perhaps due to the size of the ontologies, this was prohibitively slow.</p>
<p>Ontologies are widely used by biomedical researchers, mostly for ontology enrichment analyses.
There are easy to use tools in popular programming languages like R and Python for performing specialised analyses (such as GO enrichment, like GOATools<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id215"><span>218</span></a>]</span>, or the older <a class="reference external" href="https://github.com/jdrudolph/goenrich">goenrich</a>) but tools for querying them generally are either very specialised or browser-based (like Ontobee<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id216"><span>219</span></a>]</span>).</p>
<p>Pronto<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id214"><span>220</span></a>]</span> is a nice Python package which is one of the exceptions to this rule, however it has some “missing” (missing for my needs, out of scope for them) functionality: being able to propagate relationships between nodes (terms).</p>
<p>OntoBio<span id="id5">[<a class="reference internal" href="../cz-end-matter/reference.html#id224"><span>221</span></a>]</span> is another Python package with similar functionality, which remains in active development. It is made by the same team as GOATools.
It has a rich functionality in terms of querying common attributes of biological ontologies, for example their synonyms or definitions. Again, the missing functionality for me is to be able to merge and query ontologies for relationships.</p>
<p>It would be beneficial for Ontolopy to make use of Pronto or OntoBio’s underlying data structures as I discuss in <a class="reference internal" href="7-future-work.html#pronto-integration"><span class="std std-numref">Section 7.7.2.4</span></a>.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c07-ontolopy"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="0-index.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">7. </span>Ontolopy</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="2-functionality.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">7.2. </span>Functionality</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Natalie Zelenka<br/>
    
        &copy; Copyright 2020.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>