
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>6.2. Functionality &#8212; Phenotype from Genotype</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script >const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="6.3. Ontolopy tools and practices" href="3-how-it-works.html" />
    <link rel="prev" title="6.1. Introduction" href="1-introduction.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How the link between genotype and phenotype is researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-alspac.html">
     4.6. Testing Snowflake on ALSPAC data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.7. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     5.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     5.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     5.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     5.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     5.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   6. Ontolopy
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     6.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-how-it-works.html">
     6.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-misc-examples.html">
     6.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-mapping-example.html">
     6.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-discussion.html">
     6.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-future-work.html">
     6.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   7. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     7.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     7.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     7.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   8. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#structure">
   6.2.1. Structure
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#working-with-obo-ontologies">
   6.2.2. Working with OBO ontologies
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-obo-class">
     6.2.2.1. The
     <code class="docutils literal notranslate">
      <span class="pre">
       Obo
      </span>
     </code>
     class
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#merging-ontologies">
     6.2.2.2. Merging ontologies
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#loading-ontologies-from-file">
     6.2.2.3. Loading ontologies from file
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#downloading-obo-files">
     6.2.2.4. Downloading OBO files
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#finding-relationships">
   6.2.3. Finding relationships
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-relations-class">
     6.2.3.1. The
     <code class="docutils literal notranslate">
      <span class="pre">
       Relations
      </span>
     </code>
     class
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#converting-relation-paths-to-text">
     6.2.3.2. Converting “relation paths” to text
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#creating-uberon-mappings">
   6.2.4. Creating Uberon Mappings
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-uberon-class">
     6.2.4.1. The
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon
      </span>
     </code>
     class
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-from-sample-to-tissue-via-name-using-uberon-sample-map-by-name">
     6.2.4.2. Mapping from sample to tissue via name using
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon.sample_map_by_name
      </span>
     </code>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-from-sample-to-tissue-via-ontology-term-using-uberon-sample-map-by-ont">
     6.2.4.3. Mapping from sample to tissue via ontology term using
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon.sample_map_by_ont
      </span>
     </code>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#getting-overall-mappings-and-finding-disagreements-using-uberon-get-overall-tissue-mappings">
     6.2.4.4. Getting overall mappings and finding disagreements using
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon.get_overall_tissue_mappings
      </span>
     </code>
    </a>
   </li>
  </ul>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Functionality</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#structure">
   6.2.1. Structure
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#working-with-obo-ontologies">
   6.2.2. Working with OBO ontologies
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-obo-class">
     6.2.2.1. The
     <code class="docutils literal notranslate">
      <span class="pre">
       Obo
      </span>
     </code>
     class
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#merging-ontologies">
     6.2.2.2. Merging ontologies
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#loading-ontologies-from-file">
     6.2.2.3. Loading ontologies from file
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#downloading-obo-files">
     6.2.2.4. Downloading OBO files
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#finding-relationships">
   6.2.3. Finding relationships
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-relations-class">
     6.2.3.1. The
     <code class="docutils literal notranslate">
      <span class="pre">
       Relations
      </span>
     </code>
     class
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#converting-relation-paths-to-text">
     6.2.3.2. Converting “relation paths” to text
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#creating-uberon-mappings">
   6.2.4. Creating Uberon Mappings
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-uberon-class">
     6.2.4.1. The
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon
      </span>
     </code>
     class
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-from-sample-to-tissue-via-name-using-uberon-sample-map-by-name">
     6.2.4.2. Mapping from sample to tissue via name using
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon.sample_map_by_name
      </span>
     </code>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-from-sample-to-tissue-via-ontology-term-using-uberon-sample-map-by-ont">
     6.2.4.3. Mapping from sample to tissue via ontology term using
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon.sample_map_by_ont
      </span>
     </code>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#getting-overall-mappings-and-finding-disagreements-using-uberon-get-overall-tissue-mappings">
     6.2.4.4. Getting overall mappings and finding disagreements using
     <code class="docutils literal notranslate">
      <span class="pre">
       Uberon.get_overall_tissue_mappings
      </span>
     </code>
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="section" id="functionality">
<span id="ontolopy-functionality"></span><h1><span class="section-number">6.2. </span>Functionality<a class="headerlink" href="#functionality" title="Permalink to this headline">¶</a></h1>
<p>This section describes the low-level functionality of Ontolopy: what it can do.
For examples of how this functionality is of practical use, please see the <a class="reference internal" href="4-misc-examples.html#ontolopy-examples-2"><span class="std std-ref">examples</span></a> sections.
You can find a full and up-to-date API Reference <a class="reference external" href="https://nataliethurlby.github.io/ontolopy/contents/reference.html">in the documentation</a>.</p>
<div class="margin sidebar" id="subclass">
<p class="sidebar-title">Subclass</p>
<p>Python is an object-orientated language, meaning that it’s designed so that classes can inherit from one another.</p>
<p>If a class subclasses another, it means it inherits it’s attributes and methods.</p>
</div>
<p>The functionality of the package can be summarised as follows: Ontolopy takes OBO files and:</p>
<ol class="simple">
<li><p>makes them into an intuitive Python object (which <a class="reference internal" href="#subclass"><span class="std std-ref">subclasses</span></a> a Python <code class="docutils literal notranslate"><span class="pre">dict</span></code>, meaning that you can do everything with it that you can do with this familiar and useful data type.</p></li>
<li><p>provides a set of tools for doing some useful manipulations and queries to these objects, which are particular to ontologies. This includes for example propagating relationships between terms, finding leaf/root terms, and merging ontologies.</p></li>
<li><p>Further to this, it provides an extra class for manipulating and querying the Uberon anatomy specifically.</p></li>
</ol>
<div class="section" id="structure">
<span id="opystructure"></span><h2><span class="section-number">6.2.1. </span>Structure<a class="headerlink" href="#structure" title="Permalink to this headline">¶</a></h2>
<div class="margin sidebar">
<p class="sidebar-title"><code class="docutils literal notranslate"><span class="pre">import</span> <span class="pre">ontolopy</span> <span class="pre">as</span> <span class="pre">opy</span></code></p>
<p>Note: you will see <code class="docutils literal notranslate"><span class="pre">ontolopy</span></code> shortened to <code class="docutils literal notranslate"><span class="pre">opy</span></code> in code segments.</p>
</div>
<p>Ontolopy is organised into three submodules, each centred around classes with the same names: <code class="docutils literal notranslate"><span class="pre">opy.Obo()</span></code> for OBO ontology objects, <code class="docutils literal notranslate"><span class="pre">obo.Relations()</span></code> for finding relationships between terms in an ontology object, and <code class="docutils literal notranslate"><span class="pre">opy.Uberon()</span></code> for finding tissue mappings.
These three submodules are automatically loaded with <code class="docutils literal notranslate"><span class="pre">import</span> <span class="pre">ontolopy</span></code>.</p>
</div>
<div class="section" id="working-with-obo-ontologies">
<span id="opyobo"></span><h2><span class="section-number">6.2.2. </span>Working with OBO ontologies<a class="headerlink" href="#working-with-obo-ontologies" title="Permalink to this headline">¶</a></h2>
<div class="margin sidebar" id="callables">
<p class="sidebar-title">Callables</p>
<p>In Python callables are functions, classes, and class methods that you can call, i.e. where you use syntax like <code class="docutils literal notranslate"><span class="pre">function()</span></code> or <code class="docutils literal notranslate"><span class="pre">MyClass()</span></code> to run some code.</p>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">opy.obo</span></code> module contains the following <a class="reference internal" href="#callables"><span class="std std-ref">callables</span></a> that make it easier to work with OBO ontologies:</p>
<table class="longtable table autosummary">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#ontolopy.obo.load_obo" title="ontolopy.obo.load_obo"><code class="xref py py-obj docutils literal notranslate"><span class="pre">load_obo</span></code></a>(file_loc[, ont_ids, discard_obsolete])</p></td>
<td><p>Loads ontology from <cite>.obo</cite> file at <cite>file_loc</cite>.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#ontolopy.obo.download_obo" title="ontolopy.obo.download_obo"><code class="xref py py-obj docutils literal notranslate"><span class="pre">download_obo</span></code></a>(data_name[, out_dir])</p></td>
<td><p>Download obo from a list of known locations.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#ontolopy.obo.Obo" title="ontolopy.obo.Obo"><code class="xref py py-obj docutils literal notranslate"><span class="pre">Obo</span></code></a>([source_dict])</p></td>
<td><p>Creates <cite>Obo</cite> ontology object from <cite>dict</cite> with ontology terms for keys, mapping to term attributes and relations.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#ontolopy.obo.Obo.merge" title="ontolopy.obo.Obo.merge"><code class="xref py py-obj docutils literal notranslate"><span class="pre">Obo.merge</span></code></a>(new[, prefer])</p></td>
<td><p>Recursively merges <cite>new</cite> into <cite>self</cite> and returns a merged <cite>Obo</cite> ontology.</p></td>
</tr>
</tbody>
</table>
<div class="section" id="the-obo-class">
<span id="oboclass"></span><h3><span class="section-number">6.2.2.1. </span>The <code class="docutils literal notranslate"><span class="pre">Obo</span></code> class<a class="headerlink" href="#the-obo-class" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">Obo</span></code> class is an OBO ontology object, which subclasses <code class="docutils literal notranslate"><span class="pre">dict</span></code>.
New <code class="docutils literal notranslate"><span class="pre">Obo</span></code> objects can be created from nested dictionaries.
At the top level of the dictionary, keys are terms and values are dictionaries.
This dictionary structure also allows you to add new terms.</p>
<div class="hint admonition">
<p class="admonition-title">Obo reference</p>
<dl class="py class">
<dt id="ontolopy.obo.Obo">
<em class="property"><span class="pre">class</span> </em><code class="sig-prename descclassname"><span class="pre">ontolopy.obo.</span></code><code class="sig-name descname"><span class="pre">Obo</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">source_dict</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.obo.Obo" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates <cite>Obo</cite> ontology object from <cite>dict</cite> with ontology terms for keys, mapping to term attributes and relations.</p>
<p>Each key/term is a dictionary with key: value pairs mapping either:</p>
<ol class="arabic simple">
<li><p>Attribute (<cite>str</cite>) to value (<cite>str</cite>), e.g. <cite>‘name’: ‘scapula’</cite></p></li>
<li><p>Type of relationship (<cite>str</cite>) to term identifiers (<cite>list</cite>), e.g. <cite>‘is_a’: [‘UBERON:0002513’]</cite></p></li>
</ol>
<p>Info: Obo stands for Open Biological Ontology: a popular file format for building biological ontologies.</p>
<dl class="py method">
<dt id="ontolopy.obo.Obo.__init__">
<code class="sig-name descname"><span class="pre">__init__</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">source_dict</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">{}</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.obo.Obo.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Initialise self from a source dictionary.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>source_dict</strong> – <cite>dict</cite> mapping terms to their attributes and relationships.</p>
</dd>
</dl>
</dd></dl>

<p class="rubric">Methods</p>
<table class="longtable table autosummary">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#ontolopy.obo.Obo.merge" title="ontolopy.obo.Obo.merge"><code class="xref py py-obj docutils literal notranslate"><span class="pre">merge</span></code></a>(new[, prefer])</p></td>
<td><p>Recursively merges <cite>new</cite> into <cite>self</cite> and returns a merged <cite>Obo</cite> ontology.</p></td>
</tr>
</tbody>
</table>
<p class="rubric">Attributes</p>
<table class="longtable table autosummary">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">leaves</span></code></p></td>
<td><p>Leaf terms are the most specific terms in the ontology; they have no children, only parents (a <cite>set</cite> object).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">terms</span></code></p></td>
<td><p>The ontology terms (a <cite>dict_keys</cite> object).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="note admonition">
<p class="admonition-title">Obo usage example</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span><span class="w"> </span><span class="nn">ontolopy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">opy</span>

<span class="n">new_ontology</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">Obo</span><span class="p">({</span><span class="s1">&#39;TERM:000001&#39;</span><span class="p">:</span> <span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Example term&#39;</span><span class="p">}})</span>
<span class="n">new_ontology</span><span class="p">[</span><span class="s1">&#39;TERM:000002&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Second example term&#39;</span><span class="p">,</span> <span class="s1">&#39;is_a&#39;</span><span class="p">:</span> <span class="p">[</span><span class="s1">&#39;TERM:000001&#39;</span><span class="p">]}</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="merging-ontologies">
<span id="obomerge"></span><h3><span class="section-number">6.2.2.2. </span>Merging ontologies<a class="headerlink" href="#merging-ontologies" title="Permalink to this headline">¶</a></h3>
<p>It’s also possible to merge (a list of) ontologies into the base ontology.
This can be useful for investigating relationships between ontologies.
For example, to find relationships between samples and tissues, that might go via cells, you may want to merge a sample ontology, cell, and tissue ontology to find all possible relationships.</p>
<div class="hint admonition">
<p class="admonition-title">Obo.merge reference</p>
<dl class="py method">
<dt id="ontolopy.obo.Obo.merge">
<code class="sig-prename descclassname"><span class="pre">Obo.</span></code><code class="sig-name descname"><span class="pre">merge</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">new</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prefer</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'self'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.obo.Obo.merge" title="Permalink to this definition">¶</a></dt>
<dd><p>Recursively merges <cite>new</cite> into <cite>self</cite> and returns a merged <cite>Obo</cite> ontology.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>new</strong> – Obo object (or list of objects) to add.</p></li>
<li><p><strong>prefer</strong> – prefer ‘self’ (base Obo) or ‘new’ (new Obo)</p></li>
</ul>
</dd>
<dt class="field-even">Return merged</dt>
<dd class="field-even"><p>A merged Obo</p>
</dd>
</dl>
</dd></dl>

</div>
</div>
<div class="section" id="loading-ontologies-from-file">
<span id="oboload"></span><h3><span class="section-number">6.2.2.3. </span>Loading ontologies from file<a class="headerlink" href="#loading-ontologies-from-file" title="Permalink to this headline">¶</a></h3>
<p>While creating ontologies from dictionaries is useful for adding bespoke terms, most of the time we want to load an official and curated OBO from a file.</p>
<div class="hint admonition">
<p class="admonition-title">load_obo reference</p>
<dl class="py function">
<dt id="ontolopy.obo.load_obo">
<code class="sig-prename descclassname"><span class="pre">ontolopy.obo.</span></code><code class="sig-name descname"><span class="pre">load_obo</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file_loc</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ont_ids</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">discard_obsolete</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.obo.load_obo" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads ontology from <cite>.obo</cite> file at <cite>file_loc</cite>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>file_loc</strong> – file location - path to stored obo file.</p></li>
<li><p><strong>ont_ids</strong> – list of ontology ids, e.g. <cite>[‘UBERON’, ‘CL’]</cite></p></li>
<li><p><strong>discard_obsolete</strong> – if True discard obsolete terms.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><cite>Obo</cite> ontology object.</p>
</dd>
</dl>
</dd></dl>

</div>
</div>
<div class="section" id="downloading-obo-files">
<span id="obodownload"></span><h3><span class="section-number">6.2.2.4. </span>Downloading OBO files<a class="headerlink" href="#downloading-obo-files" title="Permalink to this headline">¶</a></h3>
<p>It’s also possible to download OBO files, either from a list of popular OBO files by name, or via a URL.</p>
<div class="hint admonition">
<p class="admonition-title">download_obo reference</p>
<dl class="py function">
<dt id="ontolopy.obo.download_obo">
<code class="sig-prename descclassname"><span class="pre">ontolopy.obo.</span></code><code class="sig-name descname"><span class="pre">download_obo</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'../data/'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.obo.download_obo" title="Permalink to this definition">¶</a></dt>
<dd><p>Download obo from a list of known locations.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data_name</strong> – Name of OBO you wish to download.</p></li>
<li><p><strong>out_dir</strong> – Directory in which to save OBO file.</p></li>
</ul>
</dd>
<dt class="field-even">Return out_file</dt>
<dd class="field-even"><p>path to saved file.</p>
</dd>
</dl>
</dd></dl>

</div>
</div>
</div>
<div class="section" id="finding-relationships">
<span id="opyrelations"></span><h2><span class="section-number">6.2.3. </span>Finding relationships<a class="headerlink" href="#finding-relationships" title="Permalink to this headline">¶</a></h2>
<p>The most key functionality in Ontolopy is the ability to infer relationships between terms, across ontologies (be it between tissue terms and phenotype terms, or something else).
This functionality is inside the <code class="docutils literal notranslate"><span class="pre">opy.relations</span></code> module and handled by the <a class="reference internal" href="#relations-class"><span class="std std-ref">Relations</span></a> class.</p>
<div class="admonition-relations-reference admonition">
<p class="admonition-title">Relations reference</p>
<table class="longtable table autosummary">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#ontolopy.relations.relation_path_to_text" title="ontolopy.relations.relation_path_to_text"><code class="xref py py-obj docutils literal notranslate"><span class="pre">relation_path_to_text</span></code></a>(relation_path, ont)</p></td>
<td><p>Converts from a relation string e.g. “UBERON:123913.is_a~UBERON:1381239” to a text version,</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#ontolopy.relations.Relations" title="ontolopy.relations.Relations"><code class="xref py py-obj docutils literal notranslate"><span class="pre">Relations</span></code></a>(allowed_relations, ont[, sources, …])</p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="the-relations-class">
<span id="relations-class"></span><h3><span class="section-number">6.2.3.1. </span>The <code class="docutils literal notranslate"><span class="pre">Relations</span></code> class<a class="headerlink" href="#the-relations-class" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">Relations</span></code> class finds relationships of certain types between sources and targets.
It subclasses a <a class="reference external" href="https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html">Pandas DataFrame</a> since that is a convenient and familiar format for the relationship information to be returned.</p>
<dl class="py class">
<dt id="ontolopy.relations.Relations">
<em class="property"><span class="pre">class</span> </em><code class="sig-prename descclassname"><span class="pre">ontolopy.relations.</span></code><code class="sig-name descname"><span class="pre">Relations</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">allowed_relations</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ont</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sources</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">targets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source_targets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">excluded</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">col_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'any'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.relations.Relations" title="Permalink to this definition">¶</a></dt>
<dd><dl class="py method">
<dt id="ontolopy.relations.Relations.__init__">
<code class="sig-name descname"><span class="pre">__init__</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">allowed_relations</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ont</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sources</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">targets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source_targets</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">excluded</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">col_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mode</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'any'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.relations.Relations.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Pandas Dataframe containing relationships between <cite>sources</cite> and <cite>targets</cite> terms according to <cite>ont</cite>.
Finds relationships that do not pass through <cite>excluded</cite> terms and uses only <cite>allowed_relations</cite>. We keep looking
until we find a relation to a target (if mode == ‘any’) or we run out of leads.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>allowed_relations</strong> – a list of allowed relations, e.g. [‘is_a’, ‘part_of’]</p></li>
<li><p><strong>sources</strong> – list of sources. For mode <cite>all</cite> must be a list of source-target tuple airs.</p></li>
<li><p><strong>mode</strong> – ‘any’ or ‘all’ - ‘all is looking for specific term1-term2 pairs, while ‘any’ is looking for any
relationship between something in specific source and anything in targets.</p></li>
<li><p><strong>targets</strong> – list of targets.</p></li>
<li><p><strong>source_targets</strong> – list of tuples of source-target pairs. Do not provide source or targets if using this
parameter. Only runs in “all” mode.</p></li>
<li><p><strong>ont</strong> – Obo ontology object.</p></li>
<li><p><strong>excluded</strong> – a list/set of terms which are explicitly not being searched for (which may otherwise match the
targets). Useful e.g. if we want to look for any tissue targets with prefix ‘UBERON’, except for very general
ones. Does not allow relationships that pass through this term.</p></li>
<li><p><strong>col_names</strong> – Alternative column names for the output of Relations Data Frame, by default is
[‘from’, ‘relation_path’, ‘relation_text’, ‘to’]</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<!--
````{admonition} Relations usage example
:class: note
```python
import ontolopy as opy


```
````
-->
<p>To find relationships, the code loops through sources, and for each source it will look at the <code class="docutils literal notranslate"><span class="pre">allowed_relations</span></code> to find relationships with other terms, then for each of these terms it will look for relationships with other terms in the same manner, etc.</p>
<p id="relation-paths">Internally, Ontolopy stores these relationships as a list of strings, where each string details the relations between the source term and other terms, e.g. <code class="docutils literal notranslate"><span class="pre">UBERON:123913.is_a~UBERON:1381239.is_a~UBERON:987890</span></code>.
Let’s call these strings <em>relation paths</em>.</p>
<p>Cyclic relationships are not permitted (a term can only be present in a relation path once).
Relationships continue to be searched for until either the ontology provided can no longer add any new relation paths OR we found what we were looking for.</p>
<p>In “any” mode, finding what we’re looking for means finding any target term as the last term in the relation string, while in “all” mode, we must find all target terms for the source term.</p>
<p><strong>The <code class="docutils literal notranslate"><span class="pre">mode</span></code> parameter</strong> can be either <code class="docutils literal notranslate"><span class="pre">any</span></code> or <code class="docutils literal notranslate"><span class="pre">all</span></code>, and this represents whether we are looking for relations from our source terms to any one target term, or to all target terms for which we can find a relationship.
It is much quicker to run in “any” mode, so this mode is the default, and it is preferable when we simply need the most direct mapping between our source and target terms, for example we want to know which (one) tissue does the sample map to best?</p>
<p>The “all” mode tends to be more useful when we are equally interested in the targets as the source terms for example: when looking at mappings between tissues and phenotypes, there is likely to be many different phenotypes that a tissue can exhibit and we are equally interested in all of them.</p>
<p><strong>Provide either <code class="docutils literal notranslate"><span class="pre">sources</span></code> and <code class="docutils literal notranslate"><span class="pre">targets</span></code> OR <code class="docutils literal notranslate"><span class="pre">source-targets</span></code></strong>.
It’s possible to provide a list of <code class="docutils literal notranslate"><span class="pre">sources</span></code> and a list of <code class="docutils literal notranslate"><span class="pre">targets</span></code>, OR a list of tuple <code class="docutils literal notranslate"><span class="pre">source-targets</span></code>.
It does not make sense to provide both.
The latter option only works in <code class="docutils literal notranslate"><span class="pre">all</span></code> mode: i.e. we are interested in all source-target pairs.
Essentially, the <code class="docutils literal notranslate"><span class="pre">sources-targets</span></code> option provides a quicker way of running <code class="docutils literal notranslate"><span class="pre">Ontolopy</span></code> in “all” mode when we know in advance which specific pairs of sources and targets we are interested in.
If <code class="docutils literal notranslate"><span class="pre">sources</span></code> and <code class="docutils literal notranslate"><span class="pre">targets</span></code> are provided and <code class="docutils literal notranslate"><span class="pre">mode==all</span></code>, then <code class="docutils literal notranslate"><span class="pre">Ontolopy</span></code> will generate a combination of all possible sources and targets (removing <code class="docutils literal notranslate"><span class="pre">excluded</span></code> target terms if provided).</p>
</div>
<div class="section" id="converting-relation-paths-to-text">
<span id="relationspathstotext"></span><h3><span class="section-number">6.2.3.2. </span>Converting “relation paths” to text<a class="headerlink" href="#converting-relation-paths-to-text" title="Permalink to this headline">¶</a></h3>
<p>Since relationships are internally stored as <a class="reference internal" href="#relation-paths"><span class="std std-ref">relation paths</span></a> as explained above, it is useful to turn these strings into more readable text, which is what the <code class="docutils literal notranslate"><span class="pre">relation_path_to_text</span></code> function does.</p>
<div class="hint admonition">
<p class="admonition-title">relation_path_to_text reference</p>
<dl class="py function">
<dt id="ontolopy.relations.relation_path_to_text">
<code class="sig-prename descclassname"><span class="pre">ontolopy.relations.</span></code><code class="sig-name descname"><span class="pre">relation_path_to_text</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">relation_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ont</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.relations.relation_path_to_text" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Converts from a relation string e.g. “UBERON:123913.is_a~UBERON:1381239” to a text version,</dt><dd><p>e.g. “heart is a circulatory organ”.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ont</strong> – opy.Obo() ontology object.</p></li>
<li><p><strong>relation_path</strong> – path describing the relationship between two terms, e.g. “UBERON:123913.is_a-UBERON:1381239”</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

</div>
<!--Simplifying relationships-->
</div>
</div>
<div class="section" id="creating-uberon-mappings">
<span id="opyuberon"></span><h2><span class="section-number">6.2.4. </span>Creating Uberon Mappings<a class="headerlink" href="#creating-uberon-mappings" title="Permalink to this headline">¶</a></h2>
<p>The <code class="docutils literal notranslate"><span class="pre">opy.uberon</span></code> submodule contains the specific tools for working with the Uberon ontology: finding mappings between tissues and phenotypes <a class="reference internal" href="#mapping-by-term"><span class="std std-ref">via ontology terms</span></a> by making use of the <a class="reference internal" href="#relations-class"><span class="std std-ref">Relations</span></a> class, as well as <a class="reference internal" href="#mapping-by-name"><span class="std std-ref">doing this mapping using text</span></a>, and <a class="reference internal" href="#comparing-mappings"><span class="std std-ref">comparing these two mappings</span></a>.
The vast majority of this functionality sits in the <code class="docutils literal notranslate"><span class="pre">Uberon</span></code> class.</p>
<table class="longtable table autosummary">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">uberon_from_obo</span></code>(obo)</p></td>
<td><p>Creates an <cite>Uberon</cite> object from an <cite>Obo</cite> object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#ontolopy.uberon.Uberon" title="ontolopy.uberon.Uberon"><code class="xref py py-obj docutils literal notranslate"><span class="pre">Uberon</span></code></a>()</p></td>
<td><p>An UBERON-specific ontology object.</p></td>
</tr>
</tbody>
</table>
<div class="section" id="the-uberon-class">
<span id="uberonclass"></span><h3><span class="section-number">6.2.4.1. </span>The <code class="docutils literal notranslate"><span class="pre">Uberon</span></code> class<a class="headerlink" href="#the-uberon-class" title="Permalink to this headline">¶</a></h3>
<p>Calling the <code class="docutils literal notranslate"><span class="pre">Uberon</span></code> class itself simply checks if there are any <code class="docutils literal notranslate"><span class="pre">Uberon</span></code> terms in the merged ontology, and then allows the ontology to be used to create Uberon sample-to-tissue mappings, through class methods (which should be called separately).</p>
<p>There are three parts to the process in creating Uberon mappings, the functionality for which lives in three different <code class="docutils literal notranslate"><span class="pre">Uberon</span></code> class methods:</p>
<ol class="simple">
<li><p><strong><a class="reference internal" href="#mapping-by-name"><span class="std std-ref">Mapping via name</span></a>:</strong> Map from sample-to-tissue via informal tissue names given in experimental design information (e.g. “eye stalk”) to an Uberon term (<code class="docutils literal notranslate"><span class="pre">UBERON:0010326</span></code>, Optic Pedicel).</p></li>
<li><p><strong><a class="reference internal" href="#mapping-by-term"><span class="std std-ref">Mapping via ontology term</span></a>:</strong> Map from CL cell types (e.g. <code class="docutils literal notranslate"><span class="pre">CL:0000235</span></code>, Macrophage), sample ontology term to Uberon tissues (e.g. <code class="docutils literal notranslate"><span class="pre">UBERON:0002405</span></code>, Immune system). Or from sample ontology terms (like FANTOM terms, such as <code class="docutils literal notranslate"><span class="pre">FF:10048-101G3</span></code>, <em>Smooth Muscle, Adult, Pool1</em>) to Uberon terms (<code class="docutils literal notranslate"><span class="pre">UBERON:0001135</span></code>, Smooth Muscle Tissue). Returns relationships between source term and Uberon term.</p></li>
<li><p><strong><a class="reference internal" href="#comparing-mappings"><span class="std std-ref">Create sample-to-tissue mappings and disagreements between mappings</span></a></strong> based on (1) and (2).</p></li>
</ol>
<dl class="py class">
<dt id="ontolopy.uberon.Uberon">
<em class="property"><span class="pre">class</span> </em><code class="sig-prename descclassname"><span class="pre">ontolopy.uberon.</span></code><code class="sig-name descname"><span class="pre">Uberon</span></code><a class="headerlink" href="#ontolopy.uberon.Uberon" title="Permalink to this definition">¶</a></dt>
<dd><p>An UBERON-specific ontology object.</p>
<dl class="py method">
<dt id="ontolopy.uberon.Uberon.__init__">
<code class="sig-name descname"><span class="pre">__init__</span></code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.uberon.Uberon.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Initialise self from a source dictionary.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>source_dict</strong> – <cite>dict</cite> mapping terms to their attributes and relationships.</p>
</dd>
</dl>
</dd></dl>

<p class="rubric">Methods</p>
<table class="longtable table autosummary">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#ontolopy.uberon.Uberon.__init__" title="ontolopy.uberon.Uberon.__init__"><code class="xref py py-obj docutils literal notranslate"><span class="pre">__init__</span></code></a>()</p></td>
<td><p>Initialise self from a source dictionary.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#ontolopy.uberon.Uberon.sample_map_by_ont" title="ontolopy.uberon.Uberon.sample_map_by_ont"><code class="xref py py-obj docutils literal notranslate"><span class="pre">sample_map_by_ont</span></code></a>(sample_ids[, exclude, …])</p></td>
<td><p>Map tissues from sample names to uberon identifiers.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#ontolopy.uberon.Uberon.sample_map_by_name" title="ontolopy.uberon.Uberon.sample_map_by_name"><code class="xref py py-obj docutils literal notranslate"><span class="pre">sample_map_by_name</span></code></a>(sample_names[, to, …])</p></td>
<td><p>Map tissues from sample identifiers to uberon identifers.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#ontolopy.uberon.Uberon.get_overall_tissue_mappings" title="ontolopy.uberon.Uberon.get_overall_tissue_mappings"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_overall_tissue_mappings</span></code></a>(map_by_name, …)</p></td>
<td><p>Combines the two mappings <cite>map_by_name</cite> and <cite>map_by_ont</cite> to create an overall mapping and disagreements.</p></td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="mapping-from-sample-to-tissue-via-name-using-uberon-sample-map-by-name">
<span id="mapping-by-name"></span><h3><span class="section-number">6.2.4.2. </span>Mapping from sample to tissue via name using <code class="docutils literal notranslate"><span class="pre">Uberon.sample_map_by_name</span></code><a class="headerlink" href="#mapping-from-sample-to-tissue-via-name-using-uberon-sample-map-by-name" title="Permalink to this headline">¶</a></h3>
<p>Informal tissue names are mapped from Uberon term identifiers by checking for exact name matches to Uberon term names and their synonyms in the extended Uberon ontology.</p>
<div class="margin sidebar" id="stop-words">
<p class="sidebar-title">Stop words</p>
<p>Stop words are words that are filtered out before processing text using Natural Language Processing (NLP) methods.
These are usually very common words (e.g. “and”, ”the”), or word which are meaningless in the context of the analysis.</p>
</div>
<p>If an exact match does not exist, individual words from the phenotype term name or synonyms are then searched for exactly.
First <a class="reference internal" href="#stop-words"><span class="std std-ref">stop words</span></a> are removed, using the base list in the Natural Language Toolkit (<code class="docutils literal notranslate"><span class="pre">nltk</span></code>) Python Package<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id207"><span>240</span></a>]</span> (e.g. and, or), and a small number of manually curated phenotypic stopwords (e.g. “phenotype”, “abnormality”).
This would mean that the <abbr title="Human Phenotype">HP</abbr> term “abnormality of the head and neck” would search for the words “head” and “neck” in the UBERON terms, and would be mapped the terms of the same name (but never to “neck of radius” - which is related to bone).
In cases where multiple terms are found, a common parent would be searched for, in this case the result is “craniocervical region” .</p>
<div class="hint admonition">
<p class="admonition-title">Uberon.sample_map_by_name reference</p>
<dl class="py method">
<dt id="ontolopy.uberon.Uberon.sample_map_by_name">
<code class="sig-prename descclassname"><span class="pre">Uberon.</span></code><code class="sig-name descname"><span class="pre">sample_map_by_name</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_names</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">to</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">col_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xref</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">synonym_types</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.uberon.Uberon.sample_map_by_name" title="Permalink to this definition">¶</a></dt>
<dd><p>Map tissues from sample identifiers to uberon identifers.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sample_names</strong> – map from sample identifiers to tissue/sample descriptors/names for values.
May be <cite>dict</cite> or <cite>pd.Series</cite></p></li>
<li><p><strong>to</strong> – list of ontology prefixes that you want to map to.</p></li>
<li><p><strong>xref</strong> – An ontology identifier (e.g. FMA) the presence of which denotes a preferred term.</p></li>
<li><p><strong>col_names</strong> – Column names of returned relationships</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

</div>
</div>
<div class="section" id="mapping-from-sample-to-tissue-via-ontology-term-using-uberon-sample-map-by-ont">
<span id="mapping-by-term"></span><h3><span class="section-number">6.2.4.3. </span>Mapping from sample to tissue via ontology term using <code class="docutils literal notranslate"><span class="pre">Uberon.sample_map_by_ont</span></code><a class="headerlink" href="#mapping-from-sample-to-tissue-via-ontology-term-using-uberon-sample-map-by-ont" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">sample_map_by_ont</span></code> function uses the Relations class in “any” mode to find relationships via ontologies in much the same way described <a class="reference internal" href="#relations-class"><span class="std std-ref">above</span></a>.
This is essentially a wrapper that provides convenient default settings for allowed relations and targets.</p>
<p>Mappings can be made via any term in the merged ontology, which allows mappings that cannot be made through Uberon alone, for example: Macrophage - monocyte derived, donor3 <code class="docutils literal notranslate"><span class="pre">is_a</span></code> Human macrophage sample <code class="docutils literal notranslate"><span class="pre">derives_from</span></code> Macrophage <code class="docutils literal notranslate"><span class="pre">is_a</span></code> Monocyte <code class="docutils literal notranslate"><span class="pre">is_a</span></code> Leukocyte <code class="docutils literal notranslate"><span class="pre">part_of</span></code> Immune System, which means this sample is derived from part of the immune system.</p>
<div class="hint admonition">
<p class="admonition-title">Uberon.sample_map_by_ont reference</p>
<dl class="py method">
<dt id="ontolopy.uberon.Uberon.sample_map_by_ont">
<code class="sig-prename descclassname"><span class="pre">Uberon.</span></code><code class="sig-name descname"><span class="pre">sample_map_by_ont</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_ids</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">exclude</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">relation_types</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">to</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">child_mapping</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.uberon.Uberon.sample_map_by_ont" title="Permalink to this definition">¶</a></dt>
<dd><p>Map tissues from sample names to uberon identifiers. Will only work if ontology contains Uberon + Sample terms.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sample_ids</strong> – list of sample identifiers</p></li>
<li><p><strong>exclude</strong> – list of tissues to exclude, i.e. because they are too general.</p></li>
<li><p><strong>relation_types</strong> – list of relation types in ontology that relate to position in body.</p></li>
<li><p><strong>to</strong> – list of ontology prefixes that you want to map to.</p></li>
<li><p><strong>child_mapping</strong> – If True, searches children instead of parents.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>

</div>
<p id="child-mapping"><strong>Mapping by term: child mapping</strong>
Some samples may be pools of cell types that may come from more than one anatomical location.
In this case, there will be no regular mapping, since no parent terms will have a mapping to a tissue.
In this case, we can look at tissue mappings (in the usual way, described above), for all of the children of our parent term of interest.
I call this mode “child mapping” and it is off by default.</p>
<p>So, for example <em>melanocytes</em> are are melanin-producing cells found in many different places in the body (skin, hair, heart), and therefore they don’t (nor any of their parents) map to a specific Uberon term.
If we choose <code class="docutils literal notranslate"><span class="pre">child_mapping==TRUE</span></code>, then for this term, we will get a list of all Uberon terms that cells of this type can come from.
This mode isn’t currently used in the context of the rest of this thesis.</p>
</div>
<div class="section" id="getting-overall-mappings-and-finding-disagreements-using-uberon-get-overall-tissue-mappings">
<span id="comparing-mappings"></span><h3><span class="section-number">6.2.4.4. </span>Getting overall mappings and finding disagreements using <code class="docutils literal notranslate"><span class="pre">Uberon.get_overall_tissue_mappings</span></code><a class="headerlink" href="#getting-overall-mappings-and-finding-disagreements-using-uberon-get-overall-tissue-mappings" title="Permalink to this headline">¶</a></h3>
<p>As described, Ontolopy has two methods of mapping to tissues, and it also provides a method of harmonising these two mappings, and for finding any disagreements between them.
This can be very useful for revealing logical inconsistencies in either the mappings or the ontologies (as was the case in the <a class="reference internal" href="5-mapping-example.html#fantom5-inconsistencies-example"><span class="std std-ref">FANTOM5 example</span></a>).</p>
<div class="hint admonition">
<p class="admonition-title">Uberon.get_overall_tissue_mappings reference</p>
<dl class="py method">
<dt id="ontolopy.uberon.Uberon.get_overall_tissue_mappings">
<code class="sig-prename descclassname"><span class="pre">Uberon.</span></code><code class="sig-name descname"><span class="pre">get_overall_tissue_mappings</span></code><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">map_by_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">map_by_ont</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ontolopy.uberon.Uberon.get_overall_tissue_mappings" title="Permalink to this definition">¶</a></dt>
<dd><p>Combines the two mappings <cite>map_by_name</cite> and <cite>map_by_ont</cite> to create an overall mapping and disagreements.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>map_by_name</strong> (class: <cite>pd.DataFrame</cite>) – mapping from sample to tissue via sample name, from <cite>Uberon.sample_map_by_name</cite>.</p></li>
<li><p><strong>map_by_ont</strong> (class: <cite>pd.DataFrame</cite>) – mapping from sample to tissue via sample ontology ID, from <cite>Uberon.sample_map_by_ont</cite>.</p></li>
</ul>
</dd>
<dt class="field-even">Parm rel</dt>
<dd class="field-even"><p>list of relation strings allowed between name and ontology mappings to count as not a disagreement.</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>(<cite>overall_mapping</cite>: mapping from sample to tissue combining both sources,
<cite>disagreements</cite>: disagreements between “by name” and “by ontology” mappings.)</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>(class: <cite>pd.DataFrame</cite>, class: <cite>pd.DataFrame</cite>)</p>
</dd>
</dl>
</dd></dl>

</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c07-ontolopy"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="1-introduction.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">6.1. </span>Introduction</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="3-how-it-works.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">6.3. </span>Ontolopy tools and practices</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Natalie Zelenka<br/>
  
      &copy; Copyright 2020.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>