
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>7.4. Example uses: mapping samples to diseases or phenotypes &#8212; Phenotype from Genotype</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script >const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="7.5. Example use: mapping samples to tissue-related phenotypes" href="5-mapping-example.html" />
    <link rel="prev" title="7.3. Ontolopy tools and practices" href="3-how-it-works.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   7. Ontolopy
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#inputs">
   7.4.1. Inputs
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fantom5">
     7.4.1.1. FANTOM5
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#uberon">
     7.4.1.2. Uberon
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#example-1-finding-disease-related-samples">
   7.4.2. Example 1: Finding disease-related samples
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#example-2-find-tissues-that-are-capable-of-cell-differentiation">
   7.4.3. Example 2: Find tissues that are capable of cell differentiation
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Example uses: mapping samples to diseases or phenotypes</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#inputs">
   7.4.1. Inputs
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fantom5">
     7.4.1.1. FANTOM5
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#uberon">
     7.4.1.2. Uberon
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#example-1-finding-disease-related-samples">
   7.4.2. Example 1: Finding disease-related samples
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#example-2-find-tissues-that-are-capable-of-cell-differentiation">
   7.4.3. Example 2: Find tissues that are capable of cell differentiation
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="section" id="example-uses-mapping-samples-to-diseases-or-phenotypes">
<span id="ontolopy-examples-2"></span><h1><span class="section-number">7.4. </span>Example uses: mapping samples to diseases or phenotypes<a class="headerlink" href="#example-uses-mapping-samples-to-diseases-or-phenotypes" title="Permalink to this headline">¶</a></h1>
<p>There are a number of potential uses for Ontolopy.
In this section, I show two simple examples to demonstrate this usefulness.
These show how Ontolopy can be used to:</p>
<ol class="simple">
<li><p><a class="reference internal" href="#opydiseaseexample"><span class="std std-ref">Find disease-related samples</span></a></p></li>
<li><p><a class="reference internal" href="#opycelldifferentiation"><span class="std std-ref">Find samples of pluripotent stem cells</span></a> (cells that can turn into different tissue types)</p></li>
</ol>
<p>Then in <a class="reference internal" href="5-mapping-example.html#ontolopy-mapping-example"><span class="std std-ref">the next section</span></a> I give the more detailed and complex example of creating a mapping between samples and tissues (which is what Ontolopy was created for specifically), and how this was used to <a class="reference internal" href="5-mapping-example.html#fantom5-inconsistencies-example"><span class="std std-ref">find inconsistencies in the FANTOM5 data</span></a>.</p>
<div class="section" id="inputs">
<span id="ontolopy-example-inputs"></span><h2><span class="section-number">7.4.1. </span>Inputs<a class="headerlink" href="#inputs" title="Permalink to this headline">¶</a></h2>
<p>The examples of using Ontolopy in this Chapter use input files from FANTOM5<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id230"><span>234</span></a>]</span> (for samples) and Uberon<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id55"><span>104</span></a>]</span> (the cross-species anatomy ontology).</p>
<div class="margin sidebar">
<p class="sidebar-title">Choosing ont_ids</p>
<p>When we read in ontologies using Ontolopy, if you do not provide <code class="docutils literal notranslate"><span class="pre">ont_ids</span></code>, Ontolopy will keep all ontology term identifiers of the form <code class="docutils literal notranslate"><span class="pre">LETTERS:NUMBERS</span></code>, but often there are many external reference terms (<code class="docutils literal notranslate"><span class="pre">xref</span></code>s) that make the ontology object larger for no gain, so it’s recommended to provide them.</p>
<p>When providing <code class="docutils literal notranslate"><span class="pre">ont_ids</span></code>, it’s important that you keep all terms that you’re interested in, as everything else is discarded.</p>
</div>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">ontolopy</span> <span class="k">as</span> <span class="nn">opy</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span> 
<span class="kn">from</span> <span class="nn">myst_nb</span> <span class="kn">import</span> <span class="n">glue</span>
<span class="kn">import</span> <span class="nn">time</span>

<span class="c1"># Read in files:</span>
<span class="c1"># -------------</span>
<span class="n">fantom_obo_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/experiments/fantom/ff-phase2-170801.obo.txt&#39;</span>
<span class="n">fantom_samples_info_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/experiments/fantom/fantom_humanSamples2.0.csv&#39;</span>
<span class="n">uberon_obo_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/uberon_ext_210321.obo&#39;</span> 

<span class="c1"># Uberon OBO:</span>
<span class="n">uberon_obo</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">load_obo</span><span class="p">(</span>
    <span class="n">file_loc</span><span class="o">=</span><span class="n">uberon_obo_file</span><span class="p">,</span> 
    <span class="n">ont_ids</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span><span class="s1">&#39;CL&#39;</span><span class="p">],</span> 
<span class="p">)</span>

<span class="c1"># FANTOM OBO:</span>
<span class="n">fantom_obo</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">load_obo</span><span class="p">(</span>
    <span class="n">file_loc</span><span class="o">=</span><span class="n">fantom_obo_file</span><span class="p">,</span> 
    <span class="n">ont_ids</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;CL&#39;</span><span class="p">,</span> <span class="s1">&#39;FF&#39;</span><span class="p">,</span> <span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span> <span class="s1">&#39;DOID&#39;</span><span class="p">],</span>
<span class="p">)</span>

<span class="c1"># FANTOM Samples Info file:</span>
<span class="n">fantom_samples_info</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fantom_samples_info_file</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>


<span class="c1"># Glue Samples Info excerpt:</span>
<span class="c1"># --------------------------</span>
<span class="n">indices</span> <span class="o">=</span> <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">9</span><span class="p">,</span><span class="mi">11</span><span class="p">]</span>  <span class="c1"># choose rows for variety</span>
<span class="n">display</span><span class="p">(</span><span class="n">fantom_samples_info</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">indices</span><span class="p">])</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;fantom-samples-info-excerpt&quot;</span><span class="p">,</span> <span class="n">fantom_samples_info</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">indices</span><span class="p">],</span> <span class="n">display</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="fantom5">
<span id="opy-fantom5"></span><h3><span class="section-number">7.4.1.1. </span>FANTOM5<a class="headerlink" href="#fantom5" title="Permalink to this headline">¶</a></h3>
<p>Large experiments sometimes include an ontology of samples instead of or (more frequently) in addition to a samples information file.
The data from the FANTOM5 experiment<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id230"><span>234</span></a>]</span> is one such example of this.
I <a class="reference internal" href="../c06-filter/3-data.html#fantom5-expression-data"><span class="std std-ref">have already explained the FANTOM5 data in more detail</span></a> but for now the only things we need to keep in mind are that:</p>
<ol class="simple">
<li><p>The FANTOM5 experiment measures transcript expression in a wide variety of samples, across many tissue and cell types.</p></li>
<li><p>FANTOM5 provide an <a class="reference external" href="https://fantom.gsc.riken.jp/5/datafiles/latest/extra/Ontology/ff-phase2-170801.obo.txt">ontology of samples</a> as well as a <a class="reference external" href="https://fantom.gsc.riken.jp/5/datafiles/reprocessed/hg38_latest/basic/HumanSamples2.0.sdrf.xlsx">sample information file</a> (containing short text descriptions of samples).</p></li>
</ol>
<div class="figure align-default" id="fantom-samples-info-excerpt" style="width: 800px">
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Source Name</th>
      <th>Charateristics [description]</th>
      <th>Characteristics [catalog_id]</th>
      <th>Characteristics [Category]</th>
      <th>Chracteristics [Species]</th>
      <th>Characteristics [Sex]</th>
      <th>Characteristics [Age]</th>
      <th>Characteristics [Developmental stage]</th>
      <th>Characteristics[Tissue]</th>
      <th>Characteristics [Cell lot]</th>
      <th>Characteristics [Cell type]</th>
      <th>Characteristics [Catalogue ID]</th>
      <th>Characteristics [Collaboration]</th>
      <th>Characteristics [Provider]</th>
      <th>Protocol REF</th>
      <th>Extract Name</th>
      <th>Material Type</th>
    </tr>
    <tr>
      <th>Charateristics [ff_ontology]</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>FF:10002-101A5</th>
      <td>10002-101A5</td>
      <td>SABiosciences XpressRef Human Universal Total ...</td>
      <td>B208251</td>
      <td>tissues</td>
      <td>Human (Homo sapiens)</td>
      <td>mixed</td>
      <td>NaN</td>
      <td>UNDEFINED</td>
      <td>unclassifiable</td>
      <td>NaN</td>
      <td>CELL MIXTURE - tissue sample</td>
      <td>NaN</td>
      <td>FANTOM5 OSC CORE (contact: Al Forrest)</td>
      <td>SABiosciences</td>
      <td>OP-RNA-extraction-totalRNA-TRIzol-isopropanol-...</td>
      <td>10002-101A5</td>
      <td>Total RNA</td>
    </tr>
    <tr>
      <th>FF:10016-101C7</th>
      <td>10016-101C7</td>
      <td>heart, adult, pool1</td>
      <td>0910061 -7</td>
      <td>tissues</td>
      <td>Human (Homo sapiens)</td>
      <td>mixed</td>
      <td>NaN</td>
      <td>70,73,74 years old adult</td>
      <td>heart</td>
      <td>NaN</td>
      <td>CELL MIXTURE - tissue sample</td>
      <td>NaN</td>
      <td>FANTOM5 OSC CORE (contact: Al Forrest)</td>
      <td>Ambion</td>
      <td>OP-RNA-extraction-totalRNA-ToTALLY-RNA-v1.0</td>
      <td>10016-101C7</td>
      <td>Total RNA</td>
    </tr>
    <tr>
      <th>FF:10018-101C9</th>
      <td>10018-101C9</td>
      <td>liver, adult, pool1</td>
      <td>0910061 -9</td>
      <td>tissues</td>
      <td>Human (Homo sapiens)</td>
      <td>mixed</td>
      <td>NaN</td>
      <td>64,69,70 years old adult</td>
      <td>liver</td>
      <td>NaN</td>
      <td>CELL MIXTURE - tissue sample</td>
      <td>NaN</td>
      <td>FANTOM5 OSC CORE (contact: Al Forrest)</td>
      <td>Ambion</td>
      <td>OP-RNA-extraction-totalRNA-ToTALLY-RNA-v1.0</td>
      <td>10018-101C9</td>
      <td>Total RNA</td>
    </tr>
  </tbody>
</table>
</div></div></div>
<p class="caption"><span class="caption-number">Fig. 7.2 </span><span class="caption-text">An excerpt of the FANTOM sample info file, showing sources of text-based information, e.g. “heart, adult, pool1” in the <em>Charateristics [description]</em> field, and mapping to ontology term in the index (<code class="docutils literal notranslate"><span class="pre">FF:10016-101C7</span></code>).</span><a class="headerlink" href="#fantom-samples-info-excerpt" title="Permalink to this image">¶</a></p>
</div>
<p><a class="reference internal" href="#fantom-samples-info-excerpt"><span class="std std-numref">Fig. 7.2</span></a> shows an excerpt of the FANTOM Samples Information file.
This kind of file is typical of transcription experiments: a csv file containing hand-entered text-based information, using non-specific lay terms for samples e.g. “heart”.</p>
<p>The FANTOM ontology file links specific FANTOM samples to more general types of FANTOM samples and to Uberon tissues and CL cell types.</p>
<p id="fantom-obo-excerpt">For example an excerpt of the FANTOM ontology OBO file is:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[Term]
id: FF:0000076
name: hepatic sinusoidal endothelial cell sample
namespace: FANTOM5
synonym: &quot;hepatic sinusoidal endothelial cell&quot; EXACT []
is_a: FF:0000002 ! in vivo cell sample
intersection_of: FF:0000002 ! in vivo cell sample
intersection_of: derives_from CL:1000398 ! endothelial cell of hepatic sinusoid
intersection_of: derives_from UBERON:0001281 ! hepatic sinusoid
relationship: derives_from CL:1000398 ! endothelial cell of hepatic sinusoid
relationship: derives_from UBERON:0001281 ! hepatic sinusoid
created_by: tmeehan
creation_date: 2011-03-01T04:51:50Z
</pre></div>
</div>
</div>
<div class="section" id="uberon">
<span id="opyuberondata"></span><h3><span class="section-number">7.4.1.2. </span>Uberon<a class="headerlink" href="#uberon" title="Permalink to this headline">¶</a></h3>
<p>As I mentioned in <a class="reference internal" href="../c03-compbio-background/3-ontologies.html#uberon-is"><span class="std std-numref">Section 3.3.3.2</span></a>, Uberon is a cross-species anatomy ontology with excellent linkage to other ontologies.
As we can see above, the FANTOM5 ontology links FANTOM samples to Uberon.
This means that the Uberon<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id55"><span>104</span></a>]</span> <a class="reference external" href="http://purl.obolibrary.org/obo/uberon/ext.obo">extended ontology OBO file</a> can then be used to further link the samples to human disease or gene ontology terms.</p>
<p>For example, here is an excerpt of the Uberon extended OBO file (non-consecutive lines for brevity), showing how the Uberon extended ontology could be used to link a FANTOM sample to a GO term:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[Term]
id: UBERON:0001281
name: hepatic sinusoid
alt_id: UBERON:0003275
def: &quot;Wide thin-walled blood vessels in the liver. In mammals they have neither veinous or arterial markers.&quot; [http://en.wikipedia.org/wiki/Hepatic_sinusoid, ZFIN:curator]
synonym: &quot;hepatic sinusoids&quot; RELATED []
synonym: &quot;liver hepatic sinusoids&quot; EXACT [EHDAA2:0000999]
synonym: &quot;liver sinusoid&quot; EXACT []
intersection_of: part_of UBERON:0002107 ! liver
relationship: part_of UBERON:0004647 ! liver lobule
relationship: part_of UBERON:0006877 {source=&quot;https://github.com/obophenotype/uberon/wiki/Inferring-part-of-relationships&quot;} ! vasculature of liver
property_value: homology_notes &quot;(...) the amphibian liver has characteristics in common with both fish and terrestrial vertebrates. (...) The histological structure of the liver is similar to that in other vertebrates, with hepatocytes arranged in clusters and cords separated by a meshwork of sinusoids and the presence of the traditional triad of portal venule, hepatic arteriole, and bile duct.[well established][VHOG]&quot; xsd:string {date_retrieved=&quot;2012-09-17&quot;, external_class=&quot;VHOG:0000708&quot;, ontology=&quot;VHOG&quot;, source=&quot;http://bgee.unil.ch/&quot;, source=&quot;DOI:10.1053/ax.2000.7133 Crawshaw GJ, Weinkle TK, Clinical and pathological aspects of the amphibian liver. Seminars in Avian and Exotic Pet Medicine (2000)&quot;}

[Term]
id: UBERON:0002107
name: liver
def: &quot;An exocrine gland which secretes bile and functions in metabolism of protein and carbohydrate and fat, synthesizes substances involved in the clotting of the blood, synthesizes vitamin A, detoxifies poisonous substances, stores glycogen, and breaks down worn-out erythrocytes[GO].&quot; [BTO:0000759, http://en.wikipedia.org/wiki/Liver]
synonym: &quot;iecur&quot; RELATED LATIN [http://en.wikipedia.org/wiki/Liver]
is_a: UBERON:0002365 {source=&quot;BTO&quot;, source=&quot;EHDAA2&quot;, source=&quot;GO-def&quot;} ! exocrine gland
is_a: UBERON:0004119 ! endoderm-derived structure
is_a: UBERON:0005172 ! abdomen element
is_a: UBERON:0006925 ! digestive system gland
disjoint_from: UBERON:0010264 ! hepatopancreas
relationship: contributes_to_morphology_of UBERON:0002423 ! hepatobiliary system
relationship: produces UBERON:0001970 ! bile
relationship: site_of GO:0002384 ! hepatic immune response
relationship: site_of GO:0005978 ! glycogen biosynthetic process
relationship: site_of GO:0005980 ! glycogen catabolic process
property_value: external_definition &quot;Organ which secretes bile and participates in formation of certain blood proteins.[AAO]&quot; xsd:string {date_retrieved=&quot;2012-06-20&quot;, external_class=&quot;AAO:0010111&quot;, ontology=&quot;AAO&quot;, source=&quot;AAO:BJB&quot;}
property_value: function_notes &quot;secretes bile and functions in metabolism of protein and carbohydrate and fat, synthesizes substances involved in the clotting of the blood, synthesizes vitamin A, detoxifies poisonous substances, stores glycogen, and breaks down worn-out erythrocytes[GO].&quot; xsd:string
</pre></div>
</div>
<p>These excerpts show how <code class="docutils literal notranslate"><span class="pre">FF:0000076</span></code> (<em>hepatic sinusoidal endothelial cell sample</em>s) are <code class="docutils literal notranslate"><span class="pre">derived_from</span></code> the <em>hepatic sinusoid</em> which is <code class="docutils literal notranslate"><span class="pre">part_of</span></code> <em>the liver</em> the <code class="docutils literal notranslate"><span class="pre">site_of</span></code> <em>hepatic immune response</em>, <em>glycogen biosynthetic process</em> and <em>glycogen catabolic process</em>.
There are many such relationships in these files: Ontolopy provides an easy way of extracting these.</p>
<div class="warning admonition" id="ontolopy-relationship-interpretation">
<p class="admonition-title">Warning: not all relationships are easy to interpret</p>
<p>In this case, we do not have enough information to infer that <em>hepatic sinusoidal endothelial cell sample</em>s are a <code class="docutils literal notranslate"><span class="pre">site_of</span></code> (for example) the <em>hepatic immune response</em> because it could be another, disjoint, part of the liver that is the site of this.
We can also not rule it out: a more specific annotation in the future might enable us to find this out with these files.</p>
<p>However, this information could still be useful in Computational Biology. If we don’t know exactly where a process takes place, we may want to cast a wider net and look at all samples which are part of a larger tissue we know exhibits the process we are interested in.</p>
<p>This is something to be aware of in general when using Ontolopy: if you are only interested in straight-forward relationships, then you often need to think carefully about the types of relationships that you ask for: <code class="docutils literal notranslate"><span class="pre">part_of</span></code> relationships need particular care.</p>
</div>
</div>
</div>
<div class="section" id="example-1-finding-disease-related-samples">
<span id="opydiseaseexample"></span><h2><span class="section-number">7.4.2. </span>Example 1: Finding disease-related samples<a class="headerlink" href="#example-1-finding-disease-related-samples" title="Permalink to this headline">¶</a></h2>
<p>This first example shows a simple use-case of Ontolopy, where we are looking for relationships to any term in an ontology: in this case any relation to a Disease Ontology term (representing human diseases).
This Ontolopy query can be done with only the FANTOM ontology.</p>
<p>As <a class="reference internal" href="#ontolopy-relationship-interpretation"><span class="std std-ref">we just saw</span></a>, to extract relationships from ontologies (whether using Ontolopy or with any other method), you have to think about the types of relations that you are interested in.
For example, if we are interested in finding samples which are models for <code class="docutils literal notranslate"><span class="pre">DOID</span></code> disease terms, then we want to ask for <code class="docutils literal notranslate"><span class="pre">DOID</span></code> targets only, and <code class="docutils literal notranslate"><span class="pre">is_a</span></code> and <code class="docutils literal notranslate"><span class="pre">is_model_for</span></code> relationships only.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">get_disease_related_samples</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">ont</span><span class="p">):</span>
    <span class="n">disease_relations_of_interest</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">,</span><span class="s1">&#39;is_model_for&#39;</span><span class="p">]</span>
    <span class="n">disease_related</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
        <span class="n">allowed_relations</span><span class="o">=</span><span class="n">disease_relations_of_interest</span><span class="p">,</span> 
        <span class="n">sources</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">samples</span><span class="p">),</span>
        <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;DOID&#39;</span><span class="p">],</span>
        <span class="n">ont</span><span class="o">=</span><span class="n">ont</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">disease_related</span>

<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">disease_related</span> <span class="o">=</span> <span class="n">get_disease_related_samples</span><span class="p">(</span><span class="n">fantom_samples_info</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">ont</span><span class="o">=</span><span class="n">fantom_obo</span><span class="p">)</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;to&#39;</span><span class="p">])</span>
<span class="n">time_taken</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Finds </span><span class="si">{</span><span class="n">disease_related</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> disease relations in </span><span class="si">{</span><span class="n">time_taken</span><span class="si">:</span><span class="s2">.3f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Finds 566 disease relations in 0.057 seconds
</pre></div>
</div>
</div>
</div>
<p>Ontolopy can quickly (less than half a second) retrieve this information, very compactly (in one line of code if we wanted).
We can see an excerpt of the output in <a class="reference internal" href="#disease-relations-found"><span class="std std-numref">Fig. 7.3</span></a>.
This would be useful for example if we wanted to remove disease samples from the samples we were looking at.</p>
<div class="figure align-default" id="disease-relations-found" style="width: 800px">
<div class="cell_output docutils container">
<div class="output text_html"><style  type="text/css" >
#T_6ae2d_row0_col0,#T_6ae2d_row0_col1,#T_6ae2d_row0_col2,#T_6ae2d_row1_col0,#T_6ae2d_row1_col1,#T_6ae2d_row1_col2,#T_6ae2d_row2_col0,#T_6ae2d_row2_col1,#T_6ae2d_row2_col2,#T_6ae2d_row3_col0,#T_6ae2d_row3_col1,#T_6ae2d_row3_col2,#T_6ae2d_row4_col0,#T_6ae2d_row4_col1,#T_6ae2d_row4_col2{
            white-space:  pre-wrap;
            white-space:  pre-wrap;
        }</style><table id="T_6ae2d_" ><thead>    <tr>        <th class="blank level0" ></th>        <th class="col_heading level0 col0" >relation_path</th>        <th class="col_heading level0 col1" >relation_text</th>        <th class="col_heading level0 col2" >to</th>    </tr>    <tr>        <th class="index_name level0" >from</th>        <th class="blank" ></th>        <th class="blank" ></th>        <th class="blank" ></th>    </tr></thead><tbody>
                <tr>
                        <th id="T_6ae2d_level0_row0" class="row_heading level0 row0" >FF:10050-101G5</th>
                        <td id="T_6ae2d_row0_col0" class="data row0 col0" >FF:10050-101G5.is_model_for~DOID:5844</td>
                        <td id="T_6ae2d_row0_col1" class="data row0 col1" >heart, adult, diseased post-infarction, donor1 is
model for myocardial infarction</td>
                        <td id="T_6ae2d_row0_col2" class="data row0 col2" >DOID:5844</td>
            </tr>
            <tr>
                        <th id="T_6ae2d_level0_row1" class="row_heading level0 row1" >FF:10051-101G6</th>
                        <td id="T_6ae2d_row1_col0" class="data row1 col0" >FF:10051-101G6.is_model_for~DOID:114</td>
                        <td id="T_6ae2d_row1_col1" class="data row1 col1" >heart, adult, diseased, donor1 is model for heart
disease</td>
                        <td id="T_6ae2d_row1_col2" class="data row1 col2" >DOID:114</td>
            </tr>
            <tr>
                        <th id="T_6ae2d_level0_row2" class="row_heading level0 row2" >FF:10399-106A3</th>
                        <td id="T_6ae2d_row2_col0" class="data row2 col0" >FF:10399-106A3.is_a~FF:0101883.is_a~FF:0100740.i
s_model_for~DOID:8692</td>
                        <td id="T_6ae2d_row2_col1" class="data row2 col1" >acute myeloid leukemia (FAB M5) cell line:THP-1,
rep3 (fresh) is a acute myeloid leukemia cell line
sample is a myeloid leukemia cell line sample is
model for myeloid leukemia</td>
                        <td id="T_6ae2d_row2_col2" class="data row2 col2" >DOID:8692</td>
            </tr>
            <tr>
                        <th id="T_6ae2d_level0_row3" class="row_heading level0 row3" >FF:10400-106A4</th>
                        <td id="T_6ae2d_row3_col0" class="data row3 col0" >FF:10400-106A4.is_a~FF:0101883.is_a~FF:0100740.i
s_model_for~DOID:8692</td>
                        <td id="T_6ae2d_row3_col1" class="data row3 col1" >acute myeloid leukemia (FAB M5) cell line:THP-1,
rep1 (revived) is a acute myeloid leukemia cell
line sample is a myeloid leukemia cell line sample
is model for myeloid leukemia</td>
                        <td id="T_6ae2d_row3_col2" class="data row3 col2" >DOID:8692</td>
            </tr>
            <tr>
                        <th id="T_6ae2d_level0_row4" class="row_heading level0 row4" >FF:10405-106A9</th>
                        <td id="T_6ae2d_row4_col0" class="data row4 col0" >FF:10405-106A9.is_a~FF:0101883.is_a~FF:0100740.i
s_model_for~DOID:8692</td>
                        <td id="T_6ae2d_row4_col1" class="data row4 col1" >acute myeloid leukemia (FAB M5) cell line:THP-1,
rep3 (thawed) is a acute myeloid leukemia cell
line sample is a myeloid leukemia cell line sample
is model for myeloid leukemia</td>
                        <td id="T_6ae2d_row4_col2" class="data row4 col2" >DOID:8692</td>
            </tr>
    </tbody></table></div></div>
<p class="caption"><span class="caption-number">Fig. 7.3 </span><span class="caption-text">The top 5 lines of the disease-related FANTOM samples that Ontolopy found.</span><a class="headerlink" href="#disease-relations-found" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="example-2-find-tissues-that-are-capable-of-cell-differentiation">
<span id="opycelldifferentiation"></span><h2><span class="section-number">7.4.3. </span>Example 2: Find tissues that are capable of cell differentiation<a class="headerlink" href="#example-2-find-tissues-that-are-capable-of-cell-differentiation" title="Permalink to this headline">¶</a></h2>
<div class="margin sidebar">
<p class="sidebar-title"></p>
<div class="info admonition">
<p class="admonition-title">The difference between “derives from” and “develops from”</p>
<p>The Uberon extended ontology contains relations <a class="reference external" href="http://www.ontobee.org/ontology/RO?iri=http://purl.obolibrary.org/obo/RO_0001000"><code class="docutils literal notranslate"><span class="pre">derives_from</span></code></a>: a very general term that just means one comes from the other in some sense and <a class="reference external" href="http://www.ontobee.org/ontology/RO?iri=http://purl.obolibrary.org/obo/RO_0002202"><code class="docutils literal notranslate"><span class="pre">develops_from</span></code></a> which means that the two are developmentally connected, i.e. <code class="docutils literal notranslate"><span class="pre">CL:0000005</span></code> <em>fibroblast neural crest derived</em> cell <code class="docutils literal notranslate"><span class="pre">develops_from</span> <span class="pre">CL:0000008</span></code> <em>migratory cranial neural crest cell</em>, but <code class="docutils literal notranslate"><span class="pre">FF:0100003</span></code> <em>intestinal cell line sample</em> <code class="docutils literal notranslate"><span class="pre">derives_from</span></code> <code class="docutils literal notranslate"><span class="pre">UBERON:0000160</span></code> <em>intestine</em>.</p>
</div>
</div>
<p>This second example showcases a different and slightly more complex example where:</p>
<ol class="simple">
<li><p><strong>We want to look for relations to a specific term</strong> rather than a general one: in this case <code class="docutils literal notranslate"><span class="pre">GO:0030154</span></code> <em>cell differentiation</em>.</p></li>
<li><p><strong>We need to use an external ontology</strong> (Uberon), so we use the <code class="docutils literal notranslate"><span class="pre">merge</span></code> function.</p></li>
<li><p><strong>We need to chain two queries and stick them together</strong>. The <code class="docutils literal notranslate"><span class="pre">derives_from</span></code> relation in the context of the FANTOM5 ontology can mean “extracted from” or “extracted from and then do lots of things to it”. To rule out the latter type of samples we only want to ask for <em>in vivo</em> samples (<code class="docutils literal notranslate"><span class="pre">is_a</span></code> <em>in vivo sample</em> <code class="docutils literal notranslate"><span class="pre">FF:0000002</span></code>) that <code class="docutils literal notranslate"><span class="pre">derives_from</span></code> cell types that are <code class="docutils literal notranslate"><span class="pre">capable_of</span></code> <em>cell differentiation</em> (<code class="docutils literal notranslate"><span class="pre">GO:0030154</span></code>).</p></li>
</ol>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">get_differentiable_samples</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">ont</span><span class="p">):</span>
    <span class="n">in_vivo</span> <span class="o">=</span> <span class="s1">&#39;FF:0000002&#39;</span>
    <span class="n">in_vivo_samples</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
        <span class="n">allowed_relations</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">],</span> 
        <span class="n">sources</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">samples</span><span class="p">),</span> 
        <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="n">in_vivo</span><span class="p">],</span> 
        <span class="n">ont</span><span class="o">=</span><span class="n">ont</span><span class="p">,</span>
    <span class="p">)</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;to&#39;</span><span class="p">])</span>
    
    <span class="n">differentiable</span> <span class="o">=</span> <span class="s1">&#39;GO:0030154&#39;</span>
    <span class="n">differentiable_samples</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
        <span class="n">allowed_relations</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">,</span> <span class="s1">&#39;derives_from&#39;</span><span class="p">,</span> <span class="s1">&#39;capable_of&#39;</span><span class="p">],</span>
        <span class="n">sources</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">in_vivo_samples</span><span class="o">.</span><span class="n">index</span><span class="p">),</span>
        <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="n">differentiable</span><span class="p">],</span>
        <span class="n">ont</span><span class="o">=</span><span class="n">ont</span><span class="p">,</span>
    <span class="p">)</span>
    <span class="k">return</span> <span class="n">differentiable_samples</span>

<span class="c1"># merge ontology:</span>
<span class="n">merged</span> <span class="o">=</span> <span class="n">uberon_obo</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">fantom_obo</span><span class="p">)</span>

<span class="c1"># get differentiable cell samples:</span>
<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">differentiable_samples</span> <span class="o">=</span> <span class="n">get_differentiable_samples</span><span class="p">(</span><span class="n">fantom_samples_info</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">ont</span><span class="o">=</span><span class="n">merged</span><span class="p">)</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;to&#39;</span><span class="p">])</span>
<span class="n">time_taken</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Finds </span><span class="si">{</span><span class="n">differentiable_samples</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s2"> relations to cell differentiation in </span><span class="si">{</span><span class="n">time_taken</span><span class="si">:</span><span class="s2">.3f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Finds 254 relations to cell differentiation in 0.108 seconds
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="differentiable-relations-found" style="width: 800px">
<div class="cell_output docutils container">
<div class="output text_html"><style  type="text/css" >
#T_3a071_row0_col0,#T_3a071_row0_col1,#T_3a071_row0_col2,#T_3a071_row1_col0,#T_3a071_row1_col1,#T_3a071_row1_col2,#T_3a071_row2_col0,#T_3a071_row2_col1,#T_3a071_row2_col2,#T_3a071_row3_col0,#T_3a071_row3_col1,#T_3a071_row3_col2,#T_3a071_row4_col0,#T_3a071_row4_col1,#T_3a071_row4_col2{
            white-space:  pre-wrap;
            white-space:  pre-wrap;
        }</style><table id="T_3a071_" ><thead>    <tr>        <th class="blank level0" ></th>        <th class="col_heading level0 col0" >relation_path</th>        <th class="col_heading level0 col1" >relation_text</th>        <th class="col_heading level0 col2" >to</th>    </tr>    <tr>        <th class="index_name level0" >from</th>        <th class="blank" ></th>        <th class="blank" ></th>        <th class="blank" ></th>    </tr></thead><tbody>
                <tr>
                        <th id="T_3a071_level0_row0" class="row_heading level0 row0" >FF:11214-116A8</th>
                        <td id="T_3a071_row0_col0" class="data row0 col0" >FF:11214-116A8.is_a~FF:0000094.derives_from~CL:0
002569.is_a~CL:0000134.is_a~CL:0000048.is_a~CL:0
000034.is_a~CL:0011115.capable_of~GO:0030154</td>
                        <td id="T_3a071_row0_col1" class="data row0 col1" >Mesenchymal stem cell - umbilical, donor0 is a
human mesenchymal stem cell of umbilical cord-
Sciencell sample derives from mesenchymal stem
cell of umbilical cord is a mesenchymal stem cell
is a multi fate stem cell is a stem cell is a
precursor cell capable of cell differentiation</td>
                        <td id="T_3a071_row0_col2" class="data row0 col2" >GO:0030154</td>
            </tr>
            <tr>
                        <th id="T_3a071_level0_row1" class="row_heading level0 row1" >FF:11224-116B9</th>
                        <td id="T_3a071_row1_col0" class="data row1 col0" >FF:11224-116B9.is_a~FF:0000024.derives_from~CL:0
000576.is_a~CL:0011026.is_a~CL:0011115.capable_o
f~GO:0030154</td>
                        <td id="T_3a071_row1_col1" class="data row1 col1" >CD14-positive Monocytes, donor1 is a human
CD14-positive monocyte sample derives from
monocyte is a progenitor cell is a precursor cell
capable of cell differentiation</td>
                        <td id="T_3a071_row1_col2" class="data row1 col2" >GO:0030154</td>
            </tr>
            <tr>
                        <th id="T_3a071_level0_row2" class="row_heading level0 row2" >FF:11227-116C3</th>
                        <td id="T_3a071_row2_col0" class="data row2 col0" >FF:11227-116C3.is_a~FF:0000044.derives_from~CL:0
000576.is_a~CL:0011026.is_a~CL:0011115.capable_o
f~GO:0030154</td>
                        <td id="T_3a071_row2_col1" class="data row2 col1" >Dendritic Cells - monocyte immature derived,
donor1, rep1 is a human monocyte immature derived
dendritic cell sample derives from monocyte is a
progenitor cell is a precursor cell capable of
cell differentiation</td>
                        <td id="T_3a071_row2_col2" class="data row2 col2" >GO:0030154</td>
            </tr>
            <tr>
                        <th id="T_3a071_level0_row3" class="row_heading level0 row3" >FF:11229-116C5</th>
                        <td id="T_3a071_row3_col0" class="data row3 col0" >FF:11229-116C5.derives_from~CL:0000576.is_a~CL:0
011026.is_a~CL:0011115.capable_of~GO:0030154</td>
                        <td id="T_3a071_row3_col1" class="data row3 col1" >CD14+ monocyte derived endothelial progenitor
cells, donor1 derives from monocyte is a
progenitor cell is a precursor cell capable of
cell differentiation</td>
                        <td id="T_3a071_row3_col2" class="data row3 col2" >GO:0030154</td>
            </tr>
            <tr>
                        <th id="T_3a071_level0_row4" class="row_heading level0 row4" >FF:11240-116D7</th>
                        <td id="T_3a071_row4_col0" class="data row4 col0" >FF:11240-116D7.is_a~FF:0000165.derives_from~CL:0
000594.is_a~CL:0000680.is_a~CL:0000055.is_a~CL:0
011115.capable_of~GO:0030154</td>
                        <td id="T_3a071_row4_col1" class="data row4 col1" >Skeletal Muscle Satellite Cells, donor1 is a human
skeletal muscle satellite cell sample derives from
skeletal muscle satellite cell is a muscle
precursor cell is a non-terminally differentiated
cell is a precursor cell capable of cell
differentiation</td>
                        <td id="T_3a071_row4_col2" class="data row4 col2" >GO:0030154</td>
            </tr>
    </tbody></table></div></div>
<p class="caption"><span class="caption-number">Fig. 7.4 </span><span class="caption-text">An excerpt of the output of Ontolopy’s found FANTOM samples that are or derive from cells that are <code class="docutils literal notranslate"><span class="pre">capable_of</span></code> cell differentiation (<code class="docutils literal notranslate"><span class="pre">GO:0030154</span></code>).</span><a class="headerlink" href="#differentiable-relations-found" title="Permalink to this image">¶</a></p>
</div>
<p>Again Ontolopy can retrieve this information compactly (2 lines of code), and in less than half a second.
An excerpt of the output is shown in <a class="reference internal" href="#differentiable-relations-found"><span class="std std-numref">Fig. 7.4</span></a>.
This would be useful if we wanted to look at expression in tissues that are capable of cell differentiation, for example.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c07-ontolopy"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="3-how-it-works.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">7.3. </span>Ontolopy tools and practices</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="5-mapping-example.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">7.5. </span>Example use: mapping samples to tissue-related phenotypes</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Natalie Zelenka<br/>
  
      &copy; Copyright 2020.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>