
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>7.5. Example use: mapping samples to tissue-related phenotypes &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.e8f53015daec13862f6db5e763c41738.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="7.6. Discussion" href="6-discussion.html" />
    <link rel="prev" title="7.4. Example uses: mapping samples to diseases or phenotypes" href="4-misc-examples.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   7. Ontolopy
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#creating-sample-to-tissue-mappings">
   7.5.1. Creating sample-to-tissue mappings
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#load-data-and-pre-filter">
     7.5.1.1. Load data and pre-filter
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-by-ontology">
     7.5.1.2. Mapping by ontology
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-by-name">
     7.5.1.3. Mapping by name
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#combining-mappings">
     7.5.1.4. Combining mappings
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#finding-inconsistencies">
     7.5.1.5. Finding inconsistencies
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#finding-samples-that-are-missing-annotations-to-tissues">
       7.5.1.5.1. Finding samples that are missing annotations to tissues
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#missing-uberon-or-cl-annotation">
       7.5.1.5.2. Missing Uberon or CL annotation
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#mislabelled-sample">
       7.5.1.5.3. Mislabelled sample
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#imprecise-annotation-to-tissue">
       7.5.1.5.4. Imprecise annotation to tissue
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-overview">
     7.5.1.6. Mapping overview
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#creating-tissue-to-phenotype-mappings">
   7.5.2. Creating tissue-to-phenotype mappings
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#propagating-relationships-up-the-tree-using-part-of">
     7.5.2.1. Propagating relationships up the tree using
     <code class="docutils literal notranslate">
      <span class="pre">
       part_of
      </span>
     </code>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#propagating-down-the-tree-has-part">
     7.5.2.2. Propagating “down” the tree:
     <code class="docutils literal notranslate">
      <span class="pre">
       has_part
      </span>
     </code>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#propagating-down-the-tree-inverse-of-part-of">
     7.5.2.3. Propagating down the tree: inverse of
     <code class="docutils literal notranslate">
      <span class="pre">
       part_of
      </span>
     </code>
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#combining-previous-mappings">
     7.5.2.4. Combining previous mappings
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#creating-sample-to-tissue-phenotype-mappings">
   7.5.3. Creating sample-to-tissue-phenotype mappings
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#final-mapping">
     7.5.3.1. Final mapping
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="example-use-mapping-samples-to-tissue-related-phenotypes">
<span id="ontolopy-mapping-example"></span><h1><span class="section-number">7.5. </span>Example use: mapping samples to tissue-related phenotypes<a class="headerlink" href="#example-use-mapping-samples-to-tissue-related-phenotypes" title="Permalink to this headline">¶</a></h1>
<!-- TODO: Give `sample_to_tissue_mapping` and `mapped_by_tissue_name` more similar names -->
<!-- TODO: Check that I've put propagate up and down the right way around -->
<p>This section presents a more sizable example of using Ontolopy, the task it was developed for: mapping from samples to tissue-related phenotypes.
This is a substantial challenge, in part because it requires:</p>
<ol class="simple">
<li><p>Mapping over many different types of ontology terms (<code class="docutils literal notranslate"><span class="pre">FF</span></code>, <code class="docutils literal notranslate"><span class="pre">UBERON</span></code>, <code class="docutils literal notranslate"><span class="pre">CL</span></code>, <code class="docutils literal notranslate"><span class="pre">GO</span></code>).</p></li>
<li><p>Using more <a class="reference internal" href="#has-part-part-of"><span class="std std-ref">complex</span></a> relations such as <code class="docutils literal notranslate"><span class="pre">part_of</span></code>.</p></li>
<li><p>Mapping from text as well as using the mapping from ontology, which then requires combining the two mappings into an overall mapping and investigating any disagreements between mappings.</p></li>
</ol>
<p>This task can be generally divided into the following parts:</p>
<ol class="simple">
<li><p>Creating sample (<code class="docutils literal notranslate"><span class="pre">FF</span></code>) to tissue (<code class="docutils literal notranslate"><span class="pre">UBERON</span></code>) <a class="reference internal" href="#sample-to-tissue"><span class="std std-ref">mapping</span></a>, including looking at <a class="reference internal" href="#fantom5-inconsistencies-example"><span class="std std-ref">disagreements</span></a> between mappings.</p></li>
<li><p>Tissue (<code class="docutils literal notranslate"><span class="pre">UBERON</span></code>) to phenotype (<code class="docutils literal notranslate"><span class="pre">GO</span></code> Biological Process - GOBP) <a class="reference internal" href="#tissue-to-phenotype"><span class="std std-ref">mapping</span></a>.</p></li>
<li><p>Combining the above, to create the final sample (<code class="docutils literal notranslate"><span class="pre">FF</span></code>) to tissue-related phenotype (<code class="docutils literal notranslate"><span class="pre">GO</span></code>) <a class="reference internal" href="#sample-tissue"><span class="std std-ref">mapping</span></a>.</p></li>
</ol>
<p>We are using the same input data as described in <a class="reference internal" href="4-misc-examples.html#ontolopy-example-inputs"><span class="std std-ref">the previous section</span></a> for this example.</p>
<div class="section" id="creating-sample-to-tissue-mappings">
<span id="sample-to-tissue"></span><h2><span class="section-number">7.5.1. </span>Creating sample-to-tissue mappings<a class="headerlink" href="#creating-sample-to-tissue-mappings" title="Permalink to this headline">¶</a></h2>
<p>To create the mapping between FANTOM sample ID (<code class="docutils literal notranslate"><span class="pre">FF:XXXXX-XXXXX</span></code>) and tissue (<code class="docutils literal notranslate"><span class="pre">UBERON:XXXXXX</span></code>), we use the <code class="docutils literal notranslate"><span class="pre">Uberon</span></code> class.
The <code class="docutils literal notranslate"><span class="pre">Uberon</span></code> class has three useful functions for creating this mapping:</p>
<ol class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">sample_map_by_ont</span></code>: creates a mapping via ontology.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sample_map_by_name</span></code>: creates a mapping via sample or tissue names.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">get_overall_tissue_mappings</span></code>: combines the two mappings to create a more comprehensive overall mapping.</p></li>
</ol>
<div class="section" id="load-data-and-pre-filter">
<span id="map-load-prefilter"></span><h3><span class="section-number">7.5.1.1. </span>Load data and pre-filter<a class="headerlink" href="#load-data-and-pre-filter" title="Permalink to this headline">¶</a></h3>
<p>In order to do this, I load the input FANTOM5 ontology and sample information files.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">ontolopy</span> <span class="k">as</span> <span class="nn">opy</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span> 
<span class="kn">from</span> <span class="nn">myst_nb</span> <span class="kn">import</span> <span class="n">glue</span>
<span class="kn">import</span> <span class="nn">time</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">logging</span>
<span class="kn">import</span> <span class="nn">plotly.graph_objects</span> <span class="k">as</span> <span class="nn">go_</span>
<span class="kn">from</span> <span class="nn">plotly.subplots</span> <span class="kn">import</span> <span class="n">make_subplots</span>

<span class="n">notebook_start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>

<span class="c1"># Read in files:</span>
<span class="c1"># -------------</span>
<span class="n">fantom_obo_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/experiments/fantom/ff-phase2-170801.obo.txt&#39;</span>
<span class="n">fantom_samples_info_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/experiments/fantom/fantom_humanSamples2.0.csv&#39;</span>
<span class="n">uberon_obo_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/uberon_ext_210321.obo&#39;</span> 

<span class="c1"># Uberon OBO:</span>
<span class="n">uberon_obo</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">load_obo</span><span class="p">(</span>
    <span class="n">file_loc</span><span class="o">=</span><span class="n">uberon_obo_file</span><span class="p">,</span> 
    <span class="n">ont_ids</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span><span class="s1">&#39;CL&#39;</span><span class="p">,</span><span class="s1">&#39;FMA&#39;</span><span class="p">],</span> 
<span class="p">)</span>

<span class="n">uberon_obo_tissue_only</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">load_obo</span><span class="p">(</span>
    <span class="n">file_loc</span><span class="o">=</span><span class="n">uberon_obo_file</span><span class="p">,</span> 
    <span class="n">ont_ids</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span><span class="s1">&#39;FMA&#39;</span><span class="p">],</span> 
<span class="p">)</span>

<span class="c1"># FANTOM OBO:</span>
<span class="n">fantom_obo</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">load_obo</span><span class="p">(</span>
    <span class="n">file_loc</span><span class="o">=</span><span class="n">fantom_obo_file</span><span class="p">,</span> 
    <span class="n">ont_ids</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;CL&#39;</span><span class="p">,</span> <span class="s1">&#39;FF&#39;</span><span class="p">,</span> <span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span> <span class="s1">&#39;DOID&#39;</span><span class="p">],</span>
<span class="p">)</span>

<span class="n">fantom_obo_no_cl</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">load_obo</span><span class="p">(</span>
    <span class="n">file_loc</span><span class="o">=</span><span class="n">fantom_obo_file</span><span class="p">,</span> 
    <span class="n">ont_ids</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;FF&#39;</span><span class="p">,</span> <span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span> <span class="s1">&#39;DOID&#39;</span><span class="p">],</span>
<span class="p">)</span>

<span class="c1"># FANTOM Samples Info file:</span>
<span class="n">samples_info</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fantom_samples_info_file</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="mapping-by-ontology">
<span id="bigexamplemapbyont"></span><h3><span class="section-number">7.5.1.2. </span>Mapping by ontology<a class="headerlink" href="#mapping-by-ontology" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal notranslate"><span class="pre">sample_map_by_ont</span></code> function is a wrapper function which calls <code class="docutils literal notranslate"><span class="pre">relations.Relations</span></code>, and excludes too-general Uberon tissues such as anatomical structure, tissue, anatomical system, embryo, and multi fate stem cell.
We use a merged sample (FANTOM) and tissue (Uberon) ontology as input.</p>
<p>The inclusion of the Cell Ontology (CL) terms (which are included in the Uberon OBO file) is important to retrieve a mapping for as many samples as possible.
<a class="reference internal" href="#cl-increases-coverage"><span class="std std-numref">Table 7.1</span></a> shows how the inclusion of CL terms in the input ontology significantly changes the mapping coverage, and that mapping via ontology alone (with CL terms used) is fairly good.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Find mappings:</span>
<span class="c1"># -------------</span>

<span class="n">total</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>

<span class="k">def</span> <span class="nf">get_mapped</span><span class="p">(</span><span class="n">mapping</span><span class="p">):</span>
    <span class="n">mapped</span> <span class="o">=</span> <span class="n">mapping</span><span class="p">[</span><span class="o">~</span><span class="n">mapping</span><span class="p">[</span><span class="s1">&#39;to&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span>
    <span class="n">unmapped</span> <span class="o">=</span> <span class="n">mapping</span><span class="p">[</span><span class="n">mapping</span><span class="p">[</span><span class="s1">&#39;to&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span>
    <span class="k">return</span> <span class="n">mapped</span><span class="p">,</span> <span class="n">unmapped</span>

<span class="c1"># Uberon tissue only (no CL cells)</span>
<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">merged_tissue_only</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">uberon_from_obo</span><span class="p">(</span><span class="n">uberon_obo_tissue_only</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">fantom_obo_no_cl</span><span class="p">))</span>
<span class="n">sample_to_tissue_mapping_tissue_only</span> <span class="o">=</span> <span class="n">merged_tissue_only</span><span class="o">.</span><span class="n">sample_map_by_ont</span><span class="p">(</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="n">mapped_nocl</span><span class="p">,</span> <span class="n">unmapped_nocl</span> <span class="o">=</span> <span class="n">get_mapped</span><span class="p">(</span><span class="n">sample_to_tissue_mapping_tissue_only</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-uberon-no-cl&quot;</span><span class="p">,</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped_nocl</span><span class="p">)</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped_nocl</span><span class="p">)</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">total</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-uberon-no-cl&quot;</span><span class="p">,</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">unmapped_nocl</span><span class="p">)</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">unmapped_nocl</span><span class="p">)</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">total</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;time-uberon-no-cl&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>

<span class="c1"># Uberon tissue and CL cells </span>
<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">merged</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">uberon_from_obo</span><span class="p">(</span><span class="n">uberon_obo</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">fantom_obo</span><span class="p">))</span>
<span class="n">sample_to_tissue_mapping</span> <span class="o">=</span> <span class="n">merged</span><span class="o">.</span><span class="n">sample_map_by_ont</span><span class="p">(</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="n">mapped_cl</span><span class="p">,</span> <span class="n">unmapped_cl</span> <span class="o">=</span> <span class="n">get_mapped</span><span class="p">(</span><span class="n">sample_to_tissue_mapping</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-uberon-cl&quot;</span><span class="p">,</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped_cl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped_cl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">total</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-uberon-cl&quot;</span><span class="p">,</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">unmapped_cl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">unmapped_cl</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">total</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;time-uberon-cl&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<table class="table" id="cl-increases-coverage">
<caption><span class="caption-number">Table 7.1 </span><span class="caption-text">Table comparing the difference in coverage (mappable samples) for different mapping techniques.</span><a class="headerlink" href="#cl-increases-coverage" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Mapping name</p></th>
<th class="head"><p>Number (and percentage) of all mapped samples</p></th>
<th class="head"><p>Number (and percentage) of all unmapped samples</p></th>
<th class="head"><p>Run time</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>By ontology: using Uberon tissues only</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>By ontology: using Uberon tissues and CL cells</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>By name: using tissue column</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Combined: combining by name and by ontology including CL</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p><strong>N/A</strong></p></td>
</tr>
</tbody>
</table>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Show some examples of things that aren&#39;t matched that should be (not mappable by ontology alone), e.g. caudate nucleus</span>
<span class="n">tissue_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics[Tissue]&#39;</span>

<span class="n">expect_to_map</span> <span class="o">=</span> <span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">unmapped_cl</span><span class="o">.</span><span class="n">index</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span> <span class="p">(</span><span class="o">~</span><span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">())</span> <span class="o">&amp;</span> <span class="o">~</span><span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;ANATOMICAL SYSTEM&#39;</span><span class="p">,</span> <span class="s1">&#39;UNDEFINED_TISSUE_TYPE&#39;</span><span class="p">,</span> <span class="s1">&#39;unclassifiable&#39;</span><span class="p">]))]</span>
<span class="n">not_expect_map</span> <span class="o">=</span> <span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">unmapped_cl</span><span class="o">.</span><span class="n">index</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()</span> <span class="o">|</span> <span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">([</span><span class="s1">&#39;ANATOMICAL SYSTEM&#39;</span><span class="p">,</span> <span class="s1">&#39;UNDEFINED_TISSUE_TYPE&#39;</span><span class="p">,</span> <span class="s1">&#39;unclassifiable&#39;</span><span class="p">]))]</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;not_expect_map&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">not_expect_map</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;expect_map&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">expect_to_map</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num_expect_map_tissues&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">expect_to_map</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()),</span> <span class="kc">False</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">list_to_text</span><span class="p">(</span><span class="n">lst</span><span class="p">):</span>
    <span class="n">text</span> <span class="o">=</span> <span class="s1">&#39;, &#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">text</span><span class="p">[:</span><span class="n">text</span><span class="o">.</span><span class="n">rindex</span><span class="p">(</span><span class="s1">&#39;, &#39;</span><span class="p">)</span> <span class="o">+</span> <span class="mi">1</span><span class="p">]</span> <span class="o">+</span> <span class="s1">&#39; and&#39;</span> <span class="o">+</span> <span class="n">text</span><span class="p">[</span><span class="n">text</span><span class="o">.</span><span class="n">rindex</span><span class="p">(</span><span class="s1">&#39;, &#39;</span><span class="p">)</span><span class="o">+</span><span class="mi">1</span><span class="p">:]</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;tissues_list&quot;</span><span class="p">,</span> <span class="n">list_to_text</span><span class="p">(</span><span class="n">expect_to_map</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>

<span class="n">chosen_examples</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;FF:10379-105H1&#39;</span><span class="p">,</span>  <span class="c1"># caudate nucleus</span>
                   <span class="s1">&#39;FF:10422-106C8&#39;</span><span class="p">,</span>  <span class="c1"># blood (lymphoma)</span>
                   <span class="s1">&#39;FF:10558-107I9&#39;</span><span class="p">,</span> <span class="c1"># osteoblast</span>
                   <span class="s1">&#39;FF:11794-124C3&#39;</span><span class="p">,</span> <span class="c1"># t-cell</span>
                  <span class="p">]</span>
<span class="n">cell_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics [Cell type]&#39;</span>
<span class="n">desc_col</span> <span class="o">=</span> <span class="s1">&#39;Charateristics [description]&#39;</span>
<span class="n">expect_to_map_table</span> <span class="o">=</span> <span class="n">expect_to_map</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chosen_examples</span><span class="p">][[</span><span class="n">desc_col</span><span class="p">,</span> <span class="n">tissue_col</span><span class="p">,</span> <span class="n">cell_col</span><span class="p">]]</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;expect-to-map-table&quot;</span><span class="p">,</span> <span class="n">expect_to_map_table</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p id="expect-to-map">The unmapped samples do contain some samples that we wouldn’t expect to be able to map to tissue (defined as those that do not have a tissue provided in the samples information file, or that are labelled as <code class="docutils literal notranslate"><span class="pre">unclassifiable</span></code>, <code class="docutils literal notranslate"><span class="pre">UNDEFINED_TISSUE_TYPE</span></code>, or <code class="docutils literal notranslate"><span class="pre">ANATOMICAL</span> <span class="pre">SYSTEM</span></code>): these account for  of the  unmapped samples.
This, however, leaves  samples which we would expect to map, spread across  tissues ().</p>
<div class="figure align-default" id="expect-to-map-examples" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.5 </span><span class="caption-text">Four FANTOM5 samples that we would expect to be able to map using Ontolopy based on the information in the samples information file (selected columns shown here), but that do not map using Ontolopy with the FANTOM5 ontology.</span><a class="headerlink" href="#expect-to-map-examples" title="Permalink to this image">¶</a></p>
</div>
<p><a class="reference internal" href="#expect-to-map-examples"><span class="std std-numref">Fig. 7.5</span></a> show four examples of such samples.
The existence of such tissues, means that mapping via name as well as by ontology could prove useful.</p>
</div>
<div class="section" id="mapping-by-name">
<span id="bigexamplemapbyname"></span><h3><span class="section-number">7.5.1.3. </span>Mapping by name<a class="headerlink" href="#mapping-by-name" title="Permalink to this headline">¶</a></h3>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># map by name</span>
<span class="n">uberon_obo_tissue_only</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">uberon_from_obo</span><span class="p">(</span><span class="n">uberon_obo_tissue_only</span><span class="p">)</span>

<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">mapped_by_tissue_name</span> <span class="o">=</span> <span class="n">uberon_obo_tissue_only</span><span class="o">.</span><span class="n">sample_map_by_name</span><span class="p">(</span><span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">],</span> <span class="n">xref</span><span class="o">=</span><span class="s1">&#39;FMA&#39;</span><span class="p">)</span>
<span class="n">mapped_name</span><span class="p">,</span> <span class="n">unmapped_name</span> <span class="o">=</span> <span class="n">get_mapped</span><span class="p">(</span><span class="n">mapped_by_tissue_name</span><span class="p">)</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;time-name&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-name&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped_name</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped_name</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">total</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-name&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">unmapped_name</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">unmapped_name</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">total</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">Uberon.sample_map_by_name</span></code> function simply looks up the strings provided (in this case those from the <code class="docutils literal notranslate"><span class="pre">Characteristics[Tissue]</span></code> column of the sample information file) and checks if any Uberon terms in the provided ontology has a matching name or synonym.
The term name is preferred over synonyms, and where there are no exactly matching term names, but there are multiple possible synonyms (e.g. <em>bladder</em> is a synonym for <em>urinary bladder</em> and <em>bladder organ</em>), we decide by whether either of the terms are linked to the Foundational Model of Anatomy (FMA) ontology, as this is a human-specific ontology by using the <code class="docutils literal notranslate"><span class="pre">xref='FMA'</span></code> option.
Since we are only looking for Uberon terms, it doesn’t make any difference whether we use the “tissue only” or “including CL” versions of the Uberon ontology that we read in earlier, aside from a negligible difference in run time).</p>
<div class="cell tag_hide-input tag_hide-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">unmapped_name</span><span class="p">[</span><span class="s1">&#39;name_matched_on&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<p>Ontolopy restricts the tissues mapped by name to <code class="docutils literal notranslate"><span class="pre">NARROW</span></code>, <code class="docutils literal notranslate"><span class="pre">EXACT</span></code>, and <code class="docutils literal notranslate"><span class="pre">BROAD</span></code> synonyms (other synonyms include “RELATED”).
These synonyms usually includes what we want, but will miss some less closely related synonyms.
Looking at the tissues that are unmapped by name can help us identify any that we might want to treat differently.
For example, <em>cartilage</em> doesn’t map to <code class="docutils literal notranslate"><span class="pre">UBERON:0002418</span></code> <em>cartilage tissue</em> as the synonym <em>cartilage</em> is <code class="docutils literal notranslate"><span class="pre">RELATED</span></code>.
It’s also useful to see that the formatting of the some names, e.g. “Fingernail (including nail plate, eponychium and hyponychium)” and “eye - vitreous humor” prevent the Ontolopy algorithm from recognising the names.
We could map these to more standardised names with a dictionary and rerun the algorithm if we had no other option, but in this case we simply use the by-ontology mapping to map terms with unmappable tissue names.</p>
<div class="cell tag_remove-input tag_remove-output docutils container">
</div>
<div class="figure align-default" id="example-imprecise" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.6 </span><span class="caption-text">Comparisons of sample mappings by ontology (inlcluding CL) and by name (by tissue column of samples information file), illustrating the tendency of by-name mappings to be less precise.</span><a class="headerlink" href="#example-imprecise" title="Permalink to this image">¶</a></p>
</div>
<p>As <a class="reference internal" href="#cl-increases-coverage"><span class="std std-numref">Table 7.1</span></a> shows, mapping by name function is quite slow, taking .
The mappings coverage is .</p>
<p>This measure doesn’t show that the mapping is constrained to be less precise than the mapping via ontology - of course this is particular to the data as it depends on how the tissues were labelled.
In this case, FANTOM5 labelled the tissues fairly broadly, so we see examples like <code class="docutils literal notranslate"><span class="pre">FF:10075-102A3</span></code> (<em>lung, right lower lobe</em>) and <code class="docutils literal notranslate"><span class="pre">FF:14331-155F2</span></code> (<em>Fibroblast - Aortic Adventitial</em>) in <a class="reference internal" href="#example-imprecise"><span class="std std-numref">Fig. 7.6</span></a> - this is particularly common when the samples are cell types.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Examples of things that would be unmappable by text alone &quot;Anatomical system&quot;</span>
<span class="n">tissue_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics[Tissue]&#39;</span>

<span class="n">anatomical_system</span> <span class="o">=</span> <span class="n">samples_info</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;ANATOMICAL SYSTEM&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="n">mapped_cl</span><span class="o">.</span><span class="n">index</span><span class="p">)][[</span><span class="n">desc_col</span><span class="p">,</span> <span class="n">tissue_col</span><span class="p">]]</span>

<span class="n">anatomical_system</span><span class="p">[</span><span class="s1">&#39;Mapped by ontology&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">mapped_cl</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">anatomical_system</span><span class="o">.</span><span class="n">index</span><span class="p">][</span><span class="s1">&#39;to&#39;</span><span class="p">]]</span>
<span class="n">anatomical_system</span><span class="p">[</span><span class="s1">&#39;Mapped by name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">mapped_by_tissue_name</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">anatomical_system</span><span class="o">.</span><span class="n">index</span><span class="p">][</span><span class="s1">&#39;to&#39;</span><span class="p">]]</span>

<span class="n">illustrative_samples</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;FF:11966-126D4&#39;</span><span class="p">,</span> 
                        <span class="s1">&#39;FF:11941-126A6&#39;</span><span class="p">,</span> 
                        <span class="s1">&#39;FF:11937-126A2&#39;</span><span class="p">,</span> 
                        <span class="s1">&#39;FF:11936-126A1&#39;</span><span class="p">,</span> 
                        <span class="s1">&#39;FF:11930-125I4&#39;</span><span class="p">,</span> 
                        <span class="s1">&#39;FF:11927-125I1&#39;</span><span class="p">]</span>

<span class="n">examples_rows</span> <span class="o">=</span> <span class="n">anatomical_system</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">illustrative_samples</span><span class="p">]</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;anatomical-system-examples-table&quot;</span><span class="p">,</span> <span class="n">examples_rows</span><span class="p">,</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>In addition, there are samples that are not usefully mapped at all (or completely missing a mapping) using the by-name approach, that we can get by-ontology.
One subset of these is the tissues which are labelled “ANATOMICAL SYSTEM”: these map to the Uberon term <em>anatomical system</em>, but this is too general to be useful.
<a class="reference internal" href="#anatomical-system"><span class="std std-numref">Fig. 7.7</span></a> shows how these terms can be mapped to more localised terms</p>
<p>This also shows the one limitation of mapping by ontology, in that we might expect that samples such as <code class="docutils literal notranslate"><span class="pre">FF:11936-126A1</span></code> to be mapped to the more specific (and useful) <code class="docutils literal notranslate"><span class="pre">UBERON:0001997</span></code> <em>olfactory epithelium</em>.
This is not the case since there is a “missing” annotation between <code class="docutils literal notranslate"><span class="pre">CL:0002167</span></code> <em>olfactory epithelial cell</em> and <em>olfactory epithelium</em>, since the definition of olfactory epithelial cell is <a class="reference external" href="https://github.com/obophenotype/cell-ontology/issues/167">still under discussion and development</a>.</p>
<div class="figure align-default" id="anatomical-system" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.7 </span><span class="caption-text">An example of <em>ANATOMICAL SYSTEM</em> tissue samples, with tissue-specific cells, which are unmapped by-name, but have useful by-ontology mappings.</span><a class="headerlink" href="#anatomical-system" title="Permalink to this image">¶</a></p>
</div>
<p id="immune-system-mappings">There are also some benefits to the name based mapping.
A quirk of the ontology-based mapping is that many cell types are identified as having being part of the <em>immune system</em>, however, this isn’t a well-defined locality.
It’s possible for samples from different physical locations (e.g. liver, blood) to map to the <em>immune system</em> term.
For the FANTOM5 data at least, the names better describe locations that these samples came from.</p>
</div>
<div class="section" id="combining-mappings">
<span id="bigexamplecombiningmappings"></span><h3><span class="section-number">7.5.1.4. </span>Combining mappings<a class="headerlink" href="#combining-mappings" title="Permalink to this headline">¶</a></h3>
<p>To get the best of both mappings, we need to combine them using the <code class="docutils literal notranslate"><span class="pre">Uberon.get_overall_tissue_mappings</span></code> function.
This function creates both an overall mapping and a list of disagreements.
Where only one mapping covers a term, it is trivial to do this (the overall mapping uses the present mapping, and there are no disagreements).
When both mappings are present and one term is an ancestor of another, we say there is no disagreement and choose the more specific mapping e.g. if mapping by ontology gives us <em>photoreceptor array</em>, but mapping by name gives us <em>eye</em>, then because <em>photoreceptor array</em> is <code class="docutils literal notranslate"><span class="pre">part_of</span></code> <em>eye</em> and <em>eye</em> <code class="docutils literal notranslate"><span class="pre">is_a</span></code> <em>sense organ</em>, we would use the overall mapping by ontology since <em>photoreceptor array</em> is the more specific term.
When both mappings are present and there is no relationship between them, this is when we say there is a disagreement, and we can choose which mapping we give precedence to, by default it is the ontology mapping.</p>
<p>Before we create the overall mapping, we will remove the <em>immune system</em> by-ontology mappings, for the reasons <a class="reference internal" href="#immune-system-mappings"><span class="std std-ref">discussed above</span></a>.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Remove immune system mappings</span>
<span class="n">immune_system</span> <span class="o">=</span> <span class="s1">&#39;UBERON:0002405&#39;</span>
<span class="k">for</span> <span class="n">ff</span> <span class="ow">in</span> <span class="n">sample_to_tissue_mapping</span><span class="p">[(</span><span class="n">sample_to_tissue_mapping</span><span class="p">[</span><span class="s1">&#39;to&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">immune_system</span><span class="p">)</span> <span class="o">&amp;</span> <span class="n">sample_to_tissue_mapping</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">mapped_by_tissue_name</span><span class="p">[</span><span class="o">~</span><span class="n">mapped_by_tissue_name</span><span class="p">[</span><span class="s1">&#39;to&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="n">sample_to_tissue_mapping</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">ff</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">]</span>

<span class="c1"># Combine mappings</span>
<span class="n">overall</span><span class="p">,</span> <span class="n">disagreements</span> <span class="o">=</span> <span class="n">merged</span><span class="o">.</span><span class="n">get_overall_tissue_mappings</span><span class="p">(</span>
    <span class="n">map_by_ont</span><span class="o">=</span><span class="n">sample_to_tissue_mapping</span><span class="p">,</span> 
    <span class="n">map_by_name</span><span class="o">=</span><span class="n">mapped_by_tissue_name</span><span class="p">,</span>
    <span class="n">rel</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">,</span> <span class="s1">&#39;part_of&#39;</span><span class="p">,</span> <span class="s1">&#39;continuous_with&#39;</span><span class="p">,</span> <span class="s1">&#39;has_potential_to_develop_into&#39;</span><span class="p">,</span> <span class="s1">&#39;develops_into&#39;</span><span class="p">]</span>
<span class="p">)</span>

<span class="n">unmapped_overall</span> <span class="o">=</span> <span class="n">overall</span><span class="p">[</span><span class="n">overall</span><span class="p">[</span><span class="s1">&#39;mapped_by&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span>
<span class="n">mapped_overall</span> <span class="o">=</span> <span class="n">overall</span><span class="p">[</span><span class="o">~</span><span class="n">overall</span><span class="p">[</span><span class="s1">&#39;mapped_by&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;overall-unique-tissues&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">overall</span><span class="p">[</span><span class="s1">&#39;overall&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()),</span> <span class="kc">False</span><span class="p">)</span>

<span class="c1"># Add overall to table</span>
<span class="k">def</span> <span class="nf">format_mapped</span><span class="p">(</span><span class="n">mapped</span><span class="p">,</span> <span class="n">total</span><span class="p">):</span>
    <span class="k">return</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped</span><span class="p">)</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">mapped</span><span class="p">)</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">total</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-overall&quot;</span><span class="p">,</span> <span class="n">format_mapped</span><span class="p">(</span><span class="n">mapped_overall</span><span class="p">,</span> <span class="n">total</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-overall&quot;</span><span class="p">,</span> <span class="n">format_mapped</span><span class="p">(</span><span class="n">unmapped_overall</span><span class="p">,</span> <span class="n">total</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p id="primary-cell-tissues-ontolopy"><a class="reference internal" href="../c06-filter/3-data.html#primary-cell-tissue"><span class="std std-ref">The FANTOM5 data contains different categories of samples</span></a> including tissues, time courses, immortal cell lines, fractionations and purturbations, and primary cells.
Some of these categories might not map in the way that we might want them to because although they might be a cell type that is usually localised to a tissue, they are unusual since they represent unusual in-between developing tissues (e.g. stem cells) or cancerous immortal cell lines.
This is likely to have led to uncertainties in the sample ontology file, so by restricting to primary cell and tissue samples, we might get a more accurate picture of the percentage of mappable samples that Ontolopy can reach.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Glue all values for the primary cell and tissues version of the table `coverage-tissue-primary-cell`</span>
<span class="k">def</span> <span class="nf">primary_and_tissues</span><span class="p">(</span><span class="n">mapped</span><span class="p">,</span> <span class="n">unmapped</span><span class="p">,</span> <span class="n">samples_info</span><span class="p">,</span> <span class="n">name_string</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;Removes non primary cell and tissue samples from the `mapped` and `unmapped` samples.&quot;&quot;&quot;</span>
    <span class="n">category_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics [Category]&#39;</span>
    <span class="n">categories</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;primary cells&#39;</span><span class="p">,</span> <span class="s1">&#39;tissues&#39;</span><span class="p">]</span>
        
    <span class="n">category_samples</span> <span class="o">=</span> <span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="p">[</span><span class="n">category_col</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">categories</span><span class="p">)]</span><span class="o">.</span><span class="n">index</span>
    <span class="n">total</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">category_samples</span><span class="p">)</span>
    
    <span class="n">new_mapped</span> <span class="o">=</span> <span class="n">mapped</span><span class="p">[</span><span class="n">mapped</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">category_samples</span><span class="p">)]</span>
    <span class="n">new_unmapped</span> <span class="o">=</span> <span class="n">unmapped</span><span class="p">[</span><span class="n">unmapped</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">category_samples</span><span class="p">)]</span>
    
    <span class="n">glue</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">name_string</span><span class="si">}</span><span class="s2">_mapped_pt&quot;</span><span class="p">,</span> <span class="n">format_mapped</span><span class="p">(</span><span class="n">new_mapped</span><span class="p">,</span> <span class="n">total</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">glue</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">name_string</span><span class="si">}</span><span class="s2">_unmapped_pt&quot;</span><span class="p">,</span> <span class="n">format_mapped</span><span class="p">(</span><span class="n">new_unmapped</span><span class="p">,</span> <span class="n">total</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">new_mapped</span><span class="p">,</span> <span class="n">new_unmapped</span>

<span class="n">mapped_nocl_pt</span><span class="p">,</span> <span class="n">unmapped_nocl_pt</span> <span class="o">=</span> <span class="n">primary_and_tissues</span><span class="p">(</span><span class="n">mapped_nocl</span><span class="p">,</span> <span class="n">unmapped_nocl</span><span class="p">,</span> <span class="n">samples_info</span><span class="p">,</span> <span class="s1">&#39;nocl&#39;</span><span class="p">)</span>
<span class="n">mapped_cl_pt</span><span class="p">,</span> <span class="n">unmapped_cl_pt</span> <span class="o">=</span> <span class="n">primary_and_tissues</span><span class="p">(</span><span class="n">mapped_cl</span><span class="p">,</span> <span class="n">unmapped_cl</span><span class="p">,</span> <span class="n">samples_info</span><span class="p">,</span> <span class="s1">&#39;cl&#39;</span><span class="p">)</span>
<span class="n">mapped_name_pt</span><span class="p">,</span> <span class="n">unmapped_name_pt</span> <span class="o">=</span> <span class="n">primary_and_tissues</span><span class="p">(</span><span class="n">mapped_name</span><span class="p">,</span> <span class="n">unmapped_name</span><span class="p">,</span> <span class="n">samples_info</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">)</span>
<span class="n">mapped_overall_pt</span><span class="p">,</span> <span class="n">unmapped_overall_pt</span> <span class="o">=</span> <span class="n">primary_and_tissues</span><span class="p">(</span><span class="n">mapped_overall</span><span class="p">,</span> <span class="n">unmapped_overall</span><span class="p">,</span> <span class="n">samples_info</span><span class="p">,</span> <span class="s1">&#39;overall&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<table class="table" id="coverage-tissue-primary-cell">
<caption><span class="caption-number">Table 7.2 </span><span class="caption-text">Table showing the difference in coverage for different mappings, when restricting the samples to tissue and primary cell samples.</span><a class="headerlink" href="#coverage-tissue-primary-cell" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Mapping name</p></th>
<th class="head"><p>Number (and percentage) of primary cell and tissue mapped samples</p></th>
<th class="head"><p>Number (and percentage) of primary cell and tissue  unmapped samples</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>By ontology: using Uberon tissues only</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>By ontology: using Uberon tissues and CL cells</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>By name: using tissue column</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Overall mapping (combining by-ontology with CL, and by-name mappings)</p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sex_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics [Sex]&#39;</span>
<span class="n">age_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics [Age]&#39;</span>
<span class="n">remaining_unmapped</span> <span class="o">=</span> <span class="n">samples_info</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">unmapped_overall_pt</span><span class="o">.</span><span class="n">index</span><span class="p">][[</span><span class="n">desc_col</span><span class="p">,</span> <span class="n">sex_col</span><span class="p">,</span> <span class="n">age_col</span><span class="p">,</span> <span class="n">tissue_col</span><span class="p">]]</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-table&quot;</span><span class="p">,</span> <span class="n">remaining_unmapped</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p><a class="reference internal" href="#coverage-tissue-primary-cell"><span class="std std-numref">Table 7.2</span></a> shows that the missing mapping seen in <a class="reference internal" href="#cl-increases-coverage"><span class="std std-numref">Table 7.1</span></a> can be explained by the presence of sample types such as developing tissues and immortal cell lines (models for diseases), i.e. not healthy adult tissues.
Again there was a benefit in combining mappings.
The only remaining unmapped tissues were:</p>
<ol class="simple">
<li><p><em>unclassifiable</em> reference RNA samples (from different providers) - shown in <a class="reference internal" href="#unmapped-samples"><span class="std std-numref">Fig. 7.8</span></a>, from mixed donors and cell types: this is reassuring as we would hope that these would not be mapped to a tissue.</p></li>
<li><p>Two <em>Buffy coat</em> sample of <em>reticulocytes</em>. We could map these by hand to the UBERON “blood” term.</p></li>
</ol>
<div class="figure align-default" id="unmapped-samples" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.8 </span><span class="caption-text">Table showing the remaining samples which could not be automatically mapped to an Uberon tissue using Ontolopy. All three are  reference RNA samples.</span><a class="headerlink" href="#unmapped-samples" title="Permalink to this image">¶</a></p>
</div>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create disagreements table</span>
<span class="n">category_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics [Category]&#39;</span>
<span class="n">categories</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;primary cells&#39;</span><span class="p">,</span> <span class="s1">&#39;tissues&#39;</span><span class="p">]</span>

<span class="n">category_samples</span> <span class="o">=</span> <span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="p">[</span><span class="n">category_col</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">categories</span><span class="p">)]</span><span class="o">.</span><span class="n">index</span>

<span class="n">disagreements_pt</span> <span class="o">=</span> <span class="n">disagreements</span><span class="p">[</span><span class="n">disagreements</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">category_samples</span><span class="p">)]</span>
<span class="n">disagreements_pt</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span><span class="s1">&#39;by name text&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">([</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">disagreements_pt</span><span class="p">[</span><span class="s1">&#39;by_name&#39;</span><span class="p">]],</span> <span class="n">disagreements_pt</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="n">disagreements_pt</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span><span class="s1">&#39;by ont text&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">([</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">disagreements_pt</span><span class="p">[</span><span class="s1">&#39;by_ont&#39;</span><span class="p">]],</span> <span class="n">disagreements_pt</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>

<span class="n">disagreements_table</span> <span class="o">=</span> <span class="n">disagreements_pt</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;by name text&#39;</span><span class="p">,</span><span class="s1">&#39;by ont text&#39;</span><span class="p">])[[</span><span class="s1">&#39;by name text&#39;</span><span class="p">,</span> <span class="s1">&#39;by ont text&#39;</span><span class="p">]]</span>
<span class="n">disagreements_table</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span><span class="s1">&#39;description&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">(</span><span class="n">samples_info</span><span class="p">[</span><span class="n">desc_col</span><span class="p">]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">disagreements_table</span><span class="o">.</span><span class="n">index</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span><span class="n">disagreements_table</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>

<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;number-disagreements&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">disagreements_pt</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;number-types-disagreements&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">disagreements_table</span><span class="p">),</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">disagreements_table</span> <span class="o">=</span> <span class="n">disagreements_table</span><span class="p">[[</span><span class="s1">&#39;description&#39;</span><span class="p">,</span><span class="s1">&#39;by name text&#39;</span><span class="p">,</span> <span class="s1">&#39;by ont text&#39;</span><span class="p">]]</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;disagreements-table&#39;</span><span class="p">,</span> <span class="n">disagreements_table</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="finding-inconsistencies">
<span id="fantom5-inconsistencies-example"></span><h3><span class="section-number">7.5.1.5. </span>Finding inconsistencies<a class="headerlink" href="#finding-inconsistencies" title="Permalink to this headline">¶</a></h3>
<p>By comparing the results of both ontology and text based searches, Ontolopy can find inconsistencies between the two representations which sign post to issues with samples data and how it is presented, or in the ontologies that it is linked to (in this case Uberon and CL): I give an example of each type.
I found this approach very useful, as it allowed me to feed back my discoveries to the maintainers of these ontologies and datasets in order to improve them, and has resulted in improvements to several of these resources.</p>
<p>There were two main ways in which inconsistencies were found:</p>
<ol class="simple">
<li><p>Through looking at samples which are not mapped by one method or another.</p></li>
<li><p>By looking at the disagreements output which compares the mapping that Ontolopy finds using one file and method (text in sample information file), to that it finds using the other (terms in sample ontology file).</p></li>
</ol>
<p>For the FANTOM5 data, disagreements between these mappings revealed problems in the biological ontologies and experiment metadata that were provided to the package in order to create the mappings.
These discrepancies may be a lack of specificity, incompleteness in, or disagreement between FANTOM, CL, or Uberon annotations, either in creating ontologies or annotating tissues to samples.
The process of mapping FANTOM to Uberon tissues found twenty-two such disagreements, of which FANTOM, Uberon, and CL where appropriate have been informed via GitHub issues, some of which have already sparked changes in the ontologies.</p>
<p>Four different types of example are described below, to give an idea of how multiple mappings may be used to improve annotation.</p>
<p>A full list of disagreements can be seen in <a class="reference internal" href="#disagreements-examples"><span class="std std-numref">Fig. 7.9</span></a>.
There were  disagreements/inconsistencies found using Ontolopy.
These disagreements can affect multiple (replicate) samples, for a total of  samples.</p>
<div class="figure align-default" id="disagreements-examples" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.9 </span><span class="caption-text">Table showing all types of disagreements found using Ontolopy, with example samples.</span><a class="headerlink" href="#disagreements-examples" title="Permalink to this image">¶</a></p>
</div>
<div class="section" id="finding-samples-that-are-missing-annotations-to-tissues">
<h4><span class="section-number">7.5.1.5.1. </span>Finding samples that are missing annotations to tissues<a class="headerlink" href="#finding-samples-that-are-missing-annotations-to-tissues" title="Permalink to this headline">¶</a></h4>
<p>When we look at the samples that we would <a class="reference internal" href="#expect-to-map"><span class="std std-ref">expect to map</span></a> by ontology, but that don’t, after filtering for tissues and primary cells only, we see that there are just two types of samples:</p>
<ol class="simple">
<li><p>One sample <code class="docutils literal notranslate"><span class="pre">FF:10379-105H1</span></code> which is missing <code class="docutils literal notranslate"><span class="pre">is_a:</span> <span class="pre">FF:0010164</span> <span class="pre">!</span> <span class="pre">human</span> <span class="pre">caudate</span> <span class="pre">nucleus</span> <span class="pre">-</span> <span class="pre">adult</span> <span class="pre">donor</span> <span class="pre">sample</span></code> in the FANTOM5 ontology file</p></li>
<li><p>21 T-cell samples, all of which appear not to have been fully classified (i.e. contain the following line in the ontology file <code class="docutils literal notranslate"><span class="pre">comment:</span> <span class="pre">Changed</span> <span class="pre">from</span> <span class="pre">previous</span> <span class="pre">label.</span> <span class="pre">TODO:</span> <span class="pre">full</span> <span class="pre">classification</span></code>). I could map all of them to the term for <em>T-cell</em>, whereas someone with more knowledge of T-cells could more accurately map these samples to more specific cell types.</p></li>
</ol>
<p>I can use Ontolopy to add these mappings to the merged ontology to improve the by ontology mapping if I needed to: this would help me to find additional mappings, for example, to <em>immune system</em> as well as <em>blood</em>.</p>
</div>
<div class="section" id="missing-uberon-or-cl-annotation">
<h4><span class="section-number">7.5.1.5.2. </span>Missing Uberon or CL annotation<a class="headerlink" href="#missing-uberon-or-cl-annotation" title="Permalink to this headline">¶</a></h4>
<p><strong>Example: Missing annotation <code class="docutils literal notranslate"><span class="pre">Bronchus</span> <span class="pre">part_of</span> <span class="pre">some</span> <span class="pre">Lung</span></code></strong></p>
<p>One type of problem that can be revealed is a missing link in an ontology.</p>
<p>An example of this that was found using the FANTOM data set was that there was no formal relation in the Uberon ontology between <em>Bronchus</em> and <em>Lung</em>, despite the fact that the description text for <em>Bronchus</em> says “the upper conducting airways of the lung”.</p>
<p>This was found because the sample <code class="docutils literal notranslate"><span class="pre">FF:11511-119G8</span></code> (<em>Bronchial Epithelial Cell, donor1</em>) is mapped by name to <code class="docutils literal notranslate"><span class="pre">UBERON:0002048</span></code> <em>Lung</em>, but by ontology to <code class="docutils literal notranslate"><span class="pre">UBERON:0002185</span></code> <em>Bronchus</em>.
This was flagged as inconsistent because there are no relations in the Uberon ontology between these terms.</p>
<p>Similar missing annotations were discovered between <em>Aorta</em> and <em>Artery</em>, <em>Hair follicle</em> and <em>Dermal papilla</em>, and <em>Skeletal muscle myoblast</em> and <em>Skeletal muscle fiber</em>, and <em>Trophoblast</em> and <em>Placenta</em>.</p>
</div>
<div class="section" id="mislabelled-sample">
<h4><span class="section-number">7.5.1.5.3. </span>Mislabelled sample<a class="headerlink" href="#mislabelled-sample" title="Permalink to this headline">¶</a></h4>
<p>Sometimes samples are simply mislabelled, this can happen in any file type.</p>
<p><strong>Example: <code class="docutils literal notranslate"><span class="pre">FF:11590-120G6</span></code> should be labelled <em>Alveolar Epithelial Cells</em> not <em>Renal Glomerular Endothelial Cells</em></strong></p>
<p>The FANTOM sample ontology file contains two samples named Renal Glomerular Endothelial Cells, donor2: <code class="docutils literal notranslate"><span class="pre">FF:11590-120G6</span></code> and <code class="docutils literal notranslate"><span class="pre">FF:11594-120H1</span></code>.
One of these is a mislabelled sample, and it is actually an Alveolar Epithelial Cell sample.
The mistake is only for the name in the FANTOM ontology file, but not the tissue annotation.</p>
<p><strong>Example: <code class="docutils literal notranslate"><span class="pre">FF:11842-124H6</span></code> should be labelled <em>ovary</em> not <em>bone marrow</em> in the samples information file</strong>
The tissue column of the samples information file lists sample <code class="docutils literal notranslate"><span class="pre">FF:11842-124H6</span></code> as a bone marrow sample, despite being an ovarian cancer sample.</p>
</div>
<div class="section" id="imprecise-annotation-to-tissue">
<h4><span class="section-number">7.5.1.5.4. </span>Imprecise annotation to tissue<a class="headerlink" href="#imprecise-annotation-to-tissue" title="Permalink to this headline">¶</a></h4>
<p><strong>Example: <em>Nucleus pulpopus</em> as <em>Spinal cord</em></strong></p>
<p>Several FANTOM5 tissues are labelled by name colloquially, rather than precisely.
For example, both <em>Nucleus pulpopus</em> and <em>Vertebra</em> are labelled <em>Spinal cord</em> (although the spinal cord itself is considered disjoint from these entities by definition, and in the Uberon ontology).
It’s for this reason that the ontology mapping is preferred over the labelled sample name in creating the overall FANTOM sample-to-tissue mapping.</p>
<p><strong>Example2: <code class="docutils literal notranslate"><span class="pre">FF:11423-118G1</span> <span class="pre">is_a</span></code> <em>dermal melanocyte</em></strong></p>
<p>Sometimes the text in the samples information file can help us to reach better mappings in the sample ontology file.
For example sample <code class="docutils literal notranslate"><span class="pre">FF:11423-118G1</span></code> (and five other similar samples) are mapped to <code class="docutils literal notranslate"><span class="pre">CL:0000148</span></code> (<em>melanocyte</em>), which is a cell that can come from many different parts of the body (skin, heart, eyes, etc), so Ontolopy can only map this term to several tissues (some of which this cell will not have come from) and only if the <a class="reference internal" href="2-functionality.html#child-mapping"><span class="std std-ref">child mapping</span></a> functionality is used.
However, since the sample was labelled as coming from the “skin”, it’s clear that this sample would have been better annotated to <code class="docutils literal notranslate"><span class="pre">CL:0002482</span></code> (<em>dermal melanocyte</em>).</p>
<!--
(tissue-group-mapping)=
### Re-grouping samples based on tissues
[//]: # (TODO: Check this renders)

Samples can be grouped by more general Uberon terms representing groups of tissues.
This {ref}`can be useful<>` when comparing to experiments that use more general tissue meta data.

```{code-cell} ipython3
:tags: [hide-input, remove-output]

# Re-grouping samples by tissue "groups"

# groups = ['brain','cardiovascular system', 'respiratory system','digestive system','skeletal system','skin of body', 'reproductive system','muscle tissue','renal system','central nervous system','connective tissue']
# groups_UBERON = []
# for group in groups:
#     for u_id in obo.ont.keys():
#         if obo.ont[u_id]['name'] == group:
#             groups_UBERON.append(u_id)
# group_name_map = dict(zip(groups_UBERON, groups))

# relations = obo.get_relations(['is_a','part_of'],tissues['UBERON'],groups_UBERON,obo.ont).relations
# groups = []
# for uberon, row in relations.iterrows():
#     try:
#         group = row[0].split('_')[-1]
#     except:
#         group = None
#     groups.append(group)
# relations['Group'] = pd.Series(groups,index=relations.index)
# relations['Group name'] = relations['Group'].map(group_name_map)
# relations['Name'] = relations.index.map(name_map)
# relations=relations.rename(columns={0:'Relation string'})
# relations.to_csv('../data/mappings/tissues_groups.csv')
```
-->
</div>
</div>
<div class="section" id="mapping-overview">
<span id="mappingstatistics"></span><h3><span class="section-number">7.5.1.6. </span>Mapping overview<a class="headerlink" href="#mapping-overview" title="Permalink to this headline">¶</a></h3>
<p>Using Ontolopy we can get a coverage of all samples that we would expect to map to a localised tissue (defining this as primary cell and tissue samples excluding reference RNA).
These mappings correspond to  unique tissues.</p>
</div>
</div>
<div class="section" id="creating-tissue-to-phenotype-mappings">
<span id="tissue-to-phenotype"></span><h2><span class="section-number">7.5.2. </span>Creating tissue-to-phenotype mappings<a class="headerlink" href="#creating-tissue-to-phenotype-mappings" title="Permalink to this headline">¶</a></h2>
<div class="margin sidebar" id="phenotype-by-name">
<p class="sidebar-title">by-ontology mapping</p>
<p>Here we are only using an ontology based mapping, but if we had information in the samples information file about phenotype (e.g. disease), we could also use this to do an additional name based mapping if we wanted to.</p>
</div>
<p>The approach to the creation of the tissue-to-phenotype mappings is different to that we just took for sample-to-tissue mappings in that we are only doing a <a class="reference internal" href="#phenotype-by-name"><span class="std std-ref">by-ontology mapping</span></a>, rather than also mapping by-name and then comparing.
However, it is also a more complex example of a by-ontology mapping since we are asking more than one question to the ontology and adding them together.
For all these questions, we start with the  tissues that we are interested in finding mappings for as source terms, and we use <code class="docutils literal notranslate"><span class="pre">opy.Relations</span></code>’s <code class="docutils literal notranslate"><span class="pre">mode='all'</span></code> option to find <em>all</em> of the Gene Ontology <code class="docutils literal notranslate"><span class="pre">targets=['GO']</span></code> terms that are related to them.</p>
<p>We are interested broadly in tissues where a phenotype can take place, so this could be something on the level of proteins (<em>calcium signalling</em>), cells (<em>cell motility</em>), or tissue (<em>protein secretion</em>).
This will affect what settings (particularly <code class="docutils literal notranslate"><span class="pre">allowed_relations</span></code>) we use when we make calls to <code class="docutils literal notranslate"><span class="pre">Relations</span></code>.</p>
<p>Only Gene Ontology <em>Biological Process</em> terms are related to phenotypes.
The quickest way to retrieve only these is to ask for all <code class="docutils literal notranslate"><span class="pre">GO</span></code> terms and then filter them afterwards.
After loading the <a class="reference external" href="http://geneontology.org/docs/download-ontology/#go_basic">GO basic ontology</a>, we can easily retrieve a list of <em>Biological Process</em> terms.</p>
<div class="cell tag_hide-input tag_remove_output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">go_obo_file</span> <span class="o">=</span> <span class="s1">&#39;../c08-combining/data/go-basic.obo&#39;</span> 
<span class="n">go_obo</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">load_obo</span><span class="p">(</span>
    <span class="n">file_loc</span><span class="o">=</span><span class="n">go_obo_file</span><span class="p">,</span> 
    <span class="n">ont_ids</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span><span class="s1">&#39;CL&#39;</span><span class="p">,</span><span class="s1">&#39;FMA&#39;</span><span class="p">],</span> 
<span class="p">)</span>

<span class="n">biological_processes</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">go_obo</span><span class="o">.</span><span class="n">terms</span> <span class="k">if</span> <span class="p">(</span><span class="s1">&#39;biological_process&#39;</span> <span class="ow">in</span> <span class="n">go_obo</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;namespace&#39;</span><span class="p">])]</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="propagating-relationships-up-the-tree-using-part-of">
<span id="partofprop"></span><h3><span class="section-number">7.5.2.1. </span>Propagating relationships up the tree using <code class="docutils literal notranslate"><span class="pre">part_of</span></code><a class="headerlink" href="#propagating-relationships-up-the-tree-using-part-of" title="Permalink to this headline">¶</a></h3>
<p>Our first example of looking for relations between tissues and phenotypes will include the <code class="docutils literal notranslate"><span class="pre">part_of</span></code> relation.
Since ontologies are often represented by DAGs, relationships are usually generally in one direction.
While there is also the <code class="docutils literal notranslate"><span class="pre">has_part</span></code> relationship that we will look at <a class="reference internal" href="#has-part"><span class="std std-ref">shortly</span></a>, <code class="docutils literal notranslate"><span class="pre">part_of</span></code> is preferred  in the Uberon ontology with almost 10 times as many instances (15,486 compared to 1,703).</p>
<p>We first combine the GO ontology with the uberon ontology, which will simply help us to be able to look up the names of the GO terms to present the output in a more accessible format.
This doesn’t make a difference to the number of mappings, only to the <code class="docutils literal notranslate"><span class="pre">relation_text</span></code> field of the output (which will contain names instead of GO term IDs if available).</p>
<p>We then use the <code class="docutils literal notranslate"><span class="pre">opy.Relations</span></code> class with <code class="docutils literal notranslate"><span class="pre">mode='all'</span></code>, and <code class="docutils literal notranslate"><span class="pre">allowed_relations</span></code> including <code class="docutils literal notranslate"><span class="pre">is_a</span></code>, <code class="docutils literal notranslate"><span class="pre">part_of</span></code> (as mentioned), and some relationships which typically define relationships between tissues and phenotypes <code class="docutils literal notranslate"><span class="pre">is_model_for</span></code>, <code class="docutils literal notranslate"><span class="pre">capable_of</span></code>, <code class="docutils literal notranslate"><span class="pre">capable_of_part_of</span></code>, and the <code class="docutils literal notranslate"><span class="pre">GO</span></code> relation which is defined by Ontology to capture references to <code class="docutils literal notranslate"><span class="pre">GO</span></code> terms within definitions.
This retrieves the mappings in .</p>
<p>The <code class="docutils literal notranslate"><span class="pre">Relations</span></code> class returns a dataframe with the same format whether you use the default <code class="docutils literal notranslate"><span class="pre">mode</span></code> (finding <code class="docutils literal notranslate"><span class="pre">any</span></code> mapping that looks like the <code class="docutils literal notranslate"><span class="pre">targets</span></code>) or the <code class="docutils literal notranslate"><span class="pre">all</span></code> mode; both are indexed by source terms.
For <code class="docutils literal notranslate"><span class="pre">all</span></code> mode, however, there can be multiple mappings for each source term, so the dataframe contains lists of mappings.
This dataframe isn’t too easy on the eyes (or analysis), so Ontolopy also has a helpful method called <code class="docutils literal notranslate"><span class="pre">format_all</span></code> which reformats the <code class="docutils literal notranslate"><span class="pre">Relations</span></code> output dataframe when the <code class="docutils literal notranslate"><span class="pre">all</span></code> mode is used into an easier-to-work-with multi-indexed dataframe.
Example output of this can be seen in <a class="reference internal" href="#phenotype-mapping-example-table"><span class="std std-numref">Fig. 7.10</span></a>.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Merge GO obo:</span>
<span class="n">go_uberon</span> <span class="o">=</span> <span class="n">uberon_obo</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">go_obo</span><span class="p">)</span>

<span class="c1"># Retrieve tissue-phenotype mapping:</span>
<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">source_tissues</span> <span class="o">=</span> <span class="n">overall</span><span class="p">[</span><span class="s1">&#39;overall&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()[</span><span class="mi">1</span><span class="p">:]</span>
<span class="n">relations</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;is_a&#39;</span><span class="p">,</span> <span class="s1">&#39;is_model_for&#39;</span><span class="p">,</span> <span class="s1">&#39;part_of&#39;</span><span class="p">,</span> <span class="s1">&#39;capable_of&#39;</span><span class="p">,</span> <span class="s1">&#39;capable_of_part_of&#39;</span><span class="p">]</span>
<span class="n">phenotype_mapping</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
    <span class="n">allowed_relations</span><span class="o">=</span><span class="n">relations</span><span class="p">,</span> 
    <span class="n">ont</span><span class="o">=</span><span class="n">go_uberon</span><span class="p">,</span> 
    <span class="n">sources</span><span class="o">=</span><span class="n">source_tissues</span><span class="p">,</span> 
    <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">],</span> 
    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;all&#39;</span>
<span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;time-tissue-up&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="p">(</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">,</span> <span class="kc">True</span><span class="p">)</span>

<span class="c1"># Create formatted version (easy to work with):</span>
<span class="n">formatted_phenotype_mapping</span> <span class="o">=</span> <span class="n">phenotype_mapping</span><span class="o">.</span><span class="n">format_all</span><span class="p">(</span><span class="n">ont</span><span class="o">=</span><span class="n">go_uberon</span><span class="p">,</span> <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">])</span>
<span class="n">formatted_phenotype_mapping</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">set_names</span><span class="p">([</span><span class="s1">&#39;Tissue&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">formatted_phenotype_mapping</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping</span><span class="p">[</span><span class="n">formatted_phenotype_mapping</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">biological_processes</span><span class="p">)]</span>

<span class="c1"># Create table view</span>
<span class="n">chosen_rows</span> <span class="o">=</span> <span class="p">[(</span><span class="s1">&#39;UBERON:0001255&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0048731&#39;</span><span class="p">),</span> <span class="c1"># urinary bladder</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0001255&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0007275&#39;</span><span class="p">),</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0001255&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0048856&#39;</span><span class="p">),</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0001255&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0032502&#39;</span><span class="p">),</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0001255&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0008015&#39;</span><span class="p">),</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0000955&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0050890&#39;</span><span class="p">),</span> <span class="c1"># brain</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0000955&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0048856&#39;</span><span class="p">),</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0000955&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0007275&#39;</span><span class="p">),</span>
               <span class="p">(</span><span class="s1">&#39;UBERON:0000955&#39;</span><span class="p">,</span> <span class="s1">&#39;GO:0021551&#39;</span><span class="p">)]</span>
<span class="n">phenotype_mapping_table</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping</span><span class="p">[[</span><span class="s1">&#39;relation_text&#39;</span><span class="p">]]</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">chosen_rows</span><span class="p">]</span>
<span class="n">pd</span><span class="o">.</span><span class="n">set_option</span><span class="p">(</span><span class="s2">&quot;display.max_colwidth&quot;</span><span class="p">,</span> <span class="mi">600</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;phenotype-mapping-table&#39;</span><span class="p">,</span> <span class="n">phenotype_mapping_table</span><span class="p">)</span>
<span class="n">pd</span><span class="o">.</span><span class="n">reset_option</span><span class="p">(</span><span class="s2">&quot;display.max_colwidth&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="phenotype-mapping-example-table" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.10 </span><span class="caption-text">Table showing a view of the output of <code class="docutils literal notranslate"><span class="pre">Relations.format_all</span></code> for the tissue-to-phenotype mapping.</span><a class="headerlink" href="#phenotype-mapping-example-table" title="Permalink to this image">¶</a></p>
</div>
<p>As <a class="reference internal" href="#phenotype-mapping-example-table"><span class="std std-numref">Fig. 7.10</span></a> shows, mappings contain a mixture of mappings to specific GO terms like <em>brain</em> and <em>cognition</em>, and very general phenotype terms, like <em>urinary bladder</em> and <em>anatomical structure development</em>.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Calculate number of mapped GO terms:</span>
<span class="n">go_counts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">formatted_phenotype_mapping</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span>
<span class="n">go_counts</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Frequency&#39;</span><span class="p">]</span>
<span class="n">name</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">go</span> <span class="ow">in</span> <span class="n">go_counts</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">name</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">go_uberon</span><span class="p">[</span><span class="n">go</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">])</span>
    <span class="k">except</span> <span class="ne">KeyError</span><span class="p">:</span>
        <span class="n">name</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="kc">None</span><span class="p">)</span>
<span class="n">go_counts</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">(</span><span class="n">name</span><span class="p">,</span> <span class="n">go_counts</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="n">go_counts_table</span> <span class="o">=</span> <span class="n">go_counts</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">20</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;go-counts-table&quot;</span><span class="p">,</span> <span class="n">go_counts_table</span><span class="p">)</span>

<span class="n">to_remove</span> <span class="o">=</span> <span class="n">go_counts_table</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">9</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">to_list</span><span class="p">()</span> <span class="o">+</span> <span class="p">[</span><span class="s1">&#39;cellular process&#39;</span><span class="p">]</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num-go-to-remove&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">to_remove</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;lst-go-to-remove&quot;</span><span class="p">,</span> <span class="n">list_to_text</span><span class="p">(</span><span class="n">to_remove</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num-go-mapped&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">go_counts</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<p>As we’ve seen in other example use cases, it’s possible to use the <code class="docutils literal notranslate"><span class="pre">exclude</span></code> option when retreiving the mapping, to exclude any terms that you might wish to avoid, for example very general terms if you have a list of these.
Since we didn’t know this, we found the 20 most frequently mapped GO terms (seen in <a class="reference internal" href="#go-counts-top20"><span class="std std-numref">Fig. 7.11</span></a>), out of  mapped overall.
From this list  tissues to remove were then manually identified: .</p>
<div class="figure align-default" id="go-counts-top20" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.11 </span><span class="caption-text">Table showing the frequency that phenotype (<code class="docutils literal notranslate"><span class="pre">GO</span></code>) terms are mapped to the provided tissue terms, for the top 20 GO terms.</span><a class="headerlink" href="#go-counts-top20" title="Permalink to this image">¶</a></p>
</div>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Calculate number of mappings per UBERON term</span>
<span class="k">def</span> <span class="nf">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">df</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Input formatted df</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">level</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Tissue&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">size</span><span class="p">()</span>
    <span class="n">counts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">counts</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">])</span>
    
    <span class="c1"># Add zeros to counts</span>
    <span class="n">unmapped_tissue_phenotype</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">tissue</span> <span class="ow">in</span> <span class="n">source_tissues</span><span class="p">:</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">tissue</span> <span class="ow">in</span> <span class="n">counts</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
            <span class="n">counts</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">tissue</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">]</span>
            <span class="n">unmapped_tissue_phenotype</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">tissue</span><span class="p">)</span>
            
    <span class="k">return</span> <span class="n">counts</span><span class="p">,</span> <span class="n">unmapped_tissue_phenotype</span>

<span class="n">counts</span><span class="p">,</span> <span class="n">unmapped_tissue_phenotype</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">formatted_phenotype_mapping</span><span class="p">)</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-tissue-up&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">unmapped_tissue_phenotype</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;lst-unmapped-tissue-phenotype&quot;</span><span class="p">,</span> <span class="n">list_to_text</span><span class="p">([</span><span class="n">merged</span><span class="p">[</span><span class="n">tissue</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">tissue</span> <span class="ow">in</span> <span class="n">unmapped_tissue_phenotype</span><span class="p">]))</span>

<span class="c1"># remove to_remove and then recalculate:</span>
<span class="n">to_remove_ids</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">go_counts_table</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:</span><span class="mi">9</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">)</span> <span class="o">+</span> <span class="p">[</span><span class="s1">&#39;GO:0009987&#39;</span><span class="p">]</span>
<span class="n">formatted_phenotype_mapping_filtered</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">to_remove_ids</span><span class="p">,</span> <span class="n">level</span><span class="o">=</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span>
<span class="n">counts_less</span><span class="p">,</span> <span class="n">unmapped_tissue_phenotype_less</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_filtered</span><span class="p">)</span>

<span class="n">num_mapped_up</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s2">&quot;Tissue&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-tissue-up&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">num_mapped_up</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="n">num_mapped_up</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">source_tissues</span><span class="p">))</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">)</span>
<span class="c1"># glue(&quot;mapped-tissue-up&quot;, len(formatted_phenotype_mapping_filtered.index.get_level_values(&quot;Tissue&quot;).unique()))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num-unique-phen-up&quot;</span><span class="p">,</span>  <span class="nb">len</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s2">&quot;Phenotype&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num-unmapped-tissue-phenotype-num&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">unmapped_tissue_phenotype_less</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;lst-unmapped-tissue-phenotype-lst&quot;</span><span class="p">,</span> <span class="n">list_to_text</span><span class="p">([</span><span class="n">merged</span><span class="p">[</span><span class="n">tissue</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">tissue</span> <span class="ow">in</span> <span class="n">unmapped_tissue_phenotype_less</span><span class="p">]))</span>
</pre></div>
</div>
</div>
</div>
<p><a class="reference internal" href="#mapped-phenotypes-per-tissue"><span class="std std-numref">Fig. 7.13</span></a> (b) shows us that after removing very general terms, the majority of terms have 1-20 phenotypes mapped to them.</p>
<p>A small number have more, and a small number have no mappings.
There are  terms in (b) which do not have a mapping except for the very general terms.
These terms are: .
Clearly there are phenotypes that affect these tissues (with the exception of the obsolete term), so the lack of mapping here may represent missing relationships or terms within the gene ontology.
An important one for our data set is <em>blood</em> (since we have many such tissue samples): there are GO phenotype terms relating to <em>blood</em> such as <em>blood circulation</em> and <em>blood coagulation</em>, so why don’t we get mappings to these terms?</p>
</div>
<div class="section" id="propagating-down-the-tree-has-part">
<span id="has-part"></span><h3><span class="section-number">7.5.2.2. </span>Propagating “down” the tree: <code class="docutils literal notranslate"><span class="pre">has_part</span></code><a class="headerlink" href="#propagating-down-the-tree-has-part" title="Permalink to this headline">¶</a></h3>
<p>The problem above happens because the annotation to these phenotype terms is not carried out at the level of tissue (<em>blood</em>) but at the level of cell type (<em>blood cell</em>).
In order to retrieve these terms, instead of propagating up the tree (to more general terms) we need to look down the tree (to more specific terms).</p>
<div class="margin sidebar" id="has-part-part-of">
<p class="sidebar-title">has_part and part_of</p>
<p>Ontological relations have strict definitions which help allow us to reason based on these relations, for example by <code class="docutils literal notranslate"><span class="pre">A</span> <span class="pre">has_part</span> <span class="pre">B</span></code>, we mean that A always has B as a part, while <code class="docutils literal notranslate"><span class="pre">B</span> <span class="pre">part_of</span> <span class="pre">A</span></code> means that whenever B exists it is part of A.</p>
<p>This means that <code class="docutils literal notranslate"><span class="pre">has_part</span></code> and <code class="docutils literal notranslate"><span class="pre">part_of</span></code> are not inverses<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id226"><span>176</span></a>]</span>, i.e. if <code class="docutils literal notranslate"><span class="pre">A</span> <span class="pre">part_of</span> <span class="pre">B</span></code>, that does not necessarily mean that <code class="docutils literal notranslate"><span class="pre">B</span> <span class="pre">has_part</span> <span class="pre">A</span></code>.
For example, wherever human ovaries exist, they are <code class="docutils literal notranslate"><span class="pre">part_of</span></code> humans, but whever humans exist, they don’t necessarily <code class="docutils literal notranslate"><span class="pre">has_part</span></code> human ovaries.</p>
<p>There are also other terms which denote similar relations, such as <code class="docutils literal notranslate"><span class="pre">composed_primarily_of</span></code>.</p>
</div>
<p>One way is to use <code class="docutils literal notranslate"><span class="pre">Relations</span></code> including relations in <code class="docutils literal notranslate"><span class="pre">allowed_relations</span></code> that denote having something as a part: <code class="docutils literal notranslate"><span class="pre">has_part</span></code> and <code class="docutils literal notranslate"><span class="pre">composed_primarily_of</span></code>.
From here onwards, I’ll use <code class="docutils literal notranslate"><span class="pre">has_part</span></code> as a shorthand for both of these terms.
It doesn’t make sense to run <code class="docutils literal notranslate"><span class="pre">Relations</span></code> with both <code class="docutils literal notranslate"><span class="pre">has_part</span></code> and <code class="docutils literal notranslate"><span class="pre">part_of</span></code>, since by running both up and down the tree, it could lead to technically true, but uninteresting and potentially misleading mappings.
A simple fictional example of this would be mapping <em>little toe</em> and <em>big toe nail</em> by finding the relation <em>little toe <strong>part of</strong> toes <strong>has part</strong> big toe <strong>has part</strong> big toe nail</em>.</p>
<p id="has-part-thoughts">Including phenotypes that are only relevant for part of the tissue makes sense for tissue samples like <em>blood</em> where we have all parts of the blood in our sample (e.g. the sample will certainly contain <em>blood cell</em>s and <em>plasma</em>).
However, they may make less sense for a tissue sample like <em>heart</em>, where we don’t know if the sample came from the <em>right ventricle</em> or the <em>left ventricle</em> and there may be phenotype terms which are specific to a part of the anatomy we didn’t sample from.
With this in mind, our choices are:</p>
<ol class="simple">
<li><p>don’t include <code class="docutils literal notranslate"><span class="pre">has_part</span></code> relations, and miss phenotype mappings that are made at the level of constituent parts</p></li>
<li><p>include <code class="docutils literal notranslate"><span class="pre">has_part</span></code> relations, but be aware that some samples may map to phenotypes that they are not capable of if the sample-to-tissue mapping is not specific enough.</p></li>
</ol>
<p>In our case, option (2) is preferable, particularly because the FANTOM5 dataset contains many blood samples, and otherwise we would be missing phenotype mappings entirely for these samples.
Running this is otherwise very similar.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Retrieve tissue-phenotype mapping (propagating down using &quot;has_part&quot;):</span>
<span class="c1"># TODO: Add test to Ontolopy checking that the order of checking the new terms doesn&#39;t matter - add issue</span>
<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">relations</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;is_a&#39;</span><span class="p">,</span> <span class="s1">&#39;is_model_for&#39;</span><span class="p">,</span> <span class="s1">&#39;has_part&#39;</span><span class="p">,</span> <span class="s1">&#39;capable_of&#39;</span><span class="p">,</span> <span class="s1">&#39;capable_of_part_of&#39;</span><span class="p">,</span> <span class="s1">&#39;channel_for&#39;</span><span class="p">,</span> <span class="s1">&#39;composed_primarily_of&#39;</span><span class="p">]</span>
<span class="n">phenotype_mapping_down</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
    <span class="n">allowed_relations</span><span class="o">=</span><span class="n">relations</span><span class="p">,</span> 
    <span class="n">ont</span><span class="o">=</span><span class="n">go_uberon</span><span class="p">,</span> 
    <span class="n">sources</span><span class="o">=</span><span class="n">source_tissues</span><span class="p">,</span> 
    <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">],</span> 
    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;all&#39;</span>
<span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;time-tissue-down-1&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="p">(</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">,</span> <span class="kc">True</span><span class="p">)</span>

<span class="c1"># Format and calculate counts of phenotypes per tissue:</span>
<span class="n">formatted_phenotype_mapping_down</span> <span class="o">=</span> <span class="n">phenotype_mapping_down</span><span class="o">.</span><span class="n">format_all</span><span class="p">(</span><span class="n">ont</span><span class="o">=</span><span class="n">go_uberon</span><span class="p">,</span> <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">])</span>
<span class="n">formatted_phenotype_mapping_down</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">set_names</span><span class="p">([</span><span class="s1">&#39;Tissue&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">formatted_phenotype_mapping_down</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping_down</span><span class="p">[</span><span class="n">formatted_phenotype_mapping_down</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">biological_processes</span><span class="p">)]</span>
<span class="n">formatted_phenotype_mapping_down_filtered</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping_down</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">to_remove_ids</span><span class="p">)</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">))),</span> <span class="n">level</span><span class="o">=</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num-unique-phen-down-1&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">num_mapped_down1</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s2">&quot;Tissue&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-tissue-down-1&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">num_mapped_down1</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="n">num_mapped_down1</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">source_tissues</span><span class="p">))</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">)</span>

<span class="n">counts_down1</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down</span><span class="p">)</span>
<span class="n">counts_down1_less</span><span class="p">,</span> <span class="n">unmapped_down1_less</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down_filtered</span><span class="p">)</span>

<span class="c1"># Combine to find as yet unmapped </span>
<span class="n">yet_unmapped_down1</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">unmapped_tissue_phenotype_less</span><span class="p">)</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">unmapped_down1_less</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-tissue-down-1&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">yet_unmapped_down1</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;lst-unmapped-updown-less&quot;</span><span class="p">,</span> <span class="n">list_to_text</span><span class="p">([</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">yet_unmapped_down1</span><span class="p">]))</span>
</pre></div>
</div>
</div>
</div>
<p>In <a class="reference internal" href="#tissue-mapping-comparison"><span class="std std-numref">Table 7.3</span></a>, which compares the number of mapped tissues and phenotypes for different tissue-to-phenotype mapping methods, we can see that this method maps more phenotypes, but for less tissues.</p>
<p>Although less tissues have been mapped overall, we can tell they do capture previously unmapped tissues since the overall number of unmapped tissues (after removal of too-general terms) reduces from  with propagating up only to  which aren’t mapped by either method.
While propagating down therefore improves the overall mapping coverage, looking at the tissues which remain unmapped gives us a clue as to what further improvements we can make.</p>
<p>The terms which remain unmapped are .
The give-away term is <em>skin of body</em>, since looking at subfigure <code class="docutils literal notranslate"><span class="pre">(d)</span></code> in <a class="reference internal" href="#mapped-phenotypes-per-tissue"><span class="std std-numref">Fig. 7.13</span></a>, by hovering over the top most well-mapped tissue we can see that it is <em>stratum basale of epidermis</em>, which is part of the <em>epidermis</em> which is in turn part of the <em>skin of body</em>.
The reason skin of body doesn’t have a mapping is because while the <em>epidermis</em> is <code class="docutils literal notranslate"><span class="pre">part_of</span></code> the <em>skin of body</em>, the <em>skin of body</em> does not have the <code class="docutils literal notranslate"><span class="pre">has_part</span></code> relation to <em>epidermis</em>.</p>
</div>
<div class="section" id="propagating-down-the-tree-inverse-of-part-of">
<span id="can-have-human-part-define"></span><h3><span class="section-number">7.5.2.3. </span>Propagating down the tree: inverse of <code class="docutils literal notranslate"><span class="pre">part_of</span></code><a class="headerlink" href="#propagating-down-the-tree-inverse-of-part-of" title="Permalink to this headline">¶</a></h3>
<p>In addition to the <code class="docutils literal notranslate"><span class="pre">has_part</span></code> approach, we could use the inverse of the <code class="docutils literal notranslate"><span class="pre">part_of</span></code> relations, however the <a class="reference internal" href="#has-part-part-of"><span class="std std-ref">definitions of these terms</span></a> mean that the inverse of <code class="docutils literal notranslate"><span class="pre">part_of</span></code> means something like <em>can have part</em>.
This would mean that in addition to the risk of potentially including mapping to more specific parts of the body that weren’t in our sample (as we <a class="reference internal" href="#has-part-thoughts"><span class="std std-ref">discussed</span></a> for the <code class="docutils literal notranslate"><span class="pre">has_part</span></code> approach), we might sometimes include mappings to tissues that were not even present in the species or gender from which the sample came.
The species problem is the much more pressing concern since Uberon is a multi-species ontology containing many non-human-specific terms and therefore we could end up mapping human samples to terms like <code class="docutils literal notranslate"><span class="pre">GO:0035844</span></code> <em>cloaca development</em>.</p>
<p>One solution to this in Ontolopy is to define relations that look something like <code class="docutils literal notranslate"><span class="pre">A</span> <span class="pre">can_have_human_part</span> <span class="pre">B</span></code> from the information in the ontology files, by using the inverse of <code class="docutils literal notranslate"><span class="pre">part_of</span></code> relations only where there is an external reference (<code class="docutils literal notranslate"><span class="pre">xref</span></code>) to the FMA human anatomy ontology.
We need to do this semi-manually as Ontolopy does not currently contain tools for automatically defining new relations.
Once we’ve created this new relationship, we can ask for relations including it in the same query as <code class="docutils literal notranslate"><span class="pre">has_part</span></code>.
One downside of this approach is that relations found using this kind of self-defined relation will not be able to use the simple reasoning that Ontolopy is capable of (i.e. collapsing relations by using definitions like <code class="docutils literal notranslate"><span class="pre">is_a</span></code><span class="math notranslate nohighlight">\(\cdot\)</span><code class="docutils literal notranslate"><span class="pre">part_of</span></code> == <code class="docutils literal notranslate"><span class="pre">part_of</span></code>, since such equivalences are not defined).</p>
<div class="admonition-sex-specific-phenotype-mappings admonition" id="sex-specific-mappings">
<p class="admonition-title">Sex-specific phenotype mappings</p>
<p>We could also create a relation like <code class="docutils literal notranslate"><span class="pre">A</span> <span class="pre">can_have_part_in_female</span> <span class="pre">B</span></code> (and an analagous term for male) when <code class="docutils literal notranslate"><span class="pre">B</span> <span class="pre">part_of</span> <span class="pre">A</span></code> and <code class="docutils literal notranslate"><span class="pre">B</span> <span class="pre">part_of</span> <span class="pre">UBERON:0003100</span></code> <em>female organism</em>.
We could then cross-reference the sex of our samples from the sample information file to ensure that we don’t create mappings between e.g. male-only samples and ovaries.
This isn’t done here, since it will only affect a very small number of mappings (given that many of the samples are mixed/unknown sexes, that there are only a small number of sexual dimorphic tissues, and that these are generally mapped at the level of specific sexes already), and wouldn’t illustrate a different aspect of using Ontolopy.
Such mappings, if they exist, are simply not included.
If they are excluded, it means that we simply do not map non sex-specific tissues like <em>gonad</em> to either testes or ovary-related phenotypes, so we might be missing such mappings.</p>
</div>
<p>We could also do the same for <a class="reference internal" href="#sex-specific-mappings"><span class="std std-ref">sex-specific tissue-phenotype mappings</span></a>, should we want to.
This could be useful, depending on the experiment data in question and the resulting sample-tissue mappings we’d previously attained.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create new ontology term &#39;can_have_human_part&#39; as inverse of &#39;part_of&#39; with &#39;xref&#39;==&#39;FMA&#39;:</span>
<span class="k">for</span> <span class="n">term</span> <span class="ow">in</span> <span class="n">go_uberon</span><span class="o">.</span><span class="n">terms</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">term</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">:</span>
        <span class="k">try</span><span class="p">:</span>
            <span class="n">parts_of</span> <span class="o">=</span> <span class="n">go_uberon</span><span class="p">[</span><span class="n">term</span><span class="p">][</span><span class="s1">&#39;part_of&#39;</span><span class="p">]</span>
        <span class="k">except</span> <span class="ne">KeyError</span><span class="p">:</span>
            <span class="n">parts_of</span> <span class="o">=</span> <span class="p">[]</span>
            
        <span class="k">try</span><span class="p">:</span>
            <span class="n">xrefs</span> <span class="o">=</span> <span class="n">go_uberon</span><span class="p">[</span><span class="n">term</span><span class="p">][</span><span class="s1">&#39;xref&#39;</span><span class="p">]</span>
        <span class="k">except</span><span class="p">:</span>
            <span class="n">xrefs</span> <span class="o">=</span> <span class="p">[]</span>
        
        <span class="k">if</span> <span class="ow">not</span> <span class="nb">any</span><span class="p">([</span><span class="n">xref</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;FMA&#39;</span><span class="p">)</span> <span class="k">for</span> <span class="n">xref</span> <span class="ow">in</span> <span class="n">xrefs</span><span class="p">]):</span>  <span class="c1"># not human, e.g. cloaca</span>
            <span class="k">continue</span>
        
        <span class="k">for</span> <span class="n">part</span> <span class="ow">in</span> <span class="n">parts_of</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">part</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">:</span>
                <span class="k">try</span><span class="p">:</span>
                    <span class="n">go_uberon</span><span class="p">[</span><span class="n">part</span><span class="p">][</span><span class="s1">&#39;can_have_human_part&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">term</span><span class="p">)</span>
                <span class="k">except</span><span class="p">:</span>
                    <span class="n">go_uberon</span><span class="p">[</span><span class="n">part</span><span class="p">][</span><span class="s1">&#39;can_have_human_part&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="p">{</span><span class="n">term</span><span class="p">}</span>

<span class="c1"># Create mapping:</span>
<span class="n">start</span> <span class="o">=</span> <span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span>
<span class="n">phenotype_mapping_down2</span> <span class="o">=</span> <span class="n">opy</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
    <span class="n">allowed_relations</span><span class="o">=</span><span class="n">relations</span><span class="o">+</span><span class="p">[</span><span class="s1">&#39;can_have_human_part&#39;</span><span class="p">],</span> 
    <span class="n">ont</span><span class="o">=</span><span class="n">go_uberon</span><span class="p">,</span> 
    <span class="n">sources</span><span class="o">=</span><span class="n">source_tissues</span><span class="p">,</span> 
    <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">],</span> 
    <span class="n">mode</span><span class="o">=</span><span class="s1">&#39;all&#39;</span>
<span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;time-tissue-down-2&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="p">(</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">start</span><span class="p">)</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">,</span> <span class="kc">True</span><span class="p">)</span>

<span class="c1"># Format:</span>
<span class="n">formatted_phenotype_mapping_down2</span> <span class="o">=</span> <span class="n">phenotype_mapping_down2</span><span class="o">.</span><span class="n">format_all</span><span class="p">(</span><span class="n">ont</span><span class="o">=</span><span class="n">go_uberon</span><span class="p">,</span> <span class="n">targets</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">])</span>
<span class="n">formatted_phenotype_mapping_down2</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">set_names</span><span class="p">([</span><span class="s1">&#39;Tissue&#39;</span><span class="p">,</span> <span class="s1">&#39;Phenotype&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">formatted_phenotype_mapping_down2</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping_down2</span><span class="p">[</span><span class="n">formatted_phenotype_mapping_down2</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">biological_processes</span><span class="p">)]</span>
<span class="n">formatted_phenotype_mapping_down2_filtered</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping_down2</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">to_remove_ids</span><span class="p">)</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">))),</span> <span class="n">level</span><span class="o">=</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span>
<span class="n">num_mapped_down2</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s2">&quot;Tissue&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-tissue-down-2&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">num_mapped_down2</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="n">num_mapped_down2</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">source_tissues</span><span class="p">))</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num-unique-phen-down-2&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>

<span class="c1"># Calculate counts:</span>
<span class="n">counts_down2</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2</span><span class="p">)</span>
<span class="n">counts_down2_less</span><span class="p">,</span> <span class="n">unmapped_down2_less</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2_filtered</span><span class="p">)</span>

<span class="n">yet_unmapped_down2</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">unmapped_tissue_phenotype_less</span><span class="p">)</span><span class="o">.</span><span class="n">intersection</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">unmapped_down1_less</span><span class="p">),</span> <span class="nb">set</span><span class="p">(</span><span class="n">unmapped_down2_less</span><span class="p">)))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-tissue-down-2&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">yet_unmapped_down2</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;lst-unmapped-updown2-less&quot;</span><span class="p">,</span> <span class="n">list_to_text</span><span class="p">([</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">yet_unmapped_down2</span><span class="p">]))</span>

<span class="k">assert</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down</span><span class="o">.</span><span class="n">index</span><span class="p">)</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2</span><span class="o">.</span><span class="n">index</span><span class="p">))</span><span class="o">==</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># Create examples of new relations table</span>
<span class="n">examples</span> <span class="o">=</span> <span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="mi">9</span><span class="p">,</span> <span class="mi">19</span><span class="p">,</span> <span class="mi">45</span><span class="p">,</span> <span class="mi">59</span><span class="p">,</span> <span class="mi">105</span><span class="p">]</span>
<span class="n">new_samples</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2_filtered</span><span class="o">.</span><span class="n">index</span><span class="p">)</span><span class="o">-</span><span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down_filtered</span><span class="o">.</span><span class="n">index</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;new-down2&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">new_samples</span><span class="p">))</span>
<span class="n">pd</span><span class="o">.</span><span class="n">set_option</span><span class="p">(</span><span class="s2">&quot;display.max_colwidth&quot;</span><span class="p">,</span> <span class="mi">600</span><span class="p">)</span>
<span class="n">table_down2_examples</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping_down2_filtered</span><span class="o">.</span><span class="n">loc</span><span class="p">[[</span><span class="n">new_samples</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">examples</span><span class="p">]][[</span><span class="s1">&#39;relation_text&#39;</span><span class="p">]]</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;table-down2-examples&#39;</span><span class="p">,</span> <span class="n">table_down2_examples</span><span class="p">)</span>
<span class="n">pd</span><span class="o">.</span><span class="n">reset_option</span><span class="p">(</span><span class="s2">&quot;display.max_colwidth&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>This resulting mapping contains  additional tissue-phenotype mappings (not found in either the <code class="docutils literal notranslate"><span class="pre">has_part</span></code> or the <code class="docutils literal notranslate"><span class="pre">part_of</span></code> approach).
Some examples of these additional tissue-phenotype mappings that were found using this <code class="docutils literal notranslate"><span class="pre">can_have_human_part</span></code> approach are given in <a class="reference internal" href="#tbl-down2-examples"><span class="std std-numref">Fig. 7.12</span></a>.
Overall, this mapping covers  tissues and  phenotypes.</p>
<p>The yet unmapped tissues (by any method) are now: .
While this list still contains tissues that we would expect to map to GOBP phenotypes, the lack of these terms in our searches now means that they are simply missing annotations.
For example we have no mapping for <em>adipose tissue</em> despite the fact that GOBP terms exists for <em>adipose tissue development</em> and <em>fat cell proliferation</em>, but there is no cross-ontology mapping of these term in the current version of the ontology, so there is no way Ontolopy could pick them up.</p>
<div class="figure align-default" id="tbl-down2-examples" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.12 </span><span class="caption-text">An example of six of the  additional Uberon-GOBP mappings found using <code class="docutils literal notranslate"><span class="pre">can_have_human_part</span></code>.</span><a class="headerlink" href="#tbl-down2-examples" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="combining-previous-mappings">
<span id="combiningexample2"></span><h3><span class="section-number">7.5.2.4. </span>Combining previous mappings<a class="headerlink" href="#combining-previous-mappings" title="Permalink to this headline">¶</a></h3>
<p>To create the final tissue-phenotype mapping, we combine the propagating up (<code class="docutils literal notranslate"><span class="pre">part_of</span></code>) mapping with the <a class="reference internal" href="#can-have-contains"><span class="std std-ref">larger</span></a> propagating down (<code class="docutils literal notranslate"><span class="pre">can_have_human_part</span></code>) mapping, by simply appending the new lines of the DataFrame.</p>
<div class="admonition-can-have-human-part-query-completely-contains-has-part-query admonition" id="can-have-contains">
<p class="admonition-title">‘can_have_human_part’ query completely contains ‘has_part’ query.</p>
<p>Because the <code class="docutils literal notranslate"><span class="pre">allowed_relations</span></code> for the  mappings of the <code class="docutils literal notranslate"><span class="pre">can_have_human_part</span></code> query completely contain those for the <code class="docutils literal notranslate"><span class="pre">has_part</span></code> query, so do the found relations.
This means that we only need to combine the <code class="docutils literal notranslate"><span class="pre">part_of</span></code> and <code class="docutils literal notranslate"><span class="pre">can_have_human_part</span></code> mappings to get the most complete set.</p>
</div>
<p>As we can see in <a class="reference internal" href="#tissue-mapping-comparison"><span class="std std-numref">Table 7.3</span></a>, the overall combined mapping covers  of the Uberon tissues searched for and maps to  unique GOBP tissues.
Since this is greater than any other individual mapping, we can see that it is necessary to combine different mapping types to get high (&gt;90%) coverage of tissues.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># COMBINING:</span>
<span class="c1"># Add new mappings:</span>
<span class="n">new_mappings</span> <span class="o">=</span>  <span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping</span><span class="o">.</span><span class="n">index</span><span class="p">)</span> <span class="o">-</span> <span class="nb">set</span><span class="p">(</span><span class="n">formatted_phenotype_mapping_down2</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="n">up_down_phenotype_mapping_formatted</span> <span class="o">=</span> <span class="n">formatted_phenotype_mapping_down2</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">new_mappings</span><span class="p">:</span>
    <span class="n">up_down_phenotype_mapping_formatted</span> <span class="o">=</span> <span class="n">up_down_phenotype_mapping_formatted</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">formatted_phenotype_mapping</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>

<span class="c1"># filter:</span>
<span class="n">up_down_phenotype_mapping_filtered</span> <span class="o">=</span> <span class="n">up_down_phenotype_mapping_formatted</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">to_remove_ids</span><span class="p">)</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">up_down_phenotype_mapping_formatted</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">))),</span> <span class="n">level</span><span class="o">=</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span>
                               
<span class="c1"># counts per phenotype:</span>
<span class="n">counts_updown</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">up_down_phenotype_mapping_formatted</span><span class="p">)</span>        
<span class="n">counts_updown_less</span><span class="p">,</span> <span class="n">unmapped_updown_less</span> <span class="o">=</span> <span class="n">mapping_to_tissue_counts</span><span class="p">(</span><span class="n">up_down_phenotype_mapping_filtered</span><span class="p">)</span>

<span class="c1"># calc stats:</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;num-unique-phen-combined&quot;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">up_down_phenotype_mapping_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Phenotype&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">num_mapped_combined</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">up_down_phenotype_mapping_filtered</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">get_level_values</span><span class="p">(</span><span class="s1">&#39;Tissue&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;mapped-tissue-combined&quot;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">num_mapped_combined</span><span class="si">}</span><span class="s2"> (</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="n">num_mapped_combined</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">source_tissues</span><span class="p">))</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%)&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create bar charts:</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">make_subplots</span><span class="p">(</span><span class="n">rows</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span>
                    <span class="n">subplot_titles</span><span class="o">=</span><span class="p">(</span>
                        <span class="s2">&quot;(a) all phenotype terms,&lt;br&gt;propagating up `part_of`&quot;</span><span class="p">,</span> 
                        <span class="s2">&quot;(b) general phenotypes removed,&lt;br&gt;propagating up `part_of`&quot;</span><span class="p">,</span>
                        <span class="s2">&quot;(c) all phenotype terms,&lt;br&gt;propagating down `has_part`&quot;</span><span class="p">,</span>
                        <span class="s2">&quot;(d) general phenotypes removed,&lt;br&gt;propagating down `has_part`&quot;</span><span class="p">,</span>
                        <span class="s2">&quot;(e) all phenotype terms,&lt;br&gt;propagating down&lt;br&gt;`can_have_human_part`&quot;</span><span class="p">,</span>
                        <span class="s2">&quot;(f) general phenotypes removed,&lt;br&gt;propagating down&lt;br&gt;`can_have_human_part`&quot;</span><span class="p">,</span>
                        <span class="s2">&quot;(g) all phenotype terms,&lt;br&gt;propagating both up and down&quot;</span><span class="p">,</span>
                        <span class="s2">&quot;(h) general phenotypes removed,&lt;br&gt;propagating both up and down&quot;</span>
                    <span class="p">),</span>
                    <span class="n">shared_yaxes</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                   <span class="n">horizontal_spacing</span><span class="o">=</span><span class="mf">0.05</span><span class="p">,</span>
                   <span class="p">)</span>

<span class="c1"># ROW 1: (a) all phenotype terms propagating up</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span> 
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;all, up&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;slateblue&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">1</span>
<span class="p">)</span>
<span class="c1"># ROW 1 (b) general phenotypes removed propagating up</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span>
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;filtered, up&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;darkslateblue&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">2</span>
<span class="p">)</span>
<span class="c1"># ROW 2: (c) all phenotype terms propagating down: has_part</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts_down1</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts_down1</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span>
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;all, down: has_part&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;lightsalmon&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">1</span>
<span class="p">)</span>
<span class="c1"># ROW 2: (d) general phenotypes removed propagating down</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts_down1_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts_down1_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span>
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;filtered, down: has_part&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;tomato&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">2</span>
<span class="p">)</span>    

<span class="c1"># ROW 3: (e) all phenotype terms, down can_have_human_part</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts_down2</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts_down2</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span>
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;all, down: can_have_human_part&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;orchid&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">1</span>
<span class="p">)</span>    
<span class="c1"># ROW 3: (f) general phenotypes removed, down can_have_human_part</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts_down2_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts_down2_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span>
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;filtered, down: can_have_human_part&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;darkorchid&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">2</span>
<span class="p">)</span>    

<span class="c1"># ROW 4: (g) all phenotype terms, propagating both up and down</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts_updown</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts_updown</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span>
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;all, both up and down&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;palevioletred&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">1</span>
<span class="p">)</span>    
<span class="c1"># ROW 4: (h) general phenotypes removed, propagating both up and down</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go_</span><span class="o">.</span><span class="n">Bar</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">counts_updown_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">],</span> 
           <span class="n">x</span><span class="o">=</span><span class="p">[</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">merged</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">x</span><span class="si">}</span><span class="s2">&quot;</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">counts_updown_less</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">)[</span><span class="s1">&#39;Number mapped phenotypes&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">],</span>
           <span class="n">name</span><span class="o">=</span><span class="s1">&#39;filtered, both up and down&#39;</span><span class="p">,</span>
           <span class="n">marker_color</span> <span class="o">=</span> <span class="s1">&#39;mediumvioletred&#39;</span><span class="p">,</span>
           <span class="p">),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">2</span>
<span class="p">)</span>    

<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">showlegend</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="mi">800</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="mi">600</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_yaxes</span><span class="p">(</span><span class="nb">range</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">150</span><span class="p">],</span> <span class="n">tickvals</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">151</span><span class="p">,</span><span class="mi">25</span><span class="p">)))</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_yaxes</span><span class="p">(</span><span class="n">title_text</span><span class="o">=</span><span class="s1">&#39;Number&lt;br&gt;phenotypes&lt;br&gt;mapped to&#39;</span><span class="p">,</span> <span class="n">col</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_annotations</span><span class="p">(</span><span class="n">font_size</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>  <span class="c1"># subplot titles are annotations in plotly</span>

<span class="n">fig</span><span class="o">.</span><span class="n">update_xaxes</span><span class="p">(</span><span class="n">showticklabels</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">update_xaxes</span><span class="p">(</span><span class="n">title_text</span><span class="o">=</span><span class="s1">&#39;Tissue term&#39;</span><span class="p">,</span> <span class="n">row</span><span class="o">=</span><span class="mi">4</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">show</span><span class="p">(</span><span class="s1">&#39;notebook&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="mapped-phenotypes-per-tissue">
<img alt="../_images/blank.png" src="../_images/blank.png" />
<p class="caption"><span class="caption-number">Fig. 7.13 </span><span class="caption-text">Bar charts showing the number of tissue-phenotype mappings (number of phenotypes mapped to each tissue - roll mouse over to see tissue names) for (a) all phenotype terms propagating down, (b) general phenotypes removed propagating down, (c) all phenotype terms propagating up, (d) general phenotypes removed propagating down, (e) all phenotype terms propagating both up and down, and (f) general phenotypes removed propagating both up and down.</span><a class="headerlink" href="#mapped-phenotypes-per-tissue" title="Permalink to this image">¶</a></p>
</div>
<!-- ../images/blank.png This is a workaround to put a 1x1px blank image after an interactive image so that it appears to have a figure label -->
<p>There are  unmapped tissues (which map to zero phenotypes), and all other tissues map to between 2 and 131 phenotype terms, (as we can see in figure <a class="reference internal" href="#mapped-phenotypes-per-tissue"><span class="std std-numref">Fig. 7.13</span></a>).
The number of mappings varies smoothly in this range with more general tissues and organs broadly appearing to have higher numbers of mappings than very specific tissues.
We can also see in <a class="reference internal" href="#mapped-phenotypes-per-tissue"><span class="std std-numref">Fig. 7.13</span></a> that the <code class="docutils literal notranslate"><span class="pre">can_have_human_part</span></code> mapping makes up the majority of the mappings in the final combined mapping.</p>
<table class="table" id="tissue-mapping-comparison">
<caption><span class="caption-number">Table 7.3 </span><span class="caption-text">Table comparing the difference in coverage (mappable tissues), phenotypes, time taken, and unmapped tissues for different mapping techniques.</span><a class="headerlink" href="#tissue-mapping-comparison" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 20%" />
<col style="width: 20%" />
<col style="width: 20%" />
<col style="width: 20%" />
<col style="width: 20%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Mapping name</p></th>
<th class="head"><p>Tissue coverage: number (percent) tissues mapped (this mapping only)</p></th>
<th class="head"><p>Number of unique phenotype mapped to (by this mapping only)</p></th>
<th class="head"><p>Number tissues remaining unmapped (by this or any previous mapping)</p></th>
<th class="head"><p>Time to retrieve mapping</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Propagating up</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Propagating down using <code class="docutils literal notranslate"><span class="pre">has_part</span></code></p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-even"><td><p>Propagating down using <code class="docutils literal notranslate"><span class="pre">can_have_human_part</span></code></p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
<td><p></p></td>
</tr>
<tr class="row-odd"><td><p>Combined (all of the above)</p></td>
<td><p></p></td>
<td><p></p></td>
<td><p><strong>N/A</strong></p></td>
<td><p><strong>N/A</strong></p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="creating-sample-to-tissue-phenotype-mappings">
<span id="sample-tissue"></span><h2><span class="section-number">7.5.3. </span>Creating sample-to-tissue-phenotype mappings<a class="headerlink" href="#creating-sample-to-tissue-phenotype-mappings" title="Permalink to this headline">¶</a></h2>
<p>Once we have both the sample-to-tissue and tissue-to-phenotype mappings, we can combine them to get the sample-to-tissue-phenotype mappings: mappings between samples and phenotypes that occur in the tissue type of that sample.
There isn’t a built-in Ontolopy function to do this, but since Ontolopy objects are built on top of Pandas DataFrames, they are fairly easy to work with.</p>
<p>Since we chose to map primary cell and tissue samples only, there are many samples which are not mapped.
Null mappings are included in the output, and where mapping by name is used, it is recorded as a <code class="docutils literal notranslate"><span class="pre">mapped_by_name_to</span></code> relationship in the relation path, e.g. <code class="docutils literal notranslate"><span class="pre">FF:11453-119A4.mapped_by_name_to~UBERON:0002048.is_a~UBERON:0000171.capable_of~GO:0007585</span></code> or in text <em><strong>Bronchial Epithelial Cell, donor4</strong> mapped by name to <strong>lung</strong> is a <strong>respiration organ</strong> capable of <strong>respiratory gaseous exchange by respiratory system</strong></em>.
Since relation strings can now contain FANTOM5, CL, UBERON and GO terms, I first merge the GO ontology into the merged FANTOM5 and Uberon ontologies, so that the names of all terms can be found for the relation text.</p>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mapped_by_name_relation</span> <span class="o">=</span> <span class="s1">&#39;mapped_by_name_to&#39;</span>

<span class="c1"># merge ontology</span>
<span class="n">merged</span> <span class="o">=</span> <span class="n">merged</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">go_obo</span><span class="p">)</span>

<span class="n">sample_to_tissue</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">sample</span><span class="p">,</span> <span class="n">row_i</span> <span class="ow">in</span> <span class="n">overall</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">row_i</span><span class="p">[</span><span class="s1">&#39;overall&#39;</span><span class="p">]):</span>
        <span class="n">tissue</span> <span class="o">=</span> <span class="n">row_i</span><span class="p">[</span><span class="s1">&#39;overall&#39;</span><span class="p">]</span>
    <span class="k">else</span><span class="p">:</span> <span class="c1"># no mapping from sample to tissue</span>
        <span class="n">tissue</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>
        <span class="n">phenotype</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span>
        <span class="n">sample_to_tissue</span><span class="p">[(</span><span class="n">sample</span><span class="p">,</span> <span class="n">tissue</span><span class="p">,</span> <span class="n">phenotype</span><span class="p">)]</span> <span class="o">=</span> <span class="p">[</span><span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">]</span>
        <span class="k">continue</span>
        
    <span class="k">try</span><span class="p">:</span>
        <span class="n">mappings</span> <span class="o">=</span> <span class="n">up_down_phenotype_mapping_filtered</span><span class="o">.</span><span class="n">xs</span><span class="p">(</span><span class="n">tissue</span><span class="p">,</span> <span class="n">level</span> <span class="o">=</span><span class="s1">&#39;Tissue&#39;</span><span class="p">)</span>
        <span class="k">for</span> <span class="n">phenotype</span><span class="p">,</span> <span class="n">row_j</span> <span class="ow">in</span> <span class="n">mappings</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">row_i</span><span class="p">[</span><span class="s1">&#39;mapped_by&#39;</span><span class="p">]</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;ontology&#39;</span><span class="p">,</span> <span class="s1">&#39;both (same)&#39;</span><span class="p">]:</span>
                <span class="n">relation_path</span> <span class="o">=</span> <span class="n">sample_to_tissue_mapping</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">sample</span><span class="p">,</span> <span class="s1">&#39;relation_path&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="n">tissue</span><span class="p">,</span> <span class="n">row_j</span><span class="p">[</span><span class="s1">&#39;relation_path&#39;</span><span class="p">])</span>          
            <span class="k">elif</span> <span class="n">row_i</span><span class="p">[</span><span class="s1">&#39;mapped_by&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;name&#39;</span><span class="p">:</span>
                <span class="n">relation_path</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">sample</span><span class="si">}</span><span class="s2">.</span><span class="si">{</span><span class="n">mapped_by_name_relation</span><span class="si">}</span><span class="s2">~</span><span class="si">{</span><span class="n">tissue</span><span class="si">}</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="n">tissue</span><span class="p">,</span> <span class="n">row_j</span><span class="p">[</span><span class="s1">&#39;relation_path&#39;</span><span class="p">])</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="n">logging</span><span class="o">.</span><span class="n">warning</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">row_i</span><span class="p">[</span><span class="s1">&#39;mapped_by&#39;</span><span class="p">]</span><span class="si">}</span><span class="s2"> has unexpected format&quot;</span><span class="p">)</span>
                <span class="k">continue</span>
            <span class="n">sample_to_tissue</span><span class="p">[(</span><span class="n">sample</span><span class="p">,</span> <span class="n">tissue</span><span class="p">,</span> <span class="n">phenotype</span><span class="p">)]</span> <span class="o">=</span> <span class="p">[</span><span class="n">relation_path</span><span class="p">,</span> 
                                                             <span class="n">opy</span><span class="o">.</span><span class="n">relation_path_to_text</span><span class="p">(</span><span class="n">relation_path</span><span class="p">,</span> <span class="n">merged</span><span class="p">)]</span>
    <span class="k">except</span> <span class="ne">KeyError</span><span class="p">:</span>
        <span class="c1"># Save phenotype as NaN if no phenotype mapping:</span>
        <span class="n">sample_to_tissue</span><span class="p">[(</span><span class="n">sample</span><span class="p">,</span> <span class="n">tissue</span><span class="p">,</span> <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">)]</span> <span class="o">=</span> <span class="p">[</span><span class="n">sample_to_tissue_mapping</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">sample</span><span class="p">,</span> <span class="s1">&#39;relation_path&#39;</span><span class="p">],</span> 
                                                      <span class="n">opy</span><span class="o">.</span><span class="n">relation_path_to_text</span><span class="p">(</span><span class="n">sample_to_tissue_mapping</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">sample</span><span class="p">,</span> <span class="s1">&#39;relation_path&#39;</span><span class="p">],</span> <span class="n">merged</span><span class="p">)]</span>
        

<span class="c1"># Save out overall mapping:</span>
<span class="n">sample_to_tissue_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_dict</span><span class="p">(</span><span class="n">sample_to_tissue</span><span class="p">,</span>
                                      <span class="n">orient</span><span class="o">=</span><span class="s1">&#39;index&#39;</span><span class="p">,</span>
                                      <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;relation_path&#39;</span><span class="p">,</span> <span class="s1">&#39;relation_text&#39;</span><span class="p">])</span>
<span class="n">sample_to_tissue_df</span><span class="o">.</span><span class="n">index</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">MultiIndex</span><span class="o">.</span><span class="n">from_tuples</span><span class="p">(</span><span class="n">sample_to_tissue_df</span><span class="o">.</span><span class="n">index</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;sample&quot;</span><span class="p">,</span> <span class="s2">&quot;tissue&quot;</span><span class="p">,</span> <span class="s2">&quot;phenotype&quot;</span><span class="p">])</span>
<span class="n">mapping_file_path</span> <span class="o">=</span> <span class="s1">&#39;../c06-filter/data/created/fantom-go-mapping.csv&#39;</span>
<span class="n">sample_to_tissue_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">mapping_file_path</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">sample_to_tissue_df</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">20</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="final-mapping">
<span id="opyexamplefinalmapping"></span><h3><span class="section-number">7.5.3.1. </span>Final mapping<a class="headerlink" href="#final-mapping" title="Permalink to this headline">¶</a></h3>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Glue final mapping statistics:</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;sample-phen-rows&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">sample_to_tissue_df</span><span class="p">))</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;total-sample-phen&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">sample_to_tissue_df</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">dropna</span><span class="p">()))</span>

<span class="n">df</span> <span class="o">=</span> <span class="n">sample_to_tissue_df</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>

<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;sample-phen-coverage-cat&#39;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="o">~</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()</span> <span class="o">&amp;</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;sample&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">category_samples</span><span class="p">)][</span><span class="s1">&#39;sample&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;sample&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">category_samples</span><span class="p">)][</span><span class="s1">&#39;sample&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%&quot;</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;sample-phen-coverage-all&#39;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="mi">100</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="o">~</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()][</span><span class="s1">&#39;sample&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;sample&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span><span class="si">:</span><span class="s2">.2f</span><span class="si">}</span><span class="s2">%&quot;</span><span class="p">)</span>

<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;sample-tissue-nan&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="o">~</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;tissue&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()</span> <span class="o">&amp;</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]))</span>
<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;total-sample-nan&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;tissue&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]))</span>

<span class="n">glue</span><span class="p">(</span><span class="s1">&#39;notebook-time-mapping-example&#39;</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">time</span><span class="o">.</span><span class="n">time</span><span class="p">()</span><span class="o">-</span><span class="n">notebook_start</span><span class="si">:</span><span class="s2">.0f</span><span class="si">}</span><span class="s2"> seconds&quot;</span><span class="p">)</span>

<span class="c1"># Why are samples unmapped? </span>
<span class="c1"># Count number of rows per unique tissue without phenotype mapping.</span>
<span class="n">rows</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">uberon</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()][</span><span class="s1">&#39;tissue&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="k">if</span> <span class="nb">type</span><span class="p">(</span><span class="n">uberon</span><span class="p">)</span> <span class="o">==</span> <span class="nb">float</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">uberon</span><span class="p">):</span>
            <span class="n">num</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()</span> <span class="o">&amp;</span> <span class="n">df</span><span class="p">[</span><span class="s1">&#39;tissue&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()])</span>
            <span class="n">row</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;NaN&#39;</span><span class="p">,</span> <span class="s1">&#39;Unmapped to tissue&#39;</span><span class="p">,</span> <span class="n">num</span><span class="p">]</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">num</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;phenotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;tissue&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">uberon</span><span class="p">)])</span>
        <span class="n">row</span> <span class="o">=</span> <span class="p">[</span><span class="n">uberon</span><span class="p">,</span> <span class="n">merged</span><span class="p">[</span><span class="n">uberon</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">],</span> <span class="n">num</span><span class="p">]</span>
    <span class="n">rows</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">row</span><span class="p">)</span>

<span class="n">unmapped_why</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">rows</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Uberon ID&#39;</span><span class="p">,</span> <span class="s1">&#39;Uberon Name&#39;</span><span class="p">,</span> <span class="s1">&#39;Number samples mapped to tissue&#39;</span><span class="p">]);</span>
<span class="n">unmapped_why</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Uberon ID&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unmapped-why&quot;</span><span class="p">,</span> <span class="n">unmapped_why</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>There are  rows of the sample-to-tissue mapping DataFrame in total.
This includes some <code class="docutils literal notranslate"><span class="pre">NaN</span></code> values, so it contains  mappings from sample to phenotype; equivalent to a sample coverage of  of filtered (<em>tissue</em> and <em>primary cell</em>) samples or  of all samples.
It also includes an additional  mappings from sample to tissue (but not to phenotype), and  samples with no mapping to tissue or phenotype.
<a class="reference internal" href="#table-unmapped-why"><span class="std std-numref">Fig. 7.14</span></a> shows why almost 10% of samples are unmapped in more detail: many samples map to the same unmapped tissues, particularly adipose tissue, epithelium, or connective tissue.</p>
<div class="figure align-default" id="table-unmapped-why" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 7.14 </span><span class="caption-text">Table showing how many samples are mapped to each unmappable tissue, showing why the coverage of samples isn’t higher.</span><a class="headerlink" href="#table-unmapped-why" title="Permalink to this image">¶</a></p>
</div>
<p>With Ontolopy, this complex task is relatively quick: it took  to run this whole notebook on a laptop without any parallelisation.
Also recall that this section is merely an example of an application of Ontolopy: this same process could be done for other datasets that provide an ontology and/or a sample information file.</p>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c07-ontolopy"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="4-misc-examples.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title"><span class="section-number">7.4. </span>Example uses: mapping samples to diseases or phenotypes</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="6-discussion.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title"><span class="section-number">7.6. </span>Discussion</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>