# Summary
[//]: # (TODO: Rewrite this section so that it is just a summary of the different sources of error AND a summary of the different excitingt types of data. BUT NOT downbeat/dramatic e.g. no wonder it doesn't work.)
[//]: # (TODO: Citations in summary section)
[//]: # (TODO: Summarise approaches to knowledge-finding: computational prediction, comparative studies - genomics, GWAS, DE exp - baseline/descriptive and linking datasets - big collaborative efforts)
[//]: # (TODO: Summarise challenges, including sources of bias, and difficulty, and error)
[//]: # (TODO: Bring it back to the big questions and the opportunities)
[//]: # (TODO: Mention different model organisms that we can gain insight from)

[//]: # (TODO: Cite data inaccessibility)
While in other fields, data inaccessibility is a major barrier to reproducible research, this is the field that had an online database system that remote computers could access in the 1960s!
So, the catalogued information populates freely available databases, vocabularies, and annotations, creating controlled and shared vocabularies that fuel computational methodologies.

Multi omics approaches that combine data types have been successful at elucidating mechanisms behind certain phenotypes{cite}`Hasin2017-tk,Ritchie2015-ds,Kristensen2014-iw`. 
There is now the opportunity for genome-wide phenotype predictions to do the same. 
Obtaining an accurate prediction of phenotype and protein function, even for a subclass of variants/proteins, has the potential to greatly impact people. 
A method which predicts phenotype directly from the molecular biology would be desirable as it would create a model for how phenotype emerges from genotype, in addition to predictions for important genes.
