
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2.1. Big questions: What is genetically determined, and how? &#8212; Phenotype from Genotype</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script >const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology." href="2-biological-molecules.html" />
    <link rel="prev" title="2. How phenotype arises from genotype" href="0-index.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#history-of-inheritable-traits">
   2.1.1. History of inheritable traits
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-future-computational-biologists-want">
   2.1.2. The future computational biologists want
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Big questions: What is genetically determined, and how?</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#history-of-inheritable-traits">
   2.1.1. History of inheritable traits
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-future-computational-biologists-want">
   2.1.2. The future computational biologists want
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="section" id="big-questions-what-is-genetically-determined-and-how">
<span id="big-questions"></span><h1><span class="section-number">2.1. </span>Big questions: What is genetically determined, and how?<a class="headerlink" href="#big-questions-what-is-genetically-determined-and-how" title="Permalink to this headline">¶</a></h1>
<p>As humans, we are curious and want to understand ourselves.
We want to know the answers to questions like: “Why are people the way we are?”, “Which aspects of ourselves have we inherited?”, and “What is fixed and what can be changed by the way we live our lives?”
Looking at what we are made from - more specifically the DNA that can be found in each of our cells - has promised answers to some of these questions.</p>
<p>We ask so many questions, not only out of curiosity, but also in order to improve and control our lives and environment.
This drive for control hasn’t always been a good thing: in recent history, genetics has been used to justify extremely harmful and unethical racist eugenics policies.
While eugenics is thankfully no longer in vogue, there is no guarantee that scientific knowledge will be used ethically.
Modern-day genetics still raises concerns about which of our traits should be medicalised or pathologised: should we be looking for cures for autism if autistic people don’t want them?</p>
<p>However, knowing more about ourselves clearly also has the capacity to be used for the good of all.
By understanding how our bodies work, researchers seek to develop new, more effective, and kinder treatments for diseases.
Alongside curiosity, these were my aims in seeking to explore the molecular link between genotype and phenotype.</p>
<div class="section" id="history-of-inheritable-traits">
<span id="history-interitable"></span><h2><span class="section-number">2.1.1. </span>History of inheritable traits<a class="headerlink" href="#history-of-inheritable-traits" title="Permalink to this headline">¶</a></h2>
<div class="margin sidebar" id="darwin-racism">
<p class="sidebar-title">Charles Darwin and racism</p>
<p>Darwin used his theory of natural selection to argue that women and non-white races were inferior to white men<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id201"><span>10</span></a>]</span>. The full title of <em>On the Origin of Species</em> was <em>On the Origin of Species by Means of Natural Selection, or the Preservation of Favoured Races in the Struggle for Life</em>.</p>
</div>
<p>Humanity has been trying to answer the big questions long before we discovered DNA.
The ancient theory of soft inheritance<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id7"><span>11</span></a>]</span> said that people can pass on traits they gained during their lives, while 16th-century alchemists theorised that sperm contains tiny fully formed humans<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id8"><span>12</span></a>]</span> (i.e. women didn’t pass down anything).</p>
<p id="darwin">In 1859, <a class="reference internal" href="#darwin-racism"><span class="std std-ref">Charles Darwin</span></a> published his book <em>On the Origin of Species</em><span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id202"><span>13</span></a>]</span>, explaining his theory of natural selection: organisms compete for resources and not all can survive to reproduce, some organisms will have traits that increase their chances of this, and those that do pass on their traits to the next generation.
The theory predicts and explains the gradual change of heritable characteristics over time: evolution.
Darwin presented <a class="reference internal" href="#homology"><span class="std std-ref">homologous</span></a> anatomical structures, like the similarity between a bat’s wing and a human hand, as evidence for evolution and shared ancestry.</p>
<div class="margin sidebar" id="homology">
<p class="sidebar-title">Homologous, orthologous, and analagous features.</p>
<p><em>Homology</em> is similarity due to a shared evolutionary ancestry.
This similarity can be between anatomy, or biological molecules and sequences (like DNA and proteins).</p>
<p>Homologous sequences are <em>orthologous</em> if they exist in different species, for example mice and humans have some orthologous genes, which are different versions of the same genes that perform the same function in the organism. This is in contrast to homology within species (paralogy) which occurs due to gene duplication, for example, humans have different versions of the histone gene.</p>
<p>Similar features and sequences which do not have shared evolutionary ancestry (i.e. which evolved independently), are <em>analagous</em>.</p>
</div>
<div class="figure align-default" id="mendel-pea">
<a class="reference internal image-reference" href="../_images/mendel.png"><img alt="../_images/mendel.png" src="../_images/mendel.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 2.1 </span><span class="caption-text">An example of Mendel’s experimental results, for white and purple pea flowers. He began with “pure line” pea plants (which always produced self-identical plants when self-pollinated). Crossing the “pure line” plants resulted in first generation offspring which always had purple flowers.
When self-pollinated, these first generation plants created plants with purple and white flowers in a 3:1 ratio.</span><a class="headerlink" href="#mendel-pea" title="Permalink to this image">¶</a></p>
</div>
<p id="mendel">Inspired by Darwin<span id="id5">[<a class="reference internal" href="../cz-end-matter/reference.html#id200"><span>14</span></a>]</span>, his contemporary Gregor Mendel’s famous pea experiments<span id="id6">[<a class="reference internal" href="../cz-end-matter/reference.html#id147"><span>15</span></a>]</span> provided the earliest scientific basis for genetics through his experiments with independently inherited traits of peas (e.g. purple or white flowers, tall or shot plants, wrinkled or round seeds…).
Importantly, he discovered rules of inheritance that indicated that offspring have combinations of discrete genetic material (rather than a blend), i.e. we don’t see pink flowers when we cross purple and white flowered peas.
He also showed that single traits (e.g. purple flowers) can actually be caused by different underlying genetics (see <a class="reference internal" href="#mendel-pea"><span class="std std-numref">Fig. 2.1</span></a>).</p>
<p>This concept was expanded by Wilhelm Johannsen, who coined the term “gene” as the name for the hidden material that caused the traits.
Johnannsen’s work also distinguished between “genotype” and “phenotype”: <em>genotype</em> being the hidden (genetic) material that organisms have, and <em>phenotype</em> being the measurable trait<span id="id7">[<a class="reference internal" href="../cz-end-matter/reference.html#id146"><span>16</span></a>]</span>.
His research showed that some phenotypes (e.g. seed size) could vary considerably even with genetically identical plants, due to their environment.</p>
<div class="margin sidebar" id="fischer-racism">
<p class="sidebar-title">Ronald Fischer, racism and eugenics</p>
<p>Fischer has a legacy of scientific racism. For example, actively campaigning for sterilisation of a tenth of the population in the name of eugenics <span id="id8">[<a class="reference internal" href="../cz-end-matter/reference.html#id151"><span>17</span></a>]</span>.</p>
</div>
<p>Phenotypes are not only strictly Mendelian, with a fixed number of “types”, or continuous and without a genetic basis.
<a class="reference internal" href="../c03-compbio-background/5-bias.html#fischer-racism"><span class="std std-ref">Ronald Fischer</span></a> showed that variation in continuous traits (such as height in humans) can be consistent with Mendelian inheritance if multiple genes contributed additively to the trait.
Many traits are <em>complex</em> in this way, meaning that they are influenced by many different genetic factors, as well as the environment.</p>
<p>When <a class="reference internal" href="#watson-racism"><span class="std std-ref">Watson</span></a> and Crick discovered the now familiar structure of DNA in 1953<span id="id9">[<a class="reference internal" href="../cz-end-matter/reference.html#id148"><span>18</span></a>]</span>, we took a huge step towards being able to answer our big questions.
We finally understood the molecular structure that underlies inherent traits.
Then in 2003, with the completion of the Human Genome Project<span id="id10">[<a class="reference internal" href="../cz-end-matter/reference.html#id16"><span>19</span></a>]</span>, it was possible to read the human version of this “code of life”.
Once researchers had access to the whole genetic code for a person, they could set about trying to decode it, with the world hoping that this landmark would aid the search for treatments for diseases like cancers and Alzheimer’s<span id="id11">[<a class="reference internal" href="../cz-end-matter/reference.html#id149"><span>20</span></a>]</span>.</p>
<div class="margin sidebar" id="watson-racism">
<p class="sidebar-title">James Watson and racism</p>
<p>Watson’s has publicly asserted that he believes differences in average measured IQ between blacks and whites are due to genetic differences<span id="id12">[<a class="reference internal" href="../cz-end-matter/reference.html#id150"><span>21</span></a>]</span>.</p>
</div>
<p>We now have thousands of human genomes to investigate, and the promises of gene therapies and personalised medicine are now beginning to become reality.
At the time of writing, there are eleven cell and gene therapies approved by the European Medicines Agency<span id="id13">[<a class="reference internal" href="../cz-end-matter/reference.html#id145"><span>22</span></a>]</span>, which treat a variety of cancers, as well as Crohn’s disease, and eye and cartilage problems.
In addition, the first personalised genomic medicine chemotherapy treatment is now available on the NHS, for cancer patients with the allele of the <a class="reference external" href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=DPYD">DYPD</a> gene that cause slower breakdown of chemotherapy toxins<span id="id14">[<a class="reference internal" href="../cz-end-matter/reference.html#id144"><span>23</span></a>]</span>.</p>
</div>
<div class="section" id="the-future-computational-biologists-want">
<span id="the-future-comp-bio"></span><h2><span class="section-number">2.1.2. </span>The future computational biologists want<a class="headerlink" href="#the-future-computational-biologists-want" title="Permalink to this headline">¶</a></h2>
<p>The eventual destination of this field is a full understanding of how our individual genomes and their interaction with the environment affects us.
With this understanding, we would anticipate a much wider application of both personalised medicine and gene therapies.
These therapies are not yet a common occurrence: the eleven approved cell and gene therapies come from a pool of such 500 clinical trials<span id="id15">[<a class="reference internal" href="../cz-end-matter/reference.html#id145"><span>22</span></a>]</span>.</p>
<p>Perhaps it makes sense that we are not finding drug targets quickly, as we still don’t know the functionality of approximately 20% of human genes<span id="id16">[<a class="reference internal" href="../cz-end-matter/reference.html#id278"><span>24</span></a>]</span>.
And genes are only a small part (1-2%) of our DNA<span id="id17">[<a class="reference internal" href="../cz-end-matter/reference.html#id155"><span>25</span></a>]</span>, and the part we understand best.
Beyond our DNA, there are many other aspects of our cellular and social environments that will have an effect on which parts of our genes are being actively used, and how much.
This section provides an overview of our current scientific model for how DNA affects phenotype, so that we can identify the sources of information that we do have and can make use of.</p>
<p>the following phenotypes in the ALSPAC cohort: <code class="docutils literal notranslate"><span class="pre">MP:0001501</span></code> <em>Abnormal Sleep Pattern</em> (measured using <code class="docutils literal notranslate"><span class="pre">FJCI250</span></code> <em>Sleep symptom score</em>), <code class="docutils literal notranslate"><span class="pre">MP:0001933</span></code> <em>Abnormal litter size</em> (measured by <code class="docutils literal notranslate"><span class="pre">mz010a</span></code> <em>Pregnancy size</em>), <code class="docutils literal notranslate"><span class="pre">MESH:D001259</span></code> <em>Ataxia</em> (measured by <code class="docutils literal notranslate"><span class="pre">kw2030</span></code> <em>Child ever thought to have a problem with clumsiness/coordination</em>), and <code class="docutils literal notranslate"><span class="pre">HP:0001249</span></code> <em>Intelligence/intellectual disability</em> (measured by <code class="docutils literal notranslate"><span class="pre">f8ws150</span></code> <em>Child had special needs</em>).
Despite what we don’t know, this is also the moment when we have a unique hope to unravel some of these mysteries.
We have openly available, expertly curated, databases containing the great collective knowledge of many experiments about our DNA, how it is being used, and what traits it affects.
These databases are being filled at an alarming speed by researchers around the world with the advent of new technologies.
Perhaps it is now possible to begin to synthesise some of this collective knowledge into a fuller understanding of complex traits.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c02-biology-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="0-index.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">2. </span>How phenotype arises from genotype</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="2-biological-molecules.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">2.2. </span>Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Natalie Zelenka<br/>
  
      &copy; Copyright 2020.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>