
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.e8f53015daec13862f6db5e763c41738.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="2.4. Looking more closely at proteins: function, structure and classification" href="4-more-proteins.html" />
    <link rel="prev" title="2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology." href="2-biological-molecules.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genomes">
   2.3.1. Genomes
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-exome-and-the-proteome">
   2.3.2. The exome and the proteome
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#genes">
   2.3.3. Genes
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#a-gene-for-x">
     2.3.3.1. “A gene for X”
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#units-of-heritability">
     2.3.3.2. Units of heritability
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#things-that-are-not-genes">
   2.3.4. Things that are not genes
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#indels-and-copy-number-variations">
   2.3.5. Indels and Copy Number Variations
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#single-nucleotide-polymorphisms">
   2.3.6. Single Nucleotide Polymorphisms
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#non-synonymous-snvs">
     2.3.6.1. Non-synonymous SNVs
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#synonymous-snvs">
     2.3.6.2. Synonymous SNVs
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="a-closer-look-at-dna-genomes-genes-and-genetic-variation">
<span id="dna-organised"></span><h1><span class="section-number">2.3. </span>A closer look at DNA: Genomes, Genes, and Genetic Variation<a class="headerlink" href="#a-closer-look-at-dna-genomes-genes-and-genetic-variation" title="Permalink to this headline">¶</a></h1>
<p>Now that we have a basic overview of how DNA can influence phenotype, we can discuss the way that DNA is organised and categorised in a little more detail. We’ll look from big (<a class="reference internal" href="#genomes"><span class="std std-ref">genomes</span></a>) to small (<a class="reference internal" href="#single-nucleotides"><span class="std std-ref">single nucleotides</span></a>).</p>
<div class="section" id="genomes">
<span id="id1"></span><h2><span class="section-number">2.3.1. </span>Genomes<a class="headerlink" href="#genomes" title="Permalink to this headline">¶</a></h2>
<p>The genome is the full amount of DNA belonging to an organism. We can talk about the genome of an individual, or about the genome of an organism (e.g. the human genome).
When we talk about an organism’s genome, we are actually talking about an example genome for that organism: the organisms’ <em>reference</em> genome.
The reference genome does not belong to any individual organism, but instead is supposed to have the most common nucleotide at each DNA location.</p>
<p>Reference genomes allow us to make general statements about an organism (e.g. “the human genome is 3 billion base pairs long”), and also to make comparisons between organisms (e.g. <a class="reference internal" href="#human-banana-dna"><span class="std std-ref">“humans share 1% of their DNA with a banana”</span></a>).
We also discuss individuals’ genomes in relation to the difference between the individual and the reference genome.</p>
<div class="margin sidebar" id="human-banana-dna">
<p class="sidebar-title">Humans and bananas</p>
<p>Humans share 50% of their <em>protein-coding</em> dna with bananas, but only 1% of their genome.</p>
</div>
</div>
<div class="section" id="the-exome-and-the-proteome">
<span id="exome-proteome"></span><h2><span class="section-number">2.3.2. </span>The exome and the proteome<a class="headerlink" href="#the-exome-and-the-proteome" title="Permalink to this headline">¶</a></h2>
<p>Still thinking big, we have the <em>exome</em> and the <em>proteome</em>.
Both of these refer to locations across the whole genome, but missing stretches in between.
The exome describes the set of all exons (protein-coding nucleotides) across the genome.
The proteome is generally used to mean the set of all proteins in in an organism (which can be much larger than the set of genes due to <a class="reference internal" href="2-biological-molecules.html#alternative-splicing"><span class="std std-ref">alternate splicing</span></a>), but it can also be used to describe the part of the genome relating to the set of protein sequences.</p>
<div class="margin sidebar" id="omics">
<p class="sidebar-title">Omics</p>
<p>As well as genomics (the science of genomes) and proteomics (proteins), there is also transcriptomics (transcripts),  metabolomics (metabolites, e.g. sugars, lipids, etc).
Together these research areas are known as <em>omics</em> and research which combines these fields is known as multi-omics.</p>
</div>
</div>
<div class="section" id="genes">
<span id="what-are-genes"></span><h2><span class="section-number">2.3.3. </span>Genes<a class="headerlink" href="#genes" title="Permalink to this headline">¶</a></h2>
<p>DNA is often considered at the level of the gene.
Genes have been so central to the historical study of dna (hence the name <em>gene</em>tics), and the gene-centric view of molecular biology continues to this day.
For example, many researchers have favourite genes, which they primarily study, and understand the mechanisms of in detail.
And for this reason, diseases and phenotypes are often attributed at the level of the gene, rather than at a more fine-grained level of the specific mutation.</p>
<p id="genes-are-complex">As <a class="reference internal" href="2-biological-molecules.html#genes-first-mentioned"><span class="std std-ref">previously mentioned</span></a>, in this thesis, I use genes to mean a stretches of DNA which can be transcribed into RNA (i.e. I include “RNA genes” in my definition).
However, the seemingly simple definition hides a lot of complexities: due to their long history, the word “gene” has had different uses and meanings.</p>
<div class="section" id="a-gene-for-x">
<span id="gene-for-x"></span><h3><span class="section-number">2.3.3.1. </span>“A gene for X”<a class="headerlink" href="#a-gene-for-x" title="Permalink to this headline">¶</a></h3>
<p>The word gene is often used as shorthand for “DNA that causes phenotypic differences” (for example in Richard Dawkin’s best-seller “The Selfish Gene”, and in news articles with titles of the form “Scientists have discovered a gene for…”).
However, there are multiple reasons why this is an incomplete and in some ways outdated understanding.
Single gene diseases do exist, however most of the time <a class="reference internal" href="2-biological-molecules.html#alternative-splicing"><span class="std std-ref">the same gene can make multiple different proteins</span></a>) which may not all cause phenotypic differences, the same protein can be involved in multiple different pathways and have multiple functions, and multiple proteins can contribute to one function.
Genes are also not guaranteed to cause phenotypic differences, and are not the only sections of DNA which can influence phenotype.
Another complication is that genes can overlap, meaning that a single nucleotide mutation could impact on multiple genes.
And finally, sometimes entirely different genes can create identical proteins after translation.</p>
<p>The interaction between DNA, RNA and proteins, and the environment is also important to consider.
Although DNA makes RNA makes proteins and proteins do pretty much everything in our bodies, which proteins are made and how they behave is highly dependent on the environment.
The function of a gene might not be evident in some environments since the protein is never transcribed, or it may behave differently.
Many traits may be mostly environmental.</p>
</div>
<div class="section" id="units-of-heritability">
<span id="units-heritability"></span><h3><span class="section-number">2.3.3.2. </span>Units of heritability<a class="headerlink" href="#units-of-heritability" title="Permalink to this headline">¶</a></h3>
<p>Genes are also often touted as a “unit of heritability/heredity”, but this is similarly not always the case.
DNA is more likely to be inherited together if it is close together on the chromosome, so generally we inherit whole copies of genes (and the regions around them) together - in fact usually we often inherit stretches of multiple genes together.
Despite this, it is also possible that genes are not inherited “in one piece” with one whole copy from each parent.</p>
</div>
</div>
<div class="section" id="things-that-are-not-genes">
<span id="not-genes"></span><h2><span class="section-number">2.3.4. </span>Things that are not genes<a class="headerlink" href="#things-that-are-not-genes" title="Permalink to this headline">¶</a></h2>
<p>There are many related concepts that contain the word gene simply because they are stretches of DNA, but that do not fit our definition.
For example <a class="reference internal" href="#transposable-elements"><span class="std std-ref">“jumping genes”</span></a> and <a class="reference internal" href="#psuedogenes"><span class="std std-ref">pseudogenes</span></a> are both important parts of the human genome, which may effect phenotype, but not via proteins.</p>
<div class="margin sidebar" id="transposable-elements">
<p class="sidebar-title">Transposable Elements</p>
<p>Transposable elements a.k.a. transposons or “jumping genes” are sections of DNA that move from one section of the genome to another. Their similarity with genes only extends as far as the fact that they are stretches of DNA. Despite making up 50% of the human genome, they only rarely overlap with protein-coding genes<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id232"><span>27</span></a>]</span>. The vast majority of human transposons are <em>silent</em>, i.e. are incapable or prevented from moving.</p>
</div>
<div class="margin sidebar" id="psuedogenes">
<p class="sidebar-title">Pseudogenes</p>
<p>Pseduogenes are segments of DNA that look like genes (have high homology to known genes), but are missing some functionality such that they do not encode proteins. Some pseudogenes cannot be transcribed (e.g. due to missing regulatory regions), and some are transcribed, but not translated (e.g. due to a premature stop codon).
Although they were originally characterised as non-functional, they have been found to have biological roles, for example through being transcribed into functional RNAs<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id203"><span>28</span></a>]</span>.</p>
</div>
<p id="genes-enhancers-promoters">In addition, there are also stretches of DNA that are of interest in relation to genes: enhancers, silencers, insulators, and promoters.
These are stretches of DNA that control the regulation of specific gene’s transcription.
Mutations in these stretches of DNA are often understood in relation to the genes that they regulate.</p>
</div>
<div class="section" id="indels-and-copy-number-variations">
<span id="other-variation"></span><h2><span class="section-number">2.3.5. </span>Indels and Copy Number Variations<a class="headerlink" href="#indels-and-copy-number-variations" title="Permalink to this headline">¶</a></h2>
<p>Smaller lengths of DNA are mostly of interest as <em>variants</em> where there are differences at those locations between individuals.
These differences include insertions and deletions (<em>indels</em>) where sections of DNA, ranging from 1 to 10,000 base pairs, are inserted or deleted from the genome, and <em>copy number variations</em> where (generally larger) stretches of the genome are repeated.
However, the most well-studied and common type of genetic variation between humans are single nucleotide polymorphisms (SNPs, pronounced “snips”).</p>
</div>
<div class="section" id="single-nucleotide-polymorphisms">
<span id="single-nucleotides"></span><h2><span class="section-number">2.3.6. </span>Single Nucleotide Polymorphisms<a class="headerlink" href="#single-nucleotide-polymorphisms" title="Permalink to this headline">¶</a></h2>
<p>A SNP is a location on an organism’s genome where there are differences of a single nucleotide (A, C, T, G) between individuals.
In some fields, these variations are only considered to be Single Nucleotide <em>Polymorphisms</em> if they are relatively commonly occurring in the population (at least 1%), while Single Nucleotide Variants (SNVs) can include both rare and common variants.</p>
<p>Variation at a location does not imply a disease-causing effect, many SNPs appear to be neutral.
Much of the time, the aim of studying such variants is to determine which are which.
This is often done through looking at their rarity, either in a specific human population (e.g. people with diabetes), the entire human population, or across the tree of life.</p>
<div class="margin sidebar" id="define-allele">
<p class="sidebar-title">Alleles</p>
<p>The different forms that a variant can take in the population are called alleles.
Alleles can be as big as different forms of a whole gene, or as small in length as an individual nucleotide.</p>
</div>
<p>SNPs are defined by their location on a human reference genome, for example “chromosome 5, position 7870860” (often written <code class="docutils literal notranslate"><span class="pre">5:7870860</span></code>).
An individual <a class="reference internal" href="#define-allele"><span class="std std-ref">allele</span></a> for a given SNP is defined as “wild” type if it matches the reference genome and “mutant” if it does not.
The reference genome does not always have the most common allele at each location, although this is it’s aim, so “wild” and “mutant” do not necessarily imply anything about rarity.</p>
<p>If there are only two nucleotide possibilities for a SNP (e.g. it could be A or C at a given position), then it is called bi-allelic; the vast majority of SNPs are of this type.
Multi-allelic SNPs (e.g. it could be A, T or C), and tri-allelic SNPs (three choices) are much rarer.</p>
<p>Since humans mostly have two copies of each chromosome (except for X/Y chromosomes in genetically male people, and people with chromosomal anomalies), an individual will usually have two alleles for each SNP.
These may match (which we call homozygous) or not (heterozygous).
Sometimes a disease-causing allele can cause problems even for heterozygotes, while in other cases a person needs two copies of the disease-causing allele in order for it to have an effect.</p>
<p>SNPs can occur either in coding or non-coding regions of the genome.
In non-coding regions, SNPs can still affect gene expression, for example by altering a regulatory site.
SNPs in coding regions have two types: synonymous or non-synonymous, based on whether they alter the amino acid sequence.</p>
<div class="section" id="non-synonymous-snvs">
<span id="non-synonymous"></span><h3><span class="section-number">2.3.6.1. </span>Non-synonymous SNVs<a class="headerlink" href="#non-synonymous-snvs" title="Permalink to this headline">¶</a></h3>
<p>If a SNP alters the amino acid makeup of a protein, it is known as non-synonymous.
Non-synonymous SNVs can cause either nonsense or missense mutations.</p>
<p>Nonsense mutations occur where the SNP substitution results in a stop codon (e.g. TAG) in an unusual position, which signals for a ribosome to stop translating RNA into a protein.
This results in an incomplete and usually nonfunctional protein.
The effect of a nonsense mutation would be more or less severe depending on the location of the new stop codon.
For example, if it was close to the end of the protein, the protein may still be functional.
Sufficiently incomplete proteins are usually destroyed by the cell.</p>
<p id="missense">On the other hand, missense mutations occur where the SNP substitution results in an amino acid substitution in the protein.
Some amino acids can be substituted without causing any difference to the function of the protein, while others can severely impede the protein.</p>
</div>
<div class="section" id="synonymous-snvs">
<span id="synonymous"></span><h3><span class="section-number">2.3.6.2. </span>Synonymous SNVs<a class="headerlink" href="#synonymous-snvs" title="Permalink to this headline">¶</a></h3>
<p>Synonymous SNVs occur where substituting the usual nucleotide with another results in the same amino acid.
The resulting protein will have the exact same functionality.
However, synonymous SNVs could still have an effect on high-level traits, since different nucleotides are translated at different speeds.</p>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c02-biology-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="2-biological-molecules.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title"><span class="section-number">2.2. </span>Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="4-more-proteins.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title"><span class="section-number">2.4. </span>Looking more closely at proteins: function, structure and classification</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>