
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>3.4. Bias &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="3.5. PQI" href="5-pqi.html" />
    <link rel="prev" title="3.3. Ontologies" href="3-ontologies.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-filter/0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html#motivation-linking-genotype-and-phenotype">
     5.2. Motivation: linking genotype and phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html#motivation-protein-function-prediction">
     5.3. Motivation: protein function prediction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html#id5">
     5.4. CAFA
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html#filip">
     5.5.
     <code class="docutils literal notranslate">
      <span class="pre">
       filip
      </span>
     </code>
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/2-data.html">
     5.6. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/3-methods.html">
     5.7. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/4-results.html">
     5.8. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/5-discussion.html">
     5.9. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/6-future-work.html">
     5.10. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/1-background.html">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/4-uberon.html">
     6.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/5-batch-correction.html">
     6.5. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html">
     6.6. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html#combined-data-set">
     6.7. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/7-discussion.html">
     6.8. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/8-future-work.html">
     6.9. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#we-want-to-trust-science">
   3.4.1. We want to trust science
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#science-s-self-correcting-mechanism">
     3.4.1.1. Science’s self correcting mechanism
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-reproducibility-crisis">
   3.4.2. The reproducibility crisis
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#null-hypothesis-significance-testing">
     3.4.2.1. Null Hypothesis Significance Testing
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#p-hacking">
     3.4.2.2. P-hacking
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#publication-bias-against-negative-results">
     3.4.2.3. Publication bias against negative results
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#other-types-of-bias">
   3.4.3. Other types of bias
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="bias">
<span id="compbio-bias"></span><h1><span class="section-number">3.4. </span>Bias<a class="headerlink" href="#bias" title="Permalink to this headline">¶</a></h1>
<div class="section" id="we-want-to-trust-science">
<h2><span class="section-number">3.4.1. </span>We want to trust science<a class="headerlink" href="#we-want-to-trust-science" title="Permalink to this headline">¶</a></h2>
<blockquote class="epigraph">
<div><p>The imposing edifice of science provides a challenging view of what can be achieved by the accumulation of many small efforts in a steady objective and dedicated search for truth.</p>
<p class="attribution">—Charles H. Townes</p>
</div></blockquote>
<p>We want to trust the results of scientific research.
Not only when it’s our own (because it’s fun and exciting to look for and find the truth), but because science builds on itself and building on shaky ground wastes time and money.
Moreover, scientific research is generally paid for by tax, and the results that are generated by it drive policy, drug treatments, and innovations.</p>
<p>In all fields, science is a search for the truth.
And in all fields, there are concerns about what makes bad, unreliable, un-useful, or biased research; what must be done or not done to uphold science’s claim to truth, or at least reliability.</p>
<p>In contrast to other fields, many bioinformatics datasets have been freely available and accessible on the internet since their inception; in this sense the field is far ahead of others.
The issues which affect the reliability of science in general, however, are likely to be present in computational biology, too.
This could have strong effects on the research that is reliant on these large ontologies and databases.</p>
<div class="section" id="science-s-self-correcting-mechanism">
<span id="self-correcting-mechanism"></span><h3><span class="section-number">3.4.1.1. </span>Science’s self correcting mechanism<a class="headerlink" href="#science-s-self-correcting-mechanism" title="Permalink to this headline">¶</a></h3>
<p>Scientific results are often based on statistics, so it’s inevitable that some proportion of published scientific results will not be true.
This isn’t a problem, as over time, researchers can double-check interesting scientific results, and the literature can be updated to reflect that.
This is sciences <em>self-correcting mechanism</em>.
If a result can be replicated in a different circumstance by a different person, it reinforces the likelihood that the result is true.
A replication doesn’t have to reveal the exact same level of statistical significance  or effect size to be successful, but (usually, depending on definitions) just a similar result.</p>
</div>
</div>
<div class="section" id="the-reproducibility-crisis">
<h2><span class="section-number">3.4.2. </span>The reproducibility crisis<a class="headerlink" href="#the-reproducibility-crisis" title="Permalink to this headline">¶</a></h2>
<blockquote class="epigraph">
<div><p>In science consensus is irrelevant. What is relevant is reproducible results.</p>
<p class="attribution">—Michael Crichton</p>
</div></blockquote>
<p>The reproducibility crisis is the realisation that worryingly large proportions of research results do not reproduce.
Replication studies have found that only 11% of cancer research findings<a href="#id1"><span class="problematic" id="id2">:cite:`Begley2012-oc`</span></a>, 20-25% of drug-target findings<a href="#id3"><span class="problematic" id="id4">:cite:`Begley2012-oc,Prinz2011-rh`</span></a>, and 39% of psychology findings<a href="#id5"><span class="problematic" id="id6">:cite:`Open_Science_Collaboration2015-vi`</span></a> reproduced.
Surveys of researchers across disciplines reveal that more than 70% of scientists say they have failed to reproduce another paper’s result, and over 50% say they have failed to reproduce their own results<a href="#id7"><span class="problematic" id="id8">:cite:`Baker2016-tb`</span></a>.
It seems that science’s self-correcting mechanism is not working as intended.</p>
<p>This surprising irreproducibility is thought to be due to a multitude of factors, including poor data management, lack of available materials/details of experiments, publication bias, poor statistical knowledge, and questionable research practices<a href="#id9"><span class="problematic" id="id10">:cite:`Bishop2019-hg`</span></a>.</p>
<div class="section" id="null-hypothesis-significance-testing">
<h3><span class="section-number">3.4.2.1. </span>Null Hypothesis Significance Testing<a class="headerlink" href="#null-hypothesis-significance-testing" title="Permalink to this headline">¶</a></h3>
<blockquote class="epigraph">
<div><p>Statistical significance is the least interesting thing about the results. You should describe the results in terms of measures of magnitude –not just, does a treatment affect people, but how much does it affect them.</p>
<p class="attribution">—Gene V. Glass</p>
</div></blockquote>
<div class="margin sidebar" id="fischer-racism">
<p class="sidebar-title">Fischer racism</p>
<p>Fischer was a racist, and campaigned for the legalisation of eugenic sterilisation.</p>
</div>
<p>Null Hypothesis Significance Testing (NHST) is the most popular method by which scientific hypotheses are tested and reported.
This reporting usually consists mostly of a p-value, a measure of statistical significance: the likelihood that a false positive at least this extreme could be obtained just by chance.
The threshold for this, usually denoted by <span class="math notranslate nohighlight">\(\alpha\)</span> is most often set to 0.05, as recommended by <a class="reference internal" href="#fischer-racism"><span class="std std-ref">Fischer</span></a>, however this is not necessarily the most sensible cut-off for science today.
Despite the dominance of p-values as main or only reported statistic across scientific fields, they do not imply that a result is interesting (the effect might be small or the hypothesis uninteresting), or even that it’s likely to be true.
Sometimes the p-value is not even reported, but only whether or not it crossed the p&lt;0.05 threshold.</p>
</div>
<div class="section" id="p-hacking">
<h3><span class="section-number">3.4.2.2. </span>P-hacking<a class="headerlink" href="#p-hacking" title="Permalink to this headline">¶</a></h3>
<div class="figure align-default" id="id11">
<a class="reference internal image-reference" href="../_images/p_hacking.png"><img alt="../_images/p_hacking.png" src="../_images/p_hacking.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3.5 </span><span class="caption-text">Images that are illustrative of researchers approaches to p-values and p-hacking. The left image is <a class="reference external" href="https://twitter.com/FaustoBustos/status/1103435523777978368">a popular tweet</a>, while the right image is <a class="reference external" href="https://xkcd.com/1478/">an xkcd comic</a>).</span><a class="headerlink" href="#id11" title="Permalink to this image">¶</a></p>
</div>
<p>The pressure on scientists to publish means that researchers may be tempted to (or may accidentally, due to statistical ignorance) employ data-mining tactics in order to harvest significant p-values.
This practice is known as “p-hacking”, and evidence for its existence can be found in distributions of p-values in scientific literature<a href="#id12"><span class="problematic" id="id13">:cite:`Head2015-ns`</span></a>, as well as popular culture (<a class="reference internal" href="#id11"><span class="std std-numref">Fig. 3.5</span></a>).
This can include rerunning analysis with different models/covariates, collecting data until a significant p-value is reached, or performing 20 experiments and only publishing the results of one.</p>
<blockquote class="epigraph">
<div><p>The first principle is that you must not fool yourself – and you are the easiest person to fool.
– Richard Feynman</p>
</div></blockquote>
<p>There are several suggested tonics to the problem of uninformative and ubiquitous p-values.
Reporting p-values as numbers rather than in relation to a threshold (e.g. p&lt;0.05) is starting point.
Information about statistical power and effect size should also be provided.
In addition to giving researchers reading a paper a better idea of the quality of it, this also allows science to self-correct a little easier, since individual p-values can then be combined into more reliable p-values, using for example Fischer’s method<a href="#id14"><span class="problematic" id="id15">:cite:`Fisher1990-ro`</span></a>.</p>
<p>For cases where many hypotheses are being generated at once (for example in GWAS), multiple hypothesis corrections (e.g. the Bonferroni correction<a href="#id16"><span class="problematic" id="id17">:cite:`Dunn1958-sj`</span></a> or the False Discovery Rate<a href="#id18"><span class="problematic" id="id19">:cite:`Benjamini1995-me`</span></a>) can be employed to adjust the p-value to account for this.</p>
</div>
<div class="section" id="publication-bias-against-negative-results">
<h3><span class="section-number">3.4.2.3. </span>Publication bias against negative results<a class="headerlink" href="#publication-bias-against-negative-results" title="Permalink to this headline">¶</a></h3>
<p>Although with standard p-value and statistical power cut-offs, negative results are more likely to be true than positive ones<a href="#id20"><span class="problematic" id="id21">:cite:`Ioannidis2005-mo`</span></a>, negative results are much harder to publish.
This bias is likely to be responsible for the draw of questionable research practices like p-hacking.
It also means that there is a lot of unpublished, negative results which are likely to be repeated, since there is no way that someone could know it has already been done.
A highly powered negative result could be very interesting, for example, we know hardly anything about which genes do not appear to affect phenotypes, since these results are not published<a href="#id22"><span class="problematic" id="id23">:cite:`Barbaric2007-zm`</span></a>, but they would help enormously with the challenge of creating a gold standard data set for gene function prediction.</p>
</div>
</div>
<div class="section" id="other-types-of-bias">
<h2><span class="section-number">3.4.3. </span>Other types of bias<a class="headerlink" href="#other-types-of-bias" title="Permalink to this headline">¶</a></h2>
<p>Two other sources of research bias come to mind which may impact the contents of computational biology databases, and therefore methodologies that make use of them.</p>
<p><strong>1. Replications:</strong>
Replications are not commonly published (as they are not novel), this discourages people from doing them, or at least from writing them up.
This is true for both computational methodologies (where often the code and computational environment needed to replicate the research are not provided), and for experimental work.
This means that it is difficult for science to self-correct this work.</p>
<p><strong>2. Biased study of genes:</strong>
Some genes are very famous, racking up thousands of publications, while others are entirely unstudied.
Even looking only at human genes, there is a huge divide between the most and least studied genes.
This means that there are many functions of genes which will be missing from the gene ontology annotations (for example) for less well-studied genes.</p>
<p><strong>3. Incentives for speed over quality:</strong>
In research, there are strong incentives to publish often (“publish or perish”).
This incentivises doing the minimum possible to publish, and disincentivises spending time on quality control or providing useful metadata wherever you can publish without it.
There are movements (like the <a class="reference external" href="http://slow-science.org/">slow science manifesto</a><a href="#id24"><span class="problematic" id="id25">:cite:`noauthor_undated-lr`</span></a>) to combat this, but it remains the default.
Clearly this impacts on the quality of entries in databases, which is important when these have such strong</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c03-compbio-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="3-ontologies.html" title="previous page"><span class="section-number">3.3. </span>Ontologies</a>
    <a class='right-next' id="next-link" href="5-pqi.html" title="next page"><span class="section-number">3.5. </span>PQI</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>