
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>3.5. Sources of bias in computational biology &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="3.6. Proteome Quality Index" href="6-pqi.html" />
    <link rel="prev" title="3.4. Predictive computational methods" href="4-comp-bio-methods.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#trusting-the-results-of-research">
   3.5.1. Trusting the results of research
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#science-s-self-correcting-mechanism">
     3.5.1.1. Science’s self correcting mechanism
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what-makes-research-trustworthy">
     3.5.1.2. What makes research trustworthy?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-reproducibility-crisis">
   3.5.2. The reproducibility crisis
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sources-of-irreproduciblity-and-how-to-combat-them">
   3.5.3. Sources of irreproduciblity and how to combat them
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#null-hypothesis-significance-testing">
     3.5.3.1. Null Hypothesis Significance Testing
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#p-hacking-and-harking">
     3.5.3.2. P-hacking and HARKing
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#publication-bias">
     3.5.3.3. Publication bias
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#code-and-data-availability">
     3.5.3.4. Code and data availability
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="sources-of-bias-in-computational-biology">
<span id="compbio-bias"></span><h1><span class="section-number">3.5. </span>Sources of bias in computational biology<a class="headerlink" href="#sources-of-bias-in-computational-biology" title="Permalink to this headline">¶</a></h1>
<p>The wealth of Open resources in computational biology, from databases to predictive methods to ontologies, hold exciting possibilities and are a credit to the collaborative spirit of the field.
It’s still important, however, to look at them with a critical eye, in order to be aware of their limits.</p>
<div class="section" id="trusting-the-results-of-research">
<span id="why-bias-bad"></span><h2><span class="section-number">3.5.1. </span>Trusting the results of research<a class="headerlink" href="#trusting-the-results-of-research" title="Permalink to this headline">¶</a></h2>
<blockquote class="epigraph">
<div><p>The imposing edifice of science provides a challenging view of what can be achieved by the accumulation of many small efforts in a steady objective and dedicated search for truth.</p>
<p class="attribution">—Charles H. Townes</p>
</div></blockquote>
<p>We all want to be able to trust the results of scientific research.
Not only when it’s our own, but because science builds on itself and building on shaky ground wastes time and money.
Moreover, scientific research is generally paid for by tax, and the results that are generated by it drive policy, drug treatments, and innovations.
Everyone has a vested interest.</p>
<p>In all fields, science is a search for knowledge.
And in all fields, there are concerns about what makes bad, unreliable, un-useful, or biased research; what must be done or not done to uphold science’s claim to truth, or at least reliability.</p>
<p>In contrast to other fields, many bioinformatics datasets have been freely available and accessible on the internet since their inception; in this sense the field is far ahead of others.
The issues which affect the reliability of science in general, however, are likely to be present in computational biology, too.
This could have strong effects on the research that is reliant on these large ontologies and databases.</p>
<div class="section" id="science-s-self-correcting-mechanism">
<span id="self-correcting-mechanism"></span><h3><span class="section-number">3.5.1.1. </span>Science’s self correcting mechanism<a class="headerlink" href="#science-s-self-correcting-mechanism" title="Permalink to this headline">¶</a></h3>
<p>Scientific results are often based on statistics, so it’s inevitable that some proportion of published scientific results will not be true simply due to the sample on which the hypothesis was tested.
The common wisdom is that this isn’t a problem, as over time, researchers can double-check interesting scientific results, and the literature can be updated to reflect that.
This is known as sciences <em>self-correcting mechanism</em>.
If a result can be replicated in a different circumstance by a different person, it reinforces the likelihood that the result is true.
A replication doesn’t have to reveal the exact same level of statistical significance or effect size to be successful, but (usually, depending on definitions) just a similar result.</p>
</div>
<div class="section" id="what-makes-research-trustworthy">
<span id="define-reproducibility"></span><h3><span class="section-number">3.5.1.2. </span>What makes research trustworthy?<a class="headerlink" href="#what-makes-research-trustworthy" title="Permalink to this headline">¶</a></h3>
<table class="table" id="reproducibility-table">
<caption><span class="caption-number">Table 3.2 </span><span class="caption-text">Definitions relating to reproducibility, adapted from <a class="reference external" href="https://the-turing-way.netlify.app/reproducible-research/overview/overview-definitions.html">The Turing Way</a>.</span><a class="headerlink" href="#reproducibility-table" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"></th>
<th class="head"><p>Data: Same</p></th>
<th class="head"><p>Data: Different</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><strong>Analysis: same</strong></p></td>
<td><p><em>Reproducible</em></p></td>
<td><p><em>Replicable</em></p></td>
</tr>
<tr class="row-odd"><td><p><strong>Analysis: different</strong></p></td>
<td><p><em>Robust</em></p></td>
<td><p><em>Generalisable</em></p></td>
</tr>
</tbody>
</table>
<p>There are different levels of trust that we might have in the results of research.
This ranges from a basic trust that the researchers didn’t make any mistakes in their implementation (their code is doing what they thought it was) to a trust that the result is trustworthy even in new contexts.
Much of the academic discussion surrounding this hinges on the concept of <em>reproducibility</em>, for which there are many contrasting definitions.
I like the definitions from <a class="reference external" href="https://the-turing-way.netlify.app/reproducible-research/overview/overview-definitions.html">The Turing Way</a><span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id219"><span>118</span></a>]</span>, shown in <a class="reference internal" href="#reproducibility-table"><span class="std std-numref">Table 3.2</span></a>.
This table says for example if you get the same result with the same data and same analysis as the original research, then the result is <em>reproducible</em>.
And if you get the same result when the data is the same, but the analysis is different (e.g. a different implementation of the code, or a different specific analysis meant to measure the same thing), then the result is <em>robust</em>.</p>
<p>Although a generalisable result is the most desirable and interesting, as long as the research is <em>reproducible</em>, it can still positively contribute towards our joint scientific knowledge.
This definition of reproducibility also requires that everything needed to run the experiment again is provided, including fine details of methods (in computational biology, this is often equivalent to code) and data.
In the absence of this, science’s self-correcting mechanism is short-circuited.</p>
</div>
</div>
<div class="section" id="the-reproducibility-crisis">
<span id="reproducibility-crisis"></span><h2><span class="section-number">3.5.2. </span>The reproducibility crisis<a class="headerlink" href="#the-reproducibility-crisis" title="Permalink to this headline">¶</a></h2>
<blockquote class="epigraph">
<div><p>In science consensus is irrelevant. What is relevant is reproducible results.</p>
<p class="attribution">—Michael Crichton</p>
</div></blockquote>
<p>The reproducibility crisis is the realisation that worryingly large proportions of research results do not replicate.
Replication studies have found that only 11% of cancer research findings<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id57"><span>119</span></a>]</span>, 20-25% of drug-target findings<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id57"><span>119</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id58"><span>120</span></a>]</span>, and 39% of psychology findings<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id59"><span>121</span></a>]</span> could be reproduced.
Surveys of researchers across disciplines reveal that more than 70% of scientists say they have failed to reproduce another paper’s result, and over 50% say they have failed to reproduce their own results<span id="id5">[<a class="reference internal" href="../cz-end-matter/reference.html#id60"><span>122</span></a>]</span>.
It seems that science’s self-correcting mechanism is not working as intended.</p>
</div>
<div class="section" id="sources-of-irreproduciblity-and-how-to-combat-them">
<h2><span class="section-number">3.5.3. </span>Sources of irreproduciblity and how to combat them<a class="headerlink" href="#sources-of-irreproduciblity-and-how-to-combat-them" title="Permalink to this headline">¶</a></h2>
<p>The surprising irreproducibility seen is thought to be explained by a range of factors<span id="id6">[<a class="reference internal" href="../cz-end-matter/reference.html#id61"><span>123</span></a>]</span> including poor data management, lack of available materials/details of experiments, publication bias, poor statistical knowledge, and questionable research practices such as <a class="reference internal" href="#harking"><span class="std std-ref">HARKing</span></a> and <a class="reference internal" href="#p-hacking"><span class="std std-ref">p-hacking</span></a>.
Although it is difficult to estimate, a very small proportion of irreproducible research is thought to be due to fraudulent practices<span id="id7">[<a class="reference internal" href="../cz-end-matter/reference.html#id234"><span>124</span></a>]</span> (although these do still happen), and arguably it’s explainable simply from our reliance on <a class="reference internal" href="#nhst"><span class="std std-ref">Null Hypothesis Significance Testing</span></a><span id="id8">[<a class="reference internal" href="../cz-end-matter/reference.html#id62"><span>125</span></a>]</span>.</p>
<div class="section" id="null-hypothesis-significance-testing">
<span id="nhst"></span><h3><span class="section-number">3.5.3.1. </span>Null Hypothesis Significance Testing<a class="headerlink" href="#null-hypothesis-significance-testing" title="Permalink to this headline">¶</a></h3>
<div class="margin sidebar" id="fischer-racism">
<p class="sidebar-title">Fischer racism</p>
<p>Fischer campaigned for the legalisation of eugenic sterilisation.</p>
</div>
<p>To discuss some of these issues, we first have to understand how scientific hypotheses are usually tested and reported: Null Hypothesis Significance Testing (NHST).
This reporting usually consists mostly of a p-value as a measure of statistical significance: the likelihood that a <a class="reference internal" href="#false-pos"><span class="std std-ref">false positive</span></a> at least this extreme could be obtained just by chance.
The threshold for this, usually denoted by <span class="math notranslate nohighlight">\(\alpha\)</span> is most often set to 0.05, as recommended by <a class="reference internal" href="#fischer-racism"><span class="std std-ref">Fischer</span></a>, however this is not necessarily the most sensible cut-off for science today<span id="id9">[<a class="reference internal" href="../cz-end-matter/reference.html#id240"><span>126</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id241"><span>127</span></a>]</span>, and different fields have differing cut-offs.</p>
<blockquote class="epigraph">
<div><p>Statistical significance is the least interesting thing about the results. You should describe the results in terms of measures of magnitude –not just, does a treatment affect people, but how much does it affect them.</p>
<p class="attribution">—Gene V. Glass</p>
</div></blockquote>
<div class="margin sidebar" id="false-pos">
<p class="sidebar-title">False positives in NHST</p>
<p>In the context of NHST in research, a false positive would be finding a p-value below our threshold (0.05) for our sample of measurements when there is not an effect in the underlying population from which we are drawing, while a false negative would be finding a p-value above our threshold for our sample when there is an effect present in the underlying population.</p>
</div>
<p>Despite the dominance of p-values as main or only reported statistic across scientific fields, they do not imply that a result is interesting (the effect might be small or the hypothesis uninteresting), or even that it’s likely to be true.
Sometimes the p-value is not even reported, but only whether or not it crossed the p&lt;0.05 threshold.</p>
</div>
<div class="section" id="p-hacking-and-harking">
<span id="p-hacking"></span><h3><span class="section-number">3.5.3.2. </span>P-hacking and HARKing<a class="headerlink" href="#p-hacking-and-harking" title="Permalink to this headline">¶</a></h3>
<div class="figure align-default" id="id10">
<a class="reference internal image-reference" href="../_images/p_hacking.png"><img alt="../_images/p_hacking.png" src="../_images/p_hacking.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3.5 </span><span class="caption-text">Images that are illustrative of researchers approaches to p-values and p-hacking. The left image is <a class="reference external" href="https://twitter.com/FaustoBustos/status/1103435523777978368">a popular tweet</a>, while the right image is <a class="reference external" href="https://xkcd.com/1478/">an xkcd comic</a>).</span><a class="headerlink" href="#id10" title="Permalink to this image">¶</a></p>
</div>
<div class="margin sidebar" id="harking">
<p class="sidebar-title">HARKing</p>
<p>HARKing (Hypothesising After Results are Known) is a variant of p-hacking in which researchers look at their data and/or run significance tests and then add or remove hypotheses based on the results, but then present “significant” results as <em>a priori</em> (previously held) hypotheses<span id="id11">[<a class="reference internal" href="../cz-end-matter/reference.html#id233"><span>128</span></a>]</span>. Xkcd has a <a class="reference external" href="https://xkcd.com/882/">comic</a> for this, too.</p>
</div>
<p>The pressure on scientists to publish means that researchers may be tempted to (or may accidentally, due to statistical ignorance) employ data-mining tactics in order to harvest significant p-values.
This practice is known as “p-hacking”, and evidence for its existence can be found in distributions of p-values in scientific literature<span id="id12">[<a class="reference internal" href="../cz-end-matter/reference.html#id63"><span>129</span></a>]</span>, as well as popular culture (<a class="reference internal" href="#id10"><span class="std std-numref">Fig. 3.5</span></a>).
This can include rerunning analysis with different models/covariates, collecting data until a significant p-value is reached, or performing 20 experiments and only publishing the results of one as in <a class="reference internal" href="#harking"><span class="std std-ref">HARKing</span></a>.</p>
<blockquote class="epigraph">
<div><p>The first principle is that you must not fool yourself – and you are the easiest person to fool.
– Richard Feynman</p>
</div></blockquote>
<p>There are several suggested tonics to the problem of uninformative and ubiquitous p-values.
Reporting p-values as numbers (e.g. “p=0.0012”) rather than in relation to a threshold (e.g. “p&lt;0.05” or “the hypothesis was found to be highly significant”) is a starting point.
Information about statistical power and effect size should also be provided.
In addition to giving researchers reading a paper a better idea of the quality of it, this also allows science to self-correct a little easier, since individual p-values can then be combined into more reliable p-values, using for example Fischer’s method<span id="id13">[<a class="reference internal" href="../cz-end-matter/reference.html#id64"><span>130</span></a>]</span>.</p>
<p>For cases where many hypotheses are being generated at once (for example in GWAS), multiple hypothesis corrections (e.g. the Bonferroni correction<span id="id14">[<a class="reference internal" href="../cz-end-matter/reference.html#id65"><span>131</span></a>]</span> or the False Discovery Rate<span id="id15">[<a class="reference internal" href="../cz-end-matter/reference.html#id66"><span>132</span></a>]</span>) can be employed to adjust the p-value to account for this.</p>
</div>
<div class="section" id="publication-bias">
<span id="id16"></span><h3><span class="section-number">3.5.3.3. </span>Publication bias<a class="headerlink" href="#publication-bias" title="Permalink to this headline">¶</a></h3>
<p>Although with standard p-value and statistical power cut-offs, negative results are more likely to be true than positive ones<span id="id17">[<a class="reference internal" href="../cz-end-matter/reference.html#id62"><span>125</span></a>]</span>, negative results are much harder to publish.
This bias is likely to be responsible for the draw of questionable research practices like p-hacking.
It also means that there is a lot of unpublished, negative results which are likely to be repeated, since there is no way that someone could know it has already been done.
A highly powered negative result could be very interesting, for example, we know hardly anything about which genes do not appear to affect phenotypes, since these results are not published<span id="id18">[<a class="reference internal" href="../cz-end-matter/reference.html#id21"><span>85</span></a>]</span>, but they would help enormously with the challenge of creating a gold standard data set for gene function prediction.</p>
<p>Publication bias is usually used to descibe the bias against negative results but there are other forms of publication bias which affect computational biology, for example the discrepency in which genes are studied.
Some genes are very famous, racking up thousands of publications, while others are entirely unstudied.
Even looking only at human genes, there is a huge divide between the most and least studied genes.
This means that there are many functions of genes which will be missing from the gene ontology annotations (for example) for less well-studied genes.</p>
<p>Another example is the bias against replications.
Repeats of studies are not commonly published (as they are not novel).
Naturally, this discourages people from doing them, or at least from writing them up.
This is true for both computational methodologies (where often the code and computational environment needed to replicate the research are not provided), and for experimental work.
This means that it is difficult for science to self-correct this work.</p>
</div>
<div class="section" id="code-and-data-availability">
<span id="code-data-availability"></span><h3><span class="section-number">3.5.3.4. </span>Code and data availability<a class="headerlink" href="#code-and-data-availability" title="Permalink to this headline">¶</a></h3>
<div class="margin sidebar" id="software-versions-r">
<p class="sidebar-title">Versions</p>
<p>Different versions of the same package, or even the same well-maintained base software (e.g. in R or Python) can have very different behaviours.
For example in <code class="docutils literal notranslate"><span class="pre">R4.0.0</span></code>, <code class="docutils literal notranslate"><span class="pre">as.numeric(data.frame(&quot;10&quot;,&quot;50&quot;,&quot;20&quot;))</span></code> would return a list <code class="docutils literal notranslate"><span class="pre">10</span>&#160; <span class="pre">50</span>&#160; <span class="pre">20</span></code>, while earlier versions process the same code assuming these characters to be factors and would return instead <code class="docutils literal notranslate"><span class="pre">1</span>&#160;&#160; <span class="pre">3</span>&#160;&#160; <span class="pre">2</span></code>.</p>
<p>This means that it is important for packages to understand how they interact with versions of each package, and also shows how small details in implementation decision-making could effect results.</p>
</div>
<p>A lack of code and data availability, while not necessarily leading to wrong or untrustworthy results, also has a place in making research irreproducible since:</p>
<ol class="simple">
<li><p>Research cannot employ it’s self-correcting mechanism unless the experiment can be repeated.</p></li>
<li><p>If the code or data is obscured, then many of the decisions may also be obscured. Decisions we make in analysis<span id="id19">[<a class="reference internal" href="../cz-end-matter/reference.html#id238"><span>133</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id239"><span>134</span></a>]</span> and even <a class="reference internal" href="#software-versions-r"><span class="std std-ref">small details of implementation</span></a> can effect the results of research, and whether we expect it to generalise to another similar context.</p></li>
</ol>
<p>This is a problem even in computational fields: in computational biology, roughly a third of papers that use code still do not provide it at all<span id="id20">[<a class="reference internal" href="../cz-end-matter/reference.html#id235"><span>135</span></a>]</span>.
Even then, providing the data or code somewhere is not in itself enough to overcome the problem of irreproducible methods: it must also be usable.
The <a class="reference internal" href="#fair"><span class="std std-ref">FAIR</span></a> principles<span id="id21">[<a class="reference internal" href="../cz-end-matter/reference.html#id237"><span>136</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id236"><span>137</span></a>]</span> provide a framework for ensuring this.
In the context of software, this includes sharing your reproducible computational environment (for example, exact versions of the packages that you used).</p>
<div class="margin sidebar" id="fair">
<p class="sidebar-title">FAIR data and code</p>
<p>Data and code are <em>FAIR</em> if they are <em>F</em>indable, <em>A</em>ccesible, <em>I</em>nteroperable, and <em>R</em>eusable for both humans and machines. This includes working with existing standards and softwares, being well-documented, etc.</p>
</div>
<p>While there is a growing movement towards transparent and available materials, and taking time to create these (e.g. the <a class="reference external" href="http://slow-science.org/">slow science manifesto</a><span id="id22">[<a class="reference internal" href="../cz-end-matter/reference.html#id184"><span>138</span></a>]</span>), there is also friction against it due to research’s deeply ingrained culture of “publish or perish”.
This incentivises doing the minimum possible to publish, and disincentivises spending time on quality control or providing useful metadata wherever you can publish without it.</p>
<!--
(other-bias)=
## Other types of bias


[//]: # (TODO: Write prediction method papers are sketchy. Compare to ML.)
**4. Lack of benchmarking and sensitivity analyses for technologies:**

**5. Bugs and lack of testing**

## Biased study of genes
[//]: # (TODO: cross ref above)
[//]: # (TODO: If time put biased study of genes here)
-->
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c03-compbio-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="4-comp-bio-methods.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">3.4. </span>Predictive computational methods</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="6-pqi.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">3.6. </span>Proteome Quality Index</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
        
        </div>
    </div>
    <footer class="footer">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>