
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>3.5. PQI &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="3.6. Summary" href="6-summary.html" />
    <link rel="prev" title="3.4. Bias" href="4-bias.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c0-front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-filter/0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/2-uberon.html">
     5.2. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/4-data.html">
     5.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/5-methods.html">
     5.4. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/6-results.html">
     5.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/7-discussion.html">
     5.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/8-future-work.html">
     5.7. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/1-background.html">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/5-batch-correction.html">
     6.4. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html">
     6.5. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html#combined-data-set">
     6.6. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/7-discussion.html">
     6.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/8-future-work.html">
     6.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#pqi-features">
   3.5.1. PQI features
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#pqi-metrics">
   3.5.2. PQI metrics
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#potential-improvements">
   3.5.3. Potential improvements
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="pqi">
<span id="id1"></span><h1><span class="section-number">3.5. </span>PQI<a class="headerlink" href="#pqi" title="Permalink to this headline">¶</a></h1>
<p>The Proteome Quality Index (PQI) is an attempt to provide quality metrics about completed genome sequences.
It was a group effort, which resulted in a paper<a href="#id2"><span class="problematic" id="id3">:cite:`Zaucha2015-ez`</span></a>, of which I am an author, as well as <a class="reference external" href="http://pqi-list.org/">an associated website (http://pqi-list.org/)</a>.
I contributed to discussions about metrics, and paper editing.</p>
<p>The motivation for creating the index came from problems of reproducibility in the field of genomics.
In creating a daily-updated tree of life (sTOL, sequenced Tree Of Life), it was found that many sequenced genomes were missing vital proteins due to poor sequencing<a href="#id4"><span class="problematic" id="id5">:cite:`Fang2013-et`</span></a>.
Such genomes are reused by many researchers, for example in comparative genomics, and omissions of whole proteins and poor accuracy of others are likely to affect research results.</p>
<p>The PQI website was intended to be both a way for users of past genomes to look up the quality of a genome in advance of some research and, more importantly as a talking point for quality guidelines for genomes/proteomes (genomes with known protein annotations).
While quality control and data submission guidelines were more developed in other areas of computational biology, similar guidelines for genome quality were lacking.</p>
<div class="section" id="pqi-features">
<h2><span class="section-number">3.5.1. </span>PQI features<a class="headerlink" href="#pqi-features" title="Permalink to this headline">¶</a></h2>
<p>The PQI website provides a scoring system for proteomes, bringing together numerous different metrics which are normalised before being averaged into an intuitive star-rating (1-5 stars) with equal weight given to each metric.
Proteomes can be searched for, filtered by the various ratings, downloaded, user-rated and commented on.
Additional proteomes and metrics can be added/suggested by users via the website and documentation describing this is provided.</p>
</div>
<div class="section" id="pqi-metrics">
<h2><span class="section-number">3.5.2. </span>PQI metrics<a class="headerlink" href="#pqi-metrics" title="Permalink to this headline">¶</a></h2>
<p>PQI originally provided 11 measures of proteome quality, that are either local i.e. “clade-based” (in which proteomes are compared to similar organisms) or global (in which case it is compared to all other proteomes).
A clade is a group of organisms that consists of a common ancestor and all its descendants, i.e. is a branch on the tree of life<a href="#id6"><span class="problematic" id="id7">:cite:`Cracraft2004-ud`</span></a>, so an appropriate ancestor must be chosen to define the clade. For PQI, since the purpose of these clades was to compare its’ constituent proteomes, we wanted clades that had similar variability.
This was achieved by choosing parent nodes that are at least 0.01 in branch length away from the proteome (leaf node), and such that the clade contains at least 10 species.
Trees and branch lengths to carry out these calculations were taken from sToL<a href="#id8"><span class="problematic" id="id9">:cite:`Fang2013-et`</span></a>.
For clade-based metrics, proteomes score well if they have similar scores to the rest of the clade.</p>
<table class="table" id="pqi-table">
<caption><span class="caption-number">Table 3.2 </span><span class="caption-text">Description of the quality metrics in PQI</span><a class="headerlink" href="#pqi-table" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
<col style="width: 25%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Metric name</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Notes</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>1. Percentage X-content</p></td>
<td><p>Global</p></td>
<td><p>Percentage of proteome with amino acids denoted by ‘X’, excluding the first residue of each protein.</p></td>
<td><p>Amino acids that cannot be identified, or can have more than one value are represented by an ‘X’ in the amino acid sequence[139]. This occurs when coverage of the sequencing is low. Uncertainty in translation start sites mean the first residue of a protein is often uncertain (‘X’) even in the highest quality proteomes, so these are excluded from this measurement.</p></td>
</tr>
<tr class="row-odd"><td><p>2. PubMed Publication Count</p></td>
<td><p>Global</p></td>
<td><p>The total number of publications related to the genome as listed for that entry in the PubMed database[140].</p></td>
<td><p>This is a measure of how well-studied a proteome is, assuming that proteomes that have been studied more will be of higher quality.</p></td>
</tr>
<tr class="row-even"><td><p>3. CEG Domain</p></td>
<td><p>Global</p></td>
<td><p>Proportion of CEG set which contains homologous domains in the proteome, according to SUPERFAMILY</p></td>
<td><p>This method assumes that all eukaryotic genomes should contain a core set of well-conserved of eukaryotic genes. This score is not calculated for non-eukaryotes. This was done using the Core Eukaryotic Gene (CEG) library used by the CEGMA tool[141], which comes from the Eukaryotic Orthologous Group (KOG) sequence orthology database[142]. Domain-architecture similarity is calculated using the SUPERFAMILY HMM library.</p></td>
</tr>
<tr class="row-odd"><td><p>4. Combination Homology</p></td>
<td><p>???</p></td>
<td><p>???</p></td>
<td><p>???</p></td>
</tr>
<tr class="row-even"><td><p>5. Percentage of sequences in Uniprot</p></td>
<td><p>Global</p></td>
<td><p>Percentage of proteome sequences that appear in Uniprot with 100% sequence identity</p></td>
<td><p>This metric assumes that the majority of discrepancies between Uniprot protein sequences and the proteome protein sequences are due to proteome inaccuracies.</p></td>
</tr>
<tr class="row-odd"><td><p>6. Number of domain superfamilies</p></td>
<td><p>Clade-based</p></td>
<td><p>Number of proteins assigned to domain superfamilies by SUPERFAMILY compared to average for clade.</p></td>
<td><p>Assignment to domain superfamilies was obtained using the SUPERFAMILY HMM Library. The number of superfamilies gives an indication of the diversity of the proteome, so a low number compared to the clade may indicate an incomplete proteome, while a high number could indicate that the proteome contains domain superfamiles that it shouldn’t.</p></td>
</tr>
<tr class="row-even"><td><p>7. Percentage of sequence covered</p></td>
<td><p>Clade-based</p></td>
<td><p>Percentage of amino acid residues in proteome sequence that are covered by SCOP domain superfamily assignments, compared to the average for the clade.</p></td>
<td><p>This metric measures the portion of structured protein sequences found in the proteome as opposed to disordered regions and gaps. This measure assumes related species have a similar breakdown of these types of proteins. A mismatch could indicate that the parts of the genome that are supposed to be protein-coding are an incorrect length, that it is missing proteins, or contains proteins that it shouldn’t.</p></td>
</tr>
<tr class="row-odd"><td><p>8. Percentage of sequences with assignment</p></td>
<td><p>Clade-based</p></td>
<td><p>Percentage of amino acid residues in proteome that have SCOP superfamily assignment according to SUPERFAMILY, compared to the average for the clade.</p></td>
<td><p>Related species are assumed to have a similar percentage of domains with SUPERFAMILY assignments to SCOP superfamilies</p></td>
</tr>
<tr class="row-even"><td><p>9. Mean sequence length</p></td>
<td><p>Clade-based</p></td>
<td><p>The average length of proteins in the proteome (in amino acids), compared to the average for the clade.</p></td>
<td><p>This measure assumes that mean sequence length of proteins should be comparable with those of related species.</p></td>
</tr>
<tr class="row-odd"><td><p>10. Mean hit length</p></td>
<td><p>Clade-based</p></td>
<td><p>Average number of amino acids in superfamily assignments, compared to the average for the clade.</p></td>
<td><p>Longer hits represent better matches to SCOP domains. These are assumed to be similar for similar species.</p></td>
</tr>
<tr class="row-even"><td><p>11. Number of domain families</p></td>
<td><p>Clade-based</p></td>
<td><p>Number of distinct SCOP protein domain families that are annotated to the proteome, compared to the average for the clade.</p></td>
<td><p>The SCOP protein domain families are annotated to the proteome using a hybrid HMM/pairwise similarity method from the SUPERFAMILY resource. Similarly to the number of domain superfamilies, the number of families gives an indication of the diversity of the proteome at the SCOP family level. Domain families were included in addition to domain superfamilies, since they are more specific and may reveal differences that are not apparent at the superfamily level.</p></td>
</tr>
<tr class="row-odd"><td><p>12. Number of domain architectures</p></td>
<td><p>Clade-based</p></td>
<td><p>Number of unique domain architectures (combinations of SCOP domain superfamilies and gaps) in the proteome, according to SUPERFAMILY, compared to the average for the clade.</p></td>
<td><p>Similarly to the number of domain families superfamilies, the number of domain architectures gives an indication of the diversity of the proteome at the SCOP family level.</p></td>
</tr>
</tbody>
</table>
<p><a class="reference internal" href="#pqi-table"><span class="std std-numref">Table 3.2</span></a> shows the original PQI metrics.</p>
<p>As a proteome will get low scores in clade-based metrics when they are unusual compared to the rest of its clade, model organisms (e.g. Homo Sapiens) can get a low score in clade-based metrics since for the wrong reasons - since they are of such higher quality compared to those in it’s clade.
The PQI website’s comment and user-rating features can be used to alert its users to these cases.</p>
<p>Since the publication of the PQI paper, the DOGMA metric<a href="#id10"><span class="problematic" id="id11">:cite:`Dohmen2016-iv`</span></a>, which scores proteomes based on conserved arrangements of protein domains, has been added to the PQI website.</p>
</div>
<div class="section" id="potential-improvements">
<h2><span class="section-number">3.5.3. </span>Potential improvements<a class="headerlink" href="#potential-improvements" title="Permalink to this headline">¶</a></h2>
<p>The mapping from raw score to star rating for some metrics could be improved.
For example, homo sapiens has ‘0’ X-content (the lowest/best possible value), but 4.1 stars.</p>
<p>PQI was created with the potential to add further quality metrics by other researchers.
Although the DOGMA metric was added and it is based on BUSCO<a href="#id12"><span class="problematic" id="id13">:cite:`Simao2015-gc`</span></a>, it might be sensible to add BUSCO, since it has become very popular, and any proteome/genome quality index would probably be seen as incomplete without it.
The continued inclusion of the CEG Domain Combination Homology metric may be questioned since CEGMA is no longer being updated (nor is the KEG database upon which it is based), but BUSCO and CEGMA may be complementary since BUSCO has a weaker requirement for inclusion in the set of proteins, which means that it has more proteins, and vice versa.</p>
<p>To impact on the problem for model organisms for clade-based metrics, one solution would be to treat model-organisms differently in these circumstances, for example score clades containing model organisms against the model organism.
An more subtle alternative to this would be using the NCBI’s Assembly database<a href="#id14"><span class="problematic" id="id15">:cite:`Kitts2016-eo`</span></a>, which has been released since PQI was published. The database tracks how many assemblies there are for each species as well as how many versions of each assembly there has been.
This information could be used to weight the importance of proteomes in clade-based metrics. The number of assemblies and versions could also form a separate score.</p>
<p>Sequencing depth (coverage) and read length are also known markers of genome quality<a href="#id16"><span class="problematic" id="id17">:cite:`Sims2014-df`</span></a>.
Average coverage for an assembly (except reference genomes) is also available in the NCBI Assembly database<a href="#id18"><span class="problematic" id="id19">:cite:`Kitts2016-eo`</span></a>.</p>
<p>Some “third-generation” sequencing technologies create much longer read lengths, but potentially lower accuracies.
A third category of metrics “technology-based” metrics could exist for proteomes where the metric is only really comparable within similar types of technologies. If technology-based metrics were implemented, it may also be sensible to have some metrics which only exist for specific technologies.
For example, for nanopore sequencing, we could implement an indel-based quality metric<a href="#id20"><span class="problematic" id="id21">:cite:`Watson2019-pz`</span></a>.
Including sequencing-technology-specific metrics may encourage contributions from other researchers who specialise in particular technologies. Sequencing technology is also available in the NCBI Assembly database<a href="#id22"><span class="problematic" id="id23">:cite:`Kitts2016-eo`</span></a>.</p>
<p>The development of additional measures is required to deal with other potential problems in genome sequencing, for example GC content, amino acid bias, or contamination from other genomes.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c03-compbio-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="4-bias.html" title="previous page"><span class="section-number">3.4. </span>Bias</a>
    <a class='right-next' id="next-link" href="6-summary.html" title="next page"><span class="section-number">3.6. </span>Summary</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>