
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>4.2. Snowflake Algorithm &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="4.3. Snowflake input data" href="3-data.html" />
    <link rel="prev" title="4.1. Introduction" href="1-introduction.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-filter/0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html#motivation-linking-genotype-and-phenotype">
     5.2. Motivation: linking Genotype and phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html#cafa">
     5.3. CAFA
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/2-data.html">
     5.4. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/3-methods.html">
     5.5. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/4-results.html">
     5.6. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/5-discussion.html">
     5.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/6-future-work.html">
     5.8. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/1-background.html">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/4-uberon.html">
     6.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/5-batch-correction.html">
     6.5. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html">
     6.6. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html#combined-data-set">
     6.7. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/7-discussion.html">
     6.8. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/8-future-work.html">
     6.9. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snowflake-predict">
   4.2.1.
   <code class="docutils literal notranslate">
    <span class="pre">
     snowflake
    </span>
    <span class="pre">
     predict
    </span>
   </code>
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#overview">
     4.2.1.1. Overview
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#snps-are-mapped-to-phenotype-terms-using-dcgo-and-dbsnp">
     4.2.1.2. SNPs are mapped to phenotype terms using DcGO and dbSNP
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#snps-are-given-deleteriousness-scores-using-fathmm">
     4.2.1.3. SNPs are given deleteriousness scores using FATHMM
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#comparison-to-a-background-via-clustering">
     4.2.1.4. Comparison to a background via clustering
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#background-data-sets-the-1000-genomes-project">
     4.2.1.5. Background data sets: The 1000 Genomes Project
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#phenotype-score">
     4.2.1.6. Phenotype score
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#features-added-to-the-predictor">
   4.2.2. Features added to the predictor
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#different-running-modes">
     4.2.2.1. Different running modes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#adding-snp-phenotype-associations-from-dbsnp">
     4.2.2.2. Adding SNP-phenotype associations from dbSNP
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#dealing-with-missing-calls">
     4.2.2.3. Dealing with missing calls
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#confidence-score-per-phenotype">
     4.2.2.4. Confidence score per phenotype
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#outputs">
     4.2.2.5. Outputs
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snowflake-create-background">
   4.2.3.
   <code class="docutils literal notranslate">
    <span class="pre">
     snowflake
    </span>
    <span class="pre">
     create_background
    </span>
   </code>
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genome-project-background-files">
     4.2.3.1. 1000 Genome Project background files
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snowflake-create-input">
   4.2.4.
   <code class="docutils literal notranslate">
    <span class="pre">
     snowflake
    </span>
    <span class="pre">
     create_input
    </span>
   </code>
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#snowflake-preprocessing">
   4.2.5.
   <code class="docutils literal notranslate">
    <span class="pre">
     snowflake
    </span>
    <span class="pre">
     preprocessing
    </span>
   </code>
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#inputs">
     4.2.5.1. Inputs
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id38">
     4.2.5.2. Outputs
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#method">
     4.2.5.3. Method
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#dcgo">
       4.2.5.3.1. DcGO
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#variant-prioritisation">
     4.2.5.4. Variant prioritisation
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#fathmm">
       4.2.5.4.1. FATHMM
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="snowflake-algorithm">
<h1><span class="section-number">4.2. </span><code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> Algorithm<a class="headerlink" href="#snowflake-algorithm" title="Permalink to this headline">¶</a></h1>
<p>The purpose of the <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> algorithm is</p>
<p>This section describes the <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> phenotype prediction algorithm that I developed alongside Jan Zaucha, Ben Smithers and Julian Gough.</p>
<p>The original idea for the predictor was Julian’s and he created an initial implementation of it in perl. This perl version was initially ported to Python by Ben, and subsequently contributed to heavily by both me and Jan, where we both fixed a number of bugs, and added essential functionality. We then forked the software into two different versions to suit our own research, mine focusing on human phenotype prediction. The project is now being taken forward by the Computational Genomics group at MRC led by Julian.</p>
<div class="section" id="snowflake-predict">
<h2><span class="section-number">4.2.1. </span><code class="docutils literal notranslate"><span class="pre">snowflake</span> <span class="pre">predict</span></code><a class="headerlink" href="#snowflake-predict" title="Permalink to this headline">¶</a></h2>
<div class="section" id="overview">
<h3><span class="section-number">4.2.1.1. </span>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h3>
<div class="figure align-default" id="snowflake-overview">
<a class="reference internal image-reference" href="../_images/snowflake-overview.png"><img alt="../_images/snowflake-overview.png" src="../_images/snowflake-overview.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4.1 </span><span class="caption-text">Flowchart showing an overview of the phenotype predictor. Scores are generated per allele using SUPERFAMILY, FATHMM and dcGO for both the input genotype(s), and the background genotypes. These data points are then combined into a matrix, which is then clustered..</span><a class="headerlink" href="#snowflake-overview" title="Permalink to this image">¶</a></p>
</div>
<p><a class="reference internal" href="#snowflake-overview"><span class="std std-numref">Fig. 4.1</span></a> shows how an overview of how <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> operates. One or more genotypes (in VCF format) are needed: this is the <em>input cohort</em>. Each genotype will be compared against all others, including (optionally) a diverse background set from the 1000 Genomes Project<a href="#id1"><span class="problematic" id="id2">:cite:`Consortium2015-ci`</span></a>.</p>
<ol class="simple">
<li><p>For each phenotype <span class="math notranslate nohighlight">\(i\)</span>, a list of SNPs is generated such that the SNPs are associated with the phenotype (according to DcGO), and the SNPs are present in the input individuals.</p></li>
<li><p>The input SNPs are given deleterious scores (using FATHMM), and this us used to construct an <span class="math notranslate nohighlight">\(N_{i}\times M\)</span> matrix of scores is created based on the individual’s alleles for each of these SNPs, (where <span class="math notranslate nohighlight">\(N\)</span>=number of SNPs and <span class="math notranslate nohighlight">\(M\)</span>=number of individuals).</p></li>
<li><p>Individuals of interest are clustered by SNP, alongside a diverse background of other people</p></li>
<li><p>A score is calculated per person per phenotype, designed to calculate how much of an outlier the person is according to their SNPs for a given phenotype.</p></li>
</ol>
<p>Further detail on these steps is provided below:</p>
</div>
<div class="section" id="snps-are-mapped-to-phenotype-terms-using-dcgo-and-dbsnp">
<h3><span class="section-number">4.2.1.2. </span>SNPs are mapped to phenotype terms using DcGO and dbSNP<a class="headerlink" href="#snps-are-mapped-to-phenotype-terms-using-dcgo-and-dbsnp" title="Permalink to this headline">¶</a></h3>
<p>DcGO<a href="#id3"><span class="problematic" id="id4">:cite:`Fang2013-ms`</span></a> is used to map combinations of protein domains to their associated phenotype terms, using a false discovery rate cut-off of <span class="math notranslate nohighlight">\(10^{-3}\)</span> or less. SNPs are therefore mapped to phenotype terms by whether they fall in a gene whose protein contains domains or combinations of domains that are statistically associated with a phenotype. In order to do this, DcGO makes use of SUPERFAMILY<a href="#id5"><span class="problematic" id="id6">:cite:`Gough2001-ct`</span></a> domain assignments, and a variety of ontology annotations (GO<a href="#id7"><span class="problematic" id="id8">:cite:`Ashburner2000-cr`</span></a>, MPO<a href="#id9"><span class="problematic" id="id10">:cite:`Smith2005-uh`</span></a>, HP<a href="#id11"><span class="problematic" id="id12">:cite:`Robinson2010-ga`</span></a>, DOID<a href="#id13"><span class="problematic" id="id14">:cite:`Schriml2012-dz`</span></a> and others).</p>
<p>Using DcGO means that phenotypes are only mapped to protein if the link is statistically significant due to the protein’s contingent domains. This leaves out some known protein-phenotype links, where the function may be due to disorder for example rather than protein domain structure. Known phenotype-associated variants are therefore added back in using dbSNP<a href="#id15"><span class="problematic" id="id16">:cite:`Sherry2001-nm`</span></a>.</p>
</div>
<div class="section" id="snps-are-given-deleteriousness-scores-using-fathmm">
<h3><span class="section-number">4.2.1.3. </span>SNPs are given deleteriousness scores using FATHMM<a class="headerlink" href="#snps-are-given-deleteriousness-scores-using-fathmm" title="Permalink to this headline">¶</a></h3>
<p>The phenotype predictor uses the unweighted FATHMM scores<a href="#id17"><span class="problematic" id="id18">:cite:`Shihab2013-pk`</span></a> to get scores per SNP for the likelihood of it causing a deleterious mutation. This is based on conservation of protein domains across all life, according to data from SUPERFAMILY<a href="#id19"><span class="problematic" id="id20">:cite:`Gough2001-ct`</span></a> and Pfam<a href="#id21"><span class="problematic" id="id22">:cite:`Bateman2002-bz`</span></a>.</p>
<p>This method gives SNPs the same base FATHMM score for being deleterious, no matter which phenotype we are predicting them for. It is therefore the combination of SNPs per phenotype, and their rarity in the population that determines the phenotype prediction score.</p>
</div>
<div class="section" id="comparison-to-a-background-via-clustering">
<h3><span class="section-number">4.2.1.4. </span>Comparison to a background via clustering<a class="headerlink" href="#comparison-to-a-background-via-clustering" title="Permalink to this headline">¶</a></h3>
<p>Individuals are compared to all others through clustering. This usually includes comparing each individual to the genetically diverse background of the 2500 genomes project<a href="#id23"><span class="problematic" id="id24">:cite:`Consortium2015-ci`</span></a>.</p>
<p>Clustering is the task of grouping objects into a number of groups (clusters) so that items in the same cluster are similar to each other by some measure. There are many clustering algorithms, but most are unsupervised learning algorithms which iterate while looking to minimise dissimilarity in the same cluster. A number of options were implemented for the predictor, but for the time being at least, OPTICS is used as a default.</p>
</div>
<div class="section" id="background-data-sets-the-1000-genomes-project">
<h3><span class="section-number">4.2.1.5. </span>Background data sets: The 1000 Genomes Project<a class="headerlink" href="#background-data-sets-the-1000-genomes-project" title="Permalink to this headline">¶</a></h3>
<p>As described in the overview, <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> requires a genetic “background” data set, so that meaningful clustering can take place. Snowflake has the functionality to be run with any background data set, but few datasets are available that contain genetic data from a wide range of populations.</p>
<p>The 1000 Genomes project<a href="#id25"><span class="problematic" id="id26">:cite:`Consortium2015-ci,1000_Genomes_Project_Consortium2012-ek`</span></a> ran from 2008 to 2015, with the aim of comprehensively mapping human genetic variation across diverse populations. The project sequenced whole genomes, and released data in two main phases:</p>
<ul class="simple">
<li><p>Phase 1: 37.9 million variants, in 1092 individuals, across 14 populations</p></li>
<li><p>Phase 3: 84.4 million, in 2504 individuals, across 26 populations</p></li>
</ul>
<p>Data from the 1000 Genomes project are always used for the background cohort to <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> in this thesis, with data from the 1000 Genomes project Phase 3<a href="#id27"><span class="problematic" id="id28">:cite:`Consortium2015-ci`</span></a> used as a default, and earlier experiments using data from Phase 1<a href="#id29"><span class="problematic" id="id30">:cite:`1000_Genomes_Project_Consortium2012-ek`</span></a>.</p>
<p>For both phases of the 1000 Genomes project, data are provided as VCF files for each Chromosome. Both data sets are available through the <a class="reference external" href="https://www.internationalgenome.org/data">International Genome Sequencing Resource</a><a href="#id31"><span class="problematic" id="id32">:cite:`Fairley2020-hp`</span></a> (IGSR); phase 1 VCFs can be downloaded <a class="reference external" href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets/">here</a>, and phase 3 VCFs can be downloaded <a class="reference external" href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/">here</a>.</p>
</div>
<div class="section" id="phenotype-score">
<h3><span class="section-number">4.2.1.6. </span>Phenotype score<a class="headerlink" href="#phenotype-score" title="Permalink to this headline">¶</a></h3>
<p>The OPTICS clustering assigns each individual to a cluster (or labels them as an outlier). Depending on the phenotype term, the cluster is expected to either correspond to a haplogroup or a phenotype. In cases where the cluster refers to a haplogroup, we are interested in the outliers of both clusters, i.e. the local outlier-ness. In cases where the cluster is the phenotype, we are interested in the outlying cluster, i.e. the global outlier-ness.</p>
<p>A local score, <span class="math notranslate nohighlight">\(L_{ij}\)</span> can be defined as the average Euclidean distance from an individual to the centre of it’s cluster, or for individuals that are identified as outliers by OPTICS, 2 multiplied by the distance to the centre of the nearest cluster.</p>
<p>A global score <span class="math notranslate nohighlight">\(G_{ij}\)</span> can be defined as the distance of the cluster to the rest of the cohort.</p>
<p>The global-local score is designed to balance these sources of interest. It sums the two scores, adjusting the weighting by a cluster size correction factor, <span class="math notranslate nohighlight">\(\mu_{\gamma}\)</span>:
$<span class="math notranslate nohighlight">\(score_{ij}=L_{ij}+\mu_{\gamma} \cdot G_{ij}\)</span>$</p>
<p>Such that: <span class="math notranslate nohighlight">\(\mu_{\gamma}=\frac{exp(\gamma \frac{n-n_j}{n})-1}{exp(\gamma)-1}\)</span> where <span class="math notranslate nohighlight">\(\gamma\)</span> is a parameter representing how strongly we wish to penalise large clusters, <span class="math notranslate nohighlight">\(n\)</span> is the over all number of individuals and <span class="math notranslate nohighlight">\(n_j\)</span> is the number of individuals in a cluster.</p>
</div>
</div>
<div class="section" id="features-added-to-the-predictor">
<h2><span class="section-number">4.2.2. </span>Features added to the predictor<a class="headerlink" href="#features-added-to-the-predictor" title="Permalink to this headline">¶</a></h2>
<p>As mentioned, the phenotype predictor was already prototyped when I began working on it. However, considerable time was spent developing, bug-fixing, and extending this prototype. Here, I describe my contributions to this.</p>
<div class="section" id="different-running-modes">
<h3><span class="section-number">4.2.2.1. </span>Different running modes<a class="headerlink" href="#different-running-modes" title="Permalink to this headline">¶</a></h3>
<p>The original version of the phenotype predictor could only be ran one individual compared to a background set at a time. In order to allow for a wider range of inputs (which will be necessary to validate the predictor), support for a wider range of genotype formats and running modes was developed, e.g.:</p>
<ul class="simple">
<li><p>Can be run with one person against a background</p></li>
<li><p>Can be run with multiple people (VCF) against the background</p></li>
<li><p>Can be run with or without the background set if there are enough people in the input set.</p></li>
<li><p>Support for different 23andMe genotype file formats (from different chips).</p></li>
</ul>
<p>As the predictor was developed to perform in different running modes, it became clear that it would be necessary to streamline the algorithm. This included parallelisation (possible due to the independence of different phenotype terms), and various data storage and algorithmic adjustments.</p>
<p>Implementing these running modes and increases in efficiency was a collaborative effort between myself, Jan, and Ben.</p>
</div>
<div class="section" id="adding-snp-phenotype-associations-from-dbsnp">
<h3><span class="section-number">4.2.2.2. </span>Adding SNP-phenotype associations from dbSNP<a class="headerlink" href="#adding-snp-phenotype-associations-from-dbsnp" title="Permalink to this headline">¶</a></h3>
<p>As mentioned in the overview, using DcGO as the only SNP-phenotype mapping leaves out some known associations that are not due to protein domain structure. Adding dbSNP<a href="#id33"><span class="problematic" id="id34">:cite:`Sherry2001-nm`</span></a> associations to the predictor was one of my contributions to this software.</p>
</div>
<div class="section" id="dealing-with-missing-calls">
<h3><span class="section-number">4.2.2.3. </span>Dealing with missing calls<a class="headerlink" href="#dealing-with-missing-calls" title="Permalink to this headline">¶</a></h3>
<p>Genotyping SNP arrays often contain missing calls, where the call can not be accurately determined. This is an obstacle to the phenotype predictor if left unchecked as it can appear that an individual has a very unusual call when it is really just unknown. Sincce most people have a call, the missing call is unusual, and this is flagged.</p>
<p>The most sensible solution to this problem is to assign the most common call for the individual’s cluster (i.e. combination of SNPs). This prevents a new cluster being formed or an individual appearing to be more unusual than they are. However, there is a downside to this approach when there are many missing calls. Adding all missing calls to a cluster that was only slightly more common than the alternatives can lead to the new cluster containing the missing data dwarfing the others. To fix this, SNPs with many missing calls were discarded. This implementation of missing calls was one of my personal contributions.</p>
<p>Alternatives such as assigning the most common call for that SNP only, or assigning an average score for that SNP dimension by carrying out a “normalised cut”<a href="#id35"><span class="problematic" id="id36">:cite:`Poland2006-gd`</span></a> are untenable since they can create the same problem we are trying to overcome: the appearance of an individual having an unusual combination of calls.</p>
</div>
<div class="section" id="confidence-score-per-phenotype">
<h3><span class="section-number">4.2.2.4. </span>Confidence score per phenotype<a class="headerlink" href="#confidence-score-per-phenotype" title="Permalink to this headline">¶</a></h3>
<p>The phenotype predictor outputs a score for each person for each phenotype. Our confidence in these scores depends on the distribution of scores, as well as the scale of them. A distribution of scores with distinct groups of individuals is generally preferable, since most phenotypes that we are interested in are categorical or it is at least more useful to highlight phenotypes that can be predicted this way (i.e. if there are 100 groups with varying risk of a disease, that would be less useful than knowing there are 2 groups with high/low risk).</p>
<p>I developed a simple method of prioritising predictions according to these requirements. A confidence score is achieved by plotting the ranked raw score and measuring the minimum area between a straight line resting on this line (resting on the two points furthest from y=x) and the line itself, as illustrated in <code class="xref std std-numref docutils literal notranslate"><span class="pre">shaded-score</span></code>. Since this measure takes into account the size of the raw scores, these confidence scores  compared across phenotypes.</p>
<p><code class="xref std std-numref docutils literal notranslate"><span class="pre">ranked-scores</span></code> shows an example of an interesting and uninteresting distribution. These distributions mostly depend on the number, population frequency, and FATHMM score of the SNPs associated with the phenotype term.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Code for: An illustration of how the confidence score is calculated (shaded area).  - `shaded-score`</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Code for: Ranked scores for `DOID:1324` - the disease ontology term Lung Cancer (left) and HP:0008518 - the human phenotype ontology term for Absent/underdeveloped sacral bone (right). These represent an interesting and uninteresting distribution of scores, respectively.  - `ranked-scores`</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="outputs">
<h3><span class="section-number">4.2.2.5. </span>Outputs<a class="headerlink" href="#outputs" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="snowflake-create-background">
<h2><span class="section-number">4.2.3. </span><code class="docutils literal notranslate"><span class="pre">snowflake</span> <span class="pre">create_background</span></code><a class="headerlink" href="#snowflake-create-background" title="Permalink to this headline">¶</a></h2>
<p>For any given run of <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code>, we are only interested in variants where:</p>
<ul class="simple">
<li><p>there is overlap between the background cohort, the predictive cohort</p></li>
<li><p>AND we have conservation information so that we can score the deleteriousnss of the variant.</p></li>
</ul>
<p>We first create base versions of the <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> background cohort input files, and then create slimmed versions of the background set (with only the overlapping variants for each input/predictive dataset) for each input run.</p>
<p>For the base versions of the background cohort, we create:</p>
<ul class="simple">
<li><p>A <code class="docutils literal notranslate"><span class="pre">.Consequence</span></code> file containing conservation scores per SNP</p></li>
<li><p>A VCF file containing the genetic data for the background set that overlaps with the information in the <code class="docutils literal notranslate"><span class="pre">.Consequence</span></code> file.</p></li>
</ul>
<p><strong>Step 1: Acquire variant information as a large temporary VCF file</strong>
We first create a large VCF file containing all possible variants.</p>
<p>In order to make the <code class="docutils literal notranslate"><span class="pre">.Consequence</span></code> file it is necessary to have a list of all variants of interest; their locations on the genome, and the possible variants at that location.</p>
<p>In the case of the 1000 Genomes project, we create a large temporary VCF file by downloading and concatenating the per chromosome VCF files (minus the meta data at the top for all bar the first, which is the same for all files).</p>
<p><strong>Step 2: Create <code class="docutils literal notranslate"><span class="pre">.Consequence</span></code> file</strong>
After obtaining the large VCF file, we then:</p>
<ul class="simple">
<li><p>trim the VCF file to contain only the necessary columns (first 10) to reduce the amount of memory required in the next step.</p></li>
<li><p>run these variants through the <a class="reference external" href="https://www.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor</a>, using the correct build.</p></li>
<li><p>grep for “missense”</p></li>
<li><p>run <code class="docutils literal notranslate"><span class="pre">snowflake</span> <span class="pre">create-consequence</span></code> to get a FATHMM score for each missense variant and create the <code class="docutils literal notranslate"><span class="pre">.Consequence</span></code> file.</p></li>
</ul>
<p><strong>Step 3: Create the base background VCF</strong>
Since the VCF file created in Step 1 is larger than needed by <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code>, we then reduce the temporary VCF file (created in Step 1) to only the proteins in the <code class="docutils literal notranslate"><span class="pre">.Consequence</span></code> file created in Step 2. Other variants will not be used by the predictor, and this substantially reduces the memory needed to run <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code>.</p>
<p><span class="xref myst">//</span>: # (TODO: Write about the diversity/limitations of the 2500 genomes project).
<span class="xref myst">//</span>: # (TODO: Write about the impact of the fact that the 2500 genomes project is whole genome sequencing, while the rest is mostly genotype data)
<span class="xref myst">//</span>: # (TODO: Put somewhere -after creating the VCF and <code class="docutils literal notranslate"><span class="pre">.Consequence</span></code> files for the input individuals, we create a slimmed version of the background cohort files)</p>
<div class="section" id="genome-project-background-files">
<h3><span class="section-number">4.2.3.1. </span>1000 Genome Project background files<a class="headerlink" href="#genome-project-background-files" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="snowflake-create-input">
<span id="id37"></span><h2><span class="section-number">4.2.4. </span><code class="docutils literal notranslate"><span class="pre">snowflake</span> <span class="pre">create_input</span></code><a class="headerlink" href="#snowflake-create-input" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="snowflake-preprocessing">
<h2><span class="section-number">4.2.5. </span><code class="docutils literal notranslate"><span class="pre">snowflake</span> <span class="pre">preprocessing</span></code><a class="headerlink" href="#snowflake-preprocessing" title="Permalink to this headline">¶</a></h2>
<div class="section" id="inputs">
<h3><span class="section-number">4.2.5.1. </span>Inputs<a class="headerlink" href="#inputs" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="id38">
<h3><span class="section-number">4.2.5.2. </span>Outputs<a class="headerlink" href="#id38" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="method">
<h3><span class="section-number">4.2.5.3. </span>Method<a class="headerlink" href="#method" title="Permalink to this headline">¶</a></h3>
<div class="section" id="dcgo">
<span id="id39"></span><h4><span class="section-number">4.2.5.3.1. </span>DcGO<a class="headerlink" href="#dcgo" title="Permalink to this headline">¶</a></h4>
<p>The aim of the Domain-centric Gene Ontology (DcGO)<a href="#id40"><span class="problematic" id="id41">:cite:`Fang2013-ms,Fang2013-ix`</span></a> tool is to give insight into uncharacterised or poorly characterised proteins by leveraging information about the content of their constituent protein domains.
It annotates domains and combinations of domains (supradomains) to phenotype terms from a variety of ontologies, including the Gene Ontology (GO), Mammalian Phenotype ontology (MP), Disease Ontology (DOID), Zebrafish ontology (ZFA).
Domain information comes from SUPERFAMILY, and annotations between (supra)domains and phenotype terms are made below a cut-off of FDR-adjusted statistical associations between the entities.
Using phenotypes from a range of species serves to make use of greater numbers of experiments, and therefore increases the number of little-known proteins across species that DcGO can make predictions about.</p>
</div>
</div>
<div class="section" id="variant-prioritisation">
<span id="variant-prioritisaton"></span><h3><span class="section-number">4.2.5.4. </span>Variant prioritisation<a class="headerlink" href="#variant-prioritisation" title="Permalink to this headline">¶</a></h3>
<p>To narrow down a long list of genes or variants to a shorter list of variants or genes which is more likely to be causal, variant prioritisation is used.</p>
<div class="section" id="fathmm">
<span id="id42"></span><h4><span class="section-number">4.2.5.4.1. </span>FATHMM<a class="headerlink" href="#fathmm" title="Permalink to this headline">¶</a></h4>
<p>Functional Analysis through Hidden Markov Models (FATHMM)<a href="#id43"><span class="problematic" id="id44">:cite:`Shihab2013-pk`</span></a> is a tool for predicting the functional effects of protein missense mutations using sequence conservation information (via HMMs), which can be (optionally) weighted by how likely a mutation in a protein/domain would be to lead to disease.
Weightings are calculated from the frequency of disease-associated and functionally neutral amino acid substitutions in protein domains from human variation databases (the Human Gene Mutation Database<a href="#id45"><span class="problematic" id="id46">:cite:`Stenson2009-jc`</span></a> and Uniprot-KB/Swiss-prot<a href="#id47"><span class="problematic" id="id48">:cite:`Pundir2016-ya`</span></a>).</p>
<p>Consequence files describing whether an amino acid results in a missense, nonsense or synonymous SNP must first be obtained by using Ensembl’s Variant Effect Predictor<a href="#id49"><span class="problematic" id="id50">:cite:`McLaren2016-di`</span></a> in order to create input to FATHMM.
FATHMM then calculates conservation scores which are a measure of the difference in amino acid probabilities for a SNP according to the HMM, i.e. between a wild type amino acid and it’s substitution.
A reduction in amino acid probabilities is interpreted as a prediction of deleteriousness, and the larger the reduction the greater the predicted harm.</p>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c04-snowflake"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="1-introduction.html" title="previous page"><span class="section-number">4.1. </span>Introduction</a>
    <a class='right-next' id="next-link" href="3-data.html" title="next page"><span class="section-number">4.3. </span>Snowflake input data</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>