
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>4.4. Preprocessing &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.e8f53015daec13862f6db5e763c41738.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="4.5. Considerations for Clustering SNPs" href="5-clustering-snps.html" />
    <link rel="prev" title="4.3. Creating Snowflake inputs" href="3-creating-inputs.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#combining-vcf-files-a-k-a-missing-snps-and-ambiguous-flips">
   4.4.1. Combining VCF files, a.k.a. missing SNPs and ambiguous flips
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#missing-snps-in-vcf-files">
     4.4.1.1. Missing SNPs in VCF files
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#ambiguous-flips">
     4.4.1.2. Ambiguous Flips
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="preprocessing">
<span id="snowflake-preprocessing"></span><h1><span class="section-number">4.4. </span>Preprocessing<a class="headerlink" href="#preprocessing" title="Permalink to this headline">¶</a></h1>
<p>In the preprocessing stage, the <code class="docutils literal notranslate"><span class="pre">snowflake</span> <span class="pre">preprocess</span></code> command looks at all the inputs together, in order to filter them only for the useful parts before running the predictor.</p>
<p>In particular, <code class="docutils literal notranslate"><span class="pre">snowflake</span> <span class="pre">preprocess</span></code> always calculates the following in all running modes:</p>
<ul class="simple">
<li><p>A list of SNPs associated with each term (<code class="docutils literal notranslate"><span class="pre">.snp</span></code> files)</p></li>
<li><p>Sets of equivalent terms, i.e. terms which have the exact same set of SNPs associated with them (<code class="docutils literal notranslate"><span class="pre">.polist</span></code> files)</p></li>
</ul>
<p>And it also calculates the following if an input cohort is provided:</p>
<ul class="simple">
<li><p>A list of overlapping SNPs between the background and input cohort</p></li>
<li><p>A combined VCF file containing only these snps, including dealing with ambiguous flips.</p></li>
</ul>
<p>In this section, I will run through and explain the preprocessing step for the 1000 genomes only (no input cohort) as this represents an approximation of the maximum number of SNPs that <code class="docutils literal notranslate"><span class="pre">snowflake</span></code> can predict on, since the 1000 genomes project uses <abbr title="Whole Genome Sequencing">WGS</abbr>.</p>
<!--
In this section, I will run through and explain the preprocessing step for:
1. The 1000 genomes only (no input cohort) as this represents an approximation of the maximum number of SNPs that `snowflake` can predict on, since the 1000 genomes project uses {abbr}`WGS (Whole Genome Sequencing)`.
2. The 1000 genomes with the ALSPAC input cohort, as this shows a more typical use of `snowflake`, where the inputs represent genotyped data.

## 1000 Genomes only: without input cohort
[//]: # (TODO: How do you run it? Etc)

### SNPs per term
[//]: # (TODO: Graph... explain these are the numbers of dimensions that we are looking through for a group of 1000 people)

### Redundant terms
[//]: # (TODO: Are the redundant terms always meaningful?)

<!--
+++

## 2500 Genomes and ALSPAC: with input cohort
[//]: # (TODO: How to run)
### Overlapping SNPs
### Ambiguous flips
### SNPs per term
### Redundant terms
--><div class="section" id="combining-vcf-files-a-k-a-missing-snps-and-ambiguous-flips">
<span id="combining-vcf-files"></span><h2><span class="section-number">4.4.1. </span>Combining VCF files, a.k.a. missing SNPs and ambiguous flips<a class="headerlink" href="#combining-vcf-files-a-k-a-missing-snps-and-ambiguous-flips" title="Permalink to this headline">¶</a></h2>
<p>Due to the cost, far more humans have been genotyped than have had their whole genomes sequenced.
Genotyped and <abbr title="Whole Genome Sequencing">WGS</abbr> data look similar once in a VCF file, but the data cannot necessarily be treated the same in both cases.</p>
<div class="section" id="missing-snps-in-vcf-files">
<span id="missing-snps-vcf"></span><h3><span class="section-number">4.4.1.1. </span>Missing SNPs in VCF files<a class="headerlink" href="#missing-snps-in-vcf-files" title="Permalink to this headline">¶</a></h3>
<p>Many VCF files only store the differences between individuals in the file, a SNP being missing from a VCF file does not necessarily mean that the original sequencing or genotyping didn’t record the calls at that position.</p>
<p>If combining two genotyped files, we would want to discard all SNPs that are not measured by both chips, but when combining a genotype VCF file with a WGS VCF file, we usually want to keep all SNPs from the genotyped VCF (since these locations will also have been sequenced by WGS).</p>
</div>
<div class="section" id="ambiguous-flips">
<span id="id1"></span><h3><span class="section-number">4.4.1.2. </span>Ambiguous Flips<a class="headerlink" href="#ambiguous-flips" title="Permalink to this headline">¶</a></h3>
<p>The majority of input data to the predictor is 23andMe data.
In testing earlier versions of Snowflake with the 2500G background and a cohort of 23andMe genomes, it became clear that for many phenotypes, the background was forming a separate cluster to the cohort.
This led to the realisation that there are 23andMe calls which had the opposite ratio of wild type:mutant than the 2500 genomes.
Some further reading revealed this to be a known problem<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id128"><span>179</span></a>]</span>, which may be due to ambiguous flips<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id129"><span>180</span></a>]</span>.</p>
<p>Implausible distributions of SNPs in the input cohort (given the background) are therefore discarded using a cutoff.</p>
<!--
### Example: ALSPAC and 2500G

### Example: CAGI5 and 2500G
--></div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c04-snowflake"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            



<div class='prev-next-bottom'>
    
    <div id="prev">
        <a class="left-prev" href="3-creating-inputs.html" title="previous page">
            <i class="prevnext-label fas fa-angle-left"></i>
            <div class="prevnext-info">
                <p class="prevnext-label">previous</p>
                <p class="prevnext-title"><span class="section-number">4.3. </span>Creating Snowflake inputs</p>
            </div>
        </a>
    </div>
     <div id="next">
        <a class="right-next" href="5-clustering-snps.html" title="next page">
            <div class="prevnext-info">
                <p class="prevnext-label">next</p>
                <p class="prevnext-title"><span class="section-number">4.5. </span>Considerations for Clustering SNPs</p>
            </div>
            <i class="prevnext-label fas fa-angle-right"></i>
        </a>
     </div>

</div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>