
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>4.6. Testing Snowflake on ALSPAC data &#8212; Phenotype from Genotype</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script >const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="4.7. Discussion" href="7-discussion.html" />
    <link rel="prev" title="4.5. Considerations for Clustering SNPs" href="5-clustering-snps.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How the link between genotype and phenotype is researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     4.6. Testing Snowflake on ALSPAC data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     4.7. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     5.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     5.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     5.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     5.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     5.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   6. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     6.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     6.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     6.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     6.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     6.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     6.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   7. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     7.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     7.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     7.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   8. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#motivation">
   4.6.1. Motivation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-alspac-cohort-study">
   4.6.2. The ALSPAC cohort study
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experiment-design">
   4.6.3. Experiment Design
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#choosing-test-phenotypes">
     4.6.3.1. Choosing test phenotypes
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#results">
   4.6.4. Results
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#phenotype-prediction">
     4.6.4.1. Phenotype prediction
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#variant-function">
     4.6.4.2. Variant function
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#discussion">
   4.6.5. Discussion
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#selection-of-phenotypes">
     4.6.5.1. Selection of phenotypes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#overlap-between-training-and-validation-data">
     4.6.5.2. Overlap between training and validation data
    </a>
   </li>
  </ul>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Testing Snowflake on ALSPAC data</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#motivation">
   4.6.1. Motivation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-alspac-cohort-study">
   4.6.2. The ALSPAC cohort study
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#experiment-design">
   4.6.3. Experiment Design
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#choosing-test-phenotypes">
     4.6.3.1. Choosing test phenotypes
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#results">
   4.6.4. Results
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#phenotype-prediction">
     4.6.4.1. Phenotype prediction
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#variant-function">
     4.6.4.2. Variant function
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#discussion">
   4.6.5. Discussion
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#selection-of-phenotypes">
     4.6.5.1. Selection of phenotypes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#overlap-between-training-and-validation-data">
     4.6.5.2. Overlap between training and validation data
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="section" id="testing-snowflake-on-alspac-data">
<span id="alspac-intro"></span><h1><span class="section-number">4.6. </span>Testing Snowflake on ALSPAC data<a class="headerlink" href="#testing-snowflake-on-alspac-data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="motivation">
<span id="alspac-motivation"></span><h2><span class="section-number">4.6.1. </span>Motivation<a class="headerlink" href="#motivation" title="Permalink to this headline">¶</a></h2>
<p>In order to test Snowflake, I needed a data set that had a wealth of phenotype and genotype information.</p>
</div>
<div class="section" id="the-alspac-cohort-study">
<span id="alspac-study"></span><h2><span class="section-number">4.6.2. </span>The ALSPAC cohort study<a class="headerlink" href="#the-alspac-cohort-study" title="Permalink to this headline">¶</a></h2>
<p>The Avon Longitudinal Study of Parents and Children, ALSPAC<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id146"><span>86</span></a>]</span> is a cohort of over 14,000 families from the Avon area with children born in 1991-1992.
It is also known as “the Children of the 90s” study.
Many of these families continue to be part of the study to this day, including some of their own children through an extension of the project: children of the children of the 90s (COCO90s).</p>
<p>A wealth of phenotype information (over 80,000 variables) has been collected from these families over the years, through a series of voluntary surveys and clinics, including genotyping of over 9000 children using 23andMe.</p>
<p>ALSPAC’s phenotype information, while extensive, is not mapped to phenotype terms in ontologies.
All data held by ALSPAC can be searched for in the <a class="reference external" href="http://www.bristol.ac.uk/alspac/researchers/our-data/">ALSPAC variable catalogue</a>, after which it can then requested per variable or data type.
At the time of writing, the cohort is around 30 years old, meaning that there is little information about phenotypes that manifest later in life, for example Alzheimer’s or heart disease.
Many phenotype terms may not have any measurements, and there may be many variables associated with some others.</p>
</div>
<div class="section" id="experiment-design">
<span id="alspac-exp-design"></span><h2><span class="section-number">4.6.3. </span>Experiment Design<a class="headerlink" href="#experiment-design" title="Permalink to this headline">¶</a></h2>
<p>Due to the identifiable nature of the data, our ethics application did not allow us to access many different phenotypes to perform a cross-phenotype validation of the predictor.
Instead, we were granted access to all of the genotype data only first, then allowed to request a small number of phenotypes of interest after running Snowflake.</p>
<div class="section" id="choosing-test-phenotypes">
<span id="choosing-phenotypes"></span><h3><span class="section-number">4.6.3.1. </span>Choosing test phenotypes<a class="headerlink" href="#choosing-test-phenotypes" title="Permalink to this headline">¶</a></h3>
<p>I created a shortlist of phenotypes of interest by first restricting the set of scores to phenotypes for which Snowflake makes a prediction within the ALSPAC cohort, then ordering this list by the <a class="reference internal" href="2-snowflake-algorithm.html#id20"><span class="std std-ref">phenotype confidence score</span></a>, to ensure that Snowflake could give confident predictions for phenotypes that were requested.
I then mapped these to ALSPAC phenotypes by searching the ALSPAC variable catalogue.
This resulted in the four : <code class="docutils literal notranslate"><span class="pre">MP:0001501</span></code> <em>Abnormal Sleep Pattern</em> (measured using <code class="docutils literal notranslate"><span class="pre">FJCI250</span></code> <em>Sleep symptom score</em>), <code class="docutils literal notranslate"><span class="pre">MP:0001933</span></code> <em>Abnormal litter size</em> (measured by <code class="docutils literal notranslate"><span class="pre">mz010a</span></code> <em>Pregnancy size</em>), <code class="docutils literal notranslate"><span class="pre">MESH:D001259</span></code> <em>Ataxia</em> (measured by <code class="docutils literal notranslate"><span class="pre">kw2030</span></code> <em>Child ever thought to have a problem with clumsiness/coordination</em>), and <code class="docutils literal notranslate"><span class="pre">HP:0001249</span></code> <em>Intelligence/intellectual disability</em> (measured by <code class="docutils literal notranslate"><span class="pre">f8ws150</span></code> <em>Child had special needs</em>).</p>
</div>
</div>
<div class="section" id="results">
<span id="alspac-discussion"></span><h2><span class="section-number">4.6.4. </span>Results<a class="headerlink" href="#results" title="Permalink to this headline">¶</a></h2>
<div class="section" id="phenotype-prediction">
<h3><span class="section-number">4.6.4.1. </span>Phenotype prediction<a class="headerlink" href="#phenotype-prediction" title="Permalink to this headline">¶</a></h3>
<p>Unfortunately, the results of phenotype prediction with Snowflake on ALSPAC data show that snowflake cannot currently be used to accurately predict the following phenotypes in the ALSPAC cohort: <code class="docutils literal notranslate"><span class="pre">MP:0001501</span></code> <em>Abnormal Sleep Pattern</em> (measured using <code class="docutils literal notranslate"><span class="pre">FJCI250</span></code> <em>Sleep symptom score</em>), <code class="docutils literal notranslate"><span class="pre">MP:0001933</span></code> <em>Abnormal litter size</em> (measured by <code class="docutils literal notranslate"><span class="pre">mz010a</span></code> <em>Pregnancy size</em>), <code class="docutils literal notranslate"><span class="pre">MESH:D001259</span></code> <em>Ataxia</em> (measured by <code class="docutils literal notranslate"><span class="pre">kw2030</span></code> <em>Child ever thought to have a problem with clumsiness/coordination</em>), and <code class="docutils literal notranslate"><span class="pre">HP:0001249</span></code> <em>Intelligence/intellectual disability</em> (measured by <code class="docutils literal notranslate"><span class="pre">f8ws150</span></code> <em>Child had special needs</em>).
These were assessed by bootstrapping the <span class="math notranslate nohighlight">\(F_{max}\)</span> scores for each phenotype, which gave very low f_max scores (&lt;0.1) and p&gt;0.05 for each phenotype.</p>
<p>Meanwhile other top-scoring phenotypes could not be tested since they did not map well to available ALSPAC phenotypes, e.g. <code class="docutils literal notranslate"><span class="pre">HP:0007703</span></code> <em>Abnormality of retinal pigmentation</em> and <code class="docutils literal notranslate"><span class="pre">HP:0001120</span></code> <em>abnormal corneal size</em>.</p>
</div>
<div class="section" id="variant-function">
<h3><span class="section-number">4.6.4.2. </span>Variant function<a class="headerlink" href="#variant-function" title="Permalink to this headline">¶</a></h3>
<p>In addition to testing the test phenotypes, it’s possible to look at highest-scoring SNPs (i.e. variants present in people with the highest snowflake “phenotype score”) for all phenotypes (even those we did not have phenotype data for).
In this way snowflake can be used as a kind of variant prioritisation tool.</p>
<p>These showed some reassuring results:</p>
<ul class="simple">
<li><p><strong>1. Rediscovering known SNPs</strong>: The highest-scoring phenotype in the ALSPAC study was <em>Abnormal fat cell morphology</em>. It’s highest scoring SNP <code class="docutils literal notranslate"><span class="pre">rs6659176</span></code> is known (via GWAS) to be associated with obesity<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id15"><span>198</span></a>]</span>.</p></li>
<li><p><strong>2. making plausible cross-species variant function predictions:</strong> One SNP, <code class="docutils literal notranslate"><span class="pre">rs2287780</span></code>, was predicted to be responsible for the phenotype of abnormal fetal growth. It is listed in SNPedia<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id24"><span>199</span></a>]</span> as being linked with vitamin B12 and folate production, which are well-known to be important for
fetal development. This SNP was flagged up by an ontology term associated with mouse genetics (“abnormal litter size”), meaning it is possible to find candidate human SNP/phenotype associations using information about other species.</p></li>
<li><p><strong>3. making plausible multi-SNP trait predictions</strong> a combination of 2 SNPs is required for the phenotype (ataxia – a disfunction of the central nervous system) to be predicted. One predicted (<code class="docutils literal notranslate"><span class="pre">rs13436090</span></code>) has been associated with autosomal dominant cerebellar ataxia (ADCA)<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id23"><span>200</span></a>]</span>. The second SNP (<code class="docutils literal notranslate"><span class="pre">rs2269961</span></code>) is a new candidate SNP, a protein from which (Tocopherol-associated protein 2) is thought to be responsible for transporting vitamin E (a deficiency of which can cause spinocerebellar ataxia<span id="id5">[<a class="reference internal" href="../cz-end-matter/reference.html#id22"><span>201</span></a>]</span>).</p></li>
</ul>
</div>
</div>
<div class="section" id="discussion">
<h2><span class="section-number">4.6.5. </span>Discussion<a class="headerlink" href="#discussion" title="Permalink to this headline">¶</a></h2>
<p id="alspac-phenotype-problems">Snowlflake’s phenotype prediction results were disappointing but could have a number of possible interpretations.</p>
<div class="section" id="selection-of-phenotypes">
<h3><span class="section-number">4.6.5.1. </span>Selection of phenotypes<a class="headerlink" href="#selection-of-phenotypes" title="Permalink to this headline">¶</a></h3>
<p>For valid ethical reasons, it was only possible to request a small number of phenotypes from ALSPAC.
Snowflake isn’t expected to work for all phenotypes, since we know there are many other mechanisms (some not even genetic) behind phenotype besides only the disruption of proteins through missense SNPs in protein-coding genes.
The fact that Snowflake was not successful in predicting any of the requested phenotypes could be an indication that none of these phenotypes has this mechanism behind it.
Alternatively, it could be an indication that Snowflake is not a successful method for predicting phenotypes even in these cases, and there are many reasons why that could be the case since Snowflake relies on a lot of other pieces of research and software to function.</p>
<p>In selecting phenotypes, I chose phenotypes where (1) Snowflake was expected to make a <a class="reference internal" href="2-snowflake-algorithm.html#id20"><span class="std std-ref">confident</span></a> prediction and (2) ALSPAC data could be used to validate this prediction.
These two restrictions narrowed down the phenotypes considerably and the majority of the top-predicted phenotype terms (Abnormal Fat Cell Morphology, Abnormal Fetal Development) did not map cleanly to ALSPAC phenotypes.</p>
<p>Since I chose phenotypes primarily by looking at the distribution of scores for Snowflake, our lack of promising results could also be an indication that the phenotype-score (finding interesting distributions of phenotypes) is unsuccessful.</p>
<p>A more interesting test of Snowflake’s abilities might be to choose phenotypes with high heritability, and specifically high missing heritability.</p>
<p>I also regret not considering the ethical implications of the choices of phenotypes to predict.</p>
</div>
<div class="section" id="overlap-between-training-and-validation-data">
<span id="alspac-overlap-problem"></span><h3><span class="section-number">4.6.5.2. </span>Overlap between training and validation data<a class="headerlink" href="#overlap-between-training-and-validation-data" title="Permalink to this headline">¶</a></h3>
<p>ALSPAC is a popular cohort study for use in GWAS analyses.
As such, many links between genotype loci and phenotypes have been published from this data set, for example loci associated with birthweight<span id="id6">[<a class="reference internal" href="../cz-end-matter/reference.html#id280"><span>202</span></a>]</span>, asthma and allergies<span id="id7">[<a class="reference internal" href="../cz-end-matter/reference.html#id276"><span>203</span></a>]</span>, autism spectrum disorders<span id="id8">[<a class="reference internal" href="../cz-end-matter/reference.html#id275"><span>204</span></a>]</span>, problematic peer relationships<span id="id9">[<a class="reference internal" href="../cz-end-matter/reference.html#id278"><span>205</span></a>]</span>, and lung function<span id="id10">[<a class="reference internal" href="../cz-end-matter/reference.html#id277"><span>206</span></a>]</span>.
This data informs GOA genotype-phenotype annotations, and through dcGO this data could already be present in Snowflake, meaning that there may be in some sense, overlap between the training and validation data, which could overestimate the accuracy of the predictor on unseen data<span id="id11">[<a class="reference internal" href="../cz-end-matter/reference.html#id279"><span>207</span></a>]</span>.</p>
<p>For this reason, I do not think that it is sensible to try any further to use ALSPAC to validate or test Snowflake.</p>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c04-snowflake"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="5-clustering-snps.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">4.5. </span>Considerations for Clustering SNPs</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="7-discussion.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">4.7. </span>Discussion</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Natalie Zelenka<br/>
  
      &copy; Copyright 2020.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>