
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>3.1. From genotype to phenotype: what is measured &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="3.2. How large computational biology datasets are used" href="2-comp-bio-methods.html" />
    <link rel="prev" title="3. Computational Biology Background" href="0-index.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Linking genotype and phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     3.1. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-comp-bio-methods.html">
     3.2. How large computational biology datasets are used
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-bias.html">
     3.3. Sources of bias in computational biology datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-summary.html">
     3.4. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-combining/0-index.html">
   5. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/1-background.html">
     5.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/2-data.html">
     5.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/3-data-wrangling.html">
     5.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html">
     5.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/5-batch-correction.html">
     5.5. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html">
     5.6. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/7-discussion.html">
     5.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/8-future-work.html">
     5.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-data.html">
     6.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-methods.html">
     6.3. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/4-results.html">
     6.4. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-discussion.html">
     6.5. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-future-work.html">
     6.6. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#introduction">
   3.1.1. Introduction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#dna">
   3.1.2. DNA
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#whole-genome">
     3.1.2.1. Whole genome
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#sequencing">
       3.1.2.1.1. Sequencing
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#assembly">
       3.1.2.1.2. Assembly
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#the-human-reference-genome">
       3.1.2.1.3. The human reference genome
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#whole-genome-sequencing-of-individuals">
       3.1.2.1.4. Whole Genome Sequencing of individuals
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genes">
     3.1.2.2. Genes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#regulatory-features">
     3.1.2.3. Regulatory features
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#variants">
     3.1.2.4. Variants
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#microarrays">
       3.1.2.4.1. Microarrays
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#variant-databases">
       3.1.2.4.2. Variant databases
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#rna">
   3.1.3. RNA
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#rna-seq">
     3.1.3.1. RNA-seq
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#differential-expression-versus-baseline">
     3.1.3.2. Differential expression versus baseline
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#gene-expression-data-pipeline">
     3.1.3.3. Gene expression data pipeline
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#quality-control">
       3.1.3.3.1. Quality Control
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#normalisation-within-sample-normalisation">
       3.1.3.3.2. Normalisation - within-sample normalisation
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#normalisation-between-sample">
       3.1.3.3.3. Normalisation - between-sample
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#proteins">
   3.1.4. Proteins
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#protein-structures">
     3.1.4.1. Protein Structures
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#protein-classification">
     3.1.4.2. Protein classification
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#blast">
       3.1.4.2.1. BLAST
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#scop">
       3.1.4.2.2. SCOP
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#superfamily">
       3.1.4.2.3. SUPERFAMILY
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#phenotypes">
   3.1.5. Phenotypes
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#a-word-on-ontologies">
     3.1.5.1. A word on ontologies
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#gene-ontology">
       3.1.5.1.1. Gene Ontology
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="from-genotype-to-phenotype-what-is-measured">
<h1><span class="section-number">3.1. </span>From genotype to phenotype: what is measured<a class="headerlink" href="#from-genotype-to-phenotype-what-is-measured" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2><span class="section-number">3.1.1. </span>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<div class="figure align-default" id="linneaus-ehret">
<a class="reference internal image-reference" href="../_images/linneaus_ehret.png"><img alt="../_images/linneaus_ehret.png" src="../_images/linneaus_ehret.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3.1 </span><span class="caption-text">Carl Linneaus developed a system of classifying plants, animals and minerals, including plant classification based on their number of stamens<a class="bibtex reference internal" href="../end-matter/reference.html#blunt2001-wr" id="id1">[4]</a>. The left image is a key to this classification system taken from his book, while the right image is a depiction of how the system works, drawn by botanist George Ehret<a class="bibtex reference internal" href="../end-matter/reference.html#ehret1748-bx" id="id2">[5]</a>.</span><a class="headerlink" href="#linneaus-ehret" title="Permalink to this image">¶</a></p>
</div>
<div class="margin sidebar" id="linneus-racist">
<p class="sidebar-title">Linneus and scientific racism</p>
<p>Linneus’ classifications included a racist hierarchical classification of human beings<a class="bibtex reference internal" href="../end-matter/reference.html#charmantier-undated-wy" id="id3">[6]</a>.</p>
</div>
<p>Cataloguing and classifying has been a successful scientific endeavour in other disciplines (e.g. the periodic table), but it’s a cornerstone of biology.
Biological classification dates back to the Linnaean taxonomy from the mid 1700s (see <a class="reference internal" href="#linneaus-ehret"><span class="std std-numref">Fig. 3.1</span></a>), which described species, their features, and the relationships between them<a class="bibtex reference internal" href="../end-matter/reference.html#jensen2010-fk" id="id4">[7]</a>.
The work contains some <a class="reference internal" href="#linneus-racist"><span class="std std-ref">unforgivable, hateful ideas</span></a>.
Nonetheless the idea of measuring and connecting the biological world also birthed an enduring tradition of classification in biology.</p>
<p>Modern biology continues in this tradition of classification, cataloguing biology in ever more (molecular) detail: cells, genes, transcripts, proteins, and pathways.
While in other fields, data inaccessibility is a major barrier to reproducible research, this is the field that had an online database system that remote computers could access in the 1960s!
So, the catalogued information populates freely available databases, vocabularies, and annotations, creating controlled and shared vocabularies that fuel computational methodologies.
We will now delve into the details of some of these datasets, looking first at <a class="reference internal" href="#dna-measurements"><span class="std std-ref">DNA</span></a>, then <a class="reference internal" href="#rna-measurements"><span class="std std-ref">RNA</span></a>, then <a class="reference internal" href="#protein-measurements"><span class="std std-ref">proteins</span></a>, then <a class="reference internal" href="#phenotype-measurements"><span class="std std-ref">phenotypes</span></a>.</p>
</div>
<div class="section" id="dna">
<span id="dna-measurements"></span><h2><span class="section-number">3.1.2. </span>DNA<a class="headerlink" href="#dna" title="Permalink to this headline">¶</a></h2>
<p>In the <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html#what-is-dna"><span class="std std-ref">previous Chapter</span></a>, we looked at what DNA is and how that links to phenotype.
Now we’re going to look at the details of how this is measured and stored in computational biology.
Since smaller pieces of DNA are generally talked about in relation to the whole genome, we’ll go from big to small, beginning with whole genomes and moving through to individual SNPs.</p>
<div class="section" id="whole-genome">
<h3><span class="section-number">3.1.2.1. </span>Whole genome<a class="headerlink" href="#whole-genome" title="Permalink to this headline">¶</a></h3>
<p>The whole genome is all the genetic material of an organism. In humans, this means all chromosomal and mitochondrial DNA, whether or not it is transcribed into RNA, or translated into proteins.
No sequencing technology can read whole chromosomes end to end: all work by reading shorter lengths of DNA (<em>reads</em>).</p>
<div class="section" id="sequencing">
<span id="sequencing-technology"></span><h4><span class="section-number">3.1.2.1.1. </span>Sequencing<a class="headerlink" href="#sequencing" title="Permalink to this headline">¶</a></h4>
<p>From the late 1970’s until the mid 2000s, <em>Sanger sequencing</em> was the most popular sequencing technology, although it underwent various improvements over this timescale.
In Sanger sequencing (and other first-generation methods), reads of around 800bp are sequenced, one at a time.
The human genome project sequenced the first human genome using this method<a class="bibtex reference internal" href="../end-matter/reference.html#venter2001-wn" id="id5">[2]</a>, and it’s still used in some circumstances, for example validating next generation sequencing.</p>
<p>Second, or <em>next generation sequencing</em> (NGS), also referred to as high-throughput sequencing, is a catch-all term for the faster and cheaper sequencing technologies which replaced the previously used Sanger sequencing.
A feature that is common to NGS methods is that many shorter reads (around 100bp, exact numbers depending on the specific technology) are sequenced in parallel.
The process is massively parallel: millions to billions of short sequences can be read at a time.
This is a huge factor in making NGS much faster (and therefore cheaper) than Sanger sequencing.
In turn, this speed and cheapness means that more repeats can be sequenced, increasing the overall accuracy of NGS over Sanger (despite the accuracy of each individual read being generally lower).</p>
<p>There are now also third generation sequencing technologies that allow much longer reads to be sequenced.</p>
</div>
<div class="section" id="assembly">
<span id="assembly-and-alignment"></span><h4><span class="section-number">3.1.2.1.2. </span>Assembly<a class="headerlink" href="#assembly" title="Permalink to this headline">¶</a></h4>
<p>Whichever <a class="reference internal" href="#sequencing-technology"><span class="std std-ref">technology</span></a> is used, DNA is sequenced in smalls sections, which must then be <em>assembled</em> to make the full genome, organised into chromosomes.</p>
<div class="figure align-default" id="alignment">
<a class="reference internal image-reference" href="../_images/alignment.png"><img alt="../_images/alignment.png" src="../_images/alignment.png" style="height: 240px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3.2 </span><span class="caption-text">Image showing how RNA-Seq reads are mapped to the genome (image from Advancing RNA-Seq Analysis <a class="bibtex reference internal" href="../end-matter/reference.html#haas2010-lm" id="id6">[8]</a>). A similar process is followed for DNA.
``` ``</span><a class="headerlink" href="#alignment" title="Permalink to this image">¶</a></p>
<div class="legend">
<p>The short reads that are the output of initial sequencing must be assembled to create longer sequences of DNA, whether that’s genes or whole genomes.
This is done by aligning reads to one another and (if available) to an existing reference sequence.
This of course requires the reads to overlap, so longer and more numerous reads make this task easier.</p>
<p>The current estimate for raw sequencing accuracy of an individual NGS read is around 0.24%<a class="bibtex reference internal" href="../end-matter/reference.html#pfeiffer2018-kt" id="id7">[9]</a>, meaning that on average one base pair will be incorrect for a 500pb read.
Multiple repeats are therefore required to obtain a more accurate measurement of the assembled sequence, which is further necessary since there are many repeated sequences (perhaps over two thirds of the human genome<a class="bibtex reference internal" href="../end-matter/reference.html#de-koning2011-ac" id="id8">[10]</a>).
The depth (or coverage) for a nucleotide is the number of reads that overlap that nucleotide. Similarly, the average depth of a sequence can be calculated.</p>
<p>After assembly, even in the most complete genomes, we are still left with some sequences that could not be placed, and some parts of the genome that we still don’t know about.</p>
</div>
</div>
</div>
<div class="section" id="the-human-reference-genome">
<span id="human-references"></span><h4><span class="section-number">3.1.2.1.3. </span>The human reference genome<a class="headerlink" href="#the-human-reference-genome" title="Permalink to this headline">¶</a></h4>
<p>As <a class="reference internal" href="../c02-biology-background/3-more-dna.html#genomes"><span class="std std-ref">previously mentioned</span></a>, reference genomes are designed to represent whole organisms: these genomes aim to have the most common allele at any given nucleotide, and are then annotated at positions where individuals differ.</p>
<p><strong>Builds and patches:</strong>
As more genomes are sequenced, more information comes to light about the nature of the human genome.
For example, some locations are revealed to be likely sequencing artifacts.
New major versions of genomes are released every few years to fix these changes.
These versions are called <em>builds</em>.
Between builds and patches, sequences may be added, removed, or moved to different locations on chromosomes.</p>
<p id="different-builds"><strong>Differences between GRC and UCSC builds:</strong>
Different versions of the builds are released by the Genome Reference Consortium (GRC) and the University of California Santa Cruz (UCSC) Genomics Institute.
Table <a class="reference internal" href="#build-table"><span class="std std-numref">Table 3.1</span></a> shows information about the most recent human reference builds, taken from the UCSC website<a class="bibtex reference internal" href="../end-matter/reference.html#noauthor-undated-ci" id="id9">[11]</a>.
For example, <code class="docutils literal notranslate"><span class="pre">hg19</span></code> (<strong>h</strong>uman <strong>g</strong>enome build <strong>19</strong>), is largely equivalent to <code class="docutils literal notranslate"><span class="pre">GRCh37</span></code> (<strong>G</strong>enome <strong>R</strong>eference <strong>C</strong>onsortium <strong>h</strong>uman build <strong>37</strong>).
These are generally used interchangeably by researchers, but there are some differences between them.
These differences including formatting differences (storing chromosome as integers rather than strings like <code class="docutils literal notranslate"><span class="pre">chr1</span></code>), the inclusion of mitochondrial DNA, as well as small numbers of differences of the locations of some variants on some chromosomes <a class="bibtex reference internal" href="../end-matter/reference.html#gatk-team2020-au" id="id10">[12]</a>.</p>
<table class="table" id="build-table">
<caption><span class="caption-number">Table 3.1 </span><span class="caption-text">Table showing human reference genome builds</span><a class="headerlink" href="#build-table" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Release name</p></th>
<th class="head"><p>UCSC</p></th>
<th class="head"><p>Release date</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>GRCh38</p></td>
<td><p>hg38</p></td>
<td><p>December 2013</p></td>
</tr>
<tr class="row-odd"><td><p>GRCh37</p></td>
<td><p>hg19</p></td>
<td><p>February 2009</p></td>
</tr>
<tr class="row-even"><td><p>NCBI Build 36.1</p></td>
<td><p>hg18</p></td>
<td><p>March 2006</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="whole-genome-sequencing-of-individuals">
<h4><span class="section-number">3.1.2.1.4. </span>Whole Genome Sequencing of individuals<a class="headerlink" href="#whole-genome-sequencing-of-individuals" title="Permalink to this headline">¶</a></h4>
<p>When individual humans have their whole genomes sequenced, this is compared to the human reference genome.
The alleles at each location are commonly stored in Variant Call Format (VCF) files.
These describe the locations on the genome of variations between individuals, given by chromosome, position, variant identifiers, and the variation between a given number of individuals.</p>
<p><strong>Position and BED formats</strong></p>
</div>
</div>
<div class="section" id="genes">
<h3><span class="section-number">3.1.2.2. </span>Genes<a class="headerlink" href="#genes" title="Permalink to this headline">¶</a></h3>
<p>There can be disagreements about the locations of genes on the genome, and popular databases of genes fundamentally disagree on this and on the number of genes<a class="bibtex reference internal" href="../end-matter/reference.html#salzberg2018-yc" id="id11">[13]</a>.
Gene names and symbols change over time, and can be difficult to map between for this reason.</p>
</div>
<div class="section" id="regulatory-features">
<h3><span class="section-number">3.1.2.3. </span>Regulatory features<a class="headerlink" href="#regulatory-features" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="variants">
<h3><span class="section-number">3.1.2.4. </span>Variants<a class="headerlink" href="#variants" title="Permalink to this headline">¶</a></h3>
<div class="section" id="microarrays">
<h4><span class="section-number">3.1.2.4.1. </span>Microarrays<a class="headerlink" href="#microarrays" title="Permalink to this headline">¶</a></h4>
<p>Through the 1970s into the early 2000s, DNA arrays/microarrays developed alongside sequencing as a way of measuring the presence of previously sequenced DNA in new samples.
These arrays contain pre-chosen fragments of DNA (probes) arranged in spots, with each spot containing many copies of the probe, on a solid surface, e.g. glass, silicon or plastic.
The probes consist of single strands of DNA, and arrays operate on the principle that the complementary DNA from the sample will bind tightly to it.</p>
<p>These arrays were originally macro-sized, one of the first being 26 × 38 cm and containing 144 probes<a class="bibtex reference internal" href="../end-matter/reference.html#bumgarner2013-hg" id="id12">[14]</a>, but are now on small chips, which can contain up to millions of probes.</p>
<p>Arrays were extremely popular for measuring gene expression, but this technology has largely been superseded by the more accurate and comprehensive RNA-seq.
However, microarrays are still commonly used by companies like 23andMe for genotyping an individual (measuring specific alleles).</p>
</div>
<div class="section" id="variant-databases">
<h4><span class="section-number">3.1.2.4.2. </span>Variant databases<a class="headerlink" href="#variant-databases" title="Permalink to this headline">¶</a></h4>
<p>Databases like dbSNP, clinVar, and SNPedia contain information about the location of SNVs, their possible alleles, and their association to diseases.
These are often used as part of a variant prioritisation pipeline, as well as by individuals who are investigating their own SNVs.</p>
<p>The NCBI’s dbSNP<a class="bibtex reference internal" href="../end-matter/reference.html#sherry2001-nm" id="id13">[15]</a> is the world’s largest database of nucleotide variations, including their locations, possible alleles, and associations to diseases.
It contains information from ten organisms (including human) and has information on indels and short tandem repeats in addition to SNVs. It was designed to store the annotations that are outputted from GWAS studies.
Anyone can submit their findings about variants, and they must indicate what sort of evidence they have for the association.
The database gives SNVs unique identifiers (Reference SNP cluster IDs, a.k.a. RSIDs) of the form rs###, which are used by many other resources.</p>
</div>
</div>
</div>
<div class="section" id="rna">
<span id="rna-measurements"></span><h2><span class="section-number">3.1.3. </span>RNA<a class="headerlink" href="#rna" title="Permalink to this headline">¶</a></h2>
<p>DNA can’t tell us the whole story. On it’s own it can’t tell us what we’d like to know about a gene’s function or a person’s traits. Nonetheless, I’d like to spell out why that is for those who do not have that background. The central dogma of molecular biology can be paraphrased as “DNA makes RNA makes proteins”. So, if we know what a person’s DNA is, then why would we also want to know about their RNA?</p>
<p>The answer boils down to the effect of the environment. “DNA makes RNA makes proteins” is shorthand for “DNA interacts with the environment to make RNA, and RNA interacts with the environment to make proteins”. The environment, of the cell, and of the individual, and even of modifications to the DNA molecules, is the reason why DNA alone is not enough to explain how we function. DNA is the collection of blueprints for what can be made, but the environment decides what goes into production; which RNA and proteins are made and how much.</p>
<p>In any given individual, the same DNA is present in their liver cells, skin cells, and neurons. The difference in how these cells/tissues look and function is due to differences in which proteins are actually being made at any given time. Gene expression is a way of measuring how the environment of the cell is affecting the process of making proteins from DNA.</p>
<p>Gene expression data is used to understand the function of genes, to identify housekeeping genes, to re-engineer gene regulatory networks, and more. This kind of insight can not be gained from looking at DNA alone.</p>
<div class="section" id="rna-seq">
<h3><span class="section-number">3.1.3.1. </span>RNA-seq<a class="headerlink" href="#rna-seq" title="Permalink to this headline">¶</a></h3>
<p>NGS can be used for sequencing either DNA or RNA (known as RNA-seq when applied to the whole transcriptome).</p>
<p>While (NGS) DNA-sequencing and RNA-seq can use the same underlying NGS technologies, there exist some notable differences. For example, RNA is reverse-transcribed into strands of complementary DNA, before being sequenced, since sequencing DNA is currently easier than sequencing RNA. RNA-seq is used much less often for de novo sequencing, and is generally mapped to a reference sequence.</p>
<p>Since transcription is dependent on time, tissue, location, cell, etc, RNA-seq experiments are also dependent on all of these conditions. Furthermore, they are sensitive to differences in laboratory conditions and experimental design, creating artefacts in the resulting data known as batch effects.</p>
<p>One of the most popular measures of gene expression, is the measure of how much RNA is in a cell at a given time.</p>
</div>
<div class="section" id="differential-expression-versus-baseline">
<h3><span class="section-number">3.1.3.2. </span>Differential expression versus baseline<a class="headerlink" href="#differential-expression-versus-baseline" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="gene-expression-data-pipeline">
<h3><span class="section-number">3.1.3.3. </span>Gene expression data pipeline<a class="headerlink" href="#gene-expression-data-pipeline" title="Permalink to this headline">¶</a></h3>
<p>Whether gene expression data gathered is gathered using microarrays or RNA-Seq, it goes through a substantial pipeline before it is used in a statistical analysis, or made available as a resource.</p>
<div class="section" id="quality-control">
<h4><span class="section-number">3.1.3.3.1. </span>Quality Control<a class="headerlink" href="#quality-control" title="Permalink to this headline">¶</a></h4>
<p>Before RNA-Seq data undergoes alignment, it undergoes quality control. This involves comparing sequencing parameters to a data set of known accuracy<a class="bibtex reference internal" href="../end-matter/reference.html#illumina2014-nd" id="id14">[16]</a> and is usually done as part of the sequencing.</p>
</div>
<div class="section" id="normalisation-within-sample-normalisation">
<h4><span class="section-number">3.1.3.3.2. </span>Normalisation - within-sample normalisation<a class="headerlink" href="#normalisation-within-sample-normalisation" title="Permalink to this headline">¶</a></h4>
<p>Within-sample normalisation methods are designed to account for sequencing depth and gene length so that gene expression values from the same sample (e.g. different replicates) can be more easily compared. Longer genes will have more reads mapped to them for an equal level of expression, so RNA-seq will report more counts. Similarly, without normalising, samples with greater sequencing depth will have higher counts for an equal level of expression.</p>
<p>RPKM/FPKM (Reads/Fragments Per Kilobase Million) and TPM (Tags Per Million) are the three major normalisation techniques used for this purpose. In RPKM and FPKM, counts are first normalised for sequencing depth, and then for gene length. This means that they are suitable for comparing within a sample (e.g. between replicates. TPM, however performs the same steps in the opposite order, which has the desirable effect of ensuring that columns corresponding to TPM normalised samples sum to the same number. This means that TPM gives us a measure of relative abundance; we can compare across samples which proportion of counts are from each gene. For this reason, TPM is now generally preferred over RPKM/FPKM<a class="bibtex reference internal" href="../end-matter/reference.html#wagner2012-ac" id="id15">[17]</a><a class="bibtex reference internal" href="../end-matter/reference.html#pimentel2014-xm" id="id16">[18]</a>.</p>
</div>
<div class="section" id="normalisation-between-sample">
<span id="rna-normalisation"></span><h4><span class="section-number">3.1.3.3.3. </span>Normalisation - between-sample<a class="headerlink" href="#normalisation-between-sample" title="Permalink to this headline">¶</a></h4>
<p>While TPM gives us a measure of relative abundance, it does not give us a measure of absolute abundance. One outlying gene which is highly expressed will have the effect of making all other genes look relatively less expressed. We might expect this to occur, particularly when samples are under different conditions (e.g. disease/treatment). Between-sample normalisation methods are designed to counter this issue, and enable researchers to compare different samples.</p>
<p>These methods adjust counts to reduce the impact of outlying expression values. Examples include scale normalisation methods like TMM (used in edgeR<a class="bibtex reference internal" href="../end-matter/reference.html#robinson2010-kl" id="id17">[19]</a>), the Log Geometric Mean (used in DESeq2<a class="bibtex reference internal" href="../end-matter/reference.html#robinson2010-kl" id="id18">[19]</a><a class="bibtex reference internal" href="../end-matter/reference.html#love2014-vx" id="id19">[20]</a>), and quantile normalisation (giving samples the same distribution of counts).</p>
</div>
</div>
</div>
<div class="section" id="proteins">
<span id="protein-measurements"></span><h2><span class="section-number">3.1.4. </span>Proteins<a class="headerlink" href="#proteins" title="Permalink to this headline">¶</a></h2>
<div class="section" id="protein-structures">
<h3><span class="section-number">3.1.4.1. </span>Protein Structures<a class="headerlink" href="#protein-structures" title="Permalink to this headline">¶</a></h3>
<p>In 1969, Margaret Dayhoff created the first bioinformatics database to store protein structures imaged using X-ray crystallography, related to her publication of Atlas of Protein Sequence and Structure<a class="bibtex reference internal" href="../end-matter/reference.html#hersh1967-ox" id="id20">[21]</a>.
Soon after, in 1972, the Protein DataBank (PDB)<a class="bibtex reference internal" href="../end-matter/reference.html#noauthor-undated-ow" id="id21">[22]</a> was established.
This continues to be well-used and updated, at the time of writing holding structures of 148,827 biological molecules.</p>
</div>
<div class="section" id="protein-classification">
<h3><span class="section-number">3.1.4.2. </span>Protein classification<a class="headerlink" href="#protein-classification" title="Permalink to this headline">¶</a></h3>
<p>Scientists are often interested in a “favourite” gene or protein, or have obtained a list of genes or proteins that they are interested in through a recent experiment. If an experiment about the specific protein has been carried out (e.g. to determine its function or structure), then a database like Uniprot (containing function and sequence information) or PDB (structure information) can be queried. However, this kind of information is not available for all proteins, so this is often necessary to make inferences about protein structure or function based on for example sequence similarity or protein classification.</p>
<div class="section" id="blast">
<h4><span class="section-number">3.1.4.2.1. </span>BLAST<a class="headerlink" href="#blast" title="Permalink to this headline">¶</a></h4>
<p>BLAST, or the The Basic Local Alignment Search Tool<a class="bibtex reference internal" href="../end-matter/reference.html#altschul1990-zf" id="id22">[23]</a>, is an extremely popular tool that is used to perform a basic search of nucleotide or amino acid sequences to known sequences, based on statistically significant similarities between parts of the sequence.</p>
</div>
<div class="section" id="scop">
<h4><span class="section-number">3.1.4.2.2. </span>SCOP<a class="headerlink" href="#scop" title="Permalink to this headline">¶</a></h4>
<p>The Structural Classification of Proteins (SCOP) database<a class="bibtex reference internal" href="../end-matter/reference.html#murzin1995-se" id="id23">[3]</a> classifies all proteins with known structure based on their structural similarities, based on the consideration of the protein’s constituent domains. The classification is mostly done at the level of families, superfamilies, and folds arranged in a tree structure. Families represent the most similar proteins, which share a “clear evolutionary relationship”, while superfamilies represent less close evolutionary relationships, and folds represent the same secondary structure. This protein classification task, while aided by automation, was carried out largely by manual visual inspection.</p>
<p>SCOP was updated until 2009, but has been succeeded by SCOP2<a class="bibtex reference internal" href="../end-matter/reference.html#andreeva2014-om" id="id24">[24]</a>. However, SCOP2 has a different underlying classification system, based on a complex graph, rather than a hierarchy. The CATH (Class, Architecture, Topology, Homologous superfamily)<a class="bibtex reference internal" href="../end-matter/reference.html#orengo1997-vf" id="id25">[25]</a> database provides another classification system, (also operating hierarchically), but created mostly via automation, which leads to major differences between the classifications<a class="bibtex reference internal" href="../end-matter/reference.html#csaba2009-of" id="id26">[26]</a>.</p>
</div>
<div class="section" id="superfamily">
<h4><span class="section-number">3.1.4.2.3. </span>SUPERFAMILY<a class="headerlink" href="#superfamily" title="Permalink to this headline">¶</a></h4>
<p>SUPERFAMILY<a class="bibtex reference internal" href="../end-matter/reference.html#gough2001-ct" id="id27">[27]</a> uses HMMs to assign sequences to SCOP domains, primarily at the superfamily level. This allows the functions of poorly understood proteins to be inferred based on how closely they match known superfamilies. HMMs are very successful at such assignments since pairwise correlations between proteins (or their domains) and other proteins in the family may be weak, but consistently for many proteins; this can be picked up by an HMM. The superfamily level is chosen since it is the broadest level which suggests evolutionary relationships, but SUPERFAMILY also generates assignments at the (stricter) family level.</p>
<p>HMMs are created by first finding closely relating protein homologs for a given protein superfamily using BLAST, and then extending it by comparing the HMM to more distantly related homologs. The resulting HMM library is fine-tuned by some manual curation.</p>
<p>The SUPERFAMILY website also contains other tools, including a database of all sequences (genomes) which are used to generate the HMM library.</p>
<p>(my-supfam-contribution=)
<strong>SUPERFAMILY update</strong>
I contributed to SUPERFAMILY’s 2014 update<a class="bibtex reference internal" href="../end-matter/reference.html#oates2015-li" id="id28">[1]</a> by editing the paper, uploading a small number of proteomes. The SUPERFAMILY database of proteomes doubled from 1400 to over 3200 from 2010 to 2014. The update paper described this development, as well as highlighting SUPERFAMILY as a resource for unique proteomes that are not found elsewhere (e.g. Uniprot). Although SUPERFAMILY’s primary resource is it’s HMM library, it also integrates a range of other tools for sequence analysis, for example protein disorder prediction (D2P2) and GO annotation (dcGO), as well as a domain-based phylogenetic tree.</p>
</div>
</div>
</div>
<div class="section" id="phenotypes">
<span id="phenotype-measurements"></span><h2><span class="section-number">3.1.5. </span>Phenotypes<a class="headerlink" href="#phenotypes" title="Permalink to this headline">¶</a></h2>
<div class="section" id="a-word-on-ontologies">
<span id="what-are-ontologies"></span><h3><span class="section-number">3.1.5.1. </span>A word on ontologies<a class="headerlink" href="#a-word-on-ontologies" title="Permalink to this headline">¶</a></h3>
<p>For many of different biological molecules, ontologies a a popular way of storing information about anatomical entities.
All ontologies contain entities names and descriptions (e.g. diseases, gene functions), different classifications of those entities (e.g. immune system disorders) and relate these classifications to one another hierarchically, sometimes with multiple types of relationships (e.g “is a”, “part of”). Hierarchical ontologies can be thought of as having a tree-like structure with one, or just a few root terms which are very general terms that all other terms in the ontology are related to, for example “biological process”, and leaf terms, which are the most specific terms in the ontology (e.g “positive regulation of cardiac muscle tissue regeneration”).</p>
<p>Relations between terms are directional, for example “positive regulation of cardiac muscle tissue regeneration” is a “regulation of cardiac muscle tissue regeneration”, but not vice versa. In such relationships the parent term is the more general term closer to the root (“positive regulation of…”) and the child term is the more specific term (“regulation of..”). It is not permitted for there to be cycles in ontologies, for example <em>term A</em> <code class="docutils literal notranslate"><span class="pre">is_a</span></code> <em>term B</em> <code class="docutils literal notranslate"><span class="pre">is_a</span></code> <em>term A</em>.</p>
<p>Terms in ontologies are given identifiers, usually of the form: <code class="docutils literal notranslate"><span class="pre">XXX:#######</span></code>, where <code class="docutils literal notranslate"><span class="pre">XXX</span></code> is an upper-case identifier for the whole ontology, e.g. <code class="docutils literal notranslate"><span class="pre">GO</span></code> for Gene Ontology, <code class="docutils literal notranslate"><span class="pre">CL</span></code> for Cell Ontology, etc. For example, <code class="docutils literal notranslate"><span class="pre">GO:0008150</span></code> is the GO term for <em>Biological Process</em>.</p>
<p>Ontologies are generally created through some combination of manual curation by highly skilled biocurators and logic-testing (checking for illogical relationships, for example using ROBOT<a class="bibtex reference internal" href="../end-matter/reference.html#overton2015-vo" id="id29">[28]</a>). Creating an ontology is generally a long-term project, with new suggestions and updates to the ontologies being made as new knowledge accumulates, or just as more people have time to add to them. As well as being the work of dedicated curators, contributions to ontologies can usually be crowd-sourced from the scientific community using GitHub issues, mailing list discussions, web forms, and dedicated workshops. In this way, they are similar to other bioinformatics community-driven efforts like structural and sequence databases.</p>
<p>There are also cross-ontology mappings and annotations, where terms from one ontology are linked to those in another (e.g. relating gene functions and tissues) or to entities in a database (e.g. gene functions to genes). These also require the work of dedicated curators, who search through literature, assessing various criteria for the inclusion of an annotation (such criteria vary by ontology). Since this is a laborious process, there are also many computational methods to annotate ontology terms automatically.</p>
<p>Ontologies can be used by researchers to investigate specific genes, tissues, functions of interest, or more generally to get a big-picture viewpoint on large groups of such entities.
Ontologies and particularly their annotations are varying degrees of incomplete, and this will have an impact on the results of any downstream use of them.</p>
<p id="obo-format">There are two major file formats in which ontologies are currently stored.
he OBO format is a human-readable format, while the OWL format is more complex, but has more functionality, and for example can be queried using SPARQL (an SQL-like querying language).</p>
<div class="section" id="gene-ontology">
<h4><span class="section-number">3.1.5.1.1. </span>Gene Ontology<a class="headerlink" href="#gene-ontology" title="Permalink to this headline">¶</a></h4>
<div class="figure align-default" id="go-rilla">
<a class="reference internal image-reference" href="../_images/go_rilla.png"><img alt="../_images/go_rilla.png" src="../_images/go_rilla.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3.3 </span><span class="caption-text">A subsection of the Gene Ontology with arrows showing the existence of relationships (image generated using GOrilla<a class="bibtex reference internal" href="../end-matter/reference.html#eden2009-ic" id="id30">[29]</a>)
``` ``</span><a class="headerlink" href="#go-rilla" title="Permalink to this image">¶</a></p>
<div class="legend">
<p>The Gene Ontology (GO)<a class="bibtex reference internal" href="../end-matter/reference.html#ashburner2000-cr" id="id31">[30]</a> is one of the first biomedical ontologies, and continues to be one of the most popular.
It is a collection of resources for cataloging the functions of gene products and designed for supporting the computational representation of biological systems<a class="bibtex reference internal" href="../end-matter/reference.html#thomas2017-vm" id="id32">[31]</a>.
It includes:</p>
<ol class="simple">
<li><p>The standard gene ontology, which is a hierarchical set of terms describing functions.</p></li>
<li><p>The gene ontology annotations (GOA) database, which contains manual and computationally derived mappings from gene products to gene ontology terms.</p></li>
<li><p>Tools for using and updating these resources.</p></li>
</ol>
<p>The Gene Ontology defines the “universe” of possible functions a gene might have, while the functions of particular genes are captured as GO annotations<a class="bibtex reference internal" href="../end-matter/reference.html#thomas2017-vm" id="id33">[31]</a>.</p>
<p>The terms in the GO ontology are subdivided into three types (molecular function, biological process, and cellular component), meaning that GO is actually a collection of three ontologies<a class="bibtex reference internal" href="../end-matter/reference.html#ashburner2000-cr" id="id34">[30]</a>.
Gene products in GO are assumed to carry out molecular-level process or activity (molecular function) in a specific location relative to the cell (cellular component), and this molecular process contributes to a larger biological objective (biological process)<a class="bibtex reference internal" href="../end-matter/reference.html#thomas2017-vm" id="id35">[31]</a>.</p>
<hr class="docutils" />
<p><strong>Chapter References</strong></p>
<p id="bibtex-bibliography-c03-compbio-background/1-measuring-genotype-phenotype-0"><dl class="citation">
<dt class="bibtex label" id="oates2015-li"><span class="brackets"><a class="fn-backref" href="#id28">1</a></span></dt>
<dd><p>Matt E Oates, Jonathan Stahlhacke, Dimitrios V Vavoulis, Ben Smithers, Owen J L Rackham, Adam J Sardar, Jan Zaucha, Natalie Thurlby, Hai Fang, and Julian Gough. The SUPERFAMILY 1.75 database in 2014: a doubling of data. <em>Nucleic Acids Res.</em>, 43(Database issue):D227–33, January 2015.</p>
</dd>
<dt class="bibtex label" id="venter2001-wn"><span class="brackets"><a class="fn-backref" href="#id5">2</a></span></dt>
<dd><p>J C Venter, M D Adams, E W Myers, P W Li, R J Mural, G G Sutton, H O Smith, M Yandell, C A Evans, R A Holt, J D Gocayne, P Amanatides, R M Ballew, D H Huson, J R Wortman, Q Zhang, C D Kodira, X H Zheng, L Chen, M Skupski, G Subramanian, P D Thomas, J Zhang, G L Gabor Miklos, C Nelson, S Broder, A G Clark, J Nadeau, V A McKusick, N Zinder, A J Levine, R J Roberts, M Simon, C Slayman, M Hunkapiller, R Bolanos, A Delcher, I Dew, D Fasulo, M Flanigan, L Florea, A Halpern, S Hannenhalli, S Kravitz, S Levy, C Mobarry, K Reinert, K Remington, J Abu-Threideh, E Beasley, K Biddick, V Bonazzi, R Brandon, M Cargill, I Chandramouliswaran, R Charlab, K Chaturvedi, Z Deng, V Di Francesco, P Dunn, K Eilbeck, C Evangelista, A E Gabrielian, W Gan, W Ge, F Gong, Z Gu, P Guan, T J Heiman, M E Higgins, R R Ji, Z Ke, K A Ketchum, Z Lai, Y Lei, Z Li, J Li, Y Liang, X Lin, F Lu, G V Merkulov, N Milshina, H M Moore, A K Naik, V A Narayan, B Neelam, D Nusskern, D B Rusch, S Salzberg, W Shao, B Shue, J Sun, Z Wang, A Wang, X Wang, J Wang, M Wei, R Wides, C Xiao, C Yan, A Yao, J Ye, M Zhan, W Zhang, H Zhang, Q Zhao, L Zheng, F Zhong, W Zhong, S Zhu, S Zhao, D Gilbert, S Baumhueter, G Spier, C Carter, A Cravchik, T Woodage, F Ali, H An, A Awe, D Baldwin, H Baden, M Barnstead, I Barrow, K Beeson, D Busam, A Carver, A Center, M L Cheng, L Curry, S Danaher, L Davenport, R Desilets, S Dietz, K Dodson, L Doup, S Ferriera, N Garg, A Gluecksmann, B Hart, J Haynes, C Haynes, C Heiner, S Hladun, D Hostin, J Houck, T Howland, C Ibegwam, J Johnson, F Kalush, L Kline, S Koduru, A Love, F Mann, D May, S McCawley, T McIntosh, I McMullen, M Moy, L Moy, B Murphy, K Nelson, C Pfannkoch, E Pratts, V Puri, H Qureshi, M Reardon, R Rodriguez, Y H Rogers, D Romblad, B Ruhfel, R Scott, C Sitter, M Smallwood, E Stewart, R Strong, E Suh, R Thomas, N N Tint, S Tse, C Vech, G Wang, J Wetter, S Williams, M Williams, S Windsor, E Winn-Deen, K Wolfe, J Zaveri, K Zaveri, J F Abril, R Guigó, M J Campbell, K V Sjolander, B Karlak, A Kejariwal, H Mi, B Lazareva, T Hatton, A Narechania, K Diemer, A Muruganujan, N Guo, S Sato, V Bafna, S Istrail, R Lippert, R Schwartz, B Walenz, S Yooseph, D Allen, A Basu, J Baxendale, L Blick, M Caminha, J Carnes-Stine, P Caulk, Y H Chiang, M Coyne, C Dahlke, A Mays, M Dombroski, M Donnelly, D Ely, S Esparham, C Fosler, H Gire, S Glanowski, K Glasser, A Glodek, M Gorokhov, K Graham, B Gropman, M Harris, J Heil, S Henderson, J Hoover, D Jennings, C Jordan, J Jordan, J Kasha, L Kagan, C Kraft, A Levitsky, M Lewis, X Liu, J Lopez, D Ma, W Majoros, J McDaniel, S Murphy, M Newman, T Nguyen, N Nguyen, M Nodell, S Pan, J Peck, M Peterson, W Rowe, R Sanders, J Scott, M Simpson, T Smith, A Sprague, T Stockwell, R Turner, E Venter, M Wang, M Wen, D Wu, M Wu, A Xia, A Zandieh, and X Zhu. The sequence of the human genome. <em>Science</em>, 291(5507):1304–1351, February 2001.</p>
</dd>
<dt class="bibtex label" id="murzin1995-se"><span class="brackets"><a class="fn-backref" href="#id23">3</a></span></dt>
<dd><p>A G Murzin, S E Brenner, T Hubbard, and C Chothia. SCOP: a structural classification of proteins database for the investigation of sequences and structures. <em>J. Mol. Biol.</em>, 247(4):536–540, April 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.1995.0159">http://dx.doi.org/10.1006/jmbi.1995.0159</a>.</p>
</dd>
<dt class="bibtex label" id="blunt2001-wr"><span class="brackets"><a class="fn-backref" href="#id1">4</a></span></dt>
<dd><p>Wilfrid Blunt. <em>The Compleat Naturalist: A Life of Linnaeus</em>. Frances Lincoln, 2001. URL: <a class="reference external" href="https://play.google.com/store/books/details?id=B3YOvgAACAAJ">https://play.google.com/store/books/details?id=B3YOvgAACAAJ</a>.</p>
</dd>
<dt class="bibtex label" id="ehret1748-bx"><span class="brackets"><a class="fn-backref" href="#id2">5</a></span></dt>
<dd><p>Ehret. <em>Plantae et papiliones rariores</em>. Volume 1748. [London :s.n.],, 1748. URL: <a class="reference external" href="https://www.biodiversitylibrary.org/item/205762">https://www.biodiversitylibrary.org/item/205762</a>.</p>
</dd>
<dt class="bibtex label" id="charmantier-undated-wy"><span class="brackets"><a class="fn-backref" href="#id3">6</a></span></dt>
<dd><p>Dr Isabelle Charmantier. Linnaeus and race. <span><a class="reference external" href="#"></a></span>https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race. Accessed: 2020-11-30. URL: <a class="reference external" href="https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race">https://www.linnean.org/learning/who-was-linnaeus/linnaeus-and-race</a>.</p>
</dd>
<dt class="bibtex label" id="jensen2010-fk"><span class="brackets"><a class="fn-backref" href="#id4">7</a></span></dt>
<dd><p>Lars J Jensen and Peer Bork. Ontologies in quantitative biology: a basis for comparison, integration, and discovery. <em>PLoS Biol.</em>, 8(5):e1000374, May 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1000374">http://dx.doi.org/10.1371/journal.pbio.1000374</a>.</p>
</dd>
<dt class="bibtex label" id="haas2010-lm"><span class="brackets"><a class="fn-backref" href="#id6">8</a></span></dt>
<dd><p>Brian J Haas and Michael C Zody. Advancing RNA-Seq analysis. <em>Nat. Biotechnol.</em>, 28(5):421–423, May 2010.</p>
</dd>
<dt class="bibtex label" id="pfeiffer2018-kt"><span class="brackets"><a class="fn-backref" href="#id7">9</a></span></dt>
<dd><p>Franziska Pfeiffer, Carsten Gröber, Michael Blank, Kristian Händler, Marc Beyer, Joachim L Schultze, and Günter Mayer. Systematic evaluation of error rates and causes in short samples in next-generation sequencing. <em>Sci. Rep.</em>, 8(1):10950, July 2018.</p>
</dd>
<dt class="bibtex label" id="de-koning2011-ac"><span class="brackets"><a class="fn-backref" href="#id8">10</a></span></dt>
<dd><p>A P Jason de Koning, Wanjun Gu, Todd A Castoe, Mark A Batzer, and David D Pollock. Repetitive elements may comprise over two-thirds of the human genome. <em>PLoS Genet.</em>, 7(12):e1002384, December 2011.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ci"><span class="brackets"><a class="fn-backref" href="#id9">11</a></span></dt>
<dd><p>Genome browser FAQ. <span><a class="reference external" href="#"></a></span>https://genome.ucsc.edu/FAQ/FAQreleases.html. Accessed: 2020-12-13. URL: <a class="reference external" href="https://genome.ucsc.edu/FAQ/FAQreleases.html">https://genome.ucsc.edu/FAQ/FAQreleases.html</a>.</p>
</dd>
<dt class="bibtex label" id="gatk-team2020-au"><span class="brackets"><a class="fn-backref" href="#id10">12</a></span></dt>
<dd><p>GATK Team. Human genome reference builds - GRCh38 or hg38 - b37 - hg19. <span><a class="reference external" href="#"></a></span>https://gatk.broadinstitute.org/hc/en-us/articles/360035890951, June 2020. Accessed: 2020-12-13. URL: <a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890951">https://gatk.broadinstitute.org/hc/en-us/articles/360035890951</a>.</p>
</dd>
<dt class="bibtex label" id="salzberg2018-yc"><span class="brackets"><a class="fn-backref" href="#id11">13</a></span></dt>
<dd><p>Steven L Salzberg. Open questions: how many genes do we have? <em>BMC Biol.</em>, 16(1):94, August 2018.</p>
</dd>
<dt class="bibtex label" id="bumgarner2013-hg"><span class="brackets"><a class="fn-backref" href="#id12">14</a></span></dt>
<dd><p>Roger Bumgarner. Overview of DNA microarrays: types, applications, and their future. <em>Curr. Protoc. Mol. Biol.</em>, Chapter 22:Unit 22.1., January 2013.</p>
</dd>
<dt class="bibtex label" id="sherry2001-nm"><span class="brackets"><a class="fn-backref" href="#id13">15</a></span></dt>
<dd><p>S T Sherry. dbSNP: the NCBI database of genetic variation. <em>Nucleic Acids Res.</em>, 29(1):308–311, 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/29.1.308">http://dx.doi.org/10.1093/nar/29.1.308</a>.</p>
</dd>
<dt class="bibtex label" id="illumina2014-nd"><span class="brackets"><a class="fn-backref" href="#id14">16</a></span></dt>
<dd><p>I Illumina. Understanding illumina quality scores. <em>Technical Note: Informatics</em>, 2014. URL: <a class="reference external" href="https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf">https://www.illumina.com/documents/products/technotes/technote_Q-Scores.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="wagner2012-ac"><span class="brackets"><a class="fn-backref" href="#id15">17</a></span></dt>
<dd><p>Günter P Wagner, Koryu Kin, and Vincent J Lynch. Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples. <em>Theory Biosci.</em>, 131(4):281–285, December 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1007/s12064-012-0162-3">http://dx.doi.org/10.1007/s12064-012-0162-3</a>.</p>
</dd>
<dt class="bibtex label" id="pimentel2014-xm"><span class="brackets"><a class="fn-backref" href="#id16">18</a></span></dt>
<dd><p>H Pimentel. What the FPKM? a review of RNA-Seq expression units. 2014. URL: <a class="reference external" href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/</a>.</p>
</dd>
<dt class="bibtex label" id="robinson2010-kl"><span class="brackets">19</span><span class="fn-backref">(<a href="#id17">1</a>,<a href="#id18">2</a>)</span></dt>
<dd><p>M D Robinson, D J McCarthy, and G K Smyth. Edger: a bioconductor package for differential expression analysis of digital gene expression data. <em>Bioinformatics</em>, 26(1):139–140, 2010. URL: <a class="reference external" href="http://dx.doi.org/10.1093/bioinformatics/btp616">http://dx.doi.org/10.1093/bioinformatics/btp616</a>.</p>
</dd>
<dt class="bibtex label" id="love2014-vx"><span class="brackets"><a class="fn-backref" href="#id19">20</a></span></dt>
<dd><p>Michael I Love, Wolfgang Huber, and Simon Anders. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. <em>Genome Biol.</em>, 15(12):550, 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-014-0550-8">http://dx.doi.org/10.1186/s13059-014-0550-8</a>.</p>
</dd>
<dt class="bibtex label" id="hersh1967-ox"><span class="brackets"><a class="fn-backref" href="#id20">21</a></span></dt>
<dd><p>Robert T Hersh. Atlas of protein sequence and structure, 1966. <em>Syst. Biol.</em>, 16(3):262–263, September 1967.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ow"><span class="brackets"><a class="fn-backref" href="#id21">22</a></span></dt>
<dd><p>Protein data bank.</p>
</dd>
<dt class="bibtex label" id="altschul1990-zf"><span class="brackets"><a class="fn-backref" href="#id22">23</a></span></dt>
<dd><p>S F Altschul, W Gish, W Miller, E W Myers, and D J Lipman. Basic local alignment search tool. <em>J. Mol. Biol.</em>, 215(3):403–410, October 1990. URL: <a class="reference external" href="http://dx.doi.org/10.1016/S0022-2836(05)80360-2">http://dx.doi.org/10.1016/S0022-2836(05)80360-2</a>.</p>
</dd>
<dt class="bibtex label" id="andreeva2014-om"><span class="brackets"><a class="fn-backref" href="#id24">24</a></span></dt>
<dd><p>Antonina Andreeva, Dave Howorth, Cyrus Chothia, Eugene Kulesha, and Alexey G Murzin. SCOP2 prototype: a new approach to protein structure mining. <em>Nucleic Acids Res.</em>, 42(Database issue):D310–4, January 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkt1242">http://dx.doi.org/10.1093/nar/gkt1242</a>.</p>
</dd>
<dt class="bibtex label" id="orengo1997-vf"><span class="brackets"><a class="fn-backref" href="#id25">25</a></span></dt>
<dd><p>C A Orengo, A D Michie, S Jones, D T Jones, M B Swindells, and J M Thornton. CATH – a hierarchic classification of protein domain structures. <em>Structure</em>, 5(8):1093–1109, 1997. URL: <a class="reference external" href="http://dx.doi.org/10.1016/s0969-2126(97)00260-8">http://dx.doi.org/10.1016/s0969-2126(97)00260-8</a>.</p>
</dd>
<dt class="bibtex label" id="csaba2009-of"><span class="brackets"><a class="fn-backref" href="#id26">26</a></span></dt>
<dd><p>Gergely Csaba, Fabian Birzele, and Ralf Zimmer. Systematic comparison of SCOP and CATH: a new gold standard for protein structure analysis. <em>BMC Struct. Biol.</em>, 9:23, April 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1472-6807-9-23">http://dx.doi.org/10.1186/1472-6807-9-23</a>.</p>
</dd>
<dt class="bibtex label" id="gough2001-ct"><span class="brackets"><a class="fn-backref" href="#id27">27</a></span></dt>
<dd><p>J Gough, K Karplus, R Hughey, and C Chothia. Assignment of homology to genome sequences using a library of hidden markov models that represent all proteins of known structure. <em>J. Mol. Biol.</em>, 313(4):903–919, November 2001. URL: <a class="reference external" href="http://dx.doi.org/10.1006/jmbi.2001.5080">http://dx.doi.org/10.1006/jmbi.2001.5080</a>.</p>
</dd>
<dt class="bibtex label" id="overton2015-vo"><span class="brackets"><a class="fn-backref" href="#id29">28</a></span></dt>
<dd><p>James A Overton, Heiko Dietze, Shahim Essaid, David Osumi-Sutherland, and Christopher J Mungall. ROBOT: a command-line tool for ontology development. In <em>ICBO</em>. ceur-ws.org, 2015. URL: <a class="reference external" href="http://ceur-ws.org/Vol-1515/demo6.pdf">http://ceur-ws.org/Vol-1515/demo6.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="eden2009-ic"><span class="brackets"><a class="fn-backref" href="#id30">29</a></span></dt>
<dd><p>Eran Eden, Roy Navon, Israel Steinfeld, Doron Lipson, and Zohar Yakhini. GOrilla: a tool for discovery and visualization of enriched GO terms in ranked gene lists. <em>BMC Bioinformatics</em>, 10:48, February 2009. URL: <a class="reference external" href="http://dx.doi.org/10.1186/1471-2105-10-48">http://dx.doi.org/10.1186/1471-2105-10-48</a>.</p>
</dd>
<dt class="bibtex label" id="ashburner2000-cr"><span class="brackets">30</span><span class="fn-backref">(<a href="#id31">1</a>,<a href="#id34">2</a>)</span></dt>
<dd><p>M Ashburner, C A Ball, J A Blake, D Botstein, H Butler, J M Cherry, A P Davis, K Dolinski, S S Dwight, J T Eppig, M A Harris, D P Hill, L Issel-Tarver, A Kasarskis, S Lewis, J C Matese, J E Richardson, M Ringwald, G M Rubin, and G Sherlock. Gene ontology: tool for the unification of biology. the gene ontology consortium. <em>Nat. Genet.</em>, 25(1):25–29, May 2000. URL: <a class="reference external" href="http://dx.doi.org/10.1038/75556">http://dx.doi.org/10.1038/75556</a>.</p>
</dd>
<dt class="bibtex label" id="thomas2017-vm"><span class="brackets">31</span><span class="fn-backref">(<a href="#id32">1</a>,<a href="#id33">2</a>,<a href="#id35">3</a>)</span></dt>
<dd><p>Paul D Thomas. The gene ontology and the meaning of biological function. <em>Methods Mol. Biol.</em>, 1446:15–24, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1007/978-1-4939-3743-1_2">http://dx.doi.org/10.1007/978-1-4939-3743-1_2</a>.</p>
</dd>
</dl>
</p>
</div>
</div>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c03-compbio-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="0-index.html" title="previous page"><span class="section-number">3. </span>Computational Biology Background</a>
    <a class='right-next' id="next-link" href="2-comp-bio-methods.html" title="next page"><span class="section-number">3.2. </span>How large computational biology datasets are used</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>