
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>3.3. Sources of bias in computational biology datasets &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="4. Phenotype prediction with Snowflake" href="../c04-snowflake/0-index.html" />
    <link rel="prev" title="3.2. How large computational biology datasets are used" href="2-comp-bio-methods.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../jupyter_book_intro.html">
   Start Here
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/title-page.html">
   Title Page
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-figures.html">
   List of Figures
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-tables.html">
   List of Tables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/declaration.html">
   Declaration
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c02-biology-background/0-bg-biology.html">
   2. Biology Background
  </a>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-measuring-genotype-phenotype.html">
     3.1. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-comp-bio-methods.html">
     3.2. How large computational biology datasets are used
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     3.3. Sources of bias in computational biology datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="#summary">
     3.4. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.2. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.3. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.4. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-combining/0-index.html">
   5. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/1-background.html">
     5.1.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/2-data.html">
     5.1.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/3-data-wrangling.html">
     5.1.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html">
     5.1.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html#functionality">
     5.1.5. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/5-batch-correction.html">
     5.1.6. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html">
     5.1.7. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/7-future-work.html">
     5.1.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-data.html">
     6.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-methods.html">
     6.3. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/4-results.html">
     6.4. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-discussion.html">
     6.5. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-future-work.html">
     6.6. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/c03-compbio-background/3-bias.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   3.3. Sources of bias in computational biology datasets
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#why-we-care">
     3.3.1. Why we care
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#we-want-to-trust-science">
       3.3.1.1. We want to trust science
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#the-reproducibility-crisis">
       3.3.1.2. The reproducibility crisis
      </a>
      <ul class="nav section-nav flex-column">
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#null-hypothesis-significance-testing">
         3.3.1.2.1. Null Hypothesis Significance Testing
        </a>
       </li>
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#publication-bias">
         3.3.1.2.2. Publication bias
        </a>
       </li>
      </ul>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#reproducibility-in-bioinformatics">
       3.3.1.3. Reproducibility in bioinformatics
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#when-is-a-model-useful">
       3.3.1.4. When is a model useful?
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#summary">
   3.4. Summary
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="sources-of-bias-in-computational-biology-datasets">
<h1><span class="section-number">3.3. </span>Sources of bias in computational biology datasets<a class="headerlink" href="#sources-of-bias-in-computational-biology-datasets" title="Permalink to this headline">¶</a></h1>
<div class="section" id="why-we-care">
<h2><span class="section-number">3.3.1. </span>Why we care<a class="headerlink" href="#why-we-care" title="Permalink to this headline">¶</a></h2>
<div class="section" id="we-want-to-trust-science">
<h3><span class="section-number">3.3.1.1. </span>We want to trust science<a class="headerlink" href="#we-want-to-trust-science" title="Permalink to this headline">¶</a></h3>
<blockquote class="epigraph">
<div><p>The imposing edifice of science provides a challenging view of what can be achieved by the accumulation of many small efforts in a steady objective and dedicated search for truth.</p>
<p class="attribution">—Charles H. Townes</p>
</div></blockquote>
<p>We want to trust the results of scientific research. Not only when it’s our own (because it’s fun and exciting to look for and find the truth), but because scientific research is generally paid for by tax, and the results that are generated by it drive policy, drug treatments, and innovations.</p>
<p>In all fields, science is a search for the truth. And in all fields, there are concerns about what makes bad, unreliable, un-useful, or biased research; what must be done or not done to uphold science’s claim to truth, or at least reliability. Computational biology has a number of unique characteristics which make these questions worth discussing in its specific context.</p>
<p>As mentioned earlier, in contrast to other fields, many bioinformatics datasets have been freely available and accessible on the internet since its inception. Computational biology might be a good place to look for what is required to really benefit from those open datasets, what can be gained from them (and what can’t), and what the next steps in reproducibility should be, beyond making data accessible.c</p>
<p>Secondly, the creation of these databases has birthed a lot of research that is highly reliant on a handful of large ontologies and databases, the completeness of which is unknown, and the cataloging process by which they are added to is extremely uneven.</p>
<p>Thirdly, many of the new publications in this field showcase novel software and models which build on these resources, despite the fact that much of it is difficult to validate due to the lack of gold standard datasets.</p>
<p>When scientific results are often based on statistics, it’s inevitable that some proportion of published scientific results will not be true. This isn’t a problem, as over time, researchers can double-check interesting scientific results, and the literature can be updated to reflect that. This is sciences self-correcting mechanism. If a result can be replicated in a different circumstance by a different person, it reinforces the likelihood that the result is true. A replication doesn’t have to reveal the exact same level of statistical significance  or effect size to be successful, but (usually, depending on definitions) just a similar result.</p>
</div>
<div class="section" id="the-reproducibility-crisis">
<h3><span class="section-number">3.3.1.2. </span>The reproducibility crisis<a class="headerlink" href="#the-reproducibility-crisis" title="Permalink to this headline">¶</a></h3>
<blockquote class="epigraph">
<div><p>In science consensus is irrelevant. What is relevant is reproducible results.</p>
<p class="attribution">—Michael Crichton</p>
</div></blockquote>
<p>The reproducibility crisis is the realisation that worryingly large proportions of research results do not reproduce. Replication studies have found that only 11% of cancer research findings<a class="bibtex reference internal" href="../end-matter/reference.html#begley2012-oc" id="id1">[2]</a>, 20-25% of drug-target findings<a class="bibtex reference internal" href="../end-matter/reference.html#begley2012-oc" id="id2">[2]</a><a class="bibtex reference internal" href="../end-matter/reference.html#prinz2011-rh" id="id3">[3]</a>, and 39% of psychology findings<a class="bibtex reference internal" href="../end-matter/reference.html#open-science-collaboration2015-vi" id="id4">[4]</a> reproduced. Surveys of researchers across disciplines reveal that more than 70% of scientists say they have failed to reproduce another paper’s result, and over 50% say they have failed to reproduce their own results<a class="bibtex reference internal" href="../end-matter/reference.html#baker2016-tb" id="id5">[5]</a>. It seems that science’s self-correcting mechanism is not working as intended.</p>
<p>This shocking irreproducibility is thought to be due to a multitude of factors, including poor data management, lack of available materials/details of experiments, publication bias, poor statistical knowledge, and questionable research practices<a class="bibtex reference internal" href="../end-matter/reference.html#bishop2019-hg" id="id6">[6]</a>.</p>
<div class="section" id="null-hypothesis-significance-testing">
<h4><span class="section-number">3.3.1.2.1. </span>Null Hypothesis Significance Testing<a class="headerlink" href="#null-hypothesis-significance-testing" title="Permalink to this headline">¶</a></h4>
<blockquote class="epigraph">
<div><p>Statistical significance is the least interesting thing about the results. You should describe the results in terms of measures of magnitude –not just, does a treatment affect people, but how much does it affect them.</p>
<p class="attribution">—Gene V. Glass</p>
</div></blockquote>
<p>Null Hypothesis Significance Testing (NHST) is the most popular method by which scientific hypotheses are tested and reported. This reporting usually consists mostly of a p-value, a measure of statistical significance: the likelihood that a false positive could be obtained just by chance. In other words p-values are a measure of type 1 error (false positives). The threshold for this, usually denoted by <span class="math notranslate nohighlight">\(\alpha\)</span> is most often set to 0.05, as recommended by Fischer, however there have been many debates about whether this is the most sensible cut-off for science today. Despite the dominance of p-values as main or only reported statistic across scientific fields, they do not imply that a result is interesting (the effect might be small or the hypothesis uninteresting), or even that it’s likely to be true. Sometimes the p-value is not even reported, but only whether or not it crossed the p&lt;0.05 threshold.</p>
<p>Two other useful quantities of NHST are effect size and statistical power. Effect size is the magnitude of the effect found by the statistical test. A very small effect can only be detected with a large enough sample size. Statistical power is a measure related to the type 2 error (false negatives), it is <span class="math notranslate nohighlight">\(1-\beta\)</span> where <span class="math notranslate nohighlight">\(\beta\)</span> is the false positive rate. A statistical power of 80% is customary, where it is calculated, in which case there is a 20% chance that a result is a false negative if the null hypothesis is accepted. A very highly powered test with a high (non-significant) p-value represents strong evidence that the null hypothesis is true, although it may often be reported as “failing to reach significance”. Low-powered tests, coinciding with low sample sizes, mean that both the acceptance or rejection of the null hypothesis is likely to be unreliable.</p>
<p>P-values do not have a high prediction value for reproducibility, since they have a high spread, even when a test is reasonably highly powered. Statistician Geoff Cumming refers to this as the “dance of the p-values”. Instead, a measure of the expected truth of a finding can be estimated from the proportion of hypotheses that are true in a given field, the statistical power, the p-value threshold as:</p>
<div class="math notranslate nohighlight">
\[
ppv=\frac{(1-\beta)p_{true}}{p_{true}(1-\beta-\alpha)+\alpha}
\]</div>
<p>Where <span class="math notranslate nohighlight">\(ppv\)</span> stands for positive predictive value and <span class="math notranslate nohighlight">\(p_{true}\)</span> is the proportion of true hypotheses in a field<a class="bibtex reference internal" href="../end-matter/reference.html#ioannidis2005-mo" id="id7">[7]</a>.</p>
<p>For this version of the formula (there is also a version that includes bias, which was instrumental in the Ionnaidis’ claim that “most published research findings are false”<a class="bibtex reference internal" href="../end-matter/reference.html#ioannidis2005-mo" id="id8">[7]</a>), and standard choices for power and statistical significance of <span class="math notranslate nohighlight">\(\alpha=0.05\)</span> and <span class="math notranslate nohighlight">\(\beta=0.2\)</span>, we would expect more findings to false than true if <span class="math notranslate nohighlight">\(p_{true}&lt;0.0588\)</span> (3.s.f). That might seem like a small number, but in some bioinformatics experiments, we hypothesise that millions of SNPs may be responsible for a trait, when only small numbers are. On the other hand, if half of researchers hypotheses were correct for a given field (<span class="math notranslate nohighlight">\(p_{true}=0.5\)</span>), the formula would yield <span class="math notranslate nohighlight">\(ppv=0.941\)</span> (3.s.f.), but the low reproducibility of GWAS results, gene annotations, etc, implies that the proportion of true hypotheses is less than this.</p>
<p>The same approach can be used to calculate the limit for <span class="math notranslate nohighlight">\(p_{true}\)</span> for which we’d expect there to be more false positives than false negatives. Using the same values for and  and , we get <span class="math notranslate nohighlight">\(p_{true}=0.2\)</span>, i.e. if less than 20% of hypotheses are true, then we are more likely to get false positives than false negatives. This is interesting as most published scientific results are claiming a positive result, so we are essentially erring on the side of publishing erroneous errors.</p>
<div class="figure align-default" id="p-hacking">
<a class="reference internal image-reference" href="../_images/p_hacking.png"><img alt="../_images/p_hacking.png" src="../_images/p_hacking.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 3.3 </span><span class="caption-text">Images that are illustrative of researchers approaches to p-values and p-hacking. The left image is <a class="reference external" href="https://twitter.com/FaustoBustos/status/1103435523777978368">a popular tweet</a>, while the right image is <a class="reference external" href="https://xkcd.com/1478/">an xkcd comic</a>).</span><a class="headerlink" href="#p-hacking" title="Permalink to this image">¶</a></p>
</div>
<p>In addition to poor reporting and underpowered tests, the pressure on scientists to publish means that researchers may be tempted to (or may accidentally, due to statistical ignorance) employ data-mining tactics in order to harvest significant p-values. This practice is known as “p-hacking”, and evidence for its existence can be found in distributions of p-values in scientific literature<a class="bibtex reference internal" href="../end-matter/reference.html#head2015-ns" id="id9">[8]</a>, as well as popular culture (<a class="reference internal" href="#p-hacking"><span class="std std-numref">Fig. 3.3</span></a>). This can include rerunning analysis with different models/covariates, collecting data until a significant p-value is reached, or performing 20 experiments and only publishing the results of one.</p>
<blockquote class="epigraph">
<div><p>The first principle is that you must not fool yourself – and you are the easiest person to fool.
– Richard Feynman</p>
</div></blockquote>
<p>There are several suggested tonics to the problem of uninformative and ubiquitous p-values. Reporting p-values as numbers rather than in relation to a threshold (e.g. p&lt;0.05) is starting point. Information about statistical power and effect size should also be provided. In addition to giving researchers reading a paper a better idea of the quality of it, this also allows science to self-correct a little easier, since individual p-values can then be combined into more reliable p-values, using for example Fischer’s method<a class="bibtex reference internal" href="../end-matter/reference.html#fisher1990-ro" id="id10">[9]</a>.</p>
<p>A more interesting solution is pre-registration, as used in clinical trials. This involves a detailed publication in advance of the analysis protocols that will be used in order to prevent tweaking analysis based on seeing the data. This solves p-hacking related problems, and makes a distinction between hypothesis-generating and hypothesis-testing research.</p>
<p>For cases where many hypotheses are being generated at once (for example in GWAS), multiple hypothesis corrections (e.g. the Bonferroni correction<a class="bibtex reference internal" href="../end-matter/reference.html#dunn1958-sj" id="id11">[10]</a> or the False Discovery Rate<a class="bibtex reference internal" href="../end-matter/reference.html#benjamini1995-me" id="id12">[11]</a>) can be employed to adjust the p-value to account for this.</p>
</div>
<div class="section" id="publication-bias">
<h4><span class="section-number">3.3.1.2.2. </span>Publication bias<a class="headerlink" href="#publication-bias" title="Permalink to this headline">¶</a></h4>
<p>Although with current standard p-value and power cut-offs, negative results are more likely to be true than positive ones, negative results are much harder to publish. This bias is likely to be responsible for the draw of questionable research practices like p-hacking. It also means that there is a lot of unpublished, negative results which are likely to be repeated, since there is no way that someone could know it has already been done. A highly powered negative result could be very interesting, for example, we know hardly anything about which genes do not appear to affect phenotypes, since these results are not published<a class="bibtex reference internal" href="../end-matter/reference.html#barbaric2007-zm" id="id13">[12]</a>, but they would help enormously with the challenge of creating a gold standard dataset for gene function prediction.</p>
<p>Registered reports are an attempt to remove these problems associated with publication bias, by linking the concept of pre-registration with that of publishing. Essentially, authors submit their introduction and methods section to a journal and at this point they undergo peer review and the journal agrees to publish the results, regardless of the result. At the time of writing over 170 journals were accepting registered reports and the number has been growing in recent months, across disciplines, although they are currently most popular in psychology and neuroscience<a class="bibtex reference internal" href="../end-matter/reference.html#hardwicke-undated-jj" id="id14">[13]</a>. This solution also offers peer-review at a more helpful stage in the manuscript, when it’s still possible to make changes to the experiment.</p>
</div>
</div>
<div class="section" id="reproducibility-in-bioinformatics">
<h3><span class="section-number">3.3.1.3. </span>Reproducibility in bioinformatics<a class="headerlink" href="#reproducibility-in-bioinformatics" title="Permalink to this headline">¶</a></h3>
<p>In a field that has long had a huge number of open data repositories, and a relatively high level of statistical knowledge among researchers, in some ways bioinformatics might be expected to be ahead of the curve in terms of reproducible research. It certainly seems that as a field, it excelling at open research. At the same time, however, it is even more important for the work to be reproducible if data and software are being reused by multiple researchers.</p>
<p>The Gene Ontology Annotations (GOA) are a combination of experimental and computational annotations. While the supporting publications for the experimental annotations are available, the GO consortium do not provide the statistical evidence that they used alongside this (e.g. p-value, effect size), etc.</p>
</div>
<div class="section" id="when-is-a-model-useful">
<h3><span class="section-number">3.3.1.4. </span>When is a model useful?<a class="headerlink" href="#when-is-a-model-useful" title="Permalink to this headline">¶</a></h3>
<blockquote class="epigraph">
<div><p>All models are wrong, but some are useful</p>
<p class="attribution">—George Box</p>
</div></blockquote>
<p>Models are most useful to us when they generate testable hypotheses about the underlying mechanisms of a process. In this way, they can help to advance science. They can also be useful to us if they generate predictions for forecasts. These predictions can be tested, which helps us to improve the model, and if they are accurate, they can also help us to generate hypotheses, or they may be useful in and of themselves.</p>
</div>
</div>
</div>
<div class="section" id="summary">
<h1><span class="section-number">3.4. </span>Summary<a class="headerlink" href="#summary" title="Permalink to this headline">¶</a></h1>
<p>The purpose of this introductory chapter was to provide an overview of how we think phenotype arises from genotype. It’s also to explain why it’s a hard problem!</p>
<p>To summarise, there are multiple kinds of genetic variation, and even the simplest (the SNP) can exist on multiple parts of the genome, and may or may not affect structure or function of transcribed proteins. Even if the SNP is in a coding region, and is non-synonymous (affects protein structure), it may fall in a disordered region of a protein (leaving us without structural information), we may not know where that protein is transcribed, or it may affect the transcription of multiple different proteins. Furthermore, the SNP may affect phenotype differently with homozygous or heterozygous calls, and the protein may affect phenotype by influencing a network of other proteins, or the protein may exist as a redundant part of a pathway which will only affect phenotype if three other SNPs have specific calls. Even after all this, the presentation of many phenotype depends heavily on the environment.</p>
<p>Given all this complexity, it may seem no wonder that phenotype prediction is currently inaccurate<a class="bibtex reference internal" href="../end-matter/reference.html#zhou2019-jk" id="id15">[4]</a>. However, diverse information about biological entities exists, for example gene and protein sequence, protein structure, variant frequencies and functions, and gene expression. Multi-omics approaches that combine these data types have been successful at elucidating mechanisms behind certain phenotypes<a class="bibtex reference internal" href="../end-matter/reference.html#hasin2017-tk" id="id16">[14]</a><a class="bibtex reference internal" href="../end-matter/reference.html#ritchie2015-ds" id="id17">[15]</a><a class="bibtex reference internal" href="../end-matter/reference.html#kristensen2014-iw" id="id18">[16]</a>. There is now the opportunity for genome-wide phenotype predictions to do the same. Obtaining an accurate prediction of phenotype and protein function, even for a subclass of variants/proteins, has the potential to greatly impact people. A method which predicts phenotype directly from the molecular biology would be desirable as it would create a model for how phenotype emerges from genotype, in addition to predictions for important genes.</p>
<hr class="docutils" />
<p><strong>Chapter References</strong></p>
<p id="bibtex-bibliography-c03-compbio-background/3-bias-0"><dl class="citation">
<dt class="bibtex label" id="zhou2019-jk"><span class="brackets"><a class="fn-backref" href="#id15">4</a></span></dt>
<dd><p>Naihui Zhou, Yuxiang Jiang, Timothy R Bergquist, Alexandra J Lee, Balint Z Kacsoh, Alex W Crocker, Kimberley A Lewis, George Georghiou, Huy N Nguyen, Md Nafiz Hamid, Larry Davis, Tunca Dogan, Volkan Atalay, Ahmet S Rifaioglu, Alperen Dalkiran, Rengul Cetin-Atalay, Chengxin Zhang, Rebecca L Hurto, Peter L Freddolino, Yang Zhang, Prajwal Bhat, Fran Supek, José M Fernández, Branislava Gemovic, Vladimir R Perovic, Radoslav S Davidović, Neven Sumonja, Nevena Veljkovic, Ehsaneddin Asgari, Mohammad R K Mofrad, Giuseppe Profiti, Castrense Savojardo, Pier Luigi Martelli, Rita Casadio, Florian Boecker, Indika Kahanda, Natalie Thurlby, Alice C McHardy, Alexandre Renaux, Rabie Saidi, Julian Gough, Alex A Freitas, Magdalena Antczak, Fabio Fabris, Mark N Wass, Jie Hou, Jianlin Cheng, Zheng Wang, Alfonso E Romero, Alberto Paccanaro, Haixuan Yang, Tatyana Goldberg, Chenguang Zhao, Liisa Holm, Petri Törönen, Alan J Medlar, Elaine Zosa, Itamar Borukhov, Ilya Novikov, Angela Wilkins, Olivier Lichtarge, Po-Han Chi, Wei-Cheng Tseng, Michal Linial, Peter W Rose, Christophe Dessimoz, Vedrana Vidulin, Saso Dzeroski, Ian Sillitoe, Sayoni Das, Jonathan Gill Lees, David T Jones, Cen Wan, Domenico Cozzetto, Rui Fa, Mateo Torres, Alex Wiarwick Vesztrocy, Jose Manuel Rodriguez, Michael L Tress, Marco Frasca, Marco Notaro, Giuliano Grossi, Alessandro Petrini, Matteo Re, Giorgio Valentini, Marco Mesiti, Daniel B Roche, Jonas Reeb, David W Ritchie, Sabeur Aridhi, Seyed Ziaeddin Alborzi, Marie-Dominique Devignes, Koo Da Chen Emily, Richard Bonneau, Vladimir Gligorijević, Meet Barot, Hai Fang, Stefano Toppo, Enrico Lavezzo, Marco Falda, Michele Berselli, Silvio C E Tosatto, Marco Carraro, Damiano Piovesan, Hafeez Ur Rehman, Qizhong Mao, Shanshan Zhang, Slobodan Vucetic, Gage S Black, Dane Jo, Dallas J Larsen, Ashton R Omdahl, Luke W Sagers, Erica Suh, Jonathan B Dayton, Liam J McGuffin, Danielle A Brackenridge, Patricia C Babbitt, Jeffrey M Yunes, Paolo Fontana, Feng Zhang, Shanfeng Zhu, Ronghui You, Zihan Zhang, Suyang Dai, Shuwei Yao, Weidong Tian, Renzhi Cao, Caleb Chandler, Miguel Amezola, Devon Johnson, Jia-Ming Chang, Wen-Hung Liao, Yi-Wei Liu, Stefano Pascarelli, Yotam Frank, Robert Hoehndorf, Maxat Kulmanov, Imane Boudellioua, Gianfranco Politano, Stefano Di Carlo, Alfredo Benso, Kai Hakala, Filip Ginter, Farrokh Mehryary, Suwisa Kaewphan, Jari Björne, Hans Moen, Martti E E Tolvanen, Tapio Salakoski, Daisuke Kihara, Aashish Jain, Tomislav Šmuc, Adrian Altenhoff, Asa Ben-Hur, Burkhard Rost, Steven E Brenner, Christine A Orengo, Constance J Jeffery, Giovanni Bosco, Deborah A Hogan, Maria J Martin, Claire O’Donovan, Sean D Mooney, Casey S Greene, Predrag Radivojac, and Iddo Friedberg. The CAFA challenge reports improved protein function prediction and new functional annotations for hundreds of genes through experimental screens. <em>bioRxiv</em>, pages 653105, May 2019.</p>
</dd>
<dt class="bibtex label" id="begley2012-oc"><span class="brackets">2</span><span class="fn-backref">(<a href="#id1">1</a>,<a href="#id2">2</a>)</span></dt>
<dd><p>C Glenn Begley and Lee M Ellis. Drug development: raise standards for preclinical cancer research. <em>Nature</em>, 483(7391):531–533, March 2012. URL: <a class="reference external" href="http://dx.doi.org/10.1038/483531a">http://dx.doi.org/10.1038/483531a</a>.</p>
</dd>
<dt class="bibtex label" id="prinz2011-rh"><span class="brackets"><a class="fn-backref" href="#id3">3</a></span></dt>
<dd><p>Florian Prinz, Thomas Schlange, and Khusru Asadullah. Believe it or not: how much can we rely on published data on potential drug targets? <em>Nat. Rev. Drug Discov.</em>, 10(9):712–712, 2011. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrd3439-c1">http://dx.doi.org/10.1038/nrd3439-c1</a>.</p>
</dd>
<dt class="bibtex label" id="open-science-collaboration2015-vi"><span class="brackets"><a class="fn-backref" href="#id4">4</a></span></dt>
<dd><p>Open Science Collaboration. PSYCHOLOGY. estimating the reproducibility of psychological science. <em>Science</em>, 349(6251):aac4716, August 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.aac4716">http://dx.doi.org/10.1126/science.aac4716</a>.</p>
</dd>
<dt class="bibtex label" id="baker2016-tb"><span class="brackets"><a class="fn-backref" href="#id5">5</a></span></dt>
<dd><p>Monya Baker. 1,500 scientists lift the lid on reproducibility. <em>Nature</em>, 533(7604):452–454, May 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1038/533452a">http://dx.doi.org/10.1038/533452a</a>.</p>
</dd>
<dt class="bibtex label" id="bishop2019-hg"><span class="brackets"><a class="fn-backref" href="#id6">6</a></span></dt>
<dd><p>Dorothy Bishop. Rein in the four horsemen of irreproducibility. <em>Nature</em>, 568(7753):435, April 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1038/d41586-019-01307-2">http://dx.doi.org/10.1038/d41586-019-01307-2</a>.</p>
</dd>
<dt class="bibtex label" id="ioannidis2005-mo"><span class="brackets">7</span><span class="fn-backref">(<a href="#id7">1</a>,<a href="#id8">2</a>)</span></dt>
<dd><p>John P A Ioannidis. Why most published research findings are false. <em>PLoS Med.</em>, 2(8):e124, August 2005. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pmed.0020124">http://dx.doi.org/10.1371/journal.pmed.0020124</a>.</p>
</dd>
<dt class="bibtex label" id="head2015-ns"><span class="brackets"><a class="fn-backref" href="#id9">8</a></span></dt>
<dd><p>Megan L Head, Luke Holman, Rob Lanfear, Andrew T Kahn, and Michael D Jennions. The extent and consequences of p-hacking in science. <em>PLoS Biol.</em>, 13(3):e1002106, March 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1371/journal.pbio.1002106">http://dx.doi.org/10.1371/journal.pbio.1002106</a>.</p>
</dd>
<dt class="bibtex label" id="fisher1990-ro"><span class="brackets"><a class="fn-backref" href="#id10">9</a></span></dt>
<dd><p>R A Fisher and F Yates. <em>Statistical Methods, Experimental Design, and Scientific Inference: A Re-Issue of Statistical Methods for Research Workers, the Design of Experiments, and Statistical Methods and Scientific Inference</em>. OUP Oxford, April 1990. URL: <a class="reference external" href="https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290">https://www.amazon.co.uk/Statistical-Methods-Experimental-Scientific-Inference/dp/0198522290</a>.</p>
</dd>
<dt class="bibtex label" id="dunn1958-sj"><span class="brackets"><a class="fn-backref" href="#id11">10</a></span></dt>
<dd><p>Olive Jean Dunn. Estimation of the means of dependent variables. <em>The Annals of Mathematical Statistics</em>, 29(4):1095–1111, 1958. URL: <a class="reference external" href="http://dx.doi.org/10.1214/aoms/1177706443">http://dx.doi.org/10.1214/aoms/1177706443</a>.</p>
</dd>
<dt class="bibtex label" id="benjamini1995-me"><span class="brackets"><a class="fn-backref" href="#id12">11</a></span></dt>
<dd><p>Yoav Benjamini and Yosef Hochberg. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal Statistical Society: Series B (Methodological)</em>, 57(1):289–300, 1995. URL: <a class="reference external" href="http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x">http://dx.doi.org/10.1111/j.2517-6161.1995.tb02031.x</a>.</p>
</dd>
<dt class="bibtex label" id="barbaric2007-zm"><span class="brackets"><a class="fn-backref" href="#id13">12</a></span></dt>
<dd><p>Ivana Barbaric, Gaynor Miller, and T Neil Dear. Appearances can be deceiving: phenotypes of knockout mice. <em>Brief. Funct. Genomic. Proteomic.</em>, 6(2):91–103, June 2007.</p>
</dd>
<dt class="bibtex label" id="hardwicke-undated-jj"><span class="brackets"><a class="fn-backref" href="#id14">13</a></span></dt>
<dd><p>Tom Elis Hardwicke and John Ioannidis. Mapping the universe of registered reports. <em>Nature Human Behaviour</em>, 2018. URL: <a class="reference external" href="https://www.nature.com/articles/s41562-018-0444-y">https://www.nature.com/articles/s41562-018-0444-y</a>.</p>
</dd>
<dt class="bibtex label" id="hasin2017-tk"><span class="brackets"><a class="fn-backref" href="#id16">14</a></span></dt>
<dd><p>Yehudit Hasin, Marcus Seldin, and Aldons Lusis. Multi-omics approaches to disease. <em>Genome Biol.</em>, 2017. URL: <a class="reference external" href="http://dx.doi.org/10.1186/s13059-017-1215-1">http://dx.doi.org/10.1186/s13059-017-1215-1</a>.</p>
</dd>
<dt class="bibtex label" id="ritchie2015-ds"><span class="brackets"><a class="fn-backref" href="#id17">15</a></span></dt>
<dd><p>Marylyn D Ritchie, Emily R Holzinger, Ruowang Li, Sarah A Pendergrass, and Dokyoon Kim. Methods of integrating data to uncover genotype-phenotype interactions. <em>Nat. Rev. Genet.</em>, 16(2):85–97, February 2015. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrg3868">http://dx.doi.org/10.1038/nrg3868</a>.</p>
</dd>
<dt class="bibtex label" id="kristensen2014-iw"><span class="brackets"><a class="fn-backref" href="#id18">16</a></span></dt>
<dd><p>Vessela N Kristensen, Ole Christian Lingjærde, Hege G Russnes, Hans Kristian M Vollan, Arnoldo Frigessi, and Anne-Lise Børresen-Dale. Principles and methods of integrative genomic analyses in cancer. <em>Nat. Rev. Cancer</em>, 14(5):299–313, May 2014. URL: <a class="reference external" href="http://dx.doi.org/10.1038/nrc3721">http://dx.doi.org/10.1038/nrc3721</a>.</p>
</dd>
</dl>
</p>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c03-compbio-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="2-comp-bio-methods.html" title="previous page"><span class="section-number">3.2. </span>How large computational biology datasets are used</a>
    <a class='right-next' id="next-link" href="../c04-snowflake/0-index.html" title="next page"><span class="section-number">4. </span>Phenotype prediction with <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code></a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>