
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>1. Thesis style and philosophy &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="2. How phenotype arises from genotype" href="../c02-biology-background/0-index.html" />
    <link rel="prev" title="Table of Contents" href="../c0-front-matter/05-full-table-of-contents.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#unusual-stylistic-choices-in-this-thesis">
   1.1. Unusual stylistic choices in this thesis
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#research-philosophy">
   1.2. Research philosophy
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#complexity-science-systems-biology-and-multi-omics">
     1.2.1. Complexity science, systems biology and multi-omics
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#team-science">
     1.2.2. Team science
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#open-and-reproducible-science">
     1.2.3. Open and reproducible science
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Thesis style and philosophy</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#unusual-stylistic-choices-in-this-thesis">
   1.1. Unusual stylistic choices in this thesis
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#research-philosophy">
   1.2. Research philosophy
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#complexity-science-systems-biology-and-multi-omics">
     1.2.1. Complexity science, systems biology and multi-omics
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#team-science">
     1.2.2. Team science
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#open-and-reproducible-science">
     1.2.3. Open and reproducible science
    </a>
   </li>
  </ul>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="section" id="thesis-style-and-philosophy">
<span id="c01-philosophy"></span><h1><span class="section-number">1. </span>Thesis style and philosophy<a class="headerlink" href="#thesis-style-and-philosophy" title="Permalink to this headline">¶</a></h1>
<p>The abstract and background chapters provide the scientific introduction to this thesis, describing the research aims and context.
This short chapter, on the other hand, explains the choices of style and research philosophy.</p>
<div class="section" id="unusual-stylistic-choices-in-this-thesis">
<span id="unusual-things"></span><h2><span class="section-number">1.1. </span>Unusual stylistic choices in this thesis<a class="headerlink" href="#unusual-stylistic-choices-in-this-thesis" title="Permalink to this headline">¶</a></h2>
<p>There are a few unusual things about the format of this thesis: some easter eggs that made it easier for me to push through and finish it, and which also represent some of the things I like most about research: making it transparent, inclusive, and accessible.</p>
<div class="margin sidebar" id="bristol-crest">
<p class="sidebar-title">Bristol Crest</p>
<a class="reference internal image-reference" href="../_images/bristol-crest.png"><img alt="../_images/bristol-crest.png" class="align-center" src="../_images/bristol-crest.png" style="width: 185px;" /></a>
<p>The official University of Bristol crest (above) has symbols for the Wills, Fry, and Colston families. These families made their wealth in industries built on slavery and used some of that wealth to found the University of Bristol.</p>
<a class="reference internal image-reference" href="../_images/bristol-alt-crest-red.png"><img alt="../_images/bristol-alt-crest-red.png" class="align-center" src="../_images/bristol-alt-crest-red.png" style="width: 185px;" /></a>
<p>I drew an alternative crest which has symbols for three of my favourite Bristol festivals instead: Upfest, the Balloon Fiesta, and St Paul’s Carnival.</p>
</div>
<p>First, in the spirit of trying to make my work as reproducible and Open as possible, this thesis is available online as a Jupyter Book<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id277"><span>1</span></a>]</span> <a class="reference external" href="https://nataliethurlby.github.io/phenotype_from_genotype/">here</a>.
I would recommend reading it online unless you really like your PDF viewer, since it includes some interactive features which don’t translate to PDF.
This book was written entirely in markdown documents and Jupyter Notebooks - which means that most of the graphs within are created directly from these notebooks.</p>
<p>The second unusual thing about this thesis is that you will see asides mentioning researchers who have been involved in eugenics and/or racism.
This is the unfortunate reality of much of the history of the field, and I didn’t want to highlight the scientific achievements of these individuals, without also acknowledging their legacy of scientific racism, particularly in the light of the Black Lives Matter movement.</p>
<div class="margin sidebar" id="illustrations">
<p class="sidebar-title">Illustrations</p>
<p>The illustrations are CC-BY licensed (use freely, with attribution) in case they are useful to anyone.</p>
</div>
<p>For the same reason, I also drew an alternative University of Bristol crest, which you can see on the title page and in <a class="reference internal" href="#bristol-crest"><span class="std std-ref">this</span></a> margin comment.
This is also an example of the third weird thing about this thesis, which is that I drew some <a class="reference internal" href="#illustrations"><span class="std std-ref">illustrations</span></a> for it (using Krita<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id194"><span>6</span></a>]</span>), particularly in the background chapters.
My aim in including the majority of these drawings was simply to illustrate concepts, and help the reader (and myself) imagine some of the incredible stuff that is going on in all of our bodies.</p>
</div>
<div class="section" id="research-philosophy">
<span id="id3"></span><h2><span class="section-number">1.2. </span>Research philosophy<a class="headerlink" href="#research-philosophy" title="Permalink to this headline">¶</a></h2>
<p>Here I explain a little about my approach to the work in this thesis.
I’m including this to add clarity about the lenses through which I did this work, as well as what I consider to be a scientific contribution.</p>
<div class="section" id="complexity-science-systems-biology-and-multi-omics">
<span id="rp-complexity-science"></span><h3><span class="section-number">1.2.1. </span>Complexity science, systems biology and multi-omics<a class="headerlink" href="#complexity-science-systems-biology-and-multi-omics" title="Permalink to this headline">¶</a></h3>
<p>This PhD was completed as part of the Bristol Centre for Complexity Science.
Complexity science is the study of systems of interacting parts, i.e. the study of parts of the world where reductionism breaks down.
Typical applications of complexity science are predator-prey models, epidemiological modelling (e.g. of pandemics), protein-protein interaction networks, or models of neurons.
When applied to biology, it often falls under the banner of Systems Biology.</p>
<div class="margin sidebar">
<p class="sidebar-title"></p>
<div class="admonition-emergence admonition" id="emergence">
<p class="admonition-title">Emergence</p>
<p>Emergent properties of systems are properties that are found only when the consistuent entities of those systems interact, for example traffic jams emerge when vehicles interact on a network of roads<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id228"><span>7</span></a>]</span>, or cheetah’s spots emerge when chemicals diffuse across cells<span id="id5">[<a class="reference internal" href="../cz-end-matter/reference.html#id229"><span>8</span></a>]</span>.</p>
</div>
</div>
<p>Everything in this thesis looks at biology at the level of whole genomes, whole organisms, whole species, or even across the tree of life, and takes the view that this is necessary if we want to understand <a class="reference internal" href="#emergence"><span class="std std-ref">emergent</span></a> properties of these interactions.
Where I do “zoom in” to a particular case study or the details of some data, I am usually doing so either to understand how the complex systems approach is working, or in order to feed back and improve the resources that make the systems approach possible.</p>
<p>The work in this thesis could also be considered multi-omics.
I integrate, combine and harmonise some of the large, collaborative “omics” (e.g. genomics, proteomics) data projects that are available in this field, to create new resources and make new predictions.</p>
<p>A complex systems approach does not mean taking into account all parts of a system.
In modelling our infinitely complicated reality, we have to simplify to some extent, whether this means not taking properties such as location or speed of reactions into account, not taking certain entities or classes of processes into account.
Discovering what must be included and what can be left out is one outcome of this type of research.</p>
</div>
<div class="section" id="team-science">
<span id="rp-team-science"></span><h3><span class="section-number">1.2.2. </span>Team science<a class="headerlink" href="#team-science" title="Permalink to this headline">¶</a></h3>
<p>Computational biology is a field which I see as characterised by excellent examples of team science, from the Human Genome Project to Biomedical Ontologies.
I think that the best progress can be made when we all work together to create robust resources and build on each other’s work and are fairly credited for that.
I recognise that not everyone shares this view, computational biologists who use the results of other scientists experiments have been referred to by some as <a class="reference external" href="https://www.nejm.org/doi/full/10.1056/nejme1516564">“research parasites”</a>, or seen as a branch of IT services to whom “real” scientists can export technical work.
In this thesis, however, I take the view that contributing to existing scientific resources (Open Source or curated information), software engineering, writing, coming up with hypotheses all fall under the banner of scientific contribution.
This is in line with policies like the <a class="reference external" href="https://www.elsevier.com/authors/policies-and-guidelines/credit-author-statement">Contributor Roles Taxonomy</a>, and the recent additions of data sets and <a class="reference external" href="https://www.software.ac.uk/REF2021guidance">research software</a> to the Research Excellent Framework (REF) research outputs.</p>
</div>
<div class="section" id="open-and-reproducible-science">
<span id="rp-open-reproducible"></span><h3><span class="section-number">1.2.3. </span>Open and reproducible science<a class="headerlink" href="#open-and-reproducible-science" title="Permalink to this headline">¶</a></h3>
<p>During my PhD, one aspect of research came as quite a surprise to me, which was that I couldn’t trust the results of peer-reviewed papers to the extent that I originally assumed.
As the <a class="reference internal" href="../c03-compbio-background/5-bias.html#reproducibility-crisis"><span class="std std-ref">reproducibility crisis</span></a> unfolded, it became clear that the “untrustworthiness” of research was also an issue for many other researchers.
Since then, it has been an important part of the way that I do research, and I have made as much of my work as reproducible and Open as I could, taking new items from the Buffet of Open Research<span id="id6">[<a class="reference internal" href="../cz-end-matter/reference.html#id230"><span>9</span></a>]</span> as I continued.
I describe the ways in which I do this as they come up.
Some earlier parts of my work, particularly <a class="reference internal" href="../c04-snowflake/0-index.html#c04-snowflake"><span class="std std-numref">chapter 4</span></a> do remain closed-source.</p>
<p>We might like to think that scientific research is the “view from nowhere”, that it is objective, and simply measuring reality.
The reality, however, is that there are many, often equally valid, decisions to make when doing research and these decisions will impact the results of that research (as I will explain later).
If these decisions are not documented, the validity of the research can be obscured, and in addition, one aspect of scientific work that makes it difficult for other people to build on is that materials or details of analyses are not freely shared.
For these reasons, in this thesis, I tried to include enough detail so that the work could be reproduced (repeated to get the same answer), and the decisions made understood.</p>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c01-introduction"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="../c0-front-matter/05-full-table-of-contents.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Table of Contents</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../c02-biology-background/0-index.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">2. </span>How phenotype arises from genotype</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Natalie Zelenka<br/>
    
        &copy; Copyright 2020.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>