
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>5.3. Input Data &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="5.4. Methods" href="5-methods.html" />
    <link rel="prev" title="5.2. The uberon-py package" href="2-uberon.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c0-front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-uberon.html">
     5.2. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     5.3. Input Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-methods.html">
     5.4. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html">
     5.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     5.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="8-future-work.html">
     5.7. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/1-background.html">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/5-batch-correction.html">
     6.4. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html">
     6.5. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html#combined-data-set">
     6.6. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/7-discussion.html">
     6.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/8-future-work.html">
     6.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#expression-data-fantom5">
   5.3.1. Expression data: FANTOM5
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-files-and-acquisition">
     5.3.1.1. Data files and acquisition
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#initial-fantom5-data-cleaning-sample-info-file">
     5.3.1.2. Initial FANTOM5 data cleaning: sample info file
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#sample-categories">
       5.3.1.2.1. Sample categories
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#technical-and-biological-replicates">
       5.3.1.2.2. Technical and biological replicates
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#age-and-age-range">
       5.3.1.2.3. Age and age range
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#sex">
       5.3.1.2.4. Sex
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#disease-and-tissue-mapping">
       5.3.1.2.5. Disease and tissue mapping
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#initial-fantom5-data-cleaning-expression-file">
     5.3.1.3. Initial FANTOM5 data cleaning: expression file
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#cage-peaks-with-associated-proteins">
       5.3.1.3.1. CAGE peaks with associated proteins
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#cage-peaks-mapped-to-one-gene-only">
       5.3.1.3.2. CAGE peaks mapped to one gene only
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#proteins-that-map-to-multiple-genes">
       5.3.1.3.3. Proteins that map to multiple genes
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#exploratory-data-analysis">
     5.3.1.4. Exploratory Data Analysis
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#samples">
       5.3.1.4.1. Samples
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#protein-centric-tpm">
       5.3.1.4.2. Protein-centric TPM
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#cell-tissue-and-phenotype-mapping-data">
   5.3.2. Cell, tissue, and phenotype mapping data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#training-set-cafa2">
   5.3.3. “Training”” set: CAFA2
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what">
     5.3.3.1. What?
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#why">
     5.3.3.2. Why?
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#how">
     5.3.3.3. How?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#test-set-cafa3">
   5.3.4. Test set: CAFA3
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="input-data">
<h1><span class="section-number">5.3. </span>Input Data<a class="headerlink" href="#input-data" title="Permalink to this headline">¶</a></h1>
<p>There are several types of input data that are needed to run <code class="docutils literal notranslate"><span class="pre">filip</span></code>. These include:</p>
<ul class="simple">
<li><p><a class="reference internal" href="#fantom5-expression-data"><span class="std std-ref">Expression Data</span></a> (and related sample metadata)</p></li>
<li><p><a class="reference internal" href="#cell-tissue-phen-map-data"><span class="std std-ref">Cell, tissue, and phenotype mapping data</span></a></p></li>
<li><p><a class="reference internal" href="#cafa2-training-set"><span class="std std-ref">Training data</span></a> (ground-truth data for use in developing <code class="docutils literal notranslate"><span class="pre">filip</span></code>)</p></li>
<li><p><a class="reference internal" href="#cafa3-test-set"><span class="std std-ref">Test data</span></a> (ground-truth data for testing <code class="docutils literal notranslate"><span class="pre">filip</span></code>)</p></li>
</ul>
<p>In addition to this, <code class="docutils literal notranslate"><span class="pre">filip</span></code> requires the input of a protein function or phenotype prediction method, but this (and the data required for this) is described in <a class="reference internal" href="5-methods.html#filter-methods"><span class="std std-ref">Methods</span></a>.</p>
<div class="section" id="expression-data-fantom5">
<span id="fantom5-expression-data"></span><h2><span class="section-number">5.3.1. </span>Expression data: FANTOM5<a class="headerlink" href="#expression-data-fantom5" title="Permalink to this headline">¶</a></h2>
<p>The <code class="docutils literal notranslate"><span class="pre">filip</span></code> method requires expression data to inform whether or not predictions should be filtered out.
The FANTOM5 data set was chosen for this purpose.
FANTOM5 represents one of the most comprehensive collections of expression data in terms of tissue and cell type.
It consists of transcript expression data, captured using the <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html#cage-method"><span class="std std-ref">CAGE technique</span></a>.
FANTOM5 collected a combination of human, mouse, health, and disease data, as well as time courses and cell perturbations.
At the time of developing it was the latest data output of the <a class="reference internal" href="#fantom-consortium"><span class="std std-ref">FANTOM consortium</span></a>.</p>
<div class="margin sidebar" id="fantom-consortium">
<p class="sidebar-title">The FANTOM Consortium</p>
<p>The Functional ANnoTation Of the MAmmalian genome (FANTOM) consortium was established as the human genome project was nearing completion when researchers had a parts list of human biology, but few of the functions of these parts (genes) were known. The consortium has run a range of large scale collaborative projects in five rounds to further this goal. The first FANTOM project used only the mouse genome, but later versions also included human.</p>
</div>
<p>My reasoning for choosing FANTOM5 data as the input gene expression data to test <code class="docutils literal notranslate"><span class="pre">filip</span></code> was:</p>
<ul class="simple">
<li><p>The data set has a good coverage of different tissue types, meaning that <code class="docutils literal notranslate"><span class="pre">filip</span></code> should be able to turn this into a good coverage of predictions.</p></li>
<li><p>The data set has an ontology of samples, which is already linked to Uberon tissue terms and CL cell terms, making the mapping process much easier.</p></li>
</ul>
<p>I chose the version of the FANTOM5 data that:</p>
<ul class="simple">
<li><p>had been reprocessed using the hg38 reference genome (the original FANTOM5 data was processed using hg19).</p></li>
<li><p>contained annotated information about the samples, as this information could be used to aid in mapping</p></li>
<li><p>available in TPM format</p></li>
</ul>
<div class="section" id="data-files-and-acquisition">
<h3><span class="section-number">5.3.1.1. </span>Data files and acquisition<a class="headerlink" href="#data-files-and-acquisition" title="Permalink to this headline">¶</a></h3>
<div class="margin sidebar" id="fantom-accession">
<p class="sidebar-title">FANTOM5 Accession numbers</p>
<p>Each FANTOM <em>sample</em> has an accession number of the form <code class="docutils literal notranslate"><span class="pre">FF:#####-#####</span></code>. These numbers are used in all three of the FANTOM5 data files.
Note: some samples have repeat measurements per sample.</p>
</div>
<p>I downloaded the following files from the FANTOM website:</p>
<ul class="simple">
<li><p>the <a class="reference external" href="http://fantom.gsc.riken.jp/5/datafiles/reprocessed/hg38_latest/extra/CAGE_peaks_expression/hg38_fair+new_CAGE_peaks_phase1and2_counts_ann.osc.txt.gz">FANTOM5 CAGE peaks expression data</a> containing expression in counts per transcript, and mappings to HGNC id and entrez gene ID. The long sample labels in this file are also a source of metadata (including <a class="reference internal" href="#fantom-accession"><span class="std std-ref">sample identifiers (FANTOM accession numbers)</span></a>).</p></li>
<li><p>FANTOM’s <a class="reference external" href="https://fantom.gsc.riken.jp/5/datafiles/reprocessed/hg38_latest/basic/HumanSamples2.0.sdrf.xlsx">human sample information file</a> containing text descriptions about sample, for example FANTOM accession numbers, tissue, age, sex, disease, etc, which is necessary for data cleaning.</p></li>
<li><p>the <a class="reference external" href="https://fantom.gsc.riken.jp/5/datafiles/latest/extra/Ontology/ff-phase2-170801.obo.txt">FANTOM5 ontology</a> containing an obo file mapping between FANTOM accession numbers, Uberon and cell ontology (CL) terms.</p></li>
</ul>
</div>
<div class="section" id="initial-fantom5-data-cleaning-sample-info-file">
<h3><span class="section-number">5.3.1.2. </span>Initial FANTOM5 data cleaning: sample info file<a class="headerlink" href="#initial-fantom5-data-cleaning-sample-info-file" title="Permalink to this headline">¶</a></h3>
<div class="cell tag_hide-input tag_remove-output docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">helper_c05.fantom_sample_clean</span> <span class="k">as</span> <span class="nn">fsc</span>
<span class="c1"># TODO: Add cell metadata to prevent output + make code dropdown</span>

<span class="c1"># read in CAGE header and samples info files:</span>
<span class="n">expression_header</span> <span class="o">=</span> <span class="n">fsc</span><span class="o">.</span><span class="n">read_CAGE_header</span><span class="p">()</span>
<span class="n">samples_info</span> <span class="o">=</span> <span class="n">fsc</span><span class="o">.</span><span class="n">read_samples_info</span><span class="p">()</span>

<span class="c1"># calculate replicates information:</span>
<span class="n">rep_info</span> <span class="o">=</span> <span class="n">fsc</span><span class="o">.</span><span class="n">get_tech_bio_reps</span><span class="p">(</span><span class="n">expression_header</span><span class="p">,</span> <span class="n">samples_info</span><span class="p">)</span>

<span class="c1"># clean samples info files:</span>
<span class="n">samples_info</span> <span class="o">=</span> <span class="n">fsc</span><span class="o">.</span><span class="n">restrict_samples</span><span class="p">(</span><span class="n">samples_info</span><span class="p">)</span>
<span class="n">samples_info</span> <span class="o">=</span> <span class="n">fsc</span><span class="o">.</span><span class="n">update_sample_info_labels</span><span class="p">(</span><span class="n">samples_info</span><span class="p">,</span> <span class="n">rep_info</span><span class="p">)</span>
<span class="n">samples_info</span> <span class="o">=</span> <span class="n">fsc</span><span class="o">.</span><span class="n">clean_samples_info</span><span class="p">(</span><span class="n">samples_info</span><span class="p">,</span> <span class="n">rep_info</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="sample-categories">
<span id="fantom-sample-categories"></span><h4><span class="section-number">5.3.1.2.1. </span>Sample categories<a class="headerlink" href="#sample-categories" title="Permalink to this headline">¶</a></h4>
<div class="margin sidebar">
<p class="sidebar-title">HeLa cell line</p>
<p>The FANTOM5 experiment contains HeLa cell lines samples (e.g. sample <code class="docutils literal notranslate"><span class="pre">FF:10815-111B5</span></code>).</p>
<p>HeLa is short for Henrietta Lacks, the woman whose cells were the source of this first immortal cell line.
Henrietta was a black woman who lived in Baltimore, Maryland.
Her cells were taken without consent during a hospital biopsy for an aggressive cervical cancer, which she died from at age 31 in 1951.</p>
<p>Companies continue to profit from the sale of these lines of cells, since such cell lines have several practical advantages over primary cells, notably their immortality, low variability (compared to primary cells, which vary depending on cell donor characteristics such as age and sex), and easiness to keep alive (without the need for e.g. additional nutrients).</p>
<p>Some companies have recently begun to pay reparations for this injustice<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id197"><span>143</span></a>]</span>.</p>
</div>
<p><strong>Restriction to  primary cell and tissue samples</strong>:</p>
<p>The human FANTOM5 sample information file contains four categories of samples (in the <code class="docutils literal notranslate"><span class="pre">Characteristics</span> <span class="pre">[Category]</span></code> field):</p>
<ul class="simple">
<li><p><strong>time courses</strong>: RNA extracted from samples being measured over time as cells change types during cell development and differentiation ( samples), e.g.  - <em></em>.</p></li>
<li><p><strong>primary cells</strong>: RNA extracted from cultures of cells recently isolated from tissues, before undergoing proliferation with nutrients specific to the cell type ( samples), e.g.  - <em></em>.</p></li>
<li><p><strong>cell lines</strong>: RNA extracted from immortal cell lines (which unlike primary cells) can keep undergoing division indefinitely ( samples), e.g.  - <em></em>.</p></li>
<li><p><strong>tissues</strong>: RNA extracted from post-mortem tissues, which may be pooled or individual donors ( samples), e.g.  - <em></em>.</p></li>
<li><p><strong>fractionations</strong>: RNA extracted from parts of cells (fractionations) ( samples), e.g.  - .</p></li>
</ul>
<p>I restricted the data set to only <em>tissues</em> and <em>primary cells</em>, as these represent the closest approximations to <em>in vivo</em> biology.
Immortal cell lines are often expressed differently than their primary counterparts<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id194"><span>144</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id195"><span>145</span></a>]</span>, and time courses and fractionations do not represent any particular tissue.</p>
<p><strong>Sample Type</strong>:</p>
<p>As mentioned tissues can come from a pool, or individual donor.
This information can be found in the <code class="docutils literal notranslate"><span class="pre">Charateristics</span> <span class="pre">[description]</span></code> field.
I combined this information with information from the <code class="docutils literal notranslate"><span class="pre">Characteristics</span> <span class="pre">[Category]</span></code> field to create an additional <code class="docutils literal notranslate"><span class="pre">Sample</span> <span class="pre">Type</span></code> field that describes whether a sample is a <code class="docutils literal notranslate"><span class="pre">tissue</span> <span class="pre">-</span> <span class="pre">pool</span></code>, <code class="docutils literal notranslate"><span class="pre">tissue</span> <span class="pre">-</span> <span class="pre">donor</span></code> or <code class="docutils literal notranslate"><span class="pre">primary</span> <span class="pre">cells</span></code> sample.</p>
</div>
<div class="section" id="technical-and-biological-replicates">
<h4><span class="section-number">5.3.1.2.2. </span>Technical and biological replicates<a class="headerlink" href="#technical-and-biological-replicates" title="Permalink to this headline">¶</a></h4>
<div class="margin sidebar" id="tech-biol-replicates">
<p class="sidebar-title">Technical and biological replicates</p>
<p>Usually <em>technical replicates</em> refer to repeated measures of the same sample, while <em>biological replicates</em> refer to separate samples which have been treated in the same way (e.g. different donors)<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id196"><span>146</span></a>]</span>.</p>
<p>In FANTOM, the “biol_rep” and “donor” label are both used to denote biological replicates.</p>
</div>
<p>The <a class="reference internal" href="#fantom-accession"><span class="std std-ref">FANTOM accession numbers</span></a> are per sample, not per measurement.
Samples for which there are repeat measurements (technical replicates) will show up multiple times in the expression file.
FANTOM technical and biological replicates are indicated in long labels of the annotated expression FANTOM file, by the inclusion of “tech_rep” or “biol_rep” in the long sample labels e.g. <code class="docutils literal notranslate"><span class="pre">tpm.Dendritic%20Cells%20-%20monocyte%20immature%20derived%2c%20donor1%2c%20tech_rep1.CNhs10855.11227-116C3.hg38.nobarcode</span></code>.
These were used to create additional fields for the human samples table.</p>
<p>Note: there is an error in the original transcript expression file for one of these identifiers (<code class="docutils literal notranslate"><span class="pre">tpm.Dendritic%20Cells%20-%20monocyte%20immature%20derived%2c%20donor1%2c%20rep2.CNhs11062.11227-116C3.hg38.nobarcode</span></code>) such that it is missing the “tech” part of the the replicate label.
There is a hard-coded fix when I read in the input file and the FANTOM data curation team was informed.</p>
<p>After restricting the dataset to <em>primary cell</em> and <em>tissue</em> type samples, there are  remaining samples which have biological replicates, and  sets of samples with technological replicates (2 replicates each).</p>
</div>
<div class="section" id="age-and-age-range">
<h4><span class="section-number">5.3.1.2.3. </span>Age and age range<a class="headerlink" href="#age-and-age-range" title="Permalink to this headline">¶</a></h4>
<p>The age of the sample source donor(s) is available through two fields in the human sample information file: <code class="docutils literal notranslate"><span class="pre">Characteristics</span> <span class="pre">[Developmental</span> <span class="pre">stage]</span></code>, and <code class="docutils literal notranslate"><span class="pre">Characteristics</span> <span class="pre">[Age]</span></code>.
These fields contain description-like text, which are somewhat inconsistent, for example, “3 year old child”, “3 years old child”, “25 year old”, “76” and “76 years old adult” all feature in the same column, amongst other errors.
These were standardised into a new field (<code class="docutils literal notranslate"><span class="pre">Age</span> <span class="pre">(years)</span></code>).
This field does not seek to include multiple ages (i.e. when the sample comes from a pool of donors).
There is a complementary (i.e. no overlap) field (<code class="docutils literal notranslate"><span class="pre">Age</span> <span class="pre">range</span> <span class="pre">(years)</span></code>), which contains age ranges for the  samples that contain multiple ages.
In both columns, some samples contain fetal samples, in which caset age (range) is given as a negative decimal (converted to years before birth).</p>
<p>There file also contained some discrepancies between ages and developmental stages in the FANTOM human samples file.
For example, sample <code class="docutils literal notranslate"><span class="pre">FF:10027-101D9</span></code> is labelled as <em>thymus, adult, pool1</em> in the <em>Description</em> field, but as <em>0.5,0.5,0.83 years old infant</em> in the <em>Developmental Stage</em> field.
Sample <code class="docutils literal notranslate"><span class="pre">FF:10209-103G2</span></code> had an age of ‘M’ and a sex of ‘28’.
I reported both these discrepencies: and the latter has since been fixed in the FANTOM file, and for the former, I hardcode the age to <code class="docutils literal notranslate"><span class="pre">NaN</span></code>.</p>
</div>
<div class="section" id="sex">
<h4><span class="section-number">5.3.1.2.4. </span>Sex<a class="headerlink" href="#sex" title="Permalink to this headline">¶</a></h4>
<p>The <code class="docutils literal notranslate"><span class="pre">Characteristics</span> <span class="pre">[Sex]</span></code> field contains information about the sex of the sample source donor(s).
Similarly to age, due to the consortium nature of FANTOM5, the entries of this field are not consistently labelled.
They undergo data cleaning into 4 categories: male, female, mixed (pool with both male and female samples), and unlabelled.</p>
</div>
<div class="section" id="disease-and-tissue-mapping">
<h4><span class="section-number">5.3.1.2.5. </span>Disease and tissue mapping<a class="headerlink" href="#disease-and-tissue-mapping" title="Permalink to this headline">¶</a></h4>
<p>The disease status of samples (e.g. healthy/non-healthy) is not straight-forwardly labelled in the human sample file, so requires some basic text-mining (and cross-referencing with ontology terms).
Similarly, there is a <code class="docutils literal notranslate"><span class="pre">Characteristics[Tissue]</span></code> field in the human samples file containing some manually mapped tissue types, but as I point out with an example in <a class="reference internal" href="#eda-sample-tissues"><span class="std std-ref">the exploratory data analysis</span></a>, these do not contain ideal mappings for <code class="docutils literal notranslate"><span class="pre">filip</span></code>.</p>
<p>The continued data processing of these components is described in <a class="reference internal" href="5-methods.html#filter-methods"><span class="std std-ref">the methdology section</span></a>, after the introduction of <code class="docutils literal notranslate"><span class="pre">uberon-py</span></code> (the package developed to do this).</p>
<div class="admonition-fantom5-cleaned-experimental-design-file admonition" id="cleaned-fantom-exp">
<p class="admonition-title">FANTOM5 cleaned experimental design file</p>
<p>The cleaned FANTOM5 experimental design file (which has undergone the cleaning mentioned in this section, and in <a class="reference internal" href="5-methods.html#filter-methods"><span class="std std-ref">the methodology section</span></a>) is available <span class="xref myst">here</span>.</p>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: Create list of allowed primary + tissue samples (in sample_header)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">helper_c05</span> <span class="kn">import</span> <span class="n">fantom_tpm_clean</span> <span class="k">as</span> <span class="n">tpm_clean</span>

<span class="c1">#tpm_file = &#39;../c06-combining/data/experiments/fantom/hg38_fair+new_CAGE_peaks_phase1and2_tpm_ann.osc.txt&#39;</span>
<span class="c1">#cage_tpm = tpm_clean.read_and_clean_tpm(tpm_file)</span>
<span class="c1">#cage_tpm</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#protein_tpm = tpm_clean.get_protein_tpm(cage_tpm)</span>
<span class="c1">#protein_tpm</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="initial-fantom5-data-cleaning-expression-file">
<h3><span class="section-number">5.3.1.3. </span>Initial FANTOM5 data cleaning: expression file<a class="headerlink" href="#initial-fantom5-data-cleaning-expression-file" title="Permalink to this headline">¶</a></h3>
<div class="section" id="cage-peaks-with-associated-proteins">
<h4><span class="section-number">5.3.1.3.1. </span>CAGE peaks with associated proteins<a class="headerlink" href="#cage-peaks-with-associated-proteins" title="Permalink to this headline">¶</a></h4>
<p>The CAGE peaks represent all kinds of transcripts, not only those which map to protein-coding genes.
The FANTOM file provides mappings to Uniprot IDs (<code class="docutils literal notranslate"><span class="pre">uniprot_id</span></code>), and these are used to discard the CAGE peaks that do not map to protein-coding genes: this takes us from  to   rows (CAGE peaks).</p>
</div>
<div class="section" id="cage-peaks-mapped-to-one-gene-only">
<h4><span class="section-number">5.3.1.3.2. </span>CAGE peaks mapped to one gene only<a class="headerlink" href="#cage-peaks-mapped-to-one-gene-only" title="Permalink to this headline">¶</a></h4>
<p>CAGE peaks are mapped to genes based on overlap with the gene, so it is not always clear which gene a CAGE peak maps to.
For simplicity, and to remove the potential of wrongly mapped genes being used in <code class="docutils literal notranslate"><span class="pre">filip</span></code>, protein-coding CAGE peaks (those which are mapped to at least one <code class="docutils literal notranslate"><span class="pre">uniprot_id</span></code> by FANTOM) that map to multiple genes are removed.
These can be found by looking at either the <code class="docutils literal notranslate"><span class="pre">hgnc_id</span></code> or <code class="docutils literal notranslate"><span class="pre">entrezgene_id</span></code> gene identifier columns.
The choice of gene ID matters, since there are discrepancies between gene ID databases: in this case, choosing <code class="docutils literal notranslate"><span class="pre">hgnc_id</span></code> finds all those CAGE peaks found by using <code class="docutils literal notranslate"><span class="pre">entrezgene_id</span></code>, and more, so these are removed.
This represents a total of  CAGE peaks that map to multiple genes according to the given identifiers.</p>
</div>
<div class="section" id="proteins-that-map-to-multiple-genes">
<h4><span class="section-number">5.3.1.3.3. </span>Proteins that map to multiple genes<a class="headerlink" href="#proteins-that-map-to-multiple-genes" title="Permalink to this headline">¶</a></h4>
<p>For <code class="docutils literal notranslate"><span class="pre">filip</span></code>, the expression was calculated per protein (since it is protein function predictions that it is filtering), rather than per CAGE peak (summing the TPMs of all CAGE peaks mapped to a protein to get the total for that protein) as in the original data, or as is often presented per gene.
This gave  rows of “protein expression” data.</p>
<p>Of these, there were then  rows of data (corresponding to  proteins) for which each protein maps to multiple genes.
This happens when different genes are translated to make identical protein products, for example the <a class="reference external" href="https://www.uniprot.org/uniprot/P62805">H4 human histone protein</a> is encoded by 14 different genes at different loci, across three different chromosomes.
It used to be the case that Uniprot would map these genes to the same Uniprot ID, but more recently different Uniprot IDs are used to capture where the proteins came from.
These rows were also removed.</p>
</div>
</div>
<div class="section" id="exploratory-data-analysis">
<h3><span class="section-number">5.3.1.4. </span>Exploratory Data Analysis<a class="headerlink" href="#exploratory-data-analysis" title="Permalink to this headline">¶</a></h3>
<div class="section" id="samples">
<h4><span class="section-number">5.3.1.4.1. </span>Samples<a class="headerlink" href="#samples" title="Permalink to this headline">¶</a></h4>
<p>After <a class="reference internal" href="#fantom-sample-categories"><span class="std std-ref">restricting the samples to those which are primary cells or tissues</span></a>, there were  remaining samples.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">helper_c05</span> <span class="kn">import</span> <span class="n">fantom_sample_eda</span> <span class="k">as</span> <span class="n">f_eda</span>
<span class="n">sex_donut</span><span class="p">,</span> <span class="n">tissues_samples</span><span class="p">,</span> <span class="n">nan_age_count</span><span class="p">,</span> <span class="n">collaborators_providers</span> <span class="o">=</span> <span class="n">f_eda</span><span class="o">.</span><span class="n">create_plot_dfs</span><span class="p">(</span><span class="n">samples_info</span><span class="p">)</span>
<span class="n">fig</span> <span class="o">=</span> <span class="n">f_eda</span><span class="o">.</span><span class="n">create_plotly_plots</span><span class="p">(</span><span class="n">samples_info</span><span class="p">,</span> <span class="n">sex_donut</span><span class="p">,</span> <span class="n">tissues_samples</span><span class="p">,</span> <span class="n">nan_age_count</span><span class="p">,</span> <span class="n">collaborators_providers</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">show</span><span class="p">(</span><span class="s1">&#39;notebook&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">NameError</span><span class="g g-Whitespace">                                 </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">dbe7181c6662</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="kn">from</span> <span class="nn">helper_c05</span> <span class="kn">import</span> <span class="n">fantom_sample_eda</span> <span class="k">as</span> <span class="n">f_eda</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">sex_donut</span><span class="p">,</span> <span class="n">tissues_samples</span><span class="p">,</span> <span class="n">nan_age_count</span><span class="p">,</span> <span class="n">collaborators_providers</span> <span class="o">=</span> <span class="n">f_eda</span><span class="o">.</span><span class="n">create_plot_dfs</span><span class="p">(</span><span class="n">samples_info</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">3</span> <span class="n">fig</span> <span class="o">=</span> <span class="n">f_eda</span><span class="o">.</span><span class="n">create_plotly_plots</span><span class="p">(</span><span class="n">samples_info</span><span class="p">,</span> <span class="n">sex_donut</span><span class="p">,</span> <span class="n">tissues_samples</span><span class="p">,</span> <span class="n">nan_age_count</span><span class="p">,</span> <span class="n">collaborators_providers</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">4</span> <span class="n">fig</span><span class="o">.</span><span class="n">show</span><span class="p">(</span><span class="s1">&#39;notebook&#39;</span><span class="p">)</span>

<span class="ne">NameError</span>: name &#39;samples_info&#39; is not defined
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="fantom-eda">
<img alt="../_images/blank.png" src="../_images/blank.png" />
<p class="caption"><span class="caption-number">Fig. 5.2 </span><span class="caption-text">(a) sex: a donut plot showing the sex labels of samples.
(b) collaborators and providers: a stacked histogram showing the  most common collaborators, and  most common providers.
(c) age: a histogram of age of sample donors (this does not include the  samples which have age ranges due to pooled donors of various ages).
(d) tissues and sample types: a histogram showing the  most common tissues, spread across the different types of samples (primary cells, tissue donors, and tissue pools).</span><a class="headerlink" href="#fantom-eda" title="Permalink to this image">¶</a></p>
</div>
<!-- ../images/blank.png This is a workaround to put a 1x1px blank image after an interactive image so that it appears to have a figure label -->
<p><strong>Sample metadata:</strong>
Looking at the FANTOM5 data (see <a class="reference internal" href="#fantom-eda"><span class="std std-numref">Fig. 5.2</span></a>), overall we see that there the samples are very varied, across ages, sex, sample providers, and collaborators, although (d) shows that the majority of samples are <em>primary cell</em> samples, and very few are <em>tissue - pool</em> samples.
Secondly, we can see that after careful cleaning, some metadata is missing, i.e. 38.4% of samples have unknown sex (a), most collaborators did not label the sample provider (b), and most samples do not have a labelled age (c).</p>
<p id="eda-sample-tissues"><strong>Sample Tissues:</strong>
In <a class="reference internal" href="#fantom-eda"><span class="std std-numref">Fig. 5.2</span></a> subplot (d), we can also note some interesting things about the tissue types provided by the Fantom Human Samples file.  primary cell samples are labeled <em>ANATOMICAL SYSTEM</em>. If we look closer at these samples, we can see that it is theoretically possible to map some of these samples to tissues (see <a class="reference internal" href="#anatomical-system-table"><span class="std std-numref">Fig. 5.3</span></a>).</p>
<p id="fantom-tissues-how-general">There is also the question of how general or specific the human sample categories are. There are  samples which are mapped to <em>blood</em> (<a class="reference internal" href="#anatomical-system-table"><span class="std std-numref">Fig. 5.3</span></a> (d)), but when we come to map the FANTOM5 tissues to phenotypes, this may be too broad a category. Similarly, there are  with less than three samples each (not pictured) that may be too narrow to map to phenotypes, and a more accurate picture of that phenotype would come from taking a more general tissue.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">anatomical_system_samples</span> <span class="o">=</span> <span class="n">f_eda</span><span class="o">.</span><span class="n">anat_system_tbl</span><span class="p">(</span><span class="n">samples_info</span><span class="p">,</span> <span class="n">chosen_samples</span> <span class="o">=</span> <span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="mi">15</span><span class="p">,</span> <span class="mi">20</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">NameError</span><span class="g g-Whitespace">                                 </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">7</span><span class="o">-</span><span class="mi">50626</span><span class="n">b1a5f3a</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="ne">----&gt; </span><span class="mi">1</span> <span class="n">anatomical_system_samples</span> <span class="o">=</span> <span class="n">f_eda</span><span class="o">.</span><span class="n">anat_system_tbl</span><span class="p">(</span><span class="n">samples_info</span><span class="p">,</span> <span class="n">chosen_samples</span> <span class="o">=</span> <span class="p">[</span><span class="mi">2</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="mi">15</span><span class="p">,</span> <span class="mi">20</span><span class="p">])</span>

<span class="ne">NameError</span>: name &#39;samples_info&#39; is not defined
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="anatomical-system-table" style="width: 800px">
<p class="caption"><span class="caption-number">Fig. 5.3 </span><span class="caption-text">An example of four <em>ANANTOMICAL SYSTEM</em> tissues, with tissue-specific cells, indicating that they could be mapped to tissues. For example sample <code class="docutils literal notranslate"><span class="pre">FF:11922-125H5</span></code> is a gingival fibroblast, which are one of the main constituent cells of gum tissue.</span><a class="headerlink" href="#anatomical-system-table" title="Permalink to this image">¶</a></p>
</div>
<p>We can also see in <a class="reference internal" href="#anatomical-system-table"><span class="std std-numref">Fig. 5.3</span></a> that this data set, though having undergone some data cleaning, still contains disease samples (e.g. “aggressive periodontitis”).</p>
</div>
<div class="section" id="protein-centric-tpm">
<h4><span class="section-number">5.3.1.4.2. </span>Protein-centric TPM<a class="headerlink" href="#protein-centric-tpm" title="Permalink to this headline">¶</a></h4>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">#display(transcript_tpm[transcript_tpm[&#39;uniprot_id&#39;].str.contains(&#39;B2R4R0&#39;, na=False)])</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: Plotly Gannt for CAGE peaks overlapping with multiple transcripts/genes (check association with transcript to check whether it&#39;s multiple transcripts or just muktiple genes with same TSS)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="cell-tissue-and-phenotype-mapping-data">
<span id="cell-tissue-phen-map-data"></span><h2><span class="section-number">5.3.2. </span>Cell, tissue, and phenotype mapping data<a class="headerlink" href="#cell-tissue-and-phenotype-mapping-data" title="Permalink to this headline">¶</a></h2>
<p>I also used the following datasets to aid in mapping to a common set of identifiers:</p>
<ul class="simple">
<li><p>the <a class="reference external" href="http://purl.obolibrary.org/obo/uberon/ext.obo">uberon extended ontology OBO file</a> from <span class="xref myst">the uberon website</span> to assist in mapping cells and tissues.</p></li>
</ul>
</div>
<div class="section" id="training-set-cafa2">
<span id="cafa2-training-set"></span><h2><span class="section-number">5.3.3. </span>“Training”” set: CAFA2<a class="headerlink" href="#training-set-cafa2" title="Permalink to this headline">¶</a></h2>
<div class="section" id="what">
<h3><span class="section-number">5.3.3.1. </span>What?<a class="headerlink" href="#what" title="Permalink to this headline">¶</a></h3>
<p>During development, I tested <code class="docutils literal notranslate"><span class="pre">filip</span></code> by comparing DcGO only and <code class="docutils literal notranslate"><span class="pre">filip</span></code> + DcGO on data from the 2nd round of the CAFA competition: CAFA2.
This was the most recent round of CAFA for which there were “groundtruth” data available at the time of development.
The data consisted of CAFA2 targets and the CAFA2 ground truth data.</p>
<p><strong>CAFA2 targets</strong>:</p>
<p><strong>CAFA2 groundtruth</strong>:</p>
</div>
<div class="section" id="why">
<h3><span class="section-number">5.3.3.2. </span>Why?<a class="headerlink" href="#why" title="Permalink to this headline">¶</a></h3>
<p>I chose to use the CAFA2 data because rather than a larger set of annotations (such as those available from SwissProt-KB or GOA) because it provided a way of validating on unknown targets.
I.e. if I made predictions with DcgO using the version of GO from the time the challenge was launched, and I use the groundtruth data provided by CAFA2, then I could compare my results with those in the CAFA2 competition and I could look at my results on unknown targets.</p>
</div>
<div class="section" id="how">
<h3><span class="section-number">5.3.3.3. </span>How?<a class="headerlink" href="#how" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="test-set-cafa3">
<span id="cafa3-test-set"></span><h2><span class="section-number">5.3.4. </span>Test set: CAFA3<a class="headerlink" href="#test-set-cafa3" title="Permalink to this headline">¶</a></h2>
<p>After initial development, I entered DcGO only, and <code class="docutils literal notranslate"><span class="pre">filip</span></code> plus DcGO into the CAFA3 competition in order to test <code class="docutils literal notranslate"><span class="pre">filip</span></code> on a new dataset.</p>
<p>This meant that I did not download the CAFA3 ground-truth, as this analysis was done by the CAFA3 team, but only the CAFA3 targets.</p>
<p><strong>CAFA3 targets</strong>:</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c05-filter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="2-uberon.html" title="previous page"><span class="section-number">5.2. </span>The <code class="docutils literal notranslate"><span class="pre">uberon-py</span></code> package</a>
    <a class='right-next' id="next-link" href="5-methods.html" title="next page"><span class="section-number">5.4. </span>Methods</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>