
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>5.3. Data &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.c441f2ba0852f4cabcb80105e3a46ae6.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="5.4. Methods" href="5-methods.html" />
    <link rel="prev" title="5.2. The uberon-py package" href="2-uberon.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c0-front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-uberon.html">
     5.2. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     5.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-methods.html">
     5.4. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html">
     5.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     5.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="8-future-work.html">
     5.7. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/1-background.html">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/5-batch-correction.html">
     6.4. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html">
     6.5. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html#combined-data-set">
     6.6. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/7-discussion.html">
     6.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/8-future-work.html">
     6.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#expression-data">
   5.3.1. Expression data
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#fantom5-data">
     5.3.1.1. FANTOM5 data
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#reasoning">
       5.3.1.1.1. Reasoning
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#additional-mapping-data">
     5.3.1.2. Additional mapping data
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-acquisition">
     5.3.1.3. Data Acquisition
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-cleaning">
     5.3.1.4. Data Cleaning
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#exploratory-data-analysis">
     5.3.1.5. Exploratory Data Analysis
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#tpm">
       5.3.1.5.1. TPM
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#counts">
       5.3.1.5.2. Counts
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#test-set-cafa2">
   5.3.2. Test set: CAFA2
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#what">
     5.3.2.1. What?
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#why">
     5.3.2.2. Why?
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#how">
     5.3.2.3. How?
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#test-set-cafa3">
   5.3.3. Test set: CAFA3
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <!-- #region -->
<div class="section" id="data">
<h1><span class="section-number">5.3. </span>Data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="expression-data">
<h2><span class="section-number">5.3.1. </span>Expression data<a class="headerlink" href="#expression-data" title="Permalink to this headline">¶</a></h2>
<div class="section" id="fantom5-data">
<h3><span class="section-number">5.3.1.1. </span>FANTOM5 data<a class="headerlink" href="#fantom5-data" title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">filip</span></code> requires expression data to inform whether or not predictions should be filtered out. The FANTOM5 data set was chosen for this purpose (at the time this was the latest data output of the <a class="reference internal" href="#fantom-consortium"><span class="std std-ref">FANTOM consortium</span></a>).</p>
<p>FANTOM5 represents one of the most comprehensive collections of expression data, in this case transcript rather than gene expression.
It contains a combination of human, mouse, health, and disease data, as well as time courses and cell perturbations.</p>
<div class="margin sidebar" id="fantom-consortium">
<p class="sidebar-title">The FANTOM Consortium</p>
<p>The Functional ANnoTation Of the MAmmalian genome (FANTOM) consortium was established as the human genome project was nearing completion when researchers had a parts list of human biology, but few of the functions of these parts (genes) were known. The consortium has run a range of large scale collaborative projects in five rounds to further this goal. The first FANTOM project used only the mouse genome, but later versions also included human.</p>
</div>
<div class="section" id="reasoning">
<h4><span class="section-number">5.3.1.1.1. </span>Reasoning<a class="headerlink" href="#reasoning" title="Permalink to this headline">¶</a></h4>
<p>I chose the FANTOM5 data as the input gene expression data for <code class="docutils literal notranslate"><span class="pre">filip</span></code>, for the following reasons:</p>
<ul class="simple">
<li><p>The data set has a good coverage of different tissue types, meaning that <code class="docutils literal notranslate"><span class="pre">filip</span></code> should be able to turn this into a good coverage of predictions.</p></li>
<li><p>The data set has an ontology of samples, which is already linked to Uberon tissue terms and CL cell terms, making the mapping process much easier.</p></li>
</ul>
<p>I chose the version of the FANTOM5 data that:</p>
<ul class="simple">
<li><p>had been reprocessed using the hg38 reference genome (the original FANTOM5 data was processed using hg19).</p></li>
<li><p>contained annotated information about the samples, as this information could be used to aid in mapping</p></li>
<li><p>was in TPM format</p></li>
</ul>
</div>
</div>
<div class="section" id="additional-mapping-data">
<h3><span class="section-number">5.3.1.2. </span>Additional mapping data<a class="headerlink" href="#additional-mapping-data" title="Permalink to this headline">¶</a></h3>
<p>I also used the following datasets to aid in mapping to a common set of identifiers:</p>
<ul class="simple">
<li><p>the <a class="reference external" href="http://purl.obolibrary.org/obo/uberon/ext.obo">uberon extended ontology OBO file</a> from <span class="xref myst">the uberon website</span> to assist in mapping cells and tissues.</p></li>
</ul>
</div>
<div class="section" id="data-acquisition">
<h3><span class="section-number">5.3.1.3. </span>Data Acquisition<a class="headerlink" href="#data-acquisition" title="Permalink to this headline">¶</a></h3>
<p>I downloaded the following files from the FANTOM website:</p>
<ul class="simple">
<li><p>the <a class="reference external" href="http://fantom.gsc.riken.jp/5/datafiles/reprocessed/hg38_latest/extra/CAGE_peaks_expression/hg38_fair+new_CAGE_peaks_phase1and2_counts_ann.osc.txt.gz">FANTOM5 CAGE peaks expression data</a> containing expression in counts per transcript, and mappings to HGNC id and entrez gene ID.</p></li>
<li><p>the <a class="reference external" href="https://fantom.gsc.riken.jp/5/datafiles/latest/extra/Ontology/ff-phase2-170801.obo.txt">FANTOM5 ontology</a> containing an obo file mapping between FANTOM sample IDs, Uberon and cell ontology (CL) terms.</p></li>
<li><p>FANTOM’s <a class="reference external" href="https://fantom.gsc.riken.jp/5/datafiles/reprocessed/hg38_latest/basic/HumanSamples2.0.sdrf.xlsx">human sample information file</a> containing text descriptions about sample, for example tissue, age, sex, disease, etc, which is necessary for data cleaning.</p></li>
</ul>
<!-- #endregion -->
</div>
<div class="section" id="data-cleaning">
<h3><span class="section-number">5.3.1.4. </span>Data Cleaning<a class="headerlink" href="#data-cleaning" title="Permalink to this headline">¶</a></h3>
<p>In order to clean the FANTOM data for use in <code class="docutils literal notranslate"><span class="pre">filip</span></code>, transcripts that didn’t match to proteins were removed, and the following samples were removed:</p>
<ul class="simple">
<li><p>Disease samples</p></li>
<li><p>Non tissue-specific samples</p></li>
<li><p>Non human samples</p></li>
</ul>
</div>
<div class="section" id="exploratory-data-analysis">
<h3><span class="section-number">5.3.1.5. </span>Exploratory Data Analysis<a class="headerlink" href="#exploratory-data-analysis" title="Permalink to this headline">¶</a></h3>
<div class="section" id="tpm">
<h4><span class="section-number">5.3.1.5.1. </span>TPM<a class="headerlink" href="#tpm" title="Permalink to this headline">¶</a></h4>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">myst_nb</span> <span class="kn">import</span> <span class="n">glue</span>

<span class="n">transcript_tpm</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;../c06-combining/data/experiments/fantom/hg38_fair+new_CAGE_peaks_phase1and2_tpm_ann.osc.txt&#39;</span><span class="p">,</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">comment</span><span class="o">=</span><span class="s1">&#39;#&#39;</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;entrezgene_id&#39;</span><span class="p">:</span> <span class="nb">object</span><span class="p">})</span>
<span class="n">transcript_tpm</span><span class="o">.</span><span class="n">drop</span><span class="p">([</span><span class="mi">0</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="s1">&#39;index&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">transcript_tpm</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;00Annotation&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">sep_in_col</span> <span class="o">=</span> <span class="s1">&#39; &#39;</span>  <span class="c1"># In the transcript expression file, genes and proteins ids with multiple values in a column are separated by a space.</span>

<span class="n">n_peaks</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="c1"># Total number of CAGE peaks in FANTOM5 data set</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;total_F5_peaks&quot;</span><span class="p">,</span> <span class="n">n_peaks</span><span class="p">)</span>

<span class="n">transcript_tpm</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">subset</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;uniprot_id&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">n_peaks_protein</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>  <span class="c1"># Number of peaks which are protein-coding (as mapped by FANTOM)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;has_protein_F5_peaks&quot;</span><span class="p">,</span> <span class="n">n_peaks_protein</span><span class="p">)</span>

<span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;uniprot_list&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;uniprot_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="n">sep_in_col</span><span class="p">)</span>  
<span class="n">transcript_tpm</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
<p>The CAGE peaks represent all kinds of transcripts, not only those which map to protein-coding genes.
The FANTOM file provides mappings to Uniprot IDs (<code class="docutils literal notranslate"><span class="pre">uniprot_id</span></code>), and these are used to discard the CAGE peaks that do not map to protein-coding genes: this takes us from  to   rows.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># How many CAGE peaks map to multiple HGNC id genes?</span>
<span class="n">hgnc_id_duplicates</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="p">[</span><span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;hgnc_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">na</span><span class="o">=</span><span class="kc">False</span><span class="p">)]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;There are </span><span class="si">{</span><span class="n">hgnc_id_duplicates</span><span class="si">}</span><span class="s2"> CAGE peaks that map to multiple HGNC ids (genes)&quot;</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;hgnc_id_duplicates&quot;</span><span class="p">,</span> <span class="n">hgnc_id_duplicates</span><span class="p">)</span>

<span class="n">entrezgene_id_duplicates</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="p">[</span><span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;entrezgene_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">na</span><span class="o">=</span><span class="kc">False</span><span class="p">)]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;There are </span><span class="si">{</span><span class="n">entrezgene_id_duplicates</span><span class="si">}</span><span class="s2"> CAGE peaks that map to multiple Entrezgene ids&quot;</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;entrezgene_id_duplicates&quot;</span><span class="p">,</span> <span class="n">entrezgene_id_duplicates</span><span class="p">)</span>

<span class="n">total_gene_id_duplicates</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="p">[(</span><span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;hgnc_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">na</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span> <span class="o">|</span> <span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;entrezgene_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">na</span><span class="o">=</span><span class="kc">False</span><span class="p">))]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;There are </span><span class="si">{</span><span class="n">total_gene_id_duplicates</span><span class="si">}</span><span class="s2"> CAGE peaks that map to multiple genes (HGNC or entrezgene IDs)&quot;</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;total_gene_id_duplicates&quot;</span><span class="p">,</span> <span class="n">total_gene_id_duplicates</span><span class="p">)</span>

<span class="c1"># Remove gene duplicates per CAGE TSS (due to overlapping CAGE TSS)</span>
<span class="n">transcript_tpm</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="p">[</span><span class="o">~</span><span class="p">(</span><span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;hgnc_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">na</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span> <span class="o">|</span> <span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;entrezgene_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">na</span><span class="o">=</span><span class="kc">False</span><span class="p">))]</span>

<span class="n">n_peaks_single_gene</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>  <span class="c1"># Number of peaks which are protein-coding (as mapped by FANTOM)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;single_gene_F5_peaks&quot;</span><span class="p">,</span> <span class="n">n_peaks_single_gene</span><span class="p">)</span>
</pre></div>
</div>
<p>CAGE peaks are mapped to genes based on overlap with the gene, so it is not always clear which gene a CAGE peak maps to.
Protein-coding CAGE peaks (map to at least one <code class="docutils literal notranslate"><span class="pre">uniprot_id</span></code>), that map to multiple genes are removed for simplicity.
These can be found by looking at either the <code class="docutils literal notranslate"><span class="pre">hgnc_id</span></code> or <code class="docutils literal notranslate"><span class="pre">entrezgene_id</span></code> gene identifier columns: the mapping to <code class="docutils literal notranslate"><span class="pre">hgnc_id</span></code> contains all those found in <code class="docutils literal notranslate"><span class="pre">entrezgene_id</span></code>, so these are removed.
This represents a total of  CAGE peaks that map to multiple genes according to the given identifers.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create expression by protein (uniprot ID), rather than by transcript</span>
<span class="n">protein_tpm</span> <span class="o">=</span> <span class="n">transcript_tpm</span><span class="o">.</span><span class="n">explode</span><span class="p">(</span><span class="s1">&#39;uniprot_list&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s1">&#39;uniprot_list&#39;</span><span class="p">,</span> <span class="s1">&#39;entrezgene_id&#39;</span><span class="p">,</span> <span class="s1">&#39;hgnc_id&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
<span class="n">protein_tpm</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">level</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;hgnc_id&#39;</span><span class="p">,</span> <span class="s1">&#39;entrezgene_id&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">protein_tpm</span><span class="o">.</span><span class="n">rename_axis</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="s1">&#39;uniprot_id&#39;</span><span class="p">)</span>
<span class="c1"># Every protein ID should have at least one gene ID</span>
<span class="k">assert</span><span class="p">(</span><span class="n">protein_tpm</span><span class="p">[</span><span class="n">protein_tpm</span><span class="p">[</span><span class="s1">&#39;entrezgene_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">)</span> 
<span class="k">assert</span><span class="p">(</span><span class="n">protein_tpm</span><span class="p">[</span><span class="n">protein_tpm</span><span class="p">[</span><span class="s1">&#39;hgnc_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">)</span>

<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;protein_expression_total&quot;</span><span class="p">,</span> <span class="n">protein_tpm</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<span class="n">protein_tpm</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">rows_left_multiple_genes</span> <span class="o">=</span> <span class="n">protein_tpm</span><span class="p">[</span><span class="n">protein_tpm</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="kc">False</span><span class="p">)]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Number of remaining rows that map to multiple genes: </span><span class="si">{</span><span class="n">rows_left_multiple_genes</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;rows_left_multiple_genes&quot;</span><span class="p">,</span> <span class="n">rows_left_multiple_genes</span><span class="p">)</span>

<span class="n">proteins_multiple_genes</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">protein_tpm</span><span class="p">[</span><span class="n">protein_tpm</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="kc">False</span><span class="p">)]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Number of proteins that map to multiple genes: </span><span class="si">{</span><span class="n">proteins_multiple_genes</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;proteins_multiple_genes&quot;</span><span class="p">,</span> <span class="n">proteins_multiple_genes</span><span class="p">)</span>

<span class="n">protein_tpm</span><span class="p">[</span><span class="n">protein_tpm</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="kc">False</span><span class="p">)]</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>

<span class="n">protein_tpm</span> <span class="o">=</span> <span class="n">protein_tpm</span><span class="p">[</span><span class="o">~</span><span class="n">protein_tpm</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="kc">False</span><span class="p">)]</span>
<span class="k">assert</span><span class="p">(</span><span class="n">protein_tpm</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">==</span><span class="nb">len</span><span class="p">(</span><span class="n">protein_tpm</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;unique_proteins&quot;</span><span class="p">,</span> <span class="n">protein_tpm</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
</pre></div>
</div>
<p>For <code class="docutils literal notranslate"><span class="pre">filip</span></code>, the expression was calculated per protein (since it is protein function predictions that it is filtering), rather than per CAGE peak (summing the TPMs of all CAGE peaks mapped to a protein to get the total for that protein) as in the original data, or as is often presented per gene.
This gave  rows of “protein expression” data.</p>
<p>Of these, however, there were  rows of data (corresponding to  proteins) for which the proteins map to multiple genes.
This happens when different genes are translated to make identical protein products.
It used to be the case that Uniprot would map these genes to the same Uniprot ID, but more recently different Uniprot IDs are used to capture where the proteins came from.
These rows were also removed.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">display</span><span class="p">(</span><span class="n">transcript_tpm</span><span class="p">[</span><span class="n">transcript_tpm</span><span class="p">[</span><span class="s1">&#39;uniprot_id&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39;B2R4R0&#39;</span><span class="p">,</span> <span class="n">na</span><span class="o">=</span><span class="kc">False</span><span class="p">)])</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># CAGE PEAKS ARE TOO SMALL TO SEE ON A STATIC PLOT, WOULD NEED AN INTERACTIVE PLOT.</span>

<span class="c1"># from Bio.SeqFeature import SeqFeature, FeatureLocation</span>
<span class="c1"># from Bio.Graphics import GenomeDiagram</span>
<span class="c1"># from IPython.core.display import Image</span>

<span class="c1"># gdd = GenomeDiagram.Diagram(&#39;Test Diagram&#39;)</span>
<span class="c1"># gdt_features = gdd.new_track(1, name=&quot;CAGE Peaks&quot;, greytrack=True)</span>
<span class="c1"># gds_features = gdt_features.new_set()</span>

<span class="c1"># #Add three features to show the strand options,</span>
<span class="c1"># p13_HIST1H4B = SeqFeature(FeatureLocation(26027058,26027076))</span>
<span class="c1"># gds_features.add_feature(p13_HIST1H4B, name=&quot;p13@HIST1H4B&quot;, label=True)</span>

<span class="c1"># p12_HIST1H4B = SeqFeature(FeatureLocation(26027093, 26027119))</span>
<span class="c1"># gds_features.add_feature(p12_HIST1H4B, name=&quot;p12@HIST1H4B&quot;, label=True)</span>

<span class="c1"># p12_HIST1H4B = SeqFeature(FeatureLocation(26189331, 26189346))</span>
<span class="c1"># gds_features.add_feature(p12_HIST1H4B, name=&quot;p3@HIST1H4D&quot;, label=True)</span>

<span class="c1"># p12_HIST1H4B = SeqFeature(FeatureLocation(26285754, 26285769))</span>
<span class="c1"># gds_features.add_feature(p12_HIST1H4B, name=&quot;p4@HIST1H4H&quot;, label=True)</span>

<span class="c1"># p12_HIST1H4B = SeqFeature(FeatureLocation(27841151, 27841161))</span>
<span class="c1"># gds_features.add_feature(p12_HIST1H4B, name=&quot;p12@HIST1H4L&quot;, label=True)</span>

<span class="c1"># gdd.draw(format=&#39;linear&#39;, fragments=1,</span>
<span class="c1">#          start=25900000, end=28100000)</span>
<span class="c1"># gdd.write(&quot;GD_labels_default.png&quot;, &quot;png&quot;)</span>
<span class="c1"># Image(&quot;GD_labels_default.png&quot;)</span>

</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># DNA FEATURES VIEWER IS NOT YET READY TO USE.</span>

<span class="c1"># from bokeh.plotting import figure, show</span>

<span class="c1"># from dna_features_viewer import GraphicFeature, GraphicRecord</span>
<span class="c1"># features=[</span>
<span class="c1">#     GraphicFeature(start=0, end=20, strand=+1, color=&quot;#ffd700&quot;,</span>
<span class="c1">#                    label=&quot;Small feature&quot;),</span>
<span class="c1">#     GraphicFeature(start=20, end=500, strand=+1, color=&quot;#ffcccc&quot;,</span>
<span class="c1">#                    label=&quot;Gene 1 with a very long name&quot;),</span>
<span class="c1">#     GraphicFeature(start=400, end=700, strand=-1, color=&quot;#cffccc&quot;,</span>
<span class="c1">#                    label=&quot;Gene 2&quot;),</span>
<span class="c1">#     GraphicFeature(start=600, end=900, strand=+1, color=&quot;#ccccff&quot;,</span>
<span class="c1">#                    label=&quot;Gene 3&quot;)</span>
<span class="c1"># ]</span>
<span class="c1"># record = GraphicRecord(sequence_length=1000, features=features)</span>
<span class="c1"># p = record.plot_with_bokeh(figure_width=5)</span>
<span class="c1"># show(p)</span>
</pre></div>
</div>
</div>
<div class="section" id="counts">
<h4><span class="section-number">5.3.1.5.2. </span>Counts<a class="headerlink" href="#counts" title="Permalink to this headline">¶</a></h4>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># # DON&#39;T WANT COUNTS</span>

<span class="c1"># import pandas as pd</span>
<span class="c1"># from myst_nb import glue</span>

<span class="c1"># transcript_counts = pd.read_csv(&#39;../c06-combining/data/experiments/fantom/hg38_fair+new_CAGE_peaks_phase1and2_counts_ann.osc.txt&#39;, delimiter=&#39;\t&#39;, comment=&#39;#&#39;, dtype={&#39;entrezgene_id&#39;: object})</span>
<span class="c1"># transcript_counts.drop([0], axis=&#39;index&#39;, inplace=True)</span>
<span class="c1"># transcript_counts.set_index(&#39;00Annotation&#39;, inplace=True)</span>

<span class="c1"># # In the transcript expression file, genes and proteins ids with multiple values per transcript are separated by a space.</span>
<span class="c1"># sep_in_col = &#39; &#39;</span>

<span class="c1"># n_transcripts = transcript_counts.shape[0] # Total number of transcripts in FANTOM5 data set</span>
<span class="c1"># glue(&quot;total_F5_transcripts&quot;, n_transcripts)</span>

<span class="c1"># transcript_counts.dropna(axis=0, subset=[&#39;uniprot_id&#39;], inplace=True)</span>
<span class="c1"># n_transcripts_protein = transcript_counts.shape[0]  # Number of transcripts which are protein-coding (as mapped by FANTOM)</span>
<span class="c1"># glue(&quot;has_protein_F5_transcripts&quot;, n_transcripts_protein)</span>

<span class="c1"># # Create expression by protein (uniprot ID), rather than by transcript</span>
<span class="c1"># transcript_counts[&#39;uniprot_id&#39;] = transcript_counts[&#39;uniprot_id&#39;].str.split(sep_in_col)  </span>
<span class="c1"># protein_counts = transcript_counts.explode(&#39;uniprot_id&#39;).groupby([&#39;uniprot_id&#39;, &#39;entrezgene_id&#39;, &#39;hgnc_id&#39;]).mean()</span>
<span class="c1"># protein_counts.reset_index(level=[&#39;hgnc_id&#39;, &#39;entrezgene_id&#39;], inplace=True)</span>

<span class="c1"># # Every protein ID should have at least one gene ID</span>
<span class="c1"># assert(protein_counts[protein_counts[&#39;entrezgene_id&#39;].isna()].shape[0]==0) </span>
<span class="c1"># assert(protein_counts[protein_counts[&#39;hgnc_id&#39;].isna()].shape[0]==0)</span>

<span class="c1"># print(f&quot;Number of rows that map to multiple genes: {protein_counts[protein_counts.index.duplicated(keep=False)].shape[0]}&quot;)</span>
<span class="c1"># print(f&quot;Number of proteins that map to multiple genes: {protein_counts[protein_counts.index.duplicated(keep=&#39;first&#39;)].shape[0]}&quot;)</span>

<span class="c1"># display(protein_counts[protein_counts.index.duplicated(keep=False)])</span>
<span class="c1"># display(protein_counts.loc[&#39;A0A024QZ90&#39;])  # example duplicate</span>

<span class="c1"># # Ideally, if we&#39;re doing a protein centric, thing, then we would like to average over the two protein averages (this should be done AFTER conversion to TPM).</span>
<span class="c1"># # Or if we&#39;re doing a gene centric thing, then obviously we&#39;d do it gene centric.</span>

<span class="c1"># protein_counts[&#39;hgnc_id&#39;] = protein_counts[&#39;hgnc_id&#39;].str.split(&#39; &#39;)</span>
<span class="c1"># protein_counts[&#39;entrezgene_id&#39;] = protein_counts[&#39;entrezgene_id&#39;].str.split(&#39; &#39;)</span>
<span class="c1"># hgnc_counts = protein_counts.explode(&#39;hgnc_id&#39;)</span>
<span class="c1"># entrez_counts = protein_counts.explode(&#39;entrezgene_id&#39;)</span>
<span class="c1"># print(&#39;hgnc&#39;,len(hgnc_counts[&#39;hgnc_id&#39;].unique()))</span>
<span class="c1"># print(&#39;entrezgene_id&#39;,len(entrez_counts[&#39;entrezgene_id&#39;].unique()))</span>

<span class="c1"># print(f&quot;Rows with multiple hgnc IDs, but only one entrezID: {protein_counts[protein_counts[&#39;hgnc_id&#39;].str.contains(&#39; &#39;, na=False) &amp; ~protein_counts[&#39;entrezgene_id&#39;].str.contains(&#39; &#39;, na=False)].shape[0]}&quot;) # two or more hgncids and only one entrez id</span>

<span class="c1"># print(f&quot;Rows with multiple entrez IDs, but only one hgnc ID: {protein_counts[protein_counts[&#39;entrezgene_id&#39;].str.contains(&#39; &#39;, na=False) &amp; ~protein_counts[&#39;hgnc_id&#39;].str.contains(&#39; &#39;, na=False)].shape[0]}&quot;) # two or more entrez ids and only one hgnc id</span>

<span class="c1"># print(transcript_counts[transcript_counts[&#39;hgnc_id&#39;].str.contains(&#39;HGNC:17992&#39;,na=False)][&#39;entrezgene_id&#39;].unique())</span>

<span class="c1"># print(&#39;hgnc&#39;,len(protein_counts[&#39;hgnc_id&#39;].unique()))</span>
<span class="c1"># print(&#39;entrezgene_id&#39;,len(protein_counts[&#39;entrezgene_id&#39;].unique()))</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="test-set-cafa2">
<h2><span class="section-number">5.3.2. </span>Test set: CAFA2<a class="headerlink" href="#test-set-cafa2" title="Permalink to this headline">¶</a></h2>
<div class="section" id="what">
<h3><span class="section-number">5.3.2.1. </span>What?<a class="headerlink" href="#what" title="Permalink to this headline">¶</a></h3>
<p>During development, I tested <code class="docutils literal notranslate"><span class="pre">filip</span></code> by comparing DcGO only and <code class="docutils literal notranslate"><span class="pre">filip</span></code> + DcGO on data from the 2nd round of the CAFA competition: CAFA2.
This was the most recent round of CAFA for which there were “groundtruth” data available at the time of development.
The data consisted of CAFA2 targets and the CAFA2 ground truth data.</p>
<p><strong>CAFA2 targets</strong>:</p>
<p><strong>CAFA2 groundtruth</strong>:</p>
</div>
<div class="section" id="why">
<h3><span class="section-number">5.3.2.2. </span>Why?<a class="headerlink" href="#why" title="Permalink to this headline">¶</a></h3>
<p>I chose to use the CAFA2 data because rather than a larger set of annotations (such as those available from SwissProt-KB or GOA) because it provided a way of validating on unknown targets.
I.e. if I made predictions with DcgO using the version of GO from the time the challenge was launched, and I use the groundtruth data provided by CAFA2, then I could compare my results with those in the CAFA2 competition and I could look at my results on unknown targets.</p>
</div>
<div class="section" id="how">
<h3><span class="section-number">5.3.2.3. </span>How?<a class="headerlink" href="#how" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="test-set-cafa3">
<h2><span class="section-number">5.3.3. </span>Test set: CAFA3<a class="headerlink" href="#test-set-cafa3" title="Permalink to this headline">¶</a></h2>
<p>After initial development, I entered DcGO only, and <code class="docutils literal notranslate"><span class="pre">filip</span></code> plus DcGO into the CAFA3 competition in order to test <code class="docutils literal notranslate"><span class="pre">filip</span></code> on a new dataset.</p>
<p>This meant that I did not download the CAFA3 groundtruth, as this analysis was done by the CAFA3 team, but only the CAFA3 targets.</p>
<p><strong>CAFA3 targets</strong>:</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c05-filter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="2-uberon.html" title="previous page"><span class="section-number">5.2. </span>The <code class="docutils literal notranslate"><span class="pre">uberon-py</span></code> package</a>
    <a class='right-next' id="next-link" href="5-methods.html" title="next page"><span class="section-number">5.4. </span>Methods</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>