
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>5.4. Methods &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="5.5. Results" href="6-results.html" />
    <link rel="prev" title="5.3. Input Data" href="4-data.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c0-front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-uberon.html">
     5.2. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-data.html">
     5.3. Input Data
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     5.4. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html">
     5.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     5.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="8-future-work.html">
     5.7. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/1-background.html">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/5-batch-correction.html">
     6.4. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html">
     6.5. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html#combined-data-set">
     6.6. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/7-discussion.html">
     6.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/8-future-work.html">
     6.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#further-data-cleaning-adding-disease-model-information-and-non-tissue-specific-samples">
   5.4.1. Further data cleaning: adding disease model information and non-tissue-specific samples
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-samples-to-uberon-terms">
     5.4.1.1. Mapping samples to uberon terms
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-cleaning">
     5.4.1.2. Data cleaning
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#creating-filip-input">
     5.4.1.3. Creating
     <code class="docutils literal notranslate">
      <span class="pre">
       filip
      </span>
     </code>
     input
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mapping-multi-species-phenotype-terms-to-samples">
   5.4.2. Mapping multi-species phenotype terms to samples
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-phenotype-terms-to-uberon-terms">
     5.4.2.1. Mapping phenotype terms to uberon terms
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#matching-via-uberon-terms">
     5.4.2.2. Matching via uberon terms
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#via-ontologies">
     5.4.2.3. Via ontologies
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#creating-an-alternate-mapping-with-text-mining">
     5.4.2.4. Creating an alternate mapping with text-mining
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#when-are-transcripts-expressed">
   5.4.3. When are transcripts “expressed”?
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#expression-cut-off">
     5.4.3.1. Expression cut-off
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#expression-score">
     5.4.3.2. Expression score
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#creation-of-input-predictions">
   5.4.4. Creation of input predictions
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#naive-predictions">
     5.4.4.1. Naive predictions
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#dcgo">
     5.4.4.2. dcGO
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#validation-methodology">
   5.4.5. Validation Methodology
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cafa-validation">
     5.4.5.1. CAFA Validation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#limitations-of-validation-method">
     5.4.5.2. Limitations of validation method
    </a>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <!-- #region -->
<div class="section" id="methods">
<span id="filter-methods"></span><h1><span class="section-number">5.4. </span>Methods<a class="headerlink" href="#methods" title="Permalink to this headline">¶</a></h1>
<div class="section" id="further-data-cleaning-adding-disease-model-information-and-non-tissue-specific-samples">
<h2><span class="section-number">5.4.1. </span>Further data cleaning: adding disease model information and non-tissue-specific samples<a class="headerlink" href="#further-data-cleaning-adding-disease-model-information-and-non-tissue-specific-samples" title="Permalink to this headline">¶</a></h2>
<div class="section" id="mapping-samples-to-uberon-terms">
<h3><span class="section-number">5.4.1.1. </span>Mapping samples to uberon terms<a class="headerlink" href="#mapping-samples-to-uberon-terms" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="data-cleaning">
<h3><span class="section-number">5.4.1.2. </span>Data cleaning<a class="headerlink" href="#data-cleaning" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="creating-filip-input">
<h3><span class="section-number">5.4.1.3. </span>Creating <code class="docutils literal notranslate"><span class="pre">filip</span></code> input<a class="headerlink" href="#creating-filip-input" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="mapping-multi-species-phenotype-terms-to-samples">
<h2><span class="section-number">5.4.2. </span>Mapping multi-species phenotype terms to samples<a class="headerlink" href="#mapping-multi-species-phenotype-terms-to-samples" title="Permalink to this headline">¶</a></h2>
<div class="section" id="mapping-phenotype-terms-to-uberon-terms">
<h3><span class="section-number">5.4.2.1. </span>Mapping phenotype terms to uberon terms<a class="headerlink" href="#mapping-phenotype-terms-to-uberon-terms" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="matching-via-uberon-terms">
<h3><span class="section-number">5.4.2.2. </span>Matching via uberon terms<a class="headerlink" href="#matching-via-uberon-terms" title="Permalink to this headline">¶</a></h3>
<!-- #endregion -->
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="o">%</span><span class="n">load_ext</span> <span class="n">autoreload</span>
<span class="o">%</span><span class="n">autoreload</span> <span class="mi">2</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">logging</span>

<span class="c1"># TODO: make sure `is_obsolete` is loaded in uberon_py. If not: make sure it now is. Also make sure `replaced_by` is captured. And make sure this is a relation of interest by default.</span>
<span class="c1"># TODO: add merge ontologies to uberon_py</span>
<span class="c1"># TODO: fix comment in uberon_py - it only saves part of the line (first word)</span>
<span class="c1"># TODO: have merge check for ontology wrongness + have that separately (for when you edit ontologies)</span>

<span class="k">def</span> <span class="nf">merge</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">b</span><span class="p">,</span> <span class="n">path</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>
    <span class="n">c</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="s2">&quot;Recursively merges (nested) dictionary b into dictionary a. Prefers a: a should be the more up-to-date ontology.&quot;</span>
    <span class="k">if</span> <span class="n">path</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span> <span class="n">path</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">b</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">c</span><span class="p">:</span>
            <span class="k">if</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">],</span> <span class="nb">dict</span><span class="p">)</span> <span class="ow">and</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">],</span> <span class="nb">dict</span><span class="p">):</span>
                <span class="n">merge</span><span class="p">(</span><span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">],</span> <span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">],</span> <span class="n">path</span> <span class="o">+</span> <span class="p">[</span><span class="nb">str</span><span class="p">(</span><span class="n">key</span><span class="p">)])</span>
            <span class="k">elif</span> <span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]</span> <span class="o">==</span> <span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">]:</span>
                <span class="k">pass</span> <span class="c1"># same leaf value</span>
            <span class="k">elif</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">],</span> <span class="nb">list</span><span class="p">)</span> <span class="ow">and</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">],</span> <span class="nb">list</span><span class="p">):</span>
                <span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">])</span> <span class="o">|</span> <span class="nb">set</span><span class="p">(</span><span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]))</span>
            <span class="k">else</span><span class="p">:</span>
                <span class="k">if</span> <span class="n">key</span> <span class="o">==</span> <span class="s1">&#39;namespace&#39;</span><span class="p">:</span>
                    <span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;Combined </span><span class="si">{</span><span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="si">}</span><span class="s2"> and </span><span class="si">{</span><span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="si">}</span><span class="s2">&quot;</span>
                <span class="k">elif</span> <span class="n">key</span> <span class="o">==</span> <span class="s1">&#39;name&#39;</span><span class="p">:</span>
                    <span class="c1"># we expect some differences in name, due to e.g. terms becoming obsolete </span>
                    <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Unmatching names: </span><span class="si">{</span><span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="si">}</span><span class="s2"> =/= </span><span class="si">{</span><span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="si">}</span><span class="s2">, keeping the former.&quot;</span><span class="p">)</span>
                <span class="k">else</span><span class="p">:</span>
                    <span class="n">logging</span><span class="o">.</span><span class="n">error</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;For key </span><span class="si">{</span><span class="n">key</span><span class="si">}</span><span class="s2">, </span><span class="si">{</span><span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="si">}</span><span class="s2"> =/= </span><span class="si">{</span><span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="si">}</span><span class="s2">. Unsure which to choose.&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">c</span><span class="p">[</span><span class="n">key</span><span class="p">]</span> <span class="o">=</span> <span class="n">b</span><span class="p">[</span><span class="n">key</span><span class="p">]</span>
    <span class="k">assert</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">c</span><span class="p">)</span> <span class="o">==</span> <span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">a</span><span class="p">)</span> <span class="o">+</span> <span class="nb">len</span><span class="p">(</span><span class="n">b</span><span class="p">)</span> <span class="o">-</span> <span class="nb">len</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">a</span><span class="o">.</span><span class="n">keys</span><span class="p">())</span> <span class="o">&amp;</span> <span class="nb">set</span><span class="p">(</span><span class="n">b</span><span class="o">.</span><span class="n">keys</span><span class="p">()))))</span>
    <span class="k">return</span> <span class="n">c</span>

</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: Map to UBERON + CELL tissues</span>

<span class="n">fantom_obo_file</span> <span class="o">=</span> <span class="s1">&#39;../c06-combining/data/experiments/fantom/ff-phase2-170801.obo.txt&#39;</span>
<span class="c1"># uberon_obo_file = &#39;../c06-combining/data/uberonext_obo.txt&#39; # TODO: maybe want to use new one and compare</span>
<span class="n">uberon_obo_file</span> <span class="o">=</span> <span class="s1">&#39;../c06-combining/data/uberon_ext_210321.obo&#39;</span> <span class="c1"># TODO: maybe want to use old one and compare</span>

<span class="n">fantom_obo</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Obo</span><span class="p">(</span><span class="n">fantom_obo_file</span><span class="p">,</span> <span class="n">ont_ids</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;HP&#39;</span><span class="p">,</span> <span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;CL&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span> <span class="s1">&#39;FF&#39;</span><span class="p">,</span> <span class="s1">&#39;DOID&#39;</span><span class="p">])</span>
<span class="n">uberon_obo</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Obo</span><span class="p">(</span><span class="n">uberon_obo_file</span><span class="p">,</span> <span class="n">ont_ids</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;GO&#39;</span><span class="p">,</span> <span class="s1">&#39;HP&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">,</span><span class="s1">&#39;CL&#39;</span><span class="p">])</span>
<span class="n">merged_ont</span> <span class="o">=</span> <span class="n">merge</span><span class="p">(</span><span class="n">uberon_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">,</span> <span class="n">fantom_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">)</span>

<span class="c1"># TODO: glue sizes of ontologies maybe</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">uberon_py</span> <span class="kn">import</span> <span class="n">obo</span>

</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: write</span>
<span class="c1"># NOTE: would need to ensure that NCBI taxon ontology is present in order to be able to do this, since there are taxon terms for vertebrates, etc, not just individual species.</span>
<span class="c1"># def restrict_to_taxon(ont, ncbi_id, valid_terms = [&#39;UBERON&#39;], keep_dubious=False, only_yes=False):</span>
<span class="c1">#     &quot;&quot;&quot;</span>
<span class="c1">#     Loops through valid_terms terms in ontology `ont`, and removes any that are never in taxon with ID `ncbi_id`</span>
<span class="c1">#     &quot;&quot;&quot;</span>
<span class="c1">#     yes_relations = [&#39;only_in_taxon&#39;, &#39;present_in_taxon&#39;]</span>
<span class="c1">#     never = &#39;never_in_taxon&#39;</span>
<span class="c1">#     dubious = &#39;dubious_for_taxon&#39;</span>
    
<span class="c1">#     for term in ont.keys():</span>
<span class="c1">#         if term.split(&#39;:&#39;)[0] not in valid_terms:</span>
<span class="c1">#             continue</span>
        
        
        
    <span class="c1"># TODO: check that you&#39;ve read in the </span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span> 
<span class="n">samples</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;data/cleaned_pre_input/ff_accessions_to_keep.txt&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">comment</span><span class="o">=</span><span class="s1">&#39;#&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span>
<span class="c1"># samples</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">samples_info</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;data/cleaned_pre_input/samples_info.csv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">comment</span><span class="o">=</span><span class="s1">&#39;#&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">samples_info</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">map_tissue_by_ontology</span><span class="p">(</span><span class="n">sample_ids</span><span class="p">,</span> 
                           <span class="n">combined_obo</span><span class="p">,</span> 
                           <span class="n">too_general_tissues</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
                           <span class="n">tissue_relations_of_interest</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
<span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">too_general_tissues</span><span class="p">:</span>
        <span class="n">too_general_tissues</span> <span class="o">=</span> <span class="p">[</span>
            <span class="s1">&#39;UBERON:0000061&#39;</span><span class="p">,</span>  <span class="c1"># anatomical structure</span>
            <span class="s1">&#39;UBERON:0000479&#39;</span><span class="p">,</span>  <span class="c1"># tissue</span>
            <span class="s1">&#39;UBERON:0000467&#39;</span><span class="p">,</span>  <span class="c1"># anatomical system</span>
            <span class="s1">&#39;UBERON:0011216&#39;</span><span class="p">,</span>  <span class="c1"># organ system subdivision</span>
            <span class="s1">&#39;UBERON:0000922&#39;</span><span class="p">,</span>  <span class="c1"># embryo</span>
        <span class="p">]</span>

    <span class="k">if</span> <span class="ow">not</span> <span class="n">tissue_relations_of_interest</span><span class="p">:</span>
        <span class="n">tissue_relations_of_interest</span> <span class="o">=</span> <span class="p">[</span>
            <span class="s1">&#39;is_a&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;related_to&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;part_of&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;derives_from&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;intersection_of&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;union_of&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;is_model_for&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;replaced_by&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;develops_from&#39;</span><span class="p">,</span>
        <span class="p">]</span>
        
    <span class="c1"># TODO: Change uberon_py and add this functionality.</span>
    <span class="c1"># TODO: Write function string</span>
    <span class="c1"># TODO: Change uberon-py excluded tissues.</span>
    <span class="n">uberon_identifier</span> <span class="o">=</span> <span class="s1">&#39;UBERON&#39;</span>  <span class="c1"># interested in relationships to uberon terms.</span>
    <span class="n">tissue_relations</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
        <span class="n">relations_of_interest</span><span class="o">=</span><span class="n">tissue_relations_of_interest</span><span class="p">,</span> 
        <span class="n">source_terms</span><span class="o">=</span><span class="n">sample_ids</span><span class="p">,</span>
        <span class="n">target_term</span><span class="o">=</span><span class="n">uberon_identifier</span><span class="p">,</span>
        <span class="n">ont</span><span class="o">=</span><span class="n">combined_obo</span><span class="p">,</span> 
        <span class="n">excluded_terms</span><span class="o">=</span><span class="n">too_general_tissues</span><span class="p">,</span>
    <span class="p">)</span>
    
    <span class="n">tissue_map_ont</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">index</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">tissue_relations</span><span class="o">.</span><span class="n">relations</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
        <span class="n">relation_string</span> <span class="o">=</span> <span class="n">row</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
        <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">relation_string</span><span class="p">):</span>
            <span class="n">uberon</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">uberon</span> <span class="o">=</span> <span class="n">relation_string</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;_&#39;</span><span class="p">)[</span><span class="o">-</span><span class="mi">1</span><span class="p">]</span>
        <span class="n">tissue_map_ont</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="n">index</span><span class="p">,</span> <span class="n">relation_string</span><span class="p">,</span> <span class="n">uberon</span><span class="p">])</span>
    <span class="n">tissue_map_ont</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">tissue_map_ont</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Sample IDs&#39;</span><span class="p">,</span> <span class="s1">&#39;Relation String&#39;</span><span class="p">,</span> <span class="s1">&#39;UBERON&#39;</span><span class="p">])</span>
    <span class="n">tissue_map_ont</span> <span class="o">=</span> <span class="n">tissue_map_ont</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Sample IDs&#39;</span><span class="p">)</span>
        
    <span class="k">return</span> <span class="n">tissue_map_ont</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">no_cl</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">merged_ont</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">key</span><span class="p">[:</span><span class="mi">2</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;CL&#39;</span><span class="p">:</span>
        <span class="n">no_cl</span><span class="p">[</span><span class="n">key</span><span class="p">]</span> <span class="o">=</span> <span class="n">merged_ont</span><span class="p">[</span><span class="n">key</span><span class="p">]</span>
<span class="n">map_by_ont_no_cl</span> <span class="o">=</span> <span class="n">map_tissue_by_ontology</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">no_cl</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">map_by_ont</span> <span class="o">=</span> <span class="n">map_tissue_by_ontology</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">merged_ont</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">get_leaves</span><span class="p">(</span><span class="n">ont</span><span class="p">,</span> <span class="n">term_types</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">relations_of_interest</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    term_types = None: get all leaf terms</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">relations_of_interest</span><span class="p">:</span>
        <span class="n">relations_of_interest</span> <span class="o">=</span> <span class="p">[</span>
            <span class="s1">&#39;is_a&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;related_to&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;part_of&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;derives_from&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;intersection_of&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;union_of&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;is_model_for&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;replaced_by&#39;</span><span class="p">,</span> 
            <span class="s1">&#39;develops_from&#39;</span><span class="p">,</span>
        <span class="p">]</span>
    
    <span class="n">leaves</span> <span class="o">=</span> <span class="nb">set</span><span class="p">(</span><span class="n">ont</span><span class="o">.</span><span class="n">keys</span><span class="p">())</span>
    <span class="k">if</span> <span class="n">term_types</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
        <span class="k">assert</span><span class="p">(</span><span class="nb">isinstance</span><span class="p">(</span><span class="n">term_types</span><span class="p">,</span> <span class="nb">list</span><span class="p">))</span>
        <span class="k">assert</span><span class="p">(</span><span class="nb">all</span><span class="p">([</span><span class="s1">&#39;:&#39;</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">term_types</span><span class="p">]))</span> <span class="c1"># don&#39;t want &#39;UBERON:1231239&#39; must be of form &#39;UBERON&#39;, &#39;GO&#39;</span>
        <span class="n">leaves</span> <span class="o">=</span> <span class="p">{</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">leaves</span> <span class="k">if</span> <span class="n">x</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">in</span> <span class="n">term_types</span><span class="p">}</span>
    <span class="k">for</span> <span class="n">key</span> <span class="ow">in</span> <span class="n">ont</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
        <span class="k">for</span> <span class="n">relation</span> <span class="ow">in</span> <span class="n">relations_of_interest</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">relation</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">ont</span><span class="p">[</span><span class="n">key</span><span class="p">]</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
                <span class="k">continue</span>
            <span class="n">related_terms</span> <span class="o">=</span> <span class="n">ont</span><span class="p">[</span><span class="n">key</span><span class="p">][</span><span class="n">relation</span><span class="p">]</span>
            <span class="k">assert</span><span class="p">(</span><span class="nb">isinstance</span><span class="p">(</span><span class="n">related_terms</span><span class="p">,</span> <span class="nb">list</span><span class="p">))</span>
            <span class="n">leaves</span> <span class="o">-=</span> <span class="nb">set</span><span class="p">(</span><span class="n">related_terms</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">leaves</span>

<span class="n">leaves</span> <span class="o">=</span> <span class="n">get_leaves</span><span class="p">(</span><span class="n">merged_ont</span><span class="p">,</span> <span class="n">term_types</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">leaves</span><span class="p">))</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: Create table showing how adding CL increases coverage (and maybe also relation types)</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">map_by_ont_no_cl</span><span class="p">[</span><span class="o">~</span><span class="n">map_by_ont_no_cl</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">map_by_ont</span><span class="p">[</span><span class="o">~</span><span class="n">map_by_ont</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">map_by_ont_no_cl</span><span class="p">[</span><span class="n">map_by_ont_no_cl</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">unmapped_by_ont</span> <span class="o">=</span> <span class="n">map_by_ont</span><span class="p">[</span><span class="n">map_by_ont</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
<span class="nb">print</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">unmapped_by_ont</span><span class="p">))</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: explain that the mapping including CL terms and all types of mapping, does find a mapping for everything that the ontology-creators finishing classifying. And then lacking any expertise, I will map these manually to T-cell.</span>
<span class="c1"># TODO: Add functionality to `uberon-py` to add new stuff.</span>

<span class="k">for</span> <span class="n">term</span> <span class="ow">in</span> <span class="n">unmapped_by_ont</span><span class="p">:</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fantom_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">[</span><span class="n">term</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">],</span> <span class="n">fantom_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">[</span><span class="n">term</span><span class="p">][</span><span class="s1">&#39;comment&#39;</span><span class="p">])</span>
    <span class="k">except</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">term</span><span class="p">,</span> <span class="n">fantom_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">[</span><span class="n">term</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">])</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">stem_cell</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;CL:0011115&#39;</span><span class="p">,</span> <span class="c1"># precursor cell --&gt; exclude precursor cells because they only develop into one cell type</span>
<span class="c1">#     &#39;CL:0000048&#39;, # multi fate stem cell</span>
<span class="c1">#     &#39;CL:0000034&#39;, # stem cell</span>
<span class="c1">#     &#39;CL:0000723&#39;, # stomatic stem cell</span>
<span class="c1">#     &#39;CL:0008001&#39;, # hematopoietic precursor cell (-&gt; missing to `is_a` precursor cell)</span>
<span class="p">]</span>

<span class="n">relations</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">,</span><span class="s1">&#39;related_to&#39;</span><span class="p">,</span><span class="s1">&#39;part_of&#39;</span><span class="p">,</span><span class="s1">&#39;intersection_of&#39;</span><span class="p">,</span><span class="s1">&#39;union_of&#39;</span><span class="p">,</span> <span class="s1">&#39;derives_from&#39;</span><span class="p">]</span>  <span class="c1"># NOT &#39;develops_from&#39;</span>
<span class="k">for</span> <span class="n">unmapped</span> <span class="ow">in</span> <span class="n">map_by_ont</span><span class="p">[</span><span class="n">map_by_ont</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">unmapped</span><span class="p">,</span> <span class="n">fantom_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">[</span><span class="n">unmapped</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">])</span>
    <span class="n">tissue_relations</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
        <span class="n">relations_of_interest</span><span class="o">=</span><span class="n">relations</span><span class="p">,</span> 
        <span class="n">source_terms</span><span class="o">=</span><span class="p">[</span><span class="n">unmapped</span><span class="p">],</span>
        <span class="n">target_term</span><span class="o">=</span><span class="n">stem_cell</span><span class="p">,</span>
        <span class="n">ont</span><span class="o">=</span><span class="n">merged_ont</span><span class="p">,</span>
<span class="c1">#         print_=True</span>
    <span class="p">)</span>

    <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">tissue_relations</span><span class="o">.</span><span class="n">relations</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]):</span>
        <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;NON STEM CELL&#39;</span><span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">unmapped</span><span class="p">,</span> <span class="n">fantom_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">[</span><span class="n">unmapped</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">])</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">samples_info</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">unmapped</span><span class="p">][</span><span class="s1">&#39;Characteristics[Tissue]&#39;</span><span class="p">])</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;STEM CELL&#39;</span><span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">unmapped</span><span class="p">,</span> <span class="n">fantom_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">[</span><span class="n">unmapped</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">])</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="n">unmapped</span><span class="p">)][</span><span class="s1">&#39;Characteristics[Tissue]&#39;</span><span class="p">])</span>

    <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n\n</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="n">map_by_ont</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: Get uberon to output a more reasonable table (of mappings (with names))</span>
<span class="c1"># TODO: uberon should use a hyphen - instad of an underscore for relation strings</span>

<span class="c1"># TODO: Write about how unclassifable stuff maps to too general tissue.</span>
<span class="n">tissue_col</span> <span class="o">=</span> <span class="s1">&#39;Characteristics[Tissue]&#39;</span>
<span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
<span class="n">unclassifiables</span> <span class="o">=</span> <span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span><span class="o">==</span><span class="s1">&#39;unclassifiable&#39;</span><span class="p">]</span>
<span class="c1"># unclassifiables.index </span>
<span class="k">for</span> <span class="n">unclassifiable</span> <span class="ow">in</span> <span class="n">unclassifiables</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">map_by_ont</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">unclassifiable</span><span class="p">][</span><span class="s1">&#39;Relation String&#39;</span><span class="p">])</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">samples_info</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;FF:11322-117D8&#39;</span><span class="p">]</span>  
<span class="c1"># Skeletal muscle myoblast, develops_into skeletal muscle</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s1">&#39;FF:12225-129F2&#39;</span><span class="p">)]</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">merged_ont</span><span class="p">[</span><span class="s1">&#39;CL:0000037&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="section" id="via-ontologies">
<h3><span class="section-number">5.4.2.3. </span>Via ontologies<a class="headerlink" href="#via-ontologies" title="Permalink to this headline">¶</a></h3>
<p>The extended Uberon ontology is first interrogated for any existing relation to the term in the ontology using <code class="docutils literal notranslate"><span class="pre">uberon-py</span></code>.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: EDA of mapping coverage (CL, UBERON, name)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: Do nothing with this (keep in disease for now), maybe add it as an uberon-py example</span>
<span class="k">def</span> <span class="nf">get_disease_related_samples</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">ont</span><span class="p">,</span> <span class="n">print_</span><span class="o">=</span><span class="kc">False</span><span class="p">):</span>
    <span class="n">disease_relations_of_interest</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">,</span><span class="s1">&#39;is_model_for&#39;</span><span class="p">]</span>
    <span class="n">disease_related</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
        <span class="n">relations_of_interest</span><span class="o">=</span><span class="n">disease_relations_of_interest</span><span class="p">,</span> 
        <span class="n">source_terms</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">samples</span><span class="p">),</span>
        <span class="n">target_term</span><span class="o">=</span><span class="s1">&#39;DOID&#39;</span><span class="p">,</span>
        <span class="n">ont</span><span class="o">=</span><span class="n">ont</span><span class="p">,</span>
        <span class="n">print_</span><span class="o">=</span><span class="n">print_</span><span class="p">,</span>
    <span class="p">)</span><span class="o">.</span><span class="n">relations</span>
    <span class="n">disease_related</span> <span class="o">=</span> <span class="n">disease_related</span><span class="p">[</span><span class="o">~</span><span class="n">disease_related</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span>
    <span class="k">return</span> <span class="n">disease_related</span>
<span class="n">disease_related</span> <span class="o">=</span> <span class="n">get_disease_related_samples</span><span class="p">(</span><span class="n">samples</span><span class="o">=</span><span class="n">samples</span><span class="p">,</span> <span class="n">ont</span><span class="o">=</span><span class="n">merged_ont</span><span class="p">)</span>
<span class="c1"># display(disease_related)</span>
<span class="c1"># print(len(disease_related.index))</span>

<span class="c1"># 32 samples are disease related</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: Do nothing with this (except maybe explain that it was checked, maybe add it as an uberon-py example)</span>
<span class="k">def</span> <span class="nf">get_missing_human_annotation</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">ont</span><span class="p">,</span> <span class="n">print_</span><span class="o">=</span><span class="kc">False</span><span class="p">):</span>
    <span class="n">human_relations_of_interest</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">]</span>
    <span class="n">human_related</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span>
        <span class="n">relations_of_interest</span><span class="o">=</span><span class="n">human_relations_of_interest</span><span class="p">,</span> 
        <span class="n">source_terms</span><span class="o">=</span><span class="n">samples</span><span class="p">,</span>
        <span class="n">target_term</span><span class="o">=</span><span class="s1">&#39;FF:0000210&#39;</span><span class="p">,</span> 
        <span class="n">ont</span><span class="o">=</span><span class="n">ont</span><span class="p">,</span>
        <span class="n">print_</span><span class="o">=</span><span class="n">print_</span><span class="p">,</span>
    <span class="p">)</span><span class="o">.</span><span class="n">relations</span>
    <span class="n">missing_human_ann</span> <span class="o">=</span> <span class="n">human_related</span><span class="p">[</span><span class="n">human_related</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span>
    <span class="k">return</span> <span class="n">missing_human_ann</span>
<span class="n">missing_human_ann</span> <span class="o">=</span> <span class="n">get_non_human_samples</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">merged_ont</span><span class="p">)</span>
<span class="n">missing_human_ann</span><span class="p">[</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">([</span><span class="n">merged_ont</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">missing_human_ann</span><span class="o">.</span><span class="n">index</span><span class="p">],</span> <span class="n">index</span><span class="o">=</span> <span class="n">missing_human_ann</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>
<span class="c1"># display(missing_human_ann)</span>

<span class="n">species_col</span> <span class="o">=</span> <span class="s1">&#39;Chracteristics [Species]&#39;</span>
<span class="n">human</span> <span class="o">=</span> <span class="s1">&#39;Human (Homo sapiens)&#39;</span>
<span class="k">assert</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="o">.</span><span class="n">all</span><span class="p">(</span><span class="n">samples_info</span><span class="p">[</span><span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">missing_human_ann</span><span class="o">.</span><span class="n">index</span><span class="p">))][</span><span class="n">species_col</span><span class="p">]</span> <span class="o">==</span> <span class="n">human</span><span class="p">))</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">map_by_name</span><span class="p">(</span><span class="n">samples_to_names</span><span class="p">,</span> <span class="n">samples</span><span class="p">,</span> <span class="n">ont</span><span class="p">,</span> <span class="n">target_terms</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    :param samples_to_names: Pandas series with samples as index and names as values</span>
<span class="sd">    :param samples:</span>
<span class="sd">    :param ont:</span>
<span class="sd">    :param target_terms: A list of target terms, can be a length-1 list. Can be either specific terms (e.g. &#39;UBERON:0000310&#39;) or general term labels (e.g. &#39;UBERON&#39;)</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="c1"># TODO: Speed up</span>
    <span class="c1"># TODO: Update uberon-py with this version</span>
    <span class="n">samples_to_names</span> <span class="o">=</span> <span class="n">samples_to_names</span><span class="p">[</span><span class="n">samples_to_names</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">samples</span><span class="p">)]</span>
    <span class="n">samples_to_names</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="k">assert</span><span class="p">(</span><span class="nb">isinstance</span><span class="p">(</span><span class="n">target_terms</span><span class="p">,</span> <span class="nb">list</span><span class="p">))</span>
    <span class="c1"># TODO: allow dict, if it&#39;s a dict convert to a series or vice versa</span>
    <span class="k">assert</span><span class="p">(</span><span class="nb">isinstance</span><span class="p">(</span><span class="n">samples_to_names</span><span class="p">,</span> <span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="p">))</span>

    <span class="n">name2uberon</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">sample_id</span><span class="p">,</span> <span class="n">tissue_name</span> <span class="ow">in</span> <span class="n">samples_to_names</span><span class="o">.</span><span class="n">iteritems</span><span class="p">():</span>
        <span class="n">found</span> <span class="o">=</span> <span class="kc">False</span>
        <span class="n">tissue_name</span> <span class="o">=</span> <span class="n">tissue_name</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
        
        <span class="k">for</span> <span class="n">ont_term</span> <span class="ow">in</span> <span class="n">ont</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
            
            <span class="c1"># Don&#39;t check terms that aren&#39;t of interest:</span>
            <span class="n">terms_general</span> <span class="o">=</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">target_terms</span><span class="p">,</span> <span class="nb">list</span><span class="p">)</span> <span class="ow">and</span> <span class="p">(</span><span class="ow">not</span> <span class="s1">&#39;:&#39;</span> <span class="ow">in</span> <span class="n">target_terms</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
            <span class="n">terms_specific</span> <span class="o">=</span> <span class="nb">isinstance</span><span class="p">(</span><span class="n">target_terms</span><span class="p">,</span> <span class="nb">list</span><span class="p">)</span> <span class="ow">and</span> <span class="p">(</span><span class="s1">&#39;:&#39;</span> <span class="ow">in</span> <span class="n">target_terms</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
            <span class="k">if</span>  <span class="n">terms_general</span> <span class="ow">and</span> <span class="n">ont_term</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;:&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">target_terms</span><span class="p">:</span>
                <span class="k">continue</span>
            <span class="k">elif</span> <span class="n">terms_specific</span> <span class="ow">and</span> <span class="p">(</span><span class="n">ont_term</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">target_terms</span><span class="p">):</span>
                <span class="k">continue</span>
            
            <span class="c1"># Check exact names and synonyms:</span>
            <span class="k">try</span><span class="p">:</span>
                <span class="n">synonyms</span> <span class="o">=</span> <span class="n">ont</span><span class="p">[</span><span class="n">ont_term</span><span class="p">][</span><span class="s1">&#39;synonyms&#39;</span><span class="p">]</span>
            <span class="k">except</span><span class="p">:</span>
                <span class="n">synonyms</span> <span class="o">=</span> <span class="p">[]</span>    
            <span class="k">if</span> <span class="p">(</span><span class="n">ont</span><span class="p">[</span><span class="n">ont_term</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span> <span class="o">==</span> <span class="n">tissue_name</span><span class="p">)</span> <span class="ow">or</span> <span class="p">(</span><span class="n">tissue_name</span> <span class="ow">in</span> <span class="n">synonyms</span><span class="p">):</span>
                <span class="n">name2uberon</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="n">sample_id</span><span class="p">,</span> <span class="n">ont_term</span><span class="p">,</span> <span class="n">tissue_name</span><span class="p">])</span>
                <span class="n">found</span> <span class="o">=</span> <span class="kc">True</span>
        
        <span class="k">if</span> <span class="n">found</span> <span class="o">==</span> <span class="kc">False</span><span class="p">:</span>
            <span class="n">name2uberon</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="n">sample_id</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="n">tissue_name</span><span class="p">])</span>

    <span class="n">name2uberon</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">name2uberon</span><span class="p">,</span> <span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">,</span><span class="s1">&#39;Ontology term&#39;</span><span class="p">,</span> <span class="s1">&#39;Name matched on&#39;</span><span class="p">])</span>
    <span class="n">name2uberon</span> <span class="o">=</span> <span class="n">name2uberon</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">)</span>
    <span class="n">name2uberon</span> <span class="o">=</span> <span class="n">name2uberon</span><span class="p">[</span><span class="o">~</span><span class="n">name2uberon</span><span class="p">[</span><span class="s1">&#39;Ontology term&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span>
    <span class="k">return</span> <span class="n">name2uberon</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">name_col</span> <span class="o">=</span> <span class="s1">&#39;Charateristics [description]&#39;</span>
<span class="n">sample_to_name</span> <span class="o">=</span> <span class="n">samples_info</span><span class="p">[</span><span class="n">tissue_col</span><span class="p">]</span>
<span class="n">map_by_tissue_name</span> <span class="o">=</span> <span class="n">map_by_name</span><span class="p">(</span><span class="n">sample_to_name</span><span class="p">,</span> <span class="n">samples</span><span class="p">,</span> <span class="n">merged_ont</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">map_by_tissue_name</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">relation_string</span><span class="p">,</span> <span class="n">uberon</span> <span class="o">=</span> <span class="n">map_by_ont</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s1">&#39;FF:10057-101H3&#39;</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="n">relation_string</span><span class="p">,</span> <span class="n">uberon</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># for sample_id in map_by_tissue_name[map_by_tissue_name.index.duplicated()].index:</span>
<span class="c1">#     terms = map_by_tissue_name.loc[sample_id][&#39;Ontology term&#39;]</span>
<span class="c1">#     # choose more specific</span>
<span class="c1"># #     print(len(terms))</span>
<span class="c1">#     if len(terms)&gt;2:</span>
<span class="c1">#         for term in terms:</span>
<span class="c1">#             if &#39;only_in_taxon&#39; in merged_ont[term].keys():</span>
<span class="c1">#                 print(term, merged_ont[term][&#39;name&#39;], merged_ont[term][&#39;only_in_taxon&#39;])</span>
<span class="c1">#             else:</span>
<span class="c1">#                 print(term, merged_ont[term][&#39;name&#39;])</span>

                
<span class="c1">#         break</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">get_overall_tissue_mappings</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">map_name</span><span class="p">,</span> <span class="n">map_ont</span><span class="p">,</span> <span class="n">ont</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">map_name</span> <span class="o">=</span> <span class="n">map_name</span><span class="p">[</span><span class="o">~</span><span class="n">map_name</span><span class="o">.</span><span class="n">index</span><span class="o">.</span><span class="n">duplicated</span><span class="p">(</span><span class="n">keep</span><span class="o">=</span><span class="s1">&#39;first&#39;</span><span class="p">)]</span>

    <span class="n">map_ont</span><span class="o">.</span><span class="n">drop_duplicates</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    <span class="n">map_ont</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="n">rel</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;is_a&#39;</span><span class="p">,</span> <span class="s1">&#39;part_of&#39;</span><span class="p">]</span>  <span class="c1"># relations of interest between by name and by ont terms.</span>
    <span class="n">overall_mapping</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">mapping_columns</span> <span class="o">=</span> <span class="p">[</span>
        <span class="s1">&#39;sample id&#39;</span><span class="p">,</span><span class="s1">&#39;by name&#39;</span><span class="p">,</span><span class="s1">&#39;by ont&#39;</span><span class="p">,</span><span class="s1">&#39;relation string&#39;</span><span class="p">,</span> <span class="s1">&#39;name label&#39;</span><span class="p">,</span> <span class="s1">&#39;name string&#39;</span><span class="p">,</span> <span class="s1">&#39;overall&#39;</span><span class="p">,</span> <span class="s1">&#39;mapped by&#39;</span>
    <span class="p">]</span>
    <span class="n">disagreements</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">disagreement_columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;sample id&#39;</span><span class="p">,</span> <span class="s1">&#39;by name&#39;</span><span class="p">,</span> <span class="s1">&#39;by ont&#39;</span><span class="p">,</span><span class="s1">&#39;relation string&#39;</span><span class="p">,</span> <span class="s1">&#39;name label&#39;</span><span class="p">]</span>
    <span class="n">unmappable</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">samples</span><span class="p">:</span>    
        <span class="c1">#check if mappable by name:</span>
        <span class="k">if</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">map_name</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
            <span class="n">by_name</span><span class="p">,</span> <span class="n">name_matched_on</span> <span class="o">=</span> <span class="n">map_name</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">by_name</span><span class="p">,</span> <span class="n">name_matched_on</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>

        <span class="c1">#check if mappable by ontology:</span>
        <span class="k">if</span> <span class="n">sample</span> <span class="ow">in</span> <span class="n">map_ont</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
            <span class="n">relation_string</span><span class="p">,</span> <span class="n">by_ont</span> <span class="o">=</span> <span class="n">map_by_ont</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">sample</span><span class="p">]</span>
            <span class="n">name_string</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">relation_string_2_name_string</span><span class="p">(</span><span class="n">ont</span><span class="p">,</span> <span class="n">relation_string</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">by_ont</span><span class="p">,</span> <span class="n">relation_string</span><span class="p">,</span> <span class="n">name_string</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>

        <span class="k">if</span> <span class="n">by_name</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">by_ont</span><span class="p">:</span>  <span class="c1"># by name only</span>
            <span class="n">overall</span> <span class="o">=</span> <span class="n">by_name</span>
            <span class="n">mapped_using</span> <span class="o">=</span> <span class="s2">&quot;name&quot;</span>
        <span class="k">elif</span> <span class="n">by_ont</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">by_name</span><span class="p">:</span>  <span class="c1"># by ont only</span>
            <span class="n">overall</span> <span class="o">=</span> <span class="n">by_ont</span>
            <span class="n">mapped_using</span> <span class="o">=</span> <span class="s2">&quot;ontology&quot;</span>
        <span class="k">elif</span> <span class="ow">not</span> <span class="n">by_name</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">by_ont</span><span class="p">:</span>  <span class="c1"># both unmappable</span>
            <span class="n">overall</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="n">unmappable</span><span class="o">.</span><span class="n">append</span><span class="p">([</span><span class="n">sample</span><span class="p">])</span>
            <span class="n">mapped_using</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="k">elif</span> <span class="n">by_name</span> <span class="o">==</span> <span class="n">by_ont</span><span class="p">:</span> <span class="c1"># both the same</span>
            <span class="n">overall</span> <span class="o">=</span> <span class="n">by_ont</span>
            <span class="n">mapped_using</span> <span class="o">=</span> <span class="s2">&quot;both (same)&quot;</span>
        <span class="k">elif</span> <span class="n">by_name</span> <span class="o">!=</span> <span class="n">by_ont</span><span class="p">:</span>  <span class="c1"># mappable but different</span>
            <span class="c1"># TODO: this is so messy, relations should be able to give back first relation.</span>
            <span class="c1"># TODO: fix &#39;one way vs other way&#39;</span>
            <span class="n">one_way</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span><span class="n">rel</span><span class="p">,</span> <span class="p">[</span><span class="n">by_name</span><span class="p">],</span> <span class="n">by_ont</span><span class="p">,</span> <span class="n">uberon_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">)</span><span class="o">.</span><span class="n">relations</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>
            <span class="n">other_way</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Relations</span><span class="p">(</span><span class="n">rel</span><span class="p">,</span> <span class="p">[</span><span class="n">by_ont</span><span class="p">],</span> <span class="n">by_name</span><span class="p">,</span> <span class="n">uberon_obo</span><span class="o">.</span><span class="n">ont</span><span class="p">)</span><span class="o">.</span><span class="n">relations</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span>

            <span class="k">if</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">one_way</span><span class="p">)</span> <span class="ow">and</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">other_way</span><span class="p">):</span> <span class="c1"># no relation between by_ont and by_name</span>
                <span class="n">dis_line</span> <span class="o">=</span> <span class="p">[</span><span class="n">sample</span><span class="p">,</span> <span class="n">by_name</span><span class="p">,</span> <span class="n">by_ont</span><span class="p">,</span> <span class="n">relation_string</span><span class="p">,</span> <span class="n">ont</span><span class="p">[</span><span class="n">sample</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">]]</span>
                <span class="n">disagreements</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dis_line</span><span class="p">)</span>
                <span class="n">overall</span> <span class="o">=</span> <span class="n">by_name</span>
                <span class="n">mapped_using</span> <span class="o">=</span> <span class="s2">&quot;name&quot;</span>
                <span class="c1">#If one is (part of) another:</span>
            <span class="k">elif</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">one_way</span><span class="p">)</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">other_way</span><span class="p">):</span>
                <span class="n">overall</span> <span class="o">=</span> <span class="n">by_name</span>
                <span class="n">mapped_using</span> <span class="o">=</span> <span class="s2">&quot;name&quot;</span>
                <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="n">other_way</span><span class="p">)</span>
            <span class="k">elif</span> <span class="ow">not</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">one_way</span><span class="p">)</span> <span class="ow">and</span> <span class="n">pd</span><span class="o">.</span><span class="n">isna</span><span class="p">(</span><span class="n">other_way</span><span class="p">):</span>
                <span class="n">overall</span> <span class="o">=</span> <span class="n">by_ont</span>
                <span class="n">mapped_using</span> <span class="o">=</span> <span class="s2">&quot;ontology&quot;</span>
                <span class="n">logging</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s2">&quot;ont&quot;</span><span class="p">,</span> <span class="n">one_way</span><span class="p">)</span>
        
        <span class="n">mapping_line</span> <span class="o">=</span> <span class="p">[</span><span class="n">sample</span><span class="p">,</span> <span class="n">by_name</span><span class="p">,</span> <span class="n">by_ont</span><span class="p">,</span> <span class="n">relation_string</span><span class="p">,</span> <span class="n">name_matched_on</span><span class="p">,</span> <span class="n">name_string</span><span class="p">,</span> <span class="n">overall</span><span class="p">,</span> <span class="n">mapped_using</span><span class="p">]</span>
        <span class="n">overall_mapping</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">mapping_line</span><span class="p">)</span>
    
    <span class="n">overall_mapping</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">overall_mapping</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="n">mapping_columns</span><span class="p">)</span>
    <span class="n">overall_mapping</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;sample id&#39;</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

    <span class="n">disagreements</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">disagreements</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="n">disagreement_columns</span><span class="p">)</span>
    <span class="n">disagreements</span> <span class="o">=</span> <span class="n">disagreements</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;sample id&#39;</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">overall_mapping</span><span class="p">,</span> <span class="n">disagreements</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">overall_mapping</span><span class="p">,</span> <span class="n">disagreements</span> <span class="o">=</span> <span class="n">get_overall_tissue_mappings</span><span class="p">(</span><span class="n">samples</span><span class="p">,</span> <span class="n">map_by_tissue_name</span><span class="p">,</span> <span class="n">map_by_ont</span><span class="p">,</span> <span class="n">merged_ont</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">overall_mapping</span><span class="p">[</span><span class="n">overall_mapping</span><span class="o">.</span><span class="n">overall</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;unmapped&#39;</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">merged_ont</span><span class="p">[</span><span class="n">x</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">])</span>

</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">num_uberon_mapped</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">overall_mapping</span><span class="p">[</span><span class="s1">&#39;overall&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">-</span> <span class="mi">1</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">overall_mapping</span><span class="o">.</span><span class="n">index</span><span class="p">)</span><span class="si">}</span><span class="s1"> samples mapped to </span><span class="si">{</span><span class="n">num_uberon_mapped</span><span class="si">}</span><span class="s1"> unique UBERON terms.&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># TODO: read in accession to new sample id</span>
<span class="n">old_to_new_ff</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">new_ff</span> <span class="ow">in</span> <span class="n">samples_info</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="n">old_ff</span> <span class="o">=</span> <span class="n">new_ff</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39;_&#39;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">old_to_new_ff</span><span class="p">[</span><span class="n">old_ff</span><span class="p">]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">new_ff</span><span class="p">)</span>
    <span class="k">except</span><span class="p">:</span>
        <span class="n">old_to_new_ff</span><span class="p">[</span><span class="n">old_ff</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="n">new_ff</span><span class="p">]</span>
</pre></div>
</div>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="mi">0</span>
<span class="k">for</span> <span class="n">uberon</span> <span class="ow">in</span> <span class="n">overall_mapping</span><span class="p">[</span><span class="s1">&#39;overall&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">uberon</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
        <span class="k">continue</span>
    <span class="n">tissue_accession</span> <span class="o">=</span> <span class="p">[</span><span class="nb">len</span><span class="p">(</span><span class="n">old_to_new_ff</span><span class="p">[</span><span class="n">x</span><span class="p">])</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">overall_mapping</span><span class="p">[</span><span class="n">overall_mapping</span><span class="p">[</span><span class="s1">&#39;overall&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="n">uberon</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">]</span>
    <span class="c1"># TODO: might be mulitple samples per accession</span>
<span class="c1">#     break</span>
    <span class="n">total</span> <span class="o">+=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">tissue_accession</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">uberon</span><span class="p">,</span> <span class="n">merged_ont</span><span class="p">[</span><span class="n">uberon</span><span class="p">][</span><span class="s1">&#39;name&#39;</span><span class="p">],</span> <span class="nb">sum</span><span class="p">(</span><span class="n">tissue_accession</span><span class="p">))</span>
    
<span class="nb">print</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
<!-- #region -->
</div>
<div class="section" id="creating-an-alternate-mapping-with-text-mining">
<h3><span class="section-number">5.4.2.4. </span>Creating an alternate mapping with text-mining<a class="headerlink" href="#creating-an-alternate-mapping-with-text-mining" title="Permalink to this headline">¶</a></h3>
<div class="margin sidebar" id="stop-words">
<p class="sidebar-title">Stop words</p>
<p>Stop words are words that are filtered out before processing text using Natural Language Processing (NLP) methods.
These are usually very common words (e.g. “and”, ”the”), or word which are meaningless in the context of the analysis.</p>
</div>
<p>Failing this, text-mining is used to assist in mapping based on phenotype term names:</p>
<ul class="simple">
<li><p>First <a class="reference internal" href="#stop-words"><span class="std std-ref">stop words</span></a> are removed, using the base list in the Natural Language Toolkit (<code class="docutils literal notranslate"><span class="pre">nltk</span></code>) Python Package<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id189"><span>158</span></a>]</span>, and a small number of manually curated phenotypic stopwords (e.g. “phenotype”, “abnormality”).</p></li>
<li><p>Then, an exact match between an UBERON term’s name or synonyms and phenotype term’s name is searched for.</p></li>
<li><p>If such a match does not exist, individual words from the phenotype term name or synonyms are then searched for exactly, such that the HP term “abnormality of the head and neck” would be mapped to UBERON’s “head” and “neck” terms (but never “neck of radius”).</p></li>
<li><p>In cases where multiple terms are found, a common parent would be searched for, in the case of this example, “craniocervical region”.</p></li>
<li><p>Failing that, a similarity measure is used to search a list of related words. This finds important words in the document using <a class="reference internal" href="#tf-idf"><span class="std std-ref">TF-IDF</span></a>, and then creates a similarity measure based on co-occurance, e.g. “mental” and “brain” have high similarity since they often appear in the same document  (documents are term descriptions).</p></li>
<li><p>This list was then used to aid in manually mapping to relevant phenotype terms where one existed.</p></li>
</ul>
<div class="margin sidebar" id="tf-idf">
<p class="sidebar-title">TF-IDF</p>
<p>Term Frequency, Inverse Document Frequency is a common and basic measure in NLP which attempts to measure how representative a term (word) is of a document.
It is defined by <span class="math notranslate nohighlight">\(tfidf=tf(t,d) \cdot idf(t,D) = (f_{t,d}) \cdot (\frac{N}{abs{d \in D : t \in d}) \)</span> where <span class="math notranslate nohighlight">\(f_{t,d}\)</span> is the frequency of a term <span class="math notranslate nohighlight">\(t\)</span> in a document <span class="math notranslate nohighlight">\(d\)</span>, <span class="math notranslate nohighlight">\(N\)</span> is the number of documents, and <span class="math notranslate nohighlight">\({abs{d \in D : t \in d}\)</span> is the number of documents containing the term.</p>
</div>
<!-- #endregion -->
<!-- #region -->
</div>
</div>
<div class="section" id="when-are-transcripts-expressed">
<h2><span class="section-number">5.4.3. </span>When are transcripts “expressed”?<a class="headerlink" href="#when-are-transcripts-expressed" title="Permalink to this headline">¶</a></h2>
<p>The idea behind <code class="docutils literal notranslate"><span class="pre">filip</span></code> is that some proteins are predicted to affect phenotypes that they are unable to affect, because the environment in the tissue or cell means that the protein isn’t around to perform it’s function (or fail to).
And, we have a measure of gene expression, for which many proteins have <code class="docutils literal notranslate"><span class="pre">0</span></code> counts (and therefore <code class="docutils literal notranslate"><span class="pre">0</span></code> TPM) in many tissues, so we could apply the filter to this cut-off. But is it meaningful to do so?</p>
<p>Like all chemical reactions, transcription is a stochastic process; there is an element of randomness; to describe if a transcription event will happen at a specific moment you have to use statistics.
Genetically identical organisms with the same environment have different measured gene expression patterns<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id204"><span>159</span></a>]</span> and the same can be said for single cells from the same organism<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id209"><span>160</span></a>]</span>.
The reason that it’s hard to predict with precision whether a given protein will be transcribed at a given moment is that it depends on the concentration of different molecules in the cell and the energy of the system.
Transcription events which have a very low probability of occuring will happen sometimes and we will measure this.
If we sequenced the transcriptome in infinite depth, we might expect all transcripts to be expressed at some level.</p>
<div class="margin sidebar" id="transcriptional-noise">
<p class="sidebar-title">Transcriptional noise</p>
<p>Transcriptional noise is variation in rates of transcription due to the implicit stochasticity of the reaction process. The implication is that many transcripts with low counts do not play a big role and cells are known to have mechanisms to protect themselves from this noise<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id204"><span>159</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id208"><span>161</span></a>]</span>. Since is difficult to distinguish between meaningful and non-meaningful and expression, in differential expression analyses it is common to remove low count transcripts<span id="id5">[<a class="reference internal" href="../cz-end-matter/reference.html#id206"><span>162</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id207"><span>163</span></a>]</span>.
Similar noise occurs in the process of translation (translational noise).</p>
</div>
<p>When we look at expression data for a sample, it will just be a snapshot of the transcription in that sample, and one that isn’t necessarily representative of what’s happening all the time.
As we saw in <a href="#id6"><span class="problematic" id="id7">:num-ref:`fantom-protein-distribution`</span></a>, very low count values in a sample are extremely common, and these are usually considered to be difficult to distinguish from <span class="xref std std-ref">transcriptional-noise</span>: low levels of transcription with little effect are often randomly happening in the cell.
In addition to the biological stochasticity (which could possibly create phenotypic differences), RNA-Seq is sensitive to technical experimental artefacts (batch effects) due to differences in RNA extraction and library preparation<span class="xref std std-ref">Conesa2016-gq</span>.
In both cases, it is low counts where this is most difficult to correct for
So, it isn’t necessarily meaningful to take all genes expressed above <code class="docutils literal notranslate"><span class="pre">0</span></code> TPM as a sensible cut-off for whether a gene counts as “expressed” or not in a tissue: when I dichotomise proteins as “expressed” or “not expressed”, I am using this as a convenient shorthand for “meaningfully expressed” or “not meaningfully expressed”.</p>
<p>We also know that proteins that do cause phenotypes are likely to be highly expressed in tissues related to the phenotype, so we definitely want to keep protein-phenotype predictions where proteins are produced at high levels in the tissue of interest, but when do TPM levels become low enough that we would want to exclude them?</p>
<div class="section" id="expression-cut-off">
<h3><span class="section-number">5.4.3.1. </span>Expression cut-off<a class="headerlink" href="#expression-cut-off" title="Permalink to this headline">¶</a></h3>
<p>One straight forward way is to choose a cut-off, and choosing this for TPM-normalised data rather than counts or R/FKPM values is sensible as it allows for comparing between different samples.
The cut-off was chosen by plotting the distribution of TPM expression and choosing a value below which there appeared to be little noise (10 TPM) between biological and technical replicates.</p>
</div>
<div class="section" id="expression-score">
<h3><span class="section-number">5.4.3.2. </span>Expression score<a class="headerlink" href="#expression-score" title="Permalink to this headline">¶</a></h3>
<p>However, for CAFA, it wasn’t necessary to choose a single cut-off, since the submission format is to choose a single score.</p>
<!-- #endregion -->
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="creation-of-input-predictions">
<h2><span class="section-number">5.4.4. </span>Creation of input predictions<a class="headerlink" href="#creation-of-input-predictions" title="Permalink to this headline">¶</a></h2>
<div class="section" id="naive-predictions">
<h3><span class="section-number">5.4.4.1. </span>Naive predictions<a class="headerlink" href="#naive-predictions" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="dcgo">
<h3><span class="section-number">5.4.4.2. </span>dcGO<a class="headerlink" href="#dcgo" title="Permalink to this headline">¶</a></h3>
<p>To create the input to DcGO, I used:</p>
<ul class="simple">
<li><p>BioPython<span id="id8">[<a class="reference internal" href="../cz-end-matter/reference.html#id190"><span>164</span></a>]</span>’s <code class="docutils literal notranslate"><span class="pre">Bio.SeqIO</span></code> interface for reading CAFA FASTA files.</p></li>
<li><p>SUPERFAMILY<span id="id9">[<a class="reference internal" href="../cz-end-matter/reference.html#id47"><span>73</span></a>]</span> <a class="reference external" href="https://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/cgi-bin/save.cgi?var=ht%3Btype=ass">domain assignments for Homo Sapiens</a>.</p></li>
<li><p>UniprotKB<span id="id10">[<a class="reference internal" href="../cz-end-matter/reference.html#id191"><span>165</span></a>]</span>’s <a class="reference external" href="https://www.uniprot.org/uploadlists/">mapping tool</a> to create a mapping between the UniprotKB id’s provided by CAFA and the ENSP ID’s used by SUPERFAMILY.</p></li>
</ul>
<p>Then, to create the DcGO only entry, I used the <code class="docutils literal notranslate"><span class="pre">DcGOR</span></code> library<span id="id11">[<a class="reference internal" href="../cz-end-matter/reference.html#id192"><span>166</span></a>]</span> (the <code class="docutils literal notranslate"><span class="pre">dcAlgoPredictMain</span></code> function).</p>
</div>
</div>
<div class="section" id="validation-methodology">
<h2><span class="section-number">5.4.5. </span>Validation Methodology<a class="headerlink" href="#validation-methodology" title="Permalink to this headline">¶</a></h2>
<div class="section" id="cafa-validation">
<h3><span class="section-number">5.4.5.1. </span>CAFA Validation<a class="headerlink" href="#cafa-validation" title="Permalink to this headline">¶</a></h3>
<p>This confidence score allows for a range of possible sets of predictions, depending on the threshold parameter <span class="math notranslate nohighlight">\(\tau\)</span>.
Precision (the proportion of selected items that are relevant), and recall (the proportion of relevant items that are selected) are defined as:</p>
<p><span class="math notranslate nohighlight">\(precision = p = \frac{t_p}{t_p + f_p}\)</span>
<span class="math notranslate nohighlight">\(recall = r = \frac{t_p}{t_p + f_n}\)</span></p>
<p>Precision-recall curves are generally used to validate a predictors performance, but the <span class="math notranslate nohighlight">\(F_1\)</span> measure combines these into a single measure of performance:</p>
<p><span class="math notranslate nohighlight">\(F_1 =2/ \frac{precision \cdot recall}{precision + recall}\)</span></p>
<p>Since the precision and recall will be different for any <span class="math notranslate nohighlight">\(\tau\)</span>, the <span class="math notranslate nohighlight">\(F_{max}\)</span> score is the maximum possible {math{<code class="docutils literal notranslate"><span class="pre">F_1</span></code>} for any value of <span class="math notranslate nohighlight">\(\tau\)</span>.</p>
<p>CAFA validation can either be term-centric or protein-centric. For each option, submissions are assessed per species and for wholly unknown and partially known genes separately.</p>
</div>
<div class="section" id="limitations-of-validation-method">
<h3><span class="section-number">5.4.5.2. </span>Limitations of validation method<a class="headerlink" href="#limitations-of-validation-method" title="Permalink to this headline">¶</a></h3>
<p>There is no penalty for making a broad guess, or reward for making a precise one. This is one of the reasons that the naive method does so well: for example it is not penalised for guessing that the root term of the GO BPO ontology Biological Process is related to every gene.</p>
<p>Due to the nature of the validation set, it’s possible that the best-scoring CAFA methods simply predict which associations are likely to be discovered soon (i.e. associations to genes people are currently studying).</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c05-filter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="4-data.html" title="previous page"><span class="section-number">5.3. </span>Input Data</a>
    <a class='right-next' id="next-link" href="6-results.html" title="next page"><span class="section-number">5.5. </span>Results</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>