(proteins-organised)=
# Looking more closely at proteins: function, structure and classification
Just as DNA has been classified at different levels (SNP, gene, genome), so too have proteins. In contrast to DNA, what's interesting (and useful!) to know about proteins is their structure.

{ref}`As mentioned earlier<what-is-translation>`, translated strings of amino acids fold automatically (or sometimes with the help of other proteins) into the 3D structure of proteins. 
{ref}`We still don't understand how this happens very well.<protein-folding-problem>`

[//]: # (TODO: Mention deep mind? Complain it's not science if no one can see it)

(protein-folding-problem)=
```{margin} The protein folding problem
Amino acid strings reliably fold into globular protein structures, but we don't know exactly how or why and can't accurately predict what form a string of amino acids will take as a protein.
Answering these related problems is known as the protein folding problem and has been one of the grandest challenges in biology for over half a century.
```

What we do understand however, is that the structure of proteins defines what molecules (large and small) they can bind to, and this has huge consequences for their functionality in the body.
Moreover, proteins have recognisable features of their structure which appear again and again. These features occur at different levels and sizes, and they allow us to learn about the evolution and functional similarity of proteins.

## Protein structure: Primary, Secondary, Tertiary, and Quaternary
Proteins are described and classified in terms of the their primary, secondary, tertiary, and quaternary structure. 

The primary structure is simply the amino acid makeup of the protein, which describes the protein's chemical makeup, but not it's three-dimensional structure. 
These amino acid strings tend to form into a small number of familiar (secondary) three-dimensional structures, for example beta strands, beta sheets, and alpha helices.
At the next level, the tertiary structure describes combinations of secondary structures, for example a TIM barrels (a torroidal structure made up of alpha helices and beta strands). 
At this level, similar structures do not imply an evolutionary or functional similarity.

### Quaternary structures: protein domains
[//]: # (TODO: Add citations.)

```{figure} ../images/lego.png
---
height: 100px
name: lego
alt: The figure shows a red lego block linked to a yellow lego block, which is then linked to another yellow lego block. The links are thin grey lego blocks.
---
An illustration of the lego analogy for protein domains. Coloured bricks represent protein domains - colour represents a specific protein domain type, while thin grey bricks represent polypeptide linkers which link domains. Image created using mecabricks{cite}`Jarraud2019-jj`
``` 

The quaternary structures of proteins - protein domains - have proved particularly interesting for research. 
A simple and oft-used metaphor is to think of protein domains as lego building blocks ({numref}`lego`) which can be linked by polypeptide chains to make up a protein. 
These polypeptide chains (known as linkers) are often inflexible, in order to allow only one conformation of the protein. 
Small and simple proteins often consist of just one domain, while bigger proteins can contain many domains. 
An individual domain can be found in many different proteins, and multiple times in the same protein. 

Protein domains are interesting because they are highly conserved in evolution, and are thought of as units of function, evolution, and/or structure. 
The functions of proteins, at both low-level (e.g. “calcium signalling protein”) and high-level (e.g. annotated to “liver disease”), are costly and difficult to discern, so there are many proteins about which little is known. 
For this reason, often proteins are classified according to their similarity to proteins about which functions are known, for example those containing the same protein domains.

### Disorder
While protein domains always exist in one confirmation, this is not the case for proteins as a whole.
One reason for this is that not all linkers are rigid. 
Flexible regions of proteins which allow for various conformations are referred to as *disordered*. 

And disordered regions are not only relegated to linkers between domains. 
Proteins can be constituted entirely of disordered regions, or may have large disordered regions.

Such *intrinsically disordered* proteins can exist in a number of conformations, rather than one fixed structure. 
On some occasions, the disordered regions are known to be functional, while on others, proteins may be non-functional until they bind with another macromolecule which forces them into a fixed conformation.

(protein-classification)=
### Classifying proteins by domain: families and superfamilies
Proteins with known domain structure can be grouped together based on their structural similarities, based on the consideration of the protein’s constituent domains into *familes*, *superfamilies* and *folds*. 
Proteins are classified into *families* representing the most similar proteins, which share a clear evolutionary relationship, while *superfamilies* represent less close evolutionary relationships, and *folds* represent the same secondary structure. 
This protein classification task, while aided by automation, were carried out largely by manual visual inspection{cite}`Murzin1995-se`.
