

<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>6. Combining RNA-seq datasets &#8212; Phenotype from Genotype</title>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css" integrity="sha384-KA6wR/X5RY4zFAHpv/CnoG2UW1uogYfdnP67Uv7eULvTveboZJg0qUpmJZb5VqzN" crossorigin="anonymous">
    <link href="../_static/css/index.css" rel="stylesheet">
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-dropdown.css" />
    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/sphinx-book-theme.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="6.1.1. Background" href="1-background.html" />
    <link rel="prev" title="5. Phenotype prediction with Snowflake" href="../c04-snowflake/0-snowflake.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="docsearch:language" content="en">



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div>

<form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>

<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../jupyter_book_intro.html">
   Start Here
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/title-page.html">
   Title Page
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-figures.html">
   List of Figures
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-tables.html">
   List of Tables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/declaration.html">
   Declaration
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c02-biology-background/0-bg-biology.html">
   2. Biology Background
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c02-biology-background/0-bg-biology.html#id18">
   3. Biology Background
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c03-compbio-background/0-compbio-bg.html">
   4. Computational Biology Background
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c04-snowflake/0-snowflake.html">
   5. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   6. Combining RNA-seq datasets
  </a>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="1-background.html">
     6.1.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-data.html">
     6.1.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-data-wrangling.html">
     6.1.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-uberon.html">
     6.1.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-uberon.html#functionality">
     6.1.5. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-batch-correction.html">
     6.1.6. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html">
     6.1.7. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-future-work.html">
     6.1.8. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c06-filter/0-filter.html">
   7. Filtering computational predictions with tissue-specific expression information
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   8. Conclusion
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav>

 <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        <div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    
    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/c05-combining/0-index.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
    
</div>
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#motivation">
   6.1. Motivation
  </a>
 </li>
</ul>

        </nav>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="combining-rna-seq-datasets">
<span id="c05-combining"></span><h1><span class="section-number">6. </span>Combining RNA-seq datasets<a class="headerlink" href="#combining-rna-seq-datasets" title="Permalink to this headline">¶</a></h1>
<p>This chapter describes my attempts to create a simulated tissue-specific dataset and to combine gene expression data from 4 large tissue-specific RNA-Seq experiments. All work in this chapter is my own.</p>
<div class="section" id="motivation">
<h2><span class="section-number">6.1. </span>Motivation<a class="headerlink" href="#motivation" title="Permalink to this headline">¶</a></h2>
<p>Gene expression data measures the number of transcribed proteins in a sample at a specific time. It’s a popular experimental technique as demonstrated by the 3,564 studies and 112,225 assays currently available on Gene Expression Atlas website <a class="bibtex reference internal" href="../end-matter/reference.html#petryszak2016-je" id="id1">[20]</a>. Researchers use this data to characterise gene function.</p>
<p>Yet each individual measure of expression is only a snapshot of what a gene can do. It only tells us about the transcription of proteins at that one time. If we want a full understanding of what a gene does, we must understand how it’s expressed in a variety of scenarios. For example, in different tissues, from different people, at different times of day, and across many repeats. That kind of data would allow us to answer questions that aren’t currently possible. Data to enable this is not easy to come by. It would be too expensive for one experiment to measure so many samples.</p>
<p>Combining expression data from many different experiments could overcome this problem. There are already an example of this happening for two experiments<a class="bibtex reference internal" href="../end-matter/reference.html#wang2018-rz" id="id2">[2]</a>. It would result in the rich information about genes that we desire. Unfortunately, this approach reveals another problem: batch effects. Batch effects are measurement artifacts that appear due to differences in experimental protocol. Their prevalence makes gene expression experiments very difficult to compare or combine.</p>
<p>This chapter combines four gene expression data sets, collating sample and tissue information about them into a common format. A simulated data set for tissue-specific batch-effected data is created as a starting point for exploring batch effect removal for this combined data set.</p>
<hr class="docutils" />
<p><strong>Page References</strong></p>
<p id="bibtex-bibliography-c05-combining/0-index-0"><dl class="citation">
<dt class="bibtex label" id="petryszak2016-je"><span class="brackets"><a class="fn-backref" href="#id1">20</a></span></dt>
<dd><p>Robert Petryszak, Maria Keays, Y Amy Tang, Nuno A Fonseca, Elisabet Barrera, Tony Burdett, Anja Füllgrabe, Alfonso Muñoz-Pomer Fuentes, Simon Jupp, Satu Koskinen, Oliver Mannion, Laura Huerta, Karine Megy, Catherine Snow, Eleanor Williams, Mitra Barzine, Emma Hastings, Hendrik Weisser, James Wright, Pankaj Jaiswal, Wolfgang Huber, Jyoti Choudhary, Helen E Parkinson, and Alvis Brazma. Expression atlas update—an integrated database of gene and protein expression in humans, animals and plants. 2016. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gkv1045">http://dx.doi.org/10.1093/nar/gkv1045</a>.</p>
</dd>
<dt class="bibtex label" id="wang2018-rz"><span class="brackets"><a class="fn-backref" href="#id2">2</a></span></dt>
<dd><p>Qingguo Wang, Joshua Armenia, Chao Zhang, Alexander V Penson, Ed Reznik, Liguo Zhang, Thais Minet, Angelica Ochoa, Benjamin E Gross, Christine A Iacobuzio-Donahue, Doron Betel, Barry S Taylor, Jianjiong Gao, and Nikolaus Schultz. Unifying cancer and normal RNA sequencing data from different sources. <em>Sci Data</em>, 5:180061, April 2018. URL: <a class="reference external" href="http://dx.doi.org/10.1038/sdata.2018.61">http://dx.doi.org/10.1038/sdata.2018.61</a>.</p>
</dd>
</dl>
</p>
<div class="toctree-wrapper compound">
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c05-combining"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="../c04-snowflake/0-snowflake.html" title="previous page"><span class="section-number">5. </span>Phenotype prediction with <code class="docutils literal notranslate"><span class="pre">Snowflake</span></code></a>
    <a class='right-next' id="next-link" href="1-background.html" title="next page"><span class="section-number">6.1.1. </span>Background</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    <script src="../_static/js/index.js"></script>
    
  </body>
</html>