
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>5.3. Data Wrangling &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="5.4. The uberon-py package" href="4-uberon.html" />
    <link rel="prev" title="5.2. Data Acquisition" href="2-data.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Sources of bias in computational biology datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-error.html">
     3.5. Sources of error in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   5. Combining RNA-seq datasets
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-background.html">
     5.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-data.html">
     5.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     5.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-uberon.html">
     5.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-batch-correction.html">
     5.5. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html">
     5.6. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-results.html#combined-data-set">
     5.7. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     5.8. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="8-future-work.html">
     5.9. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-data.html">
     6.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-methods.html">
     6.3. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/4-results.html">
     6.4. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-discussion.html">
     6.5. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-future-work.html">
     6.6. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#obtaining-raw-expression-per-gene-for-healthy-human-tissues">
   5.3.1. Obtaining raw expression per gene for healthy human tissues
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data-acquisition">
     5.3.1.1. Data Acquisition
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#mapping-from-transcript-to-gene">
     5.3.1.2. Mapping from transcript to gene
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#filtering-out-disease-samples">
     5.3.1.3. Filtering out disease samples
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#mapping-to-uberon">
   5.3.2. Mapping to UBERON
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#aggregating-metadata">
   5.3.3. Aggregating Metadata
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="data-wrangling">
<span id="c05-3-data-wrangling"></span><h1><span class="section-number">5.3. </span>Data Wrangling<a class="headerlink" href="#data-wrangling" title="Permalink to this headline">¶</a></h1>
<p>Before the data sets could be combined, substantial data wrangling was necessary. The details of these processes - obtaining, checking, mapping identifiers, and excluding irrelevant data - are described in this section.
A Python package (<code class="docutils literal notranslate"><span class="pre">uberon_py</span></code>) was developed and used to do much of this mapping.
It’s functionality is described in <a class="reference internal" href="4-uberon.html#uberon-py"><span class="std std-numref">Section 5.4</span></a>.</p>
<p>The steps required to obtain consistently formatted and labelled data can be described as follows:</p>
<ul class="simple">
<li><p>Obtaining the raw expression per gene for healthy human tissues</p>
<ul>
<li><p>Data acquisition</p></li>
<li><p>(Where required) Mapping from transcript to gene</p></li>
<li><p>(Where required) Filtering out disease samples</p></li>
<li><p>(Where required) Filtering out non-human samples</p></li>
</ul>
</li>
<li><p>Mapping from sample name to UBERON tissue using <a class="reference internal" href="4-uberon.html#uberon-py"><span class="std std-ref">uberon-py</span></a>.</p></li>
<li><p>Aggregating metadata</p></li>
</ul>
<div class="dropdown admonition">
<p class="admonition-title">FANTOM5 data pipeline code</p>
<p>This code makes use of the <span class="xref myst">fantom.py helper script</span>.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span>
<span class="c1"># General:</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">time</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">uberon_py</span> <span class="kn">import</span> <span class="n">obo</span> <span class="c1"># my package</span>

<span class="c1"># My helper functions:</span>
<span class="kn">from</span> <span class="nn">helper_c05</span> <span class="kn">import</span> <span class="n">fantom</span>

<span class="c1">#load ontologies</span>
<span class="n">fantom_obo_file</span> <span class="o">=</span> <span class="s1">&#39;data/experiments/fantom/ff-phase2-170801.obo.txt&#39;</span>
<span class="n">uberon_obo_file</span> <span class="o">=</span> <span class="s1">&#39;data/uberonext_obo.txt&#39;</span>
<span class="n">fantom_obo_root_terms</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;FF:0000001&#39;</span><span class="p">,</span><span class="s1">&#39;EFO:0000001&#39;</span><span class="p">]</span>

<span class="n">fantom_obo</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Obo</span><span class="p">(</span><span class="n">fantom_obo_file</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;FF&#39;</span><span class="p">,</span><span class="s1">&#39;CL&#39;</span><span class="p">,</span><span class="s1">&#39;UBERON&#39;</span><span class="p">],</span> <span class="n">fantom_obo_root_terms</span><span class="p">)</span>
<span class="n">uberon_obo</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Obo</span><span class="p">(</span><span class="n">uberon_obo_file</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">,</span><span class="s1">&#39;CL&#39;</span><span class="p">],</span> <span class="n">fantom_obo_root_terms</span><span class="p">)</span>

<span class="c1">#load and process fantom counts data </span>
<span class="n">fantom_count_file_loc</span> <span class="o">=</span> <span class="s1">&#39;data/experiments/fantom/hg38_fair+new_CAGE_peaks_phase1and2_counts_ann.osc.txt&#39;</span>
<span class="n">ensg_mapping_loc</span> <span class="o">=</span> <span class="s1">&#39;data/experiments/fantom/biomart_ensg_hgnc.txt&#39;</span>
<span class="n">samples_info_file</span> <span class="o">=</span> <span class="s1">&#39;data/experiments/fantom/fantom_humanSamples2.0.csv&#39;</span>
<span class="n">fantom_counts</span> <span class="o">=</span> <span class="n">fantom</span><span class="o">.</span><span class="n">Fantom</span><span class="p">(</span><span class="n">fantom_count_file_loc</span><span class="p">,</span>
                              <span class="n">ensg_mapping_loc</span><span class="p">,</span> 
                              <span class="n">fantom_obo</span><span class="p">,</span> 
                              <span class="n">uberon_obo</span><span class="p">,</span> 
                              <span class="n">samples_info_file</span><span class="p">)</span>

<span class="n">fantom_counts</span><span class="o">.</span><span class="n">gene_expression</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/fantom/fantom_gene_expression.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">fantom_counts</span><span class="o">.</span><span class="n">funnel_plot_samples</span><span class="o">.</span><span class="n">items</span><span class="p">(),</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Sample Data Stage&#39;</span><span class="p">,</span> <span class="s1">&#39;Number of samples&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/fantom/funnel_plot_samples.tsv&#39;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">fantom_counts</span><span class="o">.</span><span class="n">funnel_plot_transcripts</span><span class="o">.</span><span class="n">items</span><span class="p">(),</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Transcript Data Stage&#39;</span><span class="p">,</span> <span class="s1">&#39;Number of transcripts&#39;</span><span class="p">])</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/fantom/funnel_plot_transcripts.tsv&#39;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="obtaining-raw-expression-per-gene-for-healthy-human-tissues">
<h2><span class="section-number">5.3.1. </span>Obtaining raw expression per gene for healthy human tissues<a class="headerlink" href="#obtaining-raw-expression-per-gene-for-healthy-human-tissues" title="Permalink to this headline">¶</a></h2>
<div class="section" id="data-acquisition">
<h3><span class="section-number">5.3.1.1. </span>Data Acquisition<a class="headerlink" href="#data-acquisition" title="Permalink to this headline">¶</a></h3>
<p>As mentioned in <span class="xref std std-ref">data-aquisition</span>, for the HPA, GTeX and HDBR experiments, count data were available through the <em>ExpressionAtlas</em> R package<a class="bibtex reference internal" href="../end-matter/reference.html#keays2018-pg" id="id1">[165]</a>, and the FANTOM dataset was downloaded directly.</p>
</div>
<div class="section" id="mapping-from-transcript-to-gene">
<h3><span class="section-number">5.3.1.2. </span>Mapping from transcript to gene<a class="headerlink" href="#mapping-from-transcript-to-gene" title="Permalink to this headline">¶</a></h3>
<p>This step was only required for the FANTOM dataset.</p>
<p>FANTOM provides mappings to gene IDs based on proximity of genes to peaks according to Ensembl. Gene expression was then calculated by summing over transcripts mapped to genes. The transcripts were already mapped to HGNC gene identifiers in the downloaded FANTOM file and <a class="reference external" href="https://www.ensembl.org/biomart">Ensembl’s Biomart</a> was used to obtain a <span class="xref myst">mapping from HGNC gene identifiers to ENSG gene identifiers</span>, in order to match the gene expression atlas format.</p>
<p>Any transcripts which mapped to multiple genes were discarded, as were any HGNC ids which did not map to ENSG ids.</p>
</div>
<div class="section" id="filtering-out-disease-samples">
<h3><span class="section-number">5.3.1.3. </span>Filtering out disease samples<a class="headerlink" href="#filtering-out-disease-samples" title="Permalink to this headline">¶</a></h3>
<p>The HDBR and HPA experiments contained only healthy samples.</p>
<p><strong>GTEx</strong>
Although GTEx contained clinical data, no disease-related phenotypes were removed from the data set, since the <code class="docutils literal notranslate"><span class="pre">disease</span></code> column contains only values of “normal” and the only clinical variables (as described in the <code class="docutils literal notranslate"><span class="pre">clinical_variables</span></code> column) in the dataset were sun exposure or lack thereof for skin tissues. I judged these to be within the normal range of environments that we would expect skin to be subjected to.</p>
<p><strong>FANTOM</strong>
The FANTOM sample ontology was used to remove samples which are models for diseases. Samples which are disease models are identified using the <code class="docutils literal notranslate"><span class="pre">is_model_for</span></code> relationship and these relationships are propagated to the children terms based on the <code class="docutils literal notranslate"><span class="pre">is_a</span></code> relationship. For example, <code class="docutils literal notranslate"><span class="pre">FF:11558-120D1</span></code> (Fibroblast - skin spinal muscular atrophy, donor2) would be removed from the set of samples, since:
<code class="docutils literal notranslate"><span class="pre">FF:11558-120D1</span></code> (Fibroblast - skin spinal muscular atrophy, donor2) <code class="docutils literal notranslate"><span class="pre">is_a</span> <span class="pre">FF:0000251</span></code> (human fibroblast - skin spinal muscular atrophy sample) <code class="docutils literal notranslate"><span class="pre">is_model_for</span> <span class="pre">DOID:12377</span></code> (spinal muscular atrophy).</p>
<p><strong>1D. Filtering out non-human samples</strong></p>
<p>The GTEx, HDBR, and HPA experiments contained only human samples.</p>
<p><strong>FANTOM</strong>
The FANTOM5 data set also contains non-human (mouse) samples. The FANTOM sample ontology (which was downloaded <a class="reference external" href="http://fantom.gsc.riken.jp/5/datafiles/latest/extra/Ontology/ff-phase2-170801.obo.txt">from here</a>) was used to look-up which FANTOM samples are human samples, i.e. have an <code class="docutils literal notranslate"><span class="pre">is_a</span></code> relationship to the term <code class="docutils literal notranslate"><span class="pre">FF:0000210</span></code> (human sample) directly or indirectly.</p>
</div>
</div>
<div class="section" id="mapping-to-uberon">
<h2><span class="section-number">5.3.2. </span>Mapping to UBERON<a class="headerlink" href="#mapping-to-uberon" title="Permalink to this headline">¶</a></h2>
<p>Mapping from samples to Uberon tissue required the development of a small Python package <code class="docutils literal notranslate"><span class="pre">uberon-py</span></code>, which is described in detail in <a class="reference internal" href="4-uberon.html#uberon-py"><span class="std std-ref">the next section</span></a>. To create input to this package, informal tissue names (e.g. blood, kidney) were taken from the experimental design files (or the human sample information file for FANTOM) to create a map of samples to informal tissue names.
For FANTOM, the FANTOM ontology could also be used to create a more fine-grained mapping of samples to tissues based on FANTOM sample identifiers and/or cell type (CL) identifiers.</p>
<p><strong>HPA</strong>
The HPA samples were mapped using exact matches to Uberon names.
Three types of sample did not have exact matches: <em>transformed skin fibroblast</em>, <em>suprapubic skin</em>, and <em>ebv-transformed lymphocyte</em>.
I manually mapped <em>suprapubic skin</em> to <code class="docutils literal notranslate"><span class="pre">UBERON:0001415</span></code> <em>Skin of pelvis</em>, and excluded the other two (corresponding to excluding 869 samples).</p>
<p><strong>HDBR</strong>
For HDBR, tissue names from the “organism part’ column of the column data file were matched to Uberon names and synonyms from the Uberon extended ontology. The 96 unmatched terms corresponding to mixed brain tissues and brain fragments were defaulted to the more general Uberon Brain term.</p>
<p><strong>FANTOM</strong>
Since an experimental design file could not be obtained for FANTOM via GxA, additional sample information was obtained via the FANTOM5 website, namely the <a class="reference external" href="https://fantom.gsc.riken.jp/5/datafiles/reprocessed/hg38_latest/basic/HumanSamples2.0.sdrf.xlsx">human sample information file</a> and the FANTOM5 ontology.</p>
<p>FANTOM also contains time courses of cell differentiation (cells changing from one type to another) as well measures of perturbed cells.
Since these samples do not have a well-defined locality in the body given by cell or tissue type, they were not used in the combined dataset.
Such samples were filtered out using the human sample information file.</p>
<p>Since the FANTOM data had both an ontology file and the human sample information file, both were used to map to Uberon.
The disagreements between the two mappings revealed some inconsistencies with the data set: these are described in <a class="reference internal" href="4-uberon.html#fantom5-inconsistencies-example"><span class="std std-ref">the next section</span></a>, as they demonstrate a potential use case for the <code class="docutils literal notranslate"><span class="pre">uberon-py</span></code> package.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">plotly</span> <span class="kn">import</span> <span class="n">graph_objects</span> <span class="k">as</span> <span class="n">go</span>
<span class="kn">from</span> <span class="nn">plotly.subplots</span> <span class="kn">import</span> <span class="n">make_subplots</span>

<span class="c1"># from myst_nb import glue</span>
<span class="kn">import</span> <span class="nn">textwrap</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="n">funnel_plot_transcripts</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/fantom/funnel_plot_transcripts.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="n">funnel_plot_samples</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/fantom/funnel_plot_samples.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>

<span class="n">fig</span> <span class="o">=</span> <span class="n">make_subplots</span><span class="p">(</span><span class="n">rows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">horizontal_spacing</span><span class="o">=</span><span class="mf">0.3</span><span class="p">)</span>


<span class="n">transcript_y</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;&lt;br&gt;&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">textwrap</span><span class="o">.</span><span class="n">wrap</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="mi">15</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">list</span><span class="p">(</span><span class="n">funnel_plot_transcripts</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go</span><span class="o">.</span><span class="n">Funnel</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s1">&#39;Transcripts&#39;</span><span class="p">,</span> 
              <span class="n">y</span><span class="o">=</span><span class="n">transcript_y</span><span class="p">,</span>
              <span class="n">x</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">funnel_plot_transcripts</span><span class="p">[</span><span class="s1">&#39;Number of transcripts&#39;</span><span class="p">])),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
    <span class="n">col</span><span class="o">=</span><span class="mi">1</span>
<span class="p">)</span>

<span class="n">sample_y</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;&lt;br&gt;&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">textwrap</span><span class="o">.</span><span class="n">wrap</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="mi">27</span><span class="p">))</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">list</span><span class="p">(</span><span class="n">funnel_plot_samples</span><span class="o">.</span><span class="n">index</span><span class="p">)]</span>
<span class="n">fig</span><span class="o">.</span><span class="n">add_trace</span><span class="p">(</span>
    <span class="n">go</span><span class="o">.</span><span class="n">Funnel</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s1">&#39;Samples&#39;</span><span class="p">,</span> 
              <span class="n">y</span><span class="o">=</span><span class="n">sample_y</span><span class="p">,</span>
              <span class="n">x</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">funnel_plot_samples</span><span class="p">[</span><span class="s1">&#39;Number&#39;</span><span class="p">])),</span>
    <span class="n">row</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
    <span class="n">col</span><span class="o">=</span><span class="mi">2</span>
<span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">update_layout</span><span class="p">(</span><span class="n">showlegend</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                  <span class="n">autosize</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
                  <span class="n">width</span><span class="o">=</span><span class="mi">800</span><span class="p">,</span>
                  <span class="n">height</span><span class="o">=</span><span class="mi">400</span><span class="p">,</span>
                  <span class="n">margin</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span>
                      <span class="n">l</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span>
                      <span class="n">r</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span>
                      <span class="n">b</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span>
                      <span class="n">t</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span>
                      <span class="n">pad</span><span class="o">=</span><span class="mi">2</span>
                  <span class="p">)</span>
<span class="p">)</span>

<span class="n">fig</span><span class="o">.</span><span class="n">write_image</span><span class="p">(</span><span class="s1">&#39;../images/fantom_funnel.png&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">write_html</span><span class="p">(</span><span class="s1">&#39;../images/fantom_funnel.html&#39;</span><span class="p">)</span>

<span class="c1"># glue(&#39;transcript-funnel&#39;, fig, display=False)</span>
</pre></div>
</div>
</div>
</div>
<div class="figure align-default" id="fantom-funnel-interactive">
<img alt="../_images/fantom_funnel.html" src="../_images/fantom_funnel.html" />
<p class="caption"><span class="caption-number">Fig. 5.3 </span><span class="caption-text">Funnel plot showing the data cleaning pipeline for FANTOM transcripts/genes (left) and samples (right), along with the number which remained after each stage of data cleaning.</span><a class="headerlink" href="#fantom-funnel-interactive" title="Permalink to this image">¶</a></p>
</div>
<p>The amount of data that flows through the processing pipeline for the FANTOM5 dataset can be seen in <a class="reference internal" href="#fantom-funnel-interactive"><span class="std std-numref">Fig. 5.3</span></a>.</p>
</div>
<div class="section" id="aggregating-metadata">
<h2><span class="section-number">5.3.3. </span>Aggregating Metadata<a class="headerlink" href="#aggregating-metadata" title="Permalink to this headline">¶</a></h2>
<p>To create consistent metadata for the samples (e.g. age, developmental stage, replicate status, etc), information was extracted from multiple sources (including GxA and additional data from each experiment), and sometimes manually curated or corrected.</p>
<p><strong>HPA, HBDR, and GTEx:</strong>
Metadata about the experiments was collected from multiple sources, primarily the column data files accessed via ExpressionAtlas.
This metadata was used to describe the the experimental design for ComBat.
The metadata collected includes (where available), sample identifier, individual identifier, age (exact), age (range), developmental stage, tissue type (as Uberon term), sex, experiment, biological replicate identifier and technical replicate identifier.</p>
<p>Both age variables are given in years and may include negative values (e.g. for a developing fetus).
The age (range) variable contains uneven ranges, since this allows there to be an age-related factor that is compatible across the experiments.
These values had to be converted to common units, since they were incompatible between experiments, and age-related terms were missing in GxA for GTEx and HPA.
For GTEx it was possible to acquire this information via <a class="reference external" href="https://storage.googleapis.com/gtex_analysis_v7/annotations/GTEx_v7_Annotations_SubjectPhenotypesDS.txt">its own website</a>.</p>
<p><strong>FANTOM:</strong>
FANTOM metadata collection was mostly taken from the human sample information file.
There were discrepancies between ages and developmental stages in the FANTOM human samples file, for example, sample <code class="docutils literal notranslate"><span class="pre">FF:10027-101D9</span></code> is labelled as <em>thymus, adult, pool1</em> in the <em>Description</em> field, but as <em>0.5,0.5,0.83 years old infant</em> in the <em>Developmental Stage</em> field, and sample <code class="docutils literal notranslate"><span class="pre">FF:10209-103G2</span></code> has an age of ‘M’ and a sex of ‘28’.
There were also numerous typographical inconsistencies, for example, “3 year old child”, “3 years old child”, “25 year old”, “76” and “76 years old adult” all feature in the same column, amongst other errors.
For this reason, creating a cleaned experimental design file was laborious, but the resulting file has been sent to the FANTOM data curators so that they might make it officially available.</p>
<p>FANTOM technical and biological replicates are indicated in the annotated gene expression FANTOM file, by the inclusion of “tech_rep” or “biol_rep” in the long sample labels e.g. <code class="docutils literal notranslate"><span class="pre">counts.Dendritic%20Cells%20-%20monocyte%20immature%20derived%2c%20donor1%2c%20tech_rep1.CNhs10855.11227-116C3.hg38.nobarcode</span></code>.
These were used to create the experimental design file.</p>
<p>Note: there is an error in the original transcript expression file for one of these identifiers (<code class="docutils literal notranslate"><span class="pre">counts.Dendritic%20Cells%20-%20monocyte%20immature%20derived%2c%20donor1%2c%20rep2.CNhs11062.11227-116C3.hg38.nobarcode</span></code>) such that it is missing the “tech” part of the the replicate label.
This was manually changed in my copy of the input file and the FANTOM data curation team was informed.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># CREATE EXP DESIGN FILES: FANTOM</span>
</pre></div>
</div>
</div>
</div>
<div class="dropdown admonition">
<p class="admonition-title">Code to harmonise meta-data for GxA data sets</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span>
<span class="c1"># CREAT EXP DESIGN FILES: GXA</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">from</span> <span class="nn">uberon_py</span> <span class="kn">import</span> <span class="n">obo</span> 
<span class="kn">from</span> <span class="nn">helper_c05</span> <span class="kn">import</span> <span class="n">clean</span>
<span class="kn">from</span> <span class="nn">myst_nb</span> <span class="kn">import</span> <span class="n">glue</span>

<span class="n">uberon_obo_file</span> <span class="o">=</span> <span class="s1">&#39;data/uberonext_obo.txt&#39;</span>
<span class="n">uberon_obo</span> <span class="o">=</span> <span class="n">obo</span><span class="o">.</span><span class="n">Obo</span><span class="p">(</span><span class="n">uberon_obo_file</span><span class="p">,[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">,</span><span class="s1">&#39;CL&#39;</span><span class="p">])</span>

<span class="c1">#HPA E-MTAB-2836</span>
<span class="n">col_data_hpa_file</span> <span class="o">=</span> <span class="s1">&#39;data/experiments/hpa/E-MTAB-2836_colData.tsv&#39;</span>
<span class="n">col_data_hpa</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">col_data_hpa_file</span><span class="p">,</span><span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">quotechar</span> <span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>
<span class="n">col_data_hpa</span><span class="p">[</span><span class="n">col_data_hpa</span><span class="o">==</span><span class="s1">&#39;  &#39;</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>  <span class="c1"># make empty elements None</span>

<span class="n">tissue_map_hpa</span> <span class="o">=</span> <span class="n">uberon_obo</span><span class="o">.</span><span class="n">map_tissue_name_to_uberon</span><span class="p">(</span><span class="n">col_data_hpa</span><span class="p">,</span> <span class="s1">&#39;organism_part&#39;</span><span class="p">)</span>
<span class="k">assert</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">tissue_map_hpa</span><span class="p">[</span><span class="n">tissue_map_hpa</span><span class="o">.</span><span class="n">UBERON</span><span class="o">.</span><span class="n">isna</span><span class="p">()][</span><span class="s1">&#39;name matched on&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># Make HPA experimental file:</span>
<span class="n">lines</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">index</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">col_data_hpa</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="n">lines</span><span class="o">.</span><span class="n">append</span><span class="p">([</span>
        <span class="n">index</span><span class="p">,</span>
        <span class="s1">&#39;HPA&#39;</span><span class="p">,</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="c1">#HPA doesn&#39;t have individual IDs</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="c1">#HPA doesn&#39;t have individual age</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="c1">#HPA doesn&#39;t have age ranges</span>
        <span class="n">row</span><span class="p">[</span><span class="s1">&#39;developmental_stage&#39;</span><span class="p">],</span>
        <span class="n">clean</span><span class="o">.</span><span class="n">clean_sex</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;sex&#39;</span><span class="p">]),</span>
        <span class="n">clean</span><span class="o">.</span><span class="n">clean_tissue</span><span class="p">(</span><span class="n">tissue_map_hpa</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">index</span><span class="p">,</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]),</span>
        <span class="s1">&#39;tissues - donor&#39;</span><span class="p">,</span>
        <span class="n">clean</span><span class="o">.</span><span class="n">clean_techrep</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;technical_replicate_group&#39;</span><span class="p">],</span><span class="s1">&#39;HPA&#39;</span><span class="p">),</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="c1">#HPA doesn&#39;t have Bio Reps or individual, so can&#39;t even try to calc</span>
    <span class="p">])</span>
<span class="n">hpa_experimental_design</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">lines</span><span class="p">,</span><span class="n">columns</span> <span class="o">=</span> <span class="n">experimental_design_columns</span><span class="p">)</span>
<span class="n">hpa_experimental_design</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">hpa_experimental_design</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/hpa/hpa_experimental_design.csv&#39;</span><span class="p">)</span>

<span class="c1"># GTEx</span>
<span class="n">col_data_gtex_file</span> <span class="o">=</span> <span class="s1">&#39;data/experiments/gtex/E-MTAB-5214_colData.tsv&#39;</span>
<span class="n">col_data_gtex</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">col_data_gtex_file</span><span class="p">,</span><span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="n">quotechar</span> <span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>
<span class="n">col_data_gtex</span> <span class="o">=</span> <span class="n">col_data_gtex</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span> <span class="o">=</span> <span class="p">{</span><span class="s1">&#39;Unnamed: 0&#39;</span><span class="p">:</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">})</span>
<span class="n">col_data_gtex</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">col_data_gtex</span><span class="p">[</span><span class="n">col_data_gtex</span><span class="o">==</span><span class="s1">&#39;  &#39;</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>  <span class="c1">#make empty elements None</span>
<span class="k">assert</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">col_data_gtex</span><span class="o">.</span><span class="n">disease</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">==</span> <span class="nb">set</span><span class="p">([</span><span class="s1">&#39;normal&#39;</span><span class="p">]))</span>
<span class="k">assert</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">col_data_gtex</span><span class="o">.</span><span class="n">clinical_information</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">==</span> <span class="nb">set</span><span class="p">([</span><span class="kc">None</span><span class="p">,</span><span class="s1">&#39;not sun exposed&#39;</span><span class="p">,</span> <span class="s1">&#39;sun exposed&#39;</span><span class="p">]))</span>

<span class="n">tissue_map_gtex</span> <span class="o">=</span> <span class="n">uberon_obo</span><span class="o">.</span><span class="n">map_tissue_name_to_uberon</span><span class="p">(</span><span class="n">col_data_gtex</span><span class="p">,</span><span class="s1">&#39;organism_part&#39;</span><span class="p">)</span>
<span class="c1">#Manually map the GTEx names that don&#39;t automatically map</span>
<span class="n">manual_mapping_gtex</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s1">&#39;transformed skin fibroblast&#39;</span><span class="p">:</span> <span class="kc">None</span><span class="p">,</span> <span class="c1">#Alternative: ? &#39;UBERON:0002097&#39;,#skin of body</span>
    <span class="s1">&#39;ebv-transformed lymphocyte&#39;</span><span class="p">:</span> <span class="kc">None</span><span class="p">,</span>  <span class="c1">#Alternative: ? &#39;UBERON:0000178&#39;, #blood</span>
    <span class="s1">&#39;suprapubic skin&#39;</span><span class="p">:</span> <span class="s1">&#39;UBERON:0001415&#39;</span><span class="p">,</span> <span class="c1">#Alternative: ? &#39;UBERON:0002097&#39;,#skin of body</span>
<span class="p">}</span>

<span class="k">for</span> <span class="n">index</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">tissue_map_gtex</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="kc">None</span><span class="p">:</span>
        <span class="n">uberon_term</span> <span class="o">=</span> <span class="n">manual_mapping_gtex</span><span class="p">[</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;name matched on&#39;</span><span class="p">]]</span>
        <span class="n">tissue_map_gtex</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">index</span><span class="p">,</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">uberon_term</span>
        
<span class="n">num_excluded_samples</span> <span class="o">=</span> <span class="n">tissue_map_gtex</span><span class="p">[</span><span class="n">tissue_map_gtex</span><span class="o">.</span><span class="n">UBERON</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>  <span class="c1"># excluded because transformed cell </span>
<span class="n">glue</span><span class="p">(</span><span class="s2">&quot;Number of excluded GTEx samples&quot;</span><span class="p">,</span> <span class="n">num_excluded_samples</span><span class="p">)</span>
<span class="n">samples_to_exclude</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">tissue_map_gtex</span><span class="p">[</span><span class="n">tissue_map_gtex</span><span class="o">.</span><span class="n">UBERON</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>

<span class="c1">#Use additional file to add age information</span>
<span class="n">gtex_additional_info</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/gtex/GTEx_v7_Annotations_SubjectPhenotypesDS.txt&#39;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">gtex_additional_info</span> <span class="o">=</span> <span class="n">gtex_additional_info</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;SUBJID&#39;</span><span class="p">:</span><span class="s1">&#39;individual&#39;</span><span class="p">,</span><span class="s1">&#39;AGE&#39;</span><span class="p">:</span><span class="s1">&#39;age_range&#39;</span><span class="p">})</span>

<span class="n">col_data_gtex</span> <span class="o">=</span> <span class="n">col_data_gtex</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">gtex_additional_info</span><span class="p">[[</span><span class="s1">&#39;age_range&#39;</span><span class="p">,</span><span class="s1">&#39;individual&#39;</span><span class="p">]],</span><span class="n">on</span><span class="o">=</span><span class="s1">&#39;individual&#39;</span><span class="p">,</span><span class="n">how</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">col_data_gtex</span><span class="o">.</span><span class="n">index</span><span class="p">)</span>

<span class="c1"># Make gtex experimental file:</span>
<span class="n">lines</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">index</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">col_data_gtex</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="n">lines</span><span class="o">.</span><span class="n">append</span><span class="p">([</span>
        <span class="n">index</span><span class="p">,</span>
        <span class="s1">&#39;GTEx&#39;</span><span class="p">,</span>
        <span class="n">row</span><span class="p">[</span><span class="s1">&#39;individual&#39;</span><span class="p">],</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span><span class="c1">#GTEx doesn&#39;t have individual age</span>
        <span class="n">row</span><span class="p">[</span><span class="s1">&#39;age_range&#39;</span><span class="p">],</span>
        <span class="s1">&#39;adult&#39;</span><span class="p">,</span>
        <span class="n">clean_sex</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;sex&#39;</span><span class="p">]),</span>
        <span class="n">clean_tissue</span><span class="p">(</span><span class="n">tissue_map_gtex</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">index</span><span class="p">,</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]),</span>
        <span class="s1">&#39;tissues - donor&#39;</span><span class="p">,</span>
        <span class="n">clean_techrep</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;technical_replicate_group&#39;</span><span class="p">],</span><span class="s1">&#39;GTEx&#39;</span><span class="p">),</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="c1">#TODO: Try to calculate bio reps based on same individual + tissue, but different tech rep?</span>
    <span class="p">])</span>
<span class="n">gtex_experimental_design</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">lines</span><span class="p">,</span><span class="n">columns</span> <span class="o">=</span> <span class="n">experimental_design_columns</span><span class="p">)</span>
<span class="n">gtex_experimental_design</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">gtex_experimental_design</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/gtex/gtex_experimental_design.csv&#39;</span><span class="p">)</span>

<span class="c1"># HDBR</span>
<span class="n">col_data_hdbr_file</span> <span class="o">=</span> <span class="s1">&#39;data/experiments/hdbr/E-MTAB-4840_colData.tsv&#39;</span>
<span class="n">col_data_hdbr</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">col_data_hdbr_file</span><span class="p">,</span> <span class="n">delimiter</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="c1"># tissue mapping:</span>
<span class="n">tissue_map_hdbr</span> <span class="o">=</span> <span class="n">uberon_obo</span><span class="o">.</span><span class="n">map_tissue_name_to_uberon</span><span class="p">(</span><span class="n">col_data_hdbr</span><span class="p">,</span><span class="s1">&#39;organism_part&#39;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">index</span> <span class="ow">in</span> <span class="n">tissue_map_hdbr</span><span class="p">[</span><span class="n">tissue_map_hdbr</span><span class="o">.</span><span class="n">UBERON</span><span class="o">.</span><span class="n">isna</span><span class="p">()][</span><span class="s1">&#39;name matched on&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">index</span><span class="p">:</span>
    <span class="n">tissue_map_hdbr</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">index</span><span class="p">,</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;UBERON:0000955&#39;</span> <span class="c1"># Brain</span>
<span class="k">assert</span><span class="p">(</span><span class="n">tissue_map_hdbr</span><span class="p">[</span><span class="n">tissue_map_hdbr</span><span class="o">.</span><span class="n">UBERON</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">==</span><span class="mi">0</span><span class="p">)</span>

<span class="c1"># clean age and sex metadata:</span>
<span class="n">col_data_hdbr</span><span class="p">[</span><span class="s1">&#39;age&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">col_data_hdbr</span><span class="p">[</span><span class="s1">&#39;developmental_stage&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">clean</span><span class="o">.</span><span class="n">MapHDBR</span><span class="o">.</span><span class="n">age</span><span class="p">)</span>
<span class="n">col_data_hdbr</span><span class="p">[</span><span class="s1">&#39;sex&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">col_data_hdbr</span><span class="p">[</span><span class="s1">&#39;karyotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">clean</span><span class="o">.</span><span class="n">MapHDBR</span><span class="o">.</span><span class="n">sex</span><span class="p">)</span>

<span class="c1"># Make experimental file:</span>
<span class="n">lines</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">index</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">col_data_hdbr</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="n">lines</span><span class="o">.</span><span class="n">append</span><span class="p">([</span>
        <span class="n">index</span><span class="p">,</span>
        <span class="s1">&#39;HDBR&#39;</span><span class="p">,</span>
        <span class="s1">&#39;HDBR-&#39;</span><span class="o">+</span><span class="nb">str</span><span class="p">(</span><span class="n">row</span><span class="p">[</span><span class="s1">&#39;individual&#39;</span><span class="p">]),</span>
        <span class="n">row</span><span class="p">[</span><span class="s1">&#39;age&#39;</span><span class="p">],</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span><span class="c1">#HDBR doesn&#39;t have age ranges</span>
        <span class="s1">&#39;fetus&#39;</span><span class="p">,</span>
        <span class="n">row</span><span class="p">[</span><span class="s1">&#39;sex&#39;</span><span class="p">],</span>
        <span class="n">clean_tissue</span><span class="p">(</span><span class="n">tissue_map_hdbr</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">index</span><span class="p">,</span><span class="s1">&#39;UBERON&#39;</span><span class="p">]),</span>
        <span class="s1">&#39;tissues - donor&#39;</span><span class="p">,</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="c1"># TODO:? Calculate based on &quot;block&quot;</span>
        <span class="n">np</span><span class="o">.</span><span class="n">nan</span><span class="p">,</span> <span class="c1"># TODO:? Define from individual and tissue IDs different &quot;block&quot;</span>
    <span class="p">])</span>
<span class="n">hdbr_experimental_design</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">lines</span><span class="p">,</span><span class="n">columns</span> <span class="o">=</span> <span class="n">experimental_design_columns</span><span class="p">)</span>
<span class="n">hdbr_experimental_design</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">,</span><span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">hdbr_experimental_design</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;data/experiments/hdbr/hdbr_experimental_design.csv&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="dropdown admonition">
<p class="admonition-title">Code: combine experimental design</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">myst_nb</span> <span class="kn">import</span> <span class="n">glue</span>

<span class="c1"># read in experimental design files:</span>
<span class="n">hdbr_experimental_design</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span>
    <span class="s1">&#39;data/experiments/hdbr/hdbr_experimental_design.csv&#39;</span><span class="p">,</span> 
    <span class="n">index_col</span><span class="o">=</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">)</span>
<span class="n">fantom_experimental_design</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span>
    <span class="s1">&#39;data/experiments/fantom/fantom_experimental_design.csv&#39;</span><span class="p">,</span> 
    <span class="n">index_col</span><span class="o">=</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">)</span>
<span class="n">hpa_experimental_design</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span>
    <span class="s1">&#39;data/experiments/hpa/hpa_experimental_design.csv&#39;</span><span class="p">,</span> 
    <span class="n">index_col</span><span class="o">=</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">)</span>
<span class="n">gtex_experimental_design</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span>
    <span class="s1">&#39;data/experiments/gtex/gtex_experimental_design.csv&#39;</span><span class="p">,</span> 
    <span class="n">index_col</span><span class="o">=</span><span class="s1">&#39;Sample ID&#39;</span><span class="p">)</span>

<span class="c1"># combine:</span>
<span class="n">combined_design</span> <span class="o">=</span>  <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">hdbr_experimental_design</span><span class="p">,</span> <span class="n">fantom_experimental_design</span><span class="p">,</span> <span class="n">hpa_experimental_design</span><span class="p">,</span> <span class="n">gtex_experimental_design</span><span class="p">])</span>

<span class="c1"># drop non-tissue-specific:</span>
<span class="n">combined_design</span><span class="o">.</span><span class="n">dropna</span><span class="p">(</span><span class="n">subset</span> <span class="o">=</span><span class="p">[</span><span class="s1">&#39;Tissue (UBERON)&#39;</span><span class="p">],</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="k">assert</span><span class="p">(</span><span class="n">combined_design</span><span class="p">[</span><span class="n">combined_design</span><span class="p">[</span><span class="s1">&#39;Tissue (UBERON)&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">()]</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="mi">0</span><span class="p">)</span>

<span class="c1"># save</span>
<span class="n">combined_design</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;data/combined/combined_experimental_design.csv&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>
```{code-cell} ipython3
:tags: [hide-input]

# # load combined_design + add tissue groups info and save/print.

# # glue
# glue(&#39;number of samples in combined data set&#39;, combined_design.shape[0])
# glue(&#39;number of Uberon tissue types in combined data set&#39;, combined_design[&#39;Tissue (UBERON)&#39;].unique().shape[0])
# glue(&#39;number of individuals in combined data set&#39;, combined_design[&#39;Individual ID&#39;].unique().shape[0]-1)

# print(&#39;sex breakdown (of samples):\n&#39;,combined_design[&#39;Sex&#39;].value_counts())
# print(&#39;sample type breakdown:\n&#39;,combined_design[&#39;Sample Type&#39;].value_counts())
# print(&#39;experiment breakdown:\n&#39;,combined_design[&#39;Experiment&#39;].value_counts())
```

### Tissue groups
While the data is in general mapped to the most specific Uberon terms possible, eleven broader tissue groups (for example brain, digestive system, connective tissue) were identified by hand.
Individual tissues were mapped to these groups using `uberon-py`&#39;s `Relations()` function - the code doing this is shown in the {ref}`the second example of uberon-py usage&lt;tissue-group-mapping&gt;` in the next section.

```{code-cell} ipython3
# Code that adds tissue groups to the combined-design file
```
</pre></div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c05-combining"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="2-data.html" title="previous page"><span class="section-number">5.2. </span>Data Acquisition</a>
    <a class='right-next' id="next-link" href="4-uberon.html" title="next page"><span class="section-number">5.4. </span>The <code class="docutils literal notranslate"><span class="pre">uberon-py</span></code> package</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>