
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>4.4. Considerations for Clustering SNPs &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="4.5. Results" href="5-results.html" />
    <link rel="prev" title="4.3. Snowflake input data" href="3-data.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-combining/0-index.html">
   5. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/1-background.html">
     5.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/2-data.html">
     5.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/3-data-wrangling.html">
     5.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html">
     5.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/5-batch-correction.html">
     5.5. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html">
     5.6. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html#combined-data-set">
     5.7. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/7-discussion.html">
     5.8. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/8-future-work.html">
     5.9. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-data.html">
     6.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-methods.html">
     6.3. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/4-results.html">
     6.4. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-discussion.html">
     6.5. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-future-work.html">
     6.6. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-specificity-of-phenotypes">
   4.4.1. The specificity of phenotypes
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#choice-of-clustering-methodology">
   4.4.2. Choice of clustering methodology
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#choice-of-distance-metric">
   4.4.3. Choice of distance metric
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#input-individuals">
   4.4.4. Input individuals
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-curse-of-dimensionality">
   4.4.5. The curse of dimensionality
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#dimensionality-reduction">
     4.4.5.1. Dimensionality reduction
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#sensitivity-of-number-of-snps-on-final-score">
     4.4.5.2. Sensitivity of number of SNPs on final score
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#sensitivity-of-particular-snps-on-final-score">
     4.4.5.3. Sensitivity of particular SNPs on final score
    </a>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="considerations-for-clustering-snps">
<h1><span class="section-number">4.4. </span>Considerations for Clustering SNPs<a class="headerlink" href="#considerations-for-clustering-snps" title="Permalink to this headline">¶</a></h1>
<p>In this section, I discuss the implementation of clustering and outlier detection methodologies for SNPs.</p>
<p>The clustering takes place per phenotype, with the distinct phenotypes and the SNPs associated with them are chosen according to DcGO (see section <code class="xref std std-numref docutils literal notranslate"><span class="pre">dcgo-section</span></code> for further details). This therefore determines how many dimensions (SNPs) we are clustering on for any given phenotype.</p>
<div class="section" id="the-specificity-of-phenotypes">
<h2><span class="section-number">4.4.1. </span>The specificity of phenotypes<a class="headerlink" href="#the-specificity-of-phenotypes" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="choice-of-clustering-methodology">
<h2><span class="section-number">4.4.2. </span>Choice of clustering methodology<a class="headerlink" href="#choice-of-clustering-methodology" title="Permalink to this headline">¶</a></h2>
<p>The original implementation of the phenotype predictor used k-means clustering<a class="bibtex reference internal" href="../end-matter/reference.html#le-cam1967-ku" id="id1">[126]</a>. This wasn’t suitable for the predictor, since we expect combinations of SNPs to form non-spherical shapes (see <a class="reference internal" href="#non-spherical"><span class="std std-numref">Fig. 4.4</span></a>), and k-means cannot achieve this (see <a class="reference internal" href="#clustering-comparison"><span class="std std-numref">Fig. 4.5</span></a>).</p>
<p>Spectral clustering, DBSCAN (Density-Based Spatial Clustering of Applications with Noise), OPTICS and LOF (Local Outlier Factor) were also implemented. This involved automation of parameter selection, to enable clustering to be performed automatically on thousands of phenotype terms. These methods have theoretical pros and cons with respect to the predictor. For example, OPTICS and DBSCAN do not need the number of clusters as an input, but instead require the minimum number of points required to form a cluster and a radius from each point to consider as part of a cluster, which has more meaning in this context. They also automatically output outliers to clusters, which will affect the resulting phenotype score, potentially in unseen ways - particularly as it is difficult to visualise high-dimensional data. OPTICS is the default setting, as in addition to not requiring a number of clusters, it can identify clusters of differing densities (a quality that DBSCAN lacks - as can be seen in the second row of <a class="reference internal" href="#clustering-comparison"><span class="std std-numref">Fig. 4.5</span></a>). A final informed choice between these options requires a large benchmarking set.</p>
</div>
<div class="section" id="choice-of-distance-metric">
<h2><span class="section-number">4.4.3. </span>Choice of distance metric<a class="headerlink" href="#choice-of-distance-metric" title="Permalink to this headline">¶</a></h2>
<p>The phenotype predictor’s original distance metric was non-linear, such that the homozygous calls were further from each other than the distance via the heterozygous call, as shown in <a class="reference internal" href="#non-linear-metric"><span class="std std-numref">Fig. 4.2</span></a>. Non-linear distance metrics mean that it is not possible to create a location matrix rather than a distance matrix. This is required for some types of clustering.</p>
<p>A linear distance metric which also captured the increased likelihood of homozygous alleles to be disease-causing (<a class="reference internal" href="#linear-metric"><span class="std std-numref">Fig. 4.3</span></a>) was developed to enable this, and to better represent the biology. In this version, the popularity of an allele decides which homozygous call the heterozygous call is more functionally similar to.</p>
<div class="figure align-default" id="non-linear-metric">
<a class="reference internal image-reference" href="../_images/nonlinear_metric.png"><img alt="../_images/nonlinear_metric.png" src="../_images/nonlinear_metric.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4.2 </span><span class="caption-text">Original non-linear distance metric. <span class="math notranslate nohighlight">\(MM\)</span> denotes homozygous mutant alleles, <span class="math notranslate nohighlight">\(WW\)</span> denotes homozygous wild type alleles, and <span class="math notranslate nohighlight">\(MW\)</span> denotes heterozygous alleles. The FATHMM score for the SNP <span class="math notranslate nohighlight">\(f\)</span>, defines the distance between the wild type and mutant alleles.</span><a class="headerlink" href="#non-linear-metric" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-default" id="linear-metric">
<a class="reference internal image-reference" href="../_images/linear_metric.png"><img alt="../_images/linear_metric.png" src="../_images/linear_metric.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4.3 </span><span class="caption-text">Linear distance metrics. <span class="math notranslate nohighlight">\(MM\)</span> denotes homozygous mutant alleles, <span class="math notranslate nohighlight">\(WW\)</span> denotes homozygous wild type alleles, and <span class="math notranslate nohighlight">\(MW\)</span> denotes heterozygous alleles, and <span class="math notranslate nohighlight">\(N\)</span> represents the number of people with that allele call. The FATHMM score for the SNP <span class="math notranslate nohighlight">\(f\)</span>, defines the distance between the wild type and mutant alleles.</span><a class="headerlink" href="#linear-metric" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-default" id="non-spherical">
<a class="reference internal image-reference" href="../_images/clustering_snps.png"><img alt="../_images/clustering_snps.png" src="../_images/clustering_snps.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4.4 </span><span class="caption-text">A drawing indicating how the combinations of SNPs we might expect to cause disease would represent a non spherical relationship between SNPs.</span><a class="headerlink" href="#non-spherical" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-default" id="clustering-comparison">
<a class="reference internal image-reference" href="../_images/clustering_comparison.png"><img alt="../_images/clustering_comparison.png" src="../_images/clustering_comparison.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4.5 </span><span class="caption-text">Comparison illustrating differences between the implemented clustering methods. Image adapted from sklearn documentation<a class="bibtex reference internal" href="../end-matter/reference.html#noauthor-undated-nc" id="id2">[127]</a>.</span><a class="headerlink" href="#clustering-comparison" title="Permalink to this image">¶</a></p>
</div>
</div>
<div class="section" id="input-individuals">
<h2><span class="section-number">4.4.4. </span>Input individuals<a class="headerlink" href="#input-individuals" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="the-curse-of-dimensionality">
<h2><span class="section-number">4.4.5. </span>The curse of dimensionality<a class="headerlink" href="#the-curse-of-dimensionality" title="Permalink to this headline">¶</a></h2>
<p>The “curse of dimensionality” is a phrase coined by Richard E Bellman, during his work on dynamic programming<a class="bibtex reference internal" href="../end-matter/reference.html#bellman1957-dc" id="id3">[128]</a>, but has since proved relevant in many different mathematical and data-driven fields. In addition to it’s more colloquial use as a catch-all complaint about high-dimensional data, the “curse” refers to the sparsity of data that occurs exponentially with an increase in dimensions. This leads to various problems in different fields<a class="bibtex reference internal" href="../end-matter/reference.html#zimek2012-fg" id="id4">[129]</a> and of which we run into two in this chapter; general difficulty in reaching statistical significance and reduced usefulness of clustering, distance, and outlier metrics.</p>
<p>This problem can be reduced by choosing a clustering/outlier detection method which is more robust to the number of dimensions. It is also helpful to reduce the number of dimensions in some way (feature selection), or to project the data into a different coordinate system (feature projection) and then discard some of the newly calculated coordinates/dimensions.</p>
<p>A range of clustering methods were trialled on the phenotype predictor in an attempt to overcome the curse of dimensionality and increase the phenotype predictor’s accuracy. Jan mostly implemented the Spectral Clustering, while I implemented the other methods.</p>
<div class="section" id="dimensionality-reduction">
<h3><span class="section-number">4.4.5.1. </span>Dimensionality reduction<a class="headerlink" href="#dimensionality-reduction" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="sensitivity-of-number-of-snps-on-final-score">
<h3><span class="section-number">4.4.5.2. </span>Sensitivity of number of SNPs on final score<a class="headerlink" href="#sensitivity-of-number-of-snps-on-final-score" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="sensitivity-of-particular-snps-on-final-score">
<h3><span class="section-number">4.4.5.3. </span>Sensitivity of particular SNPs on final score<a class="headerlink" href="#sensitivity-of-particular-snps-on-final-score" title="Permalink to this headline">¶</a></h3>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c04-snowflake"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="3-data.html" title="previous page"><span class="section-number">4.3. </span>Snowflake input data</a>
    <a class='right-next' id="next-link" href="5-results.html" title="next page"><span class="section-number">4.5. </span>Results</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>