
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2. Biology Background &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="3. Computational Biology Background" href="../c03-compbio-background/0-index.html" />
    <link rel="prev" title="1. Introduction" href="../c01-introduction/intro.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../jupyter_book_intro.html">
   Start Here
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/title-page.html">
   Title Page
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-figures.html">
   List of Figures
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-tables.html">
   List of Tables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/declaration.html">
   Declaration
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   2. Biology Background
  </a>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-measuring-genotype-phenotype.html">
     3.1. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-comp-bio-methods.html">
     3.2. How large computational biology datasets are used
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-bias.html">
     3.3. Sources of bias in computational biology datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-summary.html">
     3.4. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.2. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.3. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.4. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-combining/0-index.html">
   5. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/1-background.html">
     5.1.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/2-data.html">
     5.1.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/3-data-wrangling.html">
     5.1.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html">
     5.1.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html#functionality">
     5.1.5. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/5-batch-correction.html">
     5.1.6. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html">
     5.1.7. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/7-future-work.html">
     5.1.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-data.html">
     6.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-methods.html">
     6.3. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/4-results.html">
     6.4. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-discussion.html">
     6.5. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-future-work.html">
     6.6. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/c02-biology-background/0-bg-biology.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#big-questions-what-is-genetically-determined-and-how">
   2.1. Big questions: What is genetically determined, and how?
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#history-of-inheritable-traits">
     2.1.1. History of inheritable traits
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#the-future">
     2.1.2. The future
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#biological-molecules-dna-rna-proteins-and-the-central-dogma-of-molecular-biology">
   2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#dna">
     2.2.1. DNA
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#how-dna-affects-us-the-central-dogma-of-molecular-biology">
       2.2.1.1. How DNA affects us: the central dogma of molecular biology
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#rna">
     2.2.2. RNA
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#dna-makes-rna-a-k-a-transcription">
       2.2.2.1. “DNA makes RNA”, a.k.a, transcription
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#proteins">
     2.2.3. Proteins
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#the-protein-folding-problem">
       2.2.3.1. The protein folding problem
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#post-translational-modifications">
       2.2.3.2. Post-translational modifications
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#and-proteins-do-everything">
     2.2.4. “…and proteins do everything.”
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#a-closer-look-at-dna-genomes-genes-and-genetic-variation">
   2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genomes">
     2.3.1. Genomes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genes">
     2.3.2. Genes
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genetic-variation">
     2.3.3. Genetic variation
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#snps-and-snvs">
       2.3.3.1. SNPs and SNVs
      </a>
      <ul class="nav section-nav flex-column">
       <li class="toc-h5 nav-item toc-entry">
        <a class="reference internal nav-link" href="#non-synonymous-snvs">
         2.3.3.1.1. Non-synonymous SNVs
        </a>
       </li>
       <li class="toc-h5 nav-item toc-entry">
        <a class="reference internal nav-link" href="#synonymous-snvs">
         2.3.3.1.2. Synonymous SNVs
        </a>
       </li>
      </ul>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#looking-more-closely-at-proteins-domains-and-disorder">
   2.4. Looking more closely at proteins: domains and disorder
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#primary-secondary-tertiary-and-quaternary">
     2.4.1. Primary, Secondary, Tertiary, and Quaternary
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#quaternary-structures-protein-domains">
       2.4.1.1. Quaternary structures: protein domains
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#disorder">
       2.4.1.2. Disorder
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#associations-between-regions-of-dna-and-traits">
   2.5. Associations between regions of DNA and traits
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#is-there-a-gene-for-that">
     2.5.1. Is there a gene for that?
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#genome-wide-association-studies">
     2.5.2. Genome Wide Association Studies
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#knockouts">
     2.5.3. Knockouts
    </a>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="biology-background">
<h1><span class="section-number">2. </span>Biology Background<a class="headerlink" href="#biology-background" title="Permalink to this headline">¶</a></h1>
<p>I have three aims in this background Chapter:</p>
<ol class="simple">
<li><p>to anchor the work that I’ve done within the context of the big questions for computational biology.</p></li>
<li><p>to provide information about the provenance of data that computational biologists use, in order to aid in discussions about the limitations of the data in later chapters<!--TODO: which ones-->.</p></li>
<li><p>to provide a basic run-down of molecular biology, in order to allow someone without a biology background to understand the rest of this thesis.</p></li>
</ol>
<p>The biological background presented in this section begins at the very basics of molecular biology. This includes details of what biological molecules (e.g. <a class="reference internal" href="#what-is-dna"><span class="std std-ref">DNA</span></a>, <a class="reference internal" href="#what-is-rna"><span class="std std-ref">RNA</span></a> and <a class="reference internal" href="#what-are-proteins"><span class="std std-ref">proteins</span></a>) are, what they are made of, and how current research suggests that they effect the body. These details reveal the complexity of the entities and concepts that computational biologists are interested in. The additional complexities that arise from how we store data about these entities and use it in downstream analyses, is discussed in the <a class="reference internal" href="../c03-compbio-background/0-index.html#c03-compbio-bg"><span class="std std-ref">next Chapter</span></a>.</p>
<div class="section" id="big-questions-what-is-genetically-determined-and-how">
<span id="big-questions"></span><h2><span class="section-number">2.1. </span>Big questions: What is genetically determined, and how?<a class="headerlink" href="#big-questions-what-is-genetically-determined-and-how" title="Permalink to this headline">¶</a></h2>
<p>As humans, we are curious and want to understand ourselves.
We want to know the answers to questions like: “Why are people the way we are?”, “Which aspects of ourselves have we inherited?”, and “What is fixed and what can be changed by the way we live our lives?”</p>
<p>We also seek to improve and control our lives and environment.
This drive for control hasn’t always been a good thing: in recent history, genetics has been used to justify extremely harmful and unethical racist eugenics policies.
While eugenics is thankfully no longer in vogue, there is no guarantee that scientific knowledge will be used ethically.
However, knowing more about ourselves clearly also has the capacity to be used for the good of all.
By understanding how our bodies work, researchers seek to develop new, more effective, and kinder treatments for diseases.</p>
<div class="section" id="history-of-inheritable-traits">
<h3><span class="section-number">2.1.1. </span>History of inheritable traits<a class="headerlink" href="#history-of-inheritable-traits" title="Permalink to this headline">¶</a></h3>
<p>Humanity has been trying to answer the big questions long before we discovered DNA.
The ancient theory of soft inheritance<a class="bibtex reference internal" href="../end-matter/reference.html#zirkle1935-sb" id="id1">[1]</a> said that people can pass on traits they gained during their lives, while 16th-century alchemists theorised that sperm contains tiny fully formed humans<a class="bibtex reference internal" href="../end-matter/reference.html#grafton-undated-af" id="id2">[2]</a> (i.e. women didn’t pass down anything).</p>
<p>Gregor Mendel’s famous pea experiments provided the earliest scientific basis for genetics through his experiments with independently inherited traits of peas (e.g. purple or white flowers, tall or shot plants, wrinkled or round seeds…).
Importantly, he discovered rules of inheritance that indicated that offspring have combinations of discrete genetic material (rather than a blend).
He also showed that one trait that we can measure (e.g. purple flowers) can actually be caused by a number of different genetic combinations.</p>
<p>This concept was expanded by Wilhelm Johannsen, who coined the term “gene” as the name for this genetic material.
Johnannsen’s work also distinguished between “genotype” and “phenotype”: <em>genotype</em> being is the genetic material organisms have, and <em>phenotype</em> being the measurable trait.
His research showed that some phenotypes (e.g. seed size) could vary considerably even with genetically identical plants, due to their environment.</p>
<p>Phenotypes are not only strictly Mendelian, with a fixed number of “types”, or continuous and without a genetic basis.
<a class="reference internal" href="#fischer-racism"><span class="std std-ref">Ronald Fischer</span></a> showed that variation in continuous traits (such as height in humans) can be consistent with Mendelian inheritance if multiple genes contributed additively to the trait.
Many traits are <em>complex</em> in this way, meaning that they are influenced by many different genetic factors, as well as the environment.</p>
<div class="margin sidebar" id="fischer-racism">
<p class="sidebar-title">Ronald Fischer, racism and eugenics</p>
<p>Fischer has a legacy of scientific racism. For example, actively campaigning for the forced sterilisation of a tenth of the population in the name of eugenics.</p>
</div>
<p>When <a class="reference internal" href="#watson-racism"><span class="std std-ref">Watson</span></a> and Crick discovered the now familiar structure of DNA in 1953, we took a huge step towards being able to answer our big questions.
We finally understood the molecular structure that underlies inherent traits.
Then in 2003, with the completion of the Human Genome Project, it was possible to read the human version of this “code of life”.
Once researchers had access to the whole genetic code for a person, they could set about trying to decode it, hoping to discover the key to treating diseases like cancers and Alzheimer’s.</p>
<div class="margin sidebar" id="watson-racism">
<p class="sidebar-title">James Watson and racism</p>
<p>Watson’s has publicly asserted that he believes differences in average measured IQ between blacks and whites are due to genetic differences.</p>
</div>
<p>We now have thousands of human genomes to investigate, and the promises of gene therapies and personalised medicine are now beginning to become reality.
At the time of writing, there are 11 cell and gene therapies approved by the European Medicines Agency<a class="bibtex reference internal" href="../end-matter/reference.html#cynober2020-rx" id="id3">[3]</a>, which treat a variety of cancers, as well as Crohn’s disease, and eye and cartilage problems.
In addition, the first personalised genomic medicine chemotherapy treatment is now available on the NHS, for cancer patients with the allele of the <a class="reference external" href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=DPYD">DYPD</a> gene that cause slower breakdown of chemotherapy toxins<a class="bibtex reference internal" href="../end-matter/reference.html#guardian-staff-reporter2020-lb" id="id4">[4]</a>.</p>
</div>
<div class="section" id="the-future">
<h3><span class="section-number">2.1.2. </span>The future<a class="headerlink" href="#the-future" title="Permalink to this headline">¶</a></h3>
<p>The eventual destination of this field is a full understanding of how our individual genomes and their interaction with the environment affects us.
With this understanding, we would anticipate a much wider application of both personalised medicine and gene therapies.</p>
<p>These therapies are not yet a common occurrence: the 11 approved cell and gene therapies come from a pool of such 500 clinical trials<a class="bibtex reference internal" href="../end-matter/reference.html#cynober2020-rx" id="id5">[3]</a>.
Perhaps it makes sense that we are not finding drug targets quickly, as we still don’t know the functionality of approximately 20% of human genes.
And genes are only a small part (1.2%) of our DNA, and the part we understand best.
Beyond our DNA, there are many other aspects of our cellular and social environments that will effect which parts of our genes are being actively used, and how much.
This section provides an overview of our scientific current model for how DNA effects phenotype, so that we can identify the sources of information that we do have and can make use of.</p>
<p>Despite what we don’t know, this is also the moment where when we have the fullest understanding of our genetics so far.
We have huge, expertly curated, databases containing the great collective knowledge of many experiments about our DNA, how it is being used, and what traits it effects.
Perhaps it is now possible to begin to synthesise some of this collective knowledge into a fuller understanding of complex traits.</p>
</div>
</div>
<div class="section" id="biological-molecules-dna-rna-proteins-and-the-central-dogma-of-molecular-biology">
<h2><span class="section-number">2.2. </span>Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.<a class="headerlink" href="#biological-molecules-dna-rna-proteins-and-the-central-dogma-of-molecular-biology" title="Permalink to this headline">¶</a></h2>
<p>Here I explain the classes of biological molecules that are vital in our understanding of genetics: the nucleic acids, DNA and RNA, and their product - proteins.</p>
<div class="section" id="dna">
<span id="what-is-dna"></span><h3><span class="section-number">2.2.1. </span>DNA<a class="headerlink" href="#dna" title="Permalink to this headline">¶</a></h3>
<div class="figure align-default" id="dna-metal-model">
<a class="reference internal image-reference" href="../_images/dna_metalmodel.jpg"><img alt="../_images/dna_metalmodel.jpg" src="../_images/dna_metalmodel.jpg" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 2.1 </span><span class="caption-text">A photo of the original six-foot tall metal model of DNA made by Watson and Crick in 1953. Image from the Cold Spring Harbor Archives<a class="bibtex reference internal" href="../end-matter/reference.html#noauthor-undated-ih" id="id6">[5]</a>.</span><a class="headerlink" href="#dna-metal-model" title="Permalink to this image">¶</a></p>
</div>
<p>Most people recognise the double helix of deoxyribonucleic acid (DNA) shown in <a class="reference internal" href="#dna-metal-model"><span class="std std-numref">Fig. 2.1</span></a>, it’s a twisted ladder consisting of four nucleotides; adenine, cytosine, thymine, and guanine (A, C, T, G).
It the “code of life” that contains the instructions for making all of the things which make up our bodies and the obvious starting point for understanding how they work.
A given nucleotide on one strand is always linked to its partner on the other strand - A with T, and G with C - which creates redundancy and a convenient copying mechanism.
Lengths of DNA are measured in these base pairs (bp).</p>
<div class="section" id="how-dna-affects-us-the-central-dogma-of-molecular-biology">
<span id="central-dogma"></span><h4><span class="section-number">2.2.1.1. </span>How DNA affects us: the central dogma of molecular biology<a class="headerlink" href="#how-dna-affects-us-the-central-dogma-of-molecular-biology" title="Permalink to this headline">¶</a></h4>
<p>The way in which DNA affects the body can be understood through the <em>central dogma of molecular biology</em>, and in doing so we will also become acquainted with two more important biological molecules: RNA and proteins.</p>
<p>The central dogma of molecular biology can be paraphrased as <strong>“DNA makes RNA makes proteins”</strong>. The idea is that as the “code of life”, DNA contains the instructions for making RNA, which contains the instructions for making proteins, and proteins are the molecules which constitute and make up <a class="reference internal" href="#almost-everything"><span class="std std-ref">almost everything</span></a> in our bodies.</p>
<div class="margin sidebar" id="almost-everything">
<p class="sidebar-title"><em>Almost</em> everything</p>
<p>Proteins make up a lot of molecular machinery in the body as well as structures, however there are other actors, too. There are functional RNA molecules, and fats (which may or may not be synthesised using proteins) can also make up structural parts of the body (for example in cell membranes).</p>
</div>
<p>The central dogma describes the process of gene expression. Gene expression has two parts: transcription (“DNA makes RNA”) and translation (“RNA makes proteins”).</p>
</div>
</div>
<div class="section" id="rna">
<span id="what-is-rna"></span><h3><span class="section-number">2.2.2. </span>RNA<a class="headerlink" href="#rna" title="Permalink to this headline">¶</a></h3>
<p>RNA was originally discovered alongside DNA as a nucleic acid (acidic substances found in the nucleus of cells). It was later discovered that they are also found in bacterial and archeal cells (which don’t have nuclei).</p>
<div class="section" id="dna-makes-rna-a-k-a-transcription">
<h4><span class="section-number">2.2.2.1. </span>“DNA makes RNA”, a.k.a, transcription<a class="headerlink" href="#dna-makes-rna-a-k-a-transcription" title="Permalink to this headline">¶</a></h4>
<p>The process by which “DNA makes RNA” is known as transcription. The process happens to lengths of DNA at a time. These lengths of DNA are called genes, and they may be various sizes, and be transcribed more or less frequently than one another.</p>
<p>The action of transcription is largely carried out by an enzyme called RNA polymerase, which binds to a promoter region of the DNA, close to but outside the gene. This region is only accessible in certain cellular conditions. A rough model is that in conditions in which it is unfavourable for the gene to be transcribed, other molecules will block the promoter.</p>
<p>The RNA polymerase then splits the DNA and adds complementary RNA nucleotides to the DNA, after which the RNA sugar backbone is formed. The RNA-DNA helix is then split, at which point we have what is known as pre-RNA. The pre-RNA is then processed into mature RNA (known as the transcript), ready to be translated into a protein.</p>
<p>The process of what is transcribed and how quickly is affected by many different kinds of proteins (as well as other molecules, e.g. histone modifications). Transcription factors are of particular note, which are proteins that bind to DNA close to or in the promoter region, and either activate the gene (increasing it’s rate of transcription), by for example recruiting RNA polymerase, or repress it (decrease it’s rate of transcription). These transcription factors are in turn regulated by other transcription factors, which creates a network of gene regulation; a gene regulatory network (GRN).</p>
</div>
</div>
<div class="section" id="proteins">
<span id="what-are-proteins"></span><h3><span class="section-number">2.2.3. </span>Proteins<a class="headerlink" href="#proteins" title="Permalink to this headline">¶</a></h3>
<p>Proteins were named by Dutch chemist Gerardus Mulder in his 1839 paper<a class="bibtex reference internal" href="../end-matter/reference.html#mulder1839-rf" id="id7">[6]</a>, where he found that all proteins from animals and plants have more or less the same elemental makeup -  approximately C<sub>400</sub>H<sub>620</sub>N<sub>100</sub>O<sub>120</sub>.
This intriguing result bolstered research in this area, eventually resulting in our current understanding of proteins as biological macromolecules composed of amino acids.</p>
<p>In 1969, Margaret Dayhoff created the first bioinformatics database to store protein structures imaged using X-ray crystallography, related to her publication of Atlas of Protein Sequence and Structure<a class="bibtex reference internal" href="../end-matter/reference.html#hersh1967-ox" id="id8">[7]</a>.
Soon after, in 1972, the Protein DataBank (PDB)<a class="bibtex reference internal" href="../end-matter/reference.html#noauthor-undated-ow" id="id9">[8]</a> was established.
This continues to be well-used and updated, at the time of writing holding structures of 148,827 biological molecules.</p>
<p>The second part of the central dogma is “RNA makes proteins” a.k.a. translation.</p>
<p>Translation is process by which RNA is read to create proteins. First the RNA is converted into a string of amino acids, and then the string of amino acids automatically folds into its’ usual three dimensional structure. The protein folding aspect is not part of translation.</p>
<div class="section" id="the-protein-folding-problem">
<h4><span class="section-number">2.2.3.1. </span>The protein folding problem<a class="headerlink" href="#the-protein-folding-problem" title="Permalink to this headline">¶</a></h4>
<p>As mentioned above, translated strings of amino acids fold automatically (or sometimes with the help of <em>chaperone</em> proteins) into the 3D structure of proteins. It’s the structure of proteins which define their function since their shape effects what they can bind to. Predicting what the structure of a protein will be from the string of amino acids is known as the protein folding problem. It’s has been one of the grandest challenges in biology for over half a century. If it were solved, it would revolutionise protein function and phenotype prediction.</p>
</div>
<div class="section" id="post-translational-modifications">
<h4><span class="section-number">2.2.3.2. </span>Post-translational modifications<a class="headerlink" href="#post-translational-modifications" title="Permalink to this headline">¶</a></h4>
<p>After translation, many proteins can be modified by the addition of various functional groups (parts of molecules which are responsible for chemical properties), e.g. phosphorous or ubiquitin. Such modifications change the functionality of the proteins, and are also believed to aid in protein folding.</p>
</div>
<div class="figure align-default" id="amino-acid-wheel">
<a class="reference internal image-reference" href="../_images/amino_acid_chart.png"><img alt="The figure shows an amino acid chart. It consists of three concentric circles. At the centre is a circle divided into 4 quadrants, labelled for each nucleotide (U, C, A and G). In the second circle, the arcs of the inner quadrants are again divided into 4 (again labelled U, C, A, and G), so that these letters appear 16 times in the middle circle. On the outer circle, each letter from the middle circle is divided into 4 a final time. And these are labelled with the amino acids that they map to, e.g. &quot;Val&quot;, &quot;Arg&quot;, &quot;Leu&quot;" src="../_images/amino_acid_chart.png" style="height: 270px;" /></a>
<p class="caption"><span class="caption-number">Fig. 2.2 </span><span class="caption-text">An amino acid wheel chart (from sigmaaldrich.com<a class="bibtex reference internal" href="../end-matter/reference.html#noauthor-undated-ic" id="id10">[9]</a>) showing the mapping between nucleotide codons and amino acids. The chart is read from the inside out, so for example, UGA is a stop codon, and UUG encodes for leucine (“Leu”).
``` ``</span><a class="headerlink" href="#amino-acid-wheel" title="Permalink to this image">¶</a></p>
<div class="legend">
<p>This process is mostly carried out by a large and complex piece of molecular machinery called the ribosome. The ribosome reads and processes RNA in sets of three nucleotides at a time - these are called codons. Each codon is either a flag to the ribosome (e.g “stop here”, “start here”) or corresponds to an amino acid. Although we would expect <span class="math notranslate nohighlight">\(4^3=64\)</span> permutations of nucleotides, there are only 21 different amino acids which can be incorporated into proteins in humans, so there is redundancy: multiple codons can encode for the same amino acids. We can see this clearly in <a class="reference internal" href="#amino-acid-wheel"><span class="std std-numref">Fig. 2.2</span></a>, for example, UAA, UGA, and UAG are all read as stop codons.</p>
</div>
</div>
</div>
<div class="section" id="and-proteins-do-everything">
<h3><span class="section-number">2.2.4. </span>“…and proteins do everything.”<a class="headerlink" href="#and-proteins-do-everything" title="Permalink to this headline">¶</a></h3>
<p>The unwritten addendum implied by “DNA makes RNA makes proteins” is “…and proteins do (<a class="reference internal" href="#almost-everything"><span class="std std-ref">almost</span></a>) everything”. If DNA encodes for life, then proteins are what make up life. They are the material building blocks of our bodies, and they also have a vast number of other functions: they can be enzymes catalysing reactions, hormones controlling metabolism, transporters for other proteins, signalling proteins, they might be transcription factors (controlling the expression of genes), or have many other functions.</p>
<p>In turn, these processes influence our phenotypes. A phenotype can be something like the level of a certain hormone in the bloodstream, so in a very simple case, a  could effect the phenotype by it’s difference in the speed of expression. A phenotype could alternatively be something like height, which could have a number of genetic (and non-genetic) influences.</p>
</div>
</div>
<div class="section" id="a-closer-look-at-dna-genomes-genes-and-genetic-variation">
<h2><span class="section-number">2.3. </span>A closer look at DNA: Genomes, Genes, and Genetic Variation<a class="headerlink" href="#a-closer-look-at-dna-genomes-genes-and-genetic-variation" title="Permalink to this headline">¶</a></h2>
<p>Now that we have a basic overview of how DNA can influence phenotype, we can discuss it in a little more detail.</p>
<div class="section" id="genomes">
<span id="id11"></span><h3><span class="section-number">2.3.1. </span>Genomes<a class="headerlink" href="#genomes" title="Permalink to this headline">¶</a></h3>
<p>The genome is the full amount of DNA belonging to an organism. We can talk about the genome of an individual, or about the genome of an organism (e.g. the human genome).
When we talk about an organism’s genome, we do so in relation to an example genome for that organism.
This is called a <em>reference</em> genome, and does not belong to any particular individual, but instead is supposed to have the most common nucleotide each location.
We can then discuss individuals genomes in relation to the relevant reference genome.</p>
</div>
<div class="section" id="genes">
<h3><span class="section-number">2.3.2. </span>Genes<a class="headerlink" href="#genes" title="Permalink to this headline">¶</a></h3>
<p>As mentioned, genes are stretches of DNA that encode for RNA. Only a proportion of an organism’s DNA is made up of genes. Regions of DNA that are not within genes are known as <em>non-coding</em> or <em>inter-genic</em>. These regions can still affect phenotype, by influencing the expression of other genes can include enhancers and promoters.</p>
<p>The human genome is 3 billion base pairs long, and contains approximately 20,000 genes (which make up around 2% of the genome).</p>
<p id="introns-exons">Although we define genes as regions of DNA which encode for proteins, not all of the DNA of a gene is transcribed into mature RNA. There are two types of regions inside the gene: introns (regions which cannot be transcribed) and exons (which can be transcribed).
The introns are removed during the processing step in a process known as <em>splicing</em>.
In addition, many genes have the capability to be transcribed into multiple transcripts, via a process known as alternative splicing.
This can happen, for example, by skipping some of the exons during splicing.</p>
<p>Even where DNA can be expressed as proteins, the level of expression depends on the cell environment and other DNA.</p>
<p>Genes are also not a straightforward unit.
Just because a gene is transcribed, does not mean that RNA is always translated into a protein.
Furthermore, there is not a 1:1 mapping of genes:proteins for those that do (one gene could be read to create multiple different proteins).
There can be disagreements about the locations of genes on the genome, and popular databases of genes fundamentally disagree on this and on the number of genes<a class="bibtex reference internal" href="../end-matter/reference.html#salzberg2018-yc" id="id12">[10]</a>.
Gene names and symbols change over time, and can be difficult to map between for this reason.</p>
<p>Genes are also not necessarily a “unit of heritability” - we inherit regions around genes together with them and do not necessarily inherit whole copies of each gene from each parent (although this is rare).</p>
</div>
<div class="section" id="genetic-variation">
<h3><span class="section-number">2.3.3. </span>Genetic variation<a class="headerlink" href="#genetic-variation" title="Permalink to this headline">¶</a></h3>
<p>As you are no doubt aware, individuals of the same species share the vast majority of their DNA.
Genetic variation is the difference in DNA sequence between individuals in a population, and variants are locations on an organisms genome where we observe variation between individuals.
There are different types of genetic variation, for example indels (insertions and deletions)- where small sections of DNA are inserted or deleted from the genome - and copy number variations (CNVs) where larger stretches of the genome are repeated.
However, the most well-studied and common type of genetic variation between humans are single nucleotide polymorphisms (SNPs, pronounced “snips”).</p>
<p>The different forms that a variant can take in the population are called alleles, depending on the type of variant.
Alleles could be different forms of a whole gene, or as small in length as an individual nucleotide.</p>
<div class="section" id="snps-and-snvs">
<h4><span class="section-number">2.3.3.1. </span>SNPs and SNVs<a class="headerlink" href="#snps-and-snvs" title="Permalink to this headline">¶</a></h4>
<p>A SNP is a location on an organism’s genome where there are differences of a single nucleotide (“A”,”C”,”T”,”G”) between individuals. In some fields, these variations are only considered to be Single Nucleotide <em>Polymorphisms</em> if they are relatively commonly occurring in the population (at least 1%), while Single Nucleotide Variants (SNVs) can include both rare and common variants.</p>
<p>Variation at a location does not imply a disease-causing effect, many SNPs are neutral. Much of the time, the aim of studying such variants is to determine which are which. This is often done through looking at their rarity, either in a specific human population (e.g. people with diabetes), the entire human population, or across the tree of life.</p>
<p>SNPs are defined by their location on a human reference genome, for example Chr1: . An individual allele for a given SNP is defined as “wild” type if it matches the reference genome and “mutant” if it does not. The reference genome does not always have the most common allele at each location, although this is it’s aim, so “wild” and “mutant” do not necessarily imply anything about rarity.</p>
<p>If there are only two nucleotide possibilities for a SNP (e.g. A or C), then it is called bi-allelic; the vast majority of SNPs are of this type. Multi-allelic SNPs (e.g. A, T or C) are much rarer.</p>
<p>Since (except for X/Y chromosomes) humans have two copies of each chromosome, an individual will have two alleles for each SNP. These may match (which we call homozygous) or not (heterozygous). Sometimes a disease-causing allele can cause problems even for heterozygotes, while in other cases a person needs two copies of the disease-causing allele in order for it to have an effect.</p>
<p>SNPs can occur either in coding or non-coding regions of the genome.
In non-coding regions, SNPs can still affect gene expression, for example by altering a promoter site.
SNVs in coding regions have two types: synonymous or non-synonymous, based on whether they alter the amino acid sequence.</p>
<div class="section" id="non-synonymous-snvs">
<h5><span class="section-number">2.3.3.1.1. </span>Non-synonymous SNVs<a class="headerlink" href="#non-synonymous-snvs" title="Permalink to this headline">¶</a></h5>
<p>If a SNP alters the amino acid makeup of a protein, it is known as non-synonymous.
Non-synonymous SNVs can cause either nonsense or missense mutations.</p>
<p>Nonsense mutations occur where the SNP substitution results in a stop codon (e.g. TAG) in an unusual position, which signals for a ribosome to stop translating RNA into a protein.
This results in an incomplete and usually nonfunctional protein.
The effect of a nonsense mutation would be more or less severe depending on the location of the new stop codon.
For example, if it was close to the end of the protein, the protein may still be functional.
Sufficiently incomplete proteins are usually destroyed by the cell.</p>
<p>On the other hand, missense mutations occur where the SNP substitution results in an amino acid substitution in the protein.
Some amino acids can be substituted without causing any difference to the function of the protein, while others can severely impede the protein.</p>
</div>
<div class="section" id="synonymous-snvs">
<h5><span class="section-number">2.3.3.1.2. </span>Synonymous SNVs<a class="headerlink" href="#synonymous-snvs" title="Permalink to this headline">¶</a></h5>
<p>Synonymous SNVs occur where substituting the usual nucleotide with another results in the same amino acid.
The resulting protein will have the exact same functionality.
However, synonymous SNVs could still have an effect on high-level traits, since different nucleotides are translated at different speeds.</p>
</div>
</div>
</div>
</div>
<div class="section" id="looking-more-closely-at-proteins-domains-and-disorder">
<h2><span class="section-number">2.4. </span>Looking more closely at proteins: domains and disorder<a class="headerlink" href="#looking-more-closely-at-proteins-domains-and-disorder" title="Permalink to this headline">¶</a></h2>
<p>Just as DNA has been classified at different levels (SNP, gene, genome), so too have proteins.</p>
<div class="section" id="primary-secondary-tertiary-and-quaternary">
<h3><span class="section-number">2.4.1. </span>Primary, Secondary, Tertiary, and Quaternary<a class="headerlink" href="#primary-secondary-tertiary-and-quaternary" title="Permalink to this headline">¶</a></h3>
<p>Proteins are described and classified in terms of the their primary, secondary, tertiary, and quaternary structure.</p>
<p>The primary structure is simply the amino acid makeup of the protein, which describes the protein’s chemical makeup, but not it’s three-dimensional structure.
These amino acid strings tend to form into a small number of familiar (secondary) three-dimensional structures, for example beta strands, beta sheets, and alpha helices.</p>
<p>At the next level, the tertiary structure describes folds: combinations of secondary structures, for example a TIM barrels (a torroidal structure made up of alpha helices and beta strands).
At this level, similar structures do not imply an evolutionary or functional similarity.</p>
<div class="section" id="quaternary-structures-protein-domains">
<h4><span class="section-number">2.4.1.1. </span>Quaternary structures: protein domains<a class="headerlink" href="#quaternary-structures-protein-domains" title="Permalink to this headline">¶</a></h4>
<div class="figure align-default" id="lego">
<a class="reference internal image-reference" href="../_images/lego.png"><img alt="The figure shows a red lego block linked to a yellow lego block, which is then linked to another yellow lego block. The links are thin grey lego blocks." src="../_images/lego.png" style="height: 100px;" /></a>
<p class="caption"><span class="caption-number">Fig. 2.3 </span><span class="caption-text">An illustration of the lego analogy for protein domains. Coloured bricks represent protein domains - colour represents a specific protein domain type, while thin grey bricks represent polypeptide linkers which link domains. Image created using mecabricks<a class="bibtex reference internal" href="../end-matter/reference.html#jarraud2019-jj" id="id13">[11]</a>
``` ``</span><a class="headerlink" href="#lego" title="Permalink to this image">¶</a></p>
<div class="legend">
<p>The quaternary structures of proteins - protein domains - have proved particularly interesting for research.
A simple and oft-used metaphor is to think of protein domains as lego building blocks (<a class="reference internal" href="#lego"><span class="std std-numref">Fig. 2.3</span></a>) which can be linked by polypeptide chains to make up a protein.
These polypeptide chains (known as linkers) are often inflexible, in order to allow only one conformation of the protein.
Small and simple proteins often consist of just one domain, while bigger proteins can contain many domains.
An individual domain can be found in many different proteins, and multiple times in the same protein.</p>
<p>Protein domains are interesting because they are highly conserved in evolution, and are thought of as units of function, evolution, and/or structure.
The functions of proteins, at both low-level (e.g. “calcium signalling protein”) and high-level (e.g. annotated to “liver disease”), are costly and difficult to discern, so there are many proteins about which little is known.
For this reason, often proteins are classified according to their similarity to proteins about which functions are known, for example those containing the same protein domains.</p>
</div>
</div>
</div>
<div class="section" id="disorder">
<h4><span class="section-number">2.4.1.2. </span>Disorder<a class="headerlink" href="#disorder" title="Permalink to this headline">¶</a></h4>
<p>While protein domains exist in one confirmation, not all linkers are rigid.
Flexible regions of proteins which allow for various conformations are referred to as disordered.
Disordered regions are not only relegated to linkers between domains.
Proteins can be constituted entirely of disordered regions, or may have large disordered regions.</p>
<p>Intrinsically disordered proteins can exist in a number of conformations, rather than one fixed structure.
On some occasions, the disordered regions are known to be functional, while on others, proteins may be non-functional until they bind with another macromolecule which forces them into a fixed conformation.</p>
</div>
</div>
</div>
<div class="section" id="associations-between-regions-of-dna-and-traits">
<h2><span class="section-number">2.5. </span>Associations between regions of DNA and traits<a class="headerlink" href="#associations-between-regions-of-dna-and-traits" title="Permalink to this headline">¶</a></h2>
<div class="section" id="is-there-a-gene-for-that">
<h3><span class="section-number">2.5.1. </span>Is there a gene for that?<a class="headerlink" href="#is-there-a-gene-for-that" title="Permalink to this headline">¶</a></h3>
<p>Since “DNA makes RNA makes proteins” and proteins have functions, it might seem sensible to say that there is a gene “for” a given function.
While this is sometimes true (single gene diseases do exist), most of the time the same gene can make multiple different proteins, the same protein can be involved in multiple different pathways and have multiple functions, and multiple proteins can contribute to one function.</p>
<p>The interaction between DNA, RNA and proteins, and the environment is also important to consider.
Although DNA makes RNA makes proteins and proteins do pretty much everything in our bodies, which proteins are made and how they behave is highly dependent on the environment.
The function of a gene might not be evident in some environments since the protein is never transcribed, or it may behave differently.
Furthermore, there are many traits may be entirely environmental.</p>
</div>
<div class="section" id="genome-wide-association-studies">
<h3><span class="section-number">2.5.2. </span>Genome Wide Association Studies<a class="headerlink" href="#genome-wide-association-studies" title="Permalink to this headline">¶</a></h3>
<p>Genome Wide Association Studies (GWAS) are large observational studies where the genotypes of a cohort with a specific phenotype (e.g. diabetes) are compared to the genotypes of a cohort lacking in that phenotype (i.e. a control group) in order to find genomic loci that are statistically associated with the phenotype.
This has been a popular type of scientific enquiry since the first GWAS study in 2005.
GWAS generally results in lists of SNPs, often in the hundreds, ordered by p-value.
Disentangling which of these SNPs (if any) cause the trait is a tricky, particularly since GWAS specifically interrogates common variants.
The process of identifying causal variants generally involving identifying regions in linkage disequilibrium, and re-sequencing regions of interest in further detail.</p>
<p>The GWAS catalog database<a class="bibtex reference internal" href="../end-matter/reference.html#buniello2019-cv" id="id14">[12]</a><a class="bibtex reference internal" href="../end-matter/reference.html#l-emery2017-rd" id="id15">[13]</a> was founded in 2008, to provide a consistent and accessible location for published SNP-trait associations, which extracts information about experiments from the literature (currently over 70000 associations from over 3000 publications).</p>
</div>
<div class="section" id="knockouts">
<h3><span class="section-number">2.5.3. </span>Knockouts<a class="headerlink" href="#knockouts" title="Permalink to this headline">¶</a></h3>
<p>Insight into gene function can also be gained by “knocking out” a gene, preventing it from being translated into a working protein, for example using CRISPR. Combinations of up to four genes can be knocked out in a single experiment. Knocking out a gene can lead to a difference in phenotype, and differences in gene expression, which can be used to help determine gene regulatory networks. There is a lot of existing data on the phenotypic results of mouse knockouts, since they are often used to create mouse models for diseases. Unfortunately, it is not always well-recorded when knockouts lead to no detectable phenotypic change<a class="bibtex reference internal" href="../end-matter/reference.html#barbaric2007-zm" id="id16">[14]</a>.</p>
<hr class="docutils" />
<p><strong>Page References</strong></p>
<p id="bibtex-bibliography-c02-biology-background/0-bg-biology-0"><dl class="citation">
<dt class="bibtex label" id="zirkle1935-sb"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p>Conway Zirkle. The inheritance of acquired characters and the provisional hypothesis of pangenesis. <em>Am. Nat.</em>, 69(724):417–445, September 1935.</p>
</dd>
<dt class="bibtex label" id="grafton-undated-af"><span class="brackets"><a class="fn-backref" href="#id2">2</a></span></dt>
<dd><p>Anthony Grafton and Nancy G Siraisi. Natural particulars: nature and the disciplines in renaissance europe. In <em>acls humanities e-book</em>. MPublishing, University of Michigan Library, 1999.</p>
</dd>
<dt class="bibtex label" id="cynober2020-rx"><span class="brackets">3</span><span class="fn-backref">(<a href="#id3">1</a>,<a href="#id5">2</a>)</span></dt>
<dd><p>Timothé Cynober. Why are there only 11 cell and gene therapies in europe? <em>Labiotech</em>, September 2020. URL: <a class="reference external" href="https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/">https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/</a>.</p>
</dd>
<dt class="bibtex label" id="guardian-staff-reporter2020-lb"><span class="brackets"><a class="fn-backref" href="#id4">4</a></span></dt>
<dd><p>Guardian staff reporter. Cancer patients in england to be offered chance to avoid toxic side-effects. <em>The Guardian</em>, December 2020. URL: <a class="reference external" href="http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects">http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects</a>.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ih"><span class="brackets"><a class="fn-backref" href="#id6">5</a></span></dt>
<dd><p>Gallery 19: DNA model, 1953 :: DNA learning center. <span><a class="reference external" href="#"></a></span>https://www.dnalc.org/view/16430-Gallery-19-DNA-model-1953.html. Accessed: 2019-6-2.</p>
</dd>
<dt class="bibtex label" id="mulder1839-rf"><span class="brackets"><a class="fn-backref" href="#id7">6</a></span></dt>
<dd><p>G J Mulder. Ueber die zusammensetzung einiger thierischen substanzen. <em>J. Prakt. Chem.</em>, 1839.</p>
</dd>
<dt class="bibtex label" id="hersh1967-ox"><span class="brackets"><a class="fn-backref" href="#id8">7</a></span></dt>
<dd><p>Robert T Hersh. Atlas of protein sequence and structure, 1966. <em>Syst. Biol.</em>, 16(3):262–263, September 1967.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ow"><span class="brackets"><a class="fn-backref" href="#id9">8</a></span></dt>
<dd><p>Protein data bank.</p>
</dd>
<dt class="bibtex label" id="noauthor-undated-ic"><span class="brackets"><a class="fn-backref" href="#id10">9</a></span></dt>
<dd><p>Amino acid codon wheel. <span><a class="reference external" href="#"></a></span>https://www.sigmaaldrich.com/technical-documents/articles/biology/amino-acid-codon-wheel.html. Accessed: 2019-6-2.</p>
</dd>
<dt class="bibtex label" id="salzberg2018-yc"><span class="brackets"><a class="fn-backref" href="#id12">10</a></span></dt>
<dd><p>Steven L Salzberg. Open questions: how many genes do we have? <em>BMC Biol.</em>, 16(1):94, August 2018.</p>
</dd>
<dt class="bibtex label" id="jarraud2019-jj"><span class="brackets"><a class="fn-backref" href="#id13">11</a></span></dt>
<dd><p>Nicolas Jarraud. Mecabricks. <span><a class="reference external" href="#"></a></span>https://mecabricks.com/, 2019.</p>
</dd>
<dt class="bibtex label" id="buniello2019-cv"><span class="brackets"><a class="fn-backref" href="#id14">12</a></span></dt>
<dd><p>Annalisa Buniello, Jacqueline A L MacArthur, Maria Cerezo, Laura W Harris, James Hayhurst, Cinzia Malangone, Aoife McMahon, Joannella Morales, Edward Mountjoy, Elliot Sollis, Daniel Suveges, Olga Vrousgou, Patricia L Whetzel, Ridwan Amode, Jose A Guillen, Harpreet S Riat, Stephen J Trevanion, Peggy Hall, Heather Junkins, Paul Flicek, Tony Burdett, Lucia A Hindorff, Fiona Cunningham, and Helen Parkinson. The NHGRI-EBI GWAS catalog of published genome-wide association studies, targeted arrays and summary statistics 2019. <em>Nucleic Acids Res.</em>, 47(D1):D1005–D1012, January 2019. URL: <a class="reference external" href="http://dx.doi.org/10.1093/nar/gky1120">http://dx.doi.org/10.1093/nar/gky1120</a>.</p>
</dd>
<dt class="bibtex label" id="l-emery2017-rd"><span class="brackets"><a class="fn-backref" href="#id15">13</a></span></dt>
<dd><p>J Macarthur L. Emery. GWAS catalog: exploring SNP-trait associations. <span><a class="reference external" href="#"></a></span>http://europepmc.org/article/CTX/C7914, December 2017. Accessed: 2020-9-3. URL: <a class="reference external" href="http://europepmc.org/article/CTX/C7914">http://europepmc.org/article/CTX/C7914</a>.</p>
</dd>
<dt class="bibtex label" id="barbaric2007-zm"><span class="brackets"><a class="fn-backref" href="#id16">14</a></span></dt>
<dd><p>Ivana Barbaric, Gaynor Miller, and T Neil Dear. Appearances can be deceiving: phenotypes of knockout mice. <em>Brief. Funct. Genomic. Proteomic.</em>, 6(2):91–103, June 2007.</p>
</dd>
</dl>
</p>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c02-biology-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="../c01-introduction/intro.html" title="previous page"><span class="section-number">1. </span>Introduction</a>
    <a class='right-next' id="next-link" href="../c03-compbio-background/0-index.html" title="next page"><span class="section-number">3. </span>Computational Biology Background</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>