
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2.1. Big questions: What is genetically determined, and how? &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology." href="2-biological-molecules.html" />
    <link rel="prev" title="2. Biology Background" href="0-index.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../jupyter_book_intro.html">
   Start Here
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/title-page.html">
   Title Page
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-figures.html">
   List of Figures
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/list-of-tables.html">
   List of Tables
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../front-matter/declaration.html">
   Declaration
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   2. Biology Background
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-more-proteins.html">
     2.4. Looking more closely at proteins: domains and disorder
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-phenotype.html">
     2.5. Linking genotype and phenotype: associations between regions of DNA and traits
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-measuring-genotype-phenotype.html">
     3.1. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-comp-bio-methods.html">
     3.2. How large computational biology datasets are used
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-bias.html">
     3.3. Sources of bias in computational biology datasets
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-summary.html">
     3.4. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-combining/0-index.html">
   5. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/1-background.html">
     5.1.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/2-data.html">
     5.1.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/3-data-wrangling.html">
     5.1.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html">
     5.1.4. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/4-uberon.html#functionality">
     5.1.5. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/5-batch-correction.html">
     5.1.6. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/6-results.html">
     5.1.7. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-combining/7-future-work.html">
     5.1.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-data.html">
     6.2. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-methods.html">
     6.3. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/4-results.html">
     6.4. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-discussion.html">
     6.5. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-future-work.html">
     6.6. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/c02-biology-background/1-big-questions.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#history-of-inheritable-traits">
   2.1.1. History of inheritable traits
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#the-future">
   2.1.2. The future
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="big-questions-what-is-genetically-determined-and-how">
<h1><span class="section-number">2.1. </span>Big questions: What is genetically determined, and how?<a class="headerlink" href="#big-questions-what-is-genetically-determined-and-how" title="Permalink to this headline">¶</a></h1>
<p>As humans, we are curious and want to understand ourselves.
We want to know the answers to questions like: “Why are people the way we are?”, “Which aspects of ourselves have we inherited?”, and “What is fixed and what can be changed by the way we live our lives?”</p>
<p>We also seek to improve and control our lives and environment.
This drive for control hasn’t always been a good thing: in recent history, genetics has been used to justify extremely harmful and unethical racist eugenics policies.
While eugenics is thankfully no longer in vogue, there is no guarantee that scientific knowledge will be used ethically.
However, knowing more about ourselves clearly also has the capacity to be used for the good of all.
By understanding how our bodies work, researchers seek to develop new, more effective, and kinder treatments for diseases.</p>
<div class="section" id="history-of-inheritable-traits">
<h2><span class="section-number">2.1.1. </span>History of inheritable traits<a class="headerlink" href="#history-of-inheritable-traits" title="Permalink to this headline">¶</a></h2>
<p>Humanity has been trying to answer the big questions long before we discovered DNA.
The ancient theory of soft inheritance<a class="bibtex reference internal" href="../end-matter/reference.html#zirkle1935-sb" id="id1">[1]</a> said that people can pass on traits they gained during their lives, while 16th-century alchemists theorised that sperm contains tiny fully formed humans<a class="bibtex reference internal" href="../end-matter/reference.html#grafton-undated-af" id="id2">[2]</a> (i.e. women didn’t pass down anything).</p>
<div class="figure align-default" id="mendel-pea">
<a class="reference internal image-reference" href="../_images/mendel.png"><img alt="../_images/mendel.png" src="../_images/mendel.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 2.1 </span><span class="caption-text">An example of Mendel’s experimental results, for white and purple pea flowers. He began with “pure line” pea plants (which always produced self-identical plants when self-pollinated). Crossing the “pure line” plants resulted in first generation offspring which always had purple flowers.
When self-pollinated, these first generation plants created plants with purple and white flowers in a 3:1 ratio.</span><a class="headerlink" href="#mendel-pea" title="Permalink to this image">¶</a></p>
</div>
<p>Gregor Mendel’s famous pea experiments<a class="bibtex reference internal" href="../end-matter/reference.html#mendel1996-gi" id="id3">[3]</a> provided the earliest scientific basis for genetics through his experiments with independently inherited traits of peas (e.g. purple or white flowers, tall or shot plants, wrinkled or round seeds…).
Importantly, he discovered rules of inheritance that indicated that offspring have combinations of discrete genetic material (rather than a blend), i.e. we don’t see pink flowers when we cross purple and white flowered peas.
He also showed that single traits (e.g. purple flowers) can actually be caused by different underlying genetics (see <a class="reference internal" href="#mendel-pea"><span class="std std-numref">Fig. 2.1</span></a>.</p>
<p>This concept was expanded by Wilhelm Johannsen, who coined the term “gene” as the name for the hidden material that caused the traits.
Johnannsen’s work also distinguished between “genotype” and “phenotype”: <em>genotype</em> being the hidden (genetic) material that organisms have, and <em>phenotype</em> being the measurable trait<a class="bibtex reference internal" href="../end-matter/reference.html#johannsen1911-rh" id="id4">[4]</a>.
His research showed that some phenotypes (e.g. seed size) could vary considerably even with genetically identical plants, due to their environment.</p>
<p>Phenotypes are not only strictly Mendelian, with a fixed number of “types”, or continuous and without a genetic basis.
<a class="reference internal" href="#fischer-racism"><span class="std std-ref">Ronald Fischer</span></a> showed that variation in continuous traits (such as height in humans) can be consistent with Mendelian inheritance if multiple genes contributed additively to the trait.
Many traits are <em>complex</em> in this way, meaning that they are influenced by many different genetic factors, as well as the environment.</p>
<div class="margin sidebar" id="fischer-racism">
<p class="sidebar-title">Ronald Fischer, racism and eugenics</p>
<p>Fischer has a legacy of scientific racism. For example, actively campaigning for the forced sterilisation of a tenth of the population in the name of eugenics <a class="bibtex reference internal" href="../end-matter/reference.html#evans2020-dq" id="id5">[5]</a>.</p>
</div>
<p>When <a class="reference internal" href="#watson-racism"><span class="std std-ref">Watson</span></a> and Crick discovered the now familiar structure of DNA in 1953<a class="bibtex reference internal" href="../end-matter/reference.html#watson1953-np" id="id6">[6]</a>, we took a huge step towards being able to answer our big questions.
We finally understood the molecular structure that underlies inherent traits.
Then in 2003, with the completion of the Human Genome Project<a class="bibtex reference internal" href="../end-matter/reference.html#venter2001-wn" id="id7">[7]</a>, it was possible to read the human version of this “code of life”.
Once researchers had access to the whole genetic code for a person, they could set about trying to decode it, with the world hoping that this landmark would aid the search for treatments for diseases like cancers and Alzheimer’s<a class="bibtex reference internal" href="../end-matter/reference.html#meek2000-ec" id="id8">[8]</a>.</p>
<div class="margin sidebar" id="watson-racism">
<p class="sidebar-title">James Watson and racism</p>
<p>Watson’s has publicly asserted that he believes differences in average measured IQ between blacks and whites are due to genetic differences<a class="bibtex reference internal" href="../end-matter/reference.html#harmon2019-na" id="id9">[9]</a>.</p>
</div>
<p>We now have thousands of human genomes to investigate, and the promises of gene therapies and personalised medicine are now beginning to become reality.
At the time of writing, there are eleven cell and gene therapies approved by the European Medicines Agency<a class="bibtex reference internal" href="../end-matter/reference.html#cynober2020-rx" id="id10">[10]</a>, which treat a variety of cancers, as well as Crohn’s disease, and eye and cartilage problems.
In addition, the first personalised genomic medicine chemotherapy treatment is now available on the NHS, for cancer patients with the allele of the <a class="reference external" href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=DPYD">DYPD</a> gene that cause slower breakdown of chemotherapy toxins<a class="bibtex reference internal" href="../end-matter/reference.html#guardian-staff-reporter2020-lb" id="id11">[11]</a>.</p>
</div>
<div class="section" id="the-future">
<h2><span class="section-number">2.1.2. </span>The future<a class="headerlink" href="#the-future" title="Permalink to this headline">¶</a></h2>
<p>The eventual destination of this field is a full understanding of how our individual genomes and their interaction with the environment affects us.
With this understanding, we would anticipate a much wider application of both personalised medicine and gene therapies.
These therapies are not yet a common occurrence: the eleven approved cell and gene therapies come from a pool of such 500 clinical trials<a class="bibtex reference internal" href="../end-matter/reference.html#cynober2020-rx" id="id12">[10]</a>.</p>
<p>Perhaps it makes sense that we are not finding drug targets quickly, as we still don’t know the functionality of approximately 20% of human genes.
And genes are only a small part (1-2%) of our DNA<a class="bibtex reference internal" href="../end-matter/reference.html#pennisi2007-yv" id="id13">[12]</a>, and the part we understand best.
Beyond our DNA, there are many other aspects of our cellular and social environments that will have an effect on which parts of our genes are being actively used, and how much.
This section provides an overview of our scientific current model for how DNA affects phenotype, so that we can identify the sources of information that we do have and can make use of.</p>
<p>Despite what we don’t know, this is also the moment where when we have the fullest understanding of our genetics so far.
We have huge, expertly curated, databases containing the great collective knowledge of many experiments about our DNA, how it is being used, and what traits it affects.
Perhaps it is now possible to begin to synthesise some of this collective knowledge into a fuller understanding of complex traits.</p>
<hr class="docutils" />
<p><strong>Page References</strong></p>
<p id="bibtex-bibliography-c02-biology-background/1-big-questions-0"><dl class="citation">
<dt class="bibtex label" id="zirkle1935-sb"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p>Conway Zirkle. The inheritance of acquired characters and the provisional hypothesis of pangenesis. <em>Am. Nat.</em>, 69(724):417–445, September 1935.</p>
</dd>
<dt class="bibtex label" id="grafton-undated-af"><span class="brackets"><a class="fn-backref" href="#id2">2</a></span></dt>
<dd><p>Anthony Grafton and Nancy G Siraisi. Natural particulars: nature and the disciplines in renaissance europe. In <em>acls humanities e-book</em>. MPublishing, University of Michigan Library, 1999.</p>
</dd>
<dt class="bibtex label" id="mendel1996-gi"><span class="brackets"><a class="fn-backref" href="#id3">3</a></span></dt>
<dd><p>Gregor Mendel. Experiments in plant hybridization (1865). <em>Verhandlungen des naturforschenden Vereins Brünn) Available online</em>, 1996. URL: <a class="reference external" href="http://old.esp.org/foundations/genetics/classical/gm-65-a.pdf">http://old.esp.org/foundations/genetics/classical/gm-65-a.pdf</a>.</p>
</dd>
<dt class="bibtex label" id="johannsen1911-rh"><span class="brackets"><a class="fn-backref" href="#id4">4</a></span></dt>
<dd><p>W Johannsen. The genotype conception of heredity. <em>Am. Nat.</em>, 45(531):129–159, March 1911. URL: <a class="reference external" href="https://doi.org/10.1086/279202">https://doi.org/10.1086/279202</a>.</p>
</dd>
<dt class="bibtex label" id="evans2020-dq"><span class="brackets"><a class="fn-backref" href="#id5">5</a></span></dt>
<dd><p>Richard J Evans. RA fisher and the science of hatred. <em>New Stateman</em>, June 2020. URL: <a class="reference external" href="https://www.newstatesman.com/international/science-tech/2020/07/ra-fisher-and-science-hatred">https://www.newstatesman.com/international/science-tech/2020/07/ra-fisher-and-science-hatred</a>.</p>
</dd>
<dt class="bibtex label" id="watson1953-np"><span class="brackets"><a class="fn-backref" href="#id6">6</a></span></dt>
<dd><p>J D Watson and F H Crick. Molecular structure of nucleic acids; a structure for deoxyribose nucleic acid. <em>Nature</em>, 171(4356):737–738, April 1953. URL: <a class="reference external" href="http://dx.doi.org/10.1038/171737a0">http://dx.doi.org/10.1038/171737a0</a>.</p>
</dd>
<dt class="bibtex label" id="venter2001-wn"><span class="brackets"><a class="fn-backref" href="#id7">7</a></span></dt>
<dd><p>J C Venter, M D Adams, E W Myers, P W Li, R J Mural, G G Sutton, H O Smith, M Yandell, C A Evans, R A Holt, J D Gocayne, P Amanatides, R M Ballew, D H Huson, J R Wortman, Q Zhang, C D Kodira, X H Zheng, L Chen, M Skupski, G Subramanian, P D Thomas, J Zhang, G L Gabor Miklos, C Nelson, S Broder, A G Clark, J Nadeau, V A McKusick, N Zinder, A J Levine, R J Roberts, M Simon, C Slayman, M Hunkapiller, R Bolanos, A Delcher, I Dew, D Fasulo, M Flanigan, L Florea, A Halpern, S Hannenhalli, S Kravitz, S Levy, C Mobarry, K Reinert, K Remington, J Abu-Threideh, E Beasley, K Biddick, V Bonazzi, R Brandon, M Cargill, I Chandramouliswaran, R Charlab, K Chaturvedi, Z Deng, V Di Francesco, P Dunn, K Eilbeck, C Evangelista, A E Gabrielian, W Gan, W Ge, F Gong, Z Gu, P Guan, T J Heiman, M E Higgins, R R Ji, Z Ke, K A Ketchum, Z Lai, Y Lei, Z Li, J Li, Y Liang, X Lin, F Lu, G V Merkulov, N Milshina, H M Moore, A K Naik, V A Narayan, B Neelam, D Nusskern, D B Rusch, S Salzberg, W Shao, B Shue, J Sun, Z Wang, A Wang, X Wang, J Wang, M Wei, R Wides, C Xiao, C Yan, A Yao, J Ye, M Zhan, W Zhang, H Zhang, Q Zhao, L Zheng, F Zhong, W Zhong, S Zhu, S Zhao, D Gilbert, S Baumhueter, G Spier, C Carter, A Cravchik, T Woodage, F Ali, H An, A Awe, D Baldwin, H Baden, M Barnstead, I Barrow, K Beeson, D Busam, A Carver, A Center, M L Cheng, L Curry, S Danaher, L Davenport, R Desilets, S Dietz, K Dodson, L Doup, S Ferriera, N Garg, A Gluecksmann, B Hart, J Haynes, C Haynes, C Heiner, S Hladun, D Hostin, J Houck, T Howland, C Ibegwam, J Johnson, F Kalush, L Kline, S Koduru, A Love, F Mann, D May, S McCawley, T McIntosh, I McMullen, M Moy, L Moy, B Murphy, K Nelson, C Pfannkoch, E Pratts, V Puri, H Qureshi, M Reardon, R Rodriguez, Y H Rogers, D Romblad, B Ruhfel, R Scott, C Sitter, M Smallwood, E Stewart, R Strong, E Suh, R Thomas, N N Tint, S Tse, C Vech, G Wang, J Wetter, S Williams, M Williams, S Windsor, E Winn-Deen, K Wolfe, J Zaveri, K Zaveri, J F Abril, R Guigó, M J Campbell, K V Sjolander, B Karlak, A Kejariwal, H Mi, B Lazareva, T Hatton, A Narechania, K Diemer, A Muruganujan, N Guo, S Sato, V Bafna, S Istrail, R Lippert, R Schwartz, B Walenz, S Yooseph, D Allen, A Basu, J Baxendale, L Blick, M Caminha, J Carnes-Stine, P Caulk, Y H Chiang, M Coyne, C Dahlke, A Mays, M Dombroski, M Donnelly, D Ely, S Esparham, C Fosler, H Gire, S Glanowski, K Glasser, A Glodek, M Gorokhov, K Graham, B Gropman, M Harris, J Heil, S Henderson, J Hoover, D Jennings, C Jordan, J Jordan, J Kasha, L Kagan, C Kraft, A Levitsky, M Lewis, X Liu, J Lopez, D Ma, W Majoros, J McDaniel, S Murphy, M Newman, T Nguyen, N Nguyen, M Nodell, S Pan, J Peck, M Peterson, W Rowe, R Sanders, J Scott, M Simpson, T Smith, A Sprague, T Stockwell, R Turner, E Venter, M Wang, M Wen, D Wu, M Wu, A Xia, A Zandieh, and X Zhu. The sequence of the human genome. <em>Science</em>, 291(5507):1304–1351, February 2001.</p>
</dd>
<dt class="bibtex label" id="meek2000-ec"><span class="brackets"><a class="fn-backref" href="#id8">8</a></span></dt>
<dd><p>James Meek. Decoding DNA. <em>The Guardian</em>, June 2000. URL: <a class="reference external" href="http://www.theguardian.com/science/2000/jun/27/genetics.uknews">http://www.theguardian.com/science/2000/jun/27/genetics.uknews</a>.</p>
</dd>
<dt class="bibtex label" id="harmon2019-na"><span class="brackets"><a class="fn-backref" href="#id9">9</a></span></dt>
<dd><p>Amy Harmon. James watson had a chance to salvage his reputation on race. he made things worse. <em>The New York Times</em>, January 2019. URL: <a class="reference external" href="https://www.nytimes.com/2019/01/01/science/watson-dna-genetics-race.html">https://www.nytimes.com/2019/01/01/science/watson-dna-genetics-race.html</a>.</p>
</dd>
<dt class="bibtex label" id="cynober2020-rx"><span class="brackets">10</span><span class="fn-backref">(<a href="#id10">1</a>,<a href="#id12">2</a>)</span></dt>
<dd><p>Timothé Cynober. Why are there only 11 cell and gene therapies in europe? <em>Labiotech</em>, September 2020. URL: <a class="reference external" href="https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/">https://www.labiotech.eu/in-depth/atmp-cell-gene-therapy-ema/</a>.</p>
</dd>
<dt class="bibtex label" id="guardian-staff-reporter2020-lb"><span class="brackets"><a class="fn-backref" href="#id11">11</a></span></dt>
<dd><p>Guardian staff reporter. Cancer patients in england to be offered chance to avoid toxic side-effects. <em>The Guardian</em>, December 2020. URL: <a class="reference external" href="http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects">http://www.theguardian.com/society/2020/dec/28/cancer-patients-in-england-to-be-offered-chance-to-avoid-toxic-side-effects</a>.</p>
</dd>
<dt class="bibtex label" id="pennisi2007-yv"><span class="brackets"><a class="fn-backref" href="#id13">12</a></span></dt>
<dd><p>Elizabeth Pennisi. Genomics. DNA study forces rethink of what it means to be a gene. <em>Science</em>, 316(5831):1556–1557, June 2007. URL: <a class="reference external" href="http://dx.doi.org/10.1126/science.316.5831.1556">http://dx.doi.org/10.1126/science.316.5831.1556</a>.</p>
</dd>
</dl>
</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c02-biology-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="0-index.html" title="previous page"><span class="section-number">2. </span>Biology Background</a>
    <a class='right-next' id="next-link" href="2-biological-molecules.html" title="next page"><span class="section-number">2.2. </span>Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>