
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2.5. Phenotype &#8212; Phenotype from Genotype</title>
    
  <link rel="stylesheet" href="../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/language_data.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="2.6. Summary: how genotype and phenotype are linked" href="6-summary.html" />
    <link rel="prev" title="2.4. Looking more closely at proteins: function, structure and classification" href="4-more-proteins.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../index.html">
  
  <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <p class="caption collapsible-parent">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c0-front-matter/title-page.html">
   Phenotype and Function from Genotype: Combining Data Sources to Create Explanatory Predictions
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/acknowledgements.html">
     Acknowledgements
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/abstract.html">
     Abstract
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c0-front-matter/declaration.html">
     Declaration
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Main book
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Introduction
  </a>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="0-index.html">
   2. Biology Background
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. Computational Biology Background
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-bias.html">
     3.4. Bias
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-pqi.html">
     3.5. PQI
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-summary.html">
     3.6. Summary
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with
   <code class="docutils literal notranslate">
    <span class="pre">
     Snowflake
    </span>
   </code>
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-data.html">
     4.3. Snowflake input data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-clustering-snps.html">
     4.4. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-results.html">
     4.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/6-discussion.html">
     4.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-future-work.html">
     4.7. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c05-filter/0-index.html">
   5. Filtering computational predictions with tissue-specific expression information
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/2-uberon.html">
     5.2. The
     <code class="docutils literal notranslate">
      <span class="pre">
       uberon-py
      </span>
     </code>
     package
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/4-data.html">
     5.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/5-methods.html">
     5.4. Methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/6-results.html">
     5.5. Results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/7-discussion.html">
     5.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-filter/8-future-work.html">
     5.7. Future work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../c06-combining/0-index.html">
   6. Combining RNA-seq datasets
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/1-background.html">
     6.1. Background
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/2-data.html">
     6.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/3-data-wrangling.html">
     6.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/5-batch-correction.html">
     6.4. Batch correction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html">
     6.5. Results and outputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/6-results.html#combined-data-set">
     6.6. Combined data set
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/7-discussion.html">
     6.7. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-combining/8-future-work.html">
     6.8. Future Work
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c07-conclusion/0-conclusion.html">
   7. Conclusion
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/glossary.html">
   Glossary
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
        <!-- Source interaction buttons -->


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#what-is-phenotype">
   2.5.1. What is phenotype?
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#how-do-proteins-influence-phenotype">
   2.5.2. How do proteins influence phenotype?
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#limits">
     2.5.2.1. Limits
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#ethical-considerations">
     2.5.2.2. Ethical considerations
    </a>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="phenotype">
<h1><span class="section-number">2.5. </span>Phenotype<a class="headerlink" href="#phenotype" title="Permalink to this headline">¶</a></h1>
<div class="section" id="what-is-phenotype">
<span id="id1"></span><h2><span class="section-number">2.5.1. </span>What is phenotype?<a class="headerlink" href="#what-is-phenotype" title="Permalink to this headline">¶</a></h2>
<p>Phenotypes are observable traits, which can range from neutral (like height, skin colour, or eye colour) to disabling (e.g. chronic fatigue syndrome) or life-threatening (e.g. cancers), to very specific measurements (e.g. level of calcium in blood).
Since phenotypes can have various levels of specificity, they can also be hierarchical, an individual could display “abnormal muscle morphology”, or more specifically “facial muscle atrophy”, which means we have to decide at what level to record phenotypes.
Human phenotype information is private information, and some phenotypes are not easy to measure, so information about human phenotypes is not always easy to access.</p>
</div>
<div class="section" id="how-do-proteins-influence-phenotype">
<h2><span class="section-number">2.5.2. </span>How do proteins influence phenotype?<a class="headerlink" href="#how-do-proteins-influence-phenotype" title="Permalink to this headline">¶</a></h2>
<p>The easiest phenotypes to understand genetically are <a class="reference internal" href="1-big-questions.html#mendel"><span class="std std-ref">Mendelian</span></a>.
In Mendelian phenotypes, a single mutation is responsible for a phenotype, and we can assume that the mutation changes, reduces, or stops entirely the functionality of the protein, and that this protein is the main actor involved in the trait.
An example of this in humans is the <a class="reference external" href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=OPN1MW">OPN1MW gene</a> which encodes for green-light absorbing pigment necessary to create green light absorbing cones in the eye: the allele that causes a non-functional OPN1MW gene therefore causes red-green colourblindness.</p>
<p>The way in which Mendelian genetics affect a phenotype can vary.
In humans, for a variant with two alleles, we have three options: homozygous wild type (two copies of common allele), heterozygous (one copy of common allele and one copy of rare allele), and homozygous mutant (two copies of rare allele).
Sometimes having one copy of a rare allele will cause a phenotype, and sometimes it won’t, but having two copies will.</p>
<p>As well as mutations, phenotypes can be caused by chromosomal abnormalities (extra or missing sections of chromosomes).
In this case, the mechanism is the increased or decreased gene expression of the affected section of the chromosome which is influencing phenotypic differences.</p>
<p>Proteins can effect the same phenotype indirectly, through protein-protein interaction networks,  through interaction with the metabolism (the body’s creation of small chemicals, like sugars, fatty acids, and vitamins), and through interaction with the environment of the cell.
The environment of the cell is of course in turn influenced from the human-scale environment: what we eat, whether we smoke, the air we breathe, and our body’s response to outside stimuli.</p>
<div class="section" id="limits">
<h3><span class="section-number">2.5.2.1. </span>Limits<a class="headerlink" href="#limits" title="Permalink to this headline">¶</a></h3>
<p>For many disease phenotypes (e.g. Breast Cancer, Asbestosis), a genetic mutation might predict an increased probability of having the phenotype, given similar environmental conditions.
And there are some many phenotypes which may not be linked to genetic variation at all, but may be entirely influenced by the environment: for example medical conditions that are the result of poisoning.
In these cases, we might imagine that there is a mutation that humans could have that would prevent or reduce the poison reaction, but since no one has it, we can’t study this by looking at human mutations.</p>
<div class="margin sidebar" id="social-constructs">
<p class="sidebar-title">Social Constructs</p>
<p>A social construct is an idea or concept that exists because it has been created and agreed upon by society.</p>
</div>
<p>To get a little philosophical (metaphysical) for just a paragraph, some phenotypes may not even exist (as we usually mean the word).
That is, they might not be <em>natural</em> categories such that there is a straight-forward and physical thing that decides membership to the category<a class="bibtex reference internal" href="../end-matter/reference.html#fried2017-zu" id="id2">[23]</a>.
As an example, consider an imaginary poorly-understood syndrome, it might be diagnosed if you have some of a list of symptoms, but the syndrome might actually four separate diseases with four totally separate causes and the treatments might only work for one of these diseases.
Some phenotypes might be <em><a class="reference internal" href="#social-constructs"><span class="std std-ref">social constructs</span></a></em>; there is a long-running debate among psychologists about whether some mental health conditions and other psychological and behavioural concepts are socially constructed<a class="bibtex reference internal" href="../end-matter/reference.html#hacking1999-gh" id="id3">[24]</a><a class="bibtex reference internal" href="../end-matter/reference.html#szasz1976-ee" id="id4">[25]</a>.
If phenotypes are not based in the physical, then we may have difficulty accurately predicting them from genetics.</p>
</div>
<div class="section" id="ethical-considerations">
<h3><span class="section-number">2.5.2.2. </span>Ethical considerations<a class="headerlink" href="#ethical-considerations" title="Permalink to this headline">¶</a></h3>
<p>Aside from the fact that predicting non-physical concepts is difficult, there are also ethical considerations in trying to predict socially constructed phenotypes.
If we try to predict sexual orientation from genetics <a class="bibtex reference internal" href="../end-matter/reference.html#kaiser2019-tu" id="id5">[26]</a>, then we might turn out to be measuring something else which indirectly influences sexual orientation, for example a protein that influences how open people are to new experiences, or something that in turn influences that.
And in trying to predict intelligence from genetics, for example, we are likely finding associations between variables like how much you have practiced IQ tests or whether you are in the same cultural group as those that created them<a class="bibtex reference internal" href="../end-matter/reference.html#richardson2002-nx" id="id6">[27]</a>, reinforcing racist ideas<a class="bibtex reference internal" href="../end-matter/reference.html#saini-undated-ng" id="id7">[28]</a>.</p>
<p>Even if all phenotypes were natural concepts, predicting the genetic basis of some phenotypes could be harmful<a class="bibtex reference internal" href="../end-matter/reference.html#schuklenk1997-ag" id="id8">[29]</a>, for example finding a gay gene could be motivated by, or lead to a search for “treatments” to “cure” homosexuality even if it did have a physical basis.</p>
<p>Physical measurements can also be problematic for similar reasons.
Take measurements of facial features for example: this brings to mind image of nazis measuring skulls.
Where physical measurements are proxies for measuring the social construct of race, these kinds of phenotypes can be similarly worrying.
Facial recognition technology<a class="bibtex reference internal" href="../end-matter/reference.html#wu2016-ic" id="id9">[30]</a> is often criticised on this basis<a class="bibtex reference internal" href="../end-matter/reference.html#stark2018-lk" id="id10">[31]</a>.</p>
<p>It is for these reasons, that the majority of modern concepts of phenotype are based in medical concepts, where looking for a link between genotype and phenotype can have a potential life-saving or life-improving benefit.
This scenario still comes with serious ethical considerations, however.
Many disabled people do not want cures for their disabilities<a class="bibtex reference internal" href="../end-matter/reference.html#pulrang2019-gj" id="id11">[32]</a>, and people are also worried that the development of genetic screenings for disabilities will effectively result in a genocide of disabled people<a class="bibtex reference internal" href="../end-matter/reference.html#miller2013-ws" id="id12">[33]</a>.</p>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c02-biology-background"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="4-more-proteins.html" title="previous page"><span class="section-number">2.4. </span>Looking more closely at proteins: function, structure and classification</a>
    <a class='right-next' id="next-link" href="6-summary.html" title="next page"><span class="section-number">2.6. </span>Summary: how genotype and phenotype are linked</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>